diff --git a/cmake/LIEFConfig.cmake.in b/cmake/LIEFConfig.cmake.in
index 562fda16..220208c7 100644
--- a/cmake/LIEFConfig.cmake.in
+++ b/cmake/LIEFConfig.cmake.in
@@ -76,7 +76,7 @@ macro(LIEF_load_targets lib_type)
     return()
   endif ()
 
-  if("${lib_type}" STREQUAL "static")
+  if(1)
     # Need to find all dependencies even if they're private when LIEF is
     # compiled statically
     include(CMakeFindDependencyMacro)
@@ -105,6 +105,8 @@ macro(LIEF_load_targets lib_type)
       find_dependency(tl-expected)
     endif()
 
+    find_dependency(fmt)
+    check_required_components(lief)
   endif()
 
   # Include the respective targets file
