diff --git a/include/LIEF/PE/enums.hpp b/include/LIEF/PE/enums.hpp
index 0175eb38..dec357de 100644
--- a/include/LIEF/PE/enums.hpp
+++ b/include/LIEF/PE/enums.hpp
@@ -17,6 +17,7 @@
 #include <inttypes.h>
 #include <cstddef>
 #include "LIEF/enums.hpp"
+#define LIEF_PE_FORCE_UNDEF
 #include "LIEF/PE/undef.h"
 
 namespace LIEF {
diff --git a/src/BinaryStream/VectorStream.cpp b/src/BinaryStream/VectorStream.cpp
index b029d950..c4348cca 100644
--- a/src/BinaryStream/VectorStream.cpp
+++ b/src/BinaryStream/VectorStream.cpp
@@ -13,6 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#include <array>
 #include <iterator>
 #include <vector>
 #include <string>
diff --git a/src/PE/Binary.cpp b/src/PE/Binary.cpp
index 76f5deb6..0cd58cbe 100644
--- a/src/PE/Binary.cpp
+++ b/src/PE/Binary.cpp
@@ -34,6 +34,8 @@
 #include "LIEF/PE/Binary.hpp"
 #include "LIEF/PE/Builder.hpp"
 #include "LIEF/PE/utils.hpp"
+#define LIEF_PE_FORCE_UNDEF
+#include "LIEF/PE/undef.h"
 #include "LIEF/PE/EnumToString.hpp"
 #include "LIEF/PE/ResourceDirectory.hpp"
 #include "LIEF/PE/ResourceData.hpp"
diff --git a/src/PE/Builder.cpp b/src/PE/Builder.cpp
index f6aa79de..b18fde8a 100644
--- a/src/PE/Builder.cpp
+++ b/src/PE/Builder.cpp
@@ -27,6 +27,8 @@
 #include "LIEF/PE/Builder.hpp"
 #include "LIEF/PE/ResourceData.hpp"
 #include "LIEF/PE/utils.hpp"
+#define LIEF_PE_FORCE_UNDEF
+#include "LIEF/PE/undef.h"
 #include "LIEF/PE/ImportEntry.hpp"
 #include "LIEF/PE/Import.hpp"
 #include "LIEF/PE/Section.hpp"
diff --git a/src/PE/utils.cpp b/src/PE/utils.cpp
index 3545fec3..5158ed8f 100644
--- a/src/PE/utils.cpp
+++ b/src/PE/utils.cpp
@@ -29,6 +29,8 @@
 #include "LIEF/exception.hpp"
 
 #include "LIEF/PE/utils.hpp"
+#define LIEF_PE_FORCE_UNDEF
+#include "LIEF/PE/undef.h"
 #include "LIEF/PE/Binary.hpp"
 #include "LIEF/PE/Import.hpp"
 #include "LIEF/PE/ImportEntry.hpp"
