diff --git a/CMakeLists.txt b/CMakeLists.txt
index d02b5d6..6484131 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -244,7 +244,10 @@ macro(xeus_create_target target_name linkage output_name)
         PUBLIC xtl
     )
 
-    target_link_libraries(${target_name} PUBLIC OpenSSL::Crypto)
+	if (MSVC)
+	    target_link_libraries(OpenSSL::Crypto INTERFACE crypt32)
+    endif ()
+	target_link_libraries(${target_name} PUBLIC OpenSSL::Crypto)
 
     if (NOT MSVC)
         if (APPLE)
