diff --git a/CMakeLists.txt b/CMakeLists.txt
index bff867c..cc7d706 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -113,13 +113,15 @@ endif()
 set(export_name "docopt-targets")
 
 # Runtime package
+if(BUILD_SHARED_LIBS)
 install(TARGETS docopt EXPORT ${export_name}
 	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
 	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
 	RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
-
+else()
 # Development package
 install(TARGETS docopt_s EXPORT ${export_name} DESTINATION ${CMAKE_INSTALL_LIBDIR})
+endif()
 install(FILES ${docopt_HEADERS} DESTINATION include/docopt)
 
 # CMake Package
