diff --git a/src/cpp/CMakeLists.txt b/src/cpp/CMakeLists.txt
index 14747c9..81ab383 100644
--- a/src/cpp/CMakeLists.txt
+++ b/src/cpp/CMakeLists.txt
@@ -126,81 +126,65 @@ if(UNIX AND EPROSIMA_INSTALLER)
 endif()
 
 # Install libraries
-if(MSVC OR MSVC_IDE AND EPROSIMA_INSTALLER)
-    if(IS_TOP_LEVEL)
-        install_msvc_libraries(i86Win32VS2013)
-        install_msvc_libraries(x64Win64VS2013)
-        install_msvc_libraries(i86Win32VS2015)
-        install_msvc_libraries(x64Win64VS2015)
-        install_msvc_libraries(i86Win32VS2017)
-        install_msvc_libraries(x64Win64VS2017)
-    endif()
-
-    install(FILES ${PROJECT_SOURCE_DIR}/cmake/packaging/windows/${PROJECT_NAME}-config.cmake
-        DESTINATION ${DATA_INSTALL_DIR}/${PROJECT_NAME}/cmake
-        COMPONENT cmake
-        )
-elseif(NOT EPROSIMA_INSTALLER)
-    install(TARGETS ${PROJECT_NAME}
-        EXPORT ${PROJECT_NAME}-targets
-        RUNTIME DESTINATION ${BIN_INSTALL_DIR}${MSVCARCH_DIR_EXTENSION}
-        LIBRARY DESTINATION ${LIB_INSTALL_DIR}${MSVCARCH_DIR_EXTENSION}
-        ARCHIVE DESTINATION ${LIB_INSTALL_DIR}${MSVCARCH_DIR_EXTENSION}
-        COMPONENT libraries${MSVCARCH_EXTENSION}
-        )
+install(TARGETS ${PROJECT_NAME}
+    EXPORT ${PROJECT_NAME}-targets
+    RUNTIME DESTINATION ${BIN_INSTALL_DIR}${MSVCARCH_DIR_EXTENSION}
+    LIBRARY DESTINATION ${LIB_INSTALL_DIR}${MSVCARCH_DIR_EXTENSION}
+    ARCHIVE DESTINATION ${LIB_INSTALL_DIR}${MSVCARCH_DIR_EXTENSION}
+    COMPONENT libraries${MSVCARCH_EXTENSION}
+    )
 
-    export(TARGETS ${PROJECT_NAME} FILE ${PROJECT_BINARY_DIR}/cmake/config/${PROJECT_NAME}-targets.cmake)
-    install(EXPORT ${PROJECT_NAME}-targets
-        DESTINATION ${DATA_INSTALL_DIR}/${PROJECT_NAME}/cmake${MSVCARCH_DIR_EXTENSION_EXT}
-        COMPONENT cmake
-        )
+export(TARGETS ${PROJECT_NAME} FILE ${PROJECT_BINARY_DIR}/cmake/config/${PROJECT_NAME}-targets.cmake)
+install(EXPORT ${PROJECT_NAME}-targets
+    DESTINATION ${DATA_INSTALL_DIR}/${PROJECT_NAME}/cmake${MSVCARCH_DIR_EXTENSION_EXT}
+    COMPONENT cmake
+    )
 
-    if(MSVC OR MSVC_IDE)
-        if(TARGET_TYPE STREQUAL "SHARED_LIBRARY")
-            # Install pdb files
-            install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/Debug/
-                DESTINATION ${LIB_INSTALL_DIR}${MSVCARCH_DIR_EXTENSION}
-                COMPONENT libraries${MSVCARCH_EXTENSION}
-                CONFIGURATIONS Debug
-                FILES_MATCHING
-                PATTERN "*.pdb"
-                )
-        elseif(NOT (("${MSVC_ARCH}" STREQUAL "i86Win32VS2013") OR ("${MSVC_ARCH}" STREQUAL "x64Win64VS2013")))
-            install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.dir/Debug/${PROJECT_NAME}.pdb
-                DESTINATION ${LIB_INSTALL_DIR}${MSVCARCH_DIR_EXTENSION}
-                COMPONENT libraries${MSVCARCH_EXTENSION}
-                CONFIGURATIONS Debug
-                )
-        endif()
+if(MSVC OR MSVC_IDE)
+    if(TARGET_TYPE STREQUAL "SHARED_LIBRARY")
+        # Install pdb files
+        install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/Debug/
+            DESTINATION ${LIB_INSTALL_DIR}${MSVCARCH_DIR_EXTENSION}
+            COMPONENT libraries${MSVCARCH_EXTENSION}
+            CONFIGURATIONS Debug
+            FILES_MATCHING
+            PATTERN "*.pdb"
+            )
+    elseif(NOT (("${MSVC_ARCH}" STREQUAL "i86Win32VS2013") OR ("${MSVC_ARCH}" STREQUAL "x64Win64VS2013")))
+        install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.dir/Debug/${PROJECT_NAME}.pdb
+            DESTINATION ${LIB_INSTALL_DIR}${MSVCARCH_DIR_EXTENSION}
+            COMPONENT libraries${MSVCARCH_EXTENSION}
+            CONFIGURATIONS Debug
+            )
     endif()
+endif()
 
-    ###############################################################################
-    # Create CMake package config file
-    ###############################################################################
-    include(CMakePackageConfigHelpers)
-    configure_package_config_file(${PROJECT_SOURCE_DIR}/cmake/packaging/Config.cmake.in
-        ${PROJECT_BINARY_DIR}/cmake/config/${PROJECT_NAME}-config.cmake
-        INSTALL_DESTINATION ${DATA_INSTALL_DIR}$/${PROJECT_NAME}/cmake${MSVCARCH_DIR_EXTENSION_EXT}
-        PATH_VARS BIN_INSTALL_DIR INCLUDE_INSTALL_DIR LIB_INSTALL_DIR
-        )
-    write_basic_package_version_file(${PROJECT_BINARY_DIR}/cmake/config/${PROJECT_NAME}-config-version.cmake
-        VERSION ${PROJECT_VERSION}
-        COMPATIBILITY SameMajorVersion
-        )
-    install(FILES ${PROJECT_BINARY_DIR}/cmake/config/${PROJECT_NAME}-config.cmake
-        ${PROJECT_BINARY_DIR}/cmake/config/${PROJECT_NAME}-config-version.cmake
-        DESTINATION ${DATA_INSTALL_DIR}/${PROJECT_NAME}/cmake${MSVCARCH_DIR_EXTENSION_EXT}
-        COMPONENT cmake
-        )
+###############################################################################
+# Create CMake package config file
+###############################################################################
+include(CMakePackageConfigHelpers)
+configure_package_config_file(${PROJECT_SOURCE_DIR}/cmake/packaging/Config.cmake.in
+    ${PROJECT_BINARY_DIR}/cmake/config/${PROJECT_NAME}-config.cmake
+    INSTALL_DESTINATION ${DATA_INSTALL_DIR}$/${PROJECT_NAME}/cmake${MSVCARCH_DIR_EXTENSION_EXT}
+    PATH_VARS BIN_INSTALL_DIR INCLUDE_INSTALL_DIR LIB_INSTALL_DIR
+    )
+write_basic_package_version_file(${PROJECT_BINARY_DIR}/cmake/config/${PROJECT_NAME}-config-version.cmake
+    VERSION ${PROJECT_VERSION}
+    COMPATIBILITY SameMajorVersion
+    )
+install(FILES ${PROJECT_BINARY_DIR}/cmake/config/${PROJECT_NAME}-config.cmake
+    ${PROJECT_BINARY_DIR}/cmake/config/${PROJECT_NAME}-config-version.cmake
+    DESTINATION ${DATA_INSTALL_DIR}/${PROJECT_NAME}/cmake${MSVCARCH_DIR_EXTENSION_EXT}
+    COMPONENT cmake
+    )
 
-    if(MSVCARCH_EXTENSION)
-        string(TOUPPER "${MSVC_ARCH}" MSVC_ARCH_UPPER)
-        set(CPACK_COMPONENT_LIBRARIES_${MSVC_ARCH_UPPER}_DISPLAY_NAME "${MSVC_ARCH}" PARENT_SCOPE)
-        set(CPACK_COMPONENT_LIBRARIES_${MSVC_ARCH_UPPER}_DESCRIPTION "eProsima ${PROJECT_NAME_LARGE} libraries for platform ${MSVC_ARCH}" PARENT_SCOPE)
-        set(CPACK_COMPONENT_LIBRARIES_${MSVC_ARCH_UPPER}_GROUP "Libraries" PARENT_SCOPE)
-        set(CPACK_COMPONENT_GROUP_LIBRARIES_DESCRIPTION "eProsima ${PROJECT_NAME_LARGE} libraries" PARENT_SCOPE)
-    else()
-        set(CPACK_COMPONENT_LIBRARIES_DISPLAY_NAME "Libraries" PARENT_SCOPE)
-        set(CPACK_COMPONENT_LIBRARIES_DESCRIPTION "eProsima ${PROJECT_NAME_LARGE} libraries" PARENT_SCOPE)
-    endif()
-endif()
+if(MSVCARCH_EXTENSION)
+    string(TOUPPER "${MSVC_ARCH}" MSVC_ARCH_UPPER)
+    set(CPACK_COMPONENT_LIBRARIES_${MSVC_ARCH_UPPER}_DISPLAY_NAME "${MSVC_ARCH}" PARENT_SCOPE)
+    set(CPACK_COMPONENT_LIBRARIES_${MSVC_ARCH_UPPER}_DESCRIPTION "eProsima ${PROJECT_NAME_LARGE} libraries for platform ${MSVC_ARCH}" PARENT_SCOPE)
+    set(CPACK_COMPONENT_LIBRARIES_${MSVC_ARCH_UPPER}_GROUP "Libraries" PARENT_SCOPE)
+    set(CPACK_COMPONENT_GROUP_LIBRARIES_DESCRIPTION "eProsima ${PROJECT_NAME_LARGE} libraries" PARENT_SCOPE)
+else()
+    set(CPACK_COMPONENT_LIBRARIES_DISPLAY_NAME "Libraries" PARENT_SCOPE)
+    set(CPACK_COMPONENT_LIBRARIES_DESCRIPTION "eProsima ${PROJECT_NAME_LARGE} libraries" PARENT_SCOPE)
+endif()
\ No newline at end of file
diff --git a/src/cpp/CMakeLists.txt b/src/cpp/CMakeLists.txt
index 81ab383..24c9916 100644
--- a/src/cpp/CMakeLists.txt
+++ b/src/cpp/CMakeLists.txt
@@ -140,25 +140,6 @@ install(EXPORT ${PROJECT_NAME}-targets
     COMPONENT cmake
     )
 
-if(MSVC OR MSVC_IDE)
-    if(TARGET_TYPE STREQUAL "SHARED_LIBRARY")
-        # Install pdb files
-        install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/Debug/
-            DESTINATION ${LIB_INSTALL_DIR}${MSVCARCH_DIR_EXTENSION}
-            COMPONENT libraries${MSVCARCH_EXTENSION}
-            CONFIGURATIONS Debug
-            FILES_MATCHING
-            PATTERN "*.pdb"
-            )
-    elseif(NOT (("${MSVC_ARCH}" STREQUAL "i86Win32VS2013") OR ("${MSVC_ARCH}" STREQUAL "x64Win64VS2013")))
-        install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.dir/Debug/${PROJECT_NAME}.pdb
-            DESTINATION ${LIB_INSTALL_DIR}${MSVCARCH_DIR_EXTENSION}
-            COMPONENT libraries${MSVCARCH_EXTENSION}
-            CONFIGURATIONS Debug
-            )
-    endif()
-endif()
-
 ###############################################################################
 # Create CMake package config file
 ###############################################################################
