diff --git a/cmake/install_macros.cmake b/cmake/install_macros.cmake
index d7472cf5e6e..697bd054434 100644
--- a/cmake/install_macros.cmake
+++ b/cmake/install_macros.cmake
@@ -97,7 +97,7 @@ ENDFUNCTION()
 FUNCTION(MYSQL_INSTALL_TARGET target_arg)
   CMAKE_PARSE_ARGUMENTS(ARG
     ""
-    "DESTINATION;COMPONENT"
+    "DESTINATION;COMPONENT;EXPORT"
     ""
     ${ARGN}
     )
@@ -110,7 +110,20 @@ FUNCTION(MYSQL_INSTALL_TARGET target_arg)
   IF(ARG_COMPONENT)
     SET(COMP COMPONENT ${ARG_COMPONENT})
   ENDIF()
-  INSTALL(TARGETS ${target} DESTINATION ${ARG_DESTINATION} ${COMP})
+  IF (ARG_EXPORT)
+    FILE(WRITE "${CMAKE_CURRENT_BINARY_DIR}/${ARG_EXPORT}-config.cmake"
+"include(CMakeFindDependencyMacro)
+find_dependency(ZLIB)
+find_dependency(OpenSSL)
+include(\"\${CMAKE_CURRENT_LIST_DIR}/${ARG_EXPORT}-targets.cmake\")
+")
+    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${ARG_EXPORT}-config.cmake DESTINATION share/${ARG_EXPORT})
+    set(EXPORT_ARGS EXPORT ${ARG_EXPORT}-targets)
+  ENDIF()
+  INSTALL(TARGETS ${TARGETS} ${EXPORT_ARGS} RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib ${COMP})
+  IF (ARG_EXPORT)
+    INSTALL(${EXPORT_ARGS} DESTINATION share/${ARG_EXPORT})
+  ENDIF()
   SET(INSTALL_LOCATION ${ARG_DESTINATION} )
   INSTALL_DEBUG_SYMBOLS(${target})
   SET(INSTALL_LOCATION)
diff --git a/cmake/libutils.cmake b/cmake/libutils.cmake
index 41b79e22899..18aca0852b7 100644
--- a/cmake/libutils.cmake
+++ b/cmake/libutils.cmake
@@ -285,7 +285,7 @@ MACRO(MERGE_LIBRARIES_SHARED TARGET_ARG)
     ELSE()
       SET(DESTINATION "${INSTALL_LIBDIR}")
     ENDIF()
-    MYSQL_INSTALL_TARGET(${TARGET} DESTINATION "${DESTINATION}" ${COMP})
+    MYSQL_INSTALL_TARGET(${TARGET} EXPORT unofficial-libmysql DESTINATION "${DESTINATION}" ${COMP})
   ENDIF()
 
   IF(WIN32)
@@ -461,7 +461,7 @@ MACRO(MERGE_CONVENIENCE_LIBRARIES TARGET_ARG)
       SET(COMP COMPONENT ${ARG_COMPONENT})
     ENDIF()
     IF(INSTALL_STATIC_LIBRARIES)
-      MYSQL_INSTALL_TARGET(${TARGET} DESTINATION "${INSTALL_LIBDIR}" ${COMP})
+      MYSQL_INSTALL_TARGET(${TARGET} EXPORT unofficial-libmysql DESTINATION "${INSTALL_LIBDIR}" ${COMP})
     ENDIF()
   ENDIF()
 ENDMACRO(MERGE_CONVENIENCE_LIBRARIES)
diff --git a/libmysql/CMakeLists.txt b/libmysql/CMakeLists.txt
index 1bd8bd5481d..d7be5b498a8 100644
--- a/libmysql/CMakeLists.txt
+++ b/libmysql/CMakeLists.txt
@@ -274,6 +274,14 @@ ENDIF()
 # LDAP authentication SASL client plug-in
 ADD_SUBDIRECTORY(authentication_ldap)
 
+IF (BUILD_SHARED_LIBS)
+  set(INSTALL_SHARED )
+  set(INSTALL_STATIC SKIP_INSTALL)
+ELSE()
+  set(INSTALL_SHARED SKIP_INSTALL)
+  set(INSTALL_STATIC )
+ENDIF()
+
 # FIDO authentication client plugin
 ADD_SUBDIRECTORY(authentication_fido)
 
@@ -284,8 +292,7 @@ ADD_SUBDIRECTORY(authentication_kerberos)
 ADD_SUBDIRECTORY(authentication_oci_client)
 
 # Merge several convenience libraries into one big mysqlclient
-MERGE_CONVENIENCE_LIBRARIES(mysqlclient ${LIBS_TO_MERGE}
-  COMPONENT Development
+MERGE_CONVENIENCE_LIBRARIES(mysqlclient ${LIBS_TO_MERGE} ${INSTALL_STATIC} COMPONENT Development
   LINK_LIBRARIES ${LIBS_TO_LINK}
   )
 
@@ -335,6 +342,7 @@ ENDIF()
 # Merge several convenience libraries into one big mysqlclient
 # and link them together into shared library.
 MERGE_LIBRARIES_SHARED(libmysql ${LIBS_TO_MERGE}
+  ${INSTALL_SHARED}
   EXPORTS
   ${CLIENT_API_FUNCTIONS}
   ${CLIENT_API_FUNCTIONS_UNDOCUMENTED}
@@ -400,6 +408,7 @@ CONFIGURE_FILE(api_test.c.in ${CMAKE_CURRENT_BINARY_DIR}/api_test.c)
 # from @CLIENT_API_FUNCTIONS@ are declared by <mysql.h>. It will fail
 # to run if not all of these symbols are exported by the library.
 #
+IF (ENABLE_TESTING)
 MYSQL_ADD_EXECUTABLE(libmysql_api_test
   ${CMAKE_CURRENT_BINARY_DIR}/api_test.c
   LINK_LIBRARIES libmysql ${LIBRT}
@@ -428,3 +437,4 @@ ADD_CUSTOM_COMMAND(TARGET libmysql_api_test POST_BUILD
   COMMAND libmysql_api_test
   > ${CMAKE_CURRENT_BINARY_DIR}/libmysql_api_test.out
   )
+ENDIF()
diff --git a/scripts/CMakeLists.txt b/scripts/CMakeLists.txt
index 25cc7f3ee4a..b22a5dfe945 100644
--- a/scripts/CMakeLists.txt
+++ b/scripts/CMakeLists.txt
@@ -335,12 +335,13 @@ MACRO(EXTRACT_LINK_LIBRARIES target var)
   STRING(REGEX REPLACE "[ ]+$" "" ${var} "${${var}}")
 ENDMACRO()
 
+IF (NOT BUILD_SHARED_LIBS)
 EXTRACT_LINK_LIBRARIES(mysqlclient CLIENT_LIBS)
-
-IF(MSVC)
+ENDIF()
+IF(MSVC AND NOT BUILD_SHARED_LIBS)
   GET_TARGET_PROPERTY(LIBMYSQL_OS_SHLIB_VERSION mysqlclient VERSION)
   GET_TARGET_PROPERTY(LIBMYSQL_OS_OUTPUT_NAME mysqlclient OUTPUT_NAME)
-ELSE()
+ELSEIF(BUILD_SHARED_LIBS)
   GET_TARGET_PROPERTY(LIBMYSQL_OS_SHLIB_VERSION libmysql VERSION)
   GET_TARGET_PROPERTY(LIBMYSQL_OS_OUTPUT_NAME libmysql OUTPUT_NAME)
 ENDIF()
