diff --git a/extra/re2/CMakeLists.txt b/extra/re2/CMakeLists.txt
index ca39db726..965f2c8dc 100644
--- a/extra/re2/CMakeLists.txt
+++ b/extra/re2/CMakeLists.txt
@@ -71,6 +71,7 @@ IF(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
                              HAVE_TAUTOLOGICAL_UNDEFINED_COMPARE)
   IF(HAVE_TAUTOLOGICAL_UNDEFINED_COMPARE)
     # Boost source has unused local typedefs.
+    INCLUDE(${MYSQL_CMAKE_SCRIPT_DIR}/compile_flags.cmake)
     ADD_COMPILE_FLAGS(${RE2_SOURCES}
                       COMPILE_FLAGS "-Wno-tautological-undefined-compare")
   ENDIF()
