diff --git a/cmake/AddEventLibrary.cmake b/cmake/AddEventLibrary.cmake
index 6c792db..6595550 100644
--- a/cmake/AddEventLibrary.cmake
+++ b/cmake/AddEventLibrary.cmake
@@ -77,6 +77,8 @@ macro(add_event_library LIB_NAME)
         list(APPEND ADD_EVENT_LIBRARY_TARGETS "${LIB_NAME}_static")
 
         set(ADD_EVENT_LIBRARY_INTERFACE "${LIB_NAME}_static")
+		
+		target_link_libraries("${LIB_NAME}_static" PRIVATE ${CMAKE_REQUIRED_LIBRARIES})
     endif()
 
     if (${EVENT_LIBRARY_SHARED})
@@ -85,7 +87,8 @@ macro(add_event_library LIB_NAME)
         target_link_libraries("${LIB_NAME}_shared"
             ${CMAKE_THREAD_LIBS_INIT}
             ${LIB_PLATFORM}
-            ${LIB_LIBRARIES})
+            ${LIB_LIBRARIES}
+			${CMAKE_REQUIRED_LIBRARIES})
 
         if (EVENT_SHARED_FLAGS)
             set_event_shared_lib_flags("${LIB_NAME}" "${EVENT_SHARED_FLAGS}")
