diff --git a/include/ignition/msgs/Generator.hh b/include/ignition/msgs/Generator.hh
index 62d77c6..5785e51 100644
--- a/include/ignition/msgs/Generator.hh
+++ b/include/ignition/msgs/Generator.hh
@@ -43,9 +43,9 @@ class Generator : public CodeGenerator
   /// \param[in] _generatorContext Output directory.
   /// \param[in] _error Unused string value
   public: virtual bool Generate(const FileDescriptor *_file,
-              const string &_parameter,
+              const std::string &_parameter,
               OutputDirectory *_generatorContext,
-              string *_error) const;
+              std::string *_error) const;
 
   // private: GOOGLE_DISALLOW_EVIL_CONSTRUCTORS(Generator);
 };
diff --git a/src/Generator.cc b/src/Generator.cc
index f703fb8..922d637 100644
--- a/src/Generator.cc
+++ b/src/Generator.cc
@@ -67,7 +67,7 @@ Generator::~Generator()
 
 /////////////////////////////////////////////////
 bool Generator::Generate(const FileDescriptor *_file,
-                               const string &/*_parameter*/,
+                               const std::string &/*_parameter*/,
                                OutputDirectory *_generatorContext,
                                std::string * /*_error*/) const
 {
