diff --git a/configure.ac b/configure.ac
index ab3da60..1e5fc78 100644
--- a/configure.ac
+++ b/configure.ac
@@ -149,8 +149,8 @@ AC_CHECK_FUNCS([gettimeofday localeconv setlocale getrusage])
 AC_CHECK_FUNCS([dup dup2],,
         [AC_DEFINE([MPC_NO_STREAM_REDIRECTION],1,[Do not check mpc_out_str on stdout])])
 
-AC_CHECK_LIB([gmp], [__gmpz_init],
-             [LIBS="-lgmp $LIBS"],
+PKG_CHECK_MODULES([GMP], [gmp],
+             [LIBS="$GMP_LIBS $LIBS"],
              [AC_MSG_ERROR([libgmp not found or uses a different ABI (including static vs shared).])])
 
 AC_MSG_CHECKING(for MPFR)
