diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index be89cf5..6be4c1a 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -216,7 +216,7 @@ foreach(header ${HEADERS_ROOT})
     install(FILES ${header} DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/")
 endforeach()
 foreach(header ${HEADERS_SUBDIR})
-    install(FILES ${header} DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/oneapi/dnnl/")
+    install(FILES ${header} DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/openvino/oneapi/dnnl/")
 endforeach()
 
 # Write version and package config files
