diff --git a/cmake/developer_package/compile_flags/os_flags.cmake b/cmake/developer_package/compile_flags/os_flags.cmake
index 3ce87023..1f023280 100644
--- a/cmake/developer_package/compile_flags/os_flags.cmake
+++ b/cmake/developer_package/compile_flags/os_flags.cmake
@@ -134,6 +134,11 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
         set(CMAKE_EXE_LINKER_FLAGS_RELEASE "${CMAKE_EXE_LINKER_FLAGS_RELEASE} /DEBUG")
     endif()
 
+    if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+        string(REPLACE "/Z7" "/Zi" CMAKE_C_FLAGS_DEBUG $CMAKE_C_FLAGS_DEBUG)
+        string(REPLACE "/Z7" "/Zi" CMAKE_CXX_FLAGS_DEBUG $CMAKE_CXX_FLAGS_DEBUG)
+    endif()
+
     # need to set extra flags after /DEBUG to ensure that binary size is not bloated
     set(CMAKE_SHARED_LINKER_FLAGS_RELEASE "${CMAKE_SHARED_LINKER_FLAGS_RELEASE} /OPT:REF /OPT:ICF")
     set(CMAKE_MODULE_LINKER_FLAGS_RELEASE "${CMAKE_MODULE_LINKER_FLAGS_RELEASE} /OPT:REF /OPT:ICF")
