diff --git a/CMakeLists.txt b/CMakeLists.txt
index a735c53..4adaaec 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -229,7 +229,7 @@ add_subdirectory(src)
 if(BUILD_UNIT_TESTS)
     add_subdirectory(test)
 endif()
-add_subdirectory(tools)
+#add_subdirectory(tools)
 
 file(COPY ${CMAKE_CURRENT_BINARY_DIR}/braft/
         DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/output/include/braft/
@@ -249,3 +249,9 @@ install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/output/include/
         PATTERN "*.h"
         PATTERN "*.hpp"
         )
+
+install(TARGETS braft-static EXPORT unofficial-braftTargets)
+install(EXPORT unofficial-braftTargets
+    NAMESPACE unofficial::braft::
+    DESTINATION share/unofficial-braft
+)
