diff --git a/src/lib/OpenEXRCore/unpack.c b/src/lib/OpenEXRCore/unpack.c
index 7587eb2..79061b8 100644
--- a/src/lib/OpenEXRCore/unpack.c
+++ b/src/lib/OpenEXRCore/unpack.c
@@ -37,7 +39,7 @@ half_to_float8 (float* out, const uint16_t* src)
 }
 #endif
 
-#if (defined(__x86_64__) || defined(_M_X64)) &&                                \
+#if (defined(__x86_64__) || defined(_M_X64)) && defined(__AVX__) &&                              \
     (defined(__F16C__) || defined(__GNUC__) || defined(__clang__))
 
 #    if defined(__F16C__)
