diff --git a/src/core/lib/transport/transport.cc b/src/core/lib/transport/transport.cc
index 6adcd8d..20745df 100644
--- a/src/core/lib/transport/transport.cc
+++ b/src/core/lib/transport/transport.cc
@@ -97,6 +97,8 @@ void grpc_stream_ref_init(grpc_stream_refcount* refcount, int /*initial_refs*/,
       refcount, &refcount->slice_refcount);
 }
 
+#undef move64
+
 static void move64(uint64_t* from, uint64_t* to) {
   *to += *from;
   *from = 0;
