diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1961995..d721654 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4692,7 +4692,7 @@ if (gRPC_INSTALL)
 endif()
 
 
-if (gRPC_BUILD_CODEGEN)
+if (gRPC_BUILD_CODEGEN AND NOT gRPC_USE_PROTO_LITE)
 add_library(grpcpp_channelz
   src/cpp/server/channelz/channelz_service.cc
   src/cpp/server/channelz/channelz_service_plugin.cc
@@ -4749,7 +4749,7 @@ foreach(_hdr
 endforeach()
 
 
-if (gRPC_BUILD_CODEGEN)
+if (gRPC_BUILD_CODEGEN AND NOT gRPC_USE_PROTO_LITE)
 
 if (gRPC_INSTALL)
   install(TARGETS grpcpp_channelz EXPORT gRPCTargets
@@ -4758,9 +4758,9 @@ if (gRPC_INSTALL)
     ARCHIVE DESTINATION ${gRPC_INSTALL_LIBDIR}
   )
 endif()
-endif (gRPC_BUILD_CODEGEN)
+endif (gRPC_BUILD_CODEGEN AND NOT gRPC_USE_PROTO_LITE)
 
-endif (gRPC_BUILD_CODEGEN)
+endif (gRPC_BUILD_CODEGEN AND NOT gRPC_USE_PROTO_LITE)
 if (gRPC_BUILD_TESTS)
 
 if (gRPC_BUILD_CODEGEN)
