diff --git a/src/core/lib/transport/transport.cc b/src/core/lib/transport/transport.cc
index 5d74ec2..89494b5 100644
--- a/src/core/lib/transport/transport.cc
+++ b/src/core/lib/transport/transport.cc
@@ -77,6 +77,8 @@ void grpc_stream_ref_init(grpc_stream_refcount* refcount, int /*initial_refs*/,
                                                              : nullptr);
 }
 
+#undef move64
+
 static void move64bits(uint64_t* from, uint64_t* to) {
   *to += *from;
   *from = 0;
