diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6baab5d37b..48bc6cf0de 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1813,6 +1813,8 @@ target_link_libraries(grpc
   absl::optional
   absl::strings
   absl::inlined_vector
+  absl::time
+  absl::synchronization
 )
 if(_gRPC_PLATFORM_IOS OR _gRPC_PLATFORM_MAC)
   target_link_libraries(grpc "-framework CoreFoundation")
@@ -2413,6 +2415,8 @@ target_link_libraries(grpc_unsecure
   absl::optional
   absl::strings
   absl::inlined_vector
+  absl::time
+  absl::synchronization
 )
 if(_gRPC_PLATFORM_IOS OR _gRPC_PLATFORM_MAC)
   target_link_libraries(grpc_unsecure "-framework CoreFoundation")
diff --git a/cmake/abseil-cpp.cmake b/cmake/abseil-cpp.cmake
index 078d495417..b6f50bd14c 100644
--- a/cmake/abseil-cpp.cmake
+++ b/cmake/abseil-cpp.cmake
@@ -37,4 +37,7 @@ elseif(gRPC_ABSL_PROVIDER STREQUAL "package")
   # Use "CONFIG" as there is no built-in cmake module for absl.
   find_package(absl REQUIRED CONFIG)
   set(_gRPC_FIND_ABSL "if(NOT absl_FOUND)\n  find_package(absl CONFIG)\nendif()")
+  if (gRPC_ABSL_SYNC_ENABLE)
+    add_definitions(-DGPR_ABSEIL_SYNC=1)
+  endif()
 endif()
