diff --git a/src/core/lib/event_engine/windows/grpc_polled_fd_windows.cc b/src/core/lib/event_engine/windows/grpc_polled_fd_windows.cc
index 3bfd15537e..0d4dcd5579 100644
--- a/src/core/lib/event_engine/windows/grpc_polled_fd_windows.cc
+++ b/src/core/lib/event_engine/windows/grpc_polled_fd_windows.cc
@@ -224,7 +224,7 @@ class GrpcPolledFdWindows : public GrpcPolledFd {
     // c-ares overloads this recv_from virtual socket function to receive
     // data on both UDP and TCP sockets, and from is nullptr for TCP.
     if (from != nullptr) {
-      CHECK(*from_len <= recv_from_source_addr_len_);
+      CHECK(*from_len >= recv_from_source_addr_len_);
       memcpy(from, &recv_from_source_addr_, recv_from_source_addr_len_);
       *from_len = recv_from_source_addr_len_;
     }
