diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7822c4f713..4cb75e6eea 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2047,6 +2047,8 @@ target_link_libraries(grpc
   absl::bind_front
   absl::inlined_vector
   absl::flat_hash_set
+  absl::time
+  absl::synchronization
   absl::flat_hash_map
 )
 if(_gRPC_PLATFORM_IOS OR _gRPC_PLATFORM_MAC)
@@ -2596,6 +2598,8 @@ target_link_libraries(grpc_unsecure
   absl::statusor
   absl::status
   absl::inlined_vector
+  absl::time
+  absl::synchronization
   absl::flat_hash_map
 )
 if(_gRPC_PLATFORM_IOS OR _gRPC_PLATFORM_MAC)
diff --git a/cmake/abseil-cpp.cmake b/cmake/abseil-cpp.cmake
index 078d495417..b6f50bd14c 100644
--- a/cmake/abseil-cpp.cmake
+++ b/cmake/abseil-cpp.cmake
@@ -37,4 +37,7 @@ elseif(gRPC_ABSL_PROVIDER STREQUAL "package")
   # Use "CONFIG" as there is no built-in cmake module for absl.
   find_package(absl REQUIRED CONFIG)
   set(_gRPC_FIND_ABSL "if(NOT absl_FOUND)\n  find_package(absl CONFIG)\nendif()")
+  if (gRPC_ABSL_SYNC_ENABLE)
+    add_definitions(-DGPR_ABSEIL_SYNC=1)
+  endif()
 endif()
