diff --git a/cmake/abseil-cpp.cmake b/cmake/abseil-cpp.cmake
index c89ba26f61..bdd8ef1a58 100644
--- a/cmake/abseil-cpp.cmake
+++ b/cmake/abseil-cpp.cmake
@@ -35,3 +35,4 @@ elseif(gRPC_ABSL_PROVIDER STREQUAL "package")
   find_package(absl REQUIRED CONFIG)
 endif()
 set(_gRPC_FIND_ABSL "if(NOT TARGET absl::strings)\n  find_package(absl CONFIG)\nendif()")
+add_definitions(-DGPR_ABSEIL_SYNC=1)
\ No newline at end of file
