diff --git a/CMakeLists.txt b/CMakeLists.txt
index ae06ea3..8b64438 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2180,6 +2180,8 @@ target_link_libraries(grpc
   ${_gRPC_UPB_LIBRARIES}
   ${_gRPC_ALLTARGETS_LIBRARIES}
   absl::flat_hash_map
+  absl::time
+  absl::synchronization
   absl::inlined_vector
   absl::bind_front
   absl::statusor
@@ -2733,6 +2735,8 @@ target_link_libraries(grpc_unsecure
   ${_gRPC_UPB_LIBRARIES}
   ${_gRPC_ALLTARGETS_LIBRARIES}
   absl::flat_hash_map
+  absl::time
+  absl::synchronization
   absl::inlined_vector
   absl::bind_front
   absl::statusor
diff --git a/cmake/abseil-cpp.cmake b/cmake/abseil-cpp.cmake
index c89ba26..68ae4fd 100644
--- a/cmake/abseil-cpp.cmake
+++ b/cmake/abseil-cpp.cmake
@@ -35,3 +35,6 @@ elseif(gRPC_ABSL_PROVIDER STREQUAL "package")
   find_package(absl REQUIRED CONFIG)
 endif()
 set(_gRPC_FIND_ABSL "if(NOT TARGET absl::strings)\n  find_package(absl CONFIG)\nendif()")
+  if (gRPC_ABSL_SYNC_ENABLE)
+    add_definitions(-DGPR_ABSEIL_SYNC=1)
+  endif()
\ No newline at end of file
