diff --git a/src/core/lib/debug/trace.cc b/src/core/lib/debug/trace.cc
index 84c0a3805d..47cbaa9dff 100644
--- a/src/core/lib/debug/trace.cc
+++ b/src/core/lib/debug/trace.cc
@@ -70,8 +70,10 @@ bool TraceFlagList::Set(const char* name, bool enabled) {
 }
 
 void TraceFlagList::Add(TraceFlag* flag) {
-  flag->next_tracer_ = root_tracer_;
-  root_tracer_ = flag;
+  TraceFlag** next = &root_tracer_;
+  while(*next) next = &(*next)->next_tracer_;
+  flag->next_tracer_ = nullptr;
+  *next = flag;
 }
 
 void TraceFlagList::LogAllTracers() {
