diff --git a/CMakeLists.txt b/CMakeLists.txt
index ae77375eae..ad8815d2f9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1960,10 +1960,14 @@ endif()
 
 if(gRPC_BUILD_CSHARP_EXT)
 
-add_library(grpc_csharp_ext SHARED
+add_library(grpc_csharp_ext
   src/csharp/ext/grpc_csharp_ext.c
 )
 
+if (WIN32 AND BUILD_SHARED_LIBS)
+    target_compile_definitions(grpc_csharp_ext PUBLIC GPR_WINDOWS)
+endif()
+
 set_target_properties(grpc_csharp_ext PROPERTIES
   VERSION ${gRPC_CORE_VERSION}
   SOVERSION ${gRPC_CORE_SOVERSION}
@@ -1977,6 +1981,12 @@ if(WIN32 AND MSVC)
     install(FILES ${CMAKE_CURRENT_BINARY_DIR}/grpc_csharp_ext.pdb
       DESTINATION ${gRPC_INSTALL_LIBDIR} OPTIONAL
     )
+
+    install(TARGETS grpc_csharp_ext
+      RUNTIME DESTINATION ${gRPC_INSTALL_BINDIR}
+      LIBRARY DESTINATION ${gRPC_INSTALL_LIBDIR}
+      ARCHIVE DESTINATION ${gRPC_INSTALL_LIBDIR}
+    )
   endif()
 endif()
 
