diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0dc0bd3..7f702cc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4699,6 +4699,7 @@ endif (gRPC_BUILD_TESTS)
 
 option(gRPC_INSTALL_CSHARP_EXT "" ON)
 
+if(gRPC_INSTALL_CSHARP_EXT)
 add_library(grpc_csharp_ext SHARED
   src/csharp/ext/grpc_csharp_ext.c
 )
@@ -4741,6 +4742,7 @@ if (gRPC_INSTALL AND gRPC_INSTALL_CSHARP_EXT)
     ARCHIVE DESTINATION ${gRPC_INSTALL_LIBDIR}
   )
 endif()
+endif()
 
 if (gRPC_BUILD_TESTS)
 
