diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8c285e7..cffd93f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5058,6 +5058,9 @@ endif (gRPC_BUILD_CODEGEN)
 
 endif (gRPC_BUILD_TESTS)
 
+option(gRPC_INSTALL_CSHARP_EXT "" ON)
+
+if (gRPC_INSTALL_CSHARP_EXT)
 add_library(grpc_csharp_ext SHARED
   src/csharp/ext/grpc_csharp_ext.c
 )
@@ -5101,6 +5104,7 @@ if (gRPC_INSTALL)
     ARCHIVE DESTINATION ${gRPC_INSTALL_LIBDIR}
   )
 endif()
+endif (gRPC_INSTALL_CSHARP_EXT)
 
 if (gRPC_BUILD_TESTS)
 
