diff --git a/CMakeLists.txt b/CMakeLists.txt
index d52e199..30ed816 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2072,6 +2072,8 @@ foreach(_hdr
   include/grpc/impl/codegen/sync_generic.h
   include/grpc/impl/codegen/sync_posix.h
   include/grpc/impl/codegen/sync_windows.h
+  include/grpc++/impl/codegen/proto_utils.h
+  include/grpc++/impl/codegen/config_protobuf.h
 )
   string(REPLACE "include/" "" _path ${_hdr})
   get_filename_component(_path ${_path} PATH)
