diff --git a/CMakeLists.txt b/CMakeLists.txt
index e09f729..c85a20a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11847,7 +11847,12 @@ endif (gRPC_BUILD_TESTS)
 
 
 
-
+if (gRPC_INSTALL)
+  install(EXPORT gRPCTargets
+    DESTINATION ${CMAKE_INSTALL_CMAKEDIR}
+    NAMESPACE gRPC::
+  )
+endif()
 
 foreach(_config gRPCConfig gRPCConfigVersion)
   configure_file(tools/cmake/${_config}.cmake.in
