diff --git a/src/base/PdfCompilerCompat.h b/src/base/PdfCompilerCompat.h
index 706875f..8efb349 100644
--- a/src/base/PdfCompilerCompat.h
+++ b/src/base/PdfCompilerCompat.h
@@ -113,6 +113,10 @@
 #define NOMINMAX
 #endif
 
+#if defined(WINAPI_FAMILY) && (WINAPI_FAMILY == WINAPI_FAMILY_APP)
+#include <combaseapi.h>
+#endif
+
 // Integer types - fixed size types guaranteed to work anywhere
 // because we detect the right underlying type name to use with
 // CMake. Use typedefs rather than macros for saner error messages
