diff --git a/CMakeLists.txt b/CMakeLists.txt
index df623ef..8e653b8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -569,10 +569,4 @@ CONFIGURE_FILE(${PoDoFo_SOURCE_DIR}/podofo_config.h.in ${PoDoFo_BINARY_DIR}/podo
 # To use these dependencies set PODOFO_DIR to the podofo BUILD directory in
 # your build (eg -DPODOFO_DIR=/path/to/podofo when running cmake to configure
 # the app that'll use podofo). See: FIND_PACKAGE(...) in the cmake docs.
-IF(PODOFO_BUILD_SHARED)
-  EXPORT(TARGETS podofo_shared FILE "${CMAKE_CURRENT_BINARY_DIR}/PoDoFoConfig.cmake")
-ENDIF(PODOFO_BUILD_SHARED)
-IF(PODOFO_BUILD_STATIC)
-  EXPORT(TARGETS podofo_static FILE "${CMAKE_CURRENT_BINARY_DIR}/PoDoFoConfig.cmake")
-ENDIF(PODOFO_BUILD_STATIC)
 
diff --git a/src/podofo/CMakeLists.txt b/src/podofo/CMakeLists.txt
index bba6b5f..16f0798 100644
--- a/src/podofo/CMakeLists.txt
+++ b/src/podofo/CMakeLists.txt
@@ -275,10 +275,14 @@ IF(PODOFO_BUILD_STATIC)
         CACHE INTERNAL "Which PoDoFo library variant to depend on")
     SET(USING_SHARED_PODOFO FALSE)
     INSTALL(TARGETS podofo_static
+        EXPORT PoDoFoConfig
         RUNTIME DESTINATION "bin"
         LIBRARY DESTINATION "${LIBDIRNAME}"
         ARCHIVE DESTINATION "${LIBDIRNAME}"
         )
+    INSTALL(EXPORT PoDoFoConfig
+        DESTINATION share/podofo
+        )
 ENDIF(PODOFO_BUILD_STATIC)
 
 IF(PODOFO_BUILD_SHARED)
@@ -301,10 +305,14 @@ IF(PODOFO_BUILD_SHARED)
         CACHE INTERNAL "Which PoDoFo library variant to depend on")
     SET(USING_SHARED_PODOFO TRUE)
     INSTALL(TARGETS podofo_shared
+        EXPORT PoDoFoConfig
         RUNTIME DESTINATION "bin"
         LIBRARY DESTINATION "${LIBDIRNAME}"
         ARCHIVE DESTINATION "${LIBDIRNAME}"
         )
+    INSTALL(EXPORT PoDoFoConfig
+        DESTINATION share/podofo
+        )
 
 
       # Create a pkg-config file for linking against shared library
