diff --git a/CMakeLists.txt b/CMakeLists.txt
index c8d45d2..278a57c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -135,11 +135,7 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DMORPH_FONTS_DIR=\"\\\"${PROJECT_SOURCE
 
 # Lib finding
 find_package(OpenCV QUIET) # No longer required for morph::Visual, used by some tests
-if(APPLE)
-  find_package(OpenGL REQUIRED)
-else()
-  find_package(OpenGL REQUIRED EGL) # GL ES 3.1 used in one gl compute example
-endif()
+find_package(OpenGL REQUIRED)
 
 if(USE_GLEW)
   find_package(GLEW REQUIRED)
