diff --git a/CMakeLists.txt b/CMakeLists.txt
index c8d45d2..366e213 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -216,8 +216,11 @@ add_subdirectory(morph)
 add_subdirectory(include)
 
 # Unit testing using the ctest framework
-enable_testing()
-add_subdirectory(tests)
+option(BUILD_TESTS "Build tests" OFF)
+if(BUILD_TESTS)
+  enable_testing()
+  add_subdirectory(tests)
+endif(BUILD_TESTS)
 
 # incbin for Visual Studio
 add_subdirectory(buildtools)
@@ -226,8 +229,11 @@ add_subdirectory(buildtools)
 # *installed* morphologica, as opposed to an in-tree morphologica.
 add_subdirectory(fonts)
 
-# Example code (you can also see tests/ for examples)
-add_subdirectory(examples)
+option(BUILD_EXAMPLES "Build examples" OFF)
+if(BUILD_EXAMPLES)
+  # Example code (you can also see tests/ for examples)
+  add_subdirectory(examples)
+endif(BUILD_EXAMPLES)
 
 # first we can indicate the documentation build as an option and set it to ON by default
 option(BUILD_DOC "Build documentation" OFF)
