diff --git a/CMakeLists.txt b/CMakeLists.txt
index df05f5d..5a1c727 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -189,6 +189,10 @@ add_library(zlibstatic STATIC ${ZLIB_SRCS} ${ZLIB_ASMS} ${ZLIB_PUBLIC_HDRS} ${ZL
 set_target_properties(zlib PROPERTIES DEFINE_SYMBOL ZLIB_DLL)
 set_target_properties(zlib PROPERTIES SOVERSION 1)
 
+if(MSVC)
+    target_link_options(zlib PRIVATE /APPCONTAINER)
+endif()
+
 if(NOT CYGWIN)
     # This property causes shared libraries on Linux to have the full version
     # encoded into their final filename.  We disable this on Cygwin because
