diff --git a/libs/Common/CMakeLists.txt b/libs/Common/CMakeLists.txt
index 18899af..660295f 100644
--- a/libs/Common/CMakeLists.txt
+++ b/libs/Common/CMakeLists.txt
@@ -16,7 +16,9 @@ TARGET_LINK_LIBRARIES(Common ${Boost_LIBRARIES} ${OpenCV_LIBS})
 
 # Install
 SET_TARGET_PROPERTIES(Common PROPERTIES
-	PUBLIC_HEADER "${LIBRARY_FILES_H}")
+	PUBLIC_HEADER "${LIBRARY_FILES_H}" 
+	OUTPUT_NAME "OpenMVS_Common"
+	EXPORT_NAME "OpenMVS_Common")
 INSTALL(TARGETS Common
 	EXPORT OpenMVSTargets
 	LIBRARY DESTINATION "${INSTALL_LIB_DIR}"
