 libs/Common/Common.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/libs/Common/Common.cpp b/libs/Common/Common.cpp
index 052951d..8dc404e 100644
--- a/libs/Common/Common.cpp
+++ b/libs/Common/Common.cpp
@@ -31,6 +31,11 @@ namespace boost {
 		ASSERT("boost exception thrown" == NULL);
 		exit(EXIT_FAILURE);
 	}
+	void throw_exception(std::exception const & e, boost::source_location const &) {
+		VERBOSE("exception thrown: %s", e.what());
+		ASSERT("boost exception thrown" == NULL);
+		exit(EXIT_FAILURE);
+	}
 } // namespace boost
 #endif
 #endif
