diff --git a/apps/Viewer/CMakeLists.txt b/apps/Viewer/CMakeLists.txt
index c519040..934cd50 100644
--- a/apps/Viewer/CMakeLists.txt
+++ b/apps/Viewer/CMakeLists.txt
@@ -16,18 +16,6 @@ else()
 	MESSAGE("-- Can't find GLEW. Continuing without it.")
 	RETURN()
 endif()
-if(CMAKE_COMPILER_IS_GNUCXX)
-	FIND_PACKAGE(PkgConfig QUIET)
-	pkg_search_module(GLFW QUIET glfw3)
-	if(GLFW_FOUND)
-		INCLUDE_DIRECTORIES(${GLFW_INCLUDE_DIRS})
-		ADD_DEFINITIONS(${GLFW_DEFINITIONS})
-		MESSAGE(STATUS "GLFW3 ${GLFW_VERSION} found (include: ${GLFW_INCLUDE_DIRS})")
-	else()
-		MESSAGE("-- Can't find GLFW3. Continuing without it.")
-		RETURN()
-	endif()
-else()
 	FIND_PACKAGE(glfw3 QUIET)
 	if(glfw3_FOUND)
 		INCLUDE_DIRECTORIES(${glfw3_INCLUDE_DIRS})
@@ -37,7 +25,6 @@ else()
 		MESSAGE("-- Can't find GLFW3. Continuing without it.")
 		RETURN()
 	endif()
-endif()
 
 # List sources files
 FILE(GLOB PCH_C "Common.cpp")
diff --git a/libs/IO/CMakeLists.txt b/libs/IO/CMakeLists.txt
index a354376..d4faf34 100644
--- a/libs/IO/CMakeLists.txt
+++ b/libs/IO/CMakeLists.txt
@@ -43,7 +43,7 @@ cxx_library_with_type_no_pch(IO "Libs" "STATIC" "${cxx_default}"
 set_target_pch(IO Common.h)
 
 # Link its dependencies
-TARGET_LINK_LIBRARIES(IO Common ${PNG_LIBRARIES} ${JPEG_LIBRARIES} ${TIFF_LIBRARIES} ${EXIV2_LIBS})
+TARGET_LINK_LIBRARIES(IO Common ${TIFF_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES} ${EXIV2_LIBS})
 
 # Install
 SET_TARGET_PROPERTIES(IO PROPERTIES
