diff --git a/CMakeLists.txt b/CMakeLists.txt
index e3977e1f..0eb658af 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -91,11 +91,16 @@ if(WOLFTPM_SINGLE_THREADED)
 endif()
 
 # Mutex locking
-set(WOLFTPM_NO_LOCK "no" CACHE STRING
-    "Enable thread mutex locking (default: enabled)")
+if(WOLFTPM_SINGLE_THREADED)
+    set(_WOLFTPM_NO_LOCK_DEFAULT "yes")
+else()
+    set(_WOLFTPM_NO_LOCK_DEFAULT "no")
+endif()
+set(WOLFTPM_NO_LOCK "${_WOLFTPM_NO_LOCK_DEFAULT}" CACHE STRING
+    "Disable thread mutex locking (default: ${_WOLFTPM_NO_LOCK_DEFAULT})")
 set_property(CACHE WOLFTPM_NO_LOCK
     PROPERTY STRINGS "yes;no")
-if(NOT WOLFTPM_NO_LOCK)
+if(WOLFTPM_NO_LOCK)
     list(APPEND WOLFTPM_DEFINITIONS
         "-DWOLFTPM_NO_LOCK")
 endif()
