diff --git a/lib/clang/20/include/stdatomic.h b/lib/clang/20/include/stdatomic.h
index 8ecef1a059..f1512689c8 100644
--- a/lib/clang/20/include/stdatomic.h	
+++ b/lib/clang/20/include/stdatomic.h
@@ -78,9 +78,11 @@ typedef enum memory_order {
 
 /* 7.17.4 Fences */
 
+#ifndef _MSC_VER
 /* These should be provided by the libc implementation. */
 void atomic_thread_fence(memory_order);
 void atomic_signal_fence(memory_order);
+#endif
 
 #define atomic_thread_fence(order) __c11_atomic_thread_fence(order)
 #define atomic_signal_fence(order) __c11_atomic_signal_fence(order)
