From cb2df4c372c115df84805043a7785f38eb4eb082 Mon Sep 17 00:00:00 2001
From: David Faure <david.faure@kdab.com>
Date: Sun, 26 Mar 2023 17:27:02 +0200
Subject: [PATCH] Ensure KDSoapConfig.cmake finds Qt first.

Fixes #258
---
 KDSoapConfig.cmake.in | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/KDSoapConfig.cmake.in b/KDSoapConfig.cmake.in
index d20649a4..f7bfa82b 100644
--- a/KDSoapConfig.cmake.in
+++ b/KDSoapConfig.cmake.in
@@ -8,6 +8,11 @@
 
 @PACKAGE_INIT@
 
+include(CMakeFindDependencyMacro)
+
+find_dependency(Qt@Qt_VERSION_MAJOR@Core @QT_MIN_VERSION@)
+find_dependency(Qt@Qt_VERSION_MAJOR@Network @QT_MIN_VERSION@)
+
 set_and_check(KDSoap_INCLUDE_DIR "@PACKAGE_INSTALL_INCLUDE_DIR@")
 
 set(KDSoap_INCLUDE_DIRS "${KDSoap_INCLUDE_DIR}")
