From 6d10bef865b69764f6e0ddd2b0f6a84e484cdb37 Mon Sep 17 00:00:00 2001
From: Long Nguyen <nguyen.long.908132@gmail.com>
Date: Sun, 9 May 2021 13:25:14 +0700
Subject: [PATCH 09/14] No fcntl in magic.c

---
 src/magic.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/magic.c b/src/magic.c
index e9aeafa..382bd96 100644
--- a/src/magic.c
+++ b/src/magic.c
@@ -462,7 +462,7 @@ file_or_fd(struct magic_set *ms, const char *inname, int fd)
 			rv = 0;
 			goto done;
 		}
-#if O_CLOEXEC == 0
+#if O_CLOEXEC == 0 && !defined(_WIN32)
 		(void)fcntl(fd, F_SETFD, FD_CLOEXEC);
 #endif
 	}
-- 
2.29.2.windows.2

