From 15b0f505ff9eefd41b74ffdd4230355e933263ca Mon Sep 17 00:00:00 2001
From: Long Nguyen <nguyen.long.908132@gmail.com>
Date: Sun, 9 May 2021 17:45:50 +0700
Subject: [PATCH 11/14] Remove pipe related functions in funcs.c

---
 src/funcs.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/funcs.c b/src/funcs.c
index b926625..b585486 100644
--- a/src/funcs.c
+++ b/src/funcs.c
@@ -809,6 +809,7 @@ file_print_guid(char *str, size_t len, const uint64_t *guid)
 	    g->data4[6], g->data4[7]);
 }
 
+#ifndef _WIN32
 protected int
 file_pipe_closexec(int *fds)
 {
@@ -827,6 +828,7 @@ protected int
 file_clear_closexec(int fd) {
 	return fcntl(fd, F_SETFD, 0);
 }
+#endif
 
 protected char *
 file_strtrim(char *str)
-- 
2.29.2.windows.2

