diff --git a/cmake/StlabUtil.cmake b/cmake/StlabUtil.cmake
index 3e36278..aeb284f 100644
--- a/cmake/StlabUtil.cmake
+++ b/cmake/StlabUtil.cmake
@@ -2,14 +2,14 @@
 # functionality useful for stlab builds.
 
 include( CheckCXXSymbolExists )
-include( CheckCXXSourceRuns )
+include( CheckCXXSourceCompiles )
 
 # Determine if the selected C++ compiler has functional coroutines. Set the
 # specified 'result_var' to 'TRUE' if they are determined to be disfunctional
 # and 'FALSE' otherwise. Note that this check consists of a smoke test and does
 # not check all the ways this feature may be deficient.
 function( stlab_check_disfunctional_coroutines result_var )
-  check_cxx_source_runs( "
+  check_cxx_source_compiles( "
 #include <concepts>
 #include <coroutine>
 #include <exception>
