diff --git a/CMakeLists.txt b/CMakeLists.txt
index d058bbe..260d940 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -123,8 +123,7 @@ if ( stlab.testing OR stlab.boost_variant OR stlab.boost_optional )
     endif()
   endif()
 
-
-  if(NOT TARGET Boost::unit_test_framework)
+  if (stlab.testing AND NOT TARGET Boost::unit_test_framework)
     message(FATAL_ERROR "Could not find Boost unit test framework.")
   endif()
   
