diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7e8a282..f550c2d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,7 +15,7 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 # Find Boost dependency
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
 include(find_or_download_package)
-find_or_download_package(Boost INSTALL_WITH_YOMM)
+find_package(Boost REQUIRED)
 message(STATUS "Using Boost libraries from ${Boost_INCLUDE_DIRS}")
 
 if(CMAKE_COMPILER_IS_GNUCXX OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
diff --git a/cmake/Config.cmake.in b/cmake/Config.cmake.in
index 7272504..6423a17 100644
--- a/cmake/Config.cmake.in
+++ b/cmake/Config.cmake.in
@@ -1,7 +1,7 @@
 include(CMakeFindDependencyMacro)
 
 # Tell library users about the Boost dependency
-find_dependency(Boost 1.53 HINTS ${CMAKE_CURRENT_LIST_DIR}/..)
+find_dependency(Boost 1.53)
 
 # Add the targets file
 include("${CMAKE_CURRENT_LIST_DIR}/YOMM2Targets.cmake")
