diff -urN a/c/Projects/Win32/Res/AllInOne/AllInOne.rc b/c/Projects/Win32/Res/AllInOne/AllInOne.rc
--- a/c/Projects/Win32/Res/AllInOne/AllInOne.rc	2012-03-19 16:18:11.000000000 +0000
+++ b/c/Projects/Win32/Res/AllInOne/AllInOne.rc	2018-09-07 16:08:59.871434376 +0100
@@ -7,7 +7,7 @@
 //
 // Generated from the TEXTINCLUDE 2 resource.
 //
-#include "afxres.h"
+#include "windows.h"
 
 
 /////////////////////////////////////////////////////////////////////////////
@@ -75,7 +75,7 @@
 
 2 TEXTINCLUDE 
 BEGIN
-    "#include ""afxres.h""\r\n"
+    "#include ""windows.h""\r\n"
     "\r\n"
     "\0"
 END
