diff -urN a/c/README b/c/README
--- a/c/README	2012-10-13 05:52:45.000000000 +0100
+++ b/c/README	2018-09-07 16:06:24.602562653 +0100
@@ -3,7 +3,7 @@
 
 Project Description:
     The Apache Xalan C/C++ Project provides a library and a 
-    comand line program to transform XML documents using a
+    command line program to transform XML documents using a
     stylesheet that conforms to XSLT 1.0 standards.
 
     The source package contains all the code to create the library,
@@ -84,7 +84,7 @@
     XalanMessages_1_11.exp   - Windows DLL export definitions
     XalanMessages_1_11.lib   - Windows DLL library definitions
     XalanMessages_1_11D.exp  - Windows DLL export definitions
-    XalanMessages_1_11D.lib  - Windows DLL library defnitions
+    XalanMessages_1_11D.lib  - Windows DLL library definitions
 
 SUPPORTED PLATFORMS:
     - This is only a partial list
