diff --git a/include/jwt-cpp/base.h b/include/jwt-cpp/base.h
index 375e0eb..4023709 100644
--- a/include/jwt-cpp/base.h
+++ b/include/jwt-cpp/base.h
@@ -2,6 +2,10 @@
 #include <string>
 #include <array>
 
+#ifdef _MSC_VER
+#pragma warning(disable : 4267)
+#endif
+
 #ifdef __has_cpp_attribute
 #if __has_cpp_attribute(fallthrough)
 #define JWT_FALLTHROUGH [[fallthrough]]
diff --git a/include/jwt-cpp/jwt.h b/include/jwt-cpp/jwt.h
index 6e55729..4579310 100644
--- a/include/jwt-cpp/jwt.h
+++ b/include/jwt-cpp/jwt.h
@@ -12,6 +12,11 @@
 #include <openssl/ec.h>
 #include <openssl/err.h>
 
+#ifdef _MSC_VER
+#pragma warning(disable : 4267)
+#pragma warning(disable : 4067)
+#endif
+
 //If openssl version less than 1.1
 #if OPENSSL_VERSION_NUMBER < 0x10100000L
 #define OPENSSL10
