diff --git a/src/win32/filesystem_win32.c b/src/win32/filesystem_win32.c
index 5dc36a8..c95e5a1 100644
--- a/src/win32/filesystem_win32.c
+++ b/src/win32/filesystem_win32.c
@@ -189,7 +189,15 @@ zix_canonical_path(ZixAllocator* const allocator, const char* const path)
     return NULL;
   }
 
-  const HANDLE h =
+#if WINAPI_FAMILY == WINAPI_FAMILY_APP
+    const HANDLE h = CreateFile2(
+                full,
+                GENERIC_READ,
+                FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE,
+                OPEN_EXISTING,
+                NULL);
+#else
+    const HANDLE h =
     CreateFile(full,
                FILE_READ_ATTRIBUTES,
                FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE,
@@ -197,6 +205,7 @@ zix_canonical_path(ZixAllocator* const allocator, const char* const path)
                OPEN_EXISTING,
                FILE_FLAG_BACKUP_SEMANTICS,
                NULL);
+#endif
 
   const DWORD flags      = FILE_NAME_NORMALIZED | VOLUME_NAME_DOS;
   const DWORD final_size = GetFinalPathNameByHandle(h, NULL, 0U, flags);
