diff --git a/cpp/CMakeLists.txt b/cpp/CMakeLists.txt
index 27e4680..88df056 100644
--- a/cpp/CMakeLists.txt
+++ b/cpp/CMakeLists.txt
@@ -459,11 +459,13 @@ if (BUILD_STATIC_LIB)
   add_library (phonenumber STATIC ${SOURCES})
   target_link_libraries (phonenumber ${LIBRARY_DEPS})
   target_include_directories(phonenumber PUBLIC $<INSTALL_INTERFACE:include>)
+  target_compile_features(phonenumber PUBLIC cxx_std_17)
 
   if (BUILD_GEOCODER)
     add_library (geocoding STATIC ${GEOCODING_SOURCES})
     target_link_libraries (geocoding ${LIBRARY_DEPS})
     target_include_directories(geocoding PUBLIC $<INSTALL_INTERFACE:include>)
+    target_compile_features(geocoding PUBLIC cxx_std_17)
     add_dependencies (geocoding generate_geocoding_data)
     add_dependencies (phonenumber generate_geocoding_data)
   endif ()
@@ -479,6 +481,7 @@ if (BUILD_SHARED_LIBS)
   add_library (phonenumber-shared SHARED ${SOURCES})
   target_link_libraries (phonenumber-shared ${LIBRARY_DEPS})
   target_include_directories(phonenumber-shared PUBLIC $<INSTALL_INTERFACE:include>)
+  target_compile_features(phonenumber-shared PUBLIC cxx_std_17)
 
   set_target_properties (phonenumber-shared
     PROPERTIES
@@ -495,6 +498,7 @@ if (BUILD_SHARED_LIBS)
     add_library (geocoding-shared SHARED ${GEOCODING_SOURCES})
     target_link_libraries (geocoding-shared ${LIBRARY_DEPS})
     target_include_directories(geocoding-shared PUBLIC $<INSTALL_INTERFACE:include>)
+    target_compile_features(geocoding-shared PUBLIC cxx_std_17)
     add_dependencies (geocoding-shared generate_geocoding_data)
     add_dependencies (phonenumber-shared generate_geocoding_data)
 
