diff --git a/config-msvc.h b/config-msvc.h
index 9c7f477..a812546 100644
--- a/config-msvc.h
+++ b/config-msvc.h
@@ -143,7 +143,7 @@
 #define HAVE_SYS_TYPES_H 1
 
 /* Define to 1 if you have the <unistd.h> header file. */
-#define HAVE_UNISTD_H 1
+#undef HAVE_UNISTD_H
 
 /* Define to 1 if you have the <zlib.h> header file. */
 #define HAVE_ZLIB_H 1
diff --git a/src/gaiageo/gg_extras.c b/src/gaiageo/gg_extras.c
index ff3dd7c..49ec986 100644
--- a/src/gaiageo/gg_extras.c
+++ b/src/gaiageo/gg_extras.c
@@ -59,14 +59,6 @@ the terms of any one of the MPL, the GPL or the LGPL.
 #include <spatialite/sqlite.h>
 #include <spatialite/gaiageo.h>
 
-#if defined(_WIN32) && !defined(__MINGW32__)
-static double
-rint (double x)
-{
-    return floor (x + 0.5);
-}
-#endif
-
 static void
 auxGridSnapPoint (int dimension_model, gaiaPointPtr pt, gaiaGeomCollPtr result,
 		  double origin_x, double origin_y, double origin_z,
