diff --git a/src/gaiageo/gg_shape.c b/src/gaiageo/gg_shape.c
index ee2f1cf..01f2571 100644
--- a/src/gaiageo/gg_shape.c
+++ b/src/gaiageo/gg_shape.c
@@ -58,7 +58,7 @@ the terms of any one of the MPL, the GPL or the LGPL.
 #endif
 
 #ifdef _WIN32
-#include <Windows.h>
+#include <windows.h>
 #endif
 
 #if OMIT_ICONV == 0		/* if ICONV is disabled no SHP support is available */
diff --color -ur a/configure.ac b/configure.ac
--- a/configure.ac	2022-07-21 17:23:50.490298108 +0200
+++ b/configure.ac	2022-07-21 17:25:45.671489524 +0200
@@ -116,7 +116,6 @@
 
 # Checks for installed libraries
 AC_CHECK_LIB(sqlite3,sqlite3_prepare_v2,,AC_MSG_ERROR(['libsqlite3' is required but it doesn't seem to be installed on this system.]),-lm)
-AC_CHECK_LIB(z,inflateInit_,,AC_MSG_ERROR(['libz' is required but it doesn't seem to be installed on this system.]),-lm)
 
 AC_CONFIG_FILES([Makefile \
 		src/Makefile \
