diff --git a/CMakeLists.txt b/CMakeLists.txt
index e2b69fe..aff5917 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -66,6 +66,7 @@ endif ()
 ######################################################################
 
 # Use external CommsChampion project or compile it in place
+if(FALSE)
 set (CC_EXTERNAL_TGT "comms_champion_external")
 if (OPT_BUILD_TEST OR OPT_BUILD_PLUGIN)
     set (external_cc_needed TRUE)
@@ -107,6 +108,10 @@ elseif (external_cc_needed)
     find_package(CommsChampion NO_MODULE)
     set (cc_compile_file "${OPT_CC_MAIN_INSTALL_DIR}/lib/LibComms/cmake/CC_Compile.cmake")
 endif ()
+else()
+    find_package(LibComms CONFIG REQUIRED)
+    set(cc_compile_file "${LibComms_DIR}/CC_Compile.cmake")
+endif()
 
 if (EXISTS ${cc_compile_file})
     # Set compiler flags
