diff --git a/ql/CMakeLists.txt b/ql/CMakeLists.txt
index fc137f41d..7ac42a58b 100644
--- a/ql/CMakeLists.txt
+++ b/ql/CMakeLists.txt
@@ -2220,7 +2220,6 @@ set(QL_GENERATED_HEADERS
     ${PROJECT_BINARY_DIR}/ql/cashflows/all.hpp
     ${PROJECT_BINARY_DIR}/ql/currencies/all.hpp
     ${PROJECT_BINARY_DIR}/ql/experimental/all.hpp
-    ${PROJECT_BINARY_DIR}/ql/experimental/amortizingbonds/all.hpp
     ${PROJECT_BINARY_DIR}/ql/experimental/asian/all.hpp
     ${PROJECT_BINARY_DIR}/ql/experimental/averageois/all.hpp
     ${PROJECT_BINARY_DIR}/ql/experimental/barrieroption/all.hpp
