diff --git a/profiler/CMakeLists.txt b/profiler/CMakeLists.txt
index bd7ddc0..9fb7ac6 100644
--- a/profiler/CMakeLists.txt
+++ b/profiler/CMakeLists.txt
@@ -191,7 +191,7 @@ if(NOT DEFINED GIT_REV)
 endif()
 
 find_package(Git)
-if(Git_FOUND)
+if(0)
     add_custom_target(git-ref
         COMMAND ${GIT_EXECUTABLE} -C ${CMAKE_CURRENT_SOURCE_DIR} log -1 "--format=#pragma once %nnamespace tracy { static inline const char* GitRef = %x22%h%x22; }" ${GIT_REV} > GitRef.hpp.tmp
         COMMAND ${CMAKE_COMMAND} -E copy_if_different GitRef.hpp.tmp GitRef.hpp
