diff --git a/CMakeLists.txt b/CMakeLists.txt
index 74efee3..7f87e13 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -33,13 +33,13 @@ endif()
 # Exclude any embedded CATCH
 list_filter(quickcpplib_HEADERS EXCLUDE REGEX "include/quickcpplib/CATCH")
 list_filter(quickcpplib_HEADERS EXCLUDE REGEX "include/quickcpplib/detail/config\\.hpp")
-if (QUICKCPPLIB_USE_SYSTEM_SPAN_LITE OR QUICKCPPLIB_REQUIRE_CXX20)
+if (1)
   list_filter(quickcpplib_HEADERS EXCLUDE REGEX "include/quickcpplib/span-lite/")
 endif()
-if (QUICKCPPLIB_USE_SYSTEM_BYTE_LITE OR QUICKCPPLIB_REQUIRE_CXX17)
+if (1)
   list_filter(quickcpplib_HEADERS EXCLUDE REGEX "include/quickcpplib/byte/")
 endif()
-if (QUICKCPPLIB_REQUIRE_CXX17)
+if (1)
   list_filter(quickcpplib_HEADERS EXCLUDE REGEX "include/quickcpplib/optional/")
 endif()
 # Make an interface only library so dependent CMakeLists can bring in this header-only library
