diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 49dcf02..d448928 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -384,6 +384,7 @@ endif()
 
 option(BUILD_PANGOLIN_FFMPEG "Build support for ffmpeg video input" ON)
 if(BUILD_PANGOLIN_FFMPEG AND BUILD_PANGOLIN_VIDEO)
+  add_definitions(-DHAVE_FFMPEG_AVPIXELFORMAT)
   find_package(FFMPEG QUIET)
   if(FFMPEG_FOUND)
     set(HAVE_FFMPEG 1)
