diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 7f364a7..b6567a2 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -213,6 +213,9 @@ endif()
 find_package(PythonLibs QUIET)
 if(BUILD_PANGOLIN_GUI AND BUILD_PANGOLIN_VARS AND PYTHONLIBS_FOUND AND NOT _WIN_)
   set(HAVE_PYTHON 1)
+  if (UNIX)
+    set(PYTHON_LIBRARY ${PYTHON_LIBRARY} dl)
+  endif()
   list(APPEND HEADERS
     ${INCDIR}/console/ConsoleInterpreter.h
     ${INCDIR}/console/ConsoleView.h
