diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4047714..e4e149c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -62,6 +62,11 @@ if(_OSX_)
     set(CMAKE_MACOSX_RPATH ON)
 endif()
 
+if (NOT BUILD_EXTERN_GLEW)
+    find_package(GLEW REQUIRED)
+    link_libraries(GLEW::GLEW)
+endif()
+
 # Overide with cmake -DCMAKE_BUILD_TYPE=Debug {dir}
 if( NOT CMAKE_BUILD_TYPE AND NOT _WIN_ )
    message("Build type not set (defaults to release)")
