
diff -r 398b67c0a7d3 -r 7c2f418f701b src/video/winrt/SDL_winrtgamebar.cpp
--- a/src/video/winrt/SDL_winrtgamebar.cpp	Wed Jun 10 09:36:10 2020 -0700
+++ b/src/video/winrt/SDL_winrtgamebar.cpp	Wed Jun 10 09:38:43 2020 -0700
@@ -25,7 +25,7 @@
 /* Windows includes */
 #include <roapi.h>
 #include <windows.foundation.h>
-#include <EventToken.h>
+#include <windows.system.h>
 
 
 /* SDL includes */




