diff --git a/configure.ac b/configure.ac
index f35c22949..8787a1c0b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -10,6 +10,8 @@
 
 AC_INIT([CoinUtils],[devel],[https://github.com/coin-or/CoinUtils/issues/new],[coin-or-coinutils],[https://github.com/coin-or/CoinUtils])
 
+AC_CONFIG_MACRO_DIR([m4])
+
 AC_COPYRIGHT([
 Copyright 2006 International Business Machines and others.
 All Rights Reserved.
