diff --git a/configure.ac b/configure.ac
index b9c0e73..08807bf 100644
--- a/configure.ac
+++ b/configure.ac
@@ -75,9 +75,12 @@ __asm__ (".symver versioned_symbol, versioned_symbol@IDN2_0.0.0");
 # check for gtk-doc
 #
 m4_ifdef([GTK_DOC_CHECK], [
-GTK_DOC_CHECK([1.14],[--flavour no-tmpl])
+#GTK_DOC_CHECK([1.14],[--flavour no-tmpl])
 ],[
 AM_CONDITIONAL([ENABLE_GTK_DOC], false)
+AM_CONDITIONAL([GTK_DOC_BUILD_HTML], false)
+AM_CONDITIONAL([GTK_DOC_BUILD_PDF], false)
+AM_CONDITIONAL([HAVE_GTK_DOC], false)
 ])
 # needed for some older versions of gtk-doc
 m4_ifdef([GTK_DOC_USE_LIBTOOL], [], [
diff --git a/gtk-doc.make b/gtk-doc.make
index 7d9a27f..973707f 100644
--- a/gtk-doc.make
+++ b/gtk-doc.make
@@ -268,7 +268,7 @@ maintainer-clean-local:
 
 install-data-local:
 	@installfiles=`echo $(builddir)/html/*`; \
-	if test "$$installfiles" = '$(builddir)/html/*'; \
+	if true; \
 	then echo 1>&2 'Nothing to install' ; \
 	else \
 	  if test -n "$(DOC_MODULE_VERSION)"; then \
