diff --git a/conf.d/Makefile.am b/conf.d/Makefile.am
index ff03fb2eb..769fc6005 100644
--- a/conf.d/Makefile.am
+++ b/conf.d/Makefile.am
@@ -95,9 +95,9 @@ install-data-hook:
 	@(echo cd $(DESTDIR)$(configdir);			\
 	  cd $(DESTDIR)$(configdir);				\
 	  for i in $(CONF_LINKS); do				\
-	    echo $(RM) $$i";" ln -s $(templatedir)/$$i .;	\
+	    echo $(RM) $$i";" $(LN_S) $(DESTDIR)$(templatedir)/$$i .;	\
 	    $(RM) $$i;						\
-	    ln -s $(templatedir)/$$i .;				\
+	    $(LN_S) $(DESTDIR)$(templatedir)/$$i .;				\
 	  done)
 uninstall-local:
 	@(echo cd $(DESTDIR)$(configdir);			\
diff --git a/src/Makefile.am b/src/Makefile.am
index 35e820d60..42bd7c5a2 100644
--- a/src/Makefile.am	
+++ b/src/Makefile.am
@@ -33,7 +33,7 @@ fontconfig_def_dependency = fontconfig.def
 
 install-libtool-import-lib: libfontconfig.la
 	$(MKDIR_P) $(DESTDIR)$(libdir)
-	$(INSTALL) .libs/libfontconfig.dll.a $(DESTDIR)$(libdir)/libfontconfig.dll.a
+#	$(INSTALL) .libs/libfontconfig.dll.a $(DESTDIR)$(libdir)/libfontconfig.dll.a
 	$(INSTALL) fontconfig.def $(DESTDIR)$(libdir)/fontconfig.def
 
 uninstall-libtool-import-lib:
@@ -58,7 +58,8 @@ fontconfig.lib : libfontconfig.la
 	lib -name:libfontconfig-@LIBT_CURRENT_MINUS_AGE@.dll -def:fontconfig.def -out:$@
 
 install-ms-import-lib:
-	$(INSTALL) fontconfig.lib $(DESTDIR)$(libdir)
+	$(MKDIR_P) $(DESTDIR)$(libdir)
+	$(INSTALL) .libs/fontconfig.lib $(DESTDIR)$(libdir)
 
 uninstall-ms-import-lib:
 	$(RM) $(DESTDIR)$(libdir)/fontconfig.lib
diff --git a/fc-cache/fc-cache.c b/fc-cache/fc-cache.c
index 87e302083..8b5012110 100644
--- a/fc-cache/fc-cache.c
+++ b/fc-cache/fc-cache.c
@@ -34,11 +34,20 @@
 #include <fontconfig/fontconfig.h>
 #include <stdio.h>
 #include <stdlib.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
+#ifdef _MSC_VER
+#include <io.h>
+#endif
 #include <sys/types.h>
+#if HAVE_SYS_STAT_H
 #include <sys/stat.h>
+#endif
 #include <errno.h>
+#if HAVE_FCNTL_H
 #include <fcntl.h>
+#endif
 #include <dirent.h>
 #include <string.h>
 #include <locale.h>
diff --git a/fc-cat/fc-cat.c b/fc-cat/fc-cat.c
index 69611bcc3..e7b0efbaa 100644
--- a/fc-cat/fc-cat.c
+++ b/fc-cat/fc-cat.c
@@ -36,9 +36,16 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
+#ifdef _MSC_VER
+#include <io.h>
+#endif
 #include <sys/types.h>
+#if HAVE_SYS_STAT_H
 #include <sys/stat.h>
+#endif
 #include <errno.h>
 #include <locale.h>
 
diff --git a/fc-conflist/fc-conflist.c b/fc-conflist/fc-conflist.c
index 5c40a0f09..affaca09e 100644
--- a/fc-conflist/fc-conflist.c
+++ b/fc-conflist/fc-conflist.c
@@ -35,7 +35,12 @@
 
 #include <fontconfig/fontconfig.h>
 #include <stdio.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
+#ifdef _MSC_VER
+#include <io.h>
+#endif
 #include <stdlib.h>
 #include <string.h>
 #include <locale.h>
diff --git a/fc-list/fc-list.c b/fc-list/fc-list.c
index 2039acd36..0e56b1c32 100644
--- a/fc-list/fc-list.c
+++ b/fc-list/fc-list.c
@@ -24,7 +24,12 @@
 
 #include <fontconfig/fontconfig.h>
 #include <stdio.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
+#ifdef _MSC_VER
+#include <io.h>
+#endif
 #include <stdlib.h>
 #include <string.h>
 #include <locale.h>
diff --git a/fc-match/fc-match.c b/fc-match/fc-match.c
index dee61479d..da8d820aa 100644
--- a/fc-match/fc-match.c
+++ b/fc-match/fc-match.c
@@ -33,7 +33,12 @@
 
 #include <fontconfig/fontconfig.h>
 #include <stdio.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
+#ifdef _MSC_VER
+#include <io.h>
+#endif
 #include <stdlib.h>
 #include <string.h>
 #include <locale.h>
diff --git a/fc-pattern/fc-pattern.c b/fc-pattern/fc-pattern.c
index 7989b816f..15b43e098 100644
--- a/fc-pattern/fc-pattern.c
+++ b/fc-pattern/fc-pattern.c
@@ -33,7 +33,12 @@
 
 #include <fontconfig/fontconfig.h>
 #include <stdio.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
+#ifdef _MSC_VER
+#include <io.h>
+#endif
 #include <stdlib.h>
 #include <string.h>
 #include <locale.h>
diff --git a/fc-query/fc-query.c b/fc-query/fc-query.c
index fbffb8415..58c251fe3 100644
--- a/fc-query/fc-query.c
+++ b/fc-query/fc-query.c
@@ -36,7 +36,12 @@
 #include <fontconfig/fontconfig.h>
 #include <fontconfig/fcfreetype.h>
 #include <stdio.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
+#ifdef _MSC_VER
+#include <io.h>
+#endif
 #include <stdlib.h>
 #include <string.h>
 #include <locale.h>
diff --git a/fc-scan/fc-scan.c b/fc-scan/fc-scan.c
index 9302ac559..3a0ccf188 100644
--- a/fc-scan/fc-scan.c
+++ b/fc-scan/fc-scan.c
@@ -36,7 +36,12 @@
 #include <fontconfig/fontconfig.h>
 #include <fontconfig/fcfreetype.h>
 #include <stdio.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
+#ifdef _MSC_VER
+#include <io.h>
+#endif
 #include <stdlib.h>
 #include <string.h>
 #include <locale.h>
diff --git a/fc-validate/fc-validate.c b/fc-validate/fc-validate.c
index 2ceee2024..8f702f190 100644
--- a/fc-validate/fc-validate.c
+++ b/fc-validate/fc-validate.c
@@ -36,7 +36,12 @@
 #include <fontconfig/fontconfig.h>
 #include <fontconfig/fcfreetype.h>
 #include <stdio.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
+#ifdef _MSC_VER
+#include <io.h>
+#endif
 #include <stdlib.h>
 #include <string.h>
 #include <locale.h>
diff --git a/src/fcatomic.c b/src/fcatomic.c
index d12d32408..da7234934 100644
--- a/src/fcatomic.c
+++ b/src/fcatomic.c
@@ -50,7 +50,9 @@
 #include "fcint.h"
 #include <sys/types.h>
 #include <sys/stat.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <stdlib.h>
 #include <time.h>
 
diff --git a/src/fccache.c b/src/fccache.c
index 2b604014e..50fb742c6 100644
--- a/src/fccache.c
+++ b/src/fccache.c
@@ -30,7 +30,9 @@
 #include <limits.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#ifndef _MSC_VER
 #include <sys/time.h>
+#endif
 #include <assert.h>
 #if defined(HAVE_MMAP) || defined(__CYGWIN__)
 #  include <unistd.h>
diff --git a/src/fcint.h b/src/fcint.h
index de78cd8c9..4fb269ffa 100644
--- a/src/fcint.h
+++ b/src/fcint.h
@@ -40,7 +40,18 @@
 #include <limits.h>
 #include <float.h>
 #include <math.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
+#ifdef _MSC_VER
+#include <direct.h>
+#include <io.h>
+#include <BaseTsd.h>
+#define ssize_t SSIZE_T
+#define F_OK 0
+#define W_OK 2
+typedef int mode_t;
+#endif
 #include <stddef.h>
 #include <sys/types.h>
 #include <sys/stat.h>
diff --git a/test/test-migration.c b/test/test-migration.c
index b7d96dcce..7eac59dcc 100644
--- a/test/test-migration.c
+++ b/test/test-migration.c
@@ -28,13 +28,16 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
+#ifdef _MSC_VER
+#include <io.h>
+#endif
 #include <sys/types.h>
 #include <dirent.h>
 #ifndef HAVE_STRUCT_DIRENT_D_TYPE
-#include <sys/types.h>
 #include <sys/stat.h>
-#include <unistd.h>
 #endif
 #include <fontconfig/fontconfig.h>
 
diff --git a/test/test-pthread.c b/test/test-pthread.c
index fbf397d4c..38d7dea35 100644
--- a/test/test-pthread.c
+++ b/test/test-pthread.c
@@ -24,7 +24,12 @@
  */
 #include <stdio.h>
 #include <stdlib.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
+#ifdef _MSC_VER
+#include <io.h>
+#endif
 #include <pthread.h>
 #include <fontconfig/fontconfig.h>
 
