--- 2.14.2-7f52ee670b.clean/src/meson.build	2023-11-10 05:00:12.930009000 +0900
+++ 2.14.2-7f52ee670b/src/meson.build	2023-11-10 04:52:40.375965000 +0900
@@ -85,6 +85,7 @@
   name: 'Fontconfig',
   requires: ['freetype2 ' + freetype_req],
   version: fc_version,
+  install_dir: join_paths(get_option('prefix'), get_option('libdir'), 'pkgconfig'),
   variables: [
     'sysconfdir=@0@'.format(join_paths(prefix, get_option('sysconfdir'))),
     'localstatedir=@0@'.format(join_paths(prefix, get_option('localstatedir'))),
