diff --git a/fc-cache/meson.build b/fc-cache/meson.build
index 1d66ccf..4745d94 100644
--- a/fc-cache/meson.build
+++ b/fc-cache/meson.build
@@ -1,6 +1,6 @@
 fccache = executable('fc-cache', ['fc-cache.c', fcstdint_h, alias_headers, ft_alias_headers],
   include_directories: [incbase, incsrc],
-  dependencies: [libintl_dep],
+  dependencies: [libintl_dep, getopt_dep],
   link_with: [libfontconfig],
   c_args: c_args,
   install: true,
diff --git a/fc-cat/meson.build b/fc-cat/meson.build
index dd7ee29..15da98c 100644
--- a/fc-cat/meson.build
+++ b/fc-cat/meson.build
@@ -1,6 +1,6 @@
 fccat = executable('fc-cat', ['fc-cat.c', fcstdint_h, alias_headers, ft_alias_headers],
   include_directories: [incbase, incsrc],
-  dependencies: [libintl_dep],
+  dependencies: [libintl_dep, getopt_dep],
   link_with: [libfontconfig],
   c_args: c_args,
   install: true,
diff --git a/fc-conflist/meson.build b/fc-conflist/meson.build
index 88208ff..7fcf4f5 100644
--- a/fc-conflist/meson.build
+++ b/fc-conflist/meson.build
@@ -1,6 +1,6 @@
 fcconflist = executable('fc-conflist', ['fc-conflist.c', fcstdint_h, alias_headers, ft_alias_headers],
   include_directories: [incbase, incsrc],
-  dependencies: [libintl_dep],
+  dependencies: [libintl_dep, getopt_dep],
   link_with: [libfontconfig],
   c_args: c_args,
   install: true,
diff --git a/fc-list/meson.build b/fc-list/meson.build
index 1be9df5..cfff2ec 100644
--- a/fc-list/meson.build
+++ b/fc-list/meson.build
@@ -1,6 +1,6 @@
 fclist = executable('fc-list', ['fc-list.c', fcstdint_h, alias_headers, ft_alias_headers],
   include_directories: [incbase, incsrc],
-  dependencies: [libintl_dep],
+  dependencies: [libintl_dep, getopt_dep],
   link_with: [libfontconfig],
   c_args: c_args,
   install: true,
diff --git a/fc-match/meson.build b/fc-match/meson.build
index 1aa1b1b..5b05c46 100644
--- a/fc-match/meson.build
+++ b/fc-match/meson.build
@@ -1,6 +1,6 @@
 fcmatch = executable('fc-match', ['fc-match.c', fcstdint_h, alias_headers, ft_alias_headers],
   include_directories: [incbase, incsrc],
-  dependencies: [libintl_dep],
+  dependencies: [libintl_dep, getopt_dep],
   link_with: [libfontconfig],
   c_args: c_args,
   install: true,
diff --git a/fc-pattern/meson.build b/fc-pattern/meson.build
index d9b048c..c2c6224 100644
--- a/fc-pattern/meson.build
+++ b/fc-pattern/meson.build
@@ -1,6 +1,6 @@
 fcpattern = executable('fc-pattern', ['fc-pattern.c', fcstdint_h, alias_headers, ft_alias_headers],
   include_directories: [incbase, incsrc],
-  dependencies: [libintl_dep],
+  dependencies: [libintl_dep, getopt_dep],
   link_with: [libfontconfig],
   c_args: c_args,
   install: true,
diff --git a/fc-query/meson.build b/fc-query/meson.build
index 629bc71..38c74c7 100644
--- a/fc-query/meson.build
+++ b/fc-query/meson.build
@@ -1,6 +1,6 @@
 fcquery = executable('fc-query', ['fc-query.c', fcstdint_h, alias_headers, ft_alias_headers],
   include_directories: [incbase, incsrc],
-  dependencies: [freetype_dep, libintl_dep],
+  dependencies: [freetype_dep, libintl_dep, getopt_dep],
   link_with: [libfontconfig],
   c_args: c_args,
   install: true,
diff --git a/fc-scan/meson.build b/fc-scan/meson.build
index 9049578..70bb5fe 100644
--- a/fc-scan/meson.build
+++ b/fc-scan/meson.build
@@ -1,6 +1,6 @@
 fcscan = executable('fc-scan', ['fc-scan.c', fcstdint_h, alias_headers, ft_alias_headers],
   include_directories: [incbase, incsrc],
-  dependencies: [freetype_dep, libintl_dep],
+  dependencies: [freetype_dep, libintl_dep, getopt_dep],
   link_with: [libfontconfig],
   c_args: c_args,
   install: true,
diff --git a/fc-validate/meson.build b/fc-validate/meson.build
index fcabf29..0d80df8 100644
--- a/fc-validate/meson.build
+++ b/fc-validate/meson.build
@@ -1,6 +1,6 @@
 fcvalidate = executable('fc-validate', ['fc-validate.c', fcstdint_h, alias_headers, ft_alias_headers],
   include_directories: [incbase, incsrc],
-  dependencies: [freetype_dep, libintl_dep],
+  dependencies: [freetype_dep, libintl_dep, getopt_dep],
   link_with: [libfontconfig],
   c_args: c_args,
   install: true,
diff --git a/meson.build b/meson.build
index e742ea6..20d55bd 100644
--- a/meson.build
+++ b/meson.build
@@ -319,6 +319,13 @@ if cc.links(files('meson-cc-tests/solaris-atomic-operations.c'), name: 'Solaris
   conf.set('HAVE_SOLARIS_ATOMIC_OPS', 1)
 endif
 
+if host_machine.system() == 'windows'
+  conf.set('HAVE_GETOPT', 1)
+  conf.set('HAVE_GETOPT_LONG', 1)
+  getopt_dep = cc.find_library('getopt', required: false)
+else
+  getopt_dep = dependency('', required: false)
+endif
 
 prefix = get_option('prefix')
 
