diff --git a/common/math/CMakeLists.txt b/common/math/CMakeLists.txt
index fcfa455..c5184cd 100644
--- a/common/math/CMakeLists.txt
+++ b/common/math/CMakeLists.txt
@@ -2,6 +2,11 @@
 ## SPDX-License-Identifier: Apache-2.0
 
 ADD_LIBRARY(math STATIC constants.cpp)
+SET_TARGET_PROPERTIES(math PROPERTIES # avoid conflict with vcpkg port openmvs
+  OUTPUT_NAME embree-math
+  ARCHIVE_OUTPUT_NAME embree-math
+  RUNTIME_OUTPUT_NAME embree-math
+)
 SET_PROPERTY(TARGET math PROPERTY FOLDER common)
 SET_PROPERTY(TARGET math APPEND PROPERTY COMPILE_FLAGS " ${FLAGS_LOWEST}")
 
diff --git a/common/simd/CMakeLists.txt b/common/simd/CMakeLists.txt
index 989a00d..043fec2 100644
--- a/common/simd/CMakeLists.txt
+++ b/common/simd/CMakeLists.txt
@@ -2,6 +2,11 @@
 ## SPDX-License-Identifier: Apache-2.0
 
 ADD_LIBRARY(simd STATIC sse.cpp)
+SET_TARGET_PROPERTIES(simd PROPERTIES # avoid conflict with vcpkg port simd
+  OUTPUT_NAME embree-simd
+  ARCHIVE_OUTPUT_NAME embree-simd
+  RUNTIME_OUTPUT_NAME embree-simd
+)
 SET_PROPERTY(TARGET simd PROPERTY FOLDER common)
 SET_PROPERTY(TARGET simd APPEND PROPERTY COMPILE_FLAGS " ${FLAGS_LOWEST}")
 
