diff --git a/common/cmake/embree-config.cmake b/common/cmake/embree-config.cmake
index 5087ca0..fdf8135 100644
--- a/common/cmake/embree-config.cmake
+++ b/common/cmake/embree-config.cmake
@@ -52,6 +52,21 @@ IF (EMBREE_STATIC_LIB)
   INCLUDE("${EMBREE_ROOT_DIR}/@EMBREE_CMAKEEXPORT_DIR@/tasking-targets.cmake")
 
   
+  IF(EMBREE_ISA_SSE42)
+    INCLUDE("${EMBREE_ROOT_DIR}/@EMBREE_CMAKEEXPORT_DIR@/embree_sse42-targets.cmake")
+  ENDIF()
+  IF(EMBREE_ISA_AVX)
+      INCLUDE("${EMBREE_ROOT_DIR}/@EMBREE_CMAKEEXPORT_DIR@/embree_avx-targets.cmake")
+  ENDIF()
+  IF(EMBREE_ISA_AVX2)
+      INCLUDE("${EMBREE_ROOT_DIR}/@EMBREE_CMAKEEXPORT_DIR@/embree_avx2-targets.cmake")
+  ENDIF()
+  IF(EMBREE_ISA_AVX512KNL)
+      INCLUDE("${EMBREE_ROOT_DIR}/@EMBREE_CMAKEEXPORT_DIR@/embree_avx512knl-targets.cmake")
+  ENDIF()
+  IF(EMBREE_ISA_AVX512SKX)
+      INCLUDE("${EMBREE_ROOT_DIR}/@EMBREE_CMAKEEXPORT_DIR@/embree_avx512skx-targets.cmake")
+  ENDIF()
 ENDIF()
 
 INCLUDE("${EMBREE_ROOT_DIR}/@EMBREE_CMAKEEXPORT_DIR@/embree-targets.cmake")
