diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1310f91..73aaf45 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -166,15 +166,15 @@ endif()
 if(1)
     # Install into the system (/usr/bin or /usr/local/bin)
     set(INSTALL_ROOT      "share/${project}")       # /usr/[local]/share/<project>
-    set(INSTALL_CMAKE     "share/${project}/cmake") # /usr/[local]/share/<project>/cmake
+    set(INSTALL_CMAKE     "share/${project}")       # /usr/[local]/share/<project>
     set(INSTALL_EXAMPLES  "share/${project}")       # /usr/[local]/share/<project>
     set(INSTALL_DATA      "share/${project}")       # /usr/[local]/share/<project>
     set(INSTALL_BIN       "bin")                    # /usr/[local]/bin
     set(INSTALL_SHARED    "lib")                    # /usr/[local]/lib
     set(INSTALL_LIB       "lib")                    # /usr/[local]/lib
     set(INSTALL_INCLUDE   "include")                # /usr/[local]/include
-    set(INSTALL_DOC       "share/doc/${project}")   # /usr/[local]/share/doc/<project>
-    set(INSTALL_SHORTCUTS "share/applications")     # /usr/[local]/share/applications
+    set(INSTALL_DOC       "share/${project}/doc")   # /usr/[local]/share/<project>/doc
+    set(INSTALL_SHORTCUTS "share/${project}/applications")     # /usr/[local]/share/${project}/applications
     set(INSTALL_ICONS     "share/pixmaps")          # /usr/[local]/share/pixmaps
     set(INSTALL_INIT      "etc/init")               # etc/init (upstart init scripts)
 else()
diff --git a/source/3rdparty/KHR/CMakeLists.txt b/source/3rdparty/KHR/CMakeLists.txt
index 869066a..d041df4 100644
--- a/source/3rdparty/KHR/CMakeLists.txt
+++ b/source/3rdparty/KHR/CMakeLists.txt
@@ -43,7 +43,7 @@ add_library(${target} INTERFACE)
 add_library(externals::${target} ALIAS ${target})
 
 # Export library for downstream projects
-export(TARGETS ${target} NAMESPACE ${META_PROJECT_NAME}:: FILE ${PROJECT_BINARY_DIR}/cmake/${target}/${target}-export.cmake)
+export(TARGETS ${target} NAMESPACE ${META_PROJECT_NAME}:: FILE ${PROJECT_BINARY_DIR}/${target}-export.cmake)
 
 
 #
diff --git a/source/glbinding-aux/CMakeLists.txt b/source/glbinding-aux/CMakeLists.txt
index c72cd98..7161c84 100644
--- a/source/glbinding-aux/CMakeLists.txt
+++ b/source/glbinding-aux/CMakeLists.txt
@@ -149,7 +149,7 @@ endif()
 add_library(${META_PROJECT_NAME}::${target} ALIAS ${target})
 
 # Export library for downstream projects
-export(TARGETS ${target} NAMESPACE ${META_PROJECT_NAME}:: FILE ${PROJECT_BINARY_DIR}/cmake/${target}/${target}-export.cmake)
+export(TARGETS ${target} NAMESPACE ${META_PROJECT_NAME}:: FILE ${PROJECT_BINARY_DIR}/${target}-export.cmake)
 
 # Create API export header
 generate_export_header(${target}
diff --git a/source/glbinding/CMakeLists.txt b/source/glbinding/CMakeLists.txt
index b7313f4..13aa724 100644
--- a/source/glbinding/CMakeLists.txt
+++ b/source/glbinding/CMakeLists.txt
@@ -191,7 +191,7 @@ endif()
 add_library(${META_PROJECT_NAME}::${target} ALIAS ${target})
 
 # Export library for downstream projects
-export(TARGETS ${target} NAMESPACE ${META_PROJECT_NAME}:: FILE ${PROJECT_BINARY_DIR}/cmake/${target}/${target}-export.cmake)
+export(TARGETS ${target} NAMESPACE ${META_PROJECT_NAME}:: FILE ${PROJECT_BINARY_DIR}/${target}-export.cmake)
 
 # Create API export header
 generate_export_header(${target}
