diff --git a/glbinding-config.cmake b/glbinding-config.cmake
index 07ef7cf1..2e100b50 100644
--- a/glbinding-config.cmake
+++ b/glbinding-config.cmake
@@ -44,20 +44,11 @@ endmacro()
 
 # Try install location
 set(MODULE_FOUND FALSE)
-find_modules("cmake")
+find_modules(".")
 
 if(MODULE_FOUND)
     return()
 endif()
 
-# Try common build locations
-if("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
-    find_modules("build-debug/cmake")
-    find_modules("build/cmake")
-else()
-    find_modules("build/cmake")
-    find_modules("build-debug/cmake")
-endif()
-
 # Signal success/failure to CMake
 set(glbinding_FOUND ${MODULE_FOUND})
