diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7f8572e..23f66c2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,6 +2,8 @@ cmake_minimum_required(VERSION 3.0)
 
 project( enkiTS )
 
+include(GNUInstallDirs)
+
 option( ENKITS_BUILD_C_INTERFACE    "Build C interface" ON )
 option( ENKITS_BUILD_EXAMPLES       "Build example applications" ON )
 option( ENKITS_BUILD_SHARED         "Build shared library" OFF )
@@ -60,8 +62,15 @@ if( UNIX )
 endif()
 
 if( ENKITS_INSTALL )
-    install(TARGETS enkiTS DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/enkiTS")
-    install(FILES ${ENKITS_HEADERS} DESTINATION "${CMAKE_INSTALL_PREFIX}/include/enkiTS")
+    install(
+        TARGETS enkiTS
+        EXPORT enkiTS-config
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+    install(FILES ${ENKITS_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+    install(
+        EXPORT enkiTS-config
+        NAMESPACE enkiTS::
+        DESTINATION ${DATADIR}/enkiTS)
 endif()
 
 if( UNIX )
