diff --git a/jsoncpp-namespaced-targets.cmake b/jsoncpp-namespaced-targets.cmake
index ac1504e..144f086 100644
--- a/jsoncpp-namespaced-targets.cmake
+++ b/jsoncpp-namespaced-targets.cmake
@@ -1,7 +1,9 @@
+if (NOT TARGET JsonCpp::JsonCpp)
 if (TARGET jsoncpp_static)
     add_library(JsonCpp::JsonCpp INTERFACE IMPORTED)
     set_target_properties(JsonCpp::JsonCpp PROPERTIES INTERFACE_LINK_LIBRARIES "jsoncpp_static")
 elseif (TARGET jsoncpp_lib)
     add_library(JsonCpp::JsonCpp INTERFACE IMPORTED)
     set_target_properties(JsonCpp::JsonCpp PROPERTIES INTERFACE_LINK_LIBRARIES "jsoncpp_lib")
+endif ()
 endif ()
\ No newline at end of file
