 llvm/cmake/config-ix.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/llvm/cmake/config-ix.cmake b/llvm/cmake/config-ix.cmake
index dd0aaadb47c7..279435b5b27d 100644
--- a/llvm/cmake/config-ix.cmake
+++ b/llvm/cmake/config-ix.cmake
@@ -343,7 +343,7 @@ if( LLVM_ENABLE_FFI )
     message(FATAL_ERROR "libffi includes are not found.")
   endif()
 
-  find_library(FFI_LIBRARY_PATH ffi PATHS ${FFI_LIBRARY_DIR})
+  find_library(FFI_LIBRARY_PATH NAMES ffi libffi PATHS ${FFI_LIBRARY_DIR})
   if( NOT FFI_LIBRARY_PATH )
     message(FATAL_ERROR "libffi is not found.")
   endif()
