diff --git a/projects/openmp/runtime/src/kmp_debugger.cpp b/projects/openmp/runtime/src/kmp_debugger.cpp
index 3484f7a..6f84d7f 100644
--- a/projects/openmp/runtime/src/kmp_debugger.cpp
+++ b/projects/openmp/runtime/src/kmp_debugger.cpp
@@ -1,4 +1,3 @@
-#if USE_DEBUGGER
 /*
  * kmp_debugger.cpp -- debugger support.
  */
@@ -18,7 +17,7 @@
 #include "kmp_str.h"
 
 // NOTE: All variable names are known to the debugger, do not change!
-
+#if USE_DEBUGGER
 #ifdef __cplusplus
 extern "C" {
 extern kmp_omp_struct_info_t __kmp_omp_debug_struct_info;
