diff --git a/aten/src/ATen/native/cuda/CuFFTUtils.h b/aten/src/ATen/native/cuda/CuFFTUtils.h
index f20baa9..524cdc2 100644
--- a/aten/src/ATen/native/cuda/CuFFTUtils.h
+++ b/aten/src/ATen/native/cuda/CuFFTUtils.h
@@ -38,17 +38,21 @@ static inline std::string _cudaGetErrorEnum(cufftResult error)
       return "CUFFT_INVALID_SIZE";
     case CUFFT_UNALIGNED_DATA:
       return "CUFFT_UNALIGNED_DATA";
+#if CUFFT_VERSION < 12000
     case CUFFT_INCOMPLETE_PARAMETER_LIST:
       return "CUFFT_INCOMPLETE_PARAMETER_LIST";
+#endif
     case CUFFT_INVALID_DEVICE:
       return "CUFFT_INVALID_DEVICE";
+#if CUFFT_VERSION < 12000
     case CUFFT_PARSE_ERROR:
       return "CUFFT_PARSE_ERROR";
+#endif
     case CUFFT_NO_WORKSPACE:
       return "CUFFT_NO_WORKSPACE";
     case CUFFT_NOT_IMPLEMENTED:
       return "CUFFT_NOT_IMPLEMENTED";
-#if !defined(USE_ROCM)
+#if !defined(USE_ROCM) && CUFFT_VERSION < 12000
     case CUFFT_LICENSE_ERROR:
       return "CUFFT_LICENSE_ERROR";
 #endif
diff --git a/cmake/Modules/FindCUB.cmake b/cmake/Modules/FindCUB.cmake
index e053964..7fa044d 100644
--- a/cmake/Modules/FindCUB.cmake
+++ b/cmake/Modules/FindCUB.cmake
@@ -4,7 +4,7 @@
 
 find_path(CUB_INCLUDE_DIR
         HINTS "${CUDA_TOOLKIT_INCLUDE}"
-        NAMES cub/cub.cuh
+        NAMES cub/cub.cuh cccl/cub/cub.cuh
         DOC "The directory where CUB includes reside"
 )
 
diff --git a/cmake/public/cuda.cmake b/cmake/public/cuda.cmake
index c66d32b..266f86a 100644
--- a/cmake/public/cuda.cmake
+++ b/cmake/public/cuda.cmake
@@ -299,9 +299,15 @@ endif()
 # cufft
 add_library(caffe2::cufft INTERFACE IMPORTED)
 if(CAFFE2_STATIC_LINK_CUDA AND NOT WIN32)
-    set_property(
-        TARGET caffe2::cufft PROPERTY INTERFACE_LINK_LIBRARIES
-        CUDA::cufft_static_nocallback)
+    if(TARGET CUDA::cufft_static_nocallback)
+        set_property(
+            TARGET caffe2::cufft PROPERTY INTERFACE_LINK_LIBRARIES
+            CUDA::cufft_static_nocallback)
+    else()
+        set_property(
+            TARGET caffe2::cufft PROPERTY INTERFACE_LINK_LIBRARIES
+            CUDA::cufft_static)
+    endif()
 else()
     set_property(
         TARGET caffe2::cufft PROPERTY INTERFACE_LINK_LIBRARIES
diff --git a/aten/src/ATen/native/cuda/CuFFTUtils.h b/aten/src/ATen/native/cuda/CuFFTUtils.h
index f20baa9..524cdc2 100644
--- a/aten/src/ATen/native/cuda/CuFFTUtils.h
+++ b/aten/src/ATen/native/cuda/CuFFTUtils.h
@@ -38,17 +38,21 @@ static inline std::string _cudaGetErrorEnum(cufftResult error)
       return "CUFFT_INVALID_SIZE";
     case CUFFT_UNALIGNED_DATA:
       return "CUFFT_UNALIGNED_DATA";
+#if CUFFT_VERSION < 12000
     case CUFFT_INCOMPLETE_PARAMETER_LIST:
       return "CUFFT_INCOMPLETE_PARAMETER_LIST";
+#endif
     case CUFFT_INVALID_DEVICE:
       return "CUFFT_INVALID_DEVICE";
+#if CUFFT_VERSION < 12000
     case CUFFT_PARSE_ERROR:
       return "CUFFT_PARSE_ERROR";
+#endif
     case CUFFT_NO_WORKSPACE:
       return "CUFFT_NO_WORKSPACE";
     case CUFFT_NOT_IMPLEMENTED:
       return "CUFFT_NOT_IMPLEMENTED";
-#if !defined(USE_ROCM)
+#if !defined(USE_ROCM) && CUFFT_VERSION < 12000
     case CUFFT_LICENSE_ERROR:
       return "CUFFT_LICENSE_ERROR";
 #endif
diff --git a/cmake/public/cuda.cmake b/cmake/public/cuda.cmake
index c66d32b..266f86a 100644
--- a/cmake/public/cuda.cmake
+++ b/cmake/public/cuda.cmake
@@ -299,9 +299,15 @@ endif()
 # cufft
 add_library(caffe2::cufft INTERFACE IMPORTED)
 if(CAFFE2_STATIC_LINK_CUDA AND NOT WIN32)
-    set_property(
-        TARGET caffe2::cufft PROPERTY INTERFACE_LINK_LIBRARIES
-        CUDA::cufft_static_nocallback)
+    if(TARGET CUDA::cufft_static_nocallback)
+        set_property(
+            TARGET caffe2::cufft PROPERTY INTERFACE_LINK_LIBRARIES
+            CUDA::cufft_static_nocallback)
+    else()
+        set_property(
+            TARGET caffe2::cufft PROPERTY INTERFACE_LINK_LIBRARIES
+            CUDA::cufft_static)
+    endif()
 else()
     set_property(
         TARGET caffe2::cufft PROPERTY INTERFACE_LINK_LIBRARIES
