--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -449,4 +449,7 @@ if(NOT CMAKE_SYSTEM_PROCESSOR MATCHES "^(s390x|ppc64le)$")
   # ---[ Caffe2 uses cpuinfo library in the thread pool
   # ---[ But it doesn't support s390x/powerpc and thus not used on s390x/powerpc
+  find_package(cpuinfo CONFIG REQUIRED)
+  add_library(cpuinfo ALIAS cpuinfo::cpuinfo)
+  add_library(clog ALIAS cpuinfo::clog)
   if(NOT TARGET cpuinfo AND USE_SYSTEM_CPUINFO)
     add_library(cpuinfo SHARED IMPORTED)
