--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -519,7 +519,9 @@ endif()

 # ---[ NNPACK
 if(USE_NNPACK)
-  include(${CMAKE_CURRENT_LIST_DIR}/External/nnpack.cmake)
+  find_package(unofficial-nnpack CONFIG REQUIRED)
+  add_library(nnpack ALIAS unofficial::nnpack::nnpack)
+  set(NNPACK_FOUND TRUE)
   if(NNPACK_FOUND)
     if(TARGET nnpack)
       # ---[ NNPACK is being built together with Caffe2: explicitly specify dependency
