--- a/caffe2/CMakeLists.txt
+++ b/caffe2/CMakeLists.txt
@@ -1334,8 +1334,8 @@
 target_include_directories(torch_cpu PRIVATE
   ${TORCH_ROOT}/third_party/miniz-3.0.2)
 
-target_include_directories(torch_cpu PRIVATE
-  ${TORCH_ROOT}/third_party/kineto/libkineto/include)
+find_package(unofficial-kineto CONFIG REQUIRED)
+target_link_libraries(torch_cpu PRIVATE unofficial::kineto::headers)
 
 if(USE_KINETO)
   target_include_directories(torch_cpu PRIVATE
