--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -530,3 +530,9 @@ endif()

 # ---[ XNNPACK
+if(USE_XNNPACK)
+  find_package(unofficial-xnnpack CONFIG REQUIRED)
+  add_library(XNNPACK ALIAS unofficial::xnnpack::XNNPACK)
+  add_library(microkernels-prod ALIAS unofficial::xnnpack::microkernels-prod)
+  list(APPEND Caffe2_DEPENDENCY_LIBS XNNPACK microkernels-prod)
+endif()
 if(USE_XNNPACK AND NOT USE_SYSTEM_XNNPACK)
@@ -613,7 +619,7 @@ endif()
   include_directories(SYSTEM ${XNNPACK_INCLUDE_DIR})
   list(APPEND Caffe2_DEPENDENCY_LIBS XNNPACK microkernels-prod)
 elseif(NOT TARGET XNNPACK AND USE_SYSTEM_XNNPACK)
-  add_library(XNNPACK SHARED IMPORTED)
-  add_library(microkernels-prod SHARED IMPORTED)
+  add_library(XNNPACK STATIC IMPORTED)
+  add_library(microkernels-prod STATIC IMPORTED)
   find_library(XNNPACK_LIBRARY XNNPACK)
   find_library(microkernels-prod_LIBRARY microkernels-prod)
