diff --git a/aten/src/ATen/core/dynamic_type.h b/aten/src/ATen/core/dynamic_type.h
index a84644ddde0..b1b6b2c9ecc 100644
--- a/aten/src/ATen/core/dynamic_type.h
+++ b/aten/src/ATen/core/dynamic_type.h
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <cstdint>
 #include <memory>
 #include <type_traits>
 
diff --git a/c10/util/BFloat16.h b/c10/util/BFloat16.h
index 1ada02bba1c..f6f9fc65e0e 100644
--- a/c10/util/BFloat16.h
+++ b/c10/util/BFloat16.h
@@ -3,6 +3,7 @@
 // Defines the bloat16 type (brain floating-point). This representation uses
 // 1 bit for the sign, 8 bits for the exponent and 7 bits for the mantissa.
 
+#include <cstdint>
 #include <c10/macros/Macros.h>
 #include <cmath>
 #include <cstring>
diff --git a/torch/csrc/jit/passes/quantization/quantization_type.h b/torch/csrc/jit/passes/quantization/quantization_type.h
index ea5ca10b15a..c704f9c3037 100644
--- a/torch/csrc/jit/passes/quantization/quantization_type.h
+++ b/torch/csrc/jit/passes/quantization/quantization_type.h
@@ -1,5 +1,6 @@
 #pragma once
 #include <ostream>
+#include <cstdint>
 
 namespace torch {
 namespace jit {
diff --git a/torch/csrc/jit/runtime/logging.cpp b/torch/csrc/jit/runtime/logging.cpp
index 66712990368..b326f587bc8 100644
--- a/torch/csrc/jit/runtime/logging.cpp
+++ b/torch/csrc/jit/runtime/logging.cpp
@@ -1,6 +1,7 @@
 #include <torch/csrc/jit/runtime/logging.h>
 
 #include <atomic>
+#include <stdexcept>
 #include <mutex>
 #include <unordered_map>
 
diff --git a/torch/csrc/lazy/core/multi_wait.cpp b/torch/csrc/lazy/core/multi_wait.cpp
index 6b9933518e0..db0742b32a3 100644
--- a/torch/csrc/lazy/core/multi_wait.cpp
+++ b/torch/csrc/lazy/core/multi_wait.cpp
@@ -1,5 +1,6 @@
 #include <torch/csrc/lazy/core/multi_wait.h>
 
+#include <stdexcept>
 #include <chrono>
 #include <exception>
 
