diff --git a/Wrapping/Tools/CMakeLists.txt b/Wrapping/Tools/CMakeLists.txt
index 05aa805f..911a032f 100644
--- a/Wrapping/Tools/CMakeLists.txt
+++ b/Wrapping/Tools/CMakeLists.txt
@@ -46,6 +46,7 @@ set_property(SOURCE vtkParseMain.c
 set(_vtk_module_no_namespace_abi_mangling ON)
 
 vtk_module_add_module("${ns}::WrappingTools"
+  LIBRARY_NAME_SUFFIX "_CompileTools"  # distinct from port vtk
   SOURCES ${sources}
   HEADERS ${headers})
 
