diff --git a/meson.build b/meson.build
index 9fe9cd5ac..19a19e5c6 100644
--- a/meson.build
+++ b/meson.build
@@ -792,16 +792,16 @@ subdir('docs/tools')
 subdir('docs/reference')
 
 # Keep this in sync with post-install.sh expected arguments
-if not meson.is_cross_build()
-  meson.add_install_script('build-aux/meson/post-install.py',
-                           gtk_api_version,
-                           gtk_binary_version,
-                           gtk_libdir,
-                           gtk_datadir,
-                           gtk_bindir)
-else
-  message('Not executing post-install steps automatically when cross compiling')
-endif
+#if not meson.is_cross_build()
+#  meson.add_install_script('build-aux/meson/post-install.py',
+#                           gtk_api_version,
+#                           gtk_binary_version,
+#                           gtk_libdir,
+#                           gtk_datadir,
+#                           gtk_bindir)
+#else
+#  message('Not executing post-install steps automatically when cross compiling')
+#endif
 
 if host_machine.system() != 'windows'
   # Install Valgrind suppression files (except on Windows,
