diff --git a/Source/FreeImage/PluginRAW.cpp b/Source/FreeImage/PluginRAW.cpp
index c7f8758a..a57fd5f1 100644
--- a/Source/FreeImage/PluginRAW.cpp
+++ b/Source/FreeImage/PluginRAW.cpp
@@ -46,6 +46,10 @@ private:
 	long _eof;
 	INT64 _fsize;
 
+#if LIBRAW_COMPILE_CHECK_VERSION_NOTLESS(0, 21)
+	LibRaw_abstract_datastream *substream = nullptr; // polyfill
+#endif
+
 public:
 	LibRaw_freeimage_datastream(FreeImageIO *io, fi_handle handle) : _io(io), _handle(handle) {
 		long start_pos = io->tell_proc(handle);
@@ -694,7 +698,11 @@ Load(FreeImageIO *io, fi_handle handle, int page, int flags, void *data) {
 		// --------------------------------------------
 
 		// (-s [0..N-1]) Select one raw image from input file
+#if LIBRAW_COMPILE_CHECK_VERSION_NOTLESS(0, 21)
+		RawProcessor->imgdata.rawparams.shot_select = 0;
+#else
 		RawProcessor->imgdata.params.shot_select = 0;
+#endif
 		// (-w) Use camera white balance, if possible (otherwise, fallback to auto_wb)
 		RawProcessor->imgdata.params.use_camera_wb = 1;
 		// (-M) Use any color matrix from the camera metadata. This option only affects Olympus, Leaf, and Phase One cameras.
