diff --git a/ThirdParty/protobuf/CMakeLists.txt b/ThirdParty/protobuf/CMakeLists.txt
index 811dc3721b..02f26ae757 100644
--- a/ThirdParty/protobuf/CMakeLists.txt
+++ b/ThirdParty/protobuf/CMakeLists.txt
@@ -40,7 +40,7 @@ vtk_module_third_party(
     STANDARD_INCLUDE_DIRS
   EXTERNAL
     PACKAGE Protobuf
-    VERSION "3.4"
+    CONFIG_MODE
     TARGETS protobuf::libprotobuf
     STANDARD_INCLUDE_DIRS)
 
