diff --git a/Qt/Widgets/pqCollapsedGroup.cxx b/Qt/Widgets/pqCollapsedGroup.cxx
index e912315..90040f2 100644
--- a/Qt/Widgets/pqCollapsedGroup.cxx
+++ b/Qt/Widgets/pqCollapsedGroup.cxx
@@ -39,7 +39,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <QStyleOptionGroupBox>
 #include <QStylePainter>
 
-static QStyleOptionGroupBox pqCollapseGroupGetStyleOption(const pqCollapsedGroup* p)
+static QStyleOptionGroupBox pqCollapsedGroup::pqCollapseGroupGetStyleOption(const pqCollapsedGroup* p)
 {
   QStyleOptionGroupBox option;
   p->initStyleOption(&option);
diff --git a/Qt/Widgets/pqCollapsedGroup.h b/Qt/Widgets/pqCollapsedGroup.h
index 6373115..67017b2 100644
--- a/Qt/Widgets/pqCollapsedGroup.h
+++ b/Qt/Widgets/pqCollapsedGroup.h
@@ -61,6 +61,7 @@ protected:
   bool Pressed;
 
 private:
+  static QStyleOptionGroupBox pqCollapseGroupGetStyleOption(const pqCollapsedGroup* p);
   QRect textRect();
   QRect collapseRect();
 };
diff --git a/Qt/Widgets/pqConsoleWidget.cxx b/Qt/Widgets/pqConsoleWidget.cxx
index 7de9cb7..f55e47b 100644
--- a/Qt/Widgets/pqConsoleWidget.cxx
+++ b/Qt/Widgets/pqConsoleWidget.cxx
@@ -436,7 +436,7 @@ pqConsoleWidget::pqConsoleWidget(QWidget* Parent)
   , FontSize(12)
 {
   QVBoxLayout* const l = new QVBoxLayout(this);
-  l->setMargin(0);
+  l->setContentsMargins(0,0,0,0);
   l->addWidget(this->Implementation);
 }
 
diff --git a/Qt/Widgets/pqDoubleSliderWidget.cxx b/Qt/Widgets/pqDoubleSliderWidget.cxx
index cd1e7ea..41f702b 100644
--- a/Qt/Widgets/pqDoubleSliderWidget.cxx
+++ b/Qt/Widgets/pqDoubleSliderWidget.cxx
@@ -30,7 +30,7 @@ pqDoubleSliderWidget::pqDoubleSliderWidget(QWidget* parent)
   this->DeferredValueEdited = false;
 
   QHBoxLayout* l = new QHBoxLayout(this);
-  l->setMargin(0);
+  l->setContentsMargins(0,0,0,0);
   this->Slider = new QSlider(Qt::Horizontal, this);
   this->Slider->setRange(0, 100);
   l->addWidget(this->Slider);
diff --git a/Qt/Widgets/pqExpandableTableView.cxx b/Qt/Widgets/pqExpandableTableView.cxx
index 1ef7cd5..b9affa2 100644
--- a/Qt/Widgets/pqExpandableTableView.cxx
+++ b/Qt/Widgets/pqExpandableTableView.cxx
@@ -38,7 +38,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <QItemSelectionModel>
 #include <QKeyEvent>
 #include <QMimeData>
-#include <QRegExp>
+#include <QRegularExpression>
 
 //-----------------------------------------------------------------------------
 pqExpandableTableView::pqExpandableTableView(QWidget* parentObject)
@@ -133,13 +133,13 @@ void pqExpandableTableView::keyPressEvent(QKeyEvent* e)
       // Split the lines in the text
       QString text = mimeData->text();
       QStringList lines = text.split("\n", PV_QT_SKIP_EMPTY_PARTS);
-      for (int row = 0; row < std::min(lines.size(), numModelRows); ++row)
+      for (int row = 0; row < (std::min)(lines.size(), numModelRows); ++row)
       {
         // Split within each line
-        QStringList items = lines[row].split(QRegExp("\\s+"));
+        QStringList items = lines[row].split(QRegularExpression("\\s+"));
 
         // Set the data in the table
-        for (int column = 0; column < std::min(items.size(), numModelColumns); ++column)
+        for (int column = 0; column < (std::min)(items.size(), numModelColumns); ++column)
         {
           QVariant value(items[column]);
           QModelIndex index = this->model()->index(row, column);
diff --git a/Qt/Widgets/pqFlatTreeView.cxx b/Qt/Widgets/pqFlatTreeView.cxx
index 90a2139..927d544 100644
--- a/Qt/Widgets/pqFlatTreeView.cxx
+++ b/Qt/Widgets/pqFlatTreeView.cxx
@@ -2649,7 +2649,7 @@ void pqFlatTreeView::paintEvent(QPaintEvent* e)
 QStyleOptionViewItem pqFlatTreeView::getViewOptions() const
 {
   QStyleOptionViewItem option;
-  option.init(this);
+  this->initStyleOption(option);
   option.font = this->font();
   option.state &= ~QStyle::State_HasFocus;
   int iconSize = this->getIconSize();
