diff --git a/configure.ac b/configure.ac
index 14146ec..ce926c9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -16,6 +16,8 @@ All Rights Reserved.
 This file is part of the open source package Coin which is distributed
 under the Eclipse Public License.])
 
+AC_CONFIG_MACRO_DIR([m4])
+
 # List one file in the package so that the configure script can test
 # whether the package is actually there.
 AC_CONFIG_SRCDIR(src/CbcTree.hpp)
 
