--- a/src/CMakeLists.txt	2017-10-02 14:06:57.163881000 +0000
+++ b/src/CMakeLists.txt	2017-10-02 14:08:52.815977600 +0000
@@ -313,6 +313,27 @@
 ENDIF (WIN32 AND CMAKE_COMPILER_IS_MINGW)
 
 IF (WIN32 AND MSVC)
+    # install our libraries
+    IF (WANT_STATIC)
+        INSTALL(TARGETS libwildmidi_static DESTINATION ${WILDMIDILIB_INSTALLDIR})
+        IF (WANT_PLAYERSTATIC)
+            INSTALL(TARGETS wildmidi-static DESTINATION bin)
+        ENDIF ()
+    ENDIF (WANT_STATIC)
+
+    IF (BUILD_SHARED_LIBS)
+        INSTALL(TARGETS libwildmidi_dynamic
+            ARCHIVE DESTINATION ${WILDMIDILIB_INSTALLDIR}
+            LIBRARY DESTINATION ${WILDMIDILIB_INSTALLDIR}
+            RUNTIME DESTINATION ${WILDMIDIDLL_INSTALLDIR}
+        )
+        IF (WANT_PLAYER)
+            INSTALL(TARGETS wildmidi DESTINATION bin)
+        ENDIF ()
+    ENDIF ()
+
+    INSTALL(FILES ${CMAKE_SOURCE_DIR}/include/wildmidi_lib.h DESTINATION include)
+
     IF (WANT_MP_BUILD)
         SET(MT_BUILD "/MP")
     ENDIF ()
