diff --git a/Project/CMake/CMakeLists.txt b/Project/CMake/CMakeLists.txt
index 06b0d70..0d7b15f 100644
--- a/Project/CMake/CMakeLists.txt
+++ b/Project/CMake/CMakeLists.txt
@@ -34,11 +34,11 @@ set(LIB_INSTALL_DIR "${CMAKE_INSTALL_FULL_LIBDIR}" CACHE STRING "Directory where
 set(INCLUDE_INSTALL_DIR "${CMAKE_INSTALL_FULL_INCLUDEDIR}" CACHE PATH "The directory the headers are installed in")
 
 if(VCPKG_TARGET_TRIPLET)
-    set(ZenLib_CONFIG_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/share/zenlib")
+    set(ZenLib_CONFIG_INSTALL_DIR "share/zenlib")
 elseif(MSVC)
-    set(ZenLib_CONFIG_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/cmake")
+    set(ZenLib_CONFIG_INSTALL_DIR "cmake")
 else()
-    set(ZenLib_CONFIG_INSTALL_DIR "${LIB_INSTALL_DIR}/cmake/zenlib")
+    set(ZenLib_CONFIG_INSTALL_DIR "cmake/zenlib")
 endif()
 
 set(ZenLib_HDRS
