diff --git a/transcoder/basisu_containers_impl.h b/transcoder/basisu_containers_impl.h
index d4d3eb2..3d7aadd 100644
--- a/transcoder/basisu_containers_impl.h
+++ b/transcoder/basisu_containers_impl.h
@@ -1,6 +1,8 @@
 // basisu_containers_impl.h
 // Do not include directly
 
+#include <ctype.h>
+
 #ifdef _MSC_VER
 #pragma warning (disable:4127) // warning C4127: conditional expression is constant
 #endif
