diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7b1c0d52..62c4fb43 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -70,7 +70,7 @@ find_package( Threads REQUIRED )
 include("${CMAKE_CURRENT_LIST_DIR}/src/cmake/FindLibFuzzer.cmake")
 
 # Find expat for XML parsing
-find_package(EXPAT REQUIRED)
+find_package(expat CONFIG REQUIRED)
 if(TARGET EXPAT::EXPAT)
   set(EXPAT_LIBRARIES EXPAT::EXPAT)
 elseif(TARGET expat::expat)
