diff --git a/CMakeLists.txt b/CMakeLists.txt
index 498d0b5..22815a5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -67,7 +67,7 @@ find_package(APR-Util REQUIRED)
 find_package( Threads REQUIRED )
 
 # Find expat for XML parsing
-find_package(EXPAT REQUIRED)
+find_package(expat CONFIG REQUIRED)
 if(TARGET EXPAT::EXPAT)
   set(EXPAT_LIBRARIES EXPAT::EXPAT)
 elseif(TARGET expat::expat)
