diff --git a/CMakeLists.txt b/CMakeLists.txt
index e84e497..ff8b920 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -94,7 +94,7 @@ if(UNIX)
   )
 
   install(FILES "${CMAKE_CURRENT_BINARY_DIR}/liblog4cxx.pc" 
-    DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/pkgconfig)
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
 
   if(LOG4CXX_QT_SUPPORT)
       set(prefix "${CMAKE_INSTALL_PREFIX}")
@@ -107,7 +107,7 @@ if(UNIX)
       )
 
       install(FILES "${CMAKE_CURRENT_BINARY_DIR}/liblog4cxx-qt.pc"
-	DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/pkgconfig)
+	DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
   endif(LOG4CXX_QT_SUPPORT)
 endif(UNIX)
 
