diff --git a/src/3rdparty/mapbox-gl-native/mapbox-gl-native.pro b/src/3rdparty/mapbox-gl-native/mapbox-gl-native.pro
index ed974db..f17d45a 100644
--- a/src/3rdparty/mapbox-gl-native/mapbox-gl-native.pro
+++ b/src/3rdparty/mapbox-gl-native/mapbox-gl-native.pro
@@ -34,6 +34,9 @@ android|gcc {
 }
 
 darwin {
+    QMAKE_CXXFLAGS += \
+        -Wno-enum-constexpr-conversion
+
     LIBS += \
         -framework Foundation
 }
