diff --git a/src/3rdparty/mapbox-gl-native/include/mbgl/util/unique_any.hpp b/src/3rdparty/mapbox-gl-native/include/mbgl/util/unique_any.hpp
index c7dc8b3..0fb25b8 100644
--- a/src/3rdparty/mapbox-gl-native/include/mbgl/util/unique_any.hpp
+++ b/src/3rdparty/mapbox-gl-native/include/mbgl/util/unique_any.hpp
@@ -3,6 +3,7 @@
 #include <typeinfo>
 #include <type_traits>
 #include <stdexcept>
+#include <utility>
 namespace mbgl {
 namespace util {
 
