diff --git a/CMakeLists.txt b/CMakeLists.txt
index 22d9115..da6edaf 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -220,7 +220,7 @@ endif(HAVE_SPDLOG)
 
 set(HAVE_C-ARES NO)
 if (BUILD_C-ARES)
-    find_package(c-ares)
+    find_package(c-ares CONFIG)
     if(c-ares_FOUND)
       message(STATUS "c-ares found!")
       set(HAVE_C-ARES TRUE)
@@ -231,7 +231,7 @@ if(HAVE_C-ARES)
   if (NOT BUILD_SHARED_LIBS)
     target_compile_definitions(${PROJECT_NAME} PRIVATE CARES_STATICLIB)
   endif ()
-  target_link_libraries(${PROJECT_NAME} PRIVATE c-ares_lib)
+  target_link_libraries(${PROJECT_NAME} PRIVATE c-ares::cares)
   set(TRANTOR_SOURCES
       ${TRANTOR_SOURCES}
       trantor/net/inner/AresResolver.cc)
diff --git a/cmake/templates/TrantorConfig.cmake.in b/cmake/templates/TrantorConfig.cmake.in
index 61007bd..c5dd363 100644
--- a/cmake/templates/TrantorConfig.cmake.in
+++ b/cmake/templates/TrantorConfig.cmake.in
@@ -17,7 +17,7 @@ if(@Botan_FOUND@)
   find_dependency(Botan)
 endif()
 if(@c-ares_FOUND@)
-  find_dependency(c-ares)
+  find_dependency(c-ares CONFIG)
 endif()
 find_dependency(Threads)
 if(@spdlog_FOUND@)
