diff --git a/CMakeLists.txt b/CMakeLists.txt
index d3a1dbd..44f0953 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,11 +1,32 @@
-PROJECT(kcp)
 CMAKE_MINIMUM_REQUIRED(VERSION 2.6)
 
-add_library(kcp STATIC ikcp.c)
+project(kcp LANGUAGES C)
 
-add_executable(kcp_test test.cpp)
+include(CTest)
+include(GNUInstallDirs)
 
+add_library(kcp STATIC ikcp.c)
 
+install(FILES ikcp.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
+install(TARGETS kcp
+    EXPORT kcp-targets
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+)
 
+install(EXPORT kcp-targets
+    FILE kcp-config.cmake
+    NAMESPACE kcp::
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/kcp
+)
 
+if (BUILD_TESTING)
+    enable_language(CXX)
+    
+    add_executable(kcp_test test.cpp)
+    if(MSVC AND NOT (MSVC_VERSION LESS 1900))
+        target_compile_options(${PROJECT_NAME} PRIVATE /utf-8)
+    endif()
+    target_link_libraries(kcp_test PRIVATE kcp)
+endif ()
