diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4ba3843..6697118 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -450,7 +450,7 @@ install(FILES
         DESTINATION include)
 
 install(DIRECTORY tutorial
-        DESTINATION .
+        DESTINATION share/halide
         FILES_MATCHING
         PATTERN "*.cpp"
         PATTERN "*.h"
@@ -478,12 +478,12 @@ endforeach()
 # ---- README
 file(GLOB FILES "${HALIDE_BASE_DIR}/*.md")
 install(FILES ${FILES}
-        DESTINATION .)
+        DESTINATION share/halide)
 
 # ---- halide.cmake
 file(GLOB FILES "${HALIDE_BASE_DIR}/*.cmake")
 install(FILES ${FILES}
-        DESTINATION .)
+        DESTINATION share/halide)
 
 # ---- halide_config
 file(GLOB FILES "${HALIDE_BASE_DIR}/tools/halide_config.*.tpl")
@@ -492,7 +492,7 @@ foreach(F ${FILES})
   string(REGEX REPLACE "\\.tpl$" "" FNAME "${FNAME}")  # Strip .tpl extension
   configure_file("${F}" "${CMAKE_BINARY_DIR}/${FNAME}" @ONLY)
   install(FILES "${CMAKE_BINARY_DIR}/${FNAME}"
-          DESTINATION .)
+          DESTINATION share/halide)
 endforeach()
 
 add_custom_target(distrib
