diff --git a/src/hello_imgui/internal/poor_man_log.cpp b/src/hello_imgui/internal/poor_man_log.cpp
index 3aad6f7..e9e474b 100644
--- a/src/hello_imgui/internal/poor_man_log.cpp
+++ b/src/hello_imgui/internal/poor_man_log.cpp
@@ -18,7 +18,7 @@ void PoorManLog(const char* msg, ...)
     va_end(args);

 #ifdef _MSC_VER
-    OutputDebugString(buffer);
+    OutputDebugStringA(buffer);
 #else
     printf("%s", buffer);
 #endif
