diff --git a/cmake/aui.build.cmake b/cmake/aui.build.cmake
index 90c53d4..780c872 100644
--- a/cmake/aui.build.cmake
+++ b/cmake/aui.build.cmake
@@ -140,7 +140,7 @@ else()
     set(AUI_COMPILER_MSVC 0 CACHE INTERNAL "Compiler")
 endif()
 
-if (CMAKE_GENERATOR_PLATFORM MATCHES "(arm64)|(ARM64)" OR CMAKE_SYSTEM_PROCESSOR MATCHES "(aarch64|arm64)")
+if (CMAKE_GENERATOR_PLATFORM MATCHES "(aarch64|arm64|AARCH64|ARM64)" OR CMAKE_SYSTEM_PROCESSOR MATCHES "(aarch64|arm64|AARCH64|ARM64)")
     set(AUI_ARCH_X86_64 0 CACHE INTERNAL "Arch")
     set(AUI_ARCH_X86 0 CACHE INTERNAL "Arch")
     set(AUI_ARCH_ARM_64 1 CACHE INTERNAL "Arch")
