diff --git a/src/opnmidi.cpp b/src/opnmidi.cpp
index c70197f..98226dc 100644
--- a/src/opnmidi.cpp
+++ b/src/opnmidi.cpp
@@ -433,7 +433,7 @@ OPNMIDI_EXPORT void opn2_setLoopCount(OPN2_MIDIPlayer *device, int loopCount)
 
 OPNMIDI_EXPORT void opn2_setLoopHooksOnly(OPN2_MIDIPlayer *device, int loopHooksOnly)
 {
-#ifndef ADLMIDI_DISABLE_MIDI_SEQUENCER
+#ifndef OPNMIDI_DISABLE_MIDI_SEQUENCER
     if(!device)
         return;
     MidiPlayer *play = GET_MIDI_PLAYER(device);
@@ -576,7 +576,7 @@ OPNMIDI_EXPORT int opn2_openData(OPN2_MIDIPlayer *device, const void *mem, unsig
 
 OPNMIDI_EXPORT void opn2_selectSongNum(struct OPN2_MIDIPlayer *device, int songNumber)
 {
-#ifndef ADLMIDI_DISABLE_MIDI_SEQUENCER
+#ifndef OPNMIDI_DISABLE_MIDI_SEQUENCER
     if(!device)
         return;
 
@@ -591,7 +591,7 @@ OPNMIDI_EXPORT void opn2_selectSongNum(struct OPN2_MIDIPlayer *device, int songN
 
 OPNMIDI_EXPORT int opn2_getSongsCount(struct OPN2_MIDIPlayer *device)
 {
-#ifndef ADLMIDI_DISABLE_MIDI_SEQUENCER
+#ifndef OPNMIDI_DISABLE_MIDI_SEQUENCER
     if(!device)
         return 0;
 
@@ -986,7 +986,7 @@ OPNMIDI_EXPORT void opn2_setLoopStartHook(struct OPN2_MIDIPlayer *device, OPN2_L
     assert(play);
     play->hooks.onLoopStart = loopStartHook;
     play->hooks.onLoopStart_userData = userData;
-#ifndef ADLMIDI_DISABLE_MIDI_SEQUENCER
+#ifndef OPNMIDI_DISABLE_MIDI_SEQUENCER
     play->m_sequencerInterface->onloopStart = loopStartHook;
     play->m_sequencerInterface->onloopStart_userData = userData;
 #endif
@@ -1001,7 +1001,7 @@ OPNMIDI_EXPORT void opn2_setLoopEndHook(struct OPN2_MIDIPlayer *device, OPN2_Loo
     assert(play);
     play->hooks.onLoopEnd = loopEndHook;
     play->hooks.onLoopEnd_userData = userData;
-#ifndef ADLMIDI_DISABLE_MIDI_SEQUENCER
+#ifndef OPNMIDI_DISABLE_MIDI_SEQUENCER
     play->m_sequencerInterface->onloopEnd = loopEndHook;
     play->m_sequencerInterface->onloopEnd_userData = userData;
 #endif
