diff --git a/CMakeLists.txt b/CMakeLists.txt
index 20e167d..2179296 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,7 +4,7 @@ project(libOPNMIDI VERSION 1.5.1 LANGUAGES C CXX)
 include(GNUInstallDirs)
 
 # Prefer C90 standard
-set(CMAKE_C_STANDARD 90)
+set(CMAKE_C_STANDARD 99)
 # Prefer C++98 standard
 set(CMAKE_CXX_STANDARD 98)
 
@@ -88,7 +88,7 @@ function(set_legacy_standard destTarget)
     if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX)
         # Turn on warnings and legacy C/C++ standards to support more compilers
         target_compile_options(${destTarget} PRIVATE
-            $<$<COMPILE_LANGUAGE:C>:-Wall -pedantic -std=c90>
+            $<$<COMPILE_LANGUAGE:C>:-Wall -pedantic -std=c99>
             $<$<COMPILE_LANGUAGE:CXX>:-Wall -pedantic -std=gnu++98>
         )
     endif()
