diff --git a/Makefile.in b/Makefile.in
index e29609c2e..898576e57 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -717,7 +717,7 @@ MAINTAINERCLEANFILES = Makefile.in $(srcdir)/aclocal.m4 \
 #	$(srcdir)/build/Makefile.win32.features-h \
 #	$(NULL)
 ACLOCAL_AMFLAGS = -I build ${ACLOCAL_FLAGS}
-DIST_SUBDIRS = src doc util boilerplate test perf
+DIST_SUBDIRS = src doc util boilerplate perf
 SUBDIRS = src doc util $(am__append_1)
 DISTCHECK_CONFIGURE_FLAGS = \
 	--enable-gtk-doc \
@@ -1498,14 +1498,13 @@ configure: cairo-version.h
 
 doc:
 	cd doc && $(MAKE) $(AM_MAKEFLAGS) $@
-test retest recheck: all
-	cd test && $(MAKE) $(AM_MAKEFLAGS) $@
+test retest recheck:
+	
 perf: all
 	cd perf && $(MAKE) $(AM_MAKEFLAGS) $@
 check-valgrind: all
-	cd test && $(MAKE) $(AM_MAKEFLAGS) check-valgrind
 	cd perf && $(MAKE) $(AM_MAKEFLAGS) check-valgrind
-.PHONY: doc test retest recheck perf check-valgrind
+.PHONY: doc retest recheck perf check-valgrind
 
 changelogs:
 	@$(MAKE) $(AM_MAKEFLAGS) $(CHANGELOGS)
@@ -1596,7 +1595,6 @@ release-verify-sane-changelogs: changelogs
 
 release-verify-sane-tests:
 	@echo "Checking that the test suite is sane..."
-	@cd test && $(MAKE) $(AM_MAKEFLAGS) release-verify-sane-tests
 
 release-verify-even-micro:
 	@echo -n "Checking that $(VERSION) has an even micro component..."
diff --git a/configure b/configure
index de5d4e9fe..07466e506 100644
--- a/configure
+++ b/configure
@@ -21276,7 +21276,7 @@ if ${ac_cv_lib_z_compress+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lz  $LIBS"
+LIBS="$LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
diff --git a/src/cairo-compiler-private.h b/src/cairo-compiler-private.h
index d982c101d..ad9b26ba0 100644
--- a/src/cairo-compiler-private.h
+++ b/src/cairo-compiler-private.h
@@ -201,10 +201,10 @@
 #define hypot _hypot
 #define pclose _pclose
 #define popen _popen
-#define snprintf _snprintf
+//#define snprintf _snprintf
 #define strdup _strdup
 #define unlink _unlink
-#define vsnprintf _vsnprintf
+//#define vsnprintf _vsnprintf
 #endif
 
 #ifdef _MSC_VER
diff --git a/test/bitmap-font.c b/test/bitmap-font.c
index 525bafb2a..69cd985c6 100644
--- a/test/bitmap-font.c
+++ b/test/bitmap-font.c
@@ -27,7 +27,9 @@
 
 #include <sys/types.h>
 #include <sys/stat.h>
+#if defined(HAVE_UNISTD_H)
 #include <unistd.h>
+#endif
 
 #include <cairo-ft.h>
 #include <fontconfig/fontconfig.h>
diff --git a/test/cairo-test.h b/test/cairo-test.h
index 7e9605f61..dcdf74f60 100644
--- a/test/cairo-test.h
+++ b/test/cairo-test.h
@@ -58,7 +58,7 @@ typedef unsigned __int64 uint64_t;
 #define _USE_MATH_DEFINES
 
 #include <float.h>
-#define isnan(x) _isnan(x)
+//#define isnan(x) _isnan(x)
 
 #endif
 
