diff --git a/build/Makefile.win32.common b/build/Makefile.win32.common
index 7d7e9735f..f39ea1991 100644
--- a/build/Makefile.win32.common
+++ b/build/Makefile.win32.common
@@ -44,13 +44,13 @@ else
 ifeq ($(ZLIB_PATH),)
 ZLIB_PATH := $(top_builddir)/../zlib
 endif
-ZLIB_CFLAGS += -I$(ZLIB_PATH)/
-CAIRO_LIBS += $(ZLIB_PATH)/zdll.lib
+# ZLIB_CFLAGS += -I$(ZLIB_PATH)/
+# CAIRO_LIBS += $(ZLIB_PATH)/zdll.lib
 endif
 
 DEFAULT_CFLAGS = -nologo $(CFG_CFLAGS)
 DEFAULT_CFLAGS += -I. -I$(top_srcdir) -I$(top_srcdir)/src
-DEFAULT_CFLAGS += $(PIXMAN_CFLAGS) $(LIBPNG_CFLAGS) $(ZLIB_CFLAGS)
+DEFAULT_CFLAGS += $(PIXMAN_CFLAGS) $(LIBPNG_CFLAGS)
 
 CAIRO_CFLAGS = $(DEFAULT_CFLAGS) $(CFLAGS)
 
diff --git a/configure.ac b/configure.ac
index 5e33c96ea..c9e9cfaa9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -43,7 +43,7 @@ AC_CACHE_SAVE
 
 dnl ===========================================================================
 
-AC_CHECK_LIB(z, compress,
+AC_SEARCH_LIBS(compress, z zlib zlibd,
 	 [AC_CHECK_HEADER(zlib.h, [
 	  have_libz=yes
 	  AC_DEFINE(HAVE_ZLIB, 1, [Define to 1 if you have zlib available])
@@ -481,7 +481,7 @@ CAIRO_ENABLE_SURFACE_BACKEND(script, script, yes, [
   any2ppm_cs=yes
   # The script backend requires zlib.
   use_script=$have_libz
-  script_NONPKGCONFIG_LIBS=-lz
+  script_NONPKGCONFIG_LIBS=$ac_cv_search_compress
 ])
 
 dnl ===========================================================================
@@ -582,7 +582,7 @@ dnl ===========================================================================
 CAIRO_ENABLE_SURFACE_BACKEND(ps, PostScript, yes, [
     # The ps backend requires zlib.
     use_ps=$have_libz
-    ps_NONPKGCONFIG_LIBS=-lz
+    ps_NONPKGCONFIG_LIBS=$ac_cv_search_compress
 ])
 
 dnl ===========================================================================
@@ -619,7 +619,7 @@ dnl ===========================================================================
 CAIRO_ENABLE_SURFACE_BACKEND(pdf, PDF, yes, [
     # The pdf backend requires zlib.
     use_pdf=$have_libz
-    pdf_NONPKGCONFIG_LIBS=-lz
+    pdf_NONPKGCONFIG_LIBS=$ac_cv_search_compress
 ])
 
 dnl ===========================================================================
@@ -706,7 +706,7 @@ CAIRO_ENABLE_SURFACE_BACKEND(observer, observer, always)
 CAIRO_ENABLE_SURFACE_BACKEND(tee, tee, no)
 CAIRO_ENABLE_SURFACE_BACKEND(xml, xml, no, [
     use_xml=$have_libz
-    xml_NONPKGCONFIG_LIBS=-lz
+    xml_NONPKGCONFIG_LIBS=$ac_cv_search_compress
 ])
 
 dnl ===========================================================================
diff --git a/src/Makefile.am b/src/Makefile.am
index acf0a8281..fe339a3cf 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -68,7 +68,7 @@ cairo.def: cairo-features.h $(enabled_cairo_headers)
 	@(echo EXPORTS; \
 	(cd $(srcdir); cat $(enabled_cairo_headers) || echo 'cairo_ERROR ()' ) | \
 	$(EGREP) -v '^# *include' | \
-	( cat cairo-features.h - | $(CPP) -D__cplusplus - || echo 'cairo_ERROR ()' ) | \
+	( cat cairo-features.h - | $(CPP) $(CPP_FLAGS) -D__cplusplus - || echo 'cairo_ERROR ()' ) | \
 	$(EGREP) '^cairo_.* \(' | \
 	sed -e 's/[ 	].*//' | \
 	sort; \
diff --git a/util/cairo-script/Makefile.am b/util/cairo-script/Makefile.am
index d5c2998ac..e6e23b2ed 100644
--- a/util/cairo-script/Makefile.am
+++ b/util/cairo-script/Makefile.am
@@ -15,7 +15,7 @@ libcairo_script_interpreter_la_SOURCES = \
 	$(NULL)
 libcairo_script_interpreter_la_CFLAGS = $(CAIRO_CFLAGS)
 libcairo_script_interpreter_la_LDFLAGS = -version-info $(CAIRO_LIBTOOL_VERSION_INFO) -no-undefined $(export_symbols)
-libcairo_script_interpreter_la_LIBADD = $(top_builddir)/src/libcairo.la $(CAIRO_LIBS) $(lzo_LIBS) -lz
+libcairo_script_interpreter_la_LIBADD = $(top_builddir)/src/libcairo.la $(CAIRO_LIBS) $(lzo_LIBS)
 
 csi_replay_SOURCES = csi-replay.c
 csi_replay_CFLAGS = $(CAIRO_CFLAGS)
diff --git a/util/cairo-trace/Makefile.am b/util/cairo-trace/Makefile.am
index a0091f882..64d86184f 100644
--- a/util/cairo-trace/Makefile.am
+++ b/util/cairo-trace/Makefile.am
@@ -13,7 +13,7 @@ libcairo_trace_la_CPPFLAGS = -DCAIRO_TRACE_OUTDIR="\"$(cairooutdir)\"" \
 libcairo_trace_la_CFLAGS = $(CAIRO_CFLAGS) $(real_pthread_CFLAGS)
 libcairo_trace_la_LDFLAGS = -module -no-undefined -avoid-version
 
-libcairo_trace_la_LIBADD = $(real_pthread_LIBS) -lz
+libcairo_trace_la_LIBADD = $(real_pthread_LIBS)
 if CAIRO_HAS_DL
 libcairo_trace_la_LIBADD += -ldl
 endif
