 src/win32/cairo-win32-device.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/win32/cairo-win32-device.c b/src/win32/cairo-win32-device.c
index 741e49e..c60c494 100644
--- a/src/win32/cairo-win32-device.c
+++ b/src/win32/cairo-win32-device.c
@@ -131,6 +131,8 @@ _cairo_win32_device_get (void)
 {
     cairo_win32_device_t *device;
 
+    CAIRO_MUTEX_INITIALIZE ();
+
     if (__cairo_win32_device)
 	return cairo_device_reference (__cairo_win32_device);
 
