Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 1255)
+++ CMakeLists.txt	(working copy)
@@ -475,6 +475,12 @@
                ${PROJECT_BINARY_DIR}/pcre2.h
                @ONLY)
 
+# Make sure to not link debug libs
+# against release libs and vice versa
+IF(WIN32)
+  SET(CMAKE_DEBUG_POSTFIX "d")
+ENDIF(WIN32)
+
 # Generate pkg-config files
 
 SET(PACKAGE_VERSION "${PCRE2_MAJOR}.${PCRE2_MINOR}")
@@ -483,6 +489,9 @@
 SET(exec_prefix "\${prefix}")
 SET(libdir "\${exec_prefix}/lib")
 SET(includedir "\${prefix}/include")
+IF(WIN32 AND (CMAKE_BUILD_TYPE MATCHES Debug))
+  SET(LIB_POSTFIX ${CMAKE_DEBUG_POSTFIX})
+ENDIF()
 CONFIGURE_FILE(libpcre2-posix.pc.in libpcre2-posix.pc @ONLY)
 SET(pkg_config_files ${pkg_config_files} "${CMAKE_CURRENT_BINARY_DIR}/libpcre2-posix.pc")
 
@@ -622,11 +631,6 @@
 ENDIF(MSVC)
 
 SET(CMAKE_INCLUDE_CURRENT_DIR 1)
-# needed to make sure to not link debug libs
-# against release libs and vice versa
-IF(WIN32)
-  SET(CMAKE_DEBUG_POSTFIX "d")
-ENDIF(WIN32)
 
 SET(targets)
 
Index: configure.ac
===================================================================
--- configure.ac	(revision 1255)
+++ configure.ac	(working copy)
@@ -1007,6 +1007,9 @@
 AM_CONDITIONAL([WITH_GCOV],[test "x$enable_coverage" = "xyes"])
 
 # Produce these files, in addition to config.h.
+# LIB_POSTFIX is used by CMakeLists.txt for Windows debug builds.
+# Pass empty LIB_POSTFIX to *.pc files and pcre2-config here.
+AC_SUBST(LIB_POSTFIX)
 AC_CONFIG_FILES(
 	Makefile
 	libpcre2-8.pc
Index: libpcre2-16.pc.in
===================================================================
--- libpcre2-16.pc.in	(revision 1255)
+++ libpcre2-16.pc.in	(working copy)
@@ -8,6 +8,6 @@
 Name: libpcre2-16
 Description: PCRE2 - Perl compatible regular expressions C library (2nd API) with 16 bit character support
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lpcre2-16
+Libs: -L${libdir} -lpcre2-16@LIB_POSTFIX@
 Libs.private: @PTHREAD_CFLAGS@ @PTHREAD_LIBS@
 Cflags: -I${includedir} @PCRE2_STATIC_CFLAG@
Index: libpcre2-32.pc.in
===================================================================
--- libpcre2-32.pc.in	(revision 1255)
+++ libpcre2-32.pc.in	(working copy)
@@ -8,6 +8,6 @@
 Name: libpcre2-32
 Description: PCRE2 - Perl compatible regular expressions C library (2nd API) with 32 bit character support
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lpcre2-32
+Libs: -L${libdir} -lpcre2-32@LIB_POSTFIX@
 Libs.private: @PTHREAD_CFLAGS@ @PTHREAD_LIBS@
 Cflags: -I${includedir} @PCRE2_STATIC_CFLAG@
Index: libpcre2-8.pc.in
===================================================================
--- libpcre2-8.pc.in	(revision 1255)
+++ libpcre2-8.pc.in	(working copy)
@@ -8,6 +8,6 @@
 Name: libpcre2-8
 Description: PCRE2 - Perl compatible regular expressions C library (2nd API) with 8 bit character support
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lpcre2-8
+Libs: -L${libdir} -lpcre2-8@LIB_POSTFIX@
 Libs.private: @PTHREAD_CFLAGS@ @PTHREAD_LIBS@
 Cflags: -I${includedir} @PCRE2_STATIC_CFLAG@
Index: libpcre2-posix.pc.in
===================================================================
--- libpcre2-posix.pc.in	(revision 1255)
+++ libpcre2-posix.pc.in	(working copy)
@@ -8,6 +8,6 @@
 Name: libpcre2-posix
 Description: Posix compatible interface to libpcre2-8
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -lpcre2-posix
+Libs: -L${libdir} -lpcre2-posix@LIB_POSTFIX@
 Cflags: -I${includedir} @PCRE2_STATIC_CFLAG@
 Requires.private: libpcre2-8
Index: pcre2-config.in
===================================================================
--- pcre2-config.in	(revision 1255)
+++ pcre2-config.in	(working copy)
@@ -86,7 +86,7 @@
       ;;
     --libs-posix)
       if test @enable_pcre2_8@ = yes ; then
-        echo $libS$libR -lpcre2-posix -lpcre2-8
+        echo $libS$libR -lpcre2-posix@LIB_POSTFIX@ -lpcre2-8@LIB_POSTFIX@
       else
         echo "${usage}" 1>&2
       fi
@@ -93,7 +93,7 @@
       ;;
     --libs8)
       if test @enable_pcre2_8@ = yes ; then
-        echo $libS$libR -lpcre2-8
+        echo $libS$libR -lpcre2-8@LIB_POSTFIX@
       else
         echo "${usage}" 1>&2
       fi
@@ -100,7 +100,7 @@
       ;;
     --libs16)
       if test @enable_pcre2_16@ = yes ; then
-        echo $libS$libR -lpcre2-16
+        echo $libS$libR -lpcre2-16@LIB_POSTFIX@
       else
         echo "${usage}" 1>&2
       fi
@@ -107,7 +107,7 @@
       ;;
     --libs32)
       if test @enable_pcre2_32@ = yes ; then
-        echo $libS$libR -lpcre2-32
+        echo $libS$libR -lpcre2-32@LIB_POSTFIX@
       else
         echo "${usage}" 1>&2
       fi
