diff --git a/librtmp/librtmp.pc.in b/librtmp/librtmp.pc.in
--- a/librtmp/librtmp.pc.in
+++ b/librtmp/librtmp.pc.in
@@ -5,9 +5,9 @@
 
 Name: librtmp
 Description: RTMP implementation
 Version: @VERSION@
-Requires: @CRYPTO_REQ@
+Requires: zlib,@CRYPTO_REQ@
 URL: http://rtmpdump.mplayerhq.hu
-Libs: -L${libdir} -lrtmp -lz @PUBLIC_LIBS@
+Libs: -L${libdir} -lrtmp @PUBLIC_LIBS@
 Libs.private: @PRIVATE_LIBS@
 Cflags: -I${incdir}
