commit 7ccc8044b1c42759af1122a68ae134a99341d719
Author: Capric <capric8416@gmail.com>
Date:   Thu Aug 29 22:06:27 2024 +0800

    [feature] update msvc exports def

diff --git a/msvc/libdvdnav.def b/msvc/libdvdnav.def
index 7f633d1..84e5420 100644
--- a/msvc/libdvdnav.def
+++ b/msvc/libdvdnav.def
@@ -3,29 +3,89 @@
 
 EXPORTS
 
+
+
+; libdvdcss
+dvdcss_open
+dvdcss_open_stream
+dvdcss_close
+dvdcss_seek
+dvdcss_read
+dvdcss_readv
+dvdcss_error
+dvdcss_is_scrambled
+
+
+
+; libdvdread
+dvdread_getbits_init
+dvdread_getbits
+
 DVDOpen
+DVDOpen2
+DVDOpenStream
+DVDOpenStream2
 DVDClose
+DVDFileStat
 DVDOpenFile
 DVDCloseFile
 DVDReadBlocks
+DVDFileSeek
+DVDReadBytes
+DVDFileSize
+DVDDiscID
+DVDUDFVolumeInfo
+DVDFileSeekForce
+DVDISOVolumeInfo
+DVDUDFCacheLevel
 
-navRead_DSI
-navRead_PCI
+UDFFindFile
+UDFGetVolumeIdentifier
+UDFGetVolumeSetIdentifier
 
+ifo_print
+dvdread_print_time
+
+ifoOpen
+ifoOpenVMGI
+ifoOpenVTSI
 ifoClose
-ifoRead_VOBU_ADMAP
-ifoRead_VTS_ATRT
 ifoRead_PTL_MAIT
-ifoRead_PGCI_UT
+ifoRead_VTS_ATRT
 ifoRead_TT_SRPT
+ifoRead_VTS_PTT_SRPT
 ifoRead_FP_PGC
-ifoOpenVMGI
-ifoRead_TITLE_VOBU_ADMAP
 ifoRead_PGCIT
-ifoRead_VTS_PTT_SRPT
-ifoOpenVTSI
-ifoPrint
+ifoRead_PGCI_UT
+ifoRead_VTS_TMAPT
+ifoRead_C_ADT
+ifoRead_TITLE_C_ADT
+ifoRead_VOBU_ADMAP
+ifoRead_TITLE_VOBU_ADMAP
+ifoRead_TXTDT_MGI
+ifoFree_PTL_MAIT
+ifoFree_VTS_ATRT
+ifoFree_TT_SRPT
+ifoFree_VTS_PTT_SRPT
+ifoFree_FP_PGC
+ifoFree_PGCIT
+ifoFree_PGCI_UT
+ifoFree_VTS_TMAPT
+ifoFree_C_ADT
+ifoFree_TITLE_C_ADT
+ifoFree_VOBU_ADMAP
+ifoFree_TITLE_VOBU_ADMAP
+ifoFree_TXTDT_MGI
+
+navPrint_PCI
+navPrint_DSI
 
+navRead_DSI
+navRead_PCI
+
+
+
+; libdvdnav
 dvdnav_set_readahead_flag
 dvdnav_set_region_mask
 dvdnav_spu_language_select
@@ -72,11 +132,3 @@ dvdnav_set_PGC_positioning_flag
 dvdnav_get_number_of_parts
 dvdnav_reset
 
-;------------------------------------------------------------
-; timer exports
-
-gettimeofday
-setitimer
-pause
-sleep
-nanosleep
