diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5c40326a2..b987ba2e1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -825,7 +825,11 @@ add_subdirectory(${LIBRARIES}/viewer)
 add_subdirectory(${FILAMENT}/filament)
 add_subdirectory(${FILAMENT}/shaders)
 add_external_library("basisu" "basisu/tnt" "basisu::basisu;basisu::basisu_encoder;basisu::basisu_encoder" "basisu;basis_encoder;basis_transcoder")
-add_external_library("civetweb" "civetweb/tnt" "civetweb::civetweb" "civetweb")
+add_external_library("civetweb" "civetweb/tnt" "civetweb::civetweb;civetweb::civetweb-cpp" "civetweb;civetweb-cpp")
+if(TARGET civetweb::civetweb AND TARGET civetweb::civetweb-cpp)
+    target_link_libraries(civetweb::civetweb INTERFACE civetweb::civetweb-cpp)
+endif()
+
 add_external_library("imgui" "imgui/tnt" "imgui::imgui" "imgui")
 add_external_library("smol-v" "smol-v/tnt" "smol-v::smol-v" "smol-v")
 if(FILAMENT_BENCHMARKS)
--
2.45.2
