diff --git a/hdf5-1.10.5/hl/src/H5LDprivate.h b/hdf5-1.10.5/hl/src/H5LDprivate.h
index b52928b53..5f02ea781 100644
--- a/hdf5-1.10.5/hl/src/H5LDprivate.h
+++ b/hdf5-1.10.5/hl/src/H5LDprivate.h
@@ -40,8 +40,8 @@ typedef struct H5LD_memb_t {
  * #2: these two routines are too specific to be made as public routines
  * Decide to do #3 at this point of time after some discussion.
  */
-void H5LD_clean_vector(H5LD_memb_t *listv[]);
-int H5LD_construct_vector(char *fields, H5LD_memb_t *listv[], hid_t par_tid);
+H5_HLDLL void H5LD_clean_vector(H5LD_memb_t *listv[]);
+H5_HLDLL int H5LD_construct_vector(char *fields, H5LD_memb_t *listv[], hid_t par_tid);
 
 #endif /* end _H5LDprivate_H */
 
