diff --git a/CMakeLists.txt b/CMakeLists.txt
index 13f2e35aaa..5f79aece48 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -736,6 +736,9 @@ if (H5_HAVE_PARALLEL)
   if (MPI_C_LINK_FLAGS)
     set (CMAKE_EXE_LINKER_FLAGS "${MPI_C_LINK_FLAGS} ${CMAKE_EXE_LINKER_FLAGS}")
   endif ()
+  if(UNIX)
+    set(ompi-c_PC_LIBS_PRIVATE ${MPI_C_LIBRARIES})
+  endif()
 endif ()
 
 #option (DEFAULT_API_VERSION "Enable v1.12 API (v16, v18, v110, v112)" "v112")
