diff --git a/CMakeFilters.cmake b/CMakeFilters.cmake
index 52d65e5..a73dc34 100644
--- a/CMakeFilters.cmake
+++ b/CMakeFilters.cmake
@@ -110,6 +110,7 @@ if (HDF5_ENABLE_Z_LIB_SUPPORT)
         string (REGEX REPLACE "^lib" "" libname ${libname})
         set_target_properties (ZLIB::ZLIB PROPERTIES OUTPUT_NAME zlib-static)
         set (LINK_COMP_LIBS ${LINK_COMP_LIBS} ZLIB::ZLIB)
+        set (zlib_PC_LIBS_PRIVATE "${ZLIB_LIBRARIES}")
       endif ()
     else ()
       if (HDF5_ALLOW_EXTERNAL_SUPPORT MATCHES "GIT" OR HDF5_ALLOW_EXTERNAL_SUPPORT MATCHES "TGZ")
diff --git a/CMakeLists.txt b/CMakeLists.txt
index cf9729d..7ecec8d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -779,6 +779,9 @@ if (H5_HAVE_PARALLEL)
   if (MPI_C_LINK_FLAGS)
     set (CMAKE_EXE_LINKER_FLAGS "${MPI_C_LINK_FLAGS} ${CMAKE_EXE_LINKER_FLAGS}")
   endif ()
+  if(UNIX)
+    set(ompi-c_PC_LIBS_PRIVATE ${MPI_C_LIBRARIES})
+  endif()
 endif ()
 
 # Determine whether to build the HDF5 Subfiling VFD
diff --git a/config/cmake/HDFMacros.cmake b/config/cmake/HDFMacros.cmake
index 3be3e6a..177a51e 100644
--- a/config/cmake/HDFMacros.cmake
+++ b/config/cmake/HDFMacros.cmake
@@ -483,7 +483,7 @@ macro (HDF_DIR_PATHS package_prefix)
   endif ()
 
   #set the default debug suffix for all library targets
-  if(NOT CMAKE_DEBUG_POSTFIX)
+  if(NOT DEFINED CMAKE_DEBUG_POSTFIX)
     if (WIN32)
       set (CMAKE_DEBUG_POSTFIX "_D")
     else ()
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 9224cc0..6f09256 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1213,6 +1213,16 @@ if (BUILD_SHARED_LIBS)
 endif ()
 
 set (_PKG_CONFIG_REQUIRES)
+set (_PKG_CONFIG_REQUIRES_PRIVATE)
+set (_PKG_CONFIG_LIBS_PRIVATE " ${_PKG_CONFIG_LIBS_PRIVATE} ")
+foreach(_module IN ITEMS ompi-c zlib)
+    if(${_module}_PC_LIBS_PRIVATE)
+        foreach(_lib IN LISTS ${_module}_PC_LIBS_PRIVATE)
+            string(REPLACE " -l${_lib} " " " _PKG_CONFIG_LIBS_PRIVATE "${_PKG_CONFIG_LIBS_PRIVATE}")
+        endforeach()
+        string(APPEND _PKG_CONFIG_REQUIRES_PRIVATE " ${_module}")
+    endif()
+endforeach()
 
 configure_file (
     ${HDF_CONFIG_DIR}/libhdf5.pc.in
