diff --git a/cmake/dbus-cxx-glibConfig.cmake.in b/cmake/dbus-cxx-glibConfig.cmake.in
new file mode 100644
index 0000000..1ba2959
--- /dev/null
+++ b/cmake/dbus-cxx-glibConfig.cmake.in
@@ -0,0 +1,9 @@
+@PACKAGE_INIT@
+
+include(CMakeFindDependencyMacro)
+
+find_dependency(sigc++-3)
+find_dependency(PkgConfig)
+pkg_check_modules(glib REQUIRED IMPORTED_TARGET glib-2.0)
+
+include("${CMAKE_CURRENT_LIST_DIR}/dbus-cxx-glibTargets.cmake")
diff --git a/cmake/dbus-cxx-qtConfig.cmake.in b/cmake/dbus-cxx-qtConfig.cmake.in
new file mode 100644
index 0000000..76241c1
--- /dev/null
+++ b/cmake/dbus-cxx-qtConfig.cmake.in
@@ -0,0 +1,8 @@
+@PACKAGE_INIT@
+
+include(CMakeFindDependencyMacro)
+
+find_dependency(sigc++-3)
+find_dependency(Qt5 COMPONENTS Core)
+
+include("${CMAKE_CURRENT_LIST_DIR}/dbus-cxx-qtTargets.cmake")
diff --git a/cmake/dbus-cxx-uvConfig.cmake.in b/cmake/dbus-cxx-uvConfig.cmake.in
new file mode 100644
index 0000000..b52832e
--- /dev/null
+++ b/cmake/dbus-cxx-uvConfig.cmake.in
@@ -0,0 +1,8 @@
+@PACKAGE_INIT@
+
+include(CMakeFindDependencyMacro)
+
+find_dependency(sigc++-3)
+find_dependency(libuv)
+
+include("${CMAKE_CURRENT_LIST_DIR}/dbus-cxx-uvTargets.cmake")
diff --git a/cmake/dbus-cxxConfig.cmake.in b/cmake/dbus-cxxConfig.cmake.in
new file mode 100644
index 0000000..2dbd443
--- /dev/null
+++ b/cmake/dbus-cxxConfig.cmake.in
@@ -0,0 +1,8 @@
+@PACKAGE_INIT@
+
+include(CMakeFindDependencyMacro)
+
+find_dependency(Threads)
+find_dependency(sigc++-3)
+
+include("${CMAKE_CURRENT_LIST_DIR}/dbus-cxxTargets.cmake")

