diff --git a/setup.py b/setup.py
index 85a2b26..4b7b097 100644
--- a/setup.py
+++ b/setup.py
@@ -1737,13 +1737,16 @@ def detect_compress_exts(self):
                            extra_link_args=extra_link_args))
 
         # Gustavo Niemeyer's bz2 module.
-        if (self.compiler.find_library_file(self.lib_dirs, 'bz2')):
+        bz2_name = 'bz2'
+        if (self.compiler.find_library_file(self.lib_dirs, 'bz2d')):
+            bz2_name = 'bz2d'
+        if (self.compiler.find_library_file(self.lib_dirs, bz2_name)):
             if MACOS:
                 bz2_extra_link_args = ('-Wl,-search_paths_first',)
             else:
                 bz2_extra_link_args = ()
             self.add(Extension('_bz2', ['_bz2module.c'],
-                               libraries=['bz2'],
+                               libraries=[bz2_name],
                                extra_link_args=bz2_extra_link_args))
         else:
             self.missing.append('_bz2')
