diff --git a/configure.ac b/configure.ac
index f93a9f2..30cad5a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -53,6 +53,14 @@ AC_DEFUN([WITH_SAVE_ENV],
 [RESTORE_ENV]
 )dnl
 
+if test "$ac_sys_system" != "Darwin"; then
+AS_VAR_IF([GNULD], [yes], [
+  LDFLAGS="$LDFLAGS -Wl,--enable-new-dtags,-rpath,$vcpkg_rpath"
+], [
+  LDFLAGS="$LDFLAGS -Wl,-rpath,$vcpkg_rpath"
+])
+fi
+
 dnl PY_CHECK_FUNC(FUNCTION, [INCLUDES], [AC_DEFINE-VAR])
 AC_DEFUN([PY_CHECK_FUNC],
 [ AS_VAR_PUSHDEF([py_var], [ac_cv_func_$1])
