diff --git a/Lib/sysconfig/__init__.py b/Lib/sysconfig/__init__.py
index 2ecbff2..6b969da 100644
--- a/Lib/sysconfig/__init__.py
+++ b/Lib/sysconfig/__init__.py
@@ -53,8 +53,8 @@
         'platstdlib': '{base}/Lib',
         'purelib': '{base}/Lib/site-packages',
         'platlib': '{base}/Lib/site-packages',
-        'include': '{installed_base}/Include',
-        'platinclude': '{installed_base}/Include',
+        'include': '{installed_base}/../../include/python{py_version_short}{abiflags}',
+        'platinclude': '{installed_base}/../../include/python{py_version_short}{abiflags}',
         'scripts': '{base}/Scripts',
         'data': '{base}',
         },
@@ -93,8 +93,8 @@
         'platstdlib': '{base}/Lib',
         'purelib': '{base}/Lib/site-packages',
         'platlib': '{base}/Lib/site-packages',
-        'include': '{installed_base}/Include',
-        'platinclude': '{installed_base}/Include',
+        'include': '{installed_base}/../../include/python{py_version_short}{abiflags}',
+        'platinclude': '{installed_base}/../../include/python{py_version_short}{abiflags}',
         'scripts': '{base}/Scripts',
         'data': '{base}',
         },
