
diff --git a/Lib/ctypes/__init__.py b/Lib/ctypes/__init__.py
--- a/Lib/ctypes/__init__.py
+++ b/Lib/ctypes/__init__.py
@@ -460,6 +460,8 @@
     pythonapi = PyDLL("python dll", None, _sys.dllhandle)
 elif _sys.platform == "cygwin":
     pythonapi = PyDLL("libpython%d.%d.dll" % _sys.version_info[:2])
+elif hasattr(_sys, 'getandroidapilevel'):
+    pythonapi = PyDLL("libpython%d.%d.so" % _sys.version_info[:2])
 else:
     pythonapi = PyDLL(None)
 
