diff --git a/PCbuild/_zstd.vcxproj b/PCbuild/_zstd.vcxproj
index 6f91b8d..a52ebc3 100644
--- a/PCbuild/_zstd.vcxproj
+++ b/PCbuild/_zstd.vcxproj
@@ -94,8 +94,7 @@
   </PropertyGroup>
   <ItemDefinitionGroup>
     <ClCompile>
-      <PreprocessorDefinitions>WIN32;ZSTD_MULTITHREAD=1;%(PreprocessorDefinitions)</PreprocessorDefinitions>
-      <AdditionalIncludeDirectories>$(zstdDir)lib\;$(zstdDir)lib\common;$(zstdDir)lib\compress;$(zstdDir)lib\decompress;$(zstdDir)lib\dictBuilder;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <PreprocessorDefinitions>WIN32;%(PreprocessorDefinitions)</PreprocessorDefinitions>
     </ClCompile>
   </ItemDefinitionGroup>
   <ItemGroup>
@@ -103,64 +102,11 @@
     <ClCompile Include="..\Modules\_zstd\compressor.c" />
     <ClCompile Include="..\Modules\_zstd\decompressor.c" />
     <ClCompile Include="..\Modules\_zstd\zstddict.c" />
-    <ClCompile Include="$(zstdDir)lib\common\debug.c" />
-    <ClCompile Include="$(zstdDir)lib\common\entropy_common.c" />
-    <ClCompile Include="$(zstdDir)lib\common\error_private.c" />
-    <ClCompile Include="$(zstdDir)lib\common\fse_decompress.c" />
-    <ClCompile Include="$(zstdDir)lib\common\pool.c" />
-    <ClCompile Include="$(zstdDir)lib\common\threading.c" />
-    <ClCompile Include="$(zstdDir)lib\common\xxhash.c" />
-    <ClCompile Include="$(zstdDir)lib\common\zstd_common.c" />
-    <ClCompile Include="$(zstdDir)lib\compress\fse_compress.c" />
-    <ClCompile Include="$(zstdDir)lib\compress\hist.c" />
-    <ClCompile Include="$(zstdDir)lib\compress\huf_compress.c" />
-    <ClCompile Include="$(zstdDir)lib\compress\zstd_compress.c" />
-    <ClCompile Include="$(zstdDir)lib\compress\zstd_compress_literals.c" />
-    <ClCompile Include="$(zstdDir)lib\compress\zstd_compress_sequences.c" />
-    <ClCompile Include="$(zstdDir)lib\compress\zstd_compress_superblock.c" />
-    <ClCompile Include="$(zstdDir)lib\compress\zstd_double_fast.c" />
-    <ClCompile Include="$(zstdDir)lib\compress\zstd_fast.c" />
-    <ClCompile Include="$(zstdDir)lib\compress\zstd_lazy.c" />
-    <ClCompile Include="$(zstdDir)lib\compress\zstd_ldm.c" />
-    <ClCompile Include="$(zstdDir)lib\compress\zstd_opt.c" />
-    <ClCompile Include="$(zstdDir)lib\compress\zstd_preSplit.c" />
-    <ClCompile Include="$(zstdDir)lib\compress\zstdmt_compress.c" />
-    <ClCompile Include="$(zstdDir)lib\decompress\huf_decompress.c" />
-    <ClCompile Include="$(zstdDir)lib\decompress\zstd_ddict.c" />
-    <ClCompile Include="$(zstdDir)lib\decompress\zstd_decompress.c" />
-    <ClCompile Include="$(zstdDir)lib\decompress\zstd_decompress_block.c" />
-    <ClCompile Include="$(zstdDir)lib\dictBuilder\cover.c" />
-    <ClCompile Include="$(zstdDir)lib\dictBuilder\divsufsort.c" />
-    <ClCompile Include="$(zstdDir)lib\dictBuilder\fastcover.c" />
-    <ClCompile Include="$(zstdDir)lib\dictBuilder\zdict.c" />
   </ItemGroup>
   <ItemGroup>
     <ClInclude Include="..\Modules\_zstd\_zstdmodule.h" />
     <ClInclude Include="..\Modules\_zstd\buffer.h" />
     <ClInclude Include="..\Modules\_zstd\zstddict.h" />
-    <ClInclude Include="$(zstdDir)lib\common\bitstream.h" />
-    <ClInclude Include="$(zstdDir)lib\common\error_private.h" />
-    <ClInclude Include="$(zstdDir)lib\common\fse.h" />
-    <ClInclude Include="$(zstdDir)lib\common\huf.h" />
-    <ClInclude Include="$(zstdDir)lib\common\mem.h" />
-    <ClInclude Include="$(zstdDir)lib\common\pool.h" />
-    <ClInclude Include="$(zstdDir)lib\common\threading.h" />
-    <ClInclude Include="$(zstdDir)lib\common\xxhash.h" />
-    <ClInclude Include="$(zstdDir)lib\common\zstd_internal.h" />
-    <ClInclude Include="$(zstdDir)lib\compress\zstd_compress.h" />
-    <ClInclude Include="$(zstdDir)lib\compress\zstd_compress_literals.h" />
-    <ClInclude Include="$(zstdDir)lib\compress\zstd_compress_sequences.h" />
-    <ClInclude Include="$(zstdDir)lib\compress\zstd_compress_superblock.h" />
-    <ClInclude Include="$(zstdDir)lib\compress\zstd_cwksp.h" />
-    <ClInclude Include="$(zstdDir)lib\compress\zstd_double_fast.h" />
-    <ClInclude Include="$(zstdDir)lib\compress\zstd_fast.h" />
-    <ClInclude Include="$(zstdDir)lib\compress\zstd_lazy.h" />
-    <ClInclude Include="$(zstdDir)lib\compress\zstd_ldm.h" />
-    <ClInclude Include="$(zstdDir)lib\compress\zstd_opt.h" />
-    <ClInclude Include="$(zstdDir)lib\compress\zstdmt_compress.h" />
-    <ClInclude Include="$(zstdDir)lib\decompress\zstd_ddict.h" />
-    <ClInclude Include="$(zstdDir)lib\zstd.h" />
-    <ClInclude Include="$(zstdDir)lib\zstd_errors.h" />
   </ItemGroup>
   <ItemGroup>
     <ResourceCompile Include="..\PC\python_nt.rc" />
