diff --git a/tools/CMakeLists.txt b/tools/CMakeLists.txt
index d2b721d..5d41f91 100644
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -15,9 +15,15 @@ install(TARGETS fastconv fastconvr fft
 # psdpng does not build with "simd" datatype
 if(NOT KISSFFT_DATATYPE MATCHES "simd")
     include(FindPkgConfig)
+    if(PKG_CONFIG_FOUND)
     pkg_check_modules(libpng REQUIRED IMPORTED_TARGET libpng)
     add_kissfft_executable(psdpng psdpng.c)
     target_link_libraries(psdpng PRIVATE PkgConfig::libpng)
+    else()
+        find_package(PNG REQUIRED)
+        add_kissfft_executable(psdpng psdpng.c)
+        target_link_libraries(psdpng PRIVATE PNG::PNG)
+    endif()
     install(TARGETS psdpng
         ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
         LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
