diff --git a/CMakeLists.txt b/CMakeLists.txt
index bf74604..e49dd84 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -183,6 +183,7 @@ foreach(target "nanoarrow_static" "nanoarrow_shared")
                                     $<INSTALL_INTERFACE:include>)
 endforeach()
 
+set(NANOARROW_FLATCC_EXTERNAL OFF)
 if(NANOARROW_IPC)
   # Add the flatcc (runtime) dependency
   set(FLATCC_RTONLY
@@ -218,6 +219,7 @@ if(NANOARROW_IPC)
             ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
   elseif(NOT NANOARROW_FLATCC_ROOT_DIR)
+    set(NANOARROW_FLATCC_EXTERNAL ON)
     add_library(flatccrt STATIC IMPORTED)
     set_target_properties(flatccrt
                           PROPERTIES IMPORTED_LOCATION
@@ -226,6 +228,7 @@ if(NANOARROW_IPC)
                                      "${NANOARROW_FLATCC_INCLUDE_DIR}")
 
   elseif(NOT NANOARROW_FLATCC_INCLUDE_DIR)
+    set(NANOARROW_FLATCC_EXTERNAL ON)
     set(NANOARROW_FLATCC_INCLUDE_DIR "${NANOARROW_FLATCC_ROOT_DIR}/include")
     add_library(flatccrt STATIC IMPORTED)
     set_target_properties(flatccrt
diff --git a/cmake/config.cmake.in b/cmake/config.cmake.in
index e1af6ab..4ecf151 100644
--- a/cmake/config.cmake.in
+++ b/cmake/config.cmake.in
@@ -23,6 +23,36 @@ cmake_minimum_required(VERSION @CMAKE_MINIMUM_REQUIRED_VERSION@)
 include("${CMAKE_CURRENT_LIST_DIR}/nanoarrow-targets.cmake" REQUIRED)
 include("${CMAKE_CURRENT_LIST_DIR}/nanoarrow-config-version.cmake" REQUIRED)
 
+if(@NANOARROW_FLATCC_EXTERNAL@ AND TARGET nanoarrow::nanoarrow_ipc_static
+                                  AND NOT TARGET flatccrt)
+  find_library(_nanoarrow_flatccrt_library_release
+               NAMES flatccrt
+               PATHS "${PACKAGE_PREFIX_DIR}/lib"
+               NO_DEFAULT_PATH)
+  find_library(_nanoarrow_flatccrt_library_debug
+               NAMES flatccrt_d flatccrt
+               PATHS "${PACKAGE_PREFIX_DIR}/debug/lib"
+               NO_DEFAULT_PATH)
+
+  if(NOT _nanoarrow_flatccrt_library_release OR
+     NOT _nanoarrow_flatccrt_library_debug)
+    message(FATAL_ERROR "The nanoarrow IPC static target requires flatccrt")
+  endif()
+
+  add_library(flatccrt STATIC IMPORTED)
+  set_target_properties(flatccrt
+                        PROPERTIES IMPORTED_CONFIGURATIONS "Release;Debug"
+                                   IMPORTED_LOCATION_RELEASE
+                                   "${_nanoarrow_flatccrt_library_release}"
+                                   IMPORTED_LOCATION_DEBUG
+                                   "${_nanoarrow_flatccrt_library_debug}"
+                                   MAP_IMPORTED_CONFIG_MINSIZEREL Release
+                                   MAP_IMPORTED_CONFIG_RELWITHDEBINFO Release)
+
+  unset(_nanoarrow_flatccrt_library_release CACHE)
+  unset(_nanoarrow_flatccrt_library_debug CACHE)
+endif()
+
 foreach(target nanoarrow nanoarrow_ipc nanoarrow_device nanoarrow_testing)
   if(TARGET nanoarrow::${target}_static)
     if(@NANOARROW_BUILD_SHARED_LIBS@)
