diff --git a/CMakeLists.txt b/CMakeLists.txt
index db08f06..14da6c8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -229,10 +229,21 @@ if(NANOARROW_IPC)
     set(NANOARROW_FLATCC_INCLUDE_DIR "${NANOARROW_FLATCC_ROOT_DIR}/include")
     add_library(flatccrt STATIC IMPORTED)
     set_target_properties(flatccrt
-                          PROPERTIES IMPORTED_LOCATION
-                                     "${NANOARROW_FLATCC_ROOT_DIR}/lib/libflatccrt.a"
-                                     INTERFACE_INCLUDE_DIRECTORIES
+                          PROPERTIES INTERFACE_INCLUDE_DIRECTORIES
                                      "${NANOARROW_FLATCC_INCLUDE_DIR}")
+    if (WIN32)
+      set_target_properties(flatccrt
+                            PROPERTIES IMPORTED_LOCATION_DEBUG
+                                       "${NANOARROW_FLATCC_ROOT_DIR}/debug/lib/flatccrt_d.lib"
+                                       IMPORTED_LOCATION_RELEASE
+                                       "${NANOARROW_FLATCC_ROOT_DIR}/lib/flatccrt.lib")
+    else()
+      set_target_properties(flatccrt
+                            PROPERTIES IMPORTED_LOCATION_DEBUG
+                                       "${NANOARROW_FLATCC_ROOT_DIR}/debug/lib/libflatccrt_d.a"
+                                       IMPORTED_LOCATION_RELEASE
+                                       "${NANOARROW_FLATCC_ROOT_DIR}/lib/libflatccrt.a")
+    endif()
   endif()
 
   if(NANOARROW_IPC_WITH_ZSTD)
