diff --git a/CMakeLists.txt b/CMakeLists.txt
index f7f388f..4ab8149 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -456,12 +456,14 @@ if(FLATBUFFERS_BUILD_FLATC)
   endif()
 
   target_link_libraries(flatc PRIVATE $<BUILD_INTERFACE:ProjectConfig>)
+  if(FALSE) # DONT mess with runtime flags
   target_compile_options(flatc
     PRIVATE
       $<$<AND:$<BOOL:${MSVC_LIKE}>,$<CONFIG:Release>>:
         /MT
       >
   )
+  endif()
 
   if(FLATBUFFERS_CODE_SANITIZE AND NOT WIN32)
     add_fsanitize_to_target(flatc ${FLATBUFFERS_CODE_SANITIZE})
