diff --git a/CMakeLists.txt b/CMakeLists.txt
index 32786db..4215d25 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -426,7 +426,7 @@ endif()
 
 if(GDCM_USE_SYSTEM_EXPAT)
   # If user say so, then this is a requirement !
-  find_package(EXPAT REQUIRED)
+  find_package(expat CONFIG REQUIRED)
   set(GDCM_EXPAT_LIBRARIES ${EXPAT_LIBRARIES})
 else()
   set(GDCM_EXPAT_LIBRARIES "gdcmexpat")
