diff --git a/cmake/liblas-config.cmake.in b/cmake/liblas-config.cmake.in
index 175e997..4f3b1ba 100644
--- a/cmake/liblas-config.cmake.in
+++ b/cmake/liblas-config.cmake.in
@@ -12,6 +12,12 @@ message (STATUS "Reading ${CMAKE_CURRENT_LIST_FILE}")
 # libLAS_VERSION is set by version file
 message (STATUS "libLAS configuration, version " ${libLAS_VERSION})
 
+include(CMakeFindDependencyMacro)
+
+if (@WITH_GEOTIFF@)
+    find_dependency(GeoTIFF CONFIG)
+endif()
+
 # Tell the user project where to find our headers and libraries
 get_filename_component (_DIR ${CMAKE_CURRENT_LIST_FILE} PATH)
 get_filename_component (PROJECT_ROOT_DIR "${_DIR}/@PROJECT_ROOT_DIR@" ABSOLUTE)
