diff --git a/CMakeLists.txt b/CMakeLists.txt
index d246a88..b0417ee 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -107,7 +107,7 @@ set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${LIBLAS_BUILD_OUTPUT_DIRECTORY})
 set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${LIBLAS_BUILD_OUTPUT_DIRECTORY})
 
 file(READ "doc/index.txt" README )
-file(WRITE "README.txt" "${README}")
+file(WRITE "${CMAKE_BINARY_DIR}/README.txt" "${README}")
 
 ###############################################################################
 # Platform and compiler specific settings
@@ -337,7 +337,7 @@ set(LIBLAS_DATA_DIR ${LIBLAS_DATA_SUBDIR})
 ###############################################################################
 # Installation commands
 
-install(FILES AUTHORS COPYING INSTALL LICENSE.txt README.txt
+install(FILES AUTHORS COPYING INSTALL LICENSE.txt ${CMAKE_BINARY_DIR}/README.txt
   DESTINATION ${LIBLAS_DATA_DIR}/doc)
 
 ###############################################################################
