diff --git a/CMakeLists.txt b/CMakeLists.txt
index e8d1c1b..d1f320e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -62,7 +62,7 @@ macro(blas_install_library lib)
     EXPORT ${BLAS_INSTALL_EXPORT_NAME}
     ARCHIVE DESTINATION lib COMPONENT Development
     LIBRARY DESTINATION lib COMPONENT RuntimeLibraries
-    RUNTIME DESTINATION lib COMPONENT RuntimeLibraries
+    RUNTIME DESTINATION bin COMPONENT RuntimeLibraries
   )
 endmacro()
 
