diff --git a/CMakeLists.txt b/CMakeLists.txt
index 96ea265..87eaed5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -647,6 +647,8 @@ if(QUIC_CODE_CHECK)
     endif()
 endif()
 
+include(GNUInstallDirs)
+
 add_subdirectory(src/inc)
 
 # Product code
@@ -657,6 +659,8 @@ add_subdirectory(src/bin)
 # Tool code
 if(QUIC_BUILD_TOOLS)
     add_subdirectory(src/tools)
+    install(TARGETS quicattack quicinterop quicinteropserver quicipclient quicipserver 
+                    quicpcp quicping quicpost quicreach quicsample spinquic)
 endif()
 
 # Performance code
diff --git a/src/bin/CMakeLists.txt b/src/bin/CMakeLists.txt
index 4dda469..2797670 100644
--- a/src/bin/CMakeLists.txt
+++ b/src/bin/CMakeLists.txt
@@ -29,7 +29,7 @@ endif()
 
 target_include_directories(msquic PUBLIC
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../inc>
-    $<INSTALL_INTERFACE:${include_dest}>)
+    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
 
 set(PUBLIC_HEADERS
     ../inc/msquic.h
@@ -37,14 +37,17 @@ set(PUBLIC_HEADERS
     ../inc/msquic_posix.h
     ../inc/quic_sal_stub.h)
 
-install(TARGETS msquic EXPORT msquic DESTINATION "${main_lib_dest}")
-install(FILES ${PUBLIC_HEADERS} DESTINATION "${include_dest}")
+install(TARGETS msquic EXPORT msquic-targets 
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(FILES ${PUBLIC_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
 configure_file(msquic-config.cmake.in ${CMAKE_BINARY_DIR}/msquic-config.cmake)
 
-install(FILES ${CMAKE_BINARY_DIR}/msquic-config.cmake DESTINATION ${msquic_dest})
+install(FILES ${CMAKE_BINARY_DIR}/msquic-config.cmake DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/msquic)
 
-install(EXPORT msquic DESTINATION ${msquic_dest})
+install(EXPORT msquic-targets DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/msquic)
 
 if(WIN32)
     add_library(msquic.lttng INTERFACE)
diff --git a/src/bin/msquic-config.cmake.in b/src/bin/msquic-config.cmake.in
index dd67624..c561361 100644
--- a/src/bin/msquic-config.cmake.in
+++ b/src/bin/msquic-config.cmake.in
@@ -1,4 +1,4 @@
 include(CMakeFindDependencyMacro)
 @FILENAME_DEP_REPLACE@
 
-include(${SELF_DIR}/msquic.cmake)
+include(${SELF_DIR}/msquic-targets.cmake)
