diff --git a/CMakeLists.txt b/CMakeLists.txt
index 999711e..46399d5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -525,7 +525,7 @@ if(WIN32)
         set(CMAKE_VS_SDK_LIBRARY_DIRECTORIES "$(LibraryPath);$(VC_LibraryPath_VC_${SYSTEM_PROCESSOR}_Desktop)")
     else()
         # Just doing a normal build. Use the PGD file if present.
-        if(EXISTS "${QUIC_PGO_FILE}")
+        if(FALSE)
             message(STATUS "Using profile-guided optimization")
             file(MAKE_DIRECTORY ${QUIC_PGO_DIR})
             configure_file("${QUIC_PGO_FILE}" "${QUIC_PGO_DIR}/msquic.pgd" COPYONLY)
diff --git a/submodules/CMakeLists.txt b/submodules/CMakeLists.txt
index deadd19..3616eac 100644
--- a/submodules/CMakeLists.txt
+++ b/submodules/CMakeLists.txt
@@ -53,31 +53,29 @@ if (WIN32)
 
     if (QUIC_UWP_BUILD)
         # Translate target architecture into corresponding OpenSSL build flag
-        if (${SYSTEM_PROCESSOR} STREQUAL "arm64")
+        if (VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64")
             set(QUIC_OPENSSL_WIN_ARCH "VC-WIN64-ARM")
-        elseif (${SYSTEM_PROCESSOR} STREQUAL "arm")
+        elseif (VCPKG_TARGET_ARCHITECTURE STREQUAL "arm")
             set(QUIC_OPENSSL_WIN_ARCH "VC-WIN32-ARM")
-        elseif (${SYSTEM_PROCESSOR} STREQUAL "win32")
+        elseif (VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
             set(QUIC_OPENSSL_WIN_ARCH "VC-WIN32-ONECORE")
-        elseif (${SYSTEM_PROCESSOR} STREQUAL "x64" OR ${SYSTEM_PROCESSOR} STREQUAL "amd64")
+        elseif (VCPKG_TARGET_ARCHITECTURE STREQUAL "x64")
             set(QUIC_OPENSSL_WIN_ARCH "VC-WIN64A-ONECORE")
-        else()
-            message(FATAL_ERROR "Unknown Generator Platform ${SYSTEM_PROCESSOR}")
         endif()
     else()
-        # Translate target architecture into corresponding OpenSSL build flag
-        if (${SYSTEM_PROCESSOR} STREQUAL "arm64")
+        if (VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64")
             set(QUIC_OPENSSL_WIN_ARCH "VC-WIN64-ARM")
-        elseif (${SYSTEM_PROCESSOR} STREQUAL "arm")
+        elseif (VCPKG_TARGET_ARCHITECTURE STREQUAL "arm")
             set(QUIC_OPENSSL_WIN_ARCH "VC-WIN32-ARM")
-        elseif (${SYSTEM_PROCESSOR} STREQUAL "win32")
+        elseif (VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
             set(QUIC_OPENSSL_WIN_ARCH "VC-WIN32")
-        elseif (${SYSTEM_PROCESSOR} STREQUAL "x64" OR ${SYSTEM_PROCESSOR} STREQUAL "amd64")
+        elseif (VCPKG_TARGET_ARCHITECTURE STREQUAL "x64")
             set(QUIC_OPENSSL_WIN_ARCH "VC-WIN64A")
-        else()
-            message(FATAL_ERROR "Unknown Generator Platform ${SYSTEM_PROCESSOR}")
         endif()
     endif()
+    if(NOT DEFINED QUIC_OPENSSL_WIN_ARCH)
+        message(FATAL_ERROR "Unknown Platform ${VCPKG_TARGET_ARCHITECTURE}")
+    endif()
 
     set(OPENSSL_EXTRA_CONFIGURE_ARGS "")
 

