diff --git a/CMakeLists.txt b/CMakeLists.txt
index 455221b8..51da4ca6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -663,8 +663,12 @@ set_target_properties(re-objs PROPERTIES POSITION_INDEPENDENT_CODE ON)
 target_compile_definitions(re-objs PRIVATE ${RE_DEFINITIONS})
 
 target_include_directories(re-objs PRIVATE .)
-target_include_directories(re-objs PRIVATE include
+target_include_directories(re-objs PRIVATE
   ${OPENSSL_INCLUDE_DIR} ${ZLIB_INCLUDE_DIRS})
+target_include_directories(re-objs PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include>
+  )
 
 
 ##############################################################################
@@ -680,6 +684,7 @@ if(LIBRE_BUILD_SHARED)
     ${PROJECT_SOVERSION}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH})
   set_target_properties(re-shared PROPERTIES SOVERSION ${PROJECT_SOVERSION})
   set_target_properties(re-shared PROPERTIES OUTPUT_NAME "re")
+  add_library(libre::re-shared ALIAS re-shared)
 endif()
 
 
@@ -692,8 +697,7 @@ if(LIBRE_BUILD_STATIC)
   list(APPEND RE_INSTALL_TARGETS re)
   add_library(re STATIC $<TARGET_OBJECTS:re-objs>)
   target_link_libraries(re PUBLIC ${LINKLIBS})
-  target_include_directories(re PUBLIC include)
-  set_target_properties(re PROPERTIES PUBLIC_HEADER "${HEADERS}")
+  add_library(libre::re ALIAS re)
 
   if(MSVC)
     set_target_properties(re PROPERTIES OUTPUT_NAME "re-static")
@@ -719,6 +723,7 @@ configure_file(packaging/libre.pc.in libre.pc @ONLY)
 
 
 install(TARGETS ${RE_INSTALL_TARGETS}
+  EXPORT libre
   LIBRARY
     DESTINATION ${CMAKE_INSTALL_LIBDIR}
     COMPONENT Libraries
@@ -726,9 +731,18 @@ install(TARGETS ${RE_INSTALL_TARGETS}
   ARCHIVE
     DESTINATION ${CMAKE_INSTALL_LIBDIR}
     COMPONENT Development
-  PUBLIC_HEADER
+  INCLUDES
     DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/re
-    COMPONENT Development
+)
+
+install(FILES ${HEADERS}
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/re
+  COMPONENT Development
+)
+
+install(EXPORT libre
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/libre
+  NAMESPACE libre::
 )
 
 if(LIBRE_BUILD_SHARED)
@@ -745,6 +759,11 @@ install(FILES cmake/re-config.cmake
   COMPONENT Development
 )
 
+install(FILES cmake/libre-config.cmake
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/libre
+  COMPONENT Development
+)
+
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libre.pc
   DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig
   COMPONENT Development
diff --git a/cmake/libre-config.cmake b/cmake/libre-config.cmake
new file mode 100644
index 00000000..0965b84f
--- /dev/null
+++ b/cmake/libre-config.cmake
@@ -0,0 +1 @@
+include("${CMAKE_CURRENT_LIST_DIR}/libre.cmake")
diff --git a/cmake/re-config.cmake b/cmake/re-config.cmake
index 4f65a962..96eb330e 100644
--- a/cmake/re-config.cmake
+++ b/cmake/re-config.cmake
@@ -41,7 +41,7 @@ else()
   check_symbol_exists(res_ninit resolv.h HAVE_RESOLV)
 endif()
 if(HAVE_RESOLV)
-  find_library(RESOLV_LIBRARY resolv)
+  set(RESOLV_LIBRARY resolv)
   list(APPEND RE_DEFINITIONS -DHAVE_RESOLV)
 else()
   set(RESOLV_LIBRARY)
