diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3bd5dd7..03d7aef 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -665,6 +665,7 @@ target_include_directories(re-objs PRIVATE include
 # Shared target libre.[so|dll|dylib]
 #
 
+if(BUILD_SHARED_LIBS)
 add_library(re-shared SHARED $<TARGET_OBJECTS:re-objs>)
 target_link_libraries(re-shared PRIVATE ${LINKLIBS})
 set_target_properties(re-shared PROPERTIES VERSION
@@ -672,12 +673,16 @@ set_target_properties(re-shared PROPERTIES VERSION
 set_target_properties(re-shared PROPERTIES SOVERSION ${PROJECT_SOVERSION})
 set_target_properties(re-shared PROPERTIES OUTPUT_NAME "re")
 
+set_target_properties(re-shared PROPERTIES PUBLIC_HEADER "${HEADERS}")
+endif()
+
 
 ##############################################################################
 #
 # Static target libre.a
 #
 
+if(NOT BUILD_SHARED_LIBS)
 add_library(re STATIC $<TARGET_OBJECTS:re-objs>)
 target_link_libraries(re PUBLIC ${LINKLIBS})
 target_include_directories(re PUBLIC include)
@@ -686,6 +691,7 @@ set_target_properties(re PROPERTIES PUBLIC_HEADER "${HEADERS}")
 if(MSVC)
 set_target_properties(re PROPERTIES OUTPUT_NAME "re-static")
 endif()
+endif()
 
 
 ##############################################################################
@@ -705,11 +711,18 @@ configure_file(packaging/libre.pc.in libre.pc @ONLY)
 # Install section
 #
 
-install(TARGETS re-shared re
+if(BUILD_SHARED_LIBS)
+install(TARGETS re-shared
   LIBRARY
     DESTINATION ${CMAKE_INSTALL_LIBDIR}
     COMPONENT Libraries
     NAMELINK_SKIP
+  PUBLIC_HEADER
+    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/re
+    COMPONENT Libraries
+)
+else()
+install(TARGETS re
   ARCHIVE
     DESTINATION ${CMAKE_INSTALL_LIBDIR}
     COMPONENT Development
@@ -717,13 +730,20 @@ install(TARGETS re-shared re
     DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/re
     COMPONENT Development
 )
+endif()
 
+
+if(BUILD_SHARED_LIBS)
 install(TARGETS re-shared
   LIBRARY
     DESTINATION ${CMAKE_INSTALL_LIBDIR}
     NAMELINK_ONLY
     COMPONENT Development
+  PUBLIC_HEADER
+    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/re
+    COMPONENT Development
 )
+endif()
 
 install(FILES cmake/re-config.cmake
   DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/re
