diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index fd54315..873c504 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -96,9 +96,24 @@ if(UNIX AND NOT APPLE)
     target_link_libraries (hpdf ${M_LIB})
 endif()
 
+if(BUILD_SHARED_LIBS)
+  if(WIN32 AND NOT CYGWIN)
+    target_compile_definitions(hpdf INTERFACE HPDF_DLL)
+    target_compile_definitions(hpdf PRIVATE HPDF_DLL_MAKE)
+  endif()
+endif()
+
+target_include_directories(hpdf PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include> $<INSTALL_INTERFACE:include>)
+
 install(
    TARGETS hpdf
+   EXPORT unofficial-libharu-config
    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
 )
+
+install(EXPORT unofficial-libharu-config
+  NAMESPACE unofficial::libharu::
+  DESTINATION share/unofficial-libharu
+)
