diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2f675d1..e9d6d96 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -244,6 +244,7 @@ set(SEPTEL_ROOT "${CMAKE_CURRENT_SOURCE_DIR}/../septel" CACHE PATH "Path to dire
 option(DISABLE_SNF "Disable Myricom SNF support" OFF)
 
 option(DISABLE_TC "Disable Riverbed TurboCap support" OFF)
+option(DISABLE_PACKET "Disable Packet support" OFF)
 
 #
 # Debugging options.
@@ -302,7 +303,7 @@ if(WIN32)
     if(IS_DIRECTORY ${CMAKE_HOME_DIRECTORY}/../../Common)
         include_directories(${CMAKE_HOME_DIRECTORY}/../../Common)
     endif(IS_DIRECTORY ${CMAKE_HOME_DIRECTORY}/../../Common)
-
+    if(NOT DISABLE_PACKET)
     find_package(Packet)
     if(PACKET_FOUND)
         set(HAVE_PACKET32 TRUE)
@@ -317,7 +318,8 @@ if(WIN32)
         check_function_exists(PacketGetTimestampModes HAVE_PACKET_GET_TIMESTAMP_MODES)
         cmake_pop_check_state()
     endif(PACKET_FOUND)
-
+    endif()
+    
     message(STATUS "checking for Npcap's version.h")
     check_symbol_exists(WINPCAP_PRODUCT_NAME "${CMAKE_SOURCE_DIR}/../../version.h" HAVE_VERSION_H)
     if(HAVE_VERSION_H)
