diff --git a/bindings/wxwidgets/CMakeLists.txt b/bindings/wxwidgets/CMakeLists.txt
index cf631a876..08e6455f1 100644
--- a/bindings/wxwidgets/CMakeLists.txt
+++ b/bindings/wxwidgets/CMakeLists.txt
@@ -73,6 +73,8 @@ if(ENABLE_wxwidgets)
 
   configure_library_build(plplotwxwidgets "" "${plplotwxwidgets_LIB_SRCS}" "PLPLOT::plplot;PLPLOT::plplotcxx;${wxwidgets_LINK_FLAGS}" "${LIB_INSTALL_RPATH}")
 
+  target_compile_features(${WRITEABLE_TARGET}plplotwxwidgets PUBLIC cxx_std_11)
+
   INCLUDE_DIRECTORIES( ${wxWidgets_INCLUDE_DIRS} )
   IF(wxWidgets_DEFINITIONS)
     set_property(TARGET ${WRITEABLE_TARGET}plplotwxwidgets APPEND
diff --git a/utils/CMakeLists.txt b/utils/CMakeLists.txt
index 5772910d2..e8ac61ba4 100644
--- a/utils/CMakeLists.txt
+++ b/utils/CMakeLists.txt
@@ -60,4 +60,5 @@ if(ENABLE_wxwidgets AND NOT OLD_WXWIDGETS)
     PROPERTIES
     WIN32_EXECUTABLE "ON"
     )
+  target_compile_features(${WRITEABLE_TARGET}wxPLViewer PUBLIC cxx_std_11)
 endif(ENABLE_wxwidgets AND NOT OLD_WXWIDGETS)
