diff --git a/src/geo/OCCEdge.cpp b/src/geo/OCCEdge.cpp
index 2ed6e08..248f938 100644
--- a/src/geo/OCCEdge.cpp
+++ b/src/geo/OCCEdge.cpp
@@ -1,4 +1,4 @@
-// Gmsh - Copyright (C) 1997-2025 C. Geuzaine, J.-F. Remacle
+// Gmsh - Copyright (C) 1997-2026 C. Geuzaine, J.-F. Remacle
 //
 // See the LICENSE.txt file in the Gmsh root directory for license information.
 // Please report all issues on https://gitlab.onelab.info/gmsh/gmsh/issues.
@@ -34,6 +34,12 @@
 #include <BRep_Builder.hxx>
 #include <BOPTools_AlgoTools.hxx>
 
+#if OCC_VERSION_HEX < 0x080000
+#include <Geom2dLProp_CLProps2d.hxx>
+#else
+#include <GeomLProp.hxx>
+#endif
+
 OCCEdge::OCCEdge(GModel *m, TopoDS_Edge c, int num, GVertex *v1, GVertex *v2)
   : GEdge(m, num, v1, v2), _c(c), _trimmed(nullptr)
 {
@@ -266,7 +272,7 @@ bool OCCEdge::containsPoint(const SPoint3 &pt) const
   double u;
   SPoint3 xyz;
   if(_project(pt.data(), u, xyz.data())) {
-    const Standard_Real tolerance = BRep_Tool::Tolerance(_c);
+    const double tolerance = BRep_Tool::Tolerance(_c);
     if(pt.distance(xyz) <= tolerance) return true;
   }
   return false;
@@ -411,9 +417,13 @@ double OCCEdge::curvature(double par) const
 
   if(degenerate(0)) return eps;
 
-  Standard_Real Crv;
+  double Crv;
   if(_curve.IsNull()) {
+#if OCC_VERSION_HEX < 0x080000
     Geom2dLProp_CLProps2d aCLProps(_curve2d, 2, eps);
+#else
+    GeomLProp_CLProps2d aCLProps(_curve2d, 2, eps);
+#endif
     aCLProps.SetParameter(par);
     if(!aCLProps.IsTangentDefined())
       Crv = eps;
@@ -458,4 +468,4 @@ void OCCEdge::writeGEO(FILE *fp)
     GEdge::writeGEO(fp);
 }
 
-#endif
+#endif
\ No newline at end of file
