diff --git a/single_header/CMakeLists.txt b/single_header/CMakeLists.txt
index 1d2e280..460e6c7 100644
--- a/single_header/CMakeLists.txt
+++ b/single_header/CMakeLists.txt
@@ -10,7 +10,7 @@ function(add_configuration config_name)
     # If we use FakeIt from install-tree then we use INSTALL_INTERFACE wrapper.
     target_include_directories(FakeIt-${config_name} SYSTEM INTERFACE
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/${config_name}>
-        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/${config_name}>
+        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/fakeit/single_header/${config_name}>
     )
 
     # Add alias for the library with namespace.
@@ -20,7 +20,7 @@ function(add_configuration config_name)
     # "${config_name}" instead of "${config_name}/" string means that the header will be
     # installed at ${CMAKE_INSTALL_PREFIX}/include/${config_name}/fakeit.hpp
     # instead of ${CMAKE_INSTALL_PREFIX}/include/fakeit.hpp.
-    install(DIRECTORY ${config_name} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+    install(DIRECTORY ${config_name} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/fakeit/single_header)
 
     # Add the new config target to the FakeItConfig.cmake file which will be used to find FakeIt from outside.
     install(TARGETS FakeIt-${config_name} EXPORT FakeItConfig)
