diff --git a/CMakeLists.txt b/CMakeLists.txt
index f22584b..1ba8329 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -84,7 +84,7 @@ if(BUILD_SHARED_LIBS)
 		set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /wd4251") # needs to have dll-interface
 	endif()
 
-	target_link_libraries(TinyEXIF tinyxml2)
+	target_link_libraries(TinyEXIF tinyxml2::tinyxml2)
 	set_target_properties(TinyEXIF PROPERTIES
 		COMPILE_DEFINITIONS "TINYEXIF_EXPORT"
 		VERSION "${GENERIC_LIB_VERSION}"
@@ -121,7 +121,7 @@ endif()
 if(BUILD_STATIC_LIBS)
 	add_library(TinyEXIFstatic STATIC TinyEXIF.cpp TinyEXIF.h)
 	
-	target_link_libraries(TinyEXIFstatic tinyxml2)
+	target_link_libraries(TinyEXIFstatic tinyxml2::tinyxml2)
 	set_target_properties(TinyEXIFstatic PROPERTIES
 			OUTPUT_NAME TinyEXIF
 			VERSION "${GENERIC_LIB_VERSION}"
@@ -162,7 +162,7 @@ if(BUILD_DEMO)
 		target_compile_definitions(TinyEXIFdemo PRIVATE TINYEXIF_IMPORT)
 	else(BUILD_STATIC_LIBS)
 		add_dependencies(TinyEXIFdemo TinyEXIFstatic)
-		target_link_libraries(TinyEXIFdemo TinyEXIFstatic tinyxml2)
+		target_link_libraries(TinyEXIFdemo TinyEXIFstatic tinyxml2::tinyxml2)
 	endif()
 endif()
 
