diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7c5720b..72880c4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1156,6 +1156,24 @@ if(INSTALL_PKG_CONFIG_FILES)
     endforeach(versuffix)
 endif(INSTALL_PKG_CONFIG_FILES)
 
+file(WRITE "${CMAKE_INSTALL_PREFIX}/share/unofficial-allegro5/unofficial-allegro5-config.cmake"
+[[
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-allegroTargets.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-allegro_audioTargets.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-allegro_videoTargets.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-allegro_primitivesTargets.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-allegro_fontTargets.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-allegro_ttfTargets.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-allegro_memfileTargets.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-allegro_physfsTargets.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-allegro_imageTargets.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-allegro_mainTargets.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-allegro_dialogTargets.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-allegro_colorTargets.cmake")
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-allegro_acodecTargets.cmake")
+
+]])
+
 #-----------------------------------------------------------------------------#
 #
 #   Documentation
diff --git a/cmake/Common.cmake b/cmake/Common.cmake
index b95479f..4093a47 100644
--- a/cmake/Common.cmake
+++ b/cmake/Common.cmake
@@ -166,6 +166,7 @@ function(add_our_library target framework_name sources extra_flags link_with)
     # Library dependencies are transitive by default.  Any target which links
     # with this target will therefore pull in these dependencies automatically.
     target_link_libraries(${target} ${link_with})
+    target_include_directories(${target} PUBLIC "$<INSTALL_INTERFACE:include>")
 
     # Set list of dependencies that the user would need to explicitly link with
     # if static linking.
@@ -220,6 +221,7 @@ endfunction(set_our_framework_properties)
 
 function(install_our_library target filename)
     install(TARGETS ${target}
+            EXPORT unofficial-${target}Targets
             LIBRARY DESTINATION "lib${LIB_SUFFIX}"
             ARCHIVE DESTINATION "lib${LIB_SUFFIX}"
             FRAMEWORK DESTINATION "${FRAMEWORK_INSTALL_PREFIX}"
@@ -227,6 +229,10 @@ function(install_our_library target filename)
             # Doesn't work, see below.
             # PUBLIC_HEADER DESTINATION "include"
             )
+    install(EXPORT unofficial-${target}Targets
+        NAMESPACE unofficial-allegro5::
+        DESTINATION share/unofficial-allegro5
+    )
     if(0)
         install(FILES ${CMAKE_BINARY_DIR}/lib/\${CMAKE_INSTALL_CONFIG_NAME}/${filename}.pdb
             DESTINATION lib
