diff --git a/configure.ac b/configure.ac
index ff15f2922..c254b3619 100644
--- a/configure.ac
+++ b/configure.ac
@@ -142,6 +142,7 @@ fi
 AC_ARG_WITH([geosconfig],
       [AS_HELP_STRING([--with-geosconfig=FILE], [specify an alternative geos-config file])],
 	[GEOSCONFIG="$withval"], [GEOSCONFIG=""])
+if 0; then
 if test "x$GEOSCONFIG" = "x"; then
       # GEOSCONFIG was not specified, so search within the current path
       AC_PATH_PROG([GEOSCONFIG], [geos-config])	
@@ -164,6 +165,7 @@ fi
 # Extract the linker and include flags
 GEOS_LDFLAGS=`$GEOSCONFIG --ldflags`
 GEOS_CPPFLAGS=-I`$GEOSCONFIG --includes`
+fi
 AC_SUBST([GEOS_LDFLAGS])
 AC_SUBST([GEOS_CPPFLAGS])	
 # Ensure that we can parse geos_c.h
@@ -172,11 +174,14 @@ CPPFLAGS="$GEOS_CPPFLAGS"
 AC_CHECK_HEADERS([geos_c.h],, [AC_MSG_ERROR([could not find geos_c.h - you may need to specify the directory of a geos-config file using --with-geosconfig])])
 CPPFLAGS="$CPPFLAGS_SAVE"	
 # Ensure we can link against libgeos_c
+if 0; then
 LIBS_SAVE="$LIBS"
 LIBS="$GEOS_LDFLAGS"
 AC_SEARCH_LIBS(GEOSTopologyPreserveSimplify,geos_c,,AC_MSG_ERROR([could not find libgeos_c - you may need to specify the directory of a geos-config file using --with-geosconfig]))
 LIBS="$LIBS_SAVE"
 LIBS=$LIBS$GEOS_LDFLAGS' -lgeos_c'
+fi
+LIBS="$LIBS $GEOS_LDFLAGS"
 
 PKG_CHECK_MODULES([LIBSPATIALITE], [spatialite], , AC_MSG_ERROR(['libspatialite' is required but it doesn't seem to be installed on this system.]))
 AC_SUBST(LIBSPATIALITE_CFLAGS)
