diff --git a/extern/glad/CMakeLists.txt b/extern/glad/CMakeLists.txt
index c8c8d86..27adf86 100644
--- a/extern/glad/CMakeLists.txt
+++ b/extern/glad/CMakeLists.txt
@@ -2,15 +2,19 @@ add_library(forge_glad_interface INTERFACE)
 
 target_include_directories(forge_glad_interface
     INTERFACE
-    ${CMAKE_CURRENT_SOURCE_DIR}/include
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
     )
 
 target_sources(forge_glad_interface
     INTERFACE
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/glad.c
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/glad.c>
     )
 
 target_link_libraries(forge_glad_interface
     INTERFACE
-    ${CMAKE_DL_LIBS}
+    $<BUILD_INTERFACE:${CMAKE_DL_LIBS}>
     )
+
+if (NOT BUILD_SHARED_LIBS)
+    install(TARGETS forge_glad_interface EXPORT ForgeTargets)
+endif ()
\ No newline at end of file
