diff --git a/src/backend/opengl/CMakeLists.txt b/src/backend/opengl/CMakeLists.txt
index 1aeec1e..52eff05 100755
--- a/src/backend/opengl/CMakeLists.txt
+++ b/src/backend/opengl/CMakeLists.txt
@@ -142,7 +142,7 @@ install(TARGETS forge
     INCLUDES DESTINATION ${FG_INSTALL_INC_DIR})
 
 # install dependencies
-if (WIN32)
+if (WIN32 AND BUILD_SHARED_LIBS)
     if (FG_WITH_FREEIMAGE AND NOT FG_USE_STATIC_FREEIMAGE)
         install(FILES $<TARGET_FILE:FreeImage::FreeImage>
             DESTINATION ${FG_INSTALL_BIN_DIR}
diff --git a/src/backend/opengl/glfw/CMakeLists.txt b/src/backend/opengl/glfw/CMakeLists.txt
index caf4913..3c8d71c 100644
--- a/src/backend/opengl/glfw/CMakeLists.txt
+++ b/src/backend/opengl/glfw/CMakeLists.txt
@@ -24,7 +24,7 @@ if (NOT BUILD_SHARED_LIBS)
     install(TARGETS forge_wtk_interface EXPORT ForgeTargets)
 endif ()
 
-if (WIN32)
+if (WIN32 AND BUILD_SHARED_LIBS)
     install(FILES $<TARGET_FILE:glfw>
     DESTINATION ${FG_INSTALL_BIN_DIR}
     COMPONENT dependencies)
diff --git a/src/backend/opengl/sdl/CMakeLists.txt b/src/backend/opengl/sdl/CMakeLists.txt
index d8cd312..d55f27d 100644
--- a/src/backend/opengl/sdl/CMakeLists.txt
+++ b/src/backend/opengl/sdl/CMakeLists.txt
@@ -24,7 +24,7 @@ if (NOT BUILD_SHARED_LIBS)
     install(TARGETS forge_wtk_interface EXPORT ForgeTargets)
 endif ()
 
-if (WIN32)
+if (WIN32 AND BUILD_SHARED_LIBS)
     install(FILES $<TARGET_FILE:SDL2::SDL2>
     DESTINATION ${FG_INSTALL_BIN_DIR}
     COMPONENT dependencies)
