diff --git a/caffe2/core/logging.cc b/caffe2/core/logging.cc
index 1379f3a..d1289db 100644
--- a/caffe2/core/logging.cc
+++ b/caffe2/core/logging.cc
@@ -120,17 +120,6 @@ using fLB::FLAGS_logtostderr;
 CAFFE2_DEFINE_int(caffe2_log_level, google::GLOG_ERROR,
                   "The minimum log level that caffe2 will output.");
 
-// Google glog's api does not have an external function that allows one to check
-// if glog is initialized or not. It does have an internal function - so we are
-// declaring it here. This is a hack but has been used by a bunch of others too
-// (e.g. Torch).
-namespace google {
-namespace glog_internal_namespace_ {
-bool IsGoogleLoggingInitialized();
-}  // namespace glog_internal_namespace_
-}  // namespace google
-
-
 namespace caffe2 {
 bool InitCaffeLogging(int* argc, char** argv) {
   if (*argc == 0) return true;
@@ -138,7 +127,7 @@ bool InitCaffeLogging(int* argc, char** argv) {
   // IsGoogleLoggingInitialized is not exported from the glog DLL
   // so we can't call it. If our program calls InitGoogleLogging twice glog will
   // abort it.
-  if (!::google::glog_internal_namespace_::IsGoogleLoggingInitialized())
+  if (!::google::IsGoogleLoggingInitialized())
 #endif
   {
     ::google::InitGoogleLogging(argv[0]);
