diff --git a/CMakeLists.txt b/CMakeLists.txt
index 27b8bd8..ac65300 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,6 +14,7 @@ find_package(OpenGL REQUIRED)
 ##! Library
 add_library(${PROJECT_NAME} SHARED sfml-imgui/imgui-SFML.cpp)
 target_link_libraries(${PROJECT_NAME} PUBLIC imgui::imgui sfml-graphics OpenGL::GL)
+target_compile_features(${PROJECT_NAME} PUBLIC cxx_std_11)
 
 target_include_directories(${PROJECT_NAME} PUBLIC
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/sfml-imgui>
