diff --git a/sfml-imgui/imgui-SFML.cpp b/sfml-imgui/imgui-SFML.cpp
index 36a1a91..cacc9a8 100644
--- a/sfml-imgui/imgui-SFML.cpp
+++ b/sfml-imgui/imgui-SFML.cpp
@@ -159,7 +159,7 @@ namespace
 // Returns first id of connected joystick
     unsigned int getConnectedJoystickId();
 
-    void updateJoystickActionState(ImGuiIO& io, ImGuiNavInput_ action);
+    void updateJoystickActionState(ImGuiIO& io, ImGuiNavInput action);
     void updateJoystickDPadState(ImGuiIO& io);
     void updateJoystickLStickState(ImGuiIO& io);
 }
@@ -734,7 +734,7 @@ namespace
         ImGui::SFML::SetJoytickLStickThreshold(5.f);
     }
 
-    void updateJoystickActionState(ImGuiIO& io, ImGuiNavInput_ action)
+    void updateJoystickActionState(ImGuiIO& io, ImGuiNavInput action)
     {
         bool isPressed = sf::Joystick::isButtonPressed(s_joystickId, s_joystickMapping[action]);
         io.NavInputs[action] = isPressed ? 1.0f : 0.0f;
