Index: a/build/generator/gen_win_dependencies.py
===================================================================
--- a/build/generator/gen_win_dependencies.py	(revision 1888559)
+++ b/build/generator/gen_win_dependencies.py	(working copy)
@@ -168,6 +168,7 @@ class GenDependenciesBase(gen_base.GeneratorBase):
     self.disable_shared = None
     self.static_apr = None
     self.static_openssl = None
+    self.shared_serf = None
     self.instrument_apr_pools = None
     self.instrument_purify_quantify = None
     self.sasl_path = None
@@ -225,8 +226,12 @@ class GenDependenciesBase(gen_base.GeneratorBase):
         self.disable_shared = 1
       elif opt == '--with-static-apr':
         self.static_apr = 1
+        self.shared_serf = 0 # Can't mix apr versions
       elif opt == '--with-static-openssl':
         self.static_openssl = 1
+      elif opt == '--with-shared-serf':
+        if not self.static_apr:
+          self.shared_serf = 1
       elif opt == '-D':
         self.cpp_defines.append(val)
       elif opt == '--vsnet-version':
@@ -1334,6 +1339,9 @@ class GenDependenciesBase(gen_base.GeneratorBase):
       lib_name = 'serf-%d.lib' % (serf_ver_maj,)
     else:
       lib_name = 'serf.lib'
+    
+    if self.shared_serf:
+      lib_name = 'lib' + lib_name
 
     defines = ['SVN_HAVE_SERF', 'SVN_LIBSVN_RA_LINKS_RA_SERF']
 
Index: a/gen-make.py
===================================================================
--- a/gen-make.py	(revision 1888559)
+++ b/gen-make.py	(working copy)
@@ -206,6 +206,9 @@ def _usage_exit(err=None):
   print("  --with-static-openssl")
   print("           Use static openssl")
   print("")
+  print("  --with-shared-serf")
+  print("           Use shared library version of serf")
+  print("")
   print("  --vsnet-version=VER")
   print("           generate for VS.NET version VER (2005-2017 or 9.0-15.0)")
   print("           [implies '-t vcproj']")
@@ -257,6 +260,7 @@ if __name__ == '__main__':
                             'with-apr_memcache=',
                             'with-static-apr',
                             'with-static-openssl',
+                            'with-shared-serf',
                             'enable-pool-debug',
                             'enable-purify',
                             'enable-quantify',
