diff --git a/CMakeLists.txt b/CMakeLists.txt
index d8679fb7f..ca21a7a3d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -36,10 +36,10 @@ if(LIBXSLT_WITH_PYTHON)
 	if(HAVE_UNISTD_H AND HAVE_F_GETFL)
 		find_package(Python COMPONENTS Interpreter Development REQUIRED)
 	else()
-		find_package(Python2 COMPONENTS Interpreter Development REQUIRED)
-		add_library(Python::Python ALIAS Python2::Python)
-		set(Python_EXECUTABLE ${Python2_EXECUTABLE})
-		set(Python_SITEARCH ${Python2_SITEARCH})
+		find_package(Python3 COMPONENTS Interpreter Development REQUIRED)
+		add_library(Python::Python ALIAS Python3::Python)
+		set(Python_EXECUTABLE ${Python3_EXECUTABLE})
+		set(Python_SITEARCH ${Python3_SITEARCH})
 	endif()
 	set(LIBXSLT_PYTHON_INSTALL_DIR ${Python_SITEARCH} CACHE PATH "Python bindings install directory")
 endif()
@@ -244,7 +244,7 @@ set_target_properties(
 	VERSION ${PROJECT_VERSION}
 )
 
-if(MSVC)
+if(0) # Never add suffixes which are not added by the autotools build or the nmake makefiles
 	if(BUILD_SHARED_LIBS)
 		set_target_properties(
 			LibXslt
@@ -327,7 +327,7 @@ set_target_properties(
 	VERSION ${LIBEXSLT_VERSION}
 )
 
-if(MSVC)
+if(0) # same reason as above
 	if(BUILD_SHARED_LIBS)
 		set_target_properties(
 			LibExslt
diff --git a/libexslt.pc.in b/libexslt.pc.in
index 1d605639a..c3a1edb00 100644
--- a/libexslt.pc.in
+++ b/libexslt.pc.in
@@ -7,7 +7,7 @@ includedir=@includedir@
 Name: libexslt
 Version: @LIBEXSLT_VERSION@
 Description: EXSLT Extension library
-Requires: libxml-2.0, libxslt
+Requires: libxml-2.0 libxslt
 Cflags: @EXSLT_INCLUDEDIR@
 Libs: @EXSLT_LIBDIR@ -lexslt
 Libs.private: @EXSLT_PRIVATE_LIBS@
