diff --git a/CMakeLists.txt b/CMakeLists.txt
index d36a049..fe818ff 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,7 +12,7 @@ include(CheckSymbolExists)
 include(CMakePackageConfigHelpers)
 include(GNUInstallDirs)
 
-find_package(LibXml2 CONFIG REQUIRED)
+find_package(LibXml2 REQUIRED)
 
 option(BUILD_SHARED_LIBS "Build shared libraries" ON)
 option(LIBXSLT_WITH_DEBUGGER "Add the debugging support" ON)
diff --git a/libxslt-config.cmake.cmake.in b/libxslt-config.cmake.cmake.in
index 6500370..92c5eb8 100644
--- a/libxslt-config.cmake.cmake.in
+++ b/libxslt-config.cmake.cmake.in
@@ -78,7 +78,7 @@ include(CMakeFindDependencyMacro)
 set(LIBXSLT_SHARED @BUILD_SHARED_LIBS@)
 set(LIBXSLT_WITH_CRYPTO @LIBXSLT_WITH_CRYPTO@)
 
-find_dependency(LibXml2 CONFIG)
+find_dependency(LibXml2)
 list(APPEND LIBXSLT_INCLUDE_DIRS ${LIBXML2_INCLUDE_DIRS})
 list(APPEND LIBXSLT_LIBRARIES ${LIBXML2_LIBRARIES})
 
diff --git a/libxslt-config.cmake.in b/libxslt-config.cmake.in
index 44deee1..8426995 100644
--- a/libxslt-config.cmake.in
+++ b/libxslt-config.cmake.in
@@ -67,7 +67,7 @@ include(CMakeFindDependencyMacro)
 
 set(LIBXSLT_WITH_CRYPTO @WITH_CRYPTO@)
 
-find_dependency(LibXml2 CONFIG)
+find_dependency(LibXml2)
 list(APPEND LIBXSLT_INCLUDE_DIRS ${LIBXML2_INCLUDE_DIRS})
 list(APPEND LIBXSLT_LIBRARIES ${LIBXML2_LIBRARIES})
 list(APPEND LIBXSLT_INTERFACE_LINK_LIBRARIES "LibXml2::LibXml2")
