diff --git a/src/sail-codecs/jpegxl/CMakeLists.txt b/src/sail-codecs/jpegxl/CMakeLists.txt
index 16971a4..0f46e79 100644
--- a/src/sail-codecs/jpegxl/CMakeLists.txt
+++ b/src/sail-codecs/jpegxl/CMakeLists.txt
@@ -1,7 +1,7 @@
 # Don't use SAIL_CODEC_JPEGXL_REQUIRED_OPTION as it requires CMake 3.18
 #
-find_library(JPEGXL_LIBRARY jxl                 NAMES jxl jxl-static)
-find_library(JPEGXL_THREADS_LIBRARY jxl_threads NAMES jxl_threads jxl_threads-static)
+find_library(JPEGXL_LIBRARY jxl                 NAMES jxl)
+find_library(JPEGXL_THREADS_LIBRARY jxl_threads NAMES jxl_threads)
 find_path(JPEGXL_INCLUDE_DIRS jxl/decode.h)
 
 if (NOT JPEGXL_LIBRARY OR NOT JPEGXL_THREADS_LIBRARY OR NOT JPEGXL_INCLUDE_DIRS)
@@ -25,9 +25,6 @@ endif()
 
 # For static builds libjxl requires -DJXL_STATIC_DEFINE
 #
-if (JPEGXL_LIBRARY MATCHES "jxl-static")
-    set(JXL_STATIC_DEFINE "JXL_STATIC_DEFINE")
-endif()
 
 # This will add the following CMake rules to the CMake config for static builds so a client
 # application links against the required dependencies:
@@ -38,8 +35,8 @@ endif()
 #
 # etc.
 #
-set(SAIL_CODECS_FIND_DEPENDENCIES ${SAIL_CODECS_FIND_DEPENDENCIES} "find_library,jxl jxl-static,jxl jxl-static")
-set(SAIL_CODECS_FIND_DEPENDENCIES ${SAIL_CODECS_FIND_DEPENDENCIES} "find_library,jxl_threads jxl_threads-static,jxl_threads jxl_threads-static")
+set(SAIL_CODECS_FIND_DEPENDENCIES ${SAIL_CODECS_FIND_DEPENDENCIES} "find_library,jxl,jxl")
+set(SAIL_CODECS_FIND_DEPENDENCIES ${SAIL_CODECS_FIND_DEPENDENCIES} "find_library,jxl_threads,jxl_threads")
 
 set(SAIL_CODECS_FIND_DEPENDENCIES ${SAIL_CODECS_FIND_DEPENDENCIES} "find_library,hwy,hwy")
 set(SAIL_CODECS_FIND_DEPENDENCIES ${SAIL_CODECS_FIND_DEPENDENCIES} "find_library,brotlicommon brotlicommon-static,brotlicommon brotlicommon-static")
@@ -52,7 +49,6 @@ set(SAIL_CODECS_FIND_DEPENDENCIES ${SAIL_CODECS_FIND_DEPENDENCIES} PARENT_SCOPE)
 sail_codec(NAME jpegxl
             SOURCES helpers.h helpers.c jpegxl.c memory.h memory.c
             ICON jpegxl.png
-            DEPENDENCY_COMPILE_DEFINITIONS ${JXL_STATIC_DEFINE}
             DEPENDENCY_INCLUDE_DIRS ${JPEGXL_INCLUDE_DIRS}
             DEPENDENCY_LIBS ${BROTLI_COMMON_LIBRARY} ${BROTLI_DEC_LIBRARY} ${HWY_LIBRARY}
                             ${JPEGXL_LIBRARY} ${JPEGXL_THREADS_LIBRARY})
