diff --git a/src/sail-codecs-archive/CMakeLists.txt b/src/sail-codecs-archive/CMakeLists.txt
index 0a68aee..4110cb1 100644
--- a/src/sail-codecs-archive/CMakeLists.txt
+++ b/src/sail-codecs-archive/CMakeLists.txt
@@ -69,6 +69,30 @@ if (WIN32 AND NOT BUILD_SHARED_LIBS AND "video" IN_LIST ENABLED_CODECS)
     target_link_libraries(sail-codecs INTERFACE secur32 ncrypt crypt32 bcrypt mfuuid strmiids)
 endif()
 
+# For static macOS builds, FFmpeg requires additional system frameworks
+#
+if (APPLE AND NOT BUILD_SHARED_LIBS AND "video" IN_LIST ENABLED_CODECS)
+    target_link_libraries(sail-codecs INTERFACE
+        "-framework AudioToolbox"
+        "-framework VideoToolbox"
+        "-framework CoreMedia"
+        "-framework CoreVideo"
+        "-framework CoreFoundation"
+    )
+endif()
+
+# For static Android builds, FFmpeg requires additional NDK libraries
+#
+if (ANDROID AND NOT BUILD_SHARED_LIBS AND "video" IN_LIST ENABLED_CODECS)
+    target_link_libraries(sail-codecs INTERFACE mediandk android)
+endif()
+
+# Static libjxl uses Android logging API
+#
+if (ANDROID AND NOT BUILD_SHARED_LIBS AND "jpegxl" IN_LIST ENABLED_CODECS)
+    target_link_libraries(sail-codecs INTERFACE log)
+endif()
+
 target_include_directories(sail-codecs PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../sail)
 
 install(TARGETS sail-codecs
