diff --git a/src/scn/impl/reader/float_reader.cpp b/src/scn/impl/reader/float_reader.cpp
index 095e693..e9f5e51 100644
--- a/src/scn/impl/reader/float_reader.cpp
+++ b/src/scn/impl/reader/float_reader.cpp
@@ -64,14 +64,14 @@ SCN_GCC_POP
 
 #if SCN_DISABLE_LOCALE
 #define SCN_XLOCALE SCN_XLOCALE_DISABLED
-#elif SCN_HAS_INCLUDE(<xlocale.h>)
+#elif (!defined(__ANDROID_API__)||__ANDROID_API__ >= 28 ) && SCN_HAS_INCLUDE(<xlocale.h>)
 #include <xlocale.h>
 #define SCN_XLOCALE SCN_XLOCALE_POSIX
 
 #elif defined(_MSC_VER)
 #define SCN_XLOCALE SCN_XLOCALE_MSVC
 
-#elif defined(__GLIBC__)
+#elif defined(__GLIBC__)&&!defined(__ANDROID_API__) 
 // glibc
 
 #include <features.h>
