diff --git a/include/scn/detail/pp_detect.h b/include/scn/detail/pp_detect.h
index e8c5081..2002559 100644
--- a/include/scn/detail/pp_detect.h
+++ b/include/scn/detail/pp_detect.h
@@ -527,7 +527,7 @@
 // (2021-09-29 is the date of the commit introducing `multiline`,
 //  libstdc++ doesn't support checking for minor versions)
 #if SCN_STDLIB_LIBCPP >= 15000 || SCN_STDLIB_GLIBCXX >= 12 || \
-    (SCN_STDLIB_GLIBCXX == 11 && __GLIBCXX__ >= 20210929L)
+    (SCN_STDLIB_GLIBCXX == 11 && SCN_GCC_COMPAT_MINOR >= 4 && __GLIBCXX__ >= 20210929L)
 #define SCN_HAS_STD_REGEX_MULTILINE 1
 #else
 #define SCN_HAS_STD_REGEX_MULTILINE 0
