diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -323,8 +323,8 @@
 add_subdirectory(dart)
 
 set(DART_IN_SOURCE_BUILD TRUE)
 
-if(TARGET dart)
+if(0)
 
   # Add a "tests" target to build unit tests.
   enable_testing()
diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -19,5 +10,7 @@
 add_subdirectory(dartpy)
+if(0)
 add_subdirectory(tests)
 add_subdirectory(examples)
 add_subdirectory(tutorials)
+endif()
 
