diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0d9f692..feda608 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -454,8 +454,11 @@ endif()
 set( VVENC_LIB_DIR "${CMAKE_CURRENT_SOURCE_DIR}/source/Lib" )

 add_subdirectory( "source/Lib/vvenc" )
+if(BUILD_TOOLS)
 add_subdirectory( "source/App/vvencapp" )
 add_subdirectory( "source/App/vvencFFapp" )
+endif()
+if(BUILD_TESTING)
 add_subdirectory( "test/vvenclibtest" )
 add_subdirectory( "test/vvencinterfacetest" )

@@ -464,6 +467,7 @@ if( NOT BUILD_SHARED_LIBS )
 endif()

 include( cmake/modules/vvencTests.cmake )
+endif()

 if( VVENC_ENABLE_INSTALL )
   # include installer
