diff --git a/modules/UseCorrade.cmake b/modules/UseCorrade.cmake
index 744d5a1..129e75f 100644
--- a/modules/UseCorrade.cmake
+++ b/modules/UseCorrade.cmake
@@ -164,7 +164,10 @@ elseif(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC" OR CMAKE_CXX_SIMULATE_ID STREQUAL "
         # "dllexport and extern are incompatible on an explicit instantiation".
         # Why the error is emitted only on classes? Functions are okay with
         # dllexport extern?!
-        "/wd4910")
+        "/wd4910"
+
+        #Add the /permissive- option
+        "/permissive-")
     set(CORRADE_PEDANTIC_COMPILER_DEFINITIONS
         # Disabling warning for not using "secure-but-not-standard" STL algos
         "_CRT_SECURE_NO_WARNINGS" "_SCL_SECURE_NO_WARNINGS"
