diff --git a/pxr/usd/pcp/primIndex.h b/pxr/usd/pcp/primIndex.h
index dbf1208a26..aa8269a1d3 100644
--- a/pxr/usd/pcp/primIndex.h
+++ b/pxr/usd/pcp/primIndex.h
@@ -70,7 +70,7 @@ public:
     PcpPrimIndex(const PcpPrimIndex& rhs);
 
     /// Move-construction
-    PcpPrimIndex(PcpPrimIndex &&rhs) noexcept = default;
+    PcpPrimIndex(PcpPrimIndex &&rhs) = default;
 
     /// Assignment.
     PcpPrimIndex &operator=(const PcpPrimIndex &rhs) {
@@ -79,7 +79,7 @@ public:
     }
 
     // Move-assignment.
-    PcpPrimIndex &operator=(PcpPrimIndex &&rhs) noexcept = default;
+    PcpPrimIndex &operator=(PcpPrimIndex &&rhs) = default;
 
     /// Swap the contents of this prim index with \p index.
     PCP_API
