diff --git a/pxr/pxrConfig.cmake.in b/pxr/pxrConfig.cmake.in
index 2aa4a3bd3..2f89d690c 100644
--- a/pxr/pxrConfig.cmake.in
+++ b/pxr/pxrConfig.cmake.in
@@ -25,6 +25,9 @@ if(@PXR_BUILD_IMAGING@)
     if(@PXR_BUILD_OPENIMAGEIO_PLUGIN@)
         find_dependency(OpenImageIO CONFIG)
     endif()
+    if(@PXR_ENABLE_GL_SUPPORT@)
+        find_dependency(OpenGL REQUIRED)
+    endif()
     if(@PXR_ENABLE_VULKAN_SUPPORT@)
         if (NOT DEFINED Vulkan_DIR)
             if (NOT [[@Vulkan_DIR@]] STREQUAL "")
