diff --git a/OGLCompilersDLL/CMakeLists.txt b/OGLCompilersDLL/CMakeLists.txt
index af4ab587..21d02bde 100644
--- a/OGLCompilersDLL/CMakeLists.txt
+++ b/OGLCompilersDLL/CMakeLists.txt
@@ -11,5 +11,7 @@ endif(WIN32)
 if(ENABLE_GLSLANG_INSTALL)
     install(TARGETS OGLCompiler EXPORT OGLCompilerTargets
             ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
-	install(EXPORT OGLCompilerTargets DESTINATION lib/cmake)
+    install(EXPORT OGLCompilerTargets
+            DESTINATION "${CMAKE_INSTALL_PREFIX}/share/glslang"
+            NAMESPACE glslang::)
 endif(ENABLE_GLSLANG_INSTALL)
diff --git a/SPIRV/CMakeLists.txt b/SPIRV/CMakeLists.txt
index e25ec0a1..4d73f9e3 100644
--- a/SPIRV/CMakeLists.txt
+++ b/SPIRV/CMakeLists.txt
@@ -91,8 +91,12 @@ if(ENABLE_GLSLANG_INSTALL)
                 ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
     endif()
 	
-	install(EXPORT SPVRemapperTargets DESTINATION lib/cmake)
-	install(EXPORT SPIRVTargets DESTINATION lib/cmake)
+    install(EXPORT SPVRemapperTargets
+            DESTINATION "${CMAKE_INSTALL_PREFIX}/share/glslang"
+            NAMESPACE glslang::)
+    install(EXPORT SPIRVTargets
+            DESTINATION "${CMAKE_INSTALL_PREFIX}/share/glslang"
+            NAMESPACE glslang::)
 
     install(FILES ${HEADERS} ${SPVREMAP_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/glslang/SPIRV/)
 endif(ENABLE_GLSLANG_INSTALL)
diff --git a/StandAlone/CMakeLists.txt b/StandAlone/CMakeLists.txt
index 0878965c..3d830f1e 100644
--- a/StandAlone/CMakeLists.txt
+++ b/StandAlone/CMakeLists.txt
@@ -49,12 +49,18 @@ if(ENABLE_GLSLANG_INSTALL)
     install(TARGETS spirv-remap EXPORT spirv-remapTargets
             RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
 	
-	install(EXPORT glslangValidatorTargets DESTINATION lib/cmake)
-	install(EXPORT spirv-remapTargets DESTINATION lib/cmake)
+    install(EXPORT glslangValidatorTargets 
+            DESTINATION "${CMAKE_INSTALL_PREFIX}/share/glslang"
+            NAMESPACE glslang::)
+    install(EXPORT spirv-remapTargets
+            DESTINATION "${CMAKE_INSTALL_PREFIX}/share/glslang"
+            NAMESPACE glslang::)
             
     if(BUILD_SHARED_LIBS)
         install(TARGETS glslang-default-resource-limits EXPORT glslang-default-resource-limitsTargets
                 LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
-		install(EXPORT glslang-default-resource-limitsTargets DESTINATION lib/cmake)
+        install(EXPORT glslang-default-resource-limitsTargets
+                DESTINATION "${CMAKE_INSTALL_PREFIX}/share/glslang"
+                NAMESPACE glslang::)
     endif()
 endif(ENABLE_GLSLANG_INSTALL)
diff --git a/glslang/CMakeLists.txt b/glslang/CMakeLists.txt
index 42df1d1e..0bd9bbee 100644
--- a/glslang/CMakeLists.txt
+++ b/glslang/CMakeLists.txt
@@ -112,7 +112,9 @@ if(ENABLE_GLSLANG_INSTALL)
         install(TARGETS glslang EXPORT glslangTargets
                 ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
     endif()
-	install(EXPORT glslangTargets DESTINATION lib/cmake)
+    install(EXPORT glslangTargets
+            DESTINATION "${CMAKE_INSTALL_PREFIX}/share/glslang" 
+            NAMESPACE glslang::)
 endif(ENABLE_GLSLANG_INSTALL)
 
 if(ENABLE_GLSLANG_INSTALL)
diff --git a/glslang/OSDependent/Unix/CMakeLists.txt b/glslang/OSDependent/Unix/CMakeLists.txt
index 91fb45a7..092967f0 100644
--- a/glslang/OSDependent/Unix/CMakeLists.txt
+++ b/glslang/OSDependent/Unix/CMakeLists.txt
@@ -22,5 +22,7 @@ endif()
 if(ENABLE_GLSLANG_INSTALL)
     install(TARGETS OSDependent EXPORT OSDependentTargets
             ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
-	install(EXPORT OSDependentTargets DESTINATION lib/cmake)
+    install(EXPORT OSDependentTargets
+            DESTINATION "${CMAKE_INSTALL_PREFIX}/share/glslang"
+            NAMESPACE glslang::)
 endif(ENABLE_GLSLANG_INSTALL)
diff --git a/glslang/OSDependent/Windows/CMakeLists.txt b/glslang/OSDependent/Windows/CMakeLists.txt
index f6bd4e9d..befb9a59 100644
--- a/glslang/OSDependent/Windows/CMakeLists.txt
+++ b/glslang/OSDependent/Windows/CMakeLists.txt
@@ -17,5 +17,7 @@ endif(WIN32)
 if(ENABLE_GLSLANG_INSTALL)
     install(TARGETS OSDependent EXPORT OSDependentTargets
             ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
-	install(EXPORT OSDependentTargets DESTINATION lib/cmake)
+    install(EXPORT OSDependentTargets
+            DESTINATION "${CMAKE_INSTALL_PREFIX}/share/glslang"
+            NAMESPACE glslang::)
 endif(ENABLE_GLSLANG_INSTALL)
diff --git a/gtests/CMakeLists.txt b/gtests/CMakeLists.txt
index 77d217f1..0842a0a3 100644
--- a/gtests/CMakeLists.txt
+++ b/gtests/CMakeLists.txt
@@ -33,7 +33,9 @@ if(BUILD_TESTING)
         if(ENABLE_GLSLANG_INSTALL)
             install(TARGETS glslangtests EXPORT glslangtestsTargets
                     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
-			install(EXPORT glslangtestsTargets DESTINATION lib/cmake)
+            install(EXPORT glslangtestsTargets
+                    DESTINATION "${CMAKE_INSTALL_PREFIX}/share/glslang"
+                    NAMESPACE glslang::)
         endif(ENABLE_GLSLANG_INSTALL)
 
         set(GLSLANG_TEST_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/../Test")
diff --git a/hlsl/CMakeLists.txt b/hlsl/CMakeLists.txt
index 44f9d6a2..2e99b0f2 100644
--- a/hlsl/CMakeLists.txt
+++ b/hlsl/CMakeLists.txt
@@ -41,5 +41,7 @@ if(ENABLE_GLSLANG_INSTALL)
         install(TARGETS HLSL EXPORT HLSLTargets
                 ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
     endif()
-	install(EXPORT HLSLTargets DESTINATION lib/cmake)
+    install(EXPORT HLSLTargets
+            DESTINATION "${CMAKE_INSTALL_PREFIX}/share/glslang"
+            NAMESPACE glslang::)
 endif(ENABLE_GLSLANG_INSTALL)
