diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index a682ed4..e273393 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -61,6 +61,9 @@ if(MSVC)
     list(APPEND VORBIS_SOURCES ../win32/vorbis.def)
     list(APPEND VORBISENC_SOURCES ../win32/vorbisenc.def)
     list(APPEND VORBISFILE_SOURCES ../win32/vorbisfile.def)
+    add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+    add_definitions(-D_CRT_SECURE_NO_DEPRECATE)
+    add_definitions(-D_CRT_NONSTDC_NO_DEPRECATE)
 endif()
 
 include_directories(../include)
@@ -68,7 +71,7 @@ include_directories(.)
 include_directories(${OGG_INCLUDE_DIRS})
 
 if (NOT BUILD_FRAMEWORK)
-    add_library(vorbis ${VORBIS_HEADERS} ${VORBIS_SOURCES})
+    add_library(vorbis ${VORBIS_HEADERS} ${VORBIS_SOURCES} ${VORBISENC_SOURCES})
     add_library(vorbisenc ${VORBISENC_SOURCES})
     add_library(vorbisfile ${VORBISFILE_SOURCES})
 
