diff --git a/CMakeLists.txt b/CMakeLists.txt
index bbc045ba..a23630f5 100644
--- a/CMakeLists.txt	
+++ b/CMakeLists.txt
@@ -57,10 +57,10 @@ if(NOT OGG_ROOT)
     find_package(PkgConfig QUIET)
     pkg_check_modules(PC_OGG QUIET ogg)
     find_path(OGG_INCLUDE_DIRS NAMES ogg/ogg.h HINTS ${PC_OGG_INCLUDE_DIRS} PATH_SUFFIXES ogg)
-    find_library(OGG_LIBRARIES NAMES ogg HINTS ${PC_OGG_LIBRARY_DIRS})
+    find_library(OGG_LIBRARIES NAMES ogg libogg HINTS ${PC_OGG_LIBRARY_DIRS})
 else()
     find_path(OGG_INCLUDE_DIRS NAMES ogg/ogg.h HINTS ${OGG_ROOT}/include PATH_SUFFIXES ogg)
-    find_library(OGG_LIBRARIES NAMES ogg HINTS ${OGG_ROOT}/lib ${OGG_ROOT}/lib64)
+    find_library(OGG_LIBRARIES NAMES ogg libogg HINTS ${OGG_ROOT}/lib ${OGG_ROOT}/lib64)
 endif()
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(OGG DEFAULT_MSG OGG_INCLUDE_DIRS OGG_LIBRARIES)
