diff --git a/KF5NewStuffConfig.cmake.in b/KF5NewStuffConfig.cmake.in
index 063b83ca..1be53741 100644
--- a/KF5NewStuffConfig.cmake.in
+++ b/KF5NewStuffConfig.cmake.in
@@ -13,7 +13,6 @@ find_dependency(KF5NewStuffQuick "@KF_VERSION@")
 if (NOT @BUILD_SHARED_LIBS@)
     find_dependency(Qt5Qml @REQUIRED_QT_VERSION@)
     find_dependency(Qt5Quick @REQUIRED_QT_VERSION@)
-    find_dependency(Qt5Gui @REQUIRED_QT_VERSION@)
 
     find_dependency(KF5KIO "@KF_DEP_VERSION@")
     find_dependency(KF5Completion "@KF_DEP_VERSION@")
@@ -21,13 +20,6 @@ if (NOT @BUILD_SHARED_LIBS@)
     find_dependency(KF5ItemViews "@KF_DEP_VERSION@")
     find_dependency(KF5IconThemes "@KF_DEP_VERSION@")
     find_dependency(KF5TextWidgets "@KF_DEP_VERSION@")
-    find_dependency(KF5Archive "@KF_DEP_VERSION@")
-    find_dependency(KF5Package "@KF_DEP_VERSION@")
-    find_dependency(KF5Config "@KF_DEP_VERSION@")
-
-    if(@KF5Syndication_FOUND@)
-            find_dependency(KF5Syndication "@KF_VERSION@")
-    endif()
 endif()
 
 include("${CMAKE_CURRENT_LIST_DIR}/KF5NewStuffTargets.cmake")
diff --git a/KF5NewStuffCoreConfig.cmake.in b/KF5NewStuffCoreConfig.cmake.in
index c65c7c42..b395f8cd 100644
--- a/KF5NewStuffCoreConfig.cmake.in
+++ b/KF5NewStuffCoreConfig.cmake.in
@@ -3,6 +3,19 @@
 include(CMakeFindDependencyMacro)
 find_dependency(KF5Attica "@KF_DEP_VERSION@")
 
+if (NOT @BUILD_SHARED_LIBS@)
+    find_dependency(Qt5Gui @REQUIRED_QT_VERSION@)
+
+    find_dependency(KF5I18n "@KF_DEP_VERSION@")
+    find_dependency(KF5Archive "@KF_DEP_VERSION@")
+    find_dependency(KF5Package "@KF_DEP_VERSION@")
+    find_dependency(KF5Config "@KF_DEP_VERSION@")
+
+    if(@KF5Syndication_FOUND@)
+            find_dependency(KF5Syndication "@KF_VERSION@")
+    endif()
+endif()
+
 set(KDE_INSTALL_KNSRCDIR "@KDE_INSTALL_DATADIR@/knsrcfiles")
 
 include("${CMAKE_CURRENT_LIST_DIR}/KF5NewStuffCoreTargets.cmake")
-- 
GitLab

