diff --git a/av/src/AudioDecoder.cc b/av/src/AudioDecoder.cc
index 32fef077..dcf3f473 100644
--- a/av/src/AudioDecoder.cc
+++ b/av/src/AudioDecoder.cc
@@ -345,8 +345,10 @@ bool AudioDecoder::SetFile(const std::string &_filename)
   }
 
 #if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(56, 60, 100)
+#if LIBAVCODEC_VERSION_MAJOR < 60
   if (this->data->codec->capabilities & AV_CODEC_CAP_TRUNCATED)
     this->data->codecCtx->flags |= AV_CODEC_FLAG_TRUNCATED;
+#endif
 #else
   if (this->data->codec->capabilities & CODEC_CAP_TRUNCATED)
     this->data->codecCtx->flags |= CODEC_FLAG_TRUNCATED;
diff --git a/av/src/Video.cc b/av/src/Video.cc
index 2e916f08..95d4cbdc 100644
--- a/av/src/Video.cc
+++ b/av/src/Video.cc
@@ -182,8 +182,10 @@ bool Video::Load(const std::string &_filename)
   // Inform the codec that we can handle truncated bitstreams -- i.e.,
   // bitstreams where frame boundaries can fall in the middle of packets
 #if LIBAVCODEC_VERSION_INT >= AV_VERSION_INT(56, 60, 100)
+#if LIBAVCODEC_VERSION_MAJOR < 60
   if (codec->capabilities & AV_CODEC_CAP_TRUNCATED)
     this->dataPtr->codecCtx->flags |= AV_CODEC_FLAG_TRUNCATED;
+#endif
 #else
   if (codec->capabilities & CODEC_CAP_TRUNCATED)
     this->dataPtr->codecCtx->flags |= CODEC_FLAG_TRUNCATED;
