diff --git a/KF5KCMUtilsConfig.cmake.in b/KF5KCMUtilsConfig.cmake.in
index 9871fba..ddcf6ea 100644
--- a/KF5KCMUtilsConfig.cmake.in
+++ b/KF5KCMUtilsConfig.cmake.in
@@ -4,6 +4,20 @@ include(CMakeFindDependencyMacro)
 find_dependency(KF5ConfigWidgets  "@KF_DEP_VERSION@")
 find_dependency(KF5Service        "@KF_DEP_VERSION@")
 
+if (NOT @BUILD_SHARED_LIBS@)
+    find_dependency(Qt5DBus         "@REQUIRED_QT_VERSION@")
+    find_dependency(Qt5Qml          "@REQUIRED_QT_VERSION@")
+    find_dependency(Qt5Quick        "@REQUIRED_QT_VERSION@")
+    find_dependency(Qt5QuickWidgets "@REQUIRED_QT_VERSION@")
+
+    find_dependency(KF5CoreAddons   "@KF_DEP_VERSION@")
+    find_dependency(KF5GuiAddons    "@KF_DEP_VERSION@")
+    find_dependency(KF5I18n         "@KF_DEP_VERSION@")
+    find_dependency(KF5ItemViews    "@KF_DEP_VERSION@")
+    find_dependency(KF5XmlGui       "@KF_DEP_VERSION@")
+    find_dependency(KF5Declarative  "@KF_DEP_VERSION@")
+endif()
+
 @PACKAGE_SETUP_AUTOMOC_VARIABLES@
 
 include("${CMAKE_CURRENT_LIST_DIR}/KF5KCMUtilsTargets.cmake")
-- 
GitLab

