diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index beaa54be..68aa11c8 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -41,9 +41,9 @@ install(FILES
   DESTINATION ${KDE_INSTALL_INCLUDEDIR_KF}/KCMUtilsCore COMPONENT Devel
 )
 
-install(TARGETS KF5KCMUtilsCore EXPORT KF5KCMUtilsTargets ${KF5_INSTALL_TARGETS_DEFAULT_ARGS})
-
 add_library(kcmutils_proxy_model STATIC kpluginproxymodel.cpp)
 # Needed to link this static lib to shared libs
 set_property(TARGET kcmutils_proxy_model PROPERTY POSITION_INDEPENDENT_CODE ON)
 target_link_libraries(kcmutils_proxy_model PUBLIC KF5KCMUtilsCore KF5::ItemViews)
+
+install(TARGETS KF5KCMUtilsCore kcmutils_proxy_model EXPORT KF5KCMUtilsTargets ${KF5_INSTALL_TARGETS_DEFAULT_ARGS})
-- 
