diff --git a/cmake/FindZSTD.cmake b/cmake/FindZSTD.cmake
index 11515c8a3..44d107f8d 100644
--- a/cmake/FindZSTD.cmake
+++ b/cmake/FindZSTD.cmake
@@ -18,32 +18,68 @@
 # ``ZSTD::ZSTD``
 #     The zstd library
 
-find_package(PkgConfig QUIET)
-pkg_check_modules(PC_ZSTD QUIET libzstd)
-
-find_path(ZSTD_INCLUDE_DIRS
-          NAMES zstd.h
-          HINTS ${PC_ZSTD_INCLUDEDIR}
-          PATH_SUFFIXES zstd)
-
-find_library(ZSTD_LIBRARIES
-             NAMES zstd zstd_static
-             HINTS ${PC_ZSTD_LIBDIR}
-)
-
-include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(ZSTD REQUIRED_VARS ZSTD_LIBRARIES ZSTD_INCLUDE_DIRS
-                                       VERSION_VAR PC_ZSTD_VERSION)
-
-if(ZSTD_FOUND AND NOT TARGET ZSTD::ZSTD)
-  add_library(ZSTD::ZSTD UNKNOWN IMPORTED)
-  set_target_properties(ZSTD::ZSTD PROPERTIES
-                        IMPORTED_LOCATION "${ZSTD_LIBRARIES}"
-                        INTERFACE_INCLUDE_DIRECTORIES "${ZSTD_INCLUDE_DIRS}")
-endif()
+find_package(ZSTD CONFIG)
+
+if(TARGET zstd::libzstd_static OR TARGET zstd::libzstd_shared)
+    include(FindPackageHandleStandardArgs)
+    find_package_handle_standard_args(ZSTD CONFIG_MODE)
+
+    set(zstdtargetsuffix "_shared")
+    if(TARGET zstd::libzstd_static)
+        set(zstdtargetsuffix "_static")
+    endif()
+    if(NOT TARGET ZSTD::ZSTD)
+        add_library(ZSTD::ZSTD INTERFACE IMPORTED)
+        set_target_properties(ZSTD::ZSTD PROPERTIES
+                              INTERFACE_LINK_LIBRARIES "zstd::libzstd${zstdtargetsuffix}")
+    endif()
+elseif(TARGET zstd::libzstd)
+    if(NOT TARGET ZSTD::ZSTD)
+        add_library(ZSTD::ZSTD INTERFACE IMPORTED)
+        set_target_properties(ZSTD::ZSTD PROPERTIES
+                              INTERFACE_LINK_LIBRARIES "zstd::libzstd")
+    endif()
+else()
+    find_package(PkgConfig QUIET)
+    pkg_check_modules(PC_ZSTD QUIET libzstd)
+    
+    find_path(ZSTD_INCLUDE_DIRS
+              NAMES zstd.h
+              HINTS ${PC_ZSTD_INCLUDEDIR}
+              PATH_SUFFIXES zstd)
+
+    find_library(ZSTD_LIBRARY_RELEASE
+                 NAMES zstd zstd_static
+                 HINTS ${PC_ZSTD_LIBDIR}
+    )
+    find_library(ZSTD_LIBRARY_DEBUG
+                 NAMES zstdd zstd_staticd zstd zstd_static
+                 HINTS ${PC_ZSTD_LIBDIR}
+    )
 
-mark_as_advanced(ZSTD_INCLUDE_DIRS ZSTD_LIBRARIES)
+    include(SelectLibraryConfigurations)
+    select_library_configurations(ZSTD)
 
+    include(FindPackageHandleStandardArgs)
+    find_package_handle_standard_args(ZSTD REQUIRED_VARS ZSTD_LIBRARIES ZSTD_INCLUDE_DIRS
+                                           VERSION_VAR PC_ZSTD_VERSION)
+
+    if(ZSTD_FOUND AND NOT TARGET ZSTD::ZSTD)
+      add_library(ZSTD::ZSTD UNKNOWN IMPORTED)
+      set_target_properties(ZSTD::ZSTD PROPERTIES
+                            INTERFACE_INCLUDE_DIRECTORIES "${ZSTD_INCLUDE_DIRS}")
+      if(ZSTD_LIBRARY_RELEASE)
+          set_target_properties(ZSTD::ZSTD PROPERTIES
+                                IMPORTED_LOCATION_RELEASE "${ZSTD_LIBRARY_RELEASE}")
+      endif()
+      if(ZSTD_LIBRARY_DEBUG)
+          set_target_properties(ZSTD::ZSTD PROPERTIES
+                                IMPORTED_LOCATION_DEBUG "${ZSTD_LIBRARY_DEBUG}")
+      endif()
+    endif()
+
+    mark_as_advanced(ZSTD_INCLUDE_DIRS ZSTD_LIBRARIES ZSTD_LIBRARY_RELEASE ZSTD_LIBRARY_DEBUG)
+endif()
 include(FeatureSummary)
 set_package_properties(ZSTD PROPERTIES
   URL "https://github.com/facebook/zstd"
