diff --git a/cmake/QtBuildInternalsExtra.cmake.in b/cmake/QtBuildInternalsExtra.cmake.in
index 21c40f2..c212148 100644
--- a/cmake/QtBuildInternalsExtra.cmake.in
+++ b/cmake/QtBuildInternalsExtra.cmake.in
@@ -162,6 +162,7 @@ function(qt_internal_force_set_cmake_build_type_conditionally value)
     # STREQUAL check needs to be expanded variables because an undefined var is not equal to an
     # empty defined var.
     if("${CMAKE_BUILD_TYPE}" STREQUAL "${CMAKE_BUILD_TYPE_INIT}"
+        AND NOT DEFINED CACHE{CMAKE_BUILD_TYPE}
         AND NOT __qt_toolchain_cmake_build_type_before_project_call
         AND NOT QT_NO_FORCE_SET_CMAKE_BUILD_TYPE
         AND NOT __qt_internal_extras_is_multi_config)
diff --git a/cmake/QtSetup.cmake b/cmake/QtSetup.cmake
index 762299f..f104479 100644
--- a/cmake/QtSetup.cmake
+++ b/cmake/QtSetup.cmake
@@ -56,6 +56,7 @@ get_property(QT_GENERATOR_IS_MULTI_CONFIG GLOBAL PROPERTY GENERATOR_IS_MULTI_CON
 # See also qt_internal_force_set_cmake_build_type_conditionally which is used
 # to set the build type when building other repos or tests.
 if("${CMAKE_BUILD_TYPE}" STREQUAL "${CMAKE_BUILD_TYPE_INIT}"
+    AND NOT DEFINED CACHE{CMAKE_BUILD_TYPE}
     AND NOT __qt_auto_detect_cmake_build_type_before_project_call
     AND NOT __qt_build_internals_cmake_build_type
     AND NOT CMAKE_CONFIGURATION_TYPES)
