diff --git a/cmake/QtFlagHandlingHelpers.cmake b/cmake/QtFlagHandlingHelpers.cmake
index 18ea9c2d9..52e69c108 100644
--- a/cmake/QtFlagHandlingHelpers.cmake	
+++ b/cmake/QtFlagHandlingHelpers.cmake
@@ -437,7 +437,7 @@ function(qt_internal_remove_flags_impl flag_var_name flag_values IN_CACHE)
         # Work exclusively on cache variable value only.
         set(mod_flags $CACHE{${flag_var_name}})
         foreach(flag_value IN LISTS flag_values)
-            string(${replace_type} "${flag_value}" "" mod_flags "${mod_flags}")
+            string(${replace_type} "${flag_value}" " " mod_flags "${mod_flags}")
         endforeach()
         string(STRIP "${mod_flags}" mod_flags)
         get_property(help_text CACHE ${flag_var_name} PROPERTY HELPSTRING)
