diff --git a/cmake/3rdparty/extra-cmake-modules/find-modules/FindEGL.cmake b/cmake/3rdparty/extra-cmake-modules/find-modules/FindEGL.cmake
index eafbc1ff9..dd4068838 100644
--- a/cmake/3rdparty/extra-cmake-modules/find-modules/FindEGL.cmake
+++ b/cmake/3rdparty/extra-cmake-modules/find-modules/FindEGL.cmake	
@@ -60,6 +60,13 @@
 # THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #=============================================================================
 
+find_package(unofficial-angle CONFIG)
+set(EGL_FOUND "${unofficial-angle_FOUND}")
+if(EGL_FOUND AND NOT TARGET EGL::EGL)
+  add_library(EGL::EGL INTERFACE IMPORTED)
+  target_link_libraries(EGL::EGL INTERFACE unofficial::angle::libEGL)
+endif()
+return()
 include(${CMAKE_CURRENT_LIST_DIR}/ECMFindModuleHelpersStub.cmake)
 include(CheckCXXSourceCompiles)
 include(CMakePushCheckState)
