diff --git a/src/plugins/imageformats/jpeg/qjpeghandler.cpp b/src/plugins/imageformats/jpeg/qjpeghandler.cpp
index 29bf5ab58..1a3cfd2bf 100644
--- a/src/plugins/imageformats/jpeg/qjpeghandler.cpp
+++ b/src/plugins/imageformats/jpeg/qjpeghandler.cpp
@@ -63,12 +63,12 @@
 extern "C" {
 // jpeglib.h->jmorecfg.h tries to typedef int boolean; but this conflicts with
 // some Windows headers that may or may not have been included
-#ifdef HAVE_BOOLEAN
-#  undef HAVE_BOOLEAN
-#endif
-#define boolean jboolean
+//#ifdef HAVE_BOOLEAN
+//#  undef HAVE_BOOLEAN
+//#endif
+//#define boolean jboolean
 
-#define XMD_H           // shut JPEGlib up
+//#define XMD_H           // shut JPEGlib up
 #include <jpeglib.h>
 #ifdef const
 #  undef const          // remove crazy C hackery in jconfig.h
