diff --git a/mkspecs/features/moc.prf b/mkspecs/features/moc.prf
index c5504b21..3e4eaa75 100644
--- a/mkspecs/features/moc.prf
+++ b/mkspecs/features/moc.prf
@@ -33,7 +33,7 @@ if(gcc|intel_icl|msvc):!rim_qcc:!uikit:!no_moc_predefs:if(!macos|count(QMAKE_APP
     gcc:            moc_predefs.commands = $$QMAKE_CXX $$QMAKE_CXXFLAGS -dM -E -o ${QMAKE_FILE_OUT} ${QMAKE_FILE_IN}
     else:intel_icl: moc_predefs.commands = $$QMAKE_CXX $$QMAKE_CXXFLAGS -QdM -P -Fi${QMAKE_FILE_OUT} ${QMAKE_FILE_IN}
     else:msvc {
-        moc_predefs.commands += $$QMAKE_CXX -Bx$$shell_quote($$shell_path($$QMAKE_QMAKE)) $$QMAKE_CXXFLAGS \
+        moc_predefs.commands += $$shell_quote($$QMAKE_CXX) -Bx$$shell_quote($$shell_path($$QMAKE_QMAKE)) $$QMAKE_CXXFLAGS \
                                     -E ${QMAKE_FILE_IN} 2>NUL >${QMAKE_FILE_OUT}
     } else:         error("Oops, I messed up")
     moc_predefs.output = $$MOC_DIR/moc_predefs.h
