diff --git a/CMakeLists.txt b/CMakeLists.txt
index f2eed91..f329996 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -224,6 +224,9 @@ target_compile_definitions(${PROJECT_NAME} PRIVATE ${LXW_PRIVATE_COMPILE_DEFINIT
 if(MSVC AND NOT (MSVC_VERSION LESS 1900))
     target_compile_options(${PROJECT_NAME} PRIVATE /utf-8)
 endif()
+if (WINDOWSSTORE)
+  target_compile_definitions(${PROJECT_NAME} PRIVATE -DIOWIN32_USING_WINRT_API)
+endif()
 target_include_directories(${PROJECT_NAME}
     PRIVATE ${LXW_PRIVATE_INCLUDE_DIRS}
     PUBLIC include include/xlsxwriter
@@ -319,31 +322,11 @@ endif()
 
 include(GNUInstallDirs)
 
-if(MSVC)
-    if(CMAKE_CL_64)
-        set(MSVC_FOLDER_PREFIX x64)
-    else()
-        set(MSVC_FOLDER_PREFIX Win32)
-    endif()
-
-    install(TARGETS ${PROJECT_NAME}
-        LIBRARY DESTINATION "lib/${MSVC_FOLDER_PREFIX}/\${CMAKE_INSTALL_CONFIG_NAME}"
-        ARCHIVE DESTINATION "lib/${MSVC_FOLDER_PREFIX}/\${CMAKE_INSTALL_CONFIG_NAME}"
-        RUNTIME DESTINATION "bin/${MSVC_FOLDER_PREFIX}/\${CMAKE_INSTALL_CONFIG_NAME}"
-    )
-    if (NOT BUILD_SHARED_LIBS)
-        install(FILES ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pdb
-            DESTINATION "lib/${MSVC_FOLDER_PREFIX}/\${CMAKE_INSTALL_CONFIG_NAME}"
-        )
-    endif()
-
-else(MSVC)
-    install(TARGETS ${PROJECT_NAME}
-        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-    )
-endif(MSVC)
+install(TARGETS ${PROJECT_NAME}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+)
 install(FILES include/xlsxwriter.h DESTINATION include)
 install(DIRECTORY include/xlsxwriter
     DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
