diff --git a/CMakeLists.txt b/CMakeLists.txt
index a7c70d1..4aade40 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -110,12 +110,11 @@ target_include_directories(
     INTERFACE "${CMAKE_CURRENT_LIST_DIR}/include"
     PRIVATE "${CMAKE_CURRENT_LIST_DIR}/include")
 
-if (APPLE OR (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION
+if (APPLE OR ANDROID OR (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION
     VERSION_LESS 12))
 
-    add_subdirectory(externals/fmt)
-    set(FMT_INCLUDE_DIR "${CMAKE_SOURCE_DIR}/externals/fmt/include")
-    target_link_libraries(${LIBRARY_NAME} PRIVATE fmt)
+    find_package(fmt CONFIG REQUIRED)
+    target_link_libraries(${LIBRARY_NAME} PRIVATE fmt::fmt)
 endif ()
 
 if (BUILD_FAKER_TESTS)
