diff --git a/src/draco/io/parser_utils.cc b/src/draco/io/parser_utils.cc
index 0a22ba1..f67c66d 100644
--- a/src/draco/io/parser_utils.cc
+++ b/src/draco/io/parser_utils.cc
@@ -151,6 +151,9 @@ bool ParseSignedInt(DecoderBuffer *buffer, int32_t *value) {
   if (!ParseUnsignedInt(buffer, &v))
     return false;
   *value = (sign < 0) ? -v : v;
+  if (sign < 0)
+    v *= -1;
+  *value = v;
   return true;
 }
 
