diff --git a/src/draco/core/hash_utils.h b/src/draco/core/hash_utils.h
index dd910d0..aa61523 100644
--- a/src/draco/core/hash_utils.h
+++ b/src/draco/core/hash_utils.h
@@ -17,10 +17,9 @@
 
 #include <stdint.h>
 
+#include <cstddef>
 #include <functional>
 
-// TODO(fgalligan): Move this to core.
-
 namespace draco {
 
 template <typename T1, typename T2>
diff --git a/src/draco/io/parser_utils.cc b/src/draco/io/parser_utils.cc
index 3c302b9..adca2ac 100644
--- a/src/draco/io/parser_utils.cc
+++ b/src/draco/io/parser_utils.cc
@@ -18,6 +18,7 @@
 #include <cctype>
 #include <cmath>
 #include <iterator>
+#include <limits>
 
 namespace draco {
 namespace parser {
