--- a/magick/studio.h	2019-12-24 13:18:50.000000000 -0200
+++ b/magick/studio.h	2020-01-08 20:43:08.325889700 -0200
@@ -470,10 +470,10 @@
 */
 
 #if defined(HasLTDL)
-#  define SupportMagickModules
+//#  define SupportMagickModules
 #elif !defined(__MINGW32__) && !defined(__MINGW64__)
 #  if defined(MSWINDOWS) && defined(_DLL)
-#    define SupportMagickModules
+//#    define SupportMagickModules
 #  endif
 #endif
 
 