diff --git a/src/tools/emscripten.jam b/src/tools/emscripten.jam
index 0ed289fae..4bd6c589a 100644
--- a/src/tools/emscripten.jam
+++ b/src/tools/emscripten.jam
@@ -34,6 +34,7 @@ support instead of native WebAssembly exception handling.
 import feature ;
 import os ;
 import toolset ;
+import generators ;
 import common ;
 import type ;
 import version ;
@@ -41,6 +42,9 @@ import args ;
 
 feature.extend toolset : emscripten ;
 
+generators.override builtin.lib-generator : emscripten.prebuilt ;
+generators.override emscripten.searched-lib-generator : searched-lib-generator ;
+
 feature.feature embind : off on : propagated ;
 feature.feature closure : off on full : propagated ;
 feature.feature link-optimization : off on full : propagated ;
