diff --git a/src/tools/emscripten.jam b/src/tools/emscripten.jam
index d6594c5e43..8b15d0af96 100644
--- a/src/tools/emscripten.jam
+++ b/src/tools/emscripten.jam
@@ -6,6 +6,7 @@
 import feature ;
 import os ;
 import toolset ;
+import generators ;
 import common ;
 import gcc ;
 import type ;
@@ -37,6 +38,9 @@ rule init ( version ? :  command * : options * )
 
 feature.extend toolset : emscripten ;
 
+generators.override builtin.lib-generator : emscripten.prebuilt ;
+generators.override emscripten.searched-lib-generator : searched-lib-generator ;
+
 toolset.inherit-generators emscripten <toolset>emscripten
     : gcc
     : gcc.mingw.link gcc.mingw.link.dll gcc.compile.c.pch gcc.compile.c++.pch
@@ -53,7 +57,7 @@ toolset.inherit-flags emscripten : gcc
 
 type.set-generated-target-suffix EXE : <toolset>emscripten : "js" ;
 type.set-generated-target-suffix OBJ : <toolset>emscripten : "bc" ;
-type.set-generated-target-suffix STATIC_LIB : <toolset>emscripten : "bc" ;
+type.set-generated-target-suffix STATIC_LIB : <toolset>emscripten : "a" ;
 
 toolset.flags emscripten.compile OPTIONS <flags> ;
 toolset.flags emscripten.compile OPTIONS <cflags> ;
 