diff --git a/Makefile.in b/Makefile.in
index a078eb5..7a6ab6d 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -4475,10 +4475,10 @@ uninstall-am: uninstall-binSCRIPTS uninstall-cincludeHEADERS \
 
 .PRECIOUS: Makefile
 
-AST_PAR: ast_par.source astbad
-	sed -e 's/<AST__BAD>/'`./astbad AST__BAD | tr 'E' 'D'`'/' \
-	    -e 's/<AST__NAN>/'`./astbad AST__NAN | tr 'E' 'D'`'/' \
-	    -e 's/<AST__NANR>/'`./astbad AST__NANF | tr 'E' 'D'`'/' \
+AST_PAR: ast_par.source astbad$(EXEEXT)
+	sed -e 's/<AST__BAD>/'`./astbad$(EXEEXT) AST__BAD | tr 'E' 'D'`'/' \
+	    -e 's/<AST__NAN>/'`./astbad$(EXEEXT) AST__NAN | tr 'E' 'D'`'/' \
+	    -e 's/<AST__NANR>/'`./astbad$(EXEEXT) AST__NANF | tr 'E' 'D'`'/' \
 	    ast_par.source >$@
 
 # ast.h depends on the error-facility files.  ast.h _could_ be made
 