diff --git a/src/qmlcompiler/qqmljscodegenerator.cpp b/src/qmlcompiler/qqmljscodegenerator.cpp
index 7e8ba71..0ccd3c7 100644
--- a/src/qmlcompiler/qqmljscodegenerator.cpp
+++ b/src/qmlcompiler/qqmljscodegenerator.cpp
@@ -2562,7 +2562,7 @@
         }
 
         for (int i = classSize; i < argc; i += 3) {
-            const int nameArg = argc + i + 1;
+            const int nameArg = args + i + 1;
             m_body += u"{ "_s
                     + conversion(
                               registerType(nameArg),
@@ -2570,7 +2570,7 @@
                               consumedRegisterVariable(nameArg))
                     + u", "_s;
 
-            const int valueArg = argc + i + 2;
+            const int valueArg = args + i + 2;
             m_body += convertStored(
                               registerType(valueArg).storedType(),
                               propType,
