diff --git a/CMakeLists.txt b/CMakeLists.txt
index 13c474b..34ffc06 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -398,6 +398,14 @@ if (HAVE_EXECINFO_H)
   set (HAVE_STACKTRACE 1)
 endif (HAVE_EXECINFO_H)
 
+if (HAVE_CXX11_ATOMIC)
+  message(STATUS "have atomic")
+  set (ac_cv_cxx11_atomic 1)
+else ()
+  message(STATUS "not have atomic")
+  set (ac_cv_cxx11_atomic 0)
+endif ()
+
 if (WITH_SYMBOLIZE)
   if (WIN32 OR CYGWIN)
     if(CMAKE_CROSSCOMPILING)
