diff --git a/CMakeLists.txt b/CMakeLists.txt
index ed10b22..b83a637 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,27 +22,27 @@ endif()
 
 include(CMakePackageConfigHelpers)
 
-add_library(${PROJECT_NAME} INTERFACE)
-add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
-target_compile_features(${PROJECT_NAME} INTERFACE cxx_std_17)
-target_include_directories(${PROJECT_NAME}
+add_library(magic-enum INTERFACE)
+add_library(magic-enum::magic-enum ALIAS magic-enum)
+target_compile_features(magic-enum INTERFACE cxx_std_17)
+target_include_directories(magic-enum
         INTERFACE
             $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
             $<INSTALL_INTERFACE:include>)
 
-write_basic_package_version_file(${PROJECT_NAME}ConfigVersion.cmake
+write_basic_package_version_file(magic-enumConfigVersion.cmake
         VERSION ${PROJECT_VERSION}
         COMPATIBILITY AnyNewerVersion)
 
-install(TARGETS ${PROJECT_NAME}
-        EXPORT ${PROJECT_NAME}Config)
+install(TARGETS magic-enum
+        EXPORT magic-enumConfig)
 
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake
-        DESTINATION lib/cmake/${PROJECT_NAME})
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/magic-enumConfigVersion.cmake
+        DESTINATION lib/cmake/magic-enum)
 
-install(EXPORT ${PROJECT_NAME}Config
-        NAMESPACE ${PROJECT_NAME}::
-        DESTINATION lib/cmake/${PROJECT_NAME})
+install(EXPORT magic-enumConfig
+        NAMESPACE magic-enum::
+        DESTINATION lib/cmake/magic-enum)
 
 install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/include
         DESTINATION .)
