diff --git a/util/build.info b/util/build.info
index cf06f15ae4f0e..e4aab44b38038 100644
--- a/util/build.info
+++ b/util/build.info
@@ -6,9 +6,9 @@ SCRIPTS{noinst}=wrap.pl
 SOURCE[wrap.pl]=wrap.pl.in
 DEPEND[wrap.pl]=../configdata.pm
 
-IF[{- !$disabled{quic} && !$disabled{stdio} -}]
+IF[{- !$disabled{quic} && !$disabled{stdio} && !$disabled{apps} -}]
   PROGRAMS{noinst}=quicserver
   SOURCE[quicserver]=quicserver.c
-INCLUDE[quicserver]=../include ../apps/include
-DEPEND[quicserver]=../libcrypto.a ../libssl.a
+  INCLUDE[quicserver]=../include ../apps/include
+  DEPEND[quicserver]=../libcrypto.a ../libssl.a
 ENDIF
