diff --git a/CMakeLists.txt b/CMakeLists.txt
index c7b173e..099e6a0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,6 +51,7 @@ option(MDL_LOG_PLATFORM_INFOS "Prints some infos about the current build system
 option(MDL_LOG_DEPENDENCIES "Prints the list of dependencies during the generation step." OFF)
 option(MDL_LOG_FILE_DEPENDENCIES "Prints the list of files that is copied after a successful build." OFF)
 option(MDL_INSTALL_HEADERS "Install headers." ON)
+option(MDL_INSTALL_PLUGINS "Install plugins." ON)
 set(MDL_ADDITIONAL_COMPILER_DEFINES "MDL_SOURCE_RELEASE" CACHE STRING "Additional compile defines that are passed to each of the projects")
 set(MDL_ADDITIONAL_COMPILER_OPTIONS "" CACHE STRING "Additional compile options that are passed to each of the projects")
 
@@ -148,8 +149,10 @@ add_subdirectory(${MDL_SRC_FOLDER}/prod/bin/mdlm)
 
 # PLUGINS
 #--------------------------------------------------------------------------------------------------
-add_subdirectory(${MDL_SRC_FOLDER}/shaders/plugin/dds)
-add_subdirectory(${MDL_SRC_FOLDER}/shaders/plugin/freeimage)
+if(MDL_INSTALL_PLUGINS)
+    add_subdirectory(${MDL_SRC_FOLDER}/shaders/plugin/dds)
+    add_subdirectory(${MDL_SRC_FOLDER}/shaders/plugin/freeimage)
+endif()
 
 # EXAMPLES
 #--------------------------------------------------------------------------------------------------
diff --git a/cmake/utilities.cmake b/cmake/utilities.cmake
index ef22b19..339a97f 100644
--- a/cmake/utilities.cmake
+++ b/cmake/utilities.cmake
@@ -1151,6 +1151,12 @@ set(_MDL_PROD_TARGETS
     shaders-plugin-dds
     shaders-plugin-freeimage
 )
+if(MDL_INSTALL_PLUGINS)
+    list(APPEND _MDL_PROD_TARGETS
+        shaders-plugin-dds
+        shaders-plugin-freeimage
+    )
+endif()
 function(ADD_TARGET_INSTALL)
     set(options)
     set(oneValueArgs TARGET DESTINATION)
