diff --git a/cmake/dependencies/add_freeimage.cmake b/cmake/dependencies/add_freeimage.cmake
index f63faab..675aa71 100644
--- a/cmake/dependencies/add_freeimage.cmake
+++ b/cmake/dependencies/add_freeimage.cmake
@@ -67,13 +67,4 @@ else()
                 )
         endif()
     endif()
-
-    # copy runtime dependencies
-    # copy system libraries only on windows, we assume the libraries are installed in a unix environment
-    if(NOT __TARGET_ADD_DEPENDENCY_NO_RUNTIME_COPY AND WINDOWS)
-        target_copy_to_output_dir(TARGET ${__TARGET_ADD_DEPENDENCY_TARGET}
-            FILES
-                ${MDL_DEPENDENCY_FREEIMAGE_SHARED}
-            )
-    endif()
 endif()
