diff --git a/src/nvtt/squish/CMakeLists.txt b/src/nvtt/squish/CMakeLists.txt
index 832013e1..76824137 100644
--- a/src/nvtt/squish/CMakeLists.txt	
+++ b/src/nvtt/squish/CMakeLists.txt
@@ -33,3 +33,7 @@ IF(NOT WIN32)
     ENDIF(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANGXX)
 ENDIF(NOT WIN32)
 
+INSTALL(TARGETS squish 
+    RUNTIME DESTINATION bin
+    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib/static)
\ No newline at end of file
