diff --git a/src/windows/srcversion/srcversion.vcxproj b/src/windows/srcversion/srcversion.vcxproj
index 8761197..795d1a7 100644
--- a/src/windows/srcversion/srcversion.vcxproj
+++ b/src/windows/srcversion/srcversion.vcxproj
@@ -72,7 +72,7 @@
       </Command>
     </PreBuildEvent>
     <CustomBuildStep>
-      <Command>powershell.exe -file "$(SolutionDir)..\utils\SRCVERSION.ps1" $(SRCVERSION)</Command>
+      <Command>powershell.exe -ExecutionPolicy Bypass -file "$(SolutionDir)..\utils\SRCVERSION.ps1" $(SRCVERSION)</Command>
       <Outputs>__NON_EXISTENT_FILE__</Outputs>
       <Message>generate srcversion.h</Message>
     </CustomBuildStep>
@@ -99,7 +99,7 @@
       </Command>
     </PreBuildEvent>
     <CustomBuildStep>
-      <Command>powershell.exe -file "$(SolutionDir)..\utils\SRCVERSION.ps1" $(SRCVERSION)</Command>
+      <Command>powershell.exe -ExecutionPolicy Bypass -file "$(SolutionDir)..\utils\SRCVERSION.ps1" $(SRCVERSION)</Command>
       <Outputs>__NON_EXISTENT_FILE__</Outputs>
       <Message>generate srcversion.h</Message>
     </CustomBuildStep>
