diff --git a/src/libpmemobj/critnib.c b/src/libpmemobj/critnib.c
index 41fec79..f201dbd 100644
--- a/src/libpmemobj/critnib.c
+++ b/src/libpmemobj/critnib.c
@@ -360,9 +360,9 @@ alloc_leaf(struct critnib *__restrict c)
  * crinib_insert -- write a key:value pair to the critnib structure
  *
  * Returns:
- *  • 0 on success
- *  • EEXIST if such a key already exists
- *  • ENOMEM if we're out of memory
+ *  0 on success
+ *  EEXIST if such a key already exists
+ *  ENOMEM if we're out of memory
  *
  * Takes a global write lock but doesn't stall any readers.
  */
