diff --git a/CMakeLists.txt b/CMakeLists.txt
index a8fe9e6..1631a38 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -124,8 +124,7 @@ endif()
 if (LIB_PROTO_MUTATOR_DOWNLOAD_PROTOBUF)
   include(protobuf)
 else()
-  find_package(Protobuf REQUIRED)
-  include_directories(${PROTOBUF_INCLUDE_DIRS})
+  find_package(protobuf CONFIG REQUIRED)
   include_directories(${CMAKE_CURRENT_BINARY_DIR})
 endif()
 
diff --git a/libprotobuf-mutatorConfig.cmake.in b/libprotobuf-mutatorConfig.cmake.in
index f62a501..6f905b7 100644
--- a/libprotobuf-mutatorConfig.cmake.in
+++ b/libprotobuf-mutatorConfig.cmake.in
@@ -1,5 +1,5 @@
 include(CMakeFindDependencyMacro)
 
-find_dependency(Protobuf REQUIRED)
+find_dependency(protobuf CONFIG REQUIRED)
 
 include("${CMAKE_CURRENT_LIST_DIR}/libprotobuf-mutatorTargets.cmake")
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index d0af1b6..2b4b492 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -20,7 +20,7 @@ add_library(protobuf-mutator
             text_format.cc
             utf8_fix.cc)
 target_link_libraries(protobuf-mutator
-                      ${PROTOBUF_LIBRARIES})
+                      PUBLIC protobuf::libprotobuf)
 set_target_properties(protobuf-mutator PROPERTIES
                       COMPILE_FLAGS "${NO_FUZZING_FLAGS}"
                       SOVERSION 0)
diff --git a/src/libfuzzer/CMakeLists.txt b/src/libfuzzer/CMakeLists.txt
index 85e9f4d..9863138 100644
--- a/src/libfuzzer/CMakeLists.txt
+++ b/src/libfuzzer/CMakeLists.txt
@@ -16,8 +16,7 @@ add_library(protobuf-mutator-libfuzzer
             libfuzzer_mutator.cc
             libfuzzer_macro.cc)
 target_link_libraries(protobuf-mutator-libfuzzer
-                      protobuf-mutator
-                      ${PROTOBUF_LIBRARIES})
+                      PUBLIC protobuf-mutator)
 set_target_properties(protobuf-mutator-libfuzzer PROPERTIES
                       COMPILE_FLAGS "${NO_FUZZING_FLAGS}"
                       SOVERSION 0)
