diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8543256..8ebce9c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,15 +1,26 @@
 cmake_minimum_required(VERSION 3.8 FATAL_ERROR)
 project(rapidstring LANGUAGES C VERSION 0.1.0)
 add_library(rapidstring INTERFACE)
-target_include_directories(rapidstring INTERFACE include)
+target_include_directories(rapidstring INTERFACE
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>
+    $<INSTALL_INTERFACE:include>)
 
 include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/utility.cmake)
 
 # TODO: Installation.
+install(FILES include/rapidstring.h DESTINATION include)
+
+install(TARGETS rapidstring EXPORT rapidstringConfig)
+
+install(EXPORT rapidstringConfig
+    FILE unofficial-rapidstringConfig.cmake
+    NAMESPACE unofficial::rapidstring::
+    DESTINATION share/cmake/rapidstring
+    PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ)
 
 if(BUILD_TESTING)
 	enable_testing()
 endif()
 
-add_subdirectory(test)
-add_subdirectory(benchmark)
+#add_subdirectory(test)
+#add_subdirectory(benchmark)
