diff --git a/CMakeLists.txt b/CMakeLists.txt
index d09a298..e64a6cd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,7 +12,6 @@ include(CheckIncludeFile)
 include(CheckStructHasMember)
 include(CheckSymbolExists)
 include(TestBigEndian)
-include(InstallRequiredSystemLibraries)
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${PROJECT_SOURCE_DIR}/build/cmake)
 
@@ -47,13 +46,6 @@ TEST_BIG_ENDIAN(BSON_BIG_ENDIAN)
 set (BSON_PTHREAD_ONCE_INIT_NEEDS_BRACES 0)
 set (BSON_HAVE_DECIMAL128 0)
 
-#librt needed on linux for clock_gettime
-find_library(RT_LIBRARY rt)
-if (RT_LIBRARY)
-   #set required libraries for CHECK_FUNCTION_EXISTS
-   set(CMAKE_REQUIRED_LIBRARIES ${RT_LIBRARY})
-endif()
-
 # See https://public.kitware.com/Bug/view.php?id=15659
 CHECK_SYMBOL_EXISTS(snprintf stdio.h BSON_HAVE_SNPRINTF)
 if (NOT BSON_HAVE_SNPRINTF)
@@ -314,7 +306,7 @@ install(
 )
 install(
   FILES ${HEADERS}
-  DESTINATION "include/libbson-${BSON_API_VERSION}"
+  DESTINATION "include"
 )
 
 set(VERSION "${BSON_VERSION}")
