diff --git a/src/libbson/build/cmake/libbson-static-1.0-config.cmake.in b/src/libbson/build/cmake/libbson-static-1.0-config.cmake.in
index 92933fe01..931a829ae 100644
--- a/src/libbson/build/cmake/libbson-static-1.0-config.cmake.in
+++ b/src/libbson/build/cmake/libbson-static-1.0-config.cmake.in
@@ -36,3 +36,7 @@ foreach (LIB @LIBBSON_LIBRARIES@)
 endforeach ()
 
 set (BSON_STATIC_DEFINITIONS BSON_STATIC)
+
+set (BSON_LIBRARIES ${BSON_STATIC_LIBRARIES})
+set (BSON_INCLUDE_DIRS ${BSON_STATIC_INCLUDE_DIRS})
+set (BSON_DEFINITIONS BSON_STATIC)
\ No newline at end of file
