diff --git a/src/libbson/CMakeLists.txt b/src/libbson/CMakeLists.txt
index 61e926747..715b304c0 100644
--- a/src/libbson/CMakeLists.txt
+++ b/src/libbson/CMakeLists.txt
@@ -181,6 +181,7 @@ set (HEADERS
    ${PROJECT_SOURCE_DIR}/src/bson/bson-compat.h
    ${PROJECT_SOURCE_DIR}/src/bson/bson-context.h
    ${PROJECT_SOURCE_DIR}/src/bson/bson-decimal128.h
+   ${PROJECT_SOURCE_DIR}/src/bson/bson-dsl.h  # mongo-c-driver build time only
    ${PROJECT_SOURCE_DIR}/src/bson/bson-endian.h
    ${PROJECT_SOURCE_DIR}/src/bson/bson-error.h
    ${PROJECT_SOURCE_DIR}/src/bson/bson.h
