diff --git a/configure.ac b/configure.ac
index bbe8673..7bb1b0e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -415,6 +415,7 @@ AS_HELP_STRING([--enable-install-elfh],[install elf.h in include dir]),
 AM_CONDITIONAL(INSTALL_ELFH, test "$install_elfh" = yes)
 
 AM_CONDITIONAL(BUILD_STATIC, [dnl
+test "$enable_static" = yes ||
 test "$use_gprof" = yes -o "$use_gcov" = yes])
 
 AC_ARG_ENABLE([tests-rpath],
