diff --git a/CMakeLists.txt.orig b/CMakeLists.txt
index 6f403af..d1ba4e7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -418,12 +418,6 @@ IF(WIN32)
         IF(FREEGLUT_REPLACE_GLUT)
             SET_TARGET_PROPERTIES(freeglut_static PROPERTIES OUTPUT_NAME ${LIBNAME})
         ENDIF()
-        # need to set machine:x64 for linker, at least for VC10, and
-        # doesn't hurt for older compilers:
-        # http://public.kitware.com/Bug/view.php?id=11240#c22768
-        IF (CMAKE_CL_64)
-            SET_TARGET_PROPERTIES(freeglut_static PROPERTIES STATIC_LIBRARY_FLAGS "/machine:x64")
-        ENDIF()
     ENDIF()
 ELSE()
     # on UNIX we need to make sure:
