diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8bc05ee..00ec3cc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,13 +26,16 @@ endif (POLICY CMP0077)
 
 list(APPEND CMAKE_MODULE_PATH
   ${CMAKE_CURRENT_SOURCE_DIR}/cmake
-  ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules)
-  
+  # Removed project specific Module overrides
+)
 set(eCAL_DIR ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules)
 
 project(eCAL)
 set(ECAL_PROJECT_ROOT "${CMAKE_CURRENT_LIST_DIR}")
 
+include(${CMAKE_CURRENT_LIST_DIR}/thirdparty/cmakefunctions/build-cmakefunctions.cmake)
+set(CMakeFunctions_FOUND TRUE)
+
 # --------------------------------------------------------
 # set msvc specific options
 # --------------------------------------------------------
@@ -289,7 +292,9 @@ if(MSVC)
   endif()
 endif()
 
-find_package(CMakeFunctions REQUIRED)
+if(NOT CMakeFunctions_FOUND)
+  find_package(CMakeFunctions REQUIRED)
+endif()
 
 git_revision_information(DEFAULT ${ECAL_BUILD_VERSION})
 set(eCAL_VERSION_MAJOR  ${GIT_REVISION_MAJOR})
