diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8bc05ee..00ec3cc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,13 +26,16 @@ endif (POLICY CMP0077)
 
 list(APPEND CMAKE_MODULE_PATH
   ${CMAKE_CURRENT_SOURCE_DIR}/cmake
-  ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules)
-  
+  # Removed project specific Module overrides
+)
 set(eCAL_DIR ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules)
 
 project(eCAL)
 set(ECAL_PROJECT_ROOT "${CMAKE_CURRENT_LIST_DIR}")
 
+include(${CMAKE_CURRENT_LIST_DIR}/thirdparty/cmakefunctions/build-cmakefunctions.cmake)
+set(CMakeFunctions_FOUND TRUE)
+
 # --------------------------------------------------------
 # set msvc specific options
 # --------------------------------------------------------
@@ -289,7 +292,9 @@ if(MSVC)
   endif()
 endif()
 
-find_package(CMakeFunctions REQUIRED)
+if(NOT CMakeFunctions_FOUND)
+  find_package(CMakeFunctions REQUIRED)
+endif()
 
 git_revision_information(DEFAULT ${ECAL_BUILD_VERSION})
 set(eCAL_VERSION_MAJOR  ${GIT_REVISION_MAJOR})
diff --git a/app/meas_cutter/CMakeLists.txt b/app/meas_cutter/CMakeLists.txt
index 442c308..8105682 100644
--- a/app/meas_cutter/CMakeLists.txt
+++ b/app/meas_cutter/CMakeLists.txt
@@ -19,7 +19,6 @@
 project(meas_cutter)
 
 find_package(Threads REQUIRED)
-find_package(tclap REQUIRED)
 find_package(yaml-cpp REQUIRED)
 
 include(${ECAL_PROJECT_ROOT}/thirdparty/yaml-cpp/compatibility-yaml-cpp.cmake)
diff --git a/ecal/core/CMakeLists.txt b/ecal/core/CMakeLists.txt
index 17c32c6..582a39d 100644
--- a/ecal/core/CMakeLists.txt
+++ b/ecal/core/CMakeLists.txt
@@ -40,7 +40,8 @@ project(core VERSION ${eCAL_VERSION_STRING})
 
 find_package(asio             REQUIRED)
 find_package(Threads          REQUIRED)
-find_package(ecaludp          REQUIRED)
+
+include(${CMAKE_CURRENT_LIST_DIR}/../../thirdparty/ecaludp/build-ecaludp.cmake)
 
 if (ECAL_CORE_CONFIGURATION)
   find_package(yaml-cpp       REQUIRED)
diff --git a/thirdparty/tclap/CMakeLists.txt b/thirdparty/tclap/CMakeLists.txt
index 4423eb3..dbb0305 100644
--- a/thirdparty/tclap/CMakeLists.txt
+++ b/thirdparty/tclap/CMakeLists.txt
@@ -1,3 +1,5 @@
+find_path(TCLAP_INCLUDE_DIRS "tclap/Arg.h")
+
 add_library(tclap INTERFACE)
-target_include_directories(tclap INTERFACE ${CMAKE_CURRENT_LIST_DIR}/tclap/include)
-add_library(tclap::tclap ALIAS tclap)
\ No newline at end of file
+target_include_directories(tclap INTERFACE ${TCLAP_INCLUDE_DIRS})
+add_library(tclap::tclap ALIAS tclap)
