diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8bc05ee..3cd16b0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,10 +26,13 @@ endif (POLICY CMP0077)
 
 list(APPEND CMAKE_MODULE_PATH
   ${CMAKE_CURRENT_SOURCE_DIR}/cmake
-  ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules)
-  
+  # Removed project specific Module overrides
+)
 set(eCAL_DIR ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules)
 
+include(${CMAKE_CURRENT_LIST_DIR}/thirdparty/cmakefunctions/build-cmakefunctions.cmake)
+set(CMakeFunctions_FOUND TRUE)
+
 project(eCAL)
 set(ECAL_PROJECT_ROOT "${CMAKE_CURRENT_LIST_DIR}")
 
@@ -289,9 +292,13 @@ if(MSVC)
   endif()
 endif()
 
-find_package(CMakeFunctions REQUIRED)
+if(NOT CMakeFunctions_FOUND)
+  find_package(CMakeFunctions REQUIRED)
+endif()
 
+if(NOT DEFINED ECAL_BUILD_VERSION)
 git_revision_information(DEFAULT ${ECAL_BUILD_VERSION})
+endif()
 set(eCAL_VERSION_MAJOR  ${GIT_REVISION_MAJOR})
 set(eCAL_VERSION_MINOR  ${GIT_REVISION_MINOR})
 set(eCAL_VERSION_PATCH  ${GIT_REVISION_PATCH})
