diff --git a/cmake/eCALConfig.cmake.in b/cmake/eCALConfig.cmake.in
index f285f5e..7411247 100644
--- a/cmake/eCALConfig.cmake.in
+++ b/cmake/eCALConfig.cmake.in
@@ -30,13 +30,8 @@ set(eCAL_VERSION_STRING @eCAL_VERSION_STRING@)
 set(CMAKE_MAP_IMPORTED_CONFIG_MINSIZEREL Release "")
 set(CMAKE_MAP_IMPORTED_CONFIG_RELWITHDEBINFO Release "")
 
-# Try and find protobuf in Config mode first
-# (Primarily for Windows where we provide protobuf as part of eCAL)
-find_package(Protobuf CONFIG)
-if(NOT Protobuf_FOUND)
-  # Search for Protobuf using Module mode (with no Config mode fallback)
-  find_package(Protobuf MODULE REQUIRED)
-endif()
+include(CMakeFindDependencyMacro)
+find_dependency(Protobuf CONFIG)
 
 include("@PACKAGE_eCAL_install_cmake_dir@/helper_functions/ecal_add_functions.cmake")
 include("@PACKAGE_eCAL_install_cmake_dir@/helper_functions/ecal_helper_functions.cmake")
@@ -50,6 +45,6 @@ include("@PACKAGE_eCAL_install_cmake_dir@/eCALTargets.cmake")
 #  list(APPEND CMAKE_PREFIX_PATH "${PACKAGE_PREFIX_DIR}/../../../../cmake")
 #endif()
 
-find_package(CMakeFunctions REQUIRED)
+find_dependency(CMakeFunctions CONFIG)
 
-find_package(Threads REQUIRED)
+find_dependency(Threads)
