--- a/CMakeLists.txt	Tue Aug  3 09:02:51 2021
+++ b/CMakeLists.txt	Wed Aug 25 22:31:06 2021
@@ -350,7 +350,9 @@
 
-macro(find_package)
-  if(NOT "${ARGV0}" IN_LIST as_subproject)
-    _find_package(${ARGV})
-  endif()
-endmacro()
+if (DISABLE_FIND_PACKAGE_OVERLOAD)
+  macro(find_package)
+    if(NOT "${ARGV0}" IN_LIST as_subproject)
+      _find_package(${ARGV})
+    endif()
+  endmacro()
+endif ()
 
@@ -698,2 +700,4 @@
 # --------------------------------------------------------
+if(CPACK_PACK_WITH_INNOSETUP)
 include(cpack/cpack_variables.cmake)
+endif()
\ No newline at end of file
