diff --git a/ECSUtil/ECSUtil.vcxproj b/ECSUtil/ECSUtil.vcxproj
index 05c5726..f8ac64e 100644
--- a/ECSUtil/ECSUtil.vcxproj
+++ b/ECSUtil/ECSUtil.vcxproj
@@ -274,7 +274,6 @@
       <AdditionalIncludeDirectories>$(IntDir);%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
     </ResourceCompile>
     <PostBuildEvent>
-      <Command>$(SolutionDir)bin\setversion.exe /getversion "#define ECSUTIL_VERSION" "$(SolutionDir)ECSUtil\Version.h" $(OutDir)ECSUtil.dll</Command>
     </PostBuildEvent>
     <PostBuildEvent>
       <Message>set version resource in DLL from source file</Message>
@@ -361,7 +360,6 @@
       <AdditionalIncludeDirectories>$(IntDir);%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
     </ResourceCompile>
     <PostBuildEvent>
-      <Command>$(SolutionDir)bin\setversion.exe /getversion "#define ECSUTIL_VERSION" "$(SolutionDir)ECSUtil\Version.h" $(OutDir)ECSUtil.dll</Command>
     </PostBuildEvent>
     <PostBuildEvent>
       <Message>set version resource in DLL from source file</Message>
@@ -445,7 +443,6 @@
       <AdditionalIncludeDirectories>$(IntDir);%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
     </ResourceCompile>
     <PostBuildEvent>
-      <Command>$(SolutionDir)bin\setversion.exe /getversion "#define ECSUTIL_VERSION" "$(SolutionDir)ECSUtil\Version.h" $(OutDir)ECSUtil.dll</Command>
     </PostBuildEvent>
     <PostBuildEvent>
       <Message>set version resource in DLL from source file</Message>
@@ -541,7 +538,6 @@
       <AdditionalIncludeDirectories>$(IntDir);%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
     </ResourceCompile>
     <PostBuildEvent>
-      <Command>$(SolutionDir)bin\setversion.exe /getversion "#define ECSUTIL_VERSION" "$(SolutionDir)ECSUtil\Version.h" $(OutDir)ECSUtil.dll</Command>
     </PostBuildEvent>
     <PostBuildEvent>
       <Message>set version resource in DLL from source file</Message>
