diff --git a/CMakeLists.txt b/CMakeLists.txt
index 62d5de6..ee500c7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -138,7 +138,8 @@ function (locate_zycore)
     )
 endfunction ()
 
-locate_zycore()
+find_package(zycore CONFIG REQUIRED)
+add_library(Zycore ALIAS Zycore::Zycore)
 
 # =============================================================================================== #
 # Library configuration                                                                           #
