diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6ed56a8..66d1e66 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -210,15 +210,9 @@ else()
 endif()
 
 if(ENABLE_SPIRV_TOOLS_INSTALL)
-  if(WIN32)
-    macro(spvtools_config_package_dir TARGET PATH)
-      set(${PATH} ${TARGET}/cmake)
-    endmacro()
-  else()
-    macro(spvtools_config_package_dir TARGET PATH)
-      set(${PATH} ${CMAKE_INSTALL_LIBDIR}/cmake/${TARGET})
-    endmacro()
-  endif()
+  macro(spvtools_config_package_dir TARGET PATH)
+    set(${PATH} ${CMAKE_INSTALL_DATADIR}/${TARGET})
+  endmacro()
 
   macro(spvtools_generate_config_file TARGET)
     file(WRITE ${CMAKE_BINARY_DIR}/${TARGET}Config.cmake
