diff --git a/src/tools/svg2png/meson.build b/src/tools/svg2png/meson.build
index c9fe63b6..010c6ac4 100644
--- a/src/tools/svg2png/meson.build
+++ b/src/tools/svg2png/meson.build
@@ -5,4 +5,4 @@ executable('svg2png',
            include_directories : headers,
            cpp_args : compiler_flags,
            install : true,
-           link_with : thorvg_lib)
+           link_with : thorvg_lib , install_dir : get_option('bindir'))
diff --git a/src/tools/svg2tvg/meson.build b/src/tools/svg2tvg/meson.build
index 158376b5..8d960aec 100644
--- a/src/tools/svg2tvg/meson.build
+++ b/src/tools/svg2tvg/meson.build
@@ -5,4 +5,4 @@ executable('svg2tvg',
            include_directories : headers,
            cpp_args : compiler_flags,
            install : true,
-           link_with : thorvg_lib)
+           link_with : thorvg_lib , install_dir : get_option('bindir'))
