diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 5ce1f77..1f3e863 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -10,8 +10,7 @@ set_target_properties(${PROJECT_NAME} PROPERTIES
   SOVERSION ${FCL_ABI_VERSION})
 
 target_link_libraries(${PROJECT_NAME}
-  PUBLIC ${OCTOMAP_LIBRARIES}
-  PRIVATE ${CCD_LIBRARIES}
+  PUBLIC octomap octomath ccd
   PRIVATE ${Boost_LIBRARIES})
 
 target_include_directories(${PROJECT_NAME} INTERFACE
