diff --git a/include/internal/basic_csv_parser.hpp b/include/internal/basic_csv_parser.hpp
index 341a154..2d1e70a 100644
--- a/include/internal/basic_csv_parser.hpp
+++ b/include/internal/basic_csv_parser.hpp
@@ -12,7 +12,7 @@
 #include <vector>

 #if !defined(__EMSCRIPTEN__)
-#include "../external/mio.hpp"
+#include "mio/mmap.hpp"
 #endif
 #include "basic_csv_parser_simd.hpp"
 #include "col_names.hpp"
diff --git a/include/internal/common.hpp b/include/internal/common.hpp
index 97824fd..aa5b484 100644
--- a/include/internal/common.hpp
+++ b/include/internal/common.hpp
@@ -105,7 +105,7 @@
 #endif

 // Include string_view BEFORE csv namespace to avoid namespace pollution issues
-#ifdef CSV_HAS_CXX17
+#if 1
 #include <string_view>
 #else
 #include "../external/string_view.hpp"
@@ -141,7 +141,7 @@ namespace csv {
     #define CSV_DEBUG_ASSERT(x) assert(x)
 #endif

-#ifdef CSV_HAS_CXX17
+#if 1
      /** @typedef string_view
       *  The string_view class used by this library.
       */
