diff --git a/CMakeLists.txt b/CMakeLists.txt
index 477410ba8..fcca03917 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,7 +1,6 @@
 # Enforce some CMake policies
 cmake_minimum_required(VERSION 3.4)
 
-set(CMAKE_USER_MAKE_RULES_OVERRIDE_CXX "${CMAKE_CURRENT_SOURCE_DIR}/cmake/cxx_compiler_flags_overrides.cmake")
 project(Z3 VERSION 4.8.16.0 LANGUAGES CXX)
 
 ################################################################################
