diff --git a/CMakeLists.txt b/CMakeLists.txt
index 88a8ae0..a36cb19 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -62,7 +62,7 @@ if(DEPENDENCY_CHECK AND BUILD_POSTGRESQL_CONNECTOR)
 endif()
 
 if(DEPENDENCY_CHECK AND BUILD_SQLITE3_CONNECTOR)
-  find_package(SQLite3 REQUIRED)
+  find_package(unofficial-sqlite3 CONFIG REQUIRED)
 endif()
 
 if(DEPENDENCY_CHECK AND BUILD_SQLCIPHER_CONNECTOR)
@@ -89,7 +89,7 @@ target_compile_features(sqlpp11 INTERFACE cxx_std_11)
 
 
 if(BUILD_SQLITE3_CONNECTOR)
-  add_component(NAME sqlite3 DEPENDENCIES SQLite::SQLite3)
+  add_component(NAME sqlite3 DEPENDENCIES unofficial::sqlite3::sqlite3)
 endif()
 
 if(BUILD_SQLCIPHER_CONNECTOR)
diff --git a/cmake/configs/Sqlpp11Config.cmake b/cmake/configs/Sqlpp11Config.cmake
index c25da97..b15524e 100644
--- a/cmake/configs/Sqlpp11Config.cmake
+++ b/cmake/configs/Sqlpp11Config.cmake
@@ -30,6 +30,7 @@ list(PREPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR})
 include(CMakeFindDependencyMacro)
 find_dependency(Threads)
 find_dependency(date REQUIRED)
+find_dependency(unofficial-sqlite3 CONFIG)
 
 # Work out the set of components to load
 if(${CMAKE_FIND_PACKAGE_NAME}_FIND_COMPONENTS)
