diff --git a/cpr/CMakeLists.txt b/cpr/CMakeLists.txt
index 563b785..c02d41e 100644
--- a/cpr/CMakeLists.txt
+++ b/cpr/CMakeLists.txt
@@ -1,31 +1,53 @@
 cmake_minimum_required(VERSION 3.15)
 
-add_library(cpr
-    auth.cpp
-    cookies.cpp
-    cprtypes.cpp
-    curlholder.cpp
-    digest.cpp
-    error.cpp
-    multipart.cpp
-    ntlm.cpp
-    parameters.cpp
-    payload.cpp
-    proxies.cpp
-    session.cpp
-    timeout.cpp
-    unix_socket.cpp
-    util.cpp
-    )
+add_library(${CPR_LIBRARIES}
+  auth.cpp
+  cookies.cpp
+  cprtypes.cpp
+  digest.cpp
+  error.cpp
+  multipart.cpp
+  parameters.cpp
+  payload.cpp
+  proxies.cpp
+  session.cpp
+  timeout.cpp
+  util.cpp
+  ssl_options.cpp
+)
 
-add_library(cpr::cpr ALIAS cpr)
+target_link_libraries( ${CPR_LIBRARIES}
+  PUBLIC ${CURL_LIBRARIES}
+)
 
-target_link_libraries(cpr PUBLIC CURL::libcurl) # todo should be private, but first dependencys in ssl_options need to be removed
+target_include_directories(${CPR_LIBRARIES}
+  PUBLIC 
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../include>
+    $<INSTALL_INTERFACE:include>
+  PRIVATE
+    $<BUILD_INTERFACE:${CURL_INCLUDE_DIRS}>
+)
 
 # Set version for shared libraries.
-set_target_properties(cpr
+set_target_properties(${CPR_LIBRARIES}
      PROPERTIES
      VERSION ${${PROJECT_NAME}_VERSION}
      SOVERSION ${${PROJECT_NAME}_VERSION_MAJOR})
 
-install(TARGETS cpr)
+set(TARGETS_EXPORT_NAME "${PROJECT_NAME}Targets")
+install(
+  TARGETS ${CPR_LIBRARIES}
+  EXPORT ${TARGETS_EXPORT_NAME}
+  RUNTIME DESTINATION bin
+  LIBRARY DESTINATION lib
+  ARCHIVE DESTINATION lib
+)
+
+install(
+  EXPORT ${TARGETS_EXPORT_NAME} 
+  DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}"
+)
+
+if (NOT DISABLE_INSTALL_HEADERS)
+  install(DIRECTORY ${CPR_INCLUDE_DIRS}/cpr DESTINATION include)
+endif()
\ No newline at end of file
