diff --git a/sdk/eventhubs/azure-messaging-eventhubs/CMakeLists.txt b/sdk/eventhubs/azure-messaging-eventhubs/CMakeLists.txt
index 5b69cf55c..0ea0b8e10 100644
--- a/sdk/eventhubs/azure-messaging-eventhubs/CMakeLists.txt
+++ b/sdk/eventhubs/azure-messaging-eventhubs/CMakeLists.txt
@@ -110,6 +110,8 @@ target_include_directories(
 
 target_link_libraries(azure-messaging-eventhubs PUBLIC Azure::azure-core Azure::azure-core-amqp)
 
+target_compile_definitions(azure-messaging-eventhubs PRIVATE _azure_BUILDING_SDK)
+
 # coverage. Has no effect if BUILD_CODE_COVERAGE is OFF
 create_code_coverage(eventhubs azure-messaging-eventhubs azure-messaging-eventhubs-test "tests?/*;samples?/*")
 
