diff --git a/CMakeLists.txt b/CMakeLists.txt
index ccc959e..d3e5d09 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -391,7 +391,7 @@ if(NOT CMAKE_SKIP_INSTALL_RULES)
       LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
       INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
       ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-      RUNTIME DESTINATION ${CMAKE_INSTALL_LIBDIR}
+      RUNTIME DESTINATION bin
     )
   else()
     install(
@@ -400,7 +400,7 @@ if(NOT CMAKE_SKIP_INSTALL_RULES)
       LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
       INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
       ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-      RUNTIME DESTINATION ${CMAKE_INSTALL_LIBDIR}
+      RUNTIME DESTINATION bin
     )
   endif()

diff --git a/cmake/cpptrace-config.cmake.in b/cmake/cpptrace-config.cmake.in
index c84bed9..e4c6dc7 100644
--- a/cmake/cpptrace-config.cmake.in
+++ b/cmake/cpptrace-config.cmake.in
@@ -1,3 +1,5 @@
 @PACKAGE_INIT@

 include(${CMAKE_CURRENT_LIST_DIR}/cpptrace_targets.cmake)
+
+find_dependency(ZLIB)

diff --git a/src/platform/utils.hpp b/src/platform/utils.hpp
index 7853960..51395fe 100644
--- a/src/platform/utils.hpp
+++ b/src/platform/utils.hpp
@@ -334,7 +334,7 @@ namespace detail {
         return static_cast<unsigned long long>(t);
     }
     template<typename T>
-    unsigned long long to_uintptr(T t) {
+    uintptr_t to_uintptr(T t) {
         return static_cast<uintptr_t>(t);
     }

