diff --git a/src/dft/CMakeLists.txt b/src/dft/CMakeLists.txt
index 526c4f8..667a7c2 100644
--- a/src/dft/CMakeLists.txt
+++ b/src/dft/CMakeLists.txt
@@ -443,3 +443,22 @@ install(
     INCLUDES #
     DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
 )
+
+set(items "")
+foreach(lib IN LISTS OpenMP_CXX_LIB_NAMES OpenMP_C_LIB_NAMES)
+    list(REMOVE_ITEM items " -l${lib}")
+    list(APPEND items " -l${lib}")
+endforeach()
+string(APPEND PC_LIBS_PRIVATE ${items})
+set(items "")
+foreach(flag IN LISTS OpenMP_CXX_FLAGS OpenMP_C_FLAGS)
+    list(REMOVE_ITEM items " ${flag}")
+    list(APPEND items " ${flag}")
+endforeach()
+string(APPEND PC_CFLAGS_PRIVATE ${items})
+configure_file("sleefdft.pc.in" "${CMAKE_CURRENT_BINARY_DIR}/sleefdft.pc" @ONLY)
+install(
+    FILES "${CMAKE_CURRENT_BINARY_DIR}/sleefdft.pc"
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig"
+    COMPONENT sleef_Development
+)
diff --git a/src/dft/sleefdft.pc.in b/src/dft/sleefdft.pc.in
new file mode 100644
index 0000000..58b6e2a
--- /dev/null
+++ b/src/dft/sleefdft.pc.in
@@ -0,0 +1,12 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+libdir=@CMAKE_INSTALL_FULL_LIBDIR@
+includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
+
+Name: SLEEFDFT
+Description: SIMD Library for Evaluating Elementary Functions, vectorized libm and DFT
+Version: @SLEEF_VERSION_MAJOR@.@SLEEF_VERSION_MINOR@.@SLEEF_VERSION_PATCH@
+Cflags: -I${includedir}
+Cflags.private: @PC_CFLAGS_PRIVATE@
+Libs: -L${libdir} -lsleefdft
+Libs.private: @PC_LIBS_PRIVATE@
+Requires.private: sleef
