diff --git a/src/google/protobuf/descriptor.cc b/src/google/protobuf/descriptor.cc
index b6c31fd..8b77c3d 100644
--- a/src/google/protobuf/descriptor.cc
+++ b/src/google/protobuf/descriptor.cc
@@ -794,7 +794,7 @@ class TableArena {
     size = RoundUp(size);
 
     Block* to_relocate = nullptr;
-    Block* to_use;
+    Block* to_use = nullptr;
 
     for (size_t i = 0; i < kSmallSizes.size(); ++i) {
       if (small_size_blocks_[i] != nullptr && size <= kSmallSizes[i]) {
