diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index 849679995..dba537a68 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -198,6 +198,7 @@ if (MSVC)
     /wd4506 # no definition for inline function 'function'
     /wd4800 # 'type' : forcing value to bool 'true' or 'false' (performance warning)
     /wd4996 # The compiler encountered a deprecated declaration.
+    /wd4703 # Potentially uninitialized local pointer variable 'name' used.
   )
   # Allow big object
   add_definitions(/bigobj)
