diff --git a/src/google/protobuf/compiler/main.cc b/src/google/protobuf/compiler/main.cc
index 7cb7a6375..e2716fd18 100644
--- a/src/google/protobuf/compiler/main.cc
+++ b/src/google/protobuf/compiler/main.cc	
@@ -108,6 +108,14 @@ int ProtobufMain(int argc, char* argv[]) {
 }  // namespace protobuf
 }  // namespace google
 
+#include <stdio>
+
 int main(int argc, char* argv[]) {
-  return PROTOBUF_NAMESPACE_ID::compiler::ProtobufMain(argc, argv);
+    try {
+        return PROTOBUF_NAMESPACE_ID::compiler::ProtobufMain(argc, argv);
+    } catch (std::exception &e )
+    {
+        std::puts(e.what());
+        return -1;
+    }
 }
