diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5ffbdf3..ba22109 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -75,7 +75,7 @@ if(SqliteOrm_BuildTests AND BUILD_TESTING)
     add_subdirectory(tests)
 endif()
 
-add_subdirectory(examples)
+#add_subdirectory(examples)
 
 install(TARGETS sqlite_orm EXPORT "${ProjectName}Targets"
 	INCLUDES DESTINATION "${INCLUDE_INSTALL_DIR}" COMPONENT Development
