diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 945d275..8ca2de4 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -162,6 +162,11 @@ if (MSVC)
             # C4458: declaration of 'symbol' hides class member
             /wd4458)
     endif()
+    if (CMAKE_CXX_FLAGS MATCHES "/D_UNICODE")
+        # explicitly set the entry point of the executable file,
+        # otherwise for some reason the linker will not pick up `wmain`, which is provided by the static Catch2 library
+        target_link_options(unit_tests PRIVATE "/ENTRY:wmainCRTStartup")
+    endif()
 endif()
 
 target_precompile_headers(unit_tests PRIVATE
