diff --git a/CMakeLists.txt b/CMakeLists.txt
index 34b89f2..ec53588 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,8 +17,7 @@ option(RYML_DBG "Enable (very verbose) ryml debug prints." OFF)
 
 #-------------------------------------------------------
 
-c4_require_subproject(c4core INCORPORATE
-    SUBDIRECTORY ${RYML_EXT_DIR}/c4core)
+find_package(c4core CONFIG REQUIRED)
 
 c4_add_library(ryml
     SOURCES
@@ -48,10 +47,10 @@ c4_add_library(ryml
         ryml.natvis
     SOURCE_ROOT ${RYML_SRC_DIR}
     INC_DIRS
+        $<BUILD_INTERFACE:${C4CORE_INCLUDE_DIR}>
         $<BUILD_INTERFACE:${RYML_SRC_DIR}>
         $<INSTALL_INTERFACE:include>
-    LIBS c4core
-    INCORPORATE c4core
+    LIBS c4core::c4core
     )
 
 if(NOT RYML_DEFAULT_CALLBACKS)
