diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 9d5adce..eff939d 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,5 +1,5 @@
 cmake_path(SET include_dir "${CMAKE_CURRENT_SOURCE_DIR}/../include")
-
+include(GNUInstallDirs)
 # Add the header files to the project
 set(header_files 
 	"${include_dir}/wincpp/process.hpp"
@@ -36,21 +36,21 @@ set(header_files
 
 # Add the library to the project
 add_library(_wincpp_core INTERFACE)
-
 # Add the include directory to the project
-target_sources(_wincpp_core INTERFACE ${header_files})
+target_sources(_wincpp_core INTERFACE $<BUILD_INTERFACE:${header_files}>)
 
 # Add the include directories
 target_include_directories(_wincpp_core SYSTEM INTERFACE 
 	$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../include>
-	$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+	$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/wincpp>
 )
 
 # Add the library to the project
-add_library(wincpp STATIC)
+add_library(wincpp)
+set_target_properties(wincpp PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS ON)
 
 # Link the library to the core
-target_link_libraries(wincpp INTERFACE _wincpp_core)
+target_link_libraries(wincpp PUBLIC _wincpp_core)
 
 # Add the source files to the project
 target_sources(wincpp PRIVATE 	
@@ -89,3 +89,38 @@ target_sources(wincpp PRIVATE
 
 # Add the include directory to the project
 target_include_directories(wincpp PRIVATE ${include_dir})
+
+# Install wincpp library
+install(TARGETS wincpp _wincpp_core EXPORT wincpp-export
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+)
+
+install(DIRECTORY "${CMAKE_CURRENT_LIST_DIR}/../include/wincpp" DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
+
+include(CMakePackageConfigHelpers)
+write_basic_package_version_file(
+    "${CMAKE_CURRENT_BINARY_DIR}/wincppConfigVersion.cmake" 
+    VERSION 1.5.3.6 
+    COMPATIBILITY AnyNewerVersion
+)
+
+file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/wincppConfig.cmake" 
+    "include(\${CMAKE_CURRENT_LIST_DIR}/wincppTargets.cmake)\n"
+)
+
+set(wincpp_CONFFILE_DEST "${CMAKE_INSTALL_DATAROOTDIR}/wincpp")
+
+install(FILES
+    "${CMAKE_CURRENT_BINARY_DIR}/wincppConfig.cmake"
+    "${CMAKE_CURRENT_BINARY_DIR}/wincppConfigVersion.cmake"
+    DESTINATION ${wincpp_CONFFILE_DEST}
+)
+
+install(EXPORT wincpp-export 
+    FILE wincppTargets.cmake 
+    NAMESPACE wincpp:: 
+    DESTINATION ${wincpp_CONFFILE_DEST}
+)
diff --git a/src/memory_factory.cpp b/src/memory_factory.cpp
index 50cd83d..82c1474 100644
--- a/src/memory_factory.cpp
+++ b/src/memory_factory.cpp
@@ -24,7 +24,7 @@ namespace wincpp
             }
             case wincpp::memory_type::remote_t:
             {
-                std::size_t read;
+                SIZE_T read;
 
                 if ( !ReadProcessMemory( p->handle->native, reinterpret_cast< void* >( address ), buffer, size, &read ) || read != size )
                     return false;
@@ -61,7 +61,7 @@ namespace wincpp
             }
             case wincpp::memory_type::remote_t:
             {
-                std::size_t written;
+                SIZE_T written;
 
                 WriteProcessMemory( p->handle->native, reinterpret_cast< void* >( address ), buffer, size, &written );
                 return written;
@@ -201,4 +201,4 @@ namespace wincpp
 
         return nullptr;
     }
-}  // namespace wincpp
\ No newline at end of file
+}  // namespace wincpp
