diff --git a/extras/Build/CMake/JUCEModuleSupport.cmake b/extras/Build/CMake/JUCEModuleSupport.cmake
index e18dd5e08..096bea42c 100644
--- a/extras/Build/CMake/JUCEModuleSupport.cmake
+++ b/extras/Build/CMake/JUCEModuleSupport.cmake
@@ -542,6 +542,15 @@ function(juce_add_module module_path)
         target_link_libraries(${module_name} INTERFACE oboe)
     endif()

+    if(${module_name} STREQUAL "juce_graphics")
+        if((CMAKE_SYSTEM_NAME STREQUAL "Linux") OR (CMAKE_SYSTEM_NAME MATCHES ".*BSD"))
+            find_package(Freetype)
+            if(TARGET Freetype::Freetype)
+                target_link_libraries(${module_name} INTERFACE Freetype::Freetype)
+            endif()
+        endif()
+    endif()
+
     if((${module_name} STREQUAL "juce_opengl") AND (CMAKE_SYSTEM_NAME STREQUAL "Android"))
         set(platform_supports_gl3 0)

diff --git a/extras/Build/juceaide/CMakeLists.txt b/extras/Build/juceaide/CMakeLists.txt
index 163f203cc..08a98d68f 100644
--- a/extras/Build/juceaide/CMakeLists.txt
+++ b/extras/Build/juceaide/CMakeLists.txt
@@ -35,6 +35,7 @@ if(JUCE_BUILD_HELPER_TOOLS)
     target_compile_definitions(juceaide PRIVATE
         JUCE_DISABLE_JUCE_VERSION_PRINTING=1
         JUCE_USE_CURL=0
+        JUCE_USE_FREETYPE=${JUCE_USE_FREETYPE}
         # This is a temporary workaround to allow builds to complete on Xcode 15.
         # Add -Wl,-ld_classic to the OTHER_LDFLAGS build setting if you need to
         # deploy to older versions of macOS.
--
2.34.1
