diff --git a/extras/Build/CMake/JUCEModuleSupport.cmake b/extras/Build/CMake/JUCEModuleSupport.cmake
index e18dd5e08..d53f68871 100644
--- a/extras/Build/CMake/JUCEModuleSupport.cmake
+++ b/extras/Build/CMake/JUCEModuleSupport.cmake
@@ -542,6 +542,11 @@ function(juce_add_module module_path)
         target_link_libraries(${module_name} INTERFACE oboe)
     endif()

+    if((${module_name} STREQUAL "juce_graphics") AND DEFINED JUCE_USE_FREETYPE AND JUCE_USE_FREETYPE AND (CMAKE_SYSTEM_NAME STREQUAL "Linux"))
+        find_package(Freetype REQUIRED)
+        target_link_libraries(${module_name} INTERFACE Freetype::Freetype)
+    endif()
+
     if((${module_name} STREQUAL "juce_opengl") AND (CMAKE_SYSTEM_NAME STREQUAL "Android"))
         set(platform_supports_gl3 0)

--
2.34.1
