diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4262a9852..7785d65ac 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -127,8 +127,8 @@ set(JUCE_INSTALL_DESTINATION "lib/cmake/JUCE-${JUCE_VERSION}" CACHE STRING
     "The location, relative to the install prefix, where the JUCE config file will be installed")

 set(JUCE_MODULE_PATH "include/JUCE-${JUCE_VERSION}/modules")
-set(UTILS_INSTALL_DIR "${JUCE_INSTALL_DESTINATION}")
-set(JUCEAIDE_PATH "${JUCE_TOOL_INSTALL_DIR}/${JUCE_JUCEAIDE_NAME}")
+set(UTILS_INSTALL_DIR "share/juce")
+set(JUCEAIDE_PATH "tools/juce/${JUCE_JUCEAIDE_NAME}")
 configure_package_config_file("${JUCE_CMAKE_UTILS_DIR}/JUCEConfig.cmake.in"
     "${JUCE_BINARY_DIR}/JUCEConfig.cmake"
     PATH_VARS UTILS_INSTALL_DIR JUCEAIDE_PATH JUCE_MODULE_PATH
--
2.34.1
