diff --git a/CMakeLists.txt b/CMakeLists.txt
index b2c51de83..5d42dc441 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -121,9 +121,12 @@ if (${use_prov_client})
     set(use_prov_client_core ON)
 
     if ("${hsm_custom_lib}" STREQUAL "")
-        set(hsm_type_x509 ON)
-        set(hsm_type_sastoken ON)
-        set(hsm_type_symm_key ON)
+        if ((NOT ${hsm_type_x509}) AND (NOT ${hsm_type_sastoken}) AND (NOT ${hsm_type_symm_key}))
+            # If the cmake option did not explicitly configure an hsm type, then enable them all.
+            set(hsm_type_x509 ON)
+            set(hsm_type_sastoken ON)
+            set(hsm_type_symm_key ON)
+        endif()
     else()
         set(hsm_type_custom ON)
     endif()
