diff --git a/sha1.c b/sha1.c
index fe8da83..53ad2c4 100644
--- a/sha1.c
+++ b/sha1.c
@@ -279,7 +279,7 @@ void SHA1Final(
     memset(&finalcount, '\0', sizeof(finalcount));
 }
 
-void SHA1(
+void SHA1_CLIBS(
     char *hash_out,
     const char *str,
     int len)
diff --git a/sha1.h b/sha1.h
index 96bb008..53c4b01 100644
--- a/sha1.h
+++ b/sha1.h
@@ -36,7 +36,7 @@ void SHA1Final(
     SHA1_CTX * context
     );
 
-void SHA1(
+void SHA1_CLIBS(
     char *hash_out,
     const char *str,
     int len);
