diff --git a/common/openslide-common-fd.c b/common/openslide-common-fd.c
index 3d3ce955f3c1..12fe27e8b9e8 100644
--- a/common/openslide-common-fd.c
+++ b/common/openslide-common-fd.c
@@ -23,11 +23,12 @@
 #define _WIN32_WINNT 0x0600
 #include <windows.h>
 #include <io.h>
+#else
+#include <unistd.h>
 #endif
 
 #include <sys/types.h>
 #include <sys/stat.h>
-#include <unistd.h>
 #include <fcntl.h>
 #include <glib.h>
 
diff --git a/test/extended.c b/test/extended.c
index 3e8e6def259c..0dd7de6716d8 100644
--- a/test/extended.c
+++ b/test/extended.c
@@ -22,11 +22,11 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <stdbool.h>
-#include <unistd.h>
 #include <inttypes.h>
 
 #ifndef _WIN32
 #include <sys/types.h>
+#include <unistd.h>
 #include <fcntl.h>
 #endif
 
diff --git a/test/try_open.c b/test/try_open.c
index 3a2543b54514..5cdce73b27aa 100644
--- a/test/try_open.c
+++ b/test/try_open.c
@@ -25,12 +25,15 @@
 #include <stdio.h>
 #include <stdarg.h>
 #include <string.h>
-#include <unistd.h>
 #include <glib.h>
 #include "openslide.h"
 #include "openslide-common.h"
 #include "config.h"
 
+#ifndef _WIN32
+#include <unistd.h>
+#endif
+
 #ifdef HAVE_VALGRIND
 #include <valgrind.h>
 #endif
diff --git a/tools/slidetool-util.c b/tools/slidetool-util.c
index 28901eda611c..5ebf03bbb926 100644
--- a/tools/slidetool-util.c
+++ b/tools/slidetool-util.c
@@ -20,8 +20,12 @@
  */
 
 #include <stdio.h>
-#include <unistd.h>
 #include <errno.h>
+
+#ifndef _WIN32
+#include <unistd.h>
+#endif
+
 #include "openslide-common.h"
 #include "slidetool.h"
 
