diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 873fb9f..9de74db 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -31,6 +31,6 @@ if(NOT TARGET Catch2::Catch2)
     FetchContent_MakeAvailable(Catch2)
     list(APPEND CMAKE_MODULE_PATH ${catch2_SOURCE_DIR}/contrib)
 endif()
-target_link_libraries(keyfinder-tests PRIVATE Catch2::Catch2)
+target_link_libraries(keyfinder-tests PRIVATE Catch2::Catch2WithMain)
 include(Catch)
 catch_discover_tests(keyfinder-tests)
diff --git a/tests/_testhelpers.h b/tests/_testhelpers.h
index ee6692e..5490511 100644
--- a/tests/_testhelpers.h
+++ b/tests/_testhelpers.h
@@ -23,7 +23,7 @@
 #define TESTHELPERS_H
 
 #include <cmath>
-#include <catch2/catch.hpp>
+#include <catch2/catch_all.hpp>
 #include "keyfinder.h"
 
 // all of this is a bit weak.
diff --git a/tests/main.cpp b/tests/main.cpp
index 1c321e4..1e83e5b 100644
--- a/tests/main.cpp
+++ b/tests/main.cpp
@@ -20,4 +20,4 @@
 *************************************************************************/
 
 #define CATCH_CONFIG_MAIN
-#include <catch2/catch.hpp>
+// #include <catch2/catch.hpp>
