diff --git a/onig-config.cmake.in b/onig-config.cmake.in
index 0a8a702..8d5a997 100644
--- a/onig-config.cmake.in
+++ b/onig-config.cmake.in
@@ -27,7 +27,7 @@ if test $# -eq 0; then
   show_usage
 fi
 
-prefix=@CMAKE_INSTALL_PREFIX@
+prefix=${pcfiledir}/../..
 exec_prefix=${prefix}
 libdir=@onig_pkgconfig_libdir@
 includedir=${prefix}/include
diff --git a/oniguruma.pc.cmake.in b/oniguruma.pc.cmake.in
index cb5b959..e720e86 100644
--- a/oniguruma.pc.cmake.in
+++ b/oniguruma.pc.cmake.in
@@ -1,6 +1,6 @@
-prefix=@CMAKE_INSTALL_PREFIX@
+prefix=${pcfiledir}/../..
 exec_prefix=${prefix}
-libdir=@onig_pkgconfig_libdir@
+libdir=${prefix}/lib
 includedir=${prefix}/include
 datarootdir=${prefix}/share
 datadir=${prefix}/share
