diff --git a/lzf/lzf_filter.c b/lzf/lzf_filter.c
index 951b1e4c..b2fbffc1 100644
--- a/lzf/lzf_filter.c
+++ b/lzf/lzf_filter.c
@@ -93,9 +93,9 @@ static const H5Z_class_t filter_class = {
 
 #include "H5PLextern.h"
 
-H5PL_type_t H5PLget_plugin_type(void){ return H5PL_TYPE_FILTER; }
+//H5PL_type_t H5PLget_plugin_type(void){ return H5PL_TYPE_FILTER; }
 
-const void *H5PLget_plugin_info(void){ return &filter_class; }
+//const void *H5PLget_plugin_info(void){ return &filter_class; }
 
 #endif
 #endif
