diff --git a/CMakeLists.txt b/CMakeLists.txt
index 10d10b99..fda6fa3c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1087,6 +1087,6 @@ install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/getdns DESTINATION include)
 install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/man3 DESTINATION share/man)

 set(docdir share/doc/getdns)
-install(FILES AUTHORS ChangeLog COPYING LICENSE NEWS README.md DESTINATION ${docdir})
+install(FILES AUTHORS ChangeLog LICENSE NEWS README.md DESTINATION ${docdir})
 install(FILES spec/index.html DESTINATION ${docdir}/spec)
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/getdns.pc DESTINATION lib/pkgconfig)
