diff --git a/CMakeLists.txt b/CMakeLists.txt
index a5070b1..b60462b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,18 +19,6 @@ cmake_minimum_required(VERSION 3.0.0)
 ##
 project(ctbignum VERSION 0.1 LANGUAGES CXX)
 
-##
-## INCLUDE
-##
-##
-include(ExternalProject)
-
-## External dependencies
-find_package(Boost 1.61 REQUIRED)
-if(Boost_FOUND)
-  include_directories(${Boost_INCLUDE_DIRS})
-endif()
-
 ##
 ## OPTIONS
 ##
