diff --git a/CMake/Utils/MyGUIConfigTargets.cmake b/CMake/Utils/MyGUIConfigTargets.cmake
index 6483339..71710de 100644
--- a/CMake/Utils/MyGUIConfigTargets.cmake
+++ b/CMake/Utils/MyGUIConfigTargets.cmake
@@ -170,6 +170,9 @@ function(mygui_app PROJECTNAME SOLUTIONFOLDER)
 			set(MYGUI_EXEC_TYPE WIN32)
 		endif ()
 		add_executable(${PROJECTNAME} ${MYGUI_EXEC_TYPE} ${HEADER_FILES} ${SOURCE_FILES})
+    install(TARGETS ${PROJECTNAME}
+      RUNTIME DESTINATION bin
+    )
 	endif ()
 	set_target_properties(${PROJECTNAME} PROPERTIES FOLDER ${SOLUTIONFOLDER})
 
