diff --git a/Common/CMakeLists.txt b/Common/CMakeLists.txt
index 2b160ff..559e6a8 100644
--- a/Common/CMakeLists.txt
+++ b/Common/CMakeLists.txt
@@ -99,6 +99,8 @@ add_library(${PROJECTNAME} ${HEADER_FILES} ${SOURCE_FILES})
 
 mygui_set_platform_name(${MYGUI_RENDERSYSTEM})
 add_dependencies(${PROJECTNAME} MyGUI.${MYGUI_PLATFORM_NAME}Platform)
-target_link_libraries(${PROJECTNAME} MyGUI.${MYGUI_PLATFORM_NAME}Platform)
+
+find_package(SDL2 CONFIG REQUIRED)
+target_link_libraries(${PROJECTNAME} MyGUI.${MYGUI_PLATFORM_NAME}Platform SDL2::SDL2 SDL2::SDL2main)
 
 add_dependencies(${PROJECTNAME} MyGUIEngine)
