diff --git a/CMakeLists.txt b/CMakeLists.txt
index 645bb32..3ef5c4b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -89,6 +89,11 @@ else()
     add_library(${PROJECT_NAME} STATIC ${src_MAIN} ${src_LINUX_TCP})
 endif()
 
+if(AMQP-CPP_LINUX_TCP)
+    find_path(OPENSSL_INCLUDE_DIR openssl/ssl.h)
+    include_directories(${OPENSSL_INCLUDE_DIR})
+endif()
+
 # install rules
 # ------------------------------------------------------------------------------------------------------
 
