diff --git a/plugins/input/geojson/geojson_index_featureset.cpp b/plugins/input/geojson/geojson_index_featureset.cpp
index 7f915fa..d3a4933 100644
--- a/plugins/input/geojson/geojson_index_featureset.cpp
+++ b/plugins/input/geojson/geojson_index_featureset.cpp
@@ -93,7 +93,7 @@ mapnik::feature_ptr geojson_index_featureset::next()
 #else
         std::fseek(file_.get(), pos.off, SEEK_SET);
         std::vector<char> record;
-        record.resize(pos.second);
+        record.resize(pos.size);
         auto count = std::fread(record.data(), pos.size, 1, file_.get());
         auto const* start = record.data();
         auto const*  end = (count == 1) ? start + record.size() : start;
