diff --git a/CMake/utility.cmake b/CMake/utility.cmake
index 8e7a0c6..ac5e305 100644
--- a/CMake/utility.cmake
+++ b/CMake/utility.cmake
@@ -395,7 +395,7 @@ function( set_compiler_warnings target)
     set(WARNINGS "-Werror"
                  "-Wall")
   elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-    set(WARNINGS "-Werror"
+    set(WARNINGS 
                  "-Wall")
   elseif(MSVC)
     set(WARNINGS "/WX"
