diff --git a/googlemock/CMakeLists.txt b/googlemock/CMakeLists.txt
index e7df8ec..1ecd596 100644
--- a/googlemock/CMakeLists.txt
+++ b/googlemock/CMakeLists.txt
@@ -120,7 +120,8 @@ endif()
 ########################################################################
 #
 # Install rules
-install_project(gmock gmock_main)
+install_project(gmock)
+install_project(gmock_main)
 
 ########################################################################
 #
diff --git a/googletest/CMakeLists.txt b/googletest/CMakeLists.txt
index abdd98b..7d1c01d 100644
--- a/googletest/CMakeLists.txt
+++ b/googletest/CMakeLists.txt
@@ -148,7 +148,8 @@ target_link_libraries(gtest_main PUBLIC gtest)
 ########################################################################
 #
 # Install rules
-install_project(gtest gtest_main)
+install_project(gtest)
+install_project(gtest_main)
 
 ########################################################################
 #
diff --git a/googletest/cmake/internal_utils.cmake b/googletest/cmake/internal_utils.cmake
index 8d8d60a..b08ce28 100644
--- a/googletest/cmake/internal_utils.cmake
+++ b/googletest/cmake/internal_utils.cmake
@@ -315,11 +315,15 @@ function(install_project)
     install(DIRECTORY "${PROJECT_SOURCE_DIR}/include/"
       DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
     # Install the project targets.
+    set (LIB_INSTALL_DST ${CMAKE_INSTALL_LIBDIR})
+    if (${ARGN} MATCHES "_main")
+      set (LIB_INSTALL_DST ${CMAKE_INSTALL_LIBDIR}/manual-link)
+    endif()    
     install(TARGETS ${ARGN}
       EXPORT ${targets_export_name}
       RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
-      ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-      LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+      ARCHIVE DESTINATION "${LIB_INSTALL_DST}"
+      LIBRARY DESTINATION "${LIB_INSTALL_DST}")
     if(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
       # Install PDBs
       foreach(t ${ARGN})
@@ -328,7 +332,7 @@ function(install_project)
         get_target_property(t_pdb_output_directory ${t} PDB_OUTPUT_DIRECTORY)
         install(FILES
           "${t_pdb_output_directory}/\${CMAKE_INSTALL_CONFIG_NAME}/$<$<CONFIG:Debug>:${t_pdb_name_debug}>$<$<NOT:$<CONFIG:Debug>>:${t_pdb_name}>.pdb"
-          DESTINATION ${CMAKE_INSTALL_LIBDIR}
+          DESTINATION ${LIB_INSTALL_DST}
           OPTIONAL)
       endforeach()
     endif()
