diff --git a/CMakeLists.txt b/CMakeLists.txt
index ea81ab129..179e5e13c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,6 +24,14 @@ include(GNUInstallDirs)
 #Note that googlemock target already builds googletest
 option(BUILD_GMOCK "Builds the googlemock subproject" ON)
 option(INSTALL_GTEST "Enable installation of googletest. (Projects embedding googletest may want to turn this OFF.)" ON)
+option(GTEST_USE_CXX17 "Enable CXX 17 standard" OFF)
+
+if (GTEST_USE_CXX17)
+    set(CMAKE_CXX_STANDARD 17)
+    if(MSVC AND MSVC_VERSION GREATER_EQUAL 1913)
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Zc:__cplusplus")
+    endif()
+endif()
 
 if(BUILD_GMOCK)
   add_subdirectory( googlemock )
