diff --git a/src/libgit2/CMakeLists.txt b/src/libgit2/CMakeLists.txt
index dcb4279..20acdc8 100644
--- a/src/libgit2/CMakeLists.txt
+++ b/src/libgit2/CMakeLists.txt
@@ -128,10 +128,16 @@ FILE(WRITE "${PROJECT_BINARY_DIR}/include/${LIBGIT2_FILENAME}.h" ${LIBGIT2_INCLU
 
 # Install
 
+target_include_directories(libgit2package SYSTEM PUBLIC $<INSTALL_INTERFACE:include>)
 install(TARGETS libgit2package
+	EXPORT unofficial-libgit2-targets
 	RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
 	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
 	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(EXPORT unofficial-libgit2-targets
+	NAMESPACE unofficial::libgit2::
+	DESTINATION share/unofficial-libgit2
+)
 install(DIRECTORY ${PROJECT_SOURCE_DIR}/include/git2/
         DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/${LIBGIT2_FILENAME}")
 install(FILES ${PROJECT_BINARY_DIR}/include/git2/experimental.h
