diff --git a/cmake/max_warnings.cmake b/cmake/max_warnings.cmake
index b176d30..e986461 100644
--- a/cmake/max_warnings.cmake
+++ b/cmake/max_warnings.cmake
@@ -13,6 +13,8 @@ if(MSVC)
 
   # Disable broken warnings
   add_definitions(-D_CRT_SECURE_NO_WARNINGS -D_CRT_NONSTDC_NO_DEPRECATE)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /wd4996")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /wd4996")
 elseif(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX)
   if(NOT CMAKE_CXX_FLAGS MATCHES "-Wall")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
