--- a/CMakeLists.txt	2018-11-20 15:06:35.000000000 +0100
+++ b/CMakeLists.txt	2018-11-20 14:59:20.000000000 +0100
@@ -313,7 +313,14 @@
 
 if (HB_HAVE_GLIB)
   add_definitions(-DHAVE_GLIB)
+
+  if(APPLE)
+    find_package(Threads REQUIRED)
+    find_package(unofficial-iconv REQUIRED)
+    find_library(LIBINTL_LIBRARY NAMES intl libintl)
+    list(APPEND THIRD_PARTY_LIBS ${LIBINTL_LIBRARY})
+  endif()
+  
   find_package(unofficial-glib CONFIG REQUIRED)
 
   list(APPEND project_sources ${PROJECT_SOURCE_DIR}/src/hb-glib.cc)
