diff --git a/CMakeLists.txt b/CMakeLists.txt
index c4dc524..bfd15ec 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -431,6 +431,7 @@ endif ()
 
 ## Define harfbuzz library
 add_library(harfbuzz ${project_sources} ${project_extra_sources} ${project_headers})
+target_include_directories(harfbuzz PUBLIC $<INSTALL_INTERFACE:include/harfbuzz>)
 target_link_libraries(harfbuzz ${THIRD_PARTY_LIBS})
 
 
