diff --git a/cmake/npyConfig.cmake.in b/cmake/npyConfig.cmake.in
index 06be483..899ff30 100644
--- a/cmake/npyConfig.cmake.in
+++ b/cmake/npyConfig.cmake.in
@@ -1,7 +1,10 @@
+include(CMakeFindDependencyMacro)
+find_dependency(miniz CONFIG)
+
 get_filename_component(NPY_CMAKE_DIR "${CMAKE_CURRENT_LIST_FILE}" PATH)
 
 if(NOT TARGET npy::npy)
     include("${NPY_CMAKE_DIR}/npyTargets.cmake")
 endif()
 
-set(LIBNPY_lIBRARIES npy::npy)
+set(LIBNPY_lIBRARIES npy::npy)
\ No newline at end of file
