diff --git a/CMakeLists.txt b/CMakeLists.txt
index 41ce88d..9265075 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -95,7 +95,7 @@ endif()
 
 target_include_directories(npy
   PUBLIC
-    $<INSTALL_INTERFACE:build/native/include>
+    $<INSTALL_INTERFACE:include>
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
   PRIVATE
     ${CMAKE_CURRENT_SOURCE_DIR}/src
@@ -124,11 +124,11 @@ set(INSTALL_CONFIGDIR "cmake")
 
 install(TARGETS npy 
   EXPORT npy-targets
-  ARCHIVE DESTINATION "build/native/lib"
-  LIBRARY DESTINATION "build/native/lib"
+  ARCHIVE DESTINATION "lib"
+  LIBRARY DESTINATION "lib"
 )
 
-install(DIRECTORY include/ DESTINATION "build/native/include")
+install(DIRECTORY include/ DESTINATION "include")
 
 install(EXPORT npy-targets
   FILE
