diff --git a/Project/CMake/CMakeLists.txt b/Project/CMake/CMakeLists.txt
index f983343..c6ba5ea 100644
--- a/Project/CMake/CMakeLists.txt
+++ b/Project/CMake/CMakeLists.txt
@@ -88,7 +88,7 @@ endif()
 find_package(TinyXML)
 
 # use system curl if is present
-find_package(CURL)
+find_package(CURL CONFIG REQUIRED)
 
 include_directories(${MediaInfoLib_SOURCES_PATH})
 include_directories(${MediaInfoLib_SOURCES_PATH}/ThirdParty/md5/)
@@ -400,13 +400,7 @@ endif()
 target_link_libraries(mediainfo ${ZenLib_LIBRARY} ${ZLIB_LIBRARIES})
 
 if(MSVC AND BUILD_SHARED_LIBS)
-  get_target_property(LOCATION mediainfo LOCATION_DEBUG)
-  string(REGEX REPLACE "\\.[^.]*$" ".pdb" LOCATION "${LOCATION}")
-  install(FILES ${LOCATION} DESTINATION ${BIN_INSTALL_DIR} CONFIGURATIONS Debug)
-
-  get_target_property(LOCATION mediainfo LOCATION_RELWITHDEBINFO)
-  string(REGEX REPLACE "\\.[^.]*$" ".pdb" LOCATION "${LOCATION}")
-  install(FILES ${LOCATION} DESTINATION ${BIN_INSTALL_DIR} CONFIGURATIONS RelWithDebInfo)
+  install(FILES $<TARGET_PDB_FILE:mediainfo> DESTINATION ${BIN_INSTALL_DIR} OPTIONAL)
 endif()
 
 if (APPLE)
diff --git a/Source/ThirdParty/aes-gladman/aesopt.h b/Source/ThirdParty/aes-gladman/aesopt.h
index d3ac0f9..eeefa40 100644
--- a/Source/ThirdParty/aes-gladman/aesopt.h
+++ b/Source/ThirdParty/aes-gladman/aesopt.h
@@ -183,7 +183,7 @@ Issue Date: 20/12/2007
 	AES_REV_DKS must NOT be defined when such assembler files are
 	built
 */
-#if 1 && defined( _WIN64 ) && defined( _MSC_VER )
+#if 1 && defined( _WIN64 ) && defined( _MSC_VER ) && defined( _M_AMD64 )
 #  define INTEL_AES_POSSIBLE
 #endif
 
diff --git a/Source/ThirdParty/aes-gladman/brg_endian.h b/Source/ThirdParty/aes-gladman/brg_endian.h
index b44c5cb..8c34364 100644
--- a/Source/ThirdParty/aes-gladman/brg_endian.h
+++ b/Source/ThirdParty/aes-gladman/brg_endian.h
@@ -110,7 +110,8 @@ Issue Date: 20/12/2007
       defined( __MRC__ ) || defined( __MVS__ )   || defined( __MWERKS__ ) || \
       defined( sparc )   || defined( __sparc)    || defined( SYMANTEC_C ) || \
       defined( __VOS__ ) || defined( __TIGCC__ ) || defined( __TANDEM )   || \
-      defined( THINK_C ) || defined( __VMCMS__ ) || defined( _AIX )
+      defined( THINK_C ) || defined( __VMCMS__ ) || defined( _AIX )       || \
+      defined( _M_ARM64 ) || defined ( _M_ARM )
 #  define PLATFORM_BYTE_ORDER IS_BIG_ENDIAN
 
 #elif 0     /* **** EDIT HERE IF NECESSARY **** */
