diff --git a/cmake/features.cmake b/cmake/features.cmake
index 8b5563e..a6f0ae8 100644
--- a/cmake/features.cmake
+++ b/cmake/features.cmake
@@ -57,6 +57,16 @@ if(NOT MSVC)
   if(HAVE_SSE2)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -msse2")
   endif()
+elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "arm|ARM")
+  set(HAVE_ALTIVEC OFF)
+  set(HAVE_AVX OFF)
+  set(HAVE_MMX OFF)
+  set(HAVE_SSE OFF)
+  set(HAVE_SSE2 OFF)
+  set(HAVE_SSE3 OFF)
+  set(HAVE_SSE4_1 OFF)
+  set(HAVE_SSE4_2 OFF)
+  set(HAVE_SSSE3 OFF)
 else()
   set(HAVE_ALTIVEC OFF)
   set(HAVE_AVX ON)
