diff --git a/cmake/external/onnxruntime_external_deps.cmake b/cmake/external/onnxruntime_external_deps.cmake
index 7646f3a..a25acb9 100644
--- a/cmake/external/onnxruntime_external_deps.cmake
+++ b/cmake/external/onnxruntime_external_deps.cmake
@@ -946,8 +946,12 @@ if(onnxruntime_USE_KLEIDIAI)
   # Disable the KleidiAI tests
   set(KLEIDIAI_BUILD_TESTS  OFF)
 
-  onnxruntime_fetchcontent_declare(kleidiai URL ${DEP_URL_kleidiai} URL_HASH SHA1=${DEP_SHA1_kleidiai} EXCLUDE_FROM_ALL)
-  onnxruntime_fetchcontent_makeavailable(kleidiai)
+  if(onnxruntime_USE_VCPKG)
+    find_package(KleidiAI CONFIG REQUIRED)
+  else()
+    onnxruntime_fetchcontent_declare(kleidiai URL ${DEP_URL_kleidiai} URL_HASH SHA1=${DEP_SHA1_kleidiai} EXCLUDE_FROM_ALL)
+    onnxruntime_fetchcontent_makeavailable(kleidiai)
+  endif()
   # Fetch Qualcomm's kleidiai library
   if(onnxruntime_USE_QMX_KLEIDIAI_COEXIST)
           onnxruntime_fetchcontent_declare(kleidiai-qmx URL ${DEP_URL_kleidiai-qmx} URL_HASH SHA1=${DEP_SHA1_kleidiai-qmx}
diff --git a/cmake/onnxruntime_mlas.cmake b/cmake/onnxruntime_mlas.cmake
index 56ec9f8..c5822b1 100644
--- a/cmake/onnxruntime_mlas.cmake
+++ b/cmake/onnxruntime_mlas.cmake
@@ -368,8 +368,7 @@ function(setup_kleidiai)
     ${MLAS_SRC_DIR}/kleidiai/qgemm_kleidiai.cpp
     ${MLAS_SRC_DIR}/kleidiai/qnbitgemm_kleidiai.cpp
   )
-  target_link_libraries(onnxruntime_mlas PRIVATE kleidiai)
-  list(APPEND onnxruntime_EXTERNAL_LIBRARIES kleidiai)
+  target_link_libraries(onnxruntime_mlas PRIVATE KleidiAI::kleidiai)
   if(onnxruntime_USE_QMX_KLEIDIAI_COEXIST)
           target_link_libraries(onnxruntime_mlas PRIVATE  kleidiai-qmx)
           target_compile_definitions(onnxruntime_mlas PRIVATE ENABLE_QMX_KERNELS=1)
@@ -386,14 +385,6 @@ function(setup_kleidiai)
     target_compile_definitions(onnxruntime_mlas PRIVATE KLEIDIAI_KERNEL_LOGGING=1)
   endif()
 
-  if (NOT onnxruntime_BUILD_SHARED_LIB)
-    install(TARGETS kleidiai EXPORT ${PROJECT_NAME}Targets
-    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-    FRAMEWORK DESTINATION ${CMAKE_INSTALL_BINDIR})
-  endif()
-
   if(onnxruntime_USE_QMX_KLEIDIAI_COEXIST)
     install(TARGETS kleidiai-qmx EXPORT ${PROJECT_NAME}Targets
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
