diff --git a/spine-cpp/CMakeLists.txt b/spine-cpp/CMakeLists.txt
index 9829b42cd..a5801c3d9 100644
--- a/spine-cpp/CMakeLists.txt
+++ b/spine-cpp/CMakeLists.txt
@@ -8,17 +8,26 @@ file(GLOB INCLUDES "spine-cpp/include/**/*.h")
 file(GLOB SOURCES "spine-cpp/src/**/*.cpp")
 
 add_library(spine-cpp STATIC ${SOURCES} ${INCLUDES})
-target_include_directories(spine-cpp PUBLIC spine-cpp/include)
+target_include_directories(spine-cpp PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/spine-cpp/include>
+    $<INSTALL_INTERFACE:include/spine-cpp>
+)
+
+option(SPINE_USE_STD_FUNCTION "Use std::function listeners" OFF)
+if(SPINE_USE_STD_FUNCTION)
+    target_compile_definitions(spine-cpp PUBLIC SPINE_USE_STD_FUNCTION)
+endif()
 
 add_library(spine-cpp-lite STATIC ${SOURCES} ${INCLUDES} spine-cpp-lite/spine-cpp-lite.cpp)
 target_include_directories(spine-cpp-lite PUBLIC spine-cpp/include spine-cpp-lite)
 
 # Install target
-install(TARGETS spine-cpp EXPORT spine-cpp_TARGETS DESTINATION dist/lib)
-install(FILES ${INCLUDES} DESTINATION dist/include)
+install(TARGETS spine-cpp EXPORT spine-cpp_TARGETS DESTINATION lib)
+install(FILES ${INCLUDES} DESTINATION include/spine-cpp/spine)
 
 # Export target
-export(
+install(
 	EXPORT spine-cpp_TARGETS
-	FILE ${CMAKE_CURRENT_BINARY_DIR}/spine-cppTargets.cmake
+	FILE spine-cppConfig.cmake
+	DESTINATION share/spine-cpp
 	NAMESPACE "Spine::")
\ No newline at end of file
