diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7280464e..e03f667c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -297,6 +297,7 @@ macro(add_common_target_options NAME)
 		endif()
 	elseif(MSVC)
 		target_compile_options(${NAME} PRIVATE /MP /W4 /w44062 /permissive-)
+		target_compile_definitions(${NAME} PRIVATE _CRT_SECURE_NO_WARNINGS)
 		
 		if(WARNINGS_AS_ERRORS)
 			target_compile_options(${NAME} PRIVATE /WX)
