diff --git a/CMakeLists.txt b/CMakeLists.txt
index 56205e3..d727e38 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -75,9 +75,8 @@ if(XTENSOR_USE_XSIMD)
 endif()
 
 if(XTENSOR_USE_TBB)
-    set(CMAKE_MODULE_PATH "${CMAKE_MODULE_PATH}" "${CMAKE_CURRENT_SOURCE_DIR}/cmake/")
-    find_package(TBB REQUIRED)
-    message(STATUS "Found intel TBB: ${TBB_INCLUDE_DIRS}")
+    find_package(TBB CONFIG REQUIRED)
+    message(STATUS "Found TBB: ${TBB_INCLUDE_DIRS}")
 endif()
 
 if(XTENSOR_USE_OPENMP)
