diff --git a/include/sciter-win-main.cpp b/include/sciter-win-main.cpp
index 9f2699e..104b70a 100644
--- a/include/sciter-win-main.cpp
+++ b/include/sciter-win-main.cpp
@@ -1,4 +1,6 @@
-#include "stdafx.h"
+// It is better to disable stdafx.h by default.
+// And enable it explicitly via /Fstdafx.h compiler option (C/C++ - Advanced - Force Include File).
+// #include "stdafx.h"

 #include <vector>

