diff --git a/CMakeLists.txt b/CMakeLists.txt
index c155ade..319351c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,7 +23,7 @@ target_include_directories(
         $<BUILD_INTERFACE:${${PROJECT_NAME}_SOURCE_DIR}/include>
         $<INSTALL_INTERFACE:include>)
 
-set(HEADER_FILES task_thread_pool.hpp)
+set(HEADER_FILES include/task_thread_pool.hpp)
 
 set_target_properties(${PROJECT_NAME} PROPERTIES PUBLIC_HEADER "${HEADER_FILES}")
 
