diff --git a/Common.cmake b/Common.cmake
index 31d8116..f46cbc3 100644
--- a/Common.cmake
+++ b/Common.cmake
@@ -12,9 +12,9 @@ endif()
 #
 # This method was taken from https://www.pragmaticlinux.com/2022/07/enable-compiler-warnings-with-cmake/
 target_compile_options(${PROJECT_NAME} PRIVATE
-    $<$<CXX_COMPILER_ID:MSVC>:/W4 /WX>
+    $<$<CXX_COMPILER_ID:MSVC>:/W4>
     $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:
-        -Wall -Wextra -Wpedantic -Werror
+        -Wall -Wextra -Wpedantic
 
         # We're using the `long long` type intentionally. Although it's not part of C++98, in
         # practice it is usually supported even by ancient compilers with very limited C++11
