diff -r 8df7a59b5528 -r 46ec9baae30c include/begin_code.h
--- a/include/begin_code.h	Fri Sep 22 11:25:52 2017 -0700
+++ b/include/begin_code.h	Sun Sep 24 14:32:30 2017 +0300
@@ -54,8 +54,10 @@
 #   else
 #    define DECLSPEC	__declspec(dllimport)
 #   endif
+#  elif defined(_DLL)
+#   define DECLSPEC	__declspec(dllexport)
 #  else
-#   define DECLSPEC	__declspec(dllexport)
+#   define DECLSPEC
 #  endif
 # elif defined(__OS2__)
 #  ifdef __WATCOMC__
