diff --git a/configure.py b/configure.py
index 2474860a5..90264dfb8 100755
--- a/configure.py
+++ b/configure.py
@@ -2132,7 +2132,7 @@ def create_template_vars(source_paths, build_paths, options, modules, cc, arch,
         'maintainer_mode': options.maintainer_mode,
 
         'out_dir': build_dir,
-        'build_dir': build_paths.build_dir,
+        'build_dir': os.path.abspath(build_paths.build_dir),
         'module_info_dir': build_paths.doc_module_info,
 
         'doc_stamp_file': os.path.join(build_paths.build_dir, 'doc.stamp'),
