diff --git a/cachelib/CMakeLists.txt b/cachelib/CMakeLists.txt
index e17ff27..66c47f2 100644
--- a/cachelib/CMakeLists.txt
+++ b/cachelib/CMakeLists.txt
@@ -115,7 +115,7 @@ find_package(Threads REQUIRED)
 find_package(Boost REQUIRED COMPONENTS
              system filesystem regex context program_options thread)
 find_package(Gflags REQUIRED)
-find_package(Glog REQUIRED)
+find_package(glog CONFIG REQUIRED)
 find_package(GTest REQUIRED)
 include(GoogleTest)
 enable_testing()
