diff --git a/CMakeLists.txt b/CMakeLists.txt
index 34b947883..4533874db 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -496,11 +496,11 @@ set(pkgcfg_requiresprivate "")
 if("${UA_ARCHITECTURE}" STREQUAL "posix")
     list(APPEND open62541_LIBRARIES "m")
     list(APPEND pkgcfg_libsprivate "-lm")
-    if(UA_MULTITHREADING GREATER_EQUAL 100 OR UA_BUILD_UNIT_TESTS)
+    if(UA_MULTITHREADING GREATER_EQUAL 100 OR UA_BUILD_UNIT_TESTS AND NOT ANDROID)
         list(APPEND open62541_PUBLIC_LIBRARIES "pthread")
         list(APPEND pkgcfg_libsprivate "-lpthread")
     endif()
-    if(NOT APPLE AND (NOT ${CMAKE_SYSTEM_NAME} MATCHES "OpenBSD"))
+    if(NOT APPLE AND (NOT ${CMAKE_SYSTEM_NAME} MATCHES "OpenBSD") AND NOT ANDROID)
         list(APPEND open62541_LIBRARIES "rt")
         list(APPEND pkgcfg_libsprivate "-lrt")
     endif()
