diff --git a/CMakeLists.txt b/CMakeLists.txt
index 64b9a96..8e66848 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -806,14 +806,14 @@ install(TARGETS open62541
         EXPORT open62541Targets
         LIBRARY DESTINATION ${LIB_INSTALL_DIR}
         ARCHIVE DESTINATION ${LIB_INSTALL_DIR}
-        RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}
+        RUNTIME DESTINATION ${BIN_INSTALL_DIR}
         INCLUDES DESTINATION include/open62541 include)
 else()
 install(TARGETS open62541
         EXPORT open62541Targets
         LIBRARY DESTINATION ${LIB_INSTALL_DIR}
         ARCHIVE DESTINATION ${LIB_INSTALL_DIR}
-        RUNTIME DESTINATION ${CMAKE_INSTALL_PREFIX}
+        RUNTIME DESTINATION ${BIN_INSTALL_DIR}
         INCLUDES DESTINATION include)
 endif()
 
@@ -868,7 +868,7 @@ if(NOT UA_ENABLE_AMALGAMATION)
                   ${historizing_default_plugin_headers}
             DESTINATION ${open62541_install_include_dir})
 else()
-    # Export amalgamated header open62541.h which is generated due to build of 
+    # Export amalgamated header open62541.h which is generated due to build of
     # open62541-object
     install(FILES ${PROJECT_BINARY_DIR}/open62541.h DESTINATION include)
 endif()
