diff --git a/CMakeLists.txt b/CMakeLists.txt
index 860a8be..f0bc14b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -122,15 +122,6 @@ IF (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
     SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility=hidden")
 ENDIF()

-# Set some debug vs opt flags
-if (ALEMBIC_DEBUG_WARNINGS_AS_ERRORS AND "${CMAKE_BUILD_TYPE}" MATCHES "Debug" AND NOT MSVC)
-    add_compile_options(-Wall -Werror -Wextra -Wno-unused-parameter -Wno-deprecated -Wunused-local-typedefs)
-    if((CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 7.0) OR
-        CMAKE_CXX_COMPILER_ID MATCHES "CLANG")
-        add_compile_options( -Wno-error=implicit-fallthrough)
-    endif()
-endif()
-
 IF (NOT ${WINDOWS})
     SET(EXTERNAL_MATH_LIBS "-lm")
 ELSE()
