--- a/dlib/cmake_utils/find_blas.cmake
+++ b/dlib/cmake_utils/find_blas.cmake
@@ -414,7 +414,7 @@
 if (lapack_found)
    include(CheckFunctionExists)
    include(CheckFortranFunctionExists)
-   set(CMAKE_REQUIRED_LIBRARIES ${lapack_libraries})
+   set(CMAKE_REQUIRED_LIBRARIES ${lapack_libraries} ${blas_libraries})
 
    check_function_exists("sgesv" LAPACK_FOUND_C_UNMANGLED)
    check_function_exists("sgesv_" LAPACK_FOUND_C_MANGLED)
