diff --git a/dlib/CMakeLists.txt b/dlib/CMakeLists.txt
index a1b6473f..44caafdd 100644
--- a/dlib/CMakeLists.txt
+++ b/dlib/CMakeLists.txt
@@ -38,6 +38,9 @@ endif()
 include(cmake_utils/release_build_by_default)
 include(cmake_utils/use_cpp_11.cmake)
 
+# Set preference for linking to static libraries if DLIB_PREFER_STATIC_LIBS is set
+include(cmake_utils/set_preferred_library_type.cmake)
+
 # Set DLIB_VERSION in the including CMake file so they can use it to do whatever they want. 
 get_directory_property(has_parent PARENT_DIRECTORY)
 if(has_parent)
diff --git a/dlib/cmake_utils/set_preferred_library_type.cmake b/dlib/cmake_utils/set_preferred_library_type.cmake
new file mode 100644
index 00000000..7692fb2b
--- /dev/null
+++ b/dlib/cmake_utils/set_preferred_library_type.cmake
@@ -0,0 +1,7 @@
+if(DLIB_PREFER_STATIC_LIBS)
+  if(UNIX OR MINGW)
+     set(CMAKE_FIND_LIBRARY_SUFFIXES .a .tbd .dylib .so)
+  else()
+     set(CMAKE_FIND_LIBRARY_SUFFIXES .lib .dll) 
+  endif()
+endif()
