diff --git a/include/CMakeLists.txt b/include/CMakeLists.txt
index 0dc6943..e6a45d2 100644
--- a/include/CMakeLists.txt
+++ b/include/CMakeLists.txt
@@ -14,5 +14,5 @@ install(TARGETS simdjson-headers
 install(EXPORT simdjson-headers-config
         FILE simdjson-headers-config.cmake
         DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/simdjson)
-install(DIRECTORY simdjson DESTINATION include FILES_MATCHING PATTERN *.h)
+install(DIRECTORY simdjson DESTINATION include FILES_MATCHING PATTERN *.h PATTERN *.hpp)
 install(FILES simdjson.h DESTINATION include)
