diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3861def..88265fb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -110,9 +110,11 @@ endif()
 
 # find_package Python has replaced PythonInterp and PythonLibs since cmake 3.12
 # Use the following command in the future; now this is only compatible with the latest pybind11
-# find_package(Python ${PY_VERSION} COMPONENTS Interpreter Development REQUIRED)
-find_package(PythonInterp ${PY_VERSION} REQUIRED)
-find_package(PythonLibs ${PY_VERSION})
+if(WIN32)
+  find_package(Python ${PY_VERSION} COMPONENTS Interpreter REQUIRED)
+else()
+  find_package(Python ${PY_VERSION} COMPONENTS Interpreter Development REQUIRED)
+endif()
 
 if(CMAKE_SYSTEM_NAME STREQUAL "AIX")
   set(CMAKE_NO_SYSTEM_FROM_IMPORTED 1)
@@ -441,10 +439,10 @@ if(BUILD_ONNX_PYTHON)
                              $<BUILD_INTERFACE:${ONNX_ROOT}>
                              $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
                              $<INSTALL_INTERFACE:include>
-                             ${PYTHON_INCLUDE_DIR})
+                             ${Python_INCLUDE_DIRS})
 
   # pybind11 is a header only lib
-  find_package(pybind11 2.2)
+  find_package(pybind11 2.2 CONFIG REQUIRED)
   if(pybind11_FOUND)
     target_include_directories(onnx_cpp2py_export PUBLIC
       ${pybind11_INCLUDE_DIRS})
