diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4799557..8bed591 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -181,6 +181,7 @@ endif()
 
 list(APPEND CMAKE_MODULE_PATH ${ONNX_ROOT}/cmake/external)
 if(NOT ONNX_BUILD_CUSTOM_PROTOBUF)
+  find_package(Protobuf CONFIG REQUIRED)
   if((ONNX_USE_LITE_PROTO AND TARGET protobuf::libprotobuf-lite) OR ((NOT ONNX_USE_LITE_PROTO) AND TARGET protobuf::libprotobuf))
     # Sometimes we need to use protoc compiled for host architecture while linking
     # libprotobuf against target architecture. See https://github.com/caffe2/caffe
diff --git a/cmake/ONNXConfig.cmake.in b/cmake/ONNXConfig.cmake.in
index a5129bf..19cee03 100644
--- a/cmake/ONNXConfig.cmake.in
+++ b/cmake/ONNXConfig.cmake.in
@@ -5,7 +5,8 @@
 set(ONNX_VERSION "@ONNX_VERSION@")
 
 if((NOT @@ONNX_USE_PROTOBUF_SHARED_LIBS@@) AND @@Build_Protobuf@@)
-  find_package(Protobuf REQUIRED CONFIG)
+  include(CMakeFindDependencyMacro)
+  find_dependency(Protobuf CONFIG)
 endif()
 
 # import targets
