diff --git a/CMakeLists.txt b/CMakeLists.txt
index 628dcaa..300e4ea 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -111,8 +111,8 @@ endif()
 # find_package Python has replaced PythonInterp and PythonLibs since cmake 3.12
 # Use the following command in the future; now this is only compatible with the latest pybind11
 # find_package(Python ${PY_VERSION} COMPONENTS Interpreter Development REQUIRED)
-find_package(PythonInterp ${PY_VERSION} REQUIRED)
-find_package(PythonLibs ${PY_VERSION})
+find_package(Python3 ${PY_VERSION} COMPONENTS Interpreter REQUIRED)
+set(PYTHON_EXECUTABLE ${Python3_EXECUTABLE})
 
 if(CMAKE_SYSTEM_NAME STREQUAL "AIX")
   set(CMAKE_NO_SYSTEM_FROM_IMPORTED 1)
@@ -422,6 +422,7 @@ target_link_libraries(onnx PUBLIC onnx_proto)
 add_onnx_global_defines(onnx)
 
 if(BUILD_ONNX_PYTHON)
+  find_package(Python3 ${PY_VERSION} COMPONENTS Development REQUIRED)
   if("${PY_EXT_SUFFIX}" STREQUAL "")
     if(MSVC)
       set(PY_EXT_SUFFIX ".pyd")
@@ -441,10 +442,13 @@ if(BUILD_ONNX_PYTHON)
                              $<BUILD_INTERFACE:${ONNX_ROOT}>
                              $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
                              $<INSTALL_INTERFACE:include>
-                             ${PYTHON_INCLUDE_DIR})
-
+                             ${Python3_INCLUDE_DIRS})
+  target_link_directories(onnx_cpp2py_export PRIVATE
+                          ${Python3_LIBRARY_DIRS})
+  target_link_libraries(onnx_cpp2py_export PRIVATE
+                        ${Python3_LIBRARIES})
   # pybind11 is a header only lib
-  find_package(pybind11 2.2)
+  find_package(pybind11 2.2 CONFIG REQUIRED)
   if(pybind11_FOUND)
     target_include_directories(onnx_cpp2py_export PUBLIC
       ${pybind11_INCLUDE_DIRS})
@@ -687,6 +691,27 @@ endif()
 
 include(GNUInstallDirs)
 
+# install protobuf files
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx-data.proto
+              ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx-data.proto3
+              ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx-ml.proto
+              ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx-ml.proto3
+              ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx-operators-ml.proto
+              ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx-operators-ml.proto3
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/onnx
+)
+# install python files
+if(BUILD_ONNX_PYTHON)
+  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx_data_pb.py
+                ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx_data_pb2.py
+                ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx_ml_pb2.py
+                ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx_operators_ml_pb2.py
+                ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx_operators_pb.py
+                ${CMAKE_CURRENT_BINARY_DIR}/onnx/onnx_pb.py
+          DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/onnx
+  )
+endif()
+
 install(DIRECTORY ${ONNX_ROOT}/onnx
         DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
         FILES_MATCHING

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 300e4ea..7773f7a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -356,7 +356,13 @@ list(REMOVE_ITEM __tmp_srcs "${ONNX_ROOT}/onnx/cpp2py_export.cc")
 list(REMOVE_ITEM __tmp_srcs ${onnx_gtests_src})
 list(APPEND ONNX_SRCS ${__tmp_srcs})
 
-add_library(onnx_proto ${ONNX_PROTO_SRCS} ${ONNX_PROTO_HDRS})
+if(WIN32)
+  # see .azure-pipelines/Windwos-CI.yml
+  add_library(onnx_proto STATIC ${ONNX_PROTO_SRCS} ${ONNX_PROTO_HDRS})
+else()
+  add_library(onnx_proto ${ONNX_PROTO_SRCS} ${ONNX_PROTO_HDRS})
+endif()
+
 target_include_directories(onnx_proto PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
   $<INSTALL_INTERFACE:include>
@@ -410,6 +416,8 @@ if(CMAKE_SYSTEM_NAME STREQUAL "AIX")
   # whole-archive linker option not available on AIX.
   # So, create a object library
   add_library(onnx OBJECT ${ONNX_SRCS})
+elseif(WIN32)
+  add_library(onnx STATIC ${ONNX_SRCS}) # see .azure-pipelines/Windwos-CI.yml
 else()
   add_library(onnx ${ONNX_SRCS})
 endif()
@@ -738,11 +746,17 @@ install(EXPORT ONNXTargets DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/ONNX")
 install(TARGETS
   onnx onnx_proto
   onnxifi onnxifi_dummy onnxifi_loader
-  EXPORT ONNXTargets DESTINATION ${CMAKE_INSTALL_LIBDIR})
+  EXPORT ONNXTargets
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
 if(NOT ANDROID AND NOT IOS)
   install(TARGETS onnxifi_wrapper
-    EXPORT ONNXTargets DESTINATION lib)
+    EXPORT ONNXTargets
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif()
 
 if(ONNXIFI_DUMMY_BACKEND)
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7773f7a..55047d6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -647,7 +647,12 @@ endif()
 
 if (NOT ANDROID AND NOT IOS)
   # ---[ ONNXIFI wrapper
-  add_library(onnxifi_wrapper MODULE onnx/onnxifi_wrapper.c)
+  if(WIN32)
+    add_library(onnxifi_wrapper onnx/onnxifi_wrapper.c) # for ${arch}-windows-static triplets
+  else()
+    add_library(onnxifi_wrapper onnx/onnxifi_wrapper.c)
+  endif()
+
   if(MSVC)
     add_msvc_runtime_flag(onnxifi_wrapper)
   endif()
@@ -677,7 +682,11 @@ if (NOT ANDROID AND NOT IOS)
 endif()
 
 # ---[ ONNXIFI dummy backend
-add_library(onnxifi_dummy SHARED onnx/onnxifi_dummy.c)
+if(WIN32)
+  add_library(onnxifi_dummy onnx/onnxifi_dummy.c) # for ${arch}-windows-static triplets
+else()
+  add_library(onnxifi_dummy SHARED onnx/onnxifi_dummy.c)
+endif()
 
 if(ONNXIFI_ENABLE_EXT)
   add_definitions(-DONNXIFI_ENABLE_EXT=ON)
