diff --git a/CMakeLists.txt b/CMakeLists.txt
index 044996e1b..829a92e8b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -491,6 +491,14 @@ if(ONNX_INSTALL)
           DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
           FILES_MATCHING
           PATTERN "*.h")
+  install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/onnx
+          DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+          FILES_MATCHING
+          PATTERN "*.proto")
+  install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/onnx
+          DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+          FILES_MATCHING
+          PATTERN "*.proto3")
 endif()
 
 configure_file(
