diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6f1d56949..f6b846068 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -650,6 +650,11 @@ if(ONNX_INSTALL)
           PATTERN "*.h")
 endif()
 
+install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/onnx DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+        FILES_MATCHING PATTERN "*.proto")
+install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/onnx DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+        FILES_MATCHING PATTERN "*.proto3")
+
 configure_file(
   ${PROJECT_SOURCE_DIR}/cmake/ONNXConfigVersion.cmake.in
   ${PROJECT_BINARY_DIR}/ONNXConfigVersion.cmake
