diff --git a/ci/scripts/java_jni_manylinux_build.sh b/ci/scripts/java_jni_manylinux_build.sh
index 7480aa7..0e107c2 100644
--- a/ci/scripts/java_jni_manylinux_build.sh
+++ b/ci/scripts/java_jni_manylinux_build.sh
@@ -136,12 +136,10 @@ echo "=== Checking shared dependencies for libraries ==="
 pushd ${dist_dir}
 archery linking check-dependencies \
   --allow ld-linux-x86-64 \
-  --allow libc \
   --allow libdl \
   --allow libgcc_s \
   --allow libm \
   --allow libpthread \
-  --allow librt \
   --allow libstdc++ \
   --allow libz \
   --allow linux-vdso \
