--- a/corelib/src/CMakeLists.txt
+++ b/corelib/src/CMakeLists.txt
@@ -393,10 +392,8 @@ ENDIF(WITH_TORO)
 	IF(g2o_FOUND)
 	  SET(LIBRARIES ${LIBRARIES}
 	     g2o::core
-	     g2o::solver_eigen
-	     g2o::solver_pcg
-	     g2o::types_slam2d
-	     g2o::types_slam3d
+        g2o::stuff 
+        g2o::types_icp 
         g2o::types_sba)
 	  IF(TARGET g2o::solver_csparse)
 	    SET(LIBRARIES ${LIBRARIES}
