Index: gappa-1.3.5/src/backends/coq_common.cpp
===================================================================
--- gappa-1.3.5.orig/src/backends/coq_common.cpp
+++ gappa-1.3.5/src/backends/coq_common.cpp
@@ -13,6 +13,7 @@
 #include <cstring>
 #include <iostream>
 #include <sstream>
+#include <cctype>
 #include "backends/backend.hpp"
 #include "backends/coq_common.hpp"
 #include "numbers/interval_utility.hpp"
Index: gappa-1.3.5/src/backends/holl.cpp
===================================================================
--- gappa-1.3.5.orig/src/backends/holl.cpp
+++ gappa-1.3.5/src/backends/holl.cpp
@@ -12,6 +12,7 @@
 #include <map>
 #include <iostream>
 #include <sstream>
+#include <cctype>
 #include "backends/backend.hpp"
 #include "numbers/interval_utility.hpp"
 #include "numbers/real.hpp"
Index: gappa-1.3.5/src/parser/lexer.cpp
===================================================================
--- gappa-1.3.5.orig/src/parser/lexer.cpp
+++ gappa-1.3.5/src/parser/lexer.cpp
@@ -45,7 +45,7 @@
 #include <string.h>
 #include <errno.h>
 #include <stdlib.h>
-
+#include <cctype>
 /* end standard C headers. */
 
 /* flex integer type definitions */
