diff --git a/lib/includes/gnutls/ocsp.h b/lib/includes/gnutls/ocsp.h
index 9e271476cf..80d8ccc8f5 100644
--- a/lib/includes/gnutls/ocsp.h
+++ b/lib/includes/gnutls/ocsp.h
@@ -224,9 +224,9 @@ int gnutls_ocsp_resp_get_single(gnutls_ocsp_resp_const_t resp, unsigned indx,
 				gnutls_datum_t *issuer_name_hash,
 				gnutls_datum_t *issuer_key_hash,
 				gnutls_datum_t *serial_number,
-				unsigned int *cert_status, time_t *this_update,
+				gnutls_ocsp_cert_status_t *cert_status, time_t *this_update,
 				time_t *next_update, time_t *revocation_time,
-				unsigned int *revocation_reason);
+				gnutls_x509_crl_reason_t *revocation_reason);
 int gnutls_ocsp_resp_get_extension(gnutls_ocsp_resp_const_t resp, unsigned indx,
 				   gnutls_datum_t *oid, unsigned int *critical,
 				   gnutls_datum_t *data);
diff --git a/lib/x509/ocsp.c b/lib/x509/ocsp.c
index 0120129d50..3a46f2d174 100644
--- a/lib/x509/ocsp.c
+++ b/lib/x509/ocsp.c
@@ -1433,9 +1433,9 @@ int gnutls_ocsp_resp_get_single(gnutls_ocsp_resp_const_t resp, unsigned indx,
 				gnutls_datum_t *issuer_name_hash,
 				gnutls_datum_t *issuer_key_hash,
 				gnutls_datum_t *serial_number,
-				unsigned int *cert_status, time_t *this_update,
+				gnutls_ocsp_cert_status_t *cert_status, time_t *this_update,
 				time_t *next_update, time_t *revocation_time,
-				unsigned int *revocation_reason)
+				gnutls_x509_crl_reason_t *revocation_reason)
 {
 	char name[MAX_NAME_SIZE];
 	int ret, result;
