diff --git a/src/disjoint_sets.h b/src/disjoint_sets.h
index d9da598..ae5f9b4 100644
--- a/src/disjoint_sets.h
+++ b/src/disjoint_sets.h
@@ -96,7 +96,7 @@ class DisjointSets {
     components.resize(mData.size());
     int lonelyNodes = 0;
     std::unordered_map<uint32_t, int> toLabel;
-    for (size_t i = 0; i < mData.size(); ++i) {
+    for (std::size_t i = 0; i < mData.size(); ++i) {
       // we optimize for connected component of size 1
       // no need to put them into the hashmap
       auto iParent = find(i);
