diff --git a/adm/cmake/occt_defs_flags.cmake b/adm/cmake/occt_defs_flags.cmake
index 203ca6b..4ff433a 100644
--- a/adm/cmake/occt_defs_flags.cmake
+++ b/adm/cmake/occt_defs_flags.cmake
@@ -18,7 +18,7 @@ endif()
 
 # Turn off the "improved inline analyzer" that stalls on x86 release builds using MSVC 16.7 for 32-bit
 if (NOT CMAKE_SIZEOF_VOID_P EQUAL 8)
-  if (MSVC)
+  if (MSVC AND MSVC_VERSION GREATER 1919)
     set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /d2DeepThoughtInliner-")
     set (CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} /d2DeepThoughtInliner-")
   endif()
