diff --git a/CMakeLists.txt b/CMakeLists.txt
index fbcede5..66b127d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -563,7 +563,9 @@ if (CAN_USE_FREETYPE)
   if (USE_FREETYPE)
     message (STATUS "Info: FreeType is used by OCCT")
     add_definitions (-DHAVE_FREETYPE)
-    OCCT_INCLUDE_CMAKE_FILE ("adm/cmake/freetype")
+    find_package(freetype CONFIG REQUIRED)
+	get_target_property(FREETYPE_INCLUDE_DIR freetype INTERFACE_INCLUDE_DIRECTORIES)
+    list (APPEND 3RDPARTY_INCLUDE_DIRS "${FREETYPE_INCLUDE_DIR}")
   else()
     OCCT_CHECK_AND_UNSET_GROUP ("3RDPARTY_FREETYPE")
     OCCT_CHECK_AND_UNSET ("3RDPARTY_FREETYPE_INCLUDE_DIR_freetype2")
