diff --git a/CMakeLists.txt b/CMakeLists.txt
index d3252b3..33d50b8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -634,6 +634,9 @@ if (CAN_USE_FREEIMAGE)
     add_definitions (-DHAVE_FREEIMAGE)
     find_package(freeimage CONFIG REQUIRED)
     set(CSF_FreeImagePlus freeimage::FreeImagePlus)
+    if ("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
+        add_definitions (-DDEBUG_FREEIMAGE_LIB)
+    endif()
   else()
     OCCT_CHECK_AND_UNSET_GROUP ("3RDPARTY_FREEIMAGE")
     OCCT_CHECK_AND_UNSET ("INSTALL_FREEIMAGE")
diff --git a/src/Image/Image_AlienPixMap.cxx b/src/Image/Image_AlienPixMap.cxx
index f869e8f..f12e17a 100644
--- a/src/Image/Image_AlienPixMap.cxx
+++ b/src/Image/Image_AlienPixMap.cxx
@@ -21,7 +21,11 @@
   #include <FreeImage.h>
 
   #ifdef _MSC_VER
-    #pragma comment( lib, "FreeImage.lib" )
+    #ifdef DEBUG_FREEIMAGE_LIB
+        #pragma comment( lib, "FreeImaged.lib" )
+    #else
+        #pragma comment( lib, "FreeImage.lib" )
+    #endif
   #endif
 #elif defined(HAVE_WINCODEC)
   #include <wincodec.h>
