diff --git a/adm/cmake/vtk.cmake b/adm/cmake/vtk.cmake
index a81cdf9bb2..f265dc3489 100644
--- a/adm/cmake/vtk.cmake
+++ b/adm/cmake/vtk.cmake
@@ -98,6 +98,11 @@ if (VTK_FOUND)
       # get paths from corresponding variables
       if (${VTK_LIBRARY}_INCLUDE_DIRS AND EXISTS "${${VTK_LIBRARY}_INCLUDE_DIRS}")
         list (APPEND 3RDPARTY_VTK_INCLUDE_DIRS "${${VTK_LIBRARY}_INCLUDE_DIRS}")
+      else()
+        get_target_property(${VTK_LIBRARY}_INCLUDE_DIRS ${VTK_LIBRARY} INTERFACE_INCLUDE_DIRECTORIES)
+        if (${VTK_LIBRARY}_INCLUDE_DIRS AND EXISTS "${${VTK_LIBRARY}_INCLUDE_DIRS}")
+          list (APPEND 3RDPARTY_VTK_INCLUDE_DIRS "${${VTK_LIBRARY}_INCLUDE_DIRS}")
+        endif()
       endif()
 
       if (${VTK_LIBRARY}_LIBRARY_DIRS AND EXISTS "${${VTK_LIBRARY}_LIBRARY_DIRS}")
