diff --git a/src/Makefile.am b/src/Makefile.am
index 215a434ae7..afcadb4682 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -18,8 +18,7 @@ LDFLAGS += -version-info $(B2_LIBRARY_VERSION)
 lib_LTLIBRARIES = libb2.la
 libb2_la_LIBADD = # -lgomp -lpthread
 libb2_la_CPPFLAGS =  -DSUFFIX=  \
-                     $(LTDLINCL) \
-                     ${top_builddir}/src/
+                     $(LTDLINCL)
 
 include_HEADERS = blake2.h
 
