diff --git a/meson.build b/meson.build
index 33ec4f1..fe5516f 100644
--- a/meson.build
+++ b/meson.build
@@ -144,6 +144,9 @@ install_headers(
 # src
 library_includes = include_directories('include')
 library_options = ['-DBUILDING_LIBDICOM']
+if meson.is_cross_build()
+  dict_lookup = [ 'dicom-dict-lookup.c', 'dicom-dict-lookup.h' ]
+else
 dict_build = executable(
   'dicom-dict-build',
   files('src/dicom-dict-build.c', 'src/dicom-dict-tables.c'),
@@ -156,6 +159,7 @@ dict_lookup = custom_target(
   command : [dict_build, '@OUTPUT@'],
   output : ['dicom-dict-lookup.c', 'dicom-dict-lookup.h'],
 )
+endif
 library_sources = [dict_lookup] + files(
   'src/dicom-data.c',
   'src/dicom-dict-tables.c',
