diff --git a/cds/compiler/gcc/compiler_macro.h b/cds/compiler/gcc/compiler_macro.h
index f954d6ca8e..a4846bdeb5 100644
--- a/cds/compiler/gcc/compiler_macro.h
+++ b/cds/compiler/gcc/compiler_macro.h
@@ -133,7 +133,7 @@
 #   endif
 #endif
 
-#if CDS_OS_TYPE == CDS_OS_MINGW
+#if (CDS_OS_TYPE == CDS_OS_MINGW) || defined(_MSC_VER)
 #   ifdef CDS_BUILD_LIB
 #       define CDS_EXPORT_API          __declspec(dllexport)
 #   elif !defined(CDS_BUILD_STATIC_LIB)
