--- a/src/utilities/_hypre_utilities.h	Thu Jun 09 15:56:18 2016
+++ b/src/utilities/_hypre_utilities.h	Wed Jan 04 18:08:27 2017
@@ -13,6 +13,7 @@
 #ifndef hypre_UTILITIES_HEADER
 #define hypre_UTILITIES_HEADER
 
+#include "_hypre_export.h"
 #include "HYPRE_utilities.h"
 
 #ifdef HYPRE_USING_OPENMP
@@ -610,7 +611,7 @@
 #ifdef HYPRE_TIMING_GLOBALS
 hypre_TimingType *hypre_global_timing = NULL;
 #else
-extern hypre_TimingType *hypre_global_timing;
+extern HYPRE_EXPORT hypre_TimingType *hypre_global_timing;
 #endif
 
 /*-------------------------------------------------------
@@ -781,7 +782,7 @@
  * Global variable used in hypre error checking
  *--------------------------------------------------------------------------*/
 
-extern HYPRE_Int hypre__global_error;
+extern HYPRE_EXPORT HYPRE_Int hypre__global_error;
 #define hypre_error_flag  hypre__global_error
 
 /*--------------------------------------------------------------------------
