diff -Nuar a/cmake/win32_compiler_options.cmake b/cmake/win32_compiler_options.cmake
--- a/cmake/win32_compiler_options.cmake	2018-04-06 20:24:17.000000000 +0200
+++ b/cmake/win32_compiler_options.cmake	2018-04-28 19:40:54.534593200 +0200
@@ -67,10 +67,6 @@
     endif()
 endif()
 
-set(CMAKE_INCLUDE_PATH "c:/OSGeo4W64/include;$ENV{CMAKE_INCLUDE_PATH}")
-set(CMAKE_LIBRARY_PATH "c:/OSGeo4W64/lib;$ENV{CMAKE_LIBRARY_PATH}")
-set(CMAKE_PREFIX_PATH "c:/OSGeo4W64/cmake;$ENV{CMAKE_LIBRARY_PATH}")
-
 #ABELL (& gadomski) - WHY?
 set(PDAL_PLATFORM_WIN32 1)
 set(WINSOCK_LIBRARY ws2_32)
