diff --git a/CMakeLists.txt b/CMakeLists.txt
index dc146d1..9b82544 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -140,7 +140,7 @@ get_git_head_revision(GIT_REFSPEC GIT_SHA1)
 
 configure_file(
   "${PROJECT_SOURCE_DIR}/gitsha.cpp.in"
-  "${PROJECT_SOURCE_DIR}/pdal/gitsha.cpp")
+  "${PROJECT_BINARY_DIR}/pdal/gitsha.cpp")
 
 # needs to come before configuration of pdal_features
 if(APPLE)
@@ -201,6 +201,7 @@ file(GLOB BASE_SRCS
     ${PDAL_IO_DIR}/*.cpp
     ${PDAL_KERNELS_DIR}/*.cpp
     ${PDAL_SRC_DIR}/*.cpp
+    ${PROJECT_BINARY_DIR}/pdal/gitsha.cpp
     ${PDAL_SRC_DIR}/compression/*.cpp)
 file(GLOB_RECURSE PRIVATE_SRCS
     ${PDAL_FILTERS_DIR}/private/*.cpp
