diff --git a/CMakeLists.txt b/CMakeLists.txt
index 43e446a..c4c7d6f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -339,6 +339,30 @@ install(DIRECTORY ${PDAL_FILTERS_DIR}
 install(FILES ${DIMENSION_OUTFILE} ${pdal_features_hpp}
   DESTINATION include/pdal
 )
+# Install vendor headers
+install(DIRECTORY ${PDAL_VENDOR_ARBITER_DIR}
+    DESTINATION include/pdal/vendor
+    FILES_MATCHING PATTERN "*.hpp"
+    PATTERN "private" EXCLUDE
+)
+
+install(DIRECTORY ${PDAL_VENDOR_JSONCPP_DIR}
+    DESTINATION include/pdal/vendor
+    FILES_MATCHING PATTERN "*.h"
+    PATTERN "private" EXCLUDE
+)
+
+install(DIRECTORY ${PDAL_VENDOR_KAZHDAN_DIR}
+    DESTINATION include/pdal/vendor
+    FILES_MATCHING PATTERN "*.h"
+    PATTERN "private" EXCLUDE
+)
+
+install(DIRECTORY ${PDAL_VENDOR_NANOFLANN_DIR}
+    DESTINATION include/pdal/vendor
+    FILES_MATCHING PATTERN "*.hpp"
+    PATTERN "private" EXCLUDE
+)
 
 #
 # CPACK
@@ -348,7 +373,7 @@ add_custom_target(dist COMMAND ${CMAKE_MAKE_PROGRAM} package_source)
 
 export(
     TARGETS
-        ${PDAL_BASE_LIB_NAME} ${PDAL_UTIL_LIB_NAME}
+        ${PDAL_BASE_LIB_NAME} ${PDAL_ARBITER_LIB_NAME} ${PDAL_KAZHDAN_LIB_NAME} ${PDAL_UTIL_LIB_NAME}
     FILE
         "${PDAL_BINARY_DIR}/PDALTargets.cmake")
 
diff --git a/cmake/directories.cmake b/cmake/directories.cmake
index f079ec7..8e366e8 100644
--- a/cmake/directories.cmake
+++ b/cmake/directories.cmake
@@ -11,4 +11,7 @@ set(PDAL_KERNELS_DIR ${ROOT_DIR}/kernels)
 set(PDAL_FILTERS_DIR ${ROOT_DIR}/filters)
 set(PDAL_IO_DIR ${ROOT_DIR}/io)
 set(PDAL_VENDOR_DIR ${ROOT_DIR}/vendor)
-
+set(PDAL_VENDOR_ARBITER_DIR ${PDAL_VENDOR_DIR}/arbiter)
+set(PDAL_VENDOR_JSONCPP_DIR ${PDAL_VENDOR_DIR}/jsoncpp)
+set(PDAL_VENDOR_KAZHDAN_DIR ${PDAL_VENDOR_DIR}/kazhdan)
+set(PDAL_VENDOR_NANOFLANN_DIR ${PDAL_VENDOR_DIR}/nanoflann)

\ No newline at end of file
