diff --git a/pdal/PointViewIter.hpp b/pdal/PointViewIter.hpp
index 0c387be..ccb0721 100644
--- a/pdal/PointViewIter.hpp
+++ b/pdal/PointViewIter.hpp
@@ -139,7 +139,7 @@ public:
 
     bool operator==(const PointViewIter& i)
         { return m_id == i.m_id; }
-    bool operator!=(const PointViewIter& i)
+    bool operator!=(const PointViewIter& i) const
         { return m_id != i.m_id; }
     bool operator<=(const PointViewIter& i)
         { return m_id <= i.m_id; }
