diff --git a/CMakeLists.txt b/CMakeLists.txt
index 679b4e7..08965d9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -187,7 +187,7 @@ add_subdirectory(apps)
 # so that users only need link libpdalcpp.
 #
 if (APPLE)
-    set(PDAL_REEXPORT "-Wl,-reexport_library,$<TARGET_FILE:${PDAL_UTIL_LIB_NAME}>")
+    set(PDAL_REEXPORT "-Wl$<COMMA>-reexport_library$<COMMA>$<TARGET_FILE:${PDAL_UTIL_LIB_NAME}>")
     #
     # This allows the rpath reference for the reexported library (above) to
     # be found.
