diff --git a/CMakeLists.txt b/CMakeLists.txt
index 42cca1e..43b0ced 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -266,7 +265,7 @@ target_include_directories(${PDAL_BASE_LIB_NAME}
 target_link_libraries(${PDAL_BASE_LIB_NAME}
     PRIVATE
         ${CMAKE_THREAD_LIBS_INIT}
-        ${GDAL_LIBRARY}
+        ${GDAL_LIBRARIES}
         ${GEOTIFF_LIBRARY}
         ${LASZIP_LIBRARY}
         ${LIBXML2_LIBRARIES}
diff --git a/cmake/zstd.cmake b/cmake/zstd.cmake
index bfaa5b2..e73ec9e 100644
--- a/cmake/zstd.cmake
+++ b/cmake/zstd.cmake
@@ -4,7 +4,15 @@
 option(WITH_ZSTD
     "Build support for compression/decompression with Zstd." TRUE)
 if (WITH_ZSTD)
-    find_package(ZSTD QUIET)
+    find_package(zstd CONFIG REQUIRED)
+    set(ZSTD_FOUND TRUE)
+    set(ZSTD_INCLUDE_DIRS "")
+    if(TARGET zstd::libzstd_static)
+        set(ZSTD_LIBRARIES zstd::libzstd_static)
+        set(ZSTD_STATIC_LIB zstd::libzstd_static)
+    else()
+        set(ZSTD_LIBRARIES zstd::libzstd_shared)
+    endif()
     set_package_properties(ZSTD PROPERTIES TYPE
         PURPOSE "General compression support")
     if (ZSTD_FOUND)
