diff --git a/vendor/spz/CMakeLists.txt b/vendor/spz/CMakeLists.txt
index 7377269..b866bb7 100644
--- a/vendor/spz/CMakeLists.txt
+++ b/vendor/spz/CMakeLists.txt
@@ -10,5 +10,6 @@ file(GLOB SPZ_SRCS
 PDAL_ADD_FREE_LIBRARY(${PDAL_SPZ_LIB_NAME} STATIC ${SPZ_SRCS})
 
 set_target_properties(${PDAL_SPZ_LIB_NAME} PROPERTIES POSITION_INDEPENDENT_CODE ON)
+target_link_libraries(${PDAL_SPZ_LIB_NAME} PRIVATE ZLIB::ZLIB)
 
 set_property(GLOBAL PROPERTY _PDAL_SPZ_INCLUDED TRUE)
