diff --git a/src/kpackage/CMakeLists.txt b/src/kpackage/CMakeLists.txt
index c5aca02..5f87cc9 100644
--- a/src/kpackage/CMakeLists.txt
+++ b/src/kpackage/CMakeLists.txt
@@ -49,9 +49,13 @@ ecm_generate_export_header(KF5Package
     EXCLUDE_DEPRECATED_BEFORE_AND_AT ${EXCLUDE_DEPRECATED_BEFORE_AND_AT}
 )
 
+find_package(Iconv REQUIRED)
+include_directories(${Iconv_INCLUDE_DIRS})
+
 target_link_libraries(KF5Package
 PUBLIC
      KF5::CoreAddons
+     ${Iconv_LIBRARIES}
 PRIVATE
      Qt5::DBus
      KF5::Archive
