diff --git a/source/detail/number_format/number_formatter.cpp b/source/detail/number_format/number_formatter.cpp
index b116fd0..653da0e 100644
--- a/source/detail/number_format/number_formatter.cpp
+++ b/source/detail/number_format/number_formatter.cpp
@@ -24,6 +24,7 @@
 #include <algorithm>
 #include <cctype>
 #include <cmath>
+#include <limits>
 
 #include <xlnt/utils/exceptions.hpp>
 #include <detail/default_case.hpp>
