diff --git a/subprojects/gst-plugins-bad/ext/bz2/meson.build b/subprojects/gst-plugins-bad/ext/bz2/meson.build
index e7d8feb..9908a4c 100644
--- a/subprojects/gst-plugins-bad/ext/bz2/meson.build
+++ b/subprojects/gst-plugins-bad/ext/bz2/meson.build
@@ -18,7 +18,7 @@ plugin_sources += {
   'bz2': pathsep.join(doc_sources)
 }
 
-bz2_dep = cc.find_library('bz2', required : get_option('bz2'))
+bz2_dep = dependency('bzip2', required : get_option('bz2'))
 
 if bz2_dep.found() and cc.has_header_symbol('bzlib.h', 'BZ2_bzlibVersion')
   gstbz2 = library('gstbz2',
diff --git a/subprojects/gst-plugins-good/gst/matroska/meson.build b/subprojects/gst-plugins-good/gst/matroska/meson.build
index 3dc9292..747e608 100644
--- a/subprojects/gst-plugins-good/gst/matroska/meson.build
+++ b/subprojects/gst-plugins-good/gst/matroska/meson.build
@@ -39,7 +39,7 @@ if get_option('matroska').disabled()
   subdir_done()
 endif
 
-bz2_dep = cc.find_library('bz2', required : get_option('bz2'))
+bz2_dep = dependency('bzip2', required : get_option('bz2'))
 cdata.set('HAVE_BZ2', bz2_dep.found() and cc.has_header('bzlib.h'))
 
 gstmatroska = library('gstmatroska',
