diff --git a/meson.build b/meson.build
index 9de3b01..091a7f0 100644
--- a/meson.build
+++ b/meson.build
@@ -326,7 +326,7 @@ if have_gstgl
   endif
 endif
 
-zlib_dep = dependency('zlib', required : false)
+zlib_dep = dependency('zlib', required : true)
 if not zlib_dep.found()
   zlib_dep = cc.find_library('z', required : false)
   if not zlib_dep.found() or not cc.has_header('zlib.h')
