diff --git a/sdk/tables/azure-data-tables/vcpkg/Config.cmake.in b/sdk/tables/azure-data-tables/vcpkg/Config.cmake.in
index d67a773bc..1f448dfd8 100644
--- a/sdk/tables/azure-data-tables/vcpkg/Config.cmake.in
+++ b/sdk/tables/azure-data-tables/vcpkg/Config.cmake.in
@@ -8,7 +8,10 @@
 
 include(CMakeFindDependencyMacro)
 find_dependency(azure-core-cpp)
-find_dependency(LibXml2)
+if(NOT WIN32)
+  find_dependency(LibXml2)
+  find_dependency(OpenSSL)
+endif()
 
 include("${CMAKE_CURRENT_LIST_DIR}/azure-data-tables-cppTargets.cmake")
 
