diff --git a/buildtools/pkg-config/meson.build b/buildtools/pkg-config/meson.build
index b36add17e3..a0a265ad92 100644
--- a/buildtools/pkg-config/meson.build
+++ b/buildtools/pkg-config/meson.build
@@ -47,8 +47,7 @@ pkg.generate(name: 'DPDK', # main DPDK pkgconfig file
         description: '''The Data Plane Development Kit (DPDK).
 Note that CFLAGS might contain an -march flag higher than typical baseline.
 This is required for a number of static inline functions in the public headers.''',
-        requires: ['libdpdk-libs', libbsd], # may need libbsd for string funcs
-                      # if libbsd is not enabled, then this is blank
+        requires: ['libdpdk-libs'],
         libraries_private: ['-Wl,--whole-archive'] +
             dpdk_drivers + dpdk_static_libraries +
             ['-Wl,--no-whole-archive'] + platform_flags
diff --git a/config/meson.build b/config/meson.build
index bbdad71ffc..e0a3ecc5f9 100644
--- a/config/meson.build
+++ b/config/meson.build
@@ -238,12 +238,10 @@ if meson.is_cross_build() and not meson.get_external_property('numa', true)
     find_libnuma = false
 endif
 if find_libnuma
-    numa_dep = cc.find_library('numa', required: false)
-    if numa_dep.found() and cc.has_header('numaif.h') and cc.links(min_c_code, dependencies: numa_dep)
+    numa_dep = dependency('numa', method: 'pkg-config', required: false)
+    if numa_dep.found()
         dpdk_conf.set10('RTE_HAS_LIBNUMA', true)
         has_libnuma = true
-        add_project_link_arguments('-lnuma', language: 'c')
-        dpdk_extra_ldflags += '-lnuma'
     endif
 endif
 
@@ -269,28 +267,28 @@ if libarchive.found()
 endif
 
 # check for libbsd
-libbsd = dependency('libbsd', required: false, method: 'pkg-config')
+libbsd = disabler()
 if libbsd.found()
     dpdk_conf.set('RTE_USE_LIBBSD', 1)
 endif
 
-jansson_dep = dependency('jansson', required: false, method: 'pkg-config')
+jansson_dep = disabler()
 if jansson_dep.found()
     dpdk_conf.set('RTE_HAS_JANSSON', 1)
 endif
 
 # check for OpenSSL
-openssl_dep = dependency('openssl', required: false, method: 'pkg-config')
+openssl_dep = disabler()
 if openssl_dep.found()
     dpdk_conf.set('RTE_HAS_OPENSSL', 1)
 endif
 
 # check for pcap
-pcap_dep = dependency('libpcap', required: false, method: 'pkg-config')
+pcap_dep = disabler()
 pcap_lib = is_windows ? 'wpcap' : 'pcap'
 if not pcap_dep.found()
     # pcap got a pkg-config file only in 1.9.0
-    pcap_dep = cc.find_library(pcap_lib, required: false)
+    pcap_dep = disabler()
 endif
 if (pcap_dep.found() and cc.has_header('pcap.h', dependencies: pcap_dep)
         and cc.links(min_c_code, dependencies: pcap_dep))
diff --git a/drivers/common/mlx5/linux/meson.build b/drivers/common/mlx5/linux/meson.build
index 3767e7a69b..9127dad034 100644
--- a/drivers/common/mlx5/linux/meson.build
+++ b/drivers/common/mlx5/linux/meson.build
@@ -29,9 +29,7 @@ foreach libname:libnames
             ext_deps += lib
         endif
     else
-        build = false
-        reason = 'missing dependency, "' + libname + '"'
-        subdir_done()
+        error('missing dependency: "' + libname + '"')
     endif
 endforeach
 if static_ibverbs or dlopen_ibverbs
diff --git a/drivers/common/mlx5/windows/meson.build b/drivers/common/mlx5/windows/meson.build
index f454788a9f..78dfd12d5c 100644
--- a/drivers/common/mlx5/windows/meson.build
+++ b/drivers/common/mlx5/windows/meson.build
@@ -2,16 +2,12 @@
 # Copyright 2019 Mellanox Technologies, Ltd
 
 if not cc.has_header('mlx5devx.h')
-    build = false
-    reason = 'missing dependency, "mlx5devx.h"'
-    subdir_done()
+    error('missing dependency: "mlx5devx.h"')
 endif
 
 devxlib = cc.find_library('mlx5devx', required: false)
 if not devxlib.found() or not cc.links(min_c_code, dependencies: devxlib)
-    build = false
-    reason = 'missing dependency, "mlx5devx"'
-    subdir_done()
+    error('missing dependency: "mlx5devx"')
 endif
 ext_deps += devxlib
 
diff --git a/drivers/net/intel/ipn3ke/meson.build b/drivers/net/intel/ipn3ke/meson.build
index 23c4d36b4f..a29abb1f77 100644
--- a/drivers/net/intel/ipn3ke/meson.build
+++ b/drivers/net/intel/ipn3ke/meson.build
@@ -16,9 +16,7 @@ endif
 #
 
 if not has_libfdt
-    build = false
-    reason = 'missing dependency, "libfdt"'
-    subdir_done()
+    error('missing dependency: "libfdt"')
 endif
 
 includes += include_directories('../../../raw/ifpga')
diff --git a/drivers/net/mana/meson.build b/drivers/net/mana/meson.build
index 19d4b3695e..56323879d7 100644
--- a/drivers/net/mana/meson.build
+++ b/drivers/net/mana/meson.build
@@ -31,9 +31,7 @@ foreach libname:libnames
         libs += lib
         ext_deps += lib
     else
-        build = false
-        reason = 'missing dependency, "' + libname + '"'
-        subdir_done()
+        error('missing dependency: "' + libname + '"')
     endif
 endforeach
 
diff --git a/drivers/net/mlx4/meson.build b/drivers/net/mlx4/meson.build
index 1eb67f3c47..427e6e9f89 100644
--- a/drivers/net/mlx4/meson.build
+++ b/drivers/net/mlx4/meson.build
@@ -37,9 +37,7 @@ foreach libname:libnames
             ext_deps += lib
         endif
     else
-        build = false
-        reason = 'missing dependency, "' + libname + '"'
-        subdir_done()
+        error('missing dependency: "' + libname + '"')
     endif
 endforeach
 if static_ibverbs or dlopen_ibverbs
diff --git a/drivers/raw/ifpga/meson.build b/drivers/raw/ifpga/meson.build
index 8d5f41fd5d..84c8c875b3 100644
--- a/drivers/raw/ifpga/meson.build
+++ b/drivers/raw/ifpga/meson.build
@@ -2,9 +2,7 @@
 # Copyright(c) 2018 Intel Corporation
 
 if not has_libfdt
-    build = false
-    reason = 'missing dependency, "libfdt"'
-    subdir_done()
+    error('missing dependency: "libfdt"')
 endif
 
 rtdep = dependency('librt', required: false)
@@ -12,9 +10,7 @@ if not rtdep.found()
     rtdep = cc.find_library('rt', required: false)
 endif
 if not rtdep.found() or not cc.links(min_c_code, dependencies: rtdep)
-    build = false
-    reason = 'missing dependency, "librt"'
-    subdir_done()
+    error('missing dependency: "librt"')
 endif
 
 ext_deps += rtdep
diff --git a/lib/eal/linux/meson.build b/lib/eal/linux/meson.build
index 29ba313218..5a173c43d6 100644
--- a/lib/eal/linux/meson.build
+++ b/lib/eal/linux/meson.build
@@ -25,5 +25,6 @@ endif
 
 deps += ['kvargs', 'telemetry']
 if has_libnuma
+    ext_deps += numa_dep
     dpdk_conf.set10('RTE_EAL_NUMA_AWARE_HUGEPAGES', true)
 endif
diff --git a/lib/vhost/meson.build b/lib/vhost/meson.build
index 6a24981d10..5283e7904f 100644
--- a/lib/vhost/meson.build
+++ b/lib/vhost/meson.build
@@ -6,6 +6,7 @@ if not is_linux
     reason = 'only supported on Linux'
 endif
 if has_libnuma
+    ext_deps += numa_dep
     dpdk_conf.set10('RTE_LIBRTE_VHOST_NUMA', true)
 endif
 dpdk_conf.set('RTE_LIBRTE_VHOST_POSTCOPY', cc.has_header('linux/userfaultfd.h'))
