diff --git a/meson.build b/meson.build
index 8b248d4..c546c62 100644
--- a/meson.build
+++ b/meson.build
@@ -90,10 +90,6 @@ subdir('doc')

 # build any examples explicitly requested - useful for developers - and
 # install any example code into the appropriate install path
-subdir('examples')
-install_subdir('examples',
-        install_dir: get_option('datadir') + '/dpdk',
-        exclude_files: ex_file_excludes)

 # build kernel modules
 subdir('kernel')
