diff --git a/build/config/ios/BUILD.gn b/build/config/ios/BUILD.gn
index 47d36fc2fb..11ac598170 100644
--- a/build/config/ios/BUILD.gn
+++ b/build/config/ios/BUILD.gn
@@ -21,7 +21,7 @@ declare_args() {
   # Intel assembly syntax; Yasm / Nasm do not support emitting bitcode parts.
   # That is not a limitation for now as Xcode mandates the presence of bitcode
   # only when building bitcode-enabled projects for real devices (ARM CPUs).
-  enable_ios_bitcode = false
+  enable_ios_bitcode = true
 }
 
 # This is included by reference in the //build/config/compiler config that
