diff --git a/libcpuid/cpuid_main.c b/libcpuid/cpuid_main.c
index ed3db36..04e671b 100644
--- a/libcpuid/cpuid_main.c
+++ b/libcpuid/cpuid_main.c
@@ -1452,8 +1452,12 @@ int cpuid_get_raw_data_core(struct cpu_raw_data_t* data, logical_cpu_t logical_c
 	}
 # endif /* PLATFORM_AARCH64 */
 #else
-# warning This CPU architecture is not supported by libcpuid
-	UNUSED(data);
+    #if defined(_MSC_VER)
+        #pragma message("Warning: This CPU architecture is not supported by libcpuid")
+    #else
+        #warning This CPU architecture is not supported by libcpuid
+    #endif
+    UNUSED(data);
 #endif
 
 	if (affinity_saved)
