diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1fbc4a4..075c86c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,8 +13,6 @@ set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
-# Install to the build directory
-set(CMAKE_INSTALL_PREFIX "${CMAKE_BINARY_DIR}" CACHE STRING "CMake install prefix" FORCE)
 
 # Use "d" suffix for debug builds
 # Do not use a suffix for RelWithDebInfo
