diff --git a/CMakeLists.txt b/CMakeLists.txt
index 658f994..a04cf27 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,7 +14,7 @@ if(MSVC)
 endif()
 
 #---- Include boost to add coroutines ----
-find_package(Boost COMPONENTS coroutine QUIET)
+find_package(Boost REQUIRED) # coroutine2 is header-only
 
 if(Boost_FOUND)
     string(REPLACE "." "0" Boost_VERSION_NODOT ${Boost_VERSION})
@@ -46,17 +46,18 @@ option(BUILD_SHARED_LIBS "Build shared libraries" ON)
 
 #---- Find other packages ----
 find_package(Threads)
-find_package(ZMQ)
+find_package(cppzmq)
 
 list(APPEND BEHAVIOR_TREE_PUBLIC_LIBRARIES
     ${CMAKE_THREAD_LIBS_INIT}
     ${CMAKE_DL_LIBS}
 )
 
-if( ZMQ_FOUND )
+if( ZMQ_FOUND OR ON )
     message(STATUS "ZeroMQ found.")
     add_definitions( -DZMQ_FOUND )
     list(APPEND BT_SOURCE src/loggers/bt_zmq_publisher.cpp)
+    list(APPEND BEHAVIOR_TREE_EXTERNAL_LIBRARIES cppzmq)
 else()
     message(WARNING "ZeroMQ NOT found. Skipping the build of [PublisherZMQ] and [bt_recorder].")
 endif()
