diff --git a/external/CMakeLists.txt b/external/CMakeLists.txt
index a8b2638e3..c146612d5 100644
--- a/external/CMakeLists.txt
+++ b/external/CMakeLists.txt
@@ -21,10 +21,14 @@ endif(NOT DIRECTX_HEADER_INCLUDE_DIR)
 # Enabling SPIR-V codegen requires SPIRV-Headers for spirv.hpp and
 # SPIRV-Tools for SPIR-V disassembling functionality.
 if (${ENABLE_SPIRV_CODEGEN})
-  set(DXC_SPIRV_HEADERS_DIR "${DXC_EXTERNAL_ROOT_DIR}/SPIRV-Headers"
-      CACHE STRING "Location of SPIRV-Headers source")
-  set(DXC_SPIRV_TOOLS_DIR "${DXC_EXTERNAL_ROOT_DIR}/SPIRV-Tools"
-      CACHE STRING "Location of SPIRV-Tools source")
+  if (NOT DEFINED DXC_SPIRV_HEADERS_DIR)
+    set(DXC_SPIRV_HEADERS_DIR "${DXC_EXTERNAL_ROOT_DIR}/SPIRV-Headers"
+        CACHE STRING "Location of SPIRV-Headers source")
+  endif ()
+  if (NOT DEFINED DXC_SPIRV_TOOLS_DIR)
+    set(DXC_SPIRV_TOOLS_DIR "${DXC_EXTERNAL_ROOT_DIR}/SPIRV-Tools"
+        CACHE STRING "Location of SPIRV-Tools source")
+  endif ()
 
   if (NOT DEFINED SPIRV-Headers_SOURCE_DIR)
     if (IS_DIRECTORY ${DXC_SPIRV_HEADERS_DIR})
