diff --git a/KF5GuiAddonsConfig.cmake.in b/KF5GuiAddonsConfig.cmake.in
index a3c3e58..66c2825 100644
--- a/KF5GuiAddonsConfig.cmake.in
+++ b/KF5GuiAddonsConfig.cmake.in
@@ -7,6 +7,7 @@ find_dependency(Qt5Gui @REQUIRED_QT_VERSION@)
 
 if (NOT @BUILD_SHARED_LIBS@)
     if (@WITH_WAYLAND@)
+        set(QtWaylandScanner_EXECUTABLE "${CURRENT_INSTALLED_DIR}/tools/qt5-wayland/bin/qtwaylandscanner")
         find_dependency(Wayland REQUIRED Client)
         find_dependency(Qt5WaylandClient)
         find_dependency(QtWaylandScanner)
