diff --git a/Makefile.am b/Makefile.am
index 9608be6..bdd83a2 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,5 +1,5 @@
 AM_CPPFLAGS = -I$(srcdir)/include
-AM_CFLAGS = -Wall -Wno-parentheses
+AM_CFLAGS = -Wall -Wno-parentheses -Wno-error=unused-but-set-variable -Wno-error=unused-variable
 AM_CXXFLAGS = $(AM_CFLAGS)
 ACLOCAL_AMFLAGS = -I config
 bin_PROGRAMS =
diff --git a/Makefile.in b/Makefile.in
index 5e7ad77..097ae73 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -657,7 +657,7 @@ top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AM_CPPFLAGS = -I$(srcdir)/include
-AM_CFLAGS = -Wall -Wno-parentheses
+AM_CFLAGS = -Wall -Wno-parentheses -Wno-error=unused-but-set-variable -Wno-error=unused-variable
 AM_CXXFLAGS = $(AM_CFLAGS)
 ACLOCAL_AMFLAGS = -I config
 lib_LTLIBRARIES = zbar/libzbar.la $(am__append_40) $(am__append_49) \
diff --git a/configure.ac b/configure.ac
index 56d3dd0..530632a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3,7 +3,7 @@ AC_PREREQ([2.61])
 AC_INIT([zbar], [0.10], [spadix@users.sourceforge.net])
 AC_CONFIG_AUX_DIR(config)
 AC_CONFIG_MACRO_DIR(config)
-AM_INIT_AUTOMAKE([1.10 -Wall -Werror foreign subdir-objects std-options dist-bzip2])
+AM_INIT_AUTOMAKE([1.10 foreign subdir-objects std-options dist-bzip2])
 AC_CONFIG_HEADERS([include/config.h])
 AC_CONFIG_SRCDIR(zbar/scanner.c)
 LT_PREREQ([2.2])
