diff --git a/cmake/compiler_settings.cmake b/cmake/compiler_settings.cmake
index 36795a2..77334ee 100644
--- a/cmake/compiler_settings.cmake
+++ b/cmake/compiler_settings.cmake
@@ -57,7 +57,7 @@ endif()
 if(MSVC)
     # warnings as errors, max warning level (4)
     if(NOT CMAKE_CXX_FLAGS MATCHES "/WX")
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /WX")
+        #set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /WX")
     endif()
 
     # taken from http://stackoverflow.com/questions/2368811/how-to-set-warning-level-in-cmake
