diff --git a/CMakeLists.txt b/CMakeLists.txt
index de28c31..599ddc5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -697,19 +697,10 @@ set(namespace "${PACKAGE_NAME}::")
 
 # Set up install directories. INCLUDE_INSTALL_DIR and
 # CMAKECONFIG_INSTALL_DIR must not be absolute paths.
-if(WIN32)
-    set(include_install_dir Include)
-    set(include_install_dir_full Include)
-    set(config_install_dir CMake)
-elseif(UNIX)
-    set(include_install_dir include)
-    set(include_install_dir_postfix "")
-    set(include_install_dir_full    "${include_install_dir}")
-
-    set(config_install_dir share/${PACKAGE_NAME})
-else()
-    message(FATAL_ERROR "Not supported system type. Options: UNIX or WIN32.")
-endif()
+set(include_install_dir include)
+set(include_install_dir_full "${include_install_dir}")
+
+set(config_install_dir CMake)
 
 #####  Gather git repo related information
 # Get the current working branch
