diff --git a/CMakeLists.txt b/CMakeLists.txt
index a379f549..f95a872b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,7 +16,7 @@ enable_language(C)
 include(GNUInstallDirs)
 
 if(MSVC)
-set(OpenBLAS_LIBNAME libopenblas)
+set(OpenBLAS_LIBNAME openblas)
 else()
 set(OpenBLAS_LIBNAME openblas)
 endif()
@@ -197,6 +197,10 @@ set_target_properties(${OpenBLAS_LIBNAME} PROPERTIES
   SOVERSION ${OpenBLAS_MAJOR_VERSION}
 )
 
+install(TARGETS ${OpenBLAS_LIBNAME}
+        RUNTIME DESTINATION bin
+        LIBRARY DESTINATION lib
+        ARCHIVE DESTINATION lib)
 
 # TODO: Why is the config saved here? Is this necessary with CMake?
 #Save the config files for installation
