diff --git a/KF5WindowSystemConfig.cmake.in b/KF5WindowSystemConfig.cmake.in
index b79eb0f4aabe0545493a5ab30be3d1f6299c88d3..5115daced38bdeaf62e70458c7fc58652f41327b 100644
--- a/KF5WindowSystemConfig.cmake.in
+++ b/KF5WindowSystemConfig.cmake.in
@@ -18,6 +18,7 @@ if(NOT @BUILD_SHARED_LIBS@)
 
     if(@X11_FOUND@)
         find_dependency(Qt5X11Extras @REQUIRED_QT_VERSION@)
+        find_dependency(XCB)
     endif()
 endif()
 
