diff --git a/Libraries/MiKTeX/Util/StringUtil.cpp b/Libraries/MiKTeX/Util/StringUtil.cpp
index 039a7e2..b0415da 100644
--- a/Libraries/MiKTeX/Util/StringUtil.cpp
+++ b/Libraries/MiKTeX/Util/StringUtil.cpp
@@ -230,7 +230,7 @@ u16string StringUtil::UTF8ToUTF16(const char* utf8Chars)
 {
   try
   {
-#if _MSC_VER == 1900 || _MSC_VER == 1910 || _MSC_VER == 1911
+#if _MSC_VER >= 1900
     wstring_convert<codecvt_utf8_utf16<int16_t>, int16_t> conv;
     u16string result;
     for (auto& ch : conv.from_bytes(utf8Chars))
@@ -253,7 +253,7 @@ string StringUtil::UTF16ToUTF8(const char16_t* utf16Chars)
 {
   try
   {
-#if _MSC_VER == 1900 || _MSC_VER == 1910 || _MSC_VER == 1911
+#if _MSC_VER >= 1900
     // workround for VS2015 bug: 
     // http://stackoverflow.com/questions/32055357/visual-studio-c-2015-stdcodecvt-with-char16-t-or-char32-t
     wstring_convert<codecvt_utf8_utf16<int16_t>, int16_t> conv;
@@ -274,7 +274,7 @@ u32string StringUtil::UTF8ToUTF32(const char* utf8Chars)
 {
   try
   {
-#if _MSC_VER == 1900 || _MSC_VER == 1910 || _MSC_VER == 1911
+#if _MSC_VER >= 1900
     wstring_convert<codecvt_utf8<int32_t>, int32_t> conv;
     u32string result;
     for (auto& ch : conv.from_bytes(utf8Chars))
@@ -297,7 +297,7 @@ string StringUtil::UTF32ToUTF8(const char32_t* utf32Chars)
 {
   try
   {
-#if _MSC_VER == 1900 || _MSC_VER == 1910 || _MSC_VER == 1911
+#if _MSC_VER >= 1900
     wstring_convert<codecvt_utf8<int32_t>, int32_t> conv;
     const int32_t* p = (const int32_t*)utf32Chars;
     return conv.to_bytes(p, p + StrLen(utf32Chars));
