From fd83f64efb35eb9dac6be67d1dcabfbae67d441a Mon Sep 17 00:00:00 2001
From: Matthias Kuhn <matthias@opengis.ch>
Date: Sun, 28 Feb 2021 09:25:08 +0100
Subject: [PATCH] Fix staticlib

---
 src/qscintilla.pro | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/qscintilla.pro b/src/qscintilla.pro
index 8d0acd2..2246442 100644
--- a/src/qscintilla.pro
+++ b/src/qscintilla.pro
@@ -37,13 +37,13 @@ CONFIG(debug, debug|release) {
     TARGET = qscintilla2_qt$${QT_MAJOR_VERSION}
 }
 
-macx:!CONFIG(staticlib) {
+macx:!CONFIG(static) {
     QMAKE_POST_LINK += install_name_tool -id @rpath/$(TARGET1) $(TARGET)
 }
 
 INCLUDEPATH += . ../scintilla/include ../scintilla/lexlib ../scintilla/src
 
-!CONFIG(staticlib) {
+!CONFIG(static) {
     DEFINES += QSCINTILLA_MAKE_DLL
 }
 DEFINES += SCINTILLA_QT SCI_LEXER INCLUDE_DEPRECATED_FEATURES
@@ -82,7 +82,7 @@ qsci.files = ../qsci
 INSTALLS += qsci
 
 features.path = $$[QT_HOST_DATA]/mkspecs/features
-CONFIG(staticlib) {
+CONFIG(static) {
     features.files = $$PWD/features_staticlib/qscintilla2.prf
 } else {
     features.files = $$PWD/features/qscintilla2.prf
-- 
2.30.1.windows.1

