diff --git a/CMakeLists.txt b/CMakeLists.txt
index e58d072..e582a25 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -54,12 +54,7 @@ option(WITH_THREADING "Include client library threading support?" ON)
 if (WITH_THREADING)
 	add_definitions("-DWITH_THREADING")
 	if (WIN32)
-		if (CMAKE_CL_64)
-			set (PTHREAD_LIBRARIES C:\\pthreads\\Pre-built.2\\lib\\x64\\pthreadVC2.lib)
-		else (CMAKE_CL_64)
-			set (PTHREAD_LIBRARIES C:\\pthreads\\Pre-built.2\\lib\\x86\\pthreadVC2.lib)
-		endif (CMAKE_CL_64)
-		set (PTHREAD_INCLUDE_DIR C:\\pthreads\\Pre-built.2\\include)
+		find_package(pthread REQUIRED)
 	else (WIN32)
 		find_library(LIBPTHREAD pthread)
 		if (LIBPTHREAD)
diff --git a/config.h b/config.h
index b7a7616..bceb92a 100644
--- a/config.h
+++ b/config.h
@@ -27,7 +27,7 @@
 #  define EPROTO ECONNABORTED
 #endif
 
-#ifdef WIN32
+#if defined(WIN32) || defined(WIN64)
 #  ifndef strcasecmp
 #    define strcasecmp strcmpi
 #  endif
diff --git a/lib/net_mosq.h b/lib/net_mosq.h
index b843ebe..8c077eb 100644
--- a/lib/net_mosq.h
+++ b/lib/net_mosq.h
@@ -20,6 +20,9 @@ Contributors:
 #include <unistd.h>
 #else
 #include <winsock2.h>
+#if defined(_MSC_VER) && _MSC_VER > 1900
+#undef ssize_t
+#endif
 typedef SSIZE_T ssize_t;
 #endif
 
