diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 94f2dc8..9ea9d28 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -286,7 +286,7 @@ set(REPORTER_SOURCES
 )
 set(REPORTER_FILES ${REPORTER_HEADERS} ${REPORTER_SOURCES})
 
-add_library(Catch2
+add_library(Catch2 STATIC
   ${REPORTER_FILES}
   ${INTERNAL_FILES}
   ${BENCHMARK_HEADERS}
@@ -339,7 +339,7 @@ target_include_directories(Catch2
 )
 
 
-add_library(Catch2WithMain
+add_library(Catch2WithMain STATIC
     ${SOURCES_DIR}/internal/catch_main.cpp
 )
 add_build_reproducibility_settings(Catch2WithMain)
