diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1676ee7..5231934 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -167,13 +167,13 @@ if(NOT_SUBPROJECT)
         "extras/gdbinit"
         "extras/lldbinit"
       DESTINATION
-        ${CMAKE_INSTALL_DATAROOTDIR}/Catch2
+        ${CMAKE_INSTALL_DATAROOTDIR}/catch2
     )
   endif()
 
   ## Provide some pkg-config integration
   set(PKGCONFIG_INSTALL_DIR
-    "${CMAKE_INSTALL_DATAROOTDIR}/pkgconfig"
+    "${CMAKE_INSTALL_LIBDIR}/pkgconfig"
     CACHE PATH "Path where catch2.pc is installed"
   )
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 1e3af14..265626c 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -402,7 +402,6 @@ if(NOT_SUBPROJECT)
   install(
     TARGETS
       Catch2
-      Catch2WithMain
     EXPORT
       Catch2Targets
     LIBRARY DESTINATION
@@ -413,6 +412,19 @@ if(NOT_SUBPROJECT)
       ${CMAKE_INSTALL_BINDIR}
   )
 
+  install(
+    TARGETS
+      Catch2WithMain
+    EXPORT
+      Catch2Targets
+     LIBRARY DESTINATION
+      ${CMAKE_INSTALL_LIBDIR}/manual-link
+    ARCHIVE DESTINATION
+      ${CMAKE_INSTALL_LIBDIR}/manual-link
+    RUNTIME DESTINATION
+      ${CMAKE_INSTALL_BINDIR}
+  )
+
   install(
     EXPORT
       Catch2Targets
