diff --git a/onnxoptimizer/optimize.h b/onnxoptimizer/optimize.h
index f25e6c3f..506e2554 100644
--- a/onnxoptimizer/optimize.h
+++ b/onnxoptimizer/optimize.h
@@ -9,7 +9,6 @@
 
 #include "onnx/common/ir.h"
 #include "onnx/common/ir_pb_converter.h"
-#include "onnx/common/stl_backports.h"
 #include "onnx/proto_utils.h"
 
 #include "onnxoptimizer/pass_manager.h"
diff --git a/onnxoptimizer/pass_registry.h b/onnxoptimizer/pass_registry.h
index 51a2c161..b8dc26f9 100644
--- a/onnxoptimizer/pass_registry.h
+++ b/onnxoptimizer/pass_registry.h
@@ -12,7 +12,6 @@
 
 #include "onnx/common/ir.h"
 #include "onnx/common/ir_pb_converter.h"
-#include "onnx/common/stl_backports.h"
 #include "onnx/proto_utils.h"
 #include "onnxoptimizer/passes/adjust_add.h"
 #include "onnxoptimizer/passes/adjust_slice_and_matmul.h"
