diff --git a/CMakeLists.txt b/CMakeLists.txt
index b898742..f6274c5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,7 +20,7 @@ option(BUILD_APP "Build executables" OFF)
 message(STATUS "Configuring with CMAKE_SYSTEM_PROCESSOR ${CMAKE_SYSTEM_PROCESSOR}")
 if(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "x86_64")
     set(ARCH_AMD64 ON)
-elseif(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "aarch64")
+elseif(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "aarch64" OR ${CMAKE_SYSTEM_PROCESSOR} STREQUAL "arm64")
     set(ARCH_AARCH64 ON)
 elseif(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "ppc64le")
     set(ARCH_PPC64LE ON)
