diff --git a/CMakeLists.txt b/CMakeLists.txt
index eebda35e9..c00e6b702 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -888,7 +888,7 @@ option(WITH_TESTS "build with tests" ON)
 # For test libraries, utilities, and exes that are build iff WITH_TESTS=ON and
 # in Debug mode. Add test only code that is not #ifdefed for Release here.
 if(WITH_TESTS AND CMAKE_BUILD_TYPE STREQUAL "Debug")
-  add_subdirectory(third-party/gtest-1.8.1/fused-src/gtest)
+  #add_subdirectory(third-party/gtest-1.8.1/fused-src/gtest)
   set(TESTS
         cache/cache_test.cc
         cache/lru_cache_test.cc
