diff --git a/cmake/RocksDBConfig.cmake.in b/cmake/RocksDBConfig.cmake.in
index 0bd14be11..ad07f018a 100644
--- a/cmake/RocksDBConfig.cmake.in
+++ b/cmake/RocksDBConfig.cmake.in
@@ -33,11 +33,11 @@ if(@WITH_BZ2@)
 endif()
 
 if(@WITH_LZ4@)
-  find_dependency(lz4)
+  find_dependency(lz4 CONFIG)
 endif()
 
 if(@WITH_ZSTD@)
-  find_dependency(zstd)
+  find_dependency(zstd CONFIG)
 endif()
 
 if(@WITH_NUMA@)
@@ -45,7 +45,7 @@ if(@WITH_NUMA@)
 endif()
 
 if(@WITH_TBB@)
-  find_dependency(TBB)
+  find_dependency(TBB CONFIG)
 endif()
 
 find_dependency(Threads)
