diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3d90152..b0c5d31 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1206,7 +1206,7 @@ if(NOT WIN32 OR ROCKSDB_INSTALL_ON_WINDOWS)
     endforeach()
   endforeach()
 
-  install(DIRECTORY "${PROJECT_SOURCE_DIR}/cmake/modules" COMPONENT devel DESTINATION ${package_config_destination})
+  #install(DIRECTORY "${PROJECT_SOURCE_DIR}/cmake/modules" COMPONENT devel DESTINATION ${package_config_destination})
 
   if(NOT ROCKSDB_BUILD_SHARED)
     install(
