diff --git a/CMakeLists.txt b/CMakeLists.txt
index d6bfbf2..ea6a884 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -132,6 +132,7 @@ endif()
 
 string(REGEX REPLACE "[^0-9a-f]+" "" GIT_SHA "${GIT_SHA}")
 
+if(NOT DEFINED ROCKSDB_VERSION OR NOT DEFINED ROCKSDB_VERSION_MAJOR)
 set(SH_CMD "sh")
 execute_process(COMMAND
   ${SH_CMD} -c "build_tools/version.sh full"
@@ -145,6 +146,7 @@ execute_process(COMMAND
   OUTPUT_VARIABLE ROCKSDB_VERSION_MAJOR
 )
 string(STRIP "${ROCKSDB_VERSION_MAJOR}" ROCKSDB_VERSION_MAJOR)
+endif()
 
 option(WITH_MD_LIBRARY "build with MD" ON)
 if(WIN32 AND MSVC)
