diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index 0155a7b..7850b2d 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -66,7 +66,7 @@ endif()
 enable_testing()
 
 add_library(port INTERFACE)
-add_library(descriptor_upbproto INTERFACE)
+add_library(descriptor_upb_proto INTERFACE)
 add_library(upb
   ../upb/decode.c
   ../upb/decode.int.h
@@ -167,6 +167,25 @@ target_link_libraries(upb_json
   upb_pb)
 add_library(wyhash INTERFACE)
 
+add_library(all_libs INTERFACE)
+target_link_libraries(all_libs
+  INTERFACE
+  upb
+  fastdecode
+  upb_json
+  upb_pb
+  port
+  table
+  descriptor_upb_proto
+  handlers
+  reflection
+  textformat
+)
+set_target_properties(reflection PROPERTIES OUTPUT_NAME upb_reflection)
+set_target_properties(handlers PROPERTIES OUTPUT_NAME upb_handlers)
+set_target_properties(fastdecode PROPERTIES OUTPUT_NAME upb_fastdecode)
+set_target_properties(textformat PROPERTIES OUTPUT_NAME upb_textformat)
+
 install(
   DIRECTORY ../upb
   DESTINATION include
@@ -184,9 +203,11 @@ install(TARGETS
   upb_pb
   port
   table
-  descriptor_upbproto
+  descriptor_upb_proto
   handlers
   reflection
+  textformat
+  all_libs
   EXPORT upb-config
 )
 install(EXPORT upb-config NAMESPACE upb:: DESTINATION share/upb)
