diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index b04a937..dcf54e1 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -66,6 +66,7 @@ endif()
 enable_testing()
 
 add_library(port INTERFACE)
+add_library(descriptor_upbproto INTERFACE)
 add_library(upb
   ../upb/decode.c
   ../upb/decode_internal.h
@@ -133,6 +134,21 @@ add_library(table INTERFACE)
 target_link_libraries(table INTERFACE
   port)
 
+add_library(all_libs INTERFACE)
+target_link_libraries(all_libs
+  INTERFACE
+  upb
+  fastdecode
+  json
+  port
+  table
+  descriptor_upbproto
+  reflection
+  textformat
+)
+set_target_properties(reflection PROPERTIES OUTPUT_NAME upb_reflection)
+set_target_properties(fastdecode PROPERTIES OUTPUT_NAME upb_fastdecode)
+set_target_properties(textformat PROPERTIES OUTPUT_NAME upb_textformat)
 
 install(
   DIRECTORY ../upb
