diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index 6b4c50d..b4f43f9 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -66,6 +66,10 @@ endif()
 enable_testing()
 
 add_library(port INTERFACE)
+add_library(descriptor_upb_proto INTERFACE)
+add_library(utf8_range
+  ../third_party/utf8_range/utf8_range.c
+  ../third_party/utf8_range/utf8_range.h)
 add_library(upb
   ../upb/decode.c
   ../upb/decode_internal.h
@@ -84,7 +88,7 @@ add_library(upb
 target_link_libraries(upb
   fastdecode
   port
-  /third_party/utf8_range)
+  utf8_range)
 add_library(fastdecode
   ../upb/decode.h
   ../upb/decode_internal.h
@@ -96,7 +100,7 @@ add_library(fastdecode
 target_link_libraries(fastdecode
   port
   table
-  /third_party/utf8_range)
+  utf8_range)
 add_library(generated_code_support__only_for_generated_code_do_not_use__i_give_permission_to_break_me INTERFACE)
 target_link_libraries(generated_code_support__only_for_generated_code_do_not_use__i_give_permission_to_break_me INTERFACE
   table
