diff --git a/CMakeLists.txt b/CMakeLists.txt
index cd716e3..a1f2ada 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -98,7 +98,7 @@ IF(MSVC)
   # Speedup system tests
   INCLUDE(${CC_SOURCE_DIR}/cmake/WindowsCache.cmake)
   ADD_DEFINITIONS(-DWIN32_LEAN_AND_MEAN)
-  IF (MSVC)
+  IF (MSVC AND 0)
     SET(CONFIG_TYPES "DEBUG" "RELEASE" "RELWITHDEBINFO")
     FOREACH(BUILD_TYPE ${CONFIG_TYPES})
       FOREACH(COMPILER CXX C)
