diff --git a/msh3.h b/msh3.h
index ebbdc20..c31b1b9 100644
--- a/msh3.h
+++ b/msh3.h
@@ -374,7 +374,11 @@ typedef struct MSH3_REQUEST_EVENT {
             bool AppCloseInProgress       : 1;
             bool ConnectionShutdownByApp  : 1;
             bool ConnectionClosedRemotely : 1;
-            bool RESERVED                 : 5;
+            bool RESERVED                 : 1;
+            bool RESERVED_2               : 1;
+            bool RESERVED_3               : 1;
+            bool RESERVED_4               : 1;
+            bool RESERVED_5               : 1;
             uint64_t ConnectionErrorCode;
             MSH3_STATUS ConnectionCloseStatus;
         } SHUTDOWN_COMPLETE;
@@ -487,7 +491,13 @@ typedef struct MSH3_LISTENER_EVENT {
     union {
         struct {
             bool AppCloseInProgress  : 1;
-            bool RESERVED            : 7;
+            bool RESERVED            : 1;
+            bool RESERVED_2          : 1;
+            bool RESERVED_3          : 1;
+            bool RESERVED_4          : 1;
+            bool RESERVED_5          : 1;
+            bool RESERVED_6          : 1;
+            bool RESERVED_7          : 1;
         } SHUTDOWN_COMPLETE;
         struct {
             MSH3_CONNECTION* Connection;
