diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8d959fbf7..38754c8ac 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,6 +32,7 @@ project(flint
   HOMEPAGE_URL https://flintlib.org/
   LANGUAGES C CXX)
 
+set(CMAKE_CURRENT_SOURCE_DIR ${PROJECT_SOURCE_DIR})
 file(READ "${CMAKE_CURRENT_SOURCE_DIR}/configure.ac" CONFIGURE_CONTENTS)
 foreach(version in MAJOR MINOR PATCH)
   # Set soname version for the library
@@ -115,10 +116,11 @@ find_package(mpfr ${MPFR_MIN_VERSION} REQUIRED)
 if (WITH_NTL)
     find_package(NTL REQUIRED)
 endif()
-find_package(Python REQUIRED)
 
+if(WITH_CBLAS)
 find_package(CBLAS)
 set(FLINT_USES_BLAS ${CBLAS_FOUND})
+endif()
 
 if(CMAKE_BUILD_TYPE STREQUAL Debug)
   set(FLINT_WANT_ASSERT ON)
