diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index eb692cf..11158db 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -524,7 +524,7 @@ INSTALL(
 endif()
 
   
-endif()
+else()
 
 add_library (${LIBSBML_LIBRARY}-static STATIC ${LIBSBML_SOURCES} )
 
@@ -560,7 +560,7 @@ INSTALL(FILES
   DESTINATION ${PACKAGE_CONFIG_DIR}
 )
 endif()
-
+endif()
 ##############################################################################
 #
 # add native tests
