diff --git a/CMakeLists.txt b/CMakeLists.txt
index fb10f47..114a172 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -56,12 +56,13 @@ source_group(${platformDir_} FILES ${platformHeaders_} ${platformSources_})
 
 # create the lib
 add_library(UnitTest++ STATIC ${headers_} ${sources_} ${platformHeaders_} ${platformSources_})
+target_include_directories(UnitTest++ PUBLIC $<INSTALL_INTERFACE:include>)
 
 if(${UTPP_USE_PLUS_SIGN})
 	set_target_properties(UnitTest++ PROPERTIES OUTPUT_NAME UnitTest++)
 endif()
 
-
+if(0)
 # build the test runner
 file(GLOB TEST_SRCS RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} tests/*.cpp tests/*.h)
 source_group( "" FILES ${TEST_SRCS})
@@ -82,6 +83,7 @@ add_custom_command(TARGET TestUnitTest++
 if(NOT ${UTPP_INCLUDE_TESTS_IN_BUILD})
     set_target_properties(TestUnitTest++ PROPERTIES EXCLUDE_FROM_ALL 1)
 endif()
+endif()
 
 # add install targets
 # need a custom install path?
