diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index c790fa4a8..d5f6c9b3f 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -120,9 +120,5 @@ foreach (PROJECT_VARIANT_NAME ${PROJECT_NAME} ${PROJECT_LEGACY_NAME})
     NAMESPACE ${PROJECT_NAME}::
     FILE ${PROJECT_NAME_LOWER}-targets.cmake
     DESTINATION "${CMAKECONFIGSUBDIR}")
-  install(EXPORT targets
-    NAMESPACE ${PROJECT_LEGACY_NAME}::
-    FILE ${PROJECT_LEGACY_LOWER}-targets.cmake
-    DESTINATION "${CMAKECONFIGSUBDIR}")
 endforeach ()
 

diff --git a/cmake/project-config.cmake.in b/cmake/project-config.cmake.in
index 23f997abd..c3a40116d 100644
--- a/cmake/project-config.cmake.in
+++ b/cmake/project-config.cmake.in
@@ -15,10 +15,9 @@ set (@PROJECT_VARIANT_NAME@_INCLUDE_DIRS "${_ROOT}/@INCLUDEDIR@")
 set (@PROJECT_VARIANT_NAME@_LIBRARY_DIRS "${_ROOT}/@LIBDIR@")
 set (@PROJECT_VARIANT_NAME@_BINARY_DIRS "${_ROOT}/@BINDIR@")
 
-set (@PROJECT_VARIANT_NAME@_LIBRARIES @PROJECT_VARIANT_NAME@::proj)
+set (@PROJECT_VARIANT_NAME@_LIBRARIES proj)
 # Read in the exported definition of the library
 include ("${_DIR}/@PROJECT_NAME_LOWER@-targets.cmake")
-include ("${_DIR}/@PROJECT_LEGACY_LOWER@-targets.cmake")
 
 unset (_ROOT)
 unset (_DIR)
