diff --git a/src/lib_proj.cmake b/src/lib_proj.cmake
index 3d41fe9..6632e68 100644
--- a/src/lib_proj.cmake
+++ b/src/lib_proj.cmake
@@ -392,7 +392,7 @@ set(PROJ_LIBRARIES ${PROJ_LIBRARIES} PARENT_SCOPE)
 if(UNIX)
   find_library(M_LIB m)
   if(M_LIB)
-    target_link_libraries(${PROJ_CORE_TARGET} -lm)
+    target_link_libraries(${PROJ_CORE_TARGET} PRIVATE -lm)
   endif()
   find_library(DL_LIB dl)
   if(M_LIB)
@@ -400,7 +400,7 @@ if(UNIX)
   endif()
 endif()
 if(USE_THREAD AND Threads_FOUND AND CMAKE_USE_PTHREADS_INIT)
-  target_link_libraries(${PROJ_CORE_TARGET} ${CMAKE_THREAD_LIBS_INIT})
+  target_link_libraries(${PROJ_CORE_TARGET} PRIVATE ${CMAKE_THREAD_LIBS_INIT})
 endif()
 
 target_include_directories(${PROJ_CORE_TARGET} PRIVATE ${SQLITE3_INCLUDE_DIR})
