diff --git a/src/lib_proj.cmake b/src/lib_proj.cmake
index 3d41fe9..37439a8 100644
--- a/src/lib_proj.cmake
+++ b/src/lib_proj.cmake
@@ -404,7 +404,7 @@ if(USE_THREAD AND Threads_FOUND AND CMAKE_USE_PTHREADS_INIT)
 endif()
 
 target_include_directories(${PROJ_CORE_TARGET} PRIVATE ${SQLITE3_INCLUDE_DIR})
-target_link_libraries(${PROJ_CORE_TARGET} ${SQLITE3_LIBRARY})
+target_link_libraries(${PROJ_CORE_TARGET} PRIVATE ${SQLITE3_LIBRARY})
 
 if(TIFF_ENABLED)
   target_compile_definitions(${PROJ_CORE_TARGET} PRIVATE -DTIFF_ENABLED)
