diff --git a/cmake/igl/modules/core.cmake b/cmake/igl/modules/core.cmake
index 137d30b..0520482 100644
--- a/cmake/igl/modules/core.cmake
+++ b/cmake/igl/modules/core.cmake
@@ -14,6 +14,9 @@ target_include_directories(igl_core ${IGL_SCOPE}
 # 3. Target sources
 file(GLOB INC_FILES "${libigl_SOURCE_DIR}/include/igl/*.h" "${libigl_SOURCE_DIR}/include/igl/*.hpp")
 file(GLOB SRC_FILES "${libigl_SOURCE_DIR}/include/igl/*.cpp")
+if(CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
+    list(FILTER SRC_FILES EXCLUDE REGEX "/file_dialog_(open|save)[.]cpp\$")
+endif()
 igl_target_sources(igl_core ${INC_FILES} ${SRC_FILES})
 
 # 4. Install target & headers
