diff --git a/Core/MOOSDB/CMakeLists.txt b/Core/MOOSDB/CMakeLists.txt
index 7b1a903..c7f7b50 100644
--- a/Core/MOOSDB/CMakeLists.txt
+++ b/Core/MOOSDB/CMakeLists.txt
@@ -18,6 +18,14 @@ if(DISABLE_NAMES_LOOKUP)
 endif()
 
 include_directories(${${EXECNAME}_INCLUDE_DIRS})
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/../libMOOS/include)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/../libMOOS/App/include)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/../libMOOS/Comms/include)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/../libMOOS/DB/include)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/../libMOOS/Utils/include)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/../libMOOS/Thirdparty/getpot/include)
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/../libMOOS/Thirdparty/PocoBits/include)
+
 add_executable(${EXECNAME} ${SRCS})
 target_link_libraries(${EXECNAME} MOOS)
 
diff --git a/Core/libMOOS/CMakeLists.txt b/Core/libMOOS/CMakeLists.txt
index 50c2f0a..e316121 100644
--- a/Core/libMOOS/CMakeLists.txt
+++ b/Core/libMOOS/CMakeLists.txt
@@ -152,9 +152,15 @@ elseif(WIN32)
         comctl32
         ws2_32   # to support IPv6
     )
-
+    
     # XP does not support IPv6. Therefore MOOS only supports Vista and later.
     target_compile_definitions(MOOS PUBLIC _WIN32_WINNT=0x600)
+    if (${MOOS_POCO_DLL})
+        target_compile_definitions(MOOS PUBLIC MOOS_POCO_DLL)
+    endif()
+    if (${Foundation_EXPORTS})
+        target_compile_definitions(MOOS PUBLIC Foundation_EXPORTS)
+    endif()
 endif()
 
 #do we want to disable host name lookups?
@@ -202,6 +208,7 @@ install(
 # install libraries
 install(TARGETS MOOS
   EXPORT MOOS
+  RUNTIME DESTINATION bin
   LIBRARY DESTINATION lib
   ARCHIVE DESTINATION lib
 )
