diff --git a/tcp_pubsub/cmake/tcp_pubsubConfig.cmake.in b/tcp_pubsub/cmake/tcp_pubsubConfig.cmake.in
index e4dd17a..553b0b1 100644
--- a/tcp_pubsub/cmake/tcp_pubsubConfig.cmake.in
+++ b/tcp_pubsub/cmake/tcp_pubsubConfig.cmake.in
@@ -1,4 +1,6 @@
 @PACKAGE_INIT@
 
-INCLUDE("@PACKAGE_TCP_PUBSUB_INSTALL_CMAKE_DIR@/tcp_pubsubTargets.cmake")
-find_package(Threads)
+include(CMakeFindDependencyMacro)
+find_dependency(Threads)
+
+include("${CMAKE_CURRENT_LIST_DIR}/tcp_pubsubTargets.cmake")
