diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1d8f579..b9c6d20 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -276,7 +276,9 @@ if (JAS_ENABLE_OPENGL AND OPENGL_FOUND)
 	set(JAS_HAVE_OPENGL 0)
 	message("OpenGL include directory: ${OPENGL_INCLUDE_DIR}")
 	message("OpenGL libraries: ${OPENGL_LIBRARIES}")
-	find_package(GLUT ${JAS_REQUIRED})
+	find_package(FreeGLUT CONFIG REQUIRED)
+    set(GLUT_FOUND FreeGLUT_FOUND)
+    set(GLUT_LIBRARIES FreeGLUT::freeglut)
 	message("GLUT library found: ${GLUT_FOUND}")
 	if (GLUT_FOUND)
 		message("GLUT include directory: ${GLUT_INCLUDE_DIR}")
