diff --git a/CMakeLists.txt b/CMakeLists.txt
index 76d2e9b..78f07a5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -437,6 +437,38 @@ if(LEVELDB_INSTALL)
       "${PROJECT_SOURCE_DIR}/${LEVELDB_PUBLIC_INCLUDE_DIR}/write_batch.h"
     DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/leveldb
   )
+  
+  install(
+    FILES
+      "${PROJECT_SOURCE_DIR}/util/arena.h"
+      "${PROJECT_SOURCE_DIR}/util/coding.h"
+      "${PROJECT_SOURCE_DIR}/util/crc32c.h"
+      "${PROJECT_SOURCE_DIR}/util/hash.h"
+      "${PROJECT_SOURCE_DIR}/util/logging.h"
+      "${PROJECT_SOURCE_DIR}/util/mutexlock.h"
+      "${PROJECT_SOURCE_DIR}/util/no_destructor.h"
+      "${PROJECT_SOURCE_DIR}/util/random.h"
+      "${PROJECT_SOURCE_DIR}/util/env_posix_test_helper.h"
+      "${PROJECT_SOURCE_DIR}/util/env_windows_test_helper.h"
+      "${PROJECT_SOURCE_DIR}/util/histogram.h"
+      "${PROJECT_SOURCE_DIR}/util/posix_logger.h"
+      "${PROJECT_SOURCE_DIR}/util/testharness.h"
+      "${PROJECT_SOURCE_DIR}/util/testutil.h"
+      "${PROJECT_SOURCE_DIR}/util/windows_logger.h"
+    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/leveldb/util
+  )
+
+  install(
+    FILES
+      "${PROJECT_SOURCE_DIR}/table/block.h"
+      "${PROJECT_SOURCE_DIR}/table/block_builder.h"
+      "${PROJECT_SOURCE_DIR}/table/filter_block.h"
+      "${PROJECT_SOURCE_DIR}/table/format.h"
+      "${PROJECT_SOURCE_DIR}/table/iterator_wrapper.h"
+      "${PROJECT_SOURCE_DIR}/table/merger.h"
+      "${PROJECT_SOURCE_DIR}/table/two_level_iterator.h"
+    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/leveldb/table
+  ) 
 
   include(CMakePackageConfigHelpers)
   write_basic_package_version_file(
