diff --git a/CMakeLists.txt b/CMakeLists.txt
index f8285b8..8e07863 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -451,6 +451,12 @@ if(LEVELDB_INSTALL)
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
   )
+  configure_package_config_file(${PROJECT_SOURCE_DIR}/cmake/leveldbConfig.cmake.in
+              leveldbConfig.cmake
+              INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/leveldb
+              NO_CHECK_REQUIRED_COMPONENTS_MACRO
+              PATH_VARS CMAKE_INSTALL_INCLUDEDIR
+              )
   install(
     FILES
       "${LEVELDB_PUBLIC_INCLUDE_DIR}/c.h"
