diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4158267..61828d3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -170,6 +170,7 @@ install(FILES ${CMAKE_BINARY_DIR}/libftdi1.pc ${CMAKE_BINARY_DIR}/libftdipp1.pc
         DESTINATION lib${LIB_SUFFIX}/pkgconfig)
 
 if (UNIX OR MINGW)
+elseif (0)
   configure_file ( libftdi1-config.in ${CMAKE_CURRENT_BINARY_DIR}/libftdi1-config @ONLY )
   install ( PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/libftdi1-config
             DESTINATION bin )
