diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7e8062a..ecf77fd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -436,6 +436,8 @@ if (ENABLE_ENCRYPTION)
 			set (SSL_INCLUDE_DIRS ${OPENSSL_INCLUDE_DIR})
 			set (SSL_LIBRARIES ${OPENSSL_LIBRARIES})
 			message(STATUS "SSL via find_package(OpenSSL): -I ${SSL_INCLUDE_DIRS} -l;${SSL_LIBRARIES}")
+			include_directories(${SSL_INCLUDE_DIRS})
+			link_libraries(${SSL_LIBRARIES})
 		endif()
 
 	endif()
@@ -1300,11 +1302,11 @@ if (ENABLE_APPS)
 
 	# Applications
 
-	srt_add_application(srt-live-transmit ${VIRTUAL_srtsupport})
+	srt_add_application(srt-live-transmit ${VIRTUAL_srtsupport} ${VIRTUAL_srt})
 	if (DEFINED EXTRA_stransmit)
 		set_target_properties(srt-live-transmit PROPERTIES COMPILE_FLAGS "${EXTRA_stransmit}")
 	endif()
-	srt_add_application(srt-file-transmit ${VIRTUAL_srtsupport})
+	srt_add_application(srt-file-transmit ${VIRTUAL_srtsupport} ${VIRTUAL_srt})
 
 	if (MINGW)
 		# FIXME: with MINGW, it fails to build apps that require C++11
@@ -1313,7 +1315,7 @@ if (ENABLE_APPS)
 	else()
 		# srt-multiplex temporarily blocked
 		#srt_add_application(srt-multiplex ${VIRTUAL_srtsupport})
-		srt_add_application(srt-tunnel ${VIRTUAL_srtsupport})
+		srt_add_application(srt-tunnel ${VIRTUAL_srtsupport} ${VIRTUAL_srt})
 		target_compile_definitions(srt-tunnel PUBLIC -DSRT_ENABLE_VERBOSE_LOCK)
 	endif()
 
