diff --git a/exporters/otlp/CMakeLists.txt b/exporters/otlp/CMakeLists.txt
index 227f59e..8500a78 100644
--- a/exporters/otlp/CMakeLists.txt
+++ b/exporters/otlp/CMakeLists.txt
@@ -68,7 +68,7 @@ if(OPENTELEMETRY_INSTALL)
     otlp_recordable
     "OpenTelemetry OTLP - Recordable"
     "OTLP recordable implementation for spans, metrics, and logs."
-    "opentelemetry_trace opentelemetry_metrics opentelemetry_logs opentelemetry_resources"
+    "opentelemetry_trace opentelemetry_metrics opentelemetry_logs opentelemetry_resources opentelemetry_proto"
   )
 
   opentelemetry_add_pkgconfig(
