diff --git a/CMakeLists.txt b/CMakeLists.txt
index dcf1034..f1e8a8e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -130,20 +130,6 @@ option(OPENTELEMETRY_INSTALL "Whether to install opentelemetry targets"

 include("${PROJECT_SOURCE_DIR}/cmake/tools.cmake")

-if(NOT DEFINED CMAKE_CXX_STANDARD)
-  if(WITH_STL)
-    # Require at least C++17. C++20 is needed to avoid gsl::span
-    if(CMAKE_VERSION VERSION_GREATER 3.11.999)
-      # Ask for 20, may get anything below
-      set(CMAKE_CXX_STANDARD 20)
-    else()
-      # Ask for 17, may get anything below
-      set(CMAKE_CXX_STANDARD 17)
-    endif()
-  else()
-    set(CMAKE_CXX_STANDARD 11)
-  endif()
-endif()

 if(WITH_STL)
   # These definitions are needed for test projects that do not link against
