diff --git a/CMakeLists.txt b/CMakeLists.txt
index 03472d7..daca809 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 3.2...3.10)
 
-project(WavPack VERSION 5.8.1)
+project(WavPack VERSION 5.8.1 LANGUAGES C CXX ASM)
 
 file(READ "${CMAKE_CURRENT_SOURCE_DIR}/configure.ac" CONFIGURE_AC)
 string(REGEX MATCH "LT_CURRENT=([0-9]+)" LT_CURRENT "${CONFIGURE_AC}")
@@ -33,10 +33,6 @@ endif()
 
 include(CheckLanguage)
 
-check_language(ASM)
-if(CMAKE_ASM_COMPILER)
-  enable_language(ASM)
-endif()
 
 if(MSVC)
   if(WavPack_CPU_X86 OR WavPack_CPU_X64)
