diff --git a/cmake/Config.cmake b/cmake/Config.cmake
index c6e2b40d..d8b4d689 100644
--- a/cmake/Config.cmake
+++ b/cmake/Config.cmake
@@ -35,7 +35,7 @@ if (MSVC)
     # Bond with MSVC CRT-specific code too much. More details at
     # https://msdn.microsoft.com/en-us/library/ms175759.aspx
     add_definitions (-D_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES=1 -D_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES_COUNT=1)
-    set (Boost_USE_STATIC_LIBS ON)
+    set (Boost_USE_STATIC_LIBS OFF)
 endif (MSVC)
 
 if (WIN32)
@@ -106,9 +106,6 @@ if (DEFINED ENV{APPVEYOR} AND ("$ENV{BOND_BUILD}" STREQUAL "C++"))
     endif()
 endif()
 
-# disable Boost auto-linking
-add_definitions (-DBOOST_ALL_NO_LIB)
-
 # VS2015U2 fixed a bug with atomics and emits a warning without this definition.
 add_definitions (-D_ENABLE_ATOMIC_ALIGNMENT_FIX)
 
