diff --git a/cmake/Config.cmake b/cmake/Config.cmake
index b51c76a6..4cb5a6ff 100644
--- a/cmake/Config.cmake
+++ b/cmake/Config.cmake
@@ -52,7 +52,7 @@ if (MSVC)
       add_compile_options (/permissive-)
     endif()
 
-    set (Boost_USE_STATIC_LIBS ON)
+    set (Boost_USE_STATIC_LIBS OFF)
 endif (MSVC)
 
 if (WIN32)
@@ -137,9 +137,6 @@ if (DEFINED ENV{APPVEYOR} AND ("$ENV{BOND_BUILD}" STREQUAL "C++"))
     endif()
 endif()
 
-# disable Boost auto-linking
-add_definitions (-DBOOST_ALL_NO_LIB)
-
 # VS2015U2 fixed a bug with atomics and emits a warning without this definition.
 add_definitions (-D_ENABLE_ATOMIC_ALIGNMENT_FIX)
 
