diff --git a/CMakeLists.txt b/CMakeLists.txt
index f2f8eaa..1b0c01c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,7 +23,7 @@ set (BOND_FIND_RAPIDJSON
 # settings so that we don't apply our settings to third-party code.
 add_subdirectory (thirdparty)
 
-enable_testing()
+#enable_testing()
 
 set (BOND_IDL ${CMAKE_CURRENT_SOURCE_DIR}/idl)
 set (BOND_INCLUDE ${CMAKE_CURRENT_SOURCE_DIR}/cpp/inc)
@@ -85,6 +85,6 @@ if (BOND_GBC_PATH)
 
   install (
     FILES ${BOND_GBC_PATH}
-    DESTINATION bin
+    DESTINATION tools
     RENAME ${INSTALLED_GBC_NAME})
 endif()
diff --git a/compiler/CMakeLists.txt b/compiler/CMakeLists.txt
index 1dff9d0..9a11575 100644
--- a/compiler/CMakeLists.txt
+++ b/compiler/CMakeLists.txt
@@ -108,7 +108,7 @@ set (test_sources
     tests/TestMain.hs
     ${tests})
 
-set (completion_dir etc/bash_completion.d)
+set (completion_dir tools/bond)
 set (completion ${CMAKE_CURRENT_BINARY_DIR}/gbc.comp)
 set (output ${CMAKE_CURRENT_BINARY_DIR}/build/gbc/gbc${CMAKE_EXECUTABLE_SUFFIX})
 set (GBC_EXECUTABLE ${output} PARENT_SCOPE)
@@ -130,7 +130,7 @@ endif()
 
 install (FILES ${output}
     PERMISSIONS OWNER_READ OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE
-    DESTINATION bin)
+    DESTINATION ${completion_dir})
 
 install (FILES ${completion}
          RENAME gbc
diff --git a/cpp/CMakeLists.txt b/cpp/CMakeLists.txt
index b45078e..1eebe9c 100644
--- a/cpp/CMakeLists.txt
+++ b/cpp/CMakeLists.txt
@@ -110,7 +110,7 @@ target_include_directories (bond_apply BEFORE PRIVATE
 
 install (TARGETS bond bond_apply
     EXPORT bond
-    ARCHIVE DESTINATION lib/bond
+    ARCHIVE DESTINATION lib
     INCLUDES DESTINATION include)
 
 install (DIRECTORY ${BOND_IDL}/bond/core DESTINATION include/bond)
