diff --git a/thirdparty/CMakeLists.txt b/thirdparty/CMakeLists.txt
index 8c1f368..8b13789 100644
--- a/thirdparty/CMakeLists.txt
+++ b/thirdparty/CMakeLists.txt
@@ -1,10 +1 @@
-include (Compiler)
 
-if (BOND_ENABLE_GRPC)
-    if (BOND_FIND_GRPC)
-        find_package(grpc CONFIG REQUIRED)
-    else ()
-        cxx_add_compile_options(Clang -Wno-unused-value)
-        add_subdirectory(grpc)
-    endif ()
-endif ()
