diff --git a/packaging/libaec-config.cmake b/packaging/libaec-config.cmake
index 6ef9350..d07d2ea 100644
--- a/packaging/libaec-config.cmake
+++ b/packaging/libaec-config.cmake
@@ -38,6 +38,7 @@
 include(${CMAKE_CURRENT_LIST_DIR}/libaec-targets.cmake)
 
 # Alias static or shared targets depending on libaec_USE_STATIC_LIBS
+if(DEFINED libaec_USE_STATIC_LIBS)
 if(libaec_USE_STATIC_LIBS)
   if(TARGET libaec::aec-static AND TARGET libaec::sz-static)
     add_library(libaec::aec ALIAS libaec::aec-static)
@@ -57,6 +58,15 @@ else()
     set(${CMAKE_FIND_PACKAGE_NAME}_FOUND FALSE)
   endif()
 endif()
+else()
+  if(TARGET libaec::aec-shared AND TARGET libaec::sz-shared)
+    add_library(libaec::aec ALIAS libaec::aec-shared)
+    add_library(libaec::sz ALIAS libaec::sz-shared)
+  elseif(TARGET libaec::aec-static AND TARGET libaec::sz-static)
+    add_library(libaec::aec ALIAS libaec::aec-static)
+    add_library(libaec::sz ALIAS libaec::sz-static)
+  endif()
+endif()
 
 if(TARGET libaec::sz)
   get_target_property(SZIP_INCLUDE_DIR
