From 255fc7b45c75eb1ea034beed335912b95474467e Mon Sep 17 00:00:00 2001
From: Sandy Carter <bwrsandman@gmail.com>
Date: Wed, 20 Dec 2023 17:40:35 -0500
Subject: [PATCH] Typedef POSIX types on Windows

---
 include/cdio/types.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index ef8e495c..a493bf94 100644
--- a/configure.ac
+++ b/configure.ac
@@ -470,7 +470,7 @@ int has_timeout=sizeof(test.timeout);],
          native_abs_top_srcdir=$(cd $srcdir; /bin/bash.exe -c 'builtin pwd -W')
        fi

-       AC_CHECK_HEADERS(windows.h)
+       AC_CHECK_HEADERS(windows.h ddk/scsi.h ntddcdrm.h ddk/ntddcdrm.h ntddscsi.h ddk/ntddscsi.h)
        AC_DEFINE([MINGW32], [1],
                   [Define 1 if you are compiling using MinGW])
        MINGW32=1
@@ -482,14 +482,8 @@ int has_timeout=sizeof(test.timeout);],
        ## Cross-platform LFS support in MinGW requires an override of off_t
        if test "x$enable_largefile" != "xno"; then
          AC_MSG_RESULT([enabling support for large files (_FILE_OFFSET_BITS=64)])
-          if test "x$MINGW_W64_HEADERS" = "xyes"
-          then
            LIBCDIO_CFLAGS+=' -D_FILE_OFFSET_BITS=64 '
            LIBISO9660_CFLAGS+=' -D_FILE_OFFSET_BITS=64 '
-          else
-           LIBCDIO_CFLAGS+=' -D_FILE_OFFSET_BITS=64 -D_OFF_T_DEFINED -D_OFF_T_ -D_off_t=off64_t -Doff_t=off64_t'
-           LIBISO9660_CFLAGS+=' -D_FILE_OFFSET_BITS=64 -D_OFF_T_DEFINED -D_OFF_T_ -D_off_t=off64_t -Doff_t=off64_t'
-          fi
        fi
        joliet_supported="yes"
        if test "x$enable_cdda_player" = "xyes"; then

diff --git a/include/cdio/types.h b/include/cdio/types.h
index fd735786..8fea19a4 100644
--- a/include/cdio/types.h
+++ b/include/cdio/types.h
@@ -55,7 +55,9 @@ typedef uint8_t ubyte;
    unistd.h that defines them. Such a file is provided with
    the libcdio source, in the MSVC/missing directory */
 #if defined(_MSC_VER)
-#include <unistd.h>
+#include <BaseTsd.h>
+typedef int mode_t;
+typedef SSIZE_T ssize_t;
 #endif
 
   /* default HP/UX macros are broken */
diff --git a/lib/driver/MSWindows/aspi32.c b/lib/driver/MSWindows/aspi32.c
index 20ec8c36..26654063 100644
--- a/lib/driver/MSWindows/aspi32.c
+++ b/lib/driver/MSWindows/aspi32.c
@@ -26,6 +26,8 @@
 # define __CDIO_CONFIG_H__ 1
 #endif

+#include <windows.h>
+
 #include <cdio/cdio.h>
 #include <cdio/sector.h>
 #include <cdio/util.h>
@@ -41,10 +43,8 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <errno.h>
-#include <unistd.h>
 #include <fcntl.h>
 
-#include <windows.h>
 #include "win32.h"
 
 #include <sys/stat.h>
diff --git a/lib/driver/MSWindows/win32.c b/lib/driver/MSWindows/win32.c
index 6d931514..9c563071 100644
--- a/lib/driver/MSWindows/win32.c
+++ b/lib/driver/MSWindows/win32.c
@@ -28,6 +28,8 @@
 # include <stdbool.h>
 #endif
 
+#include <windows.h>
+
 #include <cdio/cdio.h>
 #include <cdio/sector.h>
 #include <cdio/util.h>
@@ -59,7 +61,6 @@
 #include <fcntl.h>
 #endif
 
-#include <windows.h>
 #include <winioctl.h>
 #include "win32.h"
 
diff --git a/lib/driver/MSWindows/win32_ioctl.c b/lib/driver/MSWindows/win32_ioctl.c
index ea65bae6..7c102ce6 100644
--- a/lib/driver/MSWindows/win32_ioctl.c
+++ b/lib/driver/MSWindows/win32_ioctl.c
@@ -68,6 +68,10 @@
 #include <cdio/mmc.h>
 #include "cdio/logging.h"

+#ifndef __PRETTY_FUNCTION__
+#define __PRETTY_FUNCTION__ __FUNCSIG__
+#endif
+
 #if defined (_XBOX)
 #define windows_error(loglevel,i_err) {                    \
    cdio_log(loglevel, "Error: file %s: line %d (%s) %ld\n", \
diff --git a/lib/driver/image/bincue.c b/lib/driver/image/bincue.c
index d329cdc9..bc37ed71 100644
--- a/lib/driver/image/bincue.c
+++ b/lib/driver/image/bincue.c
@@ -23,6 +23,13 @@
    residing inside a disk file (*.bin) and its associated cue sheet.
    (*.cue).
 */
+
+#include "config.h"
+
+#ifdef HAVE_WINDOWS_H
+#include <windows.h>
+#endif
+
 #include "portable.h"
 #include "image.h"
 #include "cdio_assert.h"
@@ -57,9 +64,6 @@
 #else
 #define PRId64 "lld"
 #endif
-#ifdef HAVE_WINDOWS_H
-#include <windows.h>
-#endif

 #include <ctype.h>

diff --git a/lib/driver/portable.h b/lib/driver/portable.h
index 8b0ad7ba..360bd592 100644
--- a/lib/driver/portable.h
+++ b/lib/driver/portable.h
@@ -35,18 +35,6 @@
 # endif
 #endif /*HAVE_FTRUNCATE*/

-#if !defined(HAVE_SNPRINTF)
-# if defined (_MSC_VER)
-#  define snprintf _snprintf
-# endif
-#endif /*HAVE_SNPRINTF*/
-
-#if !defined(HAVE_VSNPRINTF)
-# if defined (_MSC_VER)
-#  define vsnprintf _vsnprintf
-# endif
-#endif /*HAVE_SNPRINTF*/
-
 #if !defined(HAVE_DRAND48) && defined(HAVE_RAND)
 # define drand48()   (rand() / (double)RAND_MAX)
 #endif
diff --git a/autogen.sh b/autogen.sh
index 408af0ad..275bcbe5 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -14,3 +14,5 @@ if [ $? -ne 0 ]; then
 fi
 
 ./configure --enable-maintainer-mode "$@"
+
+sed -i 's,-Xlinker,-Xlinkerxxx:,g' compile
-- 
2.39.3 (Apple Git-145)

