diff --git a/cmake/HPX_GeneratePackageUtils.cmake b/cmake/HPX_GeneratePackageUtils.cmake
index 3ec8002..b098465 100644
--- a/cmake/HPX_AddModule.cmake	
+++ b/cmake/HPX_AddModule.cmake
@@ -437,7 +437,7 @@ function(add_hpx_module libname modulename)
     elseif(MSVC)
       set(_module_target hpx_${modulename})
       target_link_libraries(
-        hpx_${libname} PRIVATE -WHOLEARCHIVE:$<TARGET_FILE:hpx_${modulename}>
+        hpx_${libname} PRIVATE -WHOLEARCHIVE:$<TARGET_FILE:$<TARGET_NAME:hpx_${modulename}>>
       )
     endif()
     target_link_libraries(hpx_${libname} PRIVATE ${_module_target})
--- a/cmake/HPX_GeneratePackageUtils.cmake	
+++ b/cmake/HPX_GeneratePackageUtils.cmake
@@ -101,7 +101,7 @@ function(
           )
             set(_libraries
                 ${_libraries}
-                $<INSTALL_INTERFACE:${CMAKE_INSTALL_LIBDIR}/$<TARGET_FILE_NAME:${target}>>$<BUILD_INTERFACE:$<TARGET_FILE:${target}>>
+                $<INSTALL_INTERFACE:${CMAKE_INSTALL_LIBDIR}/$<TARGET_FILE_NAME:$<TARGET_NAME:${target}>>>$<BUILD_INTERFACE:$<TARGET_FILE:${target}>>
             )
           endif()
         elseif("${dep_target}" MATCHES "^-l")
