diff --git a/modules/imgcodecs/src/grfmt_jpeg2000.cpp b/modules/imgcodecs/src/grfmt_jpeg2000.cpp
--- a/modules/imgcodecs/src/grfmt_jpeg2000.cpp
+++ b/modules/imgcodecs/src/grfmt_jpeg2000.cpp
@@ -375,13 +375,13 @@
     int lshift = MAX(0, -rshift);
     rshift = MAX(0, rshift);
     int delta = (rshift > 0 ? 1 << (rshift - 1) : 0) + offset;
 
     for( y = 0; y < yend - ystart; )
     {
-        jas_seqent_t* pix_row = &jas_matrix_get( buffer, y / ystep, 0 );
+        jas_seqent_t* pix_row = jas_matrix_getref( buffer, y / ystep, 0 );
         uchar* dst = data + (y - yoffset) * step - xoffset;
 
         if( xstep == 1 )
         {
             if( maxval == 256 && offset == 0 )
                 for( x = 0; x < xend - xstart; x++ )
@@ -441,13 +441,13 @@
     int lshift = MAX(0, -rshift);
     rshift = MAX(0, rshift);
     int delta = (rshift > 0 ? 1 << (rshift - 1) : 0) + offset;
 
     for( y = 0; y < yend - ystart; )
     {
-        jas_seqent_t* pix_row = &jas_matrix_get( buffer, y / ystep, 0 );
+        jas_seqent_t* pix_row = jas_matrix_getref( buffer, y / ystep, 0 );
         ushort* dst = data + (y - yoffset) * step - xoffset;
 
         if( xstep == 1 )
         {
             if( maxval == 65536 && offset == 0 )
                 for( x = 0; x < xend - xstart; x++ )
