diff --git a/CMakeLists.txt b/CMakeLists.txt
index b51bcde..15edc92 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -803,7 +803,7 @@ if(WITH_OPENVX)
 endif()
 
 if(WITH_QUIRC)
-  add_subdirectory(3rdparty/quirc)
+  find_package(quirc CONFIG REQUIRED)
   set(HAVE_QUIRC TRUE)
 endif()
 
diff --git a/modules/objdetect/CMakeLists.txt b/modules/objdetect/CMakeLists.txt
index 3fa0c5d..f4c0ad6 100644
--- a/modules/objdetect/CMakeLists.txt
+++ b/modules/objdetect/CMakeLists.txt
@@ -2,7 +2,5 @@ set(the_description "Object Detection")
 ocv_define_module(objdetect opencv_core opencv_imgproc opencv_calib3d WRAP java objc python js)
 
 if(HAVE_QUIRC)
-    get_property(QUIRC_INCLUDE GLOBAL PROPERTY QUIRC_INCLUDE_DIR)
-    ocv_include_directories(${QUIRC_INCLUDE})
-    ocv_target_link_libraries(${the_module} quirc)
+    ocv_target_link_libraries(${the_module} quirc::quirc)
 endif()
