diff --git a/CMakeLists.txt b/CMakeLists.txt
index 79a0c96270..e6b1aefb9b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -789,7 +789,9 @@ endif()
 # ----------------------------------------------------------------------------
 
 if(ANDROID AND WITH_CPUFEATURES)
-  add_subdirectory(3rdparty/cpufeatures)
+  find_package(CpuFeaturesNdkCompat CONFIG REQUIRED)
+  set(CPUFEATURES_INCLUDE_DIRS "")
+  set(CPUFEATURES_LIBRARIES CpuFeatures::ndk_compat)
   set(HAVE_CPUFEATURES 1)
 endif()
 
