diff --git a/cmake/templates/OpenCVConfig-ANDROID.cmake.in b/cmake/templates/OpenCVConfig-ANDROID.cmake.in
index 1787acab38..5d9540677e 100644
--- a/cmake/templates/OpenCVConfig-ANDROID.cmake.in
+++ b/cmake/templates/OpenCVConfig-ANDROID.cmake.in
@@ -11,3 +11,8 @@ if(OpenCV_ANDROID_NATIVE_API_LEVEL GREATER ANDROID_NATIVE_API_LEVEL)
   set(OpenCV_FOUND 0)
   return()
 endif()
+
+if(@ANDROID@ AND @HAVE_CPUFEATURES@)
+  include(CMakeFindDependencyMacro)
+  find_dependency(CpuFeaturesNdkCompat CONFIG REQUIRED)
+endif()
