--- a/cmake/OpenCVUtils.cmake.original	2024-10-01 16:59:17.788263700 +0200
+++ b/cmake/OpenCVUtils.cmake	2024-10-01 17:10:11.870857900 +0200
@@ -1638,7 +1638,7 @@
       AND NOT (CMAKE_VERSION VERSION_LESS "3.13.0")  # upgrade CMake: https://gitlab.kitware.com/cmake/cmake/-/merge_requests/2152
   )
     foreach(tgt in ${ARGN})
-      if(tgt MATCHES "^ocv\.3rdparty\.")
+      if(tgt MATCHES "^ocv\\.3rdparty\\.")
         list(FIND __OPENCV_EXPORTED_EXTERNAL_TARGETS "${tgt}" _found)
         if(_found EQUAL -1)  # don't export target twice
           install(TARGETS ${tgt} EXPORT OpenCVModules)
