diff --git a/include/fastrtps/utils/TimedMutex.hpp b/include/fastrtps/utils/TimedMutex.hpp
index b1982f5..f1e67f0 100644
--- a/include/fastrtps/utils/TimedMutex.hpp
+++ b/include/fastrtps/utils/TimedMutex.hpp
@@ -39,6 +39,12 @@ namespace fastrtps {
 #if defined(_WIN32)
 class TimedMutex
 {
+    // On MSVC 19.38.32926.95 `_Thrd_success` was changed into `_Thrd_result::_Success`.
+    // See https://github.com/eProsima/Fast-DDS/issues/3783
+    // See https://github.com/microsoft/STL/pull/3897
+#if defined(_MSC_FULL_VER) && _MSC_FULL_VER >= 193832926
+    using _Thrd_success = _Thrd_result::_Success;
+#endif
 public:

     TimedMutex()
