diff --git a/CMakeLists.txt b/CMakeLists.txt
index e5021d5..47cbae1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -511,7 +511,7 @@ endif(ENABLE_CPP_CORE_GUIDELINES_CHECK)
 
 add_library(netgen_cgns INTERFACE)
 if(USE_CGNS)
-    find_library( CGNS_LIBRARY NAMES cgns cgnsdll )
+    find_library( CGNS_LIBRARY NAMES cgnsdll cgns)
     find_path( CGNS_INCLUDE_DIR cgnslib.h )
     target_compile_definitions(netgen_cgns INTERFACE NG_CGNS)
     target_include_directories(netgen_cgns INTERFACE ${CGNS_INCLUDE_DIR})
