diff --git a/build/cmake/CMakeLists.txt b/build/cmake/CMakeLists.txt
index 9b5d7ef..585ce62 100644
--- a/build/cmake/CMakeLists.txt
+++ b/build/cmake/CMakeLists.txt
@@ -23,6 +23,10 @@ else()
 endif()
 cmake_policy(VERSION ${ZSTD_CMAKE_POLICY_VERSION})
 
+if(WIN32)
+  add_compile_options(/wd4146)
+endif()
+
 set(CMAKE_BUILD_WITH_INSTALL_RPATH on)
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/CMakeModules")
