diff --git a/w32/common/libmicrohttpd-files.vcxproj b/w32/common/libmicrohttpd-files.vcxproj
index 6f1e03b..c0ddea8 100644
--- a/w32/common/libmicrohttpd-files.vcxproj
+++ b/w32/common/libmicrohttpd-files.vcxproj
@@ -3,6 +3,7 @@
   <ItemGroup>
     <ClCompile Include="$(MhdSrc)microhttpd\base64.c" />
     <ClCompile Include="$(MhdSrc)microhttpd\basicauth.c" />
+    <ClCompile Include="$(MhdSrc)microhttpd\sha256.c" />
     <ClCompile Include="$(MhdSrc)microhttpd\connection.c" />
     <ClCompile Include="$(MhdSrc)microhttpd\daemon.c" />
     <ClCompile Include="$(MhdSrc)microhttpd\digestauth.c" />
