diff --git a/CMakeLists.txt b/CMakeLists.txt
index bb4ccd5..dfbc767 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -169,7 +169,7 @@ if (NOT ASMTK_EMBED)
     $<$<NOT:$<CONFIG:Debug>>:${ASMTK_PRIVATE_CFLAGS_REL}>)

   if(ASMJIT_EXTERNAL)
-    target_link_libraries(asmtk PUBLIC ${ASMJIT_LIBRARY})
+    target_link_libraries(asmtk PUBLIC asmjit::asmjit)
     find_path(ASMJIT_INCLUDE_DIR NAMES asmjit/core.h)
     target_include_directories(asmtk PRIVATE ${ASMJIT_INCLUDE_DIR})
   else()
