diff --git a/Source/ThirdParty/ik/CMakeLists.txt b/Source/ThirdParty/ik/CMakeLists.txt
index 029791413..603b2d13f 100644
--- a/Source/ThirdParty/ik/CMakeLists.txt
+++ b/Source/ThirdParty/ik/CMakeLists.txt
@@ -31,7 +31,7 @@ if (CMAKE_BUILD_TYPE STREQUAL Debug)
     set (DEFAULT_MEMORY_DEBUGGING 1)
 endif ()
 option (IK_MEMORY_DEBUGGING "Global switch for memory options. Keep track of the number of allocations and de-allocations and prints a report when the program shuts down" ${DEFAULT_MEMORY_DEBUGGING})
-cmake_dependent_option (IK_MEMORY_BACKTRACE "Generate backtraces for every malloc(), making it easy to track down memory leaks" "${DEFAULT_MEMORY_DEBUGGING}" "IK_MEMORY_DEBUGGING AND NOT WEB" FALSE)
+cmake_dependent_option (IK_MEMORY_BACKTRACE "Generate backtraces for every malloc(), making it easy to track down memory leaks" "${DEFAULT_MEMORY_DEBUGGING}" "IK_MEMORY_DEBUGGING AND NOT WEB AND NOT WIN32" FALSE)
 
 # Need to set IK_PLATFORM for dllimport/dllexport
 if (WIN32)
