diff --git a/Source/ThirdParty/AngelScript/CMakeLists.txt b/Source/ThirdParty/AngelScript/CMakeLists.txt
index 763dacd9c..3507c0d28 100644
--- a/Source/ThirdParty/AngelScript/CMakeLists.txt
+++ b/Source/ThirdParty/AngelScript/CMakeLists.txt
@@ -58,7 +58,7 @@ list (APPEND SOURCE_FILES ${ASM_FILES})
 set (INCLUDE_DIRS include)
 
 # Setup target
-setup_library ()
+setup_library (STATIC)
 
 # Install headers for building and using the Urho3D library (install dependency for Urho3D/Script/APITemplates.h)
 install_header_files (DIRECTORY include/ DESTINATION ${DEST_INCLUDE_DIR}/ThirdParty/AngelScript FILES_MATCHING PATTERN *.h)     # Note: the trailing slash is significant
diff --git a/Source/ThirdParty/Assimp/code/CMakeLists.txt b/Source/ThirdParty/Assimp/code/CMakeLists.txt
index bdd82340e..55839dae0 100644
--- a/Source/ThirdParty/Assimp/code/CMakeLists.txt
+++ b/Source/ThirdParty/Assimp/code/CMakeLists.txt
@@ -938,7 +938,7 @@ set (SOURCE_FILES ${assimp_src})
 list (APPEND TARGET_PROPERTIES XCODE_ATTRIBUTE_ONLY_ACTIVE_ARCH YES)
 
 # Setup target
-setup_library ()
+setup_library (STATIC)
 
 IF (ASSIMP_BUILD_NONFREE_C4D_IMPORTER)
   TARGET_LINK_LIBRARIES(Assimp optimized ${C4D_RELEASE_LIBRARIES})
diff --git a/Source/ThirdParty/Box2D/CMakeLists.txt b/Source/ThirdParty/Box2D/CMakeLists.txt
index 6768fe4c0..2ca2c89e6 100644
--- a/Source/ThirdParty/Box2D/CMakeLists.txt
+++ b/Source/ThirdParty/Box2D/CMakeLists.txt
@@ -183,7 +183,7 @@ source_group(Rope FILES ${BOX2D_Rope_SRCS} ${BOX2D_Rope_HDRS})
 set (INCLUDE_DIRS .)
 
 # Setup target
-setup_library ()
+setup_library (STATIC)
 
 # Install headers for building and using the Urho3D library (install dependency for Urho3D/Urho2D/PhysicsWorld2D.h and Urho3D/Urho2D/RigidBody2D.h)
 install_header_files (DIRECTORY Box2D DESTINATION ${DEST_INCLUDE_DIR}/ThirdParty FILES_MATCHING PATTERN *.h)
diff --git a/Source/ThirdParty/Bullet/CMakeLists.txt b/Source/ThirdParty/Bullet/CMakeLists.txt
index 3e0419087..dbd506af0 100644
--- a/Source/ThirdParty/Bullet/CMakeLists.txt
+++ b/Source/ThirdParty/Bullet/CMakeLists.txt
@@ -41,7 +41,7 @@ define_source_files (RECURSE GLOB_CPP_PATTERNS src/*.cpp GLOB_H_PATTERNS src/*.h
 set (INCLUDE_DIRS src)
 
 # Setup target
-setup_library ()
+setup_library (STATIC)
 
 # Install headers for building and using the Urho3D library (install dependency for Urho3D/Physics/PhysicsWorld.h, Urho3D/Physics/RigidBody.h, and Urho3D/Physics/PhysicsUtils.h)
 install_header_files (DIRECTORY src/ DESTINATION ${DEST_INCLUDE_DIR}/ThirdParty/Bullet FILES_MATCHING PATTERN *.h)  # Note: the trailing slash is significant
diff --git a/Source/ThirdParty/Civetweb/CMakeLists.txt b/Source/ThirdParty/Civetweb/CMakeLists.txt
index a2d41caad..6b64ed719 100644
--- a/Source/ThirdParty/Civetweb/CMakeLists.txt
+++ b/Source/ThirdParty/Civetweb/CMakeLists.txt
@@ -61,7 +61,7 @@ define_source_files (GLOB_CPP_PATTERNS src/*.c GLOB_H_PATTERNS include/*.h)
 set (INCLUDE_DIRS include)
 
 # Setup target
-setup_library ()
+setup_library (STATIC)
 
 # Link OpenSSL libraries
 if (URHO3D_SSL)
diff --git a/Source/ThirdParty/Detour/CMakeLists.txt b/Source/ThirdParty/Detour/CMakeLists.txt
index 7d78b5aeb..2fe08df47 100644
--- a/Source/ThirdParty/Detour/CMakeLists.txt
+++ b/Source/ThirdParty/Detour/CMakeLists.txt
@@ -30,7 +30,7 @@ define_source_files (GLOB_CPP_PATTERNS Source/*.cpp GLOB_H_PATTERNS Include/*.h)
 set (INCLUDE_DIRS Include)
 
 # Setup target
-setup_library ()
+setup_library (STATIC)
 
 # Install headers for building the Urho3D library
 install_header_files (DIRECTORY Include/ DESTINATION ${DEST_INCLUDE_DIR}/ThirdParty/Detour FILES_MATCHING PATTERN *.h BUILD_TREE_ONLY)  # Note: the trailing slash is significant
diff --git a/Source/ThirdParty/DetourCrowd/CMakeLists.txt b/Source/ThirdParty/DetourCrowd/CMakeLists.txt
index d8f915372..8b3d7d135 100644
--- a/Source/ThirdParty/DetourCrowd/CMakeLists.txt
+++ b/Source/ThirdParty/DetourCrowd/CMakeLists.txt
@@ -30,7 +30,7 @@ define_source_files (GLOB_CPP_PATTERNS Source/*.cpp GLOB_H_PATTERNS Include/*.h)
 set (INCLUDE_DIRS Include ../Detour/Include)
 
 # Setup target
-setup_library ()
+setup_library (STATIC)
 
 # Install headers for building the Urho3D library
 install_header_files (DIRECTORY Include/ DESTINATION ${DEST_INCLUDE_DIR}/ThirdParty/DetourCrowd FILES_MATCHING PATTERN *.h BUILD_TREE_ONLY)  # Note: the trailing slash is significant
diff --git a/Source/ThirdParty/DetourTileCache/CMakeLists.txt b/Source/ThirdParty/DetourTileCache/CMakeLists.txt
index 519fdae17..edf38ee27 100644
--- a/Source/ThirdParty/DetourTileCache/CMakeLists.txt
+++ b/Source/ThirdParty/DetourTileCache/CMakeLists.txt
@@ -30,7 +30,7 @@ define_source_files (GLOB_CPP_PATTERNS Source/*.cpp GLOB_H_PATTERNS Include/*.h)
 set (INCLUDE_DIRS Include ../Detour/Include)
 
 # Setup target
-setup_library ()
+setup_library (STATIC)
 
 # Install headers for building the Urho3D library
 install_header_files (DIRECTORY Include/ DESTINATION ${DEST_INCLUDE_DIR}/ThirdParty/DetourTileCache FILES_MATCHING PATTERN *.h BUILD_TREE_ONLY)  # Note: the trailing slash is significant
diff --git a/Source/ThirdParty/ETCPACK/CMakeLists.txt b/Source/ThirdParty/ETCPACK/CMakeLists.txt
index 555b2c28c..9f3dd54a0 100644
--- a/Source/ThirdParty/ETCPACK/CMakeLists.txt
+++ b/Source/ThirdParty/ETCPACK/CMakeLists.txt
@@ -27,6 +27,6 @@ set (TARGET_NAME ETCPACK)
 set (SOURCE_FILES source/etcdec.cxx)
 
 # Setup target
-setup_library ()
+setup_library (STATIC)
 
 
diff --git a/Source/ThirdParty/FreeType/CMakeLists.txt b/Source/ThirdParty/FreeType/CMakeLists.txt
index 306af2e34..bbc4cab0a 100644
--- a/Source/ThirdParty/FreeType/CMakeLists.txt
+++ b/Source/ThirdParty/FreeType/CMakeLists.txt
@@ -78,7 +78,7 @@ set (SOURCE_FILES
 set (INCLUDE_DIRS include)
 
 # Setup target
-setup_library ()
+setup_library (STATIC)
 
 # Install headers for building the Urho3D library
 install_header_files (DIRECTORY include/freetype DESTINATION ${DEST_INCLUDE_DIR}/ThirdParty FILES_MATCHING PATTERN *.h BUILD_TREE_ONLY)
diff --git a/Source/ThirdParty/GLEW/CMakeLists.txt b/Source/ThirdParty/GLEW/CMakeLists.txt
index 9ec4ab843..d5feebd96 100644
--- a/Source/ThirdParty/GLEW/CMakeLists.txt
+++ b/Source/ThirdParty/GLEW/CMakeLists.txt
@@ -33,7 +33,7 @@ define_source_files (GLOB_CPP_PATTERNS *.c)
 set (INCLUDE_DIRS .)
 
 # Setup target
-setup_library ()
+setup_library (STATIC)
 
 # Install headers for building and using the Urho3D library (install dependency for OGLGraphicsImpl.h)
 install_header_files (DIRECTORY ./ DESTINATION ${DEST_INCLUDE_DIR}/ThirdParty/GLEW FILES_MATCHING PATTERN *.h)  # Note: the trailing slash is significant
diff --git a/Source/ThirdParty/LZ4/CMakeLists.txt b/Source/ThirdParty/LZ4/CMakeLists.txt
index 579407338..57f93de92 100644
--- a/Source/ThirdParty/LZ4/CMakeLists.txt
+++ b/Source/ThirdParty/LZ4/CMakeLists.txt
@@ -27,7 +27,7 @@ set (TARGET_NAME LZ4)
 define_source_files (GLOB_CPP_PATTERNS *.c)
 
 # Setup target
-setup_library ()
+setup_library (STATIC)
 
 # Install headers for building the Urho3D library
 install_header_files (DIRECTORY ./ DESTINATION ${DEST_INCLUDE_DIR}/ThirdParty/LZ4 FILES_MATCHING PATTERN *.h BUILD_TREE_ONLY)  # Note: the trailing slash is significant
diff --git a/Source/ThirdParty/LibCpuId/CMakeLists.txt b/Source/ThirdParty/LibCpuId/CMakeLists.txt
index 6c4ab7291..5c4cc5069 100644
--- a/Source/ThirdParty/LibCpuId/CMakeLists.txt
+++ b/Source/ThirdParty/LibCpuId/CMakeLists.txt
@@ -47,7 +47,7 @@ list (APPEND SOURCE_FILES ${ASM_FILES})
 set (INCLUDE_DIRS src)
 
 # Setup target
-setup_library ()
+setup_library (STATIC)
 
 # Install headers for building the Urho3D library
 install_header_files (DIRECTORY src/ DESTINATION ${DEST_INCLUDE_DIR}/ThirdParty/LibCpuId FILES_MATCHING PATTERN *.h BUILD_TREE_ONLY)  # Note: the trailing slash is significant
diff --git a/Source/ThirdParty/Lua/CMakeLists.txt b/Source/ThirdParty/Lua/CMakeLists.txt
index a24c75025..f62491fa7 100644
--- a/Source/ThirdParty/Lua/CMakeLists.txt
+++ b/Source/ThirdParty/Lua/CMakeLists.txt
@@ -34,7 +34,7 @@ if (READLINE_FOUND)
 endif ()
 
 # Setup target
-setup_library ()
+setup_library (STATIC)
 
 # Install headers for building and using the Urho3D library (no direct dependencies but library user may need them)
 install_header_files (DIRECTORY src/ DESTINATION ${DEST_INCLUDE_DIR}/ThirdParty/Lua FILES_MATCHING PATTERN *.h)  # Note: the trailing slash is significant
diff --git a/Source/ThirdParty/LuaJIT/CMakeLists.txt b/Source/ThirdParty/LuaJIT/CMakeLists.txt
index c86cd0755..f1ec9250d 100644
--- a/Source/ThirdParty/LuaJIT/CMakeLists.txt
+++ b/Source/ThirdParty/LuaJIT/CMakeLists.txt
@@ -470,7 +470,7 @@ if (XCODE)
         endif ()
     endif ()
 endif ()
-setup_library (${EXCLUDE_FROM_ALL})
+setup_library (STATIC ${EXCLUDE_FROM_ALL})
 if (XCODE AND NOT ARCH)    # These variables are used to control the recursion as this script is recursive in nature on Xcode
     # Add external and custom targets to build Mach-O universal binary LuaJIT sub-library
     include (ExternalProject)
diff --git a/Source/ThirdParty/MojoShader/CMakeLists.txt b/Source/ThirdParty/MojoShader/CMakeLists.txt
index db48047ce..82394e70d 100644
--- a/Source/ThirdParty/MojoShader/CMakeLists.txt
+++ b/Source/ThirdParty/MojoShader/CMakeLists.txt
@@ -31,7 +31,7 @@ if (MSVC)
 endif (MSVC)
 
 # Setup target
-setup_library ()
+setup_library (STATIC)
 
 # Install headers for building the Urho3D library
 install_header_files (DIRECTORY ./ DESTINATION ${DEST_INCLUDE_DIR}/ThirdParty/MojoShader FILES_MATCHING PATTERN *.h BUILD_TREE_ONLY)  # Note: the trailing slash is significant
diff --git a/Source/ThirdParty/PugiXml/CMakeLists.txt b/Source/ThirdParty/PugiXml/CMakeLists.txt
index 1298ae9af..ccddcd965 100644
--- a/Source/ThirdParty/PugiXml/CMakeLists.txt
+++ b/Source/ThirdParty/PugiXml/CMakeLists.txt
@@ -27,7 +27,7 @@ set (TARGET_NAME PugiXml)
 define_source_files (GLOB_CPP_PATTERNS src/*.cpp GLOB_H_PATTERNS src/*.hpp)
 
 # Setup target
-setup_library ()
+setup_library (STATIC)
 
 # Install headers for building the Urho3D library
 install_header_files (DIRECTORY src/ DESTINATION ${DEST_INCLUDE_DIR}/ThirdParty/PugiXml FILES_MATCHING PATTERN *.hpp BUILD_TREE_ONLY)  # Note: the trailing slash is significant
diff --git a/Source/ThirdParty/Recast/CMakeLists.txt b/Source/ThirdParty/Recast/CMakeLists.txt
index 618f28a51..643278164 100644
--- a/Source/ThirdParty/Recast/CMakeLists.txt
+++ b/Source/ThirdParty/Recast/CMakeLists.txt
@@ -30,7 +30,7 @@ define_source_files (GLOB_CPP_PATTERNS Source/*.cpp GLOB_H_PATTERNS Include/*.h)
 set (INCLUDE_DIRS Include)
 
 # Setup target
-setup_library ()
+setup_library (STATIC)
 
 # Install headers for building the Urho3D library
 install_header_files (DIRECTORY Include/ DESTINATION ${DEST_INCLUDE_DIR}/ThirdParty/Recast FILES_MATCHING PATTERN *.h BUILD_TREE_ONLY)  # Note: the trailing slash is significant
diff --git a/Source/ThirdParty/SDL/CMakeLists.txt b/Source/ThirdParty/SDL/CMakeLists.txt
index 5c4ef27da..9f48ca53c 100644
--- a/Source/ThirdParty/SDL/CMakeLists.txt
+++ b/Source/ThirdParty/SDL/CMakeLists.txt
@@ -1600,7 +1600,7 @@ file (GLOB H_FILES include/*.h)     # Adding the headers into source files list
 list (APPEND SOURCE_FILES ${H_FILES})
 
 # Setup target as STATIC library (as the result we never use EXTRA_LDFLAGS linker flags)
-setup_library ()
+setup_library (STATIC)
 
 # Install headers for building and using the Urho3D library (install dependency for InputEvents.h)
 install_header_files (DIRECTORY include/ DESTINATION ${DEST_INCLUDE_DIR}/ThirdParty/SDL FILES_MATCHING PATTERN *.h USE_FILE_SYMLINK)    # Note: the trailing slash is significant
diff --git a/Source/ThirdParty/SLikeNet/CMakeLists.txt b/Source/ThirdParty/SLikeNet/CMakeLists.txt
index 0fef2defa..c2bfaabc9 100644
--- a/Source/ThirdParty/SLikeNet/CMakeLists.txt
+++ b/Source/ThirdParty/SLikeNet/CMakeLists.txt
@@ -62,7 +62,7 @@ set (SOURCE_FILES ${CPP_FILES} ${H_FILES})
 set (INCLUDE_DIRS Source/include)
 
 # Setup target
-setup_library ()
+setup_library (STATIC)
 
 # Install headers for building the Urho3D library
 install_header_files (DIRECTORY Source/include/slikenet/ DESTINATION ${DEST_INCLUDE_DIR}/ThirdParty/SLikeNet FILES_MATCHING PATTERN *.h BUILD_TREE_ONLY)    # Note: the trailing slash is significant
diff --git a/Source/ThirdParty/SQLite/CMakeLists.txt b/Source/ThirdParty/SQLite/CMakeLists.txt
index 0c42972b0..0cd33bfd3 100644
--- a/Source/ThirdParty/SQLite/CMakeLists.txt
+++ b/Source/ThirdParty/SQLite/CMakeLists.txt
@@ -42,7 +42,7 @@ endforeach ()
 set (SOURCE_FILES src/sqlite3.c)
 
 # Setup target
-setup_library ()
+setup_library (STATIC)
 
 # Install headers for building and using the Urho3D library
 install_header_files (DIRECTORY src/ DESTINATION ${DEST_INCLUDE_DIR}/ThirdParty/SQLite FILES_MATCHING PATTERN *.h)  # Note: the trailing slash is significant
diff --git a/Source/ThirdParty/StanHull/CMakeLists.txt b/Source/ThirdParty/StanHull/CMakeLists.txt
index faf4e3858..34362e419 100644
--- a/Source/ThirdParty/StanHull/CMakeLists.txt
+++ b/Source/ThirdParty/StanHull/CMakeLists.txt
@@ -27,7 +27,7 @@ set (TARGET_NAME StanHull)
 define_source_files ()
 
 # Setup target
-setup_library ()
+setup_library (STATIC)
 
 # Install headers for building the Urho3D library
 install_header_files (DIRECTORY ./ DESTINATION ${DEST_INCLUDE_DIR}/ThirdParty/StanHull FILES_MATCHING PATTERN *.h BUILD_TREE_ONLY)  # Note: the trailing slash is significant
diff --git a/Source/ThirdParty/WebP/CMakeLists.txt b/Source/ThirdParty/WebP/CMakeLists.txt
index 7072f4663..54737cc82 100644
--- a/Source/ThirdParty/WebP/CMakeLists.txt
+++ b/Source/ThirdParty/WebP/CMakeLists.txt
@@ -32,7 +32,7 @@ if (ANDROID)
 endif ()
 
 # Setup target
-setup_library ()
+setup_library (STATIC)
 
 # Install headers for building the Urho3D library
 install_header_files (DIRECTORY src/webp/ DESTINATION ${DEST_INCLUDE_DIR}/ThirdParty/webp FILES_MATCHING PATTERN *.h BUILD_TREE_ONLY)  # Note: the trailing slash is significant
diff --git a/Source/ThirdParty/ik/CMakeLists.txt b/Source/ThirdParty/ik/CMakeLists.txt
index 6868c6d88..57290a821 100644
--- a/Source/ThirdParty/ik/CMakeLists.txt
+++ b/Source/ThirdParty/ik/CMakeLists.txt
@@ -75,7 +75,7 @@ configure_file (include/ik/config.h.in ${CMAKE_CURRENT_BINARY_DIR}/include/gener
 set (INCLUDE_DIRS ${CMAKE_CURRENT_BINARY_DIR}/include/generated include)
 
 # Setup target
-setup_library ()
+setup_library (STATIC)
 
 # Install headers for building the Urho3D library
 install_header_files (DIRECTORY include/ik/ DESTINATION ${DEST_INCLUDE_DIR}/ThirdParty/ik FILES_MATCHING PATTERN *.h USE_FILE_SYMLINK BUILD_TREE_ONLY)  # Note: the trailing slash is significant
diff --git a/Source/ThirdParty/nanodbc/CMakeLists.txt b/Source/ThirdParty/nanodbc/CMakeLists.txt
index ca59b6acf..628a27d2f 100644
--- a/Source/ThirdParty/nanodbc/CMakeLists.txt
+++ b/Source/ThirdParty/nanodbc/CMakeLists.txt
@@ -36,7 +36,7 @@ if (ODBC_DEFINES)
 endif ()
 
 # Setup target
-setup_library ()
+setup_library (STATIC)
 
 # Install headers for building and using the Urho3D library
 install_header_files (DIRECTORY nanodbc/ DESTINATION ${DEST_INCLUDE_DIR}/ThirdParty/nanodbc FILES_MATCHING PATTERN *.h)  # Note: the trailing slash is significant
diff --git a/Source/ThirdParty/toluapp/src/lib/CMakeLists.txt b/Source/ThirdParty/toluapp/src/lib/CMakeLists.txt
index c84ed5c01..5a26bf2b5 100644
--- a/Source/ThirdParty/toluapp/src/lib/CMakeLists.txt
+++ b/Source/ThirdParty/toluapp/src/lib/CMakeLists.txt
@@ -30,7 +30,7 @@ define_source_files (GLOB_CPP_PATTERNS *.c)
 set (INCLUDE_DIRS ../../include ../../../Lua${JIT}/src)
 
 # Setup target
-setup_library ()
+setup_library (STATIC)
 
 # Install headers for building and using the Urho3D library (no direct dependencies but library user may need them)
 install_header_files (DIRECTORY ../../include/ DESTINATION ${DEST_INCLUDE_DIR}/ThirdParty/toluapp FILES_MATCHING PATTERN *.h)  # Note: the trailing slash is significant
