diff --git a/CMakeLists.txt b/CMakeLists.txt
index ae66913..79a1c7d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -130,7 +130,7 @@ install(TARGETS libgig EXPORT libgig-config
 	PUBLIC_HEADER DESTINATION include/libgig)
 install(EXPORT libgig-config NAMESPACE libgig:: DESTINATION share/libgig)
 
-install(TARGETS libakai EXPORT libakai-config
+install(TARGETS libakai EXPORT libgig-config
 	LIBRARY DESTINATION lib
 	ARCHIVE DESTINATION lib
 	RUNTIME DESTINATION bin
