diff --git a/src/json-validator.cpp b/src/json-validator.cpp
index f16f3ef..1f7b615 100644
--- a/src/json-validator.cpp
+++ b/src/json-validator.cpp
@@ -663,7 +663,7 @@ public:
 #ifndef NO_STD_REGEX
 		attr = sch.find("pattern");
 		if (attr != sch.end()) {
-			patternString_ = attr.value();
+			patternString_ = attr.value().get<std::string>();
 			pattern_ = {true, REGEX_NAMESPACE::regex(attr.value().get<std::string>(),
 			                                         REGEX_NAMESPACE::regex::ECMAScript)};
 			sch.erase(attr);
