diff --git a/google/cloud/storage/parallel_upload.h b/google/cloud/storage/parallel_upload.h
index 44f542d..4fd97bd 100644
--- a/google/cloud/storage/parallel_upload.h
+++ b/google/cloud/storage/parallel_upload.h
@@ -515,7 +515,7 @@ StatusOr<std::vector<ParallelUploadFileShard>> CreateUploadShards(
 
   std::size_t const wanted_num_streams =
       (std::max<std::size_t>)(1,
-                              (std::min)(max_streams,
+                              (std::min<std::size_t>)(max_streams,
                                          div_ceil(file_size, min_stream_size)));
 
   std::uintmax_t const stream_size =
