diff --git a/glib/gwin32.h b/glib/gwin32.h
--- a/glib/gwin32.h
+++ b/glib/gwin32.h
@@ -45,7 +45,7 @@
 
 /* needed include for C++ builds; including this in C mode will cause havoc of type conflicts */
 #ifdef __cplusplus
-#include <unknwn.h>
+struct IUnknown;
 #endif
 
 /*
@@ -180,11 +180,11 @@
  * from the Windows SDK (albeit a C interface is provided for the mingw-w64 toolchain),
  * so we need to have a C++ version for this
  */
-template <typename com_interface>
+template <typename com_interface, TUnknown = IUnknown>
 static inline void
 g_win32_clear_com (com_interface **com_obj)
 {
-  IUnknown **unknown_com_obj = reinterpret_cast<IUnknown **>(com_obj);
+  TUnknown **unknown_com_obj = reinterpret_cast<TUnknown **>(com_obj);
 
   if (*unknown_com_obj != NULL)
     {
