diff --color -ur a/src/CMakeLists.txt b/src/CMakeLists.txt
--- a/src/CMakeLists.txt	2022-07-19 15:21:35.950966519 +0200
+++ b/src/CMakeLists.txt	2022-07-19 15:21:50.251002371 +0200
@@ -208,7 +208,7 @@
     PROPERTIES VERSION "${SIDX_LIB_VERSION}"
                SOVERSION "${SIDX_LIB_SOVERSION}" )
 
-if(WIN32)
+if(MSVC)
     target_compile_options(${SIDX_LIB_NAME} PRIVATE "/wd4068")
     target_compile_options(${SIDX_C_LIB_NAME} PRIVATE "/wd4068")
 
