diff --git a/CMakeLists.txt b/CMakeLists.txt
index dae5c58..1c7aa96 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,6 +20,7 @@ add_library(Boost::gil ALIAS boost_gil)
 
 target_include_directories(boost_gil INTERFACE include)
 
+# Boost-Exclude: Boost::filesystem
 target_link_libraries(boost_gil
   INTERFACE
     Boost::assert
@@ -27,7 +28,6 @@ target_link_libraries(boost_gil
     Boost::config
     Boost::container_hash
     Boost::core
-    Boost::filesystem
     Boost::integer
     Boost::iterator
     Boost::mp11
