diff --git a/include/usbmuxd-proto.h b/include/usbmuxd-proto.h
index a2a731e..38473f5 100644
--- a/include/usbmuxd-proto.h
+++ b/include/usbmuxd-proto.h
@@ -26,7 +26,7 @@
 #include <stdint.h>
 #define USBMUXD_PROTOCOL_VERSION 0
 
-#if defined(WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32) || defined(__CYGWIN__)
 #define USBMUXD_SOCKET_PORT 27015
 #else
 #define USBMUXD_SOCKET_FILE "/var/run/usbmuxd"
