diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 5905613..a5e6c91 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -16,10 +16,16 @@ if (NOT CMAKE_BUILD_TYPE)
 endif()
 
 # Import mimalloc (if installed)
-find_package(mimalloc 1.9 CONFIG REQUIRED)
+find_package(mimalloc 3.0 CONFIG REQUIRED)
 message(STATUS "Found mimalloc installed at: ${MIMALLOC_LIBRARY_DIR} (${MIMALLOC_VERSION_DIR})")
 
+if(WIN32)
+  add_library(mimalloc-test-override-dep SHARED main-override-dep.cpp)
+else()
+  add_library(mimalloc-test-override-dep INTERFACE)
+endif()
 
+if(BUILD_SHARED_LIBS)
 # link with a dynamic shared library
 # use `LD_PRELOAD` to actually override malloc/free at runtime with mimalloc
 add_executable(dynamic-override  main-override.c)
@@ -27,15 +33,20 @@ target_link_libraries(dynamic-override PUBLIC mimalloc)
 
 add_executable(dynamic-override-cxx  main-override.cpp)
 target_link_libraries(dynamic-override-cxx PUBLIC mimalloc)
+target_link_libraries(dynamic-override-cxx PRIVATE mimalloc-test-override-dep)
+endif()
 
 
+if(0)
 # overriding with a static object file works reliable as the symbols in the
 # object file have priority over those in library files
 add_executable(static-override-obj main-override.c ${MIMALLOC_OBJECT_DIR}/mimalloc${CMAKE_C_OUTPUT_EXTENSION})
 target_include_directories(static-override-obj PUBLIC ${MIMALLOC_INCLUDE_DIR})
 target_link_libraries(static-override-obj PUBLIC mimalloc-static)
+endif()
 
 
+if(NOT BUILD_SHARED_LIBS)
 # overriding with a static library works too if using the `mimalloc-override.h`
 # header to redefine malloc/free. (the library already overrides new/delete)
 add_executable(static-override-static main-override-static.c)
@@ -49,7 +60,10 @@ target_link_libraries(static-override PUBLIC mimalloc-static)
 
 add_executable(static-override-cxx  main-override.cpp)
 target_link_libraries(static-override-cxx PUBLIC mimalloc-static)
+target_link_libraries(static-override-cxx PRIVATE mimalloc-test-override-dep)
+endif()
 
+include("${VCPKG_TESTS}")
 
 ## test memory errors
 add_executable(test-wrong  test-wrong.c)
