diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index c7a504b..40157c7 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -16,9 +16,14 @@ if (NOT CMAKE_BUILD_TYPE)
 endif()
 
 # Import mimalloc (if installed)
-find_package(mimalloc 1.9 CONFIG REQUIRED)
+find_package(mimalloc 3.1 CONFIG REQUIRED)
 message(STATUS "Found mimalloc installed at: ${MIMALLOC_LIBRARY_DIR} (${MIMALLOC_VERSION_DIR})")
 
+if(WIN32)
+  add_library(mimalloc-test-override-dep SHARED main-override-dep.cpp)
+else()
+  add_library(mimalloc-test-override-dep INTERFACE)
+endif()
 
 if(BUILD_SHARED_LIBS)
 # link with a dynamic shared library
@@ -28,6 +33,7 @@ target_link_libraries(dynamic-override PUBLIC mimalloc)
 
 add_executable(dynamic-override-cxx  main-override.cpp)
 target_link_libraries(dynamic-override-cxx PUBLIC mimalloc)
+target_link_libraries(dynamic-override-cxx PRIVATE mimalloc-test-override-dep)
 endif()
 
 
@@ -54,8 +60,10 @@ target_link_libraries(static-override PUBLIC mimalloc-static)
 
 add_executable(static-override-cxx  main-override.cpp)
 target_link_libraries(static-override-cxx PUBLIC mimalloc-static)
+target_link_libraries(static-override-cxx PRIVATE mimalloc-test-override-dep)
 endif()
 
+include("${VCPKG_TESTS}")
 
 ## test memory errors
 add_executable(test-wrong  test-wrong.c)
