--- a/lib/ome/common/xml/Platform.h
+++ b/lib/ome/common/xml/Platform.h
@@ -40,7 +40,7 @@
 #define OME_COMMON_XML_PLATFORM_H
 
 #include <mutex>
-
+#include <ome_common_export.h>
 #include <xercesc/util/PlatformUtils.hpp>
 
 namespace ome
@@ -93,7 +93,7 @@
         }
 
 	/// Mutex to lock libxerces access.
-        static std::mutex mutex;
+        static OME_COMMON_EXPORT std::mutex mutex;
       };
 
     }
--- a/lib/ome/common/xml/Platform.cpp
+++ b/lib/ome/common/xml/Platform.cpp
@@ -45,7 +45,7 @@
     namespace xml
     {
 
-      std::mutex Platform::mutex;
+      std::mutex OME_COMMON_EXPORT Platform::mutex;
 
     }
   }
--- a/lib/ome/common/xsl/Platform.cpp
+++ a/lib/ome/common/xsl/Platform.cpp
@@ -45,9 +45,9 @@
     namespace xsl
     {
 
-      std::mutex Platform::mutex;
+      std::mutex OME_COMMON_EXPORT Platform::mutex;
 
-      uint32_t Platform::refcount = 0;
+      uint32_t OME_COMMON_EXPORT Platform::refcount = 0;
 
     }
   }
--- a/lib/ome/common/xsl/Platform.h
+++ b/lib/ome/common/xsl/Platform.h
@@ -43,6 +43,7 @@
 
 #include <mutex>
 
+#include <ome_common_export.h>
 #include <ome/common/xml/Platform.h>
 
 #include <xalanc/XalanTransformer/XalanTransformer.hpp>
@@ -118,9 +119,9 @@
         /// Skip initialize and terminate calls.
         bool skip;
         /// Mutex to lock libxalan access.
-        static std::mutex mutex;
+        static OME_COMMON_EXPORT std::mutex mutex;
         /// Reference count.
-        static uint32_t refcount;
+        static OME_COMMON_EXPORT uint32_t refcount;
       };
 
     }
 