diff --git a/CMake/dcmtkPrepare.cmake b/CMake/dcmtkPrepare.cmake
index e9f77b5..4f7ebe3 100644
--- a/CMake/dcmtkPrepare.cmake
+++ b/CMake/dcmtkPrepare.cmake
@@ -192,9 +192,9 @@ include(GNUInstallDirs)
 # CMake's files (DCMTKTarget.cmake, DCMTKConfigVersion.cmake and DCMTKConfig.cmake) are installed
 # to different installation paths under Unix- and Windows-based systems
 if(UNIX)
-  set(DCMTK_INSTALL_CMKDIR "${CMAKE_INSTALL_LIBDIR}/cmake/dcmtk")
+  SET(DCMTK_INSTALL_CMKDIR "share/dcmtk")
 elseif(WIN32)
-  set(DCMTK_INSTALL_CMKDIR "cmake")
+  SET(DCMTK_INSTALL_CMKDIR "share/dcmtk")
 endif()
 
 #-----------------------------------------------------------------------------
