diff --git a/CMake/3rdparty.cmake b/CMake/3rdparty.cmake
index 2c974e8..a6a5d65 100644
--- a/CMake/3rdparty.cmake
+++ b/CMake/3rdparty.cmake
@@ -86,7 +86,7 @@ if(DCMTK_USE_FIND_PACKAGE)
 
   # Find libXML2
   if(DCMTK_WITH_XML)
-    find_package(LibXml2 QUIET)
+    find_package(LibXml2 REQUIRED)
     if(NOT LIBXML2_FOUND)
       message(STATUS "Warning: XML support will be disabled because libxml2 was not found.")
       set(WITH_LIBXML "")
@@ -95,7 +95,7 @@ if(DCMTK_USE_FIND_PACKAGE)
       message(STATUS "Info: DCMTK XML support will be enabled")
       set(WITH_LIBXML 1)
       include_directories(${LIBXML2_INCLUDE_DIR})
-      set(LIBXML_LIBS ${LIBXML2_LIBRARIES} ${LIBXML2_EXTRA_LIBS_STATIC})
+      set(LIBXML_LIBS LibXml2::LibXml2 ${LIBXML2_EXTRA_LIBS_STATIC})
     endif()
   endif()
 
