diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4571a0e..2263378 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -369,6 +369,10 @@ else()
   message(STATUS "nanoflann was not found, so some PCL classes will not be available.")
 endif()
 
+if(UNIX AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 6.0)
+    set (CMAKE_CXX_STANDARD 17)
+endif()
+
 # libusb
 option(WITH_LIBUSB "Build USB RGBD-Camera drivers" TRUE)
 if(WITH_LIBUSB)
