From b27fdca720eb258b79c46e60c91f606ebd709ebd Mon Sep 17 00:00:00 2001
Date: Wed, 7 Jul 2021 11:55:29 -0400
Subject: [PATCH] numeric literals

---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 99ddd2cc9..50124a298 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -423,6 +423,9 @@ endif()
 
 # Boost (required)
 include("${PCL_SOURCE_DIR}/cmake/pcl_find_boost.cmake")
+if (NOT APPLE AND NOT WIN32 AND CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    add_definitions(-fext-numeric-literals)
+endif()
 
 ### ---[ Create the config.h file
 set(pcl_config_h_in "${CMAKE_CURRENT_SOURCE_DIR}/pcl_config.h.in")
-- 
2.32.0.windows.1

