diff --git a/cmake/pcl_utils.cmake b/cmake/pcl_utils.cmake
index 55b0820f2..034fbc227 100644
--- a/cmake/pcl_utils.cmake
+++ b/cmake/pcl_utils.cmake
@@ -109,7 +109,7 @@ macro(SET_INSTALL_DIRS)
     set(BIN_INSTALL_DIR "bin")
     set(PKGCFG_INSTALL_DIR "${LIB_INSTALL_DIR}/pkgconfig")
     if(WIN32 AND NOT MINGW)
-        set(PCLCONFIG_INSTALL_DIR "cmake")
+        set(PCLCONFIG_INSTALL_DIR "share/pcl")
       else(WIN32 AND NOT MINGW)
         set(PCLCONFIG_INSTALL_DIR "share/${PROJECT_NAME_LOWER}-${PCL_MAJOR_VERSION}.${PCL_MINOR_VERSION}")
       endif(WIN32 AND NOT MINGW)
