diff --git a/cuda/common/CMakeLists.txt b/cuda/common/CMakeLists.txt
index b5775af..e7a2ff2 100644
--- a/cuda/common/CMakeLists.txt
+++ b/cuda/common/CMakeLists.txt
@@ -29,10 +29,10 @@ set(common_incs
     )
 
 include_directories(./include)
-#set(LIB_NAME pcl_${SUBSYS_NAME})
+set(LIB_NAME "pcl_${SUBSYS_NAME}")
 set(EXT_DEPS CUDA)
-#PCL_MAKE_PKGCONFIG(${LIB_NAME} ${SUBSYS_NAME} "${SUBSYS_DESC}"
-#    "${SUBSYS_DEPS}" "${EXT_DEPS}" "" "" "")
+PCL_MAKE_PKGCONFIG(${LIB_NAME} COMPONENT ${SUBSYS_NAME} DESC "${SUBSYS_DESC}"
+    PCL_DEPS "${SUBSYS_DEPS}" EXT_DEPS "" HEADER_ONLY)
 
 # Install include files
 PCL_ADD_INCLUDES(${SUBSYS_NAME} "cuda" ${incs})
