diff --git a/common/include/pcl/impl/instantiate.hpp b/common/include/pcl/impl/instantiate.hpp
index ea73a0c..4cfc92b 100644
--- a/common/include/pcl/impl/instantiate.hpp
+++ b/common/include/pcl/impl/instantiate.hpp
@@ -95,7 +95,7 @@
 //
 //    ((x)(y)(z))((1)(2)(3))((dracula)(radu))
 //
-#ifdef _MSC_VER
+#if defined(_MSC_VER) && ! defined(__clang__)
 #define PCL_INSTANTIATE_PRODUCT_IMPL(r, product) \
   BOOST_PP_CAT(PCL_INSTANTIATE_, BOOST_PP_SEQ_HEAD(product)) \
           BOOST_PP_EXPAND(BOOST_PP_SEQ_TO_TUPLE(BOOST_PP_SEQ_TAIL(product))) 
diff --git a/common/src/colors.cpp b/common/src/colors.cpp
index 4d53ab1..ba0c114 100644
--- a/common/src/colors.cpp
+++ b/common/src/colors.cpp
@@ -39,6 +39,7 @@
 #include <pcl/common/colors.h>
 
 #include <array>
+#include <cassert>
 
 /// Glasbey lookup table
 static constexpr std::array<unsigned char, 256 * 3> GLASBEY_LUT =
diff --git a/common/src/gaussian.cpp b/common/src/gaussian.cpp
index 69bbd12..2b6a667 100644
--- a/common/src/gaussian.cpp
+++ b/common/src/gaussian.cpp
@@ -36,6 +36,7 @@
  */
 
 #include <pcl/common/gaussian.h>
+#include <cassert>
 
 void 
 pcl::GaussianKernel::compute (float sigma, 
diff --git a/common/src/range_image_planar.cpp b/common/src/range_image_planar.cpp
index 97b249b..dddd7f7 100644
--- a/common/src/range_image_planar.cpp
+++ b/common/src/range_image_planar.cpp
@@ -35,6 +35,7 @@
 /** \author Bastian Steder */
 
 #include <iostream>
+#include <cassert>
 using std::cout;
 using std::cerr;
 
diff --git a/CMakeLists.txt b/CMakeLists.txt
index fad95c9..b4ca305 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,7 +51,7 @@ set(CMAKE_BUILD_TYPE "${CMAKE_BUILD_TYPE}" CACHE STRING
 # Compiler identification
 # Define a variable CMAKE_COMPILER_IS_X where X is the compiler short name.
 # Note: CMake automatically defines one for GNUCXX, nothing to do in this case.
-if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND NOT MSVC)
   set(CMAKE_COMPILER_IS_CLANG 1)
 elseif(__COMPILER_PATHSCALE)
   set(CMAKE_COMPILER_IS_PATHSCALE 1)
