diff --git a/io/CMakeLists.txt b/io/CMakeLists.txt
index d26e60421..893104b77 100644
--- a/io/CMakeLists.txt
+++ b/io/CMakeLists.txt
@@ -334,6 +334,10 @@ PCL_ADD_LIBRARY(${LIB_NAME} COMPONENT ${SUBSYS_NAME} SOURCES ${srcs} ${incs} ${c
 
 target_include_directories(${LIB_NAME} PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include")
 
+if(PNG_FOUND)
+  target_link_libraries("${LIB_NAME}" ${PNG_LIBRARIES})
+endif()
+
 target_link_libraries("${LIB_NAME}" pcl_common pcl_io_ply)
 if(VTK_FOUND)
   if(${VTK_VERSION} VERSION_LESS 9.0)
@@ -347,10 +351,6 @@ if(VTK_FOUND)
   endif()
 endif()
 
-if(PNG_FOUND)
-  target_link_libraries("${LIB_NAME}" ${PNG_LIBRARIES})
-endif()
-
 if(LIBUSB_FOUND)
   target_link_libraries("${LIB_NAME}" libusb::libusb)
 endif()
