diff --git a/core/CMakeLists.txt b/core/CMakeLists.txt
index 3483f9c..57e149b 100644
--- a/core/CMakeLists.txt
+++ b/core/CMakeLists.txt
@@ -100,7 +100,7 @@ endif()
 
 
 # coordinate systems
-if(VXL_BUILD_CORE_GEOMETRY AND VXL_BUILD_CORE_NUMERICS)
+if(VXL_BUILD_CORE_GEOMETRY AND VXL_BUILD_CORE_NUMERICS AND VXL_BUILD_CORE_IMAGING)
   add_subdirectory(vcsl)
   add_subdirectory(vpgl)
 endif()
diff --git a/core/vgl/vgl_clip.hxx b/core/vgl/vgl_clip.hxx
index 9ff64f8..10c986a 100644
--- a/core/vgl/vgl_clip.hxx
+++ b/core/vgl/vgl_clip.hxx
@@ -144,7 +144,7 @@ namespace {
 
 #elif HAS_CLIPPER
 
-#include <clipper.hxx>
+#include <clipper.hpp>
 
 namespace {
   //: Creates a Clipper polygon from a vgl_polygon.
