diff --git a/v3p/openjpeg2/CMakeLists.txt b/v3p/openjpeg2/CMakeLists.txt
index 67b184e..cdd7c0d 100644
--- a/v3p/openjpeg2/CMakeLists.txt
+++ b/v3p/openjpeg2/CMakeLists.txt
@@ -1,3 +1,8 @@
+if(NOT VXL_FORCE_V3P_OPENJPEG2)
+  set(OPENJPEG2_FOUND FALSE)
+  return()
+endif()
+
 project(OpenJPEG C)
 
 include(${VXL_CMAKE_DIR}/FindOpenJPEG2.cmake)
@@ -54,6 +54,7 @@ if(VXL_MATH_LIBRARY_FOUND)
 endif()
 mark_as_advanced(VXL_MATH_LIBRARY_FOUND)
 set_target_properties(openjpeg2 PROPERTIES
+  OUTPUT_NAME   vxl_openjpeg
   VERSION       2.0.0
   DEFINE_SYMBOL OPJ_EXPORTS)
 endif()
