diff --git a/CMakeLists.txt b/CMakeLists.txt
index 490cc19..51c0a92 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -388,6 +388,9 @@ if(USE_NGHTTP2)
   find_package(NGHTTP2 REQUIRED)
   include_directories(${NGHTTP2_INCLUDE_DIRS})
   list(APPEND CURL_LIBS ${NGHTTP2_LIBRARIES})
+  if(NOT BUILD_SHARED_LIBS)
+    add_definitions(-DNGHTTP2_STATICLIB)
+  endif()
 endif()

 if(NOT CURL_DISABLE_LDAP)
