diff --git a/CMakeLists.txt b/CMakeLists.txt
index 38b7b7d..73d323e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1150,7 +1150,11 @@ endfunction()
 
 include(GNUInstallDirs)
 
-set(CURL_INSTALL_CMAKE_DIR ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
+IF(WIN32)
+  set(CURL_INSTALL_CMAKE_DIR ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
+ELSE()
+  set(CURL_INSTALL_CMAKE_DIR ${CMAKE_INSTALL_LIBDIR}/../share/cmake/${PROJECT_NAME})
+ENDIF()
 set(TARGETS_EXPORT_NAME "${PROJECT_NAME}Targets")
 set(generated_dir "${CMAKE_CURRENT_BINARY_DIR}/generated")
 set(project_config "${generated_dir}/${PROJECT_NAME}Config.cmake")
diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index eca9a8a..fc8605e 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -80,6 +80,10 @@ if(NOT BUILD_SHARED_LIBS)
     set_target_properties(${LIB_NAME} PROPERTIES INTERFACE_COMPILE_DEFINITIONS CURL_STATICLIB)
 endif()
 
+string(TOUPPER ${CMAKE_SYSTEM_NAME} HOST_OS)
+if (${HOST_OS} STREQUAL "WINDOWSSTORE")
+  add_definitions(-D_WINSOCK_DEPRECATED_NO_WARNINGS)
+endif()
 target_link_libraries(${LIB_NAME} ${CURL_LIBS})
 
 if(WIN32)
