diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 0e567e7..697dfea 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -71,7 +71,7 @@ target_link_libraries(${EXE_NAME} libcurl ${CURL_LIBS} ZLIB::ZLIB)
 
 #INCLUDE(ModuleInstall OPTIONAL)
 
-install(TARGETS ${EXE_NAME} DESTINATION ${CMAKE_INSTALL_BINDIR})
+install(TARGETS ${EXE_NAME} DESTINATION tools)
 #export(TARGETS ${EXE_NAME}
 #       APPEND FILE ${PROJECT_BINARY_DIR}/curl-target.cmake
 #       NAMESPACE CURL::
