diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 5838805..d2ca468 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -103,7 +103,7 @@ target_link_libraries(${EXE_NAME} libcurl ${CURL_LIBS} ZLIB::ZLIB)
 
 #INCLUDE(ModuleInstall OPTIONAL)
 
-install(TARGETS ${EXE_NAME} DESTINATION ${CMAKE_INSTALL_BINDIR})
+install(TARGETS ${EXE_NAME} DESTINATION tools)
 #export(TARGETS ${EXE_NAME}
 #       APPEND FILE ${PROJECT_BINARY_DIR}/curl-target.cmake
 #       NAMESPACE ${PROJECT_NAME}::
