diff --git a/CMake/OtherTests.cmake b/CMake/OtherTests.cmake
index c1c9aa3..378a5bd 100644
--- a/CMake/OtherTests.cmake
+++ b/CMake/OtherTests.cmake
@@ -185,6 +185,7 @@ if(HAVE_SIGNAL_H)
   set(CMAKE_REQUIRED_FLAGS "-DHAVE_SIGNAL_H")
   set(CMAKE_EXTRA_INCLUDE_FILES "signal.h")
 endif()
+unset(CMAKE_TRY_COMPILE_TARGET_TYPE)
 check_type_size("sig_atomic_t" SIZEOF_SIG_ATOMIC_T)
 if(HAVE_SIZEOF_SIG_ATOMIC_T)
   check_c_source_compiles("
