diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0cba6f6..851efaf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2129,7 +2129,9 @@ if(NOT CURL_DISABLE_INSTALL)
     set(_implicit_libs ${CMAKE_C_IMPLICIT_LINK_LIBRARIES})
   endif()
 
-  foreach(_lib IN LISTS _implicit_libs _custom_libs CURL_LIBS)
+  set(CURL_LIBS_FLAT "")
+  vcpkg_curl_flatten(CURL_LIBS CURL_LIBS_FLAT)
+  foreach(_lib IN LISTS CURL_LIBS_FLAT)
     if(TARGET "${_lib}")
       set(_libname "${_lib}")
       get_target_property(_imported "${_libname}" IMPORTED)
