diff --git a/CMakeLists.txt b/CMakeLists.txt
index da69699..3a9b844 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2129,7 +2129,7 @@ if(NOT CURL_DISABLE_INSTALL)
     set(_implicit_libs ${CMAKE_C_IMPLICIT_LINK_LIBRARIES})
   endif()
 
-  foreach(_lib IN LISTS _implicit_libs _custom_libs CURL_LIBS)
+  foreach(_lib IN LISTS                _custom_libs          )
     if(TARGET "${_lib}")
       set(_libname "${_lib}")
       get_target_property(_imported "${_libname}" IMPORTED)
@@ -2190,9 +2190,6 @@ if(NOT CURL_DISABLE_INSTALL)
     set(LIBCURL_PC_CFLAGS   "")
   else()
     set(ENABLE_SHARED       "no")
-    set(LIBCURL_PC_REQUIRES "${LIBCURL_PC_REQUIRES_PRIVATE}")
-    set(LIBCURL_PC_LIBS     "${LIBCURL_PC_LIBS_PRIVATE}")
-    set(LIBCURL_PC_CFLAGS   "${LIBCURL_PC_CFLAGS_PRIVATE}")
   endif()
   if(BUILD_STATIC_LIBS)
     set(ENABLE_STATIC       "yes")
