diff --git a/CMakeLists.txt b/CMakeLists.txt
index 580cc4357..735a9e234 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -540,11 +540,12 @@ if(CURL_USE_WOLFSSL)
 endif()
 
 if(CURL_USE_GNUTLS)
-  find_package(GnuTLS REQUIRED)
-  find_package(nettle REQUIRED)
+  find_package(PkgConfig REQUIRED)
+  pkg_check_modules(GNUTLS "gnutls")
+  pkg_check_modules(NETTLE "nettle")
   set(SSL_ENABLED ON)
   set(USE_GNUTLS ON)
-  list(APPEND CURL_LIBS ${GNUTLS_LIBRARIES} ${NETTLE_LIBRARIES})
+  list(APPEND CURL_LIBS ${GNUTLS_LINK_LIBRARIES} ${NETTLE_LINK_LIBRARIES})
   list(APPEND LIBCURL_PC_REQUIRES_PRIVATE "gnutls" "nettle")
   include_directories(${GNUTLS_INCLUDE_DIRS} ${NETTLE_INCLUDE_DIRS})
 
