diff --git a/CMakeLists.txt b/CMakeLists.txt
index c4ba74d..0e8d45f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -59,7 +59,7 @@ foreach(_cache_var ${_cache_vars})
     else()
       set(_cache_var_type ":${_cache_var_type}")
     endif()
-    set(_cmake_args "${_cmake_args} -D${_cache_var}${_cache_var_type}=\"${${_cache_var}}\"")
+    set(_cmake_args "${_cmake_args} -D${_cache_var}${_cache_var_type}=\"<redacted>\"")
   endif()
 endforeach()
 
