diff --git a/CMakeLists.txt b/CMakeLists.txt
index 53e4598..bf7177b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -60,7 +60,7 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND
 endif()
 
 hunter_add_package(thrift)
-find_package(thrift ${hunter_config} REQUIRED)
+find_package(Thrift CONFIG REQUIRED)
 if(HUNTER_ENABLED)
   list(APPEND LIBS thrift::thrift_static)
 else()
@@ -75,10 +75,10 @@ hunter_add_package(opentracing-cpp)
 find_package(OpenTracing CONFIG REQUIRED)
 # Under Windows, link dynamically with OpenTracing
 if (WIN32)
-  list(APPEND LIBS OpenTracing::opentracing)
+  list(APPEND LIBS $<IF:$<TARGET_EXISTS:OpenTracing::opentracing>,OpenTracing::opentracing,OpenTracing::opentracing-static>)
   set(OPENTRACING_LIB OpenTracing::opentracing)
 else()
-  list(APPEND LIBS OpenTracing::opentracing-static)
+  list(APPEND LIBS $<IF:$<TARGET_EXISTS:OpenTracing::opentracing>,OpenTracing::opentracing,OpenTracing::opentracing-static>)
   set(OPENTRACING_LIB OpenTracing::opentracing-static)
 endif()
 list(APPEND package_deps OpenTracing)
@@ -273,6 +273,9 @@ if(JAEGERTRACING_PLUGIN)
   file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/export.map
     "{ global: OpenTracingMakeTracerFactory; local: *; };")
   add_library(jaegertracing_plugin MODULE ${SRC})
+  if (WIN32)
+    target_link_libraries(jaegertracing_plugin PUBLIC Iphlpapi Ws2_32)
+  endif()
   add_lib_deps(jaegertracing_plugin)
   target_link_libraries(jaegertracing_plugin PUBLIC
     -static-libgcc
@@ -418,7 +421,7 @@ include(GNUInstallDirs)
 #   * <prefix>/lib*/cmake/<PROJECT-NAME>
 #   * <prefix>/lib*/
 #   * <prefix>/include/
-set(config_install_dir "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}")
+set(config_install_dir "${CMAKE_INSTALL_DATAROOTDIR}/jaeger-client-cpp")
 
 set(generated_dir "${CMAKE_CURRENT_BINARY_DIR}/generated")
 
