diff --git a/src/libtriton/CMakeLists.txt b/src/libtriton/CMakeLists.txt
index 7a62a92..778d801 100644
--- a/src/libtriton/CMakeLists.txt
+++ b/src/libtriton/CMakeLists.txt
@@ -275,7 +275,7 @@ if(MSVC AND MSVC_STATIC)
 endif()
 
 if(WIN32 AND BUILD_SHARED_LIBS)
-    target_compile_definitions(triton PUBLIC TRITON_BUILDING_DLL)
+    target_compile_definitions(triton PRIVATE BUILDING_DLL)
 endif()
 
 # Install tritonTargets.cmake
@@ -289,6 +289,7 @@ install(
 install(
     TARGETS triton
     EXPORT tritonTargets
+    RUNTIME DESTINATION bin
     PUBLIC_HEADER DESTINATION include/triton
     INCLUDES DESTINATION include
     ARCHIVE DESTINATION lib
