diff --git a/src/3rdparty/chromium/third_party/blink/renderer/core/layout/svg/layout_svg_resource_paint_server.cc b/src/3rdparty/chromium/third_party/blink/renderer/core/layout/svg/layout_svg_resource_paint_server.cc
index 99ba54e790c..abf8467912d 100644
--- a/src/3rdparty/chromium/third_party/blink/renderer/core/layout/svg/layout_svg_resource_paint_server.cc
+++ b/src/3rdparty/chromium/third_party/blink/renderer/core/layout/svg/layout_svg_resource_paint_server.cc
@@ -94,7 +94,7 @@ void SVGPaintServer::PrependTransform(const AffineTransform& transform) {
   transform_ = transform * transform_;
 }

-static base::Optional<Color> ResolveColor(const ComputedStyle& style,
+static base::Optional<Color> ResolveColorSvgPaintServer(const ComputedStyle& style,
                                           const SVGPaint& paint,
                                           const SVGPaint& visited_paint) {
   if (!paint.HasColor())
@@ -122,7 +122,7 @@ static SVGPaintDescription RequestPaint(const LayoutObject& object,
   const SVGPaint& visited_paint = apply_to_fill
                                       ? svg_style.InternalVisitedFillPaint()
                                       : svg_style.InternalVisitedStrokePaint();
-  base::Optional<Color> color = ResolveColor(style, paint, visited_paint);
+  base::Optional<Color> color = ResolveColorSvgPaintServer(style, paint, visited_paint);

   if (paint.HasUrl()) {
     LayoutSVGResourcePaintServer* uri_resource = nullptr;
--
2.39.3
