diff --git a/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/base/circular_queue.h b/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/base/circular_queue.h
index 59dbfad04..134357395 100644
--- a/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/base/circular_queue.h
+++ b/src/3rdparty/chromium/third_party/perfetto/include/perfetto/ext/base/circular_queue.h
@@ -74,7 +74,7 @@ class CircularQueue {
       return queue_->Get(pos_);
     }
 
-    const T* operator->() const {
+    T* operator->() const {
 #if PERFETTO_DCHECK_IS_ON()
       PERFETTO_DCHECK(generation_ == queue_->generation());
 #endif
@@ -82,13 +82,13 @@ class CircularQueue {
     }
 
     T& operator*() { return *(operator->()); }
-    const T& operator*() const { return *(operator->()); }
+    T& operator*() const { return *(operator->()); }
 
     value_type& operator[](difference_type i) {
       return *(*this + i);
     }
 
-    const value_type& operator[](difference_type i) const {
+    value_type& operator[](difference_type i) const {
       return *(*this + i);
     }
 
