--- a/src/3rdparty/chromium/third_party/libxml/src/encoding.c
+++ b/src/3rdparty/chromium/third_party/libxml/src/encoding.c
@@ -25,6 +25,7 @@
 
 #include <string.h>
 #include <limits.h>
+#include <stdbool.h>
 
 #ifdef HAVE_CTYPE_H
 #include <ctype.h>
@@ -2004,7 +2005,7 @@ xmlEncOutputChunk(xmlCharEncodingHandler *handler, unsigned char *out,
 #ifdef LIBXML_ICU_ENABLED
     else if (handler->uconv_out != NULL) {
         ret = xmlUconvWrapper(handler->uconv_out, 0, out, outlen, in, inlen,
-                              TRUE);
+                              true);
     }
 #endif /* LIBXML_ICU_ENABLED */
     else {
