diff --git a/CMakeLists.txt b/CMakeLists.txt
index f478fb4..5a54eee 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -196,9 +196,9 @@ list_source_directories(LIBHV_SRCS ${LIBHV_SRCDIRS})
 if(BUILD_SHARED)
     add_library(hv SHARED ${LIBHV_SRCS})
     target_compile_definitions(hv PRIVATE HV_DYNAMICLIB)
-    target_include_directories(hv PRIVATE ${LIBHV_SRCDIRS})
+    target_include_directories(hv PRIVATE ${LIBHV_SRCDIRS} INTERFACE $<INSTALL_INTERFACE:include>)
     target_link_libraries(hv ${LIBS})
-    install(TARGETS hv
+    install(TARGETS hv EXPORT unofficial-hv-config
         ARCHIVE DESTINATION lib
         LIBRARY DESTINATION lib
         RUNTIME DESTINATION bin)
@@ -208,12 +208,18 @@ endif()
 if(BUILD_STATIC)
     add_library(hv_static STATIC ${LIBHV_SRCS})
     target_compile_definitions(hv_static PRIVATE HV_STATICLIB)
-    target_include_directories(hv_static PRIVATE ${LIBHV_SRCDIRS})
+    target_include_directories(hv_static PRIVATE ${LIBHV_SRCDIRS} INTERFACE $<INSTALL_INTERFACE:include>)
     target_link_libraries(hv_static ${LIBS})
-    install(TARGETS hv_static DESTINATION lib)
+    install(TARGETS hv_static EXPORT unofficial-hv-config DESTINATION lib)
     add_custom_target(libhv_static DEPENDS hv_static)
 endif()
 
+INSTALL(EXPORT unofficial-hv-config
+    NAMESPACE unofficial::hv::
+    DESTINATION share/unofficial-hv
+    FILE unofficial-hv-config.cmake
+)
+
 file(INSTALL ${LIBHV_HEADERS} DESTINATION include/hv)
 install(FILES ${LIBHV_HEADERS} DESTINATION include/hv)
 
