diff --git a/src/cgnstypes.h.in b/src/cgnstypes.h.in
index 98c4e7fdc..8c6bcea62 100644
--- a/src/cgnstypes.h.in
+++ b/src/cgnstypes.h.in
@@ -29,6 +29,11 @@
 #ifndef CGNSTYPES_H
 #define CGNSTYPES_H
 
+#define CG_BUILD_HDF5      @BUILDHDF5@
+#define CG_BUILD_FORTRAN   @BUILDFORTRAN@
+#define CG_BUILD_PARALLEL  @BUILDPARALLEL@
+#define CG_BUILD_BASESCOPE @BUILDBASESCOPE@
+
 #define CG_BUILD_LEGACY @BUILDLEGACY@
 #define CG_BUILD_64BIT  @BUILD64BIT@
 #define CG_BUILD_SCOPE  @BUILDSCOPE@
