diff -r 29939b8a25be -r 8a0a85e63950 CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,7 +9,7 @@
 # Find ignition-cmake
 #============================================================================
 # If you get an error at this line, you need to install ignition-cmake
-find_package(ignition-cmake0 REQUIRED)
+find_package(ignition-cmake0 0.4.1 REQUIRED)
 
 #============================================================================
 # Configure the project
diff -r 29939b8a25be -r 8a0a85e63950 src/CMakeLists.txt
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -52,7 +52,7 @@
     curl::curl
     ignition-common${IGN_COMMON_MAJOR_VER}::ignition-common${IGN_COMMON_MAJOR_VER}
     ${JSONCPP_TARGET}
-    YAML::YAML
+    ${YAML_TARGET}
     ZIP::ZIP
 )
 
