diff --git a/ompl/src/ompl/util/src/RandomNumbers.cpp b/ompl/src/ompl/util/src/RandomNumbers.cpp
index fe30070..048b42d 100644
--- a/ompl/src/ompl/util/src/RandomNumbers.cpp
+++ b/ompl/src/ompl/util/src/RandomNumbers.cpp
@@ -37,6 +37,7 @@
 #include "ompl/util/RandomNumbers.h"
 #include "ompl/util/Exception.h"
 #include "ompl/util/Console.h"
+#include <chrono>
 #include <mutex>
 #include <memory>
 #include <boost/math/constants/constants.hpp>
