diff --git a/src/native/corehost/nethost/CMakeLists.txt b/src/native/corehost/nethost/CMakeLists.txt
index a9f44e720..7149ea6c6 100644
--- a/src/native/corehost/nethost/CMakeLists.txt
+++ b/src/native/corehost/nethost/CMakeLists.txt
@@ -1,11 +1,14 @@
 # Licensed to the .NET Foundation under one or more agreements.
 # The .NET Foundation licenses this file to you under the MIT license.
 
+cmake_minimum_required (VERSION 3.14.5)
 project(nethost)
 
 set(DOTNET_PROJECT_NAME "nethost")
 
 # Include directories
+include(../../../../eng/native/configurepaths.cmake)
+include("${CLR_ENG_NATIVE_DIR}/configurecompiler.cmake")
 include_directories(../fxr)
 
 # CMake does not recommend using globbing since it messes with the freshness checks
@@ -20,7 +23,9 @@ if(CLR_CMAKE_TARGET_WIN32)
 endif()
 
 include(../lib.cmake)
+
 include(../lib_static.cmake)
+target_compile_definitions(libnethost PUBLIC NETHOST_USE_AS_STATIC)
 
 add_definitions(-DFEATURE_LIBHOST=1)
 add_definitions(-DNETHOST_EXPORT)
@@ -33,14 +38,29 @@ if (WIN32)
     )
 endif(WIN32)
 
-install(FILES ../coreclr_delegates.h DESTINATION corehost)
-install(FILES ../hostfxr.h DESTINATION corehost)
-install(FILES nethost.h DESTINATION corehost)
-install_with_stripped_symbols(nethost TARGETS corehost)
-
-# Only Windows creates a symbols file for static libs.
-if (WIN32)
-    install_with_stripped_symbols(libnethost TARGETS corehost)
+set(NETHOST_HEADERS nethost.h ../coreclr_delegates.h ../hostfxr.h)
+install(FILES ${NETHOST_HEADERS} DESTINATION include)
+# Original src/native/corehost/nethost/CMakeLists.txt uses install_with_stripped_symbols() to install
+# targets with symbols. However, on Windows symbols will be put into PDB
+# folder which doesn't seem to be recognizable by vcpkg functions.
+# Furthermore, install_with_stripped_symbols() may install files instead of
+# targets.
+# For these reason, they are replaced with a plain install().
+if (BUILD_SHARED_LIBS)
+    set(NETHOST_ACTIVE_TARGET nethost)
 else()
-    install(TARGETS libnethost DESTINATION corehost)
-endif(WIN32)
\ No newline at end of file
+    set(NETHOST_ACTIVE_TARGET libnethost)
+endif(BUILD_SHARED_LIBS)
+
+target_include_directories(${NETHOST_ACTIVE_TARGET} INTERFACE $<INSTALL_INTERFACE:include>)
+install(TARGETS ${NETHOST_ACTIVE_TARGET} EXPORT unofficial-nethost-config
+    ARCHIVE DESTINATION lib
+    LIBRARY DESTINATION lib
+    RUNTIME DESTINATION bin
+)
+
+install(EXPORT unofficial-nethost-config
+    DESTINATION share/unofficial-nethost
+    FILE unofficial-nethost-config.cmake
+    NAMESPACE unofficial::nethost::
+)
