From 2625b4a85b6cacf9b6ede3cf30af16eb795044e9 Mon Sep 17 00:00:00 2001
From: Hiroshi Miura <miurahr@linux.com>
Date: Sat, 24 Feb 2018 08:36:31 +0900
Subject: [PATCH 6/6] Fix utility link error

Signed-off-by: Hiroshi Miura <miurahr@linux.com>
---
 bin/CMakeLists.txt | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/bin/CMakeLists.txt b/bin/CMakeLists.txt
index 583ae36..6c6cc21 100644
--- a/bin/CMakeLists.txt
+++ b/bin/CMakeLists.txt
@@ -6,10 +6,6 @@
 #
 ###############################################################################
 
-INCLUDE_DIRECTORIES(
-    .
-    ${CMAKE_SOURCE_DIR})
-
 IF(WIN32 AND MSVC)
     SET(GETOPT_SOURCE getopt.c)
 ENDIF()
@@ -22,12 +18,11 @@ MESSAGE(STATUS "Adding GeoTIFF utilities to build")
 
 FOREACH(utility ${GEOTIFF_UTILITIES})
     ADD_EXECUTABLE(${utility} ${utility}.c ${GETOPT_SOURCE})
-    TARGET_LINK_LIBRARIES(${utility} xtiff ${GEOTIFF_LIBRARY_TARGET})
+    TARGET_LINK_LIBRARIES(${utility} ${GEOTIFF_LIBRARY_TARGET})
 ENDFOREACH()
 
 ADD_EXECUTABLE(geotifcp geotifcp.c ${GETOPT_SOURCE})
 TARGET_LINK_LIBRARIES(geotifcp
-    xtiff
     ${GEOTIFF_LIBRARY_TARGET}
     ${JPEG_LIBRARIES}
     ${ZLIB_LIBRARIES})
-- 
2.16.1

