diff --git a/libgeotiff/cmake/CMakeLists.txt b/libgeotiff/cmake/CMakeLists.txt
index 3809ba3..c92eb47 100644
--- a/libgeotiff/cmake/CMakeLists.txt
+++ b/libgeotiff/cmake/CMakeLists.txt
@@ -25,13 +25,13 @@ configure_file (project-config-version.cmake.in
 install (FILES
   "${CMAKE_CURRENT_BINARY_DIR}/project-config.cmake"
   DESTINATION "${INSTALL_CMAKE_DIR}"
-  RENAME "${PROJECT_NAME_LOWER}-config.cmake")
+  RENAME "${PROJECT_NAME}-config.cmake")
 install (FILES
   "${CMAKE_CURRENT_BINARY_DIR}/project-config-version.cmake"
   DESTINATION "${INSTALL_CMAKE_DIR}"
-  RENAME "${PROJECT_NAME_LOWER}-config-version.cmake")
+  RENAME "${PROJECT_NAME}-config-version.cmake")
 # Make information about the cmake targets (the library and the tools)
 # available.
 install (EXPORT depends
-  FILE ${PROJECT_NAME_LOWER}-depends.cmake
+  FILE ${PROJECT_NAME}-depends.cmake
   DESTINATION "${INSTALL_CMAKE_DIR}")
