diff --git a/cmake/project-config.cmake.in b/cmake/project-config.cmake.in
index 0dd376aca..0615081bd 100644
--- a/libgeotiff/cmake/project-config.cmake.in	
+++ b/libgeotiff/cmake/project-config.cmake.in
@@ -23,6 +23,12 @@ message (STATUS "Reading ${CMAKE_CURRENT_LIST_FILE}")
 message (STATUS
   "@PROJECT_NAME@ configuration, version ${@PROJECT_NAME@_VERSION}")
 
+#Find dependencies
+if(@WITH_PROJ4@)
+  include(CMakeFindDependencyMacro)
+  find_dependency(PROJ4)
+endif()
+
 # Tell the user project where to find our headers and libraries
 get_filename_component (_DIR ${CMAKE_CURRENT_LIST_FILE} PATH)
 get_filename_component (_ROOT "${_DIR}/@PROJECT_ROOT_DIR@" ABSOLUTE)
