diff --git a/CMake/macros.cmake b/CMake/macros.cmake
index e54d984..09e515f 100644
--- a/CMake/macros.cmake
+++ b/CMake/macros.cmake
@@ -97,6 +97,8 @@ macro(FL_ADD_LIBRARY LIBNAME LIBTYPE LIBFILES)
 		COMPILE_DEFINITIONS "FL_DLL"
 		)
 	endif (${LIBTYPE} STREQUAL "SHARED")
+
+	target_link_libraries(${LIBRARY_NAME} gdi32 gdiplus user32 advapi32 ole32 shell32 comdlg32)
     endif (MSVC)
 
     install(TARGETS ${LIBRARY_NAME}
