diff --git a/CMake/fl_add_library.cmake b/CMake/fl_add_library.cmake
index 6fa2fc5..e5ba493 100644
--- a/CMake/fl_add_library.cmake
+++ b/CMake/fl_add_library.cmake
@@ -38,6 +38,11 @@ macro (FL_ADD_LIBRARY LIBNAME LIBTYPE LIBFILES)
   add_library(${TARGET_NAME} ${LIBTYPE} ${LIBFILES})
   
   target_include_directories(${TARGET_NAME} PUBLIC $<INSTALL_INTERFACE:include>)
+  
+    if (APPLE)
+        find_library(Cocoa Cocoa)
+        target_link_libraries(${TARGET_NAME} $<$<PLATFORM_ID:Darwin>:${Cocoa}>)
+    endif()
 
   # target properties for all libraries
 
