diff --git a/CMake/fl_add_library.cmake b/CMake/fl_add_library.cmake
index c4e5f91c4..5bb4eafad 100644
--- a/CMake/fl_add_library.cmake
+++ b/CMake/fl_add_library.cmake
@@ -65,7 +65,7 @@ function(fl_add_library LIBNAME LIBTYPE SOURCES)
 
   set(OUTPUT_NAME_RELEASE "${LIBNAME}")
 
-  add_library(${TARGET_NAME} ${LIBTYPE} ${SOURCES})
+  target_include_directories(${TARGET_NAME} PUBLIC $<INSTALL_INTERFACE:include>)
 
   # Create an alias 'fltk::alias_name' for the library
   # where 'alias_name' is the library name without the prefix 'fltk_'
