diff --git a/glib/glibmm/varianttype.h b/glib/glibmm/varianttype.h
index a232e70..64eb4a1 100644
--- a/glib/glibmm/varianttype.h
+++ b/glib/glibmm/varianttype.h
@@ -500,54 +500,79 @@ public:
 
 };
 
+GLIBMM_API
 extern const VariantType VARIANT_TYPE_BOOL;
 
+GLIBMM_API
 extern const VariantType VARIANT_TYPE_BYTE;
 
+GLIBMM_API
 extern const VariantType VARIANT_TYPE_INT16;
 
+GLIBMM_API
 extern const VariantType VARIANT_TYPE_UINT16;
 
+GLIBMM_API
 extern const VariantType VARIANT_TYPE_INT32;
 
+GLIBMM_API
 extern const VariantType VARIANT_TYPE_UINT32;
 
+GLIBMM_API
 extern const VariantType VARIANT_TYPE_INT64;
 
+GLIBMM_API
 extern const VariantType VARIANT_TYPE_UINT64;
 
+GLIBMM_API
 extern const VariantType VARIANT_TYPE_DOUBLE;
 
+GLIBMM_API
 extern const VariantType VARIANT_TYPE_STRING;
 
+GLIBMM_API
 extern const VariantType VARIANT_TYPE_OBJECT_PATH;
 
+GLIBMM_API
 extern const VariantType VARIANT_TYPE_SIGNATURE;
 
+GLIBMM_API
 extern const VariantType VARIANT_TYPE_VARIANT;
 
+GLIBMM_API
 extern const VariantType VARIANT_TYPE_HANDLE;
 
+GLIBMM_API
 extern const VariantType VARIANT_TYPE_UNIT;
 
+GLIBMM_API
 extern const VariantType VARIANT_TYPE_ANY;
 
+GLIBMM_API
 extern const VariantType VARIANT_TYPE_BASIC;
 
+GLIBMM_API
 extern const VariantType VARIANT_TYPE_MAYBE;
 
+GLIBMM_API
 extern const VariantType VARIANT_TYPE_ARRAY;
 
+GLIBMM_API
 extern const VariantType VARIANT_TYPE_TUPLE;
 
+GLIBMM_API
 extern const VariantType VARIANT_TYPE_DICT_ENTRY;
 
+GLIBMM_API
 extern const VariantType VARIANT_TYPE_DICTIONARY;
 
+GLIBMM_API
 extern const VariantType VARIANT_TYPE_STRING_ARRAY;
 
+GLIBMM_API
 extern const VariantType VARIANT_TYPE_BYTESTRING;
 
+GLIBMM_API
 extern const VariantType VARIANT_TYPE_BYTESTRING_ARRAY;
 
 
