diff --git a/src/ptex/PtexWriter.cpp b/src/ptex/PtexWriter.cpp
index 99df951..0032275 100644
--- a/src/ptex/PtexWriter.cpp
+++ b/src/ptex/PtexWriter.cpp
@@ -42,7 +42,7 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGES.
 #include <algorithm>
 #include <iostream>
 #include <sstream>
-#if defined(__FreeBSD__)
+#if defined(__FreeBSD__) || defined(__ANDROID__)
     #include <unistd.h>
     #include <stddef.h>
 #endif
