diff --git a/include/litehtml.h b/include/litehtml.h
index 98a24e0..d20addd 100644
--- a/include/litehtml.h
+++ b/include/litehtml.h
@@ -1,11 +1,11 @@
 #ifndef LITEHTML_H
 #define LITEHTML_H
 
-#include <litehtml/html.h>
-#include <litehtml/document.h>
-#include <litehtml/html_tag.h>
-#include <litehtml/stylesheet.h>
-#include <litehtml/element.h>
-#include <litehtml/utf8_strings.h>
+#include "html.h"
+#include "document.h"
+#include "html_tag.h"
+#include "stylesheet.h"
+#include "element.h"
+#include "utf8_strings.h"
 
 #endif  // LITEHTML_H
