diff --git a/CMakeLists.txt b/CMakeLists.txt
index 00e058c..489b884 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -199,7 +199,8 @@ install(EXPORT ${PROJECT_NAME}Config DESTINATION lib/cmake/${PROJECT_NAME})
 ## Build provided copy of SQLite3 C library ##
 
 # TODO 
-#find_package(sqlite3)
+find_package(sqlite3 CONFIG)
+target_link_libraries(SQLiteCpp PRIVATE sqlite3)
 #if(sqlite3_VERSION VERSION_LESS "3.19")
 #    set_target_properties(SQLiteCpp PROPERTIES COMPILE_FLAGS "-DSQLITECPP_HAS_MEM_STRUCT")
 #endif()
