diff --git a/cmake/SQLiteCppConfig.cmake.in b/cmake/SQLiteCppConfig.cmake.in
index 2b48df4..d0feda9 100644
--- a/cmake/SQLiteCppConfig.cmake.in
+++ b/cmake/SQLiteCppConfig.cmake.in
@@ -1,6 +1,6 @@
 include(CMakeFindDependencyMacro)
 if(NOT @SQLITECPP_INTERNAL_SQLITE@)
-    find_dependency(SQLite3 REQUIRED)
+    find_dependency(unofficial-sqlite3 REQUIRED)
 endif()
 if(@UNIX@)
     set(THREADS_PREFER_PTHREAD_FLAG @THREADS_PREFER_PTHREAD_FLAG@)
