 CMakeLists.txt | 18 +++++++++++++-----
 1 file changed, 13 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index de8ccc8..e2484d2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -239,17 +245,19 @@ install(
         telnetpp
     EXPORT
         telnetpp-config
+    RUNTIME DESTINATION
+        bin
     ARCHIVE DESTINATION
-        lib/telnetpp-${TELNETPP_VERSION}
+        lib
     LIBRARY DESTINATION
-        lib/telnetpp-${TELNETPP_VERSION}
+        lib
 )
 
 install(
     DIRECTORY
         include/
     DESTINATION
-        include/telnetpp-${TELNETPP_VERSION}
+        include
 )
 
 export(
@@ -263,7 +271,7 @@ install(
     EXPORT
         telnetpp-config
     DESTINATION
-        lib/telnetpp-${TELNETPP_VERSION}
+        share/telnetpp
 )
 
 include(CMakePackageConfigHelpers)
@@ -278,7 +286,7 @@ install(
     FILES
         "${CMAKE_CURRENT_BINARY_DIR}/telnetpp-config-version.cmake"
     DESTINATION
-        lib/telnetpp-${TELNETPP_VERSION}
+        share/telnetpp
 )
 
 if (TELNETPP_WITH_TESTS)
