
---
 include/telnetpp/options/msdp/variable.hpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/telnetpp/options/msdp/variable.hpp b/include/telnetpp/options/msdp/variable.hpp
index 16dae0c..d74a6da 100644
--- a/include/telnetpp/options/msdp/variable.hpp
+++ b/include/telnetpp/options/msdp/variable.hpp
@@ -5,6 +5,7 @@
 #include <boost/variant.hpp>
 #include <iosfwd>
 #include <string>
+#include <ostream>
 
 namespace telnetpp { namespace options { namespace msdp {
 


---
 include/telnetpp/options/msdp/variable.hpp | 1 -
 src/options/msdp/variable.cpp              | 1 +
 2 files changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/telnetpp/options/msdp/variable.hpp b/include/telnetpp/options/msdp/variable.hpp
index d74a6da..16dae0c 100644
--- a/include/telnetpp/options/msdp/variable.hpp
+++ b/include/telnetpp/options/msdp/variable.hpp
@@ -5,7 +5,6 @@
 #include <boost/variant.hpp>
 #include <iosfwd>
 #include <string>
-#include <ostream>
 
 namespace telnetpp { namespace options { namespace msdp {
 
diff --git a/src/options/msdp/variable.cpp b/src/options/msdp/variable.cpp
index dc779ea..82549e9 100644
--- a/src/options/msdp/variable.cpp
+++ b/src/options/msdp/variable.cpp
@@ -1,5 +1,6 @@
 #include "telnetpp/options/msdp/variable.hpp"
 #include "telnetpp/detail/lambda_visitor.hpp"
+#include <ostream>
 
 namespace telnetpp { namespace options { namespace msdp {
 
