diff --git a/gui-builder/src/main.cpp b/gui-builder/src/main.cpp
index 4211c9c..649f226 100644
--- a/gui-builder/src/main.cpp
+++ b/gui-builder/src/main.cpp
@@ -24,8 +24,15 @@
 
 #include "GuiBuilder.hpp"
 
+// SDL has its own main function on some platforms but it has a define so that we can still call our version "main".
+// So if we are using the SDL backend then we must include it in this file.
+#if !TGUI_HAS_BACKEND_SFML && TGUI_HAS_BACKEND_SDL
+    #include <SDL_main.h>
+#endif
+
 int main(int, char* argv[])
 {
     GuiBuilder builder(argv[0]);
     builder.mainLoop();
+    return 0;
 }
