diff --git a/call/rtp_config.cc b/call/rtp_config.cc
index 4ab26a2..56079d2 100644
--- a/call/rtp_config.cc
+++ b/call/rtp_config.cc
@@ -258,7 +258,8 @@ RtpStreamConfig RtpConfig::GetStreamConfig(size_t index) const {
   stream_config.raw_payload = raw_payload;
   if (!rtx.ssrcs.empty()) {
     RTC_DCHECK_EQ(ssrcs.size(), rtx.ssrcs.size());
-    auto& stream_config_rtx = stream_config.rtx.emplace();
+    stream_config.rtx = RtpStreamConfig::Rtx{};
+    auto& stream_config_rtx = *stream_config.rtx;
     stream_config_rtx.ssrc = rtx.ssrcs[index];
     stream_config_rtx.payload_type = rtx.payload_type;
   }
