diff --git a/core/CMakeLists.txt b/core/CMakeLists.txt
index 4be4a9e..b858c08 100644
--- a/core/CMakeLists.txt
+++ b/core/CMakeLists.txt
@@ -16,7 +16,8 @@ set(LIBJSONNET_HEADERS
     static_error.h
     string_utils.h
     unicode.h
-    vm.h)
+    vm.h
+    ../third_party/md5/md5.h)
 
 set(LIBJSONNET_SOURCE
     desugarer.cpp
@@ -28,12 +29,13 @@ set(LIBJSONNET_SOURCE
     path_utils.cpp
     static_analysis.cpp
     string_utils.cpp
-    vm.cpp)
+    vm.cpp
+    ../third_party/md5/md5.cpp)
 
 if (BUILD_SHARED_BINARIES)
 add_library(libjsonnet ${LIBJSONNET_HEADERS} ${LIBJSONNET_SOURCE})
-add_dependencies(libjsonnet md5 stdlib)
-target_link_libraries(libjsonnet md5 nlohmann_json::nlohmann_json ryml)
+add_dependencies(libjsonnet stdlib)
+target_link_libraries(libjsonnet nlohmann_json::nlohmann_json ryml)
 
 file(STRINGS ${CMAKE_CURRENT_SOURCE_DIR}/../include/libjsonnet.h JSONNET_VERSION_DEF
      REGEX "[#]define[ \t]+LIB_JSONNET_VERSION[ \t]+")
@@ -59,8 +61,8 @@ endif()
 if (BUILD_STATIC_LIBS)
     # Static library for jsonnet command-line tool.
     add_library(libjsonnet_static STATIC ${LIBJSONNET_SOURCE})
-    add_dependencies(libjsonnet_static md5 stdlib)
-    target_link_libraries(libjsonnet_static md5 nlohmann_json::nlohmann_json ryml)
+    add_dependencies(libjsonnet_static stdlib)
+    target_link_libraries(libjsonnet_static nlohmann_json::nlohmann_json ryml)
     set_target_properties(libjsonnet_static PROPERTIES OUTPUT_NAME jsonnet)
     install(TARGETS libjsonnet_static DESTINATION "${CMAKE_INSTALL_LIBDIR}")
     target_include_directories(libjsonnet_static INTERFACE
