diff --git a/CMakeLists.txt b/CMakeLists.txt
index c3595a1..d38dc21 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -136,6 +136,7 @@ else()
   message(WARNING "ZLIB not found - install zlib for xnpz file support")
 endif()
 
+if (0)
 message(STATUS "Trying to find HighFive for npz file support")
 find_package(HighFive)
 find_package(HDF5)
@@ -154,6 +155,7 @@ if (${HighFive_FOUND} AND ${HDF5_FOUND})
 else()
   message(WARNING "HighFive and/or HDF5 not found - install HighFive and HDF5 for HDF5 file support")
 endif()
+endif()
 
 if(DOWNLOAD_GTEST OR GTEST_SRC_DIR)
     set(BUILD_TESTS ON)
