diff --git a/gtk/gtkmm/meson.build b/gtk/gtkmm/meson.build
index bf6c5804d..3b2ae866d 100644
--- a/gtk/gtkmm/meson.build
+++ b/gtk/gtkmm/meson.build
@@ -489,6 +489,10 @@ if is_host_windows
   extra_gtkmm_objects += gtkmm_res
 endif
 
+if meson.get_compiler('cpp').get_id() == 'msvc'
+    gtkmm_cpp_args += ['-std:c++17']
+endif
+
 gtkmm_library = library(gtkmm_libname, extra_gtkmm_objects,
   gtk_gen_sources, built_dummy_h_file_target, gtkmm_extra_used_cc_files,
   include_directories: extra_include_dirs,
diff --git a/gtk/src/expressionwatch.hg b/gtk/src/expressionwatch.hg
index 78308d221..04e593f8b 100644
--- a/gtk/src/expressionwatch.hg
+++ b/gtk/src/expressionwatch.hg
@@ -16,6 +16,7 @@
 
 _CONFIGINCLUDE(gtkmmconfig.h)
 
+#include <optional>
 #include <glibmm/object.h>
 #include <gtk/gtk.h>
 
diff --git a/untracked/gtk/gtkmm/expressionwatch.h b/untracked/gtk/gtkmm/expressionwatch.h
index 3a80efd64..279870ed7 100644
--- a/untracked/gtk/gtkmm/expressionwatch.h
+++ b/untracked/gtk/gtkmm/expressionwatch.h
@@ -4,7 +4,7 @@
 
 #include <gtkmmconfig.h>
 
-
+#include <optional>
 #include <glibmm/ustring.h>
 #include <sigc++/sigc++.h>
 
