diff --git a/untracked/gtk/gtkmm/expression.h b/untracked/gtk/gtkmm/expression.h
index fc3bf1e5b..ff70c720a 100644
--- a/untracked/gtk/gtkmm/expression.h
+++ b/untracked/gtk/gtkmm/expression.h
@@ -377,7 +377,7 @@ private:
 namespace Expression_Private
 {
 
-void watch_callback(gpointer data);
+GTKMM_API void watch_callback(gpointer data);
 
 template<class T, class... ATs>
 class Invoker
@@ -406,7 +406,7 @@ void closure_marshal(GClosure* closure,
                      gpointer invocation_hint,
                      gpointer marshal_data);
 
-void closure_callback_func();
+GTKMM_API void closure_callback_func();
 
 template<class TI>
 void closure_destroy(gpointer data, GClosure* closure);
diff --git a/untracked/gtk/gtkmm/expressionwatch.h b/untracked/gtk/gtkmm/expressionwatch.h
index 279870ed7..e9c760664 100644
--- a/untracked/gtk/gtkmm/expressionwatch.h
+++ b/untracked/gtk/gtkmm/expressionwatch.h
@@ -83,7 +83,7 @@ public:
 };
 
 template<class T>
-class GTKMM_API ExpressionWatch final : public ExpressionWatchBase
+class ExpressionWatch final : public ExpressionWatchBase
 {
 public:
   /** Evaluates the watched expression and on success returns the result.
