diff --git a/CMakeLists.txt b/CMakeLists.txt
index f3cfc20..9826bff 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -385,6 +385,12 @@ if (BUILD_TESTS)
     target_link_libraries (bench ${fftw3_lib})
   endif ()
 
+  if (ENABLE_OPENMP)
+    target_link_libraries (bench ${fftw3_lib}_omp)
+  else ()
+    target_link_libraries (bench ${fftw3_lib})
+  endif ()
+
 
   enable_testing ()
 
