diff --git a/src/tools/msvc/Install.pm b/src/tools/msvc/Install.pm
index d109b2c90..75c7446e6 100644
--- a/src/tools/msvc/Install.pm
+++ b/src/tools/msvc/Install.pm
@@ -22,13 +22,13 @@ our (@ISA, @EXPORT_OK);
 
 my $insttype;
 my @client_contribs = ('oid2name', 'pgbench', 'vacuumlo');
-my @client_program_files = (
-	'clusterdb',     'createdb',       'createuser', 'dropdb',
-	'dropuser',      'ecpg',           'libecpg',    'libecpg_compat',
-	'libpgtypes',    'libpq',          'pg_amcheck', 'pg_basebackup',
-	'pg_config',     'pg_dump',        'pg_dumpall', 'pg_isready',
-	'pg_receivewal', 'pg_recvlogical', 'pg_restore', 'psql',
-	'reindexdb',     'vacuumdb',       @client_contribs);
+my @client_program_files = ( 'libecpg','libecpg_compat', 'libpgtypes', 'libpq' );
+	#'clusterdb',     'createdb',       'createuser', 'dropdb',
+	#'dropuser',      'ecpg',           'libecpg',    'libecpg_compat',
+	#'libpgtypes',    'libpq',          'pg_amcheck', 'pg_basebackup',
+	#'pg_config',     'pg_dump',        'pg_dumpall', 'pg_isready',
+	#'pg_receivewal', 'pg_recvlogical', 'pg_restore', 'psql',
+	#'reindexdb',     'vacuumdb',       @client_contribs);
 
 sub lcopy
 {
@@ -121,7 +121,7 @@ sub Install
 	CopyFiles(
 		'Import libraries',
 		$target . '/lib/',
-		"$conf\\", "postgres\\postgres.lib", "libpgcommon\\libpgcommon.lib",
+		"$conf\\", "libpgcommon\\libpgcommon.lib",
 		"libpgport\\libpgport.lib");
 	CopyContribFiles($config, $target);
 	CopyIncludeFiles($target);

