diff --git a/openvdb/openvdb/cmd/CMakeLists.txt b/openvdb/openvdb/cmd/CMakeLists.txt
index c173566..53f8575 100644
--- a/openvdb/openvdb/cmd/CMakeLists.txt
+++ b/openvdb/openvdb/cmd/CMakeLists.txt
@@ -137,9 +137,11 @@ if(OPENVDB_BUILD_VDB_RENDER)
 
   if(WIN32)
     # @note OPENVDB_OPENEXR_STATICLIB is old functionality and should be removed
-    get_target_property(ILMBASE_LIB_TYPE IlmBase::Half TYPE)
-    if(OPENEXR_USE_STATIC_LIBS OR (${ILMBASE_LIB_TYPE} STREQUAL STATIC_LIBRARY))
-      target_compile_definitions(vdb_render PUBLIC -DOPENVDB_OPENEXR_STATICLIB)
+    if(USE_IMATH_HALF)
+      get_target_property(ILMBASE_LIB_TYPE IlmBase::Half TYPE)
+      if(OPENEXR_USE_STATIC_LIBS OR (${ILMBASE_LIB_TYPE} STREQUAL STATIC_LIBRARY))
+        target_compile_definitions(vdb_render PUBLIC -DOPENVDB_OPENEXR_STATICLIB)
+      endif()
     endif()
   endif()
 
