diff --git a/nanovdb/nanovdb/CMakeLists.txt b/nanovdb/nanovdb/CMakeLists.txt
index 4e0284e..62fc68b 100644
--- a/nanovdb/nanovdb/CMakeLists.txt
+++ b/nanovdb/nanovdb/CMakeLists.txt
@@ -126,7 +126,7 @@ if(NANOVDB_USE_TBB AND NOT TARGET TBB::tbb)
 endif()
 
 if(NANOVDB_USE_BLOSC AND NOT TARGET Blosc::blosc)
-  find_package(Blosc ${MINIMUM_BLOSC_VERSION} REQUIRED)
+  find_package(Blosc ${MINIMUM_BLOSC_VERSION} REQUIRED NAMES blosc)
 endif()
 
 if(NANOVDB_USE_ZLIB AND NOT TARGET ZLIB::ZLIB)
@@ -290,7 +290,7 @@ if(NANOVDB_USE_TBB)
 endif()
 
 if(NANOVDB_USE_BLOSC)
-  target_link_libraries(nanovdb INTERFACE Blosc::blosc)
+  target_link_libraries(nanovdb INTERFACE $<IF:$<TARGET_EXISTS:blosc_shared>,blosc_shared,blosc_static>)
   target_compile_definitions(nanovdb INTERFACE -DNANOVDB_USE_BLOSC)
 endif()
 
