diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index a366e51..257f87d 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -132,7 +132,9 @@ endif()  #!MSVC
 include(jxl.cmake)
 
 # Other libraries outside the core jxl library.
+if(JPEGXL_ENABLE_TOOLS)
 include(jxl_extras.cmake)
+endif()
 include(jxl_threads.cmake)
 
 # Install all the library headers from the source and the generated ones. There
