diff --git a/lz4.h b/lz4.h
index 7420ad8..c1c3e56 100644
--- a/lz4.h
+++ b/lz4.h
@@ -73,6 +73,9 @@ extern "C" {
 *  LZ4_DLL_EXPORT :
 *  Enable exporting of functions when building a Windows DLL
 */
+
+#define LZ4_DLL_IMPORT 1
+
 #if defined(LZ4_DLL_EXPORT) && (LZ4_DLL_EXPORT==1)
 #  define LZ4LIB_API __declspec(dllexport)
 #elif defined(LZ4_DLL_IMPORT) && (LZ4_DLL_IMPORT==1)
diff --git a/lz4frame.h b/lz4frame.h
index 3104d2e..4442461 100644
--- a/lz4frame.h
+++ b/lz4frame.h
@@ -55,6 +55,9 @@ extern "C" {
 *  LZ4_DLL_EXPORT :
 *  Enable exporting of functions when building a Windows DLL
 */
+
+#define LZ4_DLL_IMPORT 1
+
 #if defined(LZ4_DLL_EXPORT) && (LZ4_DLL_EXPORT==1)
 #  define LZ4FLIB_API __declspec(dllexport)
 #elif defined(LZ4_DLL_IMPORT) && (LZ4_DLL_IMPORT==1)
