--- a/OpenXLSX/CMakeLists.txt.old	2025-07-14 15:07:03.000000000 +0200
+++ a/OpenXLSX/CMakeLists.txt	2025-07-29 10:04:55.753385000 +0200
@@ -41,8 +41,7 @@ set(OPENXLSX_LIBRARY_TYPE "STATIC" CACHE
 #=======================================================================================================================
 
 if (OPENXLSX_ENABLE_NOWIDE)
-    add_library(NoWide INTERFACE IMPORTED)
-    target_include_directories(NoWide SYSTEM INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/external/nowide/>)
+    find_package(nowide CONFIG REQUIRED)
 endif()
 
 add_library(Zippy INTERFACE IMPORTED)
@@ -51,8 +50,7 @@ if (OPENXLSX_ENABLE_NOWIDE)
     target_compile_definitions(Zippy INTERFACE ENABLE_NOWIDE)
 endif ()
 
-add_library(PugiXML INTERFACE IMPORTED)
-target_include_directories(PugiXML SYSTEM INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/external/pugixml/>)
+find_package(PugiXML CONFIG REQUIRED)
 
 if (${OPENXLSX_COMPACT_MODE})
     target_compile_definitions(PugiXML INTERFACE PUGIXML_COMPACT)
@@ -143,12 +141,13 @@ if ("${OPENXLSX_LIBRARY_TYPE}" STREQUAL
     target_link_libraries(OpenXLSX
             PRIVATE
             $<BUILD_INTERFACE:Zippy>
-            $<BUILD_INTERFACE:PugiXML>)
+            PUBLIC
+            pugixml::pugixml)
 
     if (OPENXLSX_ENABLE_NOWIDE)
         target_link_libraries(OpenXLSX
-                PRIVATE
-                $<BUILD_INTERFACE:NoWide>)
+                PUBLIC
+                nowide::nowide)
     endif ()
 
     target_compile_definitions(OpenXLSX PUBLIC OPENXLSX_STATIC_DEFINE)
@@ -171,12 +170,13 @@ if ("${OPENXLSX_LIBRARY_TYPE}" STREQUAL
     target_link_libraries(OpenXLSX
             PRIVATE
             $<BUILD_INTERFACE:Zippy>
-            $<BUILD_INTERFACE:PugiXML>)
+            PUBLIC
+            pugixml::pugixml)
 
     if (OPENXLSX_ENABLE_NOWIDE)
         target_link_libraries(OpenXLSX
-                PRIVATE
-                $<BUILD_INTERFACE:NoWide>)
+                PUBLIC
+                nowide::nowide)
     endif ()
 
     # Enable Link-Time Optimization (LTO)
@@ -334,9 +334,16 @@ install(
 # Package configuration
 configure_file(OpenXLSXConfig.cmake
         "${CMAKE_CURRENT_BINARY_DIR}/OpenXLSX/OpenXLSXConfig.cmake"
-        COPYONLY
+        @ONLY
         )
 
+install(
+        FILES
+        "${CMAKE_CURRENT_BINARY_DIR}/OpenXLSX/OpenXLSXConfig.cmake"
+        "${CMAKE_CURRENT_BINARY_DIR}/OpenXLSX/OpenXLSXConfigVersion.cmake"
+        DESTINATION ${ConfigPackageLocation}
+)
+
 # Package export targets
 export(
         EXPORT OpenXLSXTargets
--- a/OpenXLSX/OpenXLSXConfig.cmake.old	2025-07-29 10:01:35.707328300 +0200
+++ a/OpenXLSX/OpenXLSXConfig.cmake	2025-07-29 10:01:39.321206200 +0200
@@ -1 +1,7 @@
-include("${CMAKE_CURRENT_LIST_DIR}/OpenXLSXTargets.cmake")
\ No newline at end of file
+include(CMakeFindDependencyMacro)
+find_dependency(pugixml CONFIG)
+if(@OPENXLSX_ENABLE_NOWIDE@)
+    find_dependency(nowide CONFIG)
+endif()
+
+include("${CMAKE_CURRENT_LIST_DIR}/OpenXLSXTargets.cmake")
