diff --git a/OpenXLSX/CMakeLists.txt b/OpenXLSX/CMakeLists.txt
index 9bfe1f8..3b054a6 100644
--- a/OpenXLSX/CMakeLists.txt
+++ b/OpenXLSX/CMakeLists.txt
@@ -117,6 +117,10 @@ set(OPENXLSX_SOURCES
         ${CMAKE_CURRENT_LIST_DIR}/sources/XLZipArchive.cpp
         )
 
+
+if(OPENXLSX_ENABLE_NOWIDE)
+        find_package(nowide CONFIG REQUIRED)
+endif()
 
 #=======================================================================================================================
 # STATIC AND SHARED LIBRARY
@@ -142,12 +146,12 @@ if ("${OPENXLSX_LIBRARY_TYPE}" STREQUAL "STATIC")
     target_link_libraries(OpenXLSX
             PRIVATE
             $<BUILD_INTERFACE:Zippy>
            $<BUILD_INTERFACE:PugiXML>)
 
     if (OPENXLSX_ENABLE_NOWIDE)
         target_link_libraries(OpenXLSX
                 PRIVATE
-                $<BUILD_INTERFACE:NoWide>)
+                nowide::nowide)
     endif ()
 
     target_compile_definitions(OpenXLSX PUBLIC OPENXLSX_STATIC_DEFINE)
@@ -170,12 +174,12 @@ if ("${OPENXLSX_LIBRARY_TYPE}" STREQUAL "SHARED")
     target_link_libraries(OpenXLSX
             PRIVATE
             $<BUILD_INTERFACE:Zippy>
             $<BUILD_INTERFACE:PugiXML>)
 
     if (OPENXLSX_ENABLE_NOWIDE)
         target_link_libraries(OpenXLSX
                 PRIVATE
-                $<BUILD_INTERFACE:NoWide>)
+                nowide::nowide)
     endif ()
 
     # Enable Link-Time Optimization (LTO)
