diff --git a/CMakeLists.txt b/CMakeLists.txt
index 92846d4f..96fc1cc9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -155,2 +155,5 @@ if(SPDLOG_FMT_EXTERNAL OR SPDLOG_FMT_EXTERNAL_HO)
         target_link_libraries(spdlog_header_only INTERFACE fmt::fmt)
+        if (WIN32)
+            target_compile_options(spdlog PRIVATE /wd4275)
+        endif()
     endif()

diff --git a/bench/CMakeLists.txt b/bench/CMakeLists.txt
index 9913ae64..a57462b3 100644
--- a/bench/CMakeLists.txt
+++ b/bench/CMakeLists.txt
@@ -17,2 +17,5 @@ spdlog_enable_warnings(bench)
 target_link_libraries(bench PRIVATE spdlog::spdlog)
+if (WIN32)
+    target_compile_options(bench PRIVATE /wd4275)
+endif()

@@ -26,2 +29,5 @@ add_executable(formatter-bench formatter-bench.cpp)
 target_link_libraries(formatter-bench PRIVATE benchmark::benchmark spdlog::spdlog)
+if (WIN32)
+    target_compile_options(formatter-bench PRIVATE /wd4275)
+endif()

