diff --git a/visualc/vngspice-fftw.vcxproj b/visualc/vngspice-fftw.vcxproj
index fb9c4ea..3b094e7 100644
--- a/visualc/vngspice-fftw.vcxproj
+++ b/visualc/vngspice-fftw.vcxproj
@@ -238,8 +238,6 @@ lib /machine:x86 /def:..\..\fftw-3.3-dll32\libfftw3-3.def /out:$(IntDir)libfftw3
     </Link>
     <PostBuildEvent>
       <Command>
-      copy /y "..\..\fftw-3.3-dll32\libfftw3-3.dll" "$(OutDir)"
-      make-install-vngspiced.bat $(OutDir) fftw
       </Command>
     </PostBuildEvent>
   </ItemDefinitionGroup>
@@ -289,8 +287,6 @@ lib /machine:x86 /def:..\..\fftw-3.3-dll32\libfftw3-3.def /out:$(IntDir)libfftw3
     </Link>
     <PostBuildEvent>
       <Command>
-      copy /y "..\..\fftw-3.3-dll32\libfftw3-3.dll" "$(OutDir)"
-      make-install-vngspice.bat $(OutDir) fftw
       </Command>
     </PostBuildEvent>
   </ItemDefinitionGroup>
@@ -339,8 +335,6 @@ lib /machine:x64 /def:..\..\fftw-3.3-dll64\libfftw3-3.def /out:$(IntDir)libfftw3
     </Link>
     <PostBuildEvent>
       <Command>
-      copy /y "..\..\fftw-3.3-dll64\libfftw3-3.dll" "$(OutDir)"
-      make-install-vngspiced.bat $(OutDir) fftw 64
       </Command>
     </PostBuildEvent>
   </ItemDefinitionGroup>
@@ -397,8 +391,6 @@ lib /machine:x64 /def:..\..\fftw-3.3-dll64\libfftw3-3.def /out:$(IntDir)libfftw3
     </Link>
     <PostBuildEvent>
       <Command>
-      copy /y "..\..\fftw-3.3-dll64\libfftw3-3.dll" "$(OutDir)"
-      make-install-vngspice.bat $(OutDir) fftw 64
       </Command>
     </PostBuildEvent>
   </ItemDefinitionGroup>
@@ -443,8 +435,6 @@ lib /machine:x86 /def:..\..\fftw-3.3-dll32\libfftw3-3.def /out:$(IntDir)libfftw3
     </Link>
     <PostBuildEvent>
       <Command>
-      copy /y "..\..\fftw-3.3-dll32\libfftw3-3.dll" "$(OutDir)"
-      make-install-vngspiced.bat $(OutDir) fftw
       </Command>
     </PostBuildEvent>
   </ItemDefinitionGroup>
@@ -495,8 +485,6 @@ lib /machine:x86 /def:..\..\fftw-3.3-dll32\libfftw3-3.def /out:$(IntDir)libfftw3
     </Link>
     <PostBuildEvent>
       <Command>
-      copy /y "..\..\fftw-3.3-dll32\libfftw3-3.dll" "$(OutDir)"
-      make-install-vngspice.bat $(OutDir) fftw
       </Command>
     </PostBuildEvent>
   </ItemDefinitionGroup>
@@ -545,8 +533,6 @@ lib /machine:x64 /def:..\..\fftw-3.3-dll64\libfftw3-3.def /out:$(IntDir)libfftw3
     </Link>
     <PostBuildEvent>
       <Command>
-      copy /y "..\..\fftw-3.3-dll64\libfftw3-3.dll" "$(OutDir)"
-      make-install-vngspiced.bat $(OutDir) fftw 64
       </Command>
     </PostBuildEvent>
   </ItemDefinitionGroup>
@@ -601,8 +587,6 @@ lib /machine:x64 /def:..\..\fftw-3.3-dll64\libfftw3-3.def /out:$(IntDir)libfftw3
     </Link>
     <PostBuildEvent>
       <Command>
-      copy /y "..\..\fftw-3.3-dll64\libfftw3-3.dll" "$(OutDir)"
-      make-install-vngspice.bat $(OutDir) fftw 64
       </Command>
     </PostBuildEvent>
   </ItemDefinitionGroup>
@@ -654,8 +638,6 @@ lib /machine:x86 /def:..\..\fftw-3.3-dll32\libfftw3-3.def /out:$(IntDir)libfftw3
     </Link>
     <PostBuildEvent>
       <Command>
-      copy /y "..\..\fftw-3.3-dll32\libfftw3-3.dll" "$(OutDir)"
-      make-install-vngspice.bat $(OutDir) fftw
       </Command>
     </PostBuildEvent>
   </ItemDefinitionGroup>
@@ -714,8 +696,6 @@ lib /machine:x64 /def:..\..\fftw-3.3-dll64\libfftw3-3.def /out:$(IntDir)libfftw3
     </Link>
     <PostBuildEvent>
       <Command>
-      copy /y "..\..\fftw-3.3-dll64\libfftw3-3.dll" "$(OutDir)"
-      make-install-vngspice.bat $(OutDir) fftw 64
       </Command>
     </PostBuildEvent>
   </ItemDefinitionGroup>
@@ -766,8 +746,6 @@ lib /machine:x86 /def:..\..\fftw-3.3-dll32\libfftw3-3.def /out:$(IntDir)libfftw3
     </Link>
     <PostBuildEvent>
       <Command>
-      copy /y "..\..\fftw-3.3-dll32\libfftw3-3.dll" "$(OutDir)"
-      make-install-vngspice.bat $(OutDir) fftw
       </Command>
     </PostBuildEvent>
   </ItemDefinitionGroup>
@@ -824,8 +802,6 @@ lib /machine:x64 /def:..\..\fftw-3.3-dll64\libfftw3-3.def /out:$(IntDir)libfftw3
     </Link>
     <PostBuildEvent>
       <Command>
-      copy /y "..\..\fftw-3.3-dll64\libfftw3-3.dll" "$(OutDir)"
-      make-install-vngspice.bat $(OutDir) fftw 64
       </Command>
     </PostBuildEvent>
   </ItemDefinitionGroup>
diff --git a/visualc/vngspice.vcxproj b/visualc/vngspice.vcxproj
index 0321228..b4fe958 100644
--- a/visualc/vngspice.vcxproj
+++ b/visualc/vngspice.vcxproj
@@ -244,7 +244,6 @@
       <AdditionalLibraryDirectories>KLU/Debug/;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
     </Link>
     <PostBuildEvent>
-      <Command>make-install-vngspiced.bat $(OutDir)</Command>
     </PostBuildEvent>
     <Manifest>
       <AdditionalManifestFiles>$(ProjectDir)ngspice-x86.exe.manifest</AdditionalManifestFiles>
@@ -297,7 +296,6 @@
     </Link>
     <PostBuildEvent>
       <Command>
-      make-install-vngspice.bat $(OutDir)
       </Command>
     </PostBuildEvent>
     <Manifest>
@@ -348,7 +346,6 @@
     </Link>
     <PostBuildEvent>
       <Command>
-      make-install-vngspiced.bat $(OutDir) 64
       </Command>
     </PostBuildEvent>
     <Manifest>
@@ -407,7 +404,6 @@
     </Link>
     <PostBuildEvent>
       <Command>
-      make-install-vngspice.bat $(OutDir) 64
       </Command>
     </PostBuildEvent>
     <Manifest>
@@ -454,7 +450,6 @@
       <AdditionalLibraryDirectories>KLU/Debug/;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
     </Link>
     <PostBuildEvent>
-      <Command>make-install-vngspiced.bat $(OutDir)</Command>
     </PostBuildEvent>
     <Manifest>
       <AdditionalManifestFiles>$(ProjectDir)ngspice-x86.exe.manifest</AdditionalManifestFiles>
@@ -506,7 +501,6 @@
       <AdditionalLibraryDirectories>KLU/Release/;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
     </Link>
     <PostBuildEvent>
-      <Command>make-install-vngspice.bat $(OutDir)</Command>
     </PostBuildEvent>
     <Manifest>
       <AdditionalManifestFiles>$(ProjectDir)ngspice-x86.exe.manifest</AdditionalManifestFiles>
@@ -556,7 +550,6 @@
     </Link>
     <PostBuildEvent>
       <Command>
-      make-install-vngspiced.bat $(OutDir) 64
       </Command>
     </PostBuildEvent>
     <Manifest>
@@ -613,7 +606,6 @@
     </Link>
     <PostBuildEvent>
       <Command>
-      make-install-vngspice.bat $(OutDir) 64
       </Command>
     </PostBuildEvent>
     <Manifest>
@@ -666,7 +658,6 @@
       <AdditionalLibraryDirectories>KLU/Release/;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
     </Link>
     <PostBuildEvent>
-      <Command>make-install-vngspice.bat $(OutDir)</Command>
     </PostBuildEvent>
     <Manifest>
       <AdditionalManifestFiles>$(ProjectDir)ngspice-x86.exe.manifest</AdditionalManifestFiles>
@@ -726,7 +717,6 @@
     </Link>
     <PostBuildEvent>
       <Command>
-      make-install-vngspice.bat $(OutDir) 64
       </Command>
     </PostBuildEvent>
     <Manifest>
@@ -779,7 +769,6 @@
       <AdditionalLibraryDirectories>KLU/Release/;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
     </Link>
     <PostBuildEvent>
-      <Command>make-install-vngspice.bat $(OutDir)</Command>
     </PostBuildEvent>
     <Manifest>
       <AdditionalManifestFiles>$(ProjectDir)ngspice-x86.exe.manifest</AdditionalManifestFiles>
@@ -837,7 +826,6 @@
     </Link>
     <PostBuildEvent>
       <Command>
-      make-install-vngspice.bat $(OutDir) 64
       </Command>
     </PostBuildEvent>
     <Manifest>
