diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7974e8b..8afdb5a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -255,8 +255,8 @@ if(NOT BUILD_SHARED_LIBS)
 endif()
 
 eprosima_find_package(fastcdr 2 REQUIRED)
-eprosima_find_thirdparty(Asio asio VERSION 1.13.0)
-eprosima_find_thirdparty(TinyXML2 tinyxml2)
+find_package(asio CONFIG REQUIRED)
+find_package(tinyxml2 CONFIG REQUIRED)
 
 find_package(foonathan_memory REQUIRED)
 message(STATUS "Found foonathan_memory: ${foonathan_memory_DIR}")
@@ -268,7 +268,7 @@ if(ANDROID)
     endif()
 endif()
 
-include_directories(thirdparty/nlohmann-json)
+find_package(nlohmann_json CONFIG REQUIRED)
 include_directories(thirdparty/filewatch)
 
 ###############################################################################
diff --git a/src/cpp/CMakeLists.txt b/src/cpp/CMakeLists.txt
index 3f4a3aa..3394cc3 100644
--- a/src/cpp/CMakeLists.txt
+++ b/src/cpp/CMakeLists.txt
@@ -508,8 +508,9 @@ target_link_libraries(${PROJECT_NAME}
     ${PRIVACY}
     fastcdr
     foonathan_memory
+    nlohmann_json::nlohmann_json
     ${CMAKE_THREAD_LIBS_INIT} ${CMAKE_DL_LIBS}
-    ${TINYXML2_LIBRARY}
+    tinyxml2::tinyxml2
     $<$<BOOL:${LINK_SSL}>:OpenSSL::SSL$<SEMICOLON>OpenSSL::Crypto$<$<BOOL:${WIN32}>:$<SEMICOLON>crypt32.lib>>
     $<$<BOOL:${WIN32}>:iphlpapi$<SEMICOLON>Shlwapi>
     ${THIRDPARTY_BOOST_LINK_LIBS}
