diff --git a/CMakeLists.txt b/CMakeLists.txt
index e3b30fd..3828816 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -59,7 +59,7 @@ if(WIN32)
     set(Boost_USE_STATIC_LIBS ON)
 endif(WIN32)
 
-find_package(Boost REQUIRED COMPONENTS system date_time regex unit_test_framework)
+find_package(Boost REQUIRED COMPONENTS system date_time regex)
 find_package(OpenSSL)
 set(CMAKE_THREAD_PREFER_PTHREAD)
 # "Use of both the imported target as well as this switch is highly recommended for new code."
@@ -123,7 +123,7 @@ if (MAILIO_BUILD_SHARED_LIBRARY)
 else()
     set(LIBRARY_TYPE "STATIC")
 endif()
-add_library(${PROJECT_NAME} ${LIBRARY_TYPE} ${project_sources} ${project_headers})
+add_library(${PROJECT_NAME} ${project_sources} ${project_headers})
 
 
 # pkg-config support
