diff --git a/makefile.vc b/makefile.vc
index 5841cd3..5767a4a 100644
--- a/makefile.vc
+++ b/makefile.vc
@@ -82,7 +82,7 @@ staticlib:	$(LIB_DEPENDS)
 	call <<clean_main_build_output.bat
 $(CLEAN_MAIN_BUILD_OUTPUT_CMDS)
 <<
-	lib /nologo /out:gdal.lib $(LIBOBJ) $(EXTERNAL_LIBS)
+	lib /nologo /out:gdal.lib $(LIBOBJ) $(EXTERNAL_LIBS) /LTCG
 
 appslib-target:
 	cd apps
@@ -225,23 +225,12 @@ install: default
 	if exist $(GDAL_DLL) $(INSTALL) $(GDAL_DLL) $(BINDIR)
 	cd frmts
 	$(MAKE) /f makefile.vc plugins-install
-	cd ..\apps
-	$(MAKE) /f makefile.vc install
 	cd ..
 	$(INSTALL) data\*.* $(DATADIR)
 	$(INSTALL) LICENSE.TXT $(DATADIR)
-!IFDEF HTMLDIR
-	-mkdir $(HTMLDIR)
-	cd frmts
-	$(MAKE) /f makefile.vc html-install
-	cd ..
-!ENDIF
 !IFDEF INCLUDE_OGR_FRMTS
 	cd ogr\ogrsf_frmts
 	$(MAKE) /f makefile.vc plugins-install
-!IFDEF HTMLDIR
-	$(MAKE) /f makefile.vc html-install
-!ENDIF
 	cd ..\..
 !ENDIF
 
