diff --git a/makefile.vc b/makefile.vc
index 8559f79..87f8fc9 100644
--- a/makefile.vc
+++ b/makefile.vc
@@ -61,7 +61,7 @@ TARGET_LIB = staticlib
 
 DEFAULT_TARGETS = 
 
-default:	$(TARGET_LIB) $(PLUGIN_TARGET) apps_dir
+default:	$(TARGET_LIB) $(PLUGIN_TARGET)
 
 # Batch commands to clean main build output are stored in a macro so they can be reused by different
 # targets in an inline temporary batch file.
@@ -230,8 +230,6 @@ install: default docs
 	if exist $(GDAL_DLL) $(INSTALL) $(GDAL_DLL) $(BINDIR)
 	cd frmts
 	$(MAKE) /f makefile.vc plugins-install
-	cd ..\apps
-	$(MAKE) /f makefile.vc install
 	cd ..
 	$(INSTALL) data\*.* "$(DATADIR)"
 	$(INSTALL) LICENSE.TXT "$(DATADIR)"
