From 9da0cb6eed442ebf3eea232cd85e26c155c963ef Mon Sep 17 00:00:00 2001
From: Manuel Massing <m.massing@warped-space.de>
Date: Tue, 29 Nov 2016 12:43:58 +0100
Subject: [PATCH 3/4] - Ensures inclusion of PDB in release dll if so requested

---
 gdal/nmake.opt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/nmake.opt b/nmake.opt
index 47c2df8..4844d22 100644
--- a/nmake.opt
+++ b/nmake.opt
@@ -174,6 +174,9 @@ SOFTWARNFLAGS= /wd4244 /wd4702 /wd4701 /wd4013 /wd4706 /wd4057 /wd4210 /wd4305
 # Linker debug options
 !IFDEF DEBUG
 LDEBUG= /debug
+!ELSEIFDEF WITH_PDB
+# Ensures that PDB is included in release DLL if so requested
+LDEBUG= /debug /opt:ref /opt:icf
 !ENDIF
 
 # Uncomment the following if you are building for 64-bit windows
-- 
2.10.1.windows.1

