diff --git a/cmake/dateConfig.cmake b/cmake/dateConfig.cmake
index 20f86e8..2198ad1 100644
--- a/cmake/dateConfig.cmake
+++ b/cmake/dateConfig.cmake
@@ -1,8 +1,8 @@
 include( CMakeFindDependencyMacro )
 include( "${CMAKE_CURRENT_LIST_DIR}/dateTargets.cmake" )
-if( NOT MSVC AND TARGET date::tz )
+if( NOT MSVC AND TARGET date::date-tz )
     find_dependency( Threads REQUIRED)
-    get_target_property( _tzill date::tz  INTERFACE_LINK_LIBRARIES )
+    get_target_property( _tzill date::date-tz  INTERFACE_LINK_LIBRARIES )
     if( _tzill AND "${_tzill}" MATCHES "libcurl" )
         find_dependency( CURL )
     endif( )
