diff --git a/cmake/Helper.cmake b/cmake/Helper.cmake
index b73a6cb..d3bdf29 100644
--- a/cmake/Helper.cmake
+++ b/cmake/Helper.cmake
@@ -179,8 +179,8 @@ if((WASMEDGE_LINK_LLVM_STATIC OR WASMEDGE_BUILD_STATIC_LIB) AND WASMEDGE_BUILD_A
   endif()
   if (APPLE AND LLVM_VERSION_MAJOR GREATER_EQUAL 15)
     # For LLVM 15 or greater on MacOS
-    find_package(zstd REQUIRED)
-    get_filename_component(ZSTD_PATH "${zstd_LIBRARY}" DIRECTORY)
+    find_package(zstd CONFIG REQUIRED)
+    set(ZSTD_PATH "${VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/lib")
     list(APPEND WASMEDGE_LLVM_LINK_STATIC_COMPONENTS
       ${ZSTD_PATH}/libzstd.a
     )
