diff --git a/src/io/scene/CMakeLists.txt b/src/io/scene/CMakeLists.txt
index 5a0a1c6..37a0a19 100644
--- a/src/io/scene/CMakeLists.txt
+++ b/src/io/scene/CMakeLists.txt
@@ -104,5 +104,10 @@ target_add_dependencies(TARGET ${PROJECT_NAME}
         boost
     )
 
+target_compile_definitions(${PROJECT_NAME}
+    PRIVATE
+        BOOST_UUID_FORCE_AUTO_LINK # for some reason this is not working while using vcpkg boost-uuid
+)
+
 # add tests if available
 add_tests(POST)
