diff --git a/src/mdl/jit/llvm/dist/cmake/modules/AddLLVM.cmake b/src/mdl/jit/llvm/dist/cmake/modules/AddLLVM.cmake
index ce2057f..7ad8dfe 100644
--- a/src/mdl/jit/llvm/dist/cmake/modules/AddLLVM.cmake
+++ b/src/mdl/jit/llvm/dist/cmake/modules/AddLLVM.cmake
@@ -616,11 +616,7 @@ macro(add_llvm_library name)
     ""
     ""
     ${ARGN})
-  if( BUILD_SHARED_LIBS OR ARG_SHARED )
-    llvm_add_library(${name} SHARED ${ARG_UNPARSED_ARGUMENTS})
-  else()
-    llvm_add_library(${name} ${ARG_UNPARSED_ARGUMENTS})
-  endif()
+  llvm_add_library(${name} STATIC ${ARG_UNPARSED_ARGUMENTS})
 
   # Libraries that are meant to only be exposed via the build tree only are
   # never installed and are only exported as a target in the special build tree
