diff --git a/cmake/dependencies/add_llvm.cmake b/cmake/dependencies/add_llvm.cmake
index e477c57..0e323d0 100644
--- a/cmake/dependencies/add_llvm.cmake
+++ b/cmake/dependencies/add_llvm.cmake
@@ -100,7 +100,7 @@ set(_LLVM_LIB_NAMES
     LLVMSupport
     )
     
-target_include_directories(${__TARGET_ADD_DEPENDENCY_TARGET} 
+target_include_directories(${__TARGET_ADD_DEPENDENCY_TARGET} BEFORE
     PRIVATE
         ${mdl-jit-llvm_SOURCE_DIR}/dist/include
         ${mdl-jit-llvm_BINARY_DIR}/dist/include
