diff --git a/gettext-runtime/intl/xsize.h b/gettext-runtime/intl/xsize.h
index c256665..7400ea1 100644
--- a/gettext-runtime/intl/xsize.h
+++ b/gettext-runtime/intl/xsize.h
@@ -27,6 +27,12 @@
 # include <stdint.h>
 #endif
 
+#if defined _WIN32
+# define _GL_INLINE_HEADER_BEGIN
+# define _GL_INLINE_HEADER_END
+# define _GL_INLINE static inline
+#endif
+
 #ifndef _GL_INLINE_HEADER_BEGIN
  #error "Please include config.h first."
 #endif
