diff --git a/gettext-runtime/gnulib-m4/iconv.m4 b/gettext-runtime/gnulib-m4/iconv.m4
index dc57b45..ff57298 100644
--- a/gettext-runtime/gnulib-m4/iconv.m4
+++ b/gettext-runtime/gnulib-m4/iconv.m4
@@ -122,6 +122,7 @@ AC_DEFUN([AM_ICONV_LINK],
      But here, the conversion always does transliteration (the suffixes
      "//TRANSLIT" and "//IGNORE" have no effect, nor does iconvctl()) and
      does not report when it does a non-identical conversion.  */
+  if(0)
   {
     iconv_t cd_utf8_to_88591 = iconv_open ("ISO-8859-1", "UTF-8");
     if (cd_utf8_to_88591 != (iconv_t)(-1))
