diff --color -ruN a/gettext-runtime/intl/dcigettext.c src/gettext-runtime/intl/dcigettext.c
--- a/gettext-runtime/intl/dcigettext.c	2021-05-26 16:27:55.420544597 +0200
+++ src/gettext-runtime/intl/dcigettext.c	2021-05-26 16:29:14.546516701 +0200
@@ -143,7 +143,7 @@
 # else
 #  if VMS
 #   define getcwd(buf, max) (getcwd) (buf, max, 0)
-#  else
+#  elif !(defined(__clang__) && defined(__BIONIC_FORTIFY))
 char *getcwd ();
 #  endif
 # endif
