diff --git a/Makefile.in b/Makefile.in
index ab71ec3..25fb7a0 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -323,7 +323,7 @@ top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = 1.5 gnu
 ACLOCAL_AMFLAGS = -I m4
-SUBDIRS = gnulib-local gettext-runtime libtextstyle gettext-tools
+SUBDIRS = gnulib-local $(VCPKG_GETTEXT_SUBDIRS)
 changelog_etc = \
   gettext-runtime/ChangeLog.0 \
   gettext-runtime/doc/ChangeLog.0 \
diff --git a/configure b/configure
index 79ae793..3685874 100755
--- a/configure
+++ b/configure
@@ -3040,7 +3040,7 @@ am__tar='{ find "$$tardir" -type d | LC_ALL=C sort; find "$$tardir" ! -type d -p
 
 
 
-subdirs="$subdirs gettext-runtime libtextstyle gettext-tools"
+subdirs="$subdirs gettext-runtime $(echo ${VCPKG_GETTEXT_SUBDIRS} | sed 's/gettext-runtime//')"
 
 
 
diff --git a/gettext-runtime/Makefile.in b/gettext-runtime/Makefile.in
index 365fd4e..8521d1e 100644
--- a/gettext-runtime/Makefile.in
+++ b/gettext-runtime/Makefile.in
@@ -2347,6 +2347,7 @@ ACLOCAL_AMFLAGS = -I m4 -I ../m4 -I gnulib-m4
 # specified.
 @ENABLE_LIBASPRINTF_TRUE@SUBDIR_libasprintf = libasprintf
 SUBDIRS = doc intl intl-java intl-csharp intl-d intl-modula2 gnulib-lib $(SUBDIR_libasprintf) src po man m4 tests install-tests
+SUBDIRS =     $(VCPKG_INTL)                                  gnulib-lib                       src po
 
 # Allow users to use "gnulib-tool --update".
 
diff --git a/gettext-runtime/configure b/gettext-runtime/configure
index 3fab6ea..79bf7ad 100755
--- a/gettext-runtime/configure
+++ b/gettext-runtime/configure
@@ -26018,7 +26018,7 @@ printf '%s\n' "$ac_res" >&6; }
 
                                         if { eval "gt_val=\$$gt_func_gnugettext_libc"; test "$gt_val" = "yes"; } \
            || { { eval "gt_val=\$$gt_func_gnugettext_libintl"; test "$gt_val" = "yes"; } \
-                && test "$PACKAGE" != gettext-runtime \
+                && test "$PACKAGE" != pristine-gettext-runtime \
                 && test "$PACKAGE" != gettext-tools \
                 && test "$PACKAGE" != libintl; }; then
           gt_use_preinstalled_gnugettext=yes
diff --git a/gettext-runtime/libasprintf/configure b/gettext-runtime/libasprintf/configure
index bd754f9..059e1c9 100755
--- a/gettext-runtime/libasprintf/configure
+++ b/gettext-runtime/libasprintf/configure
@@ -9,6 +9,7 @@
 #
 # This configure script is free software; the Free Software Foundation
 # gives unlimited permission to copy, distribute and modify it.
+exit 0;
 ## -------------------- ##
 ## M4sh Initialization. ##
 ## -------------------- ##
diff --git a/gettext-tools/Makefile.in b/gettext-tools/Makefile.in
index 18767d7..a201661 100644
--- a/gettext-tools/Makefile.in
+++ b/gettext-tools/Makefile.in
@@ -3834,7 +3834,7 @@ top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = 1.5 gnu
 ACLOCAL_AMFLAGS = -I m4 -I ../gettext-runtime/m4 -I ../m4 -I gnulib-m4 -I libgrep/gnulib-m4 -I libgettextpo/gnulib-m4 -I tests/gnulib-m4
-SUBDIRS = gnulib-lib libgrep src libgettextpo po its projects styles emacs misc man m4 tests system-tests gnulib-tests install-tests examples doc
+SUBDIRS = gnulib-lib libgrep src              po its projects styles       misc
 
 # Files imported from tree-sitter.
 
diff --git a/gettext-tools/configure b/gettext-tools/configure
index 049d8fe..d40476a 100755
--- a/gettext-tools/configure
+++ b/gettext-tools/configure
@@ -29766,7 +29766,7 @@ printf '%s\n' "$ac_res" >&6; }
                                         if { eval "gt_val=\$$gt_func_gnugettext_libc"; test "$gt_val" = "yes"; } \
            || { { eval "gt_val=\$$gt_func_gnugettext_libintl"; test "$gt_val" = "yes"; } \
                 && test "$PACKAGE" != gettext-runtime \
-                && test "$PACKAGE" != gettext-tools \
+                && test "$PACKAGE" != pristine-gettext-tools \
                 && test "$PACKAGE" != libintl; }; then
           gt_use_preinstalled_gnugettext=yes
         else
diff --git a/gettext-tools/examples/configure b/gettext-tools/examples/configure
index 0bb9816..3b523e7 100755
--- a/gettext-tools/examples/configure
+++ b/gettext-tools/examples/configure
@@ -11,6 +11,7 @@
 #
 # This configure script is free software; the Free Software Foundation
 # gives unlimited permission to copy, distribute and modify it.
+exit 0;
 ## -------------------- ##
 ## M4sh Initialization. ##
 ## -------------------- ##
diff --git a/gettext-tools/src/Makefile.in b/gettext-tools/src/Makefile.in
index e44593b..2ce98d5 100644
--- a/gettext-tools/src/Makefile.in
+++ b/gettext-tools/src/Makefile.in
@@ -5280,6 +5280,7 @@ msgfilter$(EXEEXT): $(msgfilter_OBJECTS) $(msgfilter_DEPENDENCIES) $(EXTRA_msgfi
 	$(AM_V_GEN)$(msgfilter_LINK) $(msgfilter_OBJECTS) $(msgfilter_LDADD) $(LIBS)
 ../../gettext-runtime/intl/$(am__dirstamp):
 	@$(MKDIR_P) ../../gettext-runtime/intl
+	@$(MAKE) -C ../../gettext-runtime/intl libgnuintl.h
 	@: >>../../gettext-runtime/intl/$(am__dirstamp)
 ../../gettext-runtime/intl/$(DEPDIR)/$(am__dirstamp):
 	@$(MKDIR_P) ../../gettext-runtime/intl/$(DEPDIR)
diff --git a/libtextstyle/Makefile.in b/libtextstyle/Makefile.in
index cdaf2b5..d76310c 100644
--- a/libtextstyle/Makefile.in
+++ b/libtextstyle/Makefile.in
@@ -2268,7 +2268,7 @@ AUTOMAKE_OPTIONS = 1.13 gnu
 ACLOCAL_AMFLAGS = -I m4 -I gnulib-m4
 
 # The list of subdirectories containing Makefiles.
-SUBDIRS = lib tests adhoc-tests install-tests doc
+SUBDIRS = lib
 
 # Allow users to use "gnulib-tool --update".
 
