diff --git a/gettext-runtime/intl/Makefile.in b/gettext-runtime/intl/Makefile.in
index 7999d3456..b3ef40340 100644
--- a/gettext-runtime/intl/Makefile.in
+++ b/gettext-runtime/intl/Makefile.in
@@ -252,7 +252,7 @@ am__libgnuintl_la_SOURCES_DIST = bindtextdom.c dcgettext.c dgettext.c \
 	gettext.c finddomain.c hash-string.c loadmsgcat.c \
 	localealias.c textdomain.c l10nflist.c explodename.c \
 	dcigettext.c dcngettext.c dngettext.c ngettext.c plural.y \
-	plural-exp.c localcharset.c threadlib.c lock.c relocatable.c \
+	plural-exp.c threadlib.c lock.c relocatable.c \
 	langprefs.c localename.c localename-table.c log.c printf.c \
 	setlocale.c setlocale-lock.c setlocale_null.c version.c \
 	xsize.c osdep.c intl-compat.c windows-mutex.c windows-rwlock.c \
@@ -264,7 +264,7 @@ am__objects_2 = bindtextdom.lo dcgettext.lo dgettext.lo gettext.lo \
 	finddomain.lo hash-string.lo loadmsgcat.lo localealias.lo \
 	textdomain.lo l10nflist.lo explodename.lo dcigettext.lo \
 	dcngettext.lo dngettext.lo ngettext.lo plural.lo plural-exp.lo \
-	localcharset.lo threadlib.lo lock.lo relocatable.lo \
+	threadlib.lo lock.lo relocatable.lo \
 	langprefs.lo localename.lo localename-table.lo log.lo \
 	printf.lo setlocale.lo setlocale-lock.lo setlocale_null.lo \
 	version.lo xsize.lo osdep.lo intl-compat.lo $(am__objects_1)
@@ -282,7 +282,7 @@ am__libintl_la_SOURCES_DIST = bindtextdom.c dcgettext.c dgettext.c \
 	gettext.c finddomain.c hash-string.c loadmsgcat.c \
 	localealias.c textdomain.c l10nflist.c explodename.c \
 	dcigettext.c dcngettext.c dngettext.c ngettext.c plural.y \
-	plural-exp.c localcharset.c threadlib.c lock.c relocatable.c \
+	plural-exp.c threadlib.c lock.c relocatable.c \
 	langprefs.c localename.c localename-table.c log.c printf.c \
 	setlocale.c setlocale-lock.c setlocale_null.c version.c \
 	xsize.c osdep.c intl-compat.c windows-mutex.c windows-rwlock.c \
@@ -1387,7 +1387,7 @@ AUTOMAKE_OPTIONS = 1.10 gnu no-dependencies
 
 # Miscellaneous files.
 EXTRA_DIST = gmo.h gettextP.h hash-string.h loadinfo.h plural-exp.h \
-	eval-plural.h localcharset.h lock.h windows-mutex.h \
+	eval-plural.h lock.h windows-mutex.h \
 	windows-rwlock.h windows-recmutex.h windows-once.h \
 	windows-initguard.h relocatable.h arg-nonnull.h attribute.h \
 	filename.h flexmember.h localename-table.in.h setlocale_null.h \
@@ -1543,7 +1543,7 @@ LIBINTLSOURCES = bindtextdom.c dcgettext.c dgettext.c gettext.c \
 	finddomain.c hash-string.c loadmsgcat.c localealias.c \
 	textdomain.c l10nflist.c explodename.c dcigettext.c \
 	dcngettext.c dngettext.c ngettext.c plural.y plural-exp.c \
-	localcharset.c threadlib.c lock.c relocatable.c langprefs.c \
+	threadlib.c lock.c relocatable.c langprefs.c \
 	localename.c localename-table.c log.c printf.c setlocale.c \
 	setlocale-lock.c setlocale_null.c version.c xsize.c osdep.c \
 	intl-compat.c $(am__append_2)
@@ -2016,8 +2016,6 @@ plural.lo: $(srcdir)/plural.c
 	$(AM_V_CC)$(LTCOMPILE) -c -o $@ $(srcdir)/plural.c
 plural-exp.lo: $(srcdir)/plural-exp.c
 	$(AM_V_CC)$(LTCOMPILE) -c -o $@ $(srcdir)/plural-exp.c
-localcharset.lo: $(srcdir)/localcharset.c
-	$(AM_V_CC)$(LTCOMPILE) -c -o $@ $(srcdir)/localcharset.c
 threadlib.lo: $(srcdir)/threadlib.c
 	$(AM_V_CC)$(LTCOMPILE) -c -o $@ $(srcdir)/threadlib.c
 lock.lo: $(srcdir)/lock.c
@@ -2074,7 +2072,6 @@ dngettext.lo:        ../config.h $(srcdir)/gettextP.h libgnuintl.h $(srcdir)/gmo
 ngettext.lo:         ../config.h $(srcdir)/gettextP.h libgnuintl.h $(srcdir)/gmo.h $(srcdir)/loadinfo.h $(srcdir)/lock.h $(srcdir)/windows-mutex.h $(srcdir)/windows-rwlock.h $(srcdir)/windows-recmutex.h $(srcdir)/windows-once.h $(srcdir)/windows-initguard.h
 plural.lo:           ../config.h $(srcdir)/plural-exp.h $(PLURAL_DEPS)
 plural-exp.lo:       ../config.h $(srcdir)/plural-exp.h
-localcharset.lo:     ../config.h $(srcdir)/localcharset.h
 threadlib.lo:        ../config.h
 lock.lo:             ../config.h $(srcdir)/lock.h $(srcdir)/windows-mutex.h $(srcdir)/windows-rwlock.h $(srcdir)/windows-recmutex.h $(srcdir)/windows-once.h $(srcdir)/windows-initguard.h
 relocatable.lo:      ../config.h $(srcdir)/relocatable.h
diff --git a/gettext-runtime/intl/localcharset.c b/gettext-runtime/intl/localcharset.c
index 1a1627888..3e6582f17 100644
--- a/gettext-runtime/intl/localcharset.c
+++ b/gettext-runtime/intl/localcharset.c
@@ -1021,7 +1021,7 @@ locale_charset (void)
 
 # else
 
-#  error "Add code for other platforms here."
+//#  error "Add code for other platforms here."
 
 # endif
 
