diff --git a/include/BoostInstall.cmake b/include/BoostInstall.cmake
index e4a65c43e..b20f3856a 100644
--- a/include/BoostInstall.cmake
+++ b/include/BoostInstall.cmake	
@@ -310,9 +310,9 @@ function(boost_install_target)
     foreach(dep IN LISTS link_libraries)
 
       if(dep MATCHES "^Boost::(.*)$")
-
+        string(APPEND CONFIG_FILE_CONTENTS "if(NOT boost_${CMAKE_MATCH_1}_FOUND)\n")
         string(APPEND CONFIG_FILE_CONTENTS "find_dependency(boost_${CMAKE_MATCH_1} ${__VERSION} EXACT)\n")
-
+        string(APPEND CONFIG_FILE_CONTENTS "endif()\n")
       elseif(dep STREQUAL "Threads::Threads")
 
         string(APPEND CONFIG_FILE_CONTENTS "set(THREADS_PREFER_PTHREAD_FLAG ON)\n")

