diff --git a/include/BoostInstall.cmake b/include/BoostInstall.cmake
index 49c0086fea..ad16dc5e0a 100644
--- a/include/BoostInstall.cmake	
+++ b/include/BoostInstall.cmake
@@ -69,7 +69,7 @@ function(__boost_install_set_output_name LIB TYPE VERSION)
 
   # prefix
   if(WIN32 AND TYPE STREQUAL "STATIC_LIBRARY")
-    set_target_properties(${LIB} PROPERTIES PREFIX "lib")
+    #set_target_properties(${LIB} PROPERTIES PREFIX "lib")
   endif()
 
   # toolset
