diff --git a/zzip/CMakeLists.txt b/zzip/CMakeLists.txt
index 52555d4..de06eae 100644
--- a/zzip/CMakeLists.txt
+++ b/zzip/CMakeLists.txt
@@ -27,7 +27,7 @@ set(ZZIP_VERSION "${PROJECT_VERSION}")
 set(ZZIP_PACKAGE_NAME "${PROJECT_NAME}lib")
 set(ZZIP_PACKAGE_VERSION "${PROJECT_VERSION}")
 
-if(ZZIPCOMPAT)
+if(UNIX)
 find_package ( UnixCommands REQUIRED ) # bash cp mv rm gzip tar
 endif()
 
