diff --git a/libmysofa.pc.cmake b/libmysofa.pc.cmake
index 3c597fa..b09b0b7 100644
--- a/libmysofa.pc.cmake
+++ b/libmysofa.pc.cmake
@@ -8,4 +8,4 @@ Libs: -L${libdir} -lmysofa
 Cflags: -I${includedir}
 
 Libs.private: @PKG_CONFIG_PRIVATELIBS@
-#Requires.private:
+Requires.private: zlib
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 37fe41f..66978a9 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -4,29 +4,13 @@ configure_file(config.h.in config.h)
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 if(NOT MSVC)
-  if(NOT WIN32)
-    find_library(MATH m)
-  else()
-    set(MATH "")
-  endif()
-  include(FindZLIB)
-  include_directories(${ZLIB_INCLUDE_DIRS})
+  find_library(HAVE_MATH m PATHS ${CMAKE_C_IMPLICIT_LINK_DIRECTORIES})
   set(PKG_CONFIG_PRIVATELIBS "-lm ${PKG_CONFIG_PRIVATELIBS}")
-  set(PKG_CONFIG_PRIVATELIBS "-lz ${PKG_CONFIG_PRIVATELIBS}")
 else()
   set(MATH "")
-  find_program(NUGET nuget)
-  if(NOT NUGET)
-    message(
-      FATAL
-      "Cannot find nuget command line tool.\nInstall it with e.g. choco install nuget.commandline"
-    )
-  else()
-    execute_process(COMMAND ${NUGET} install zlib)
-  endif()
-  include_directories(
-    ${PROJECT_SOURCE_DIR}/windows/third-party/zlib-1.3.1/include/)
 endif()
+find_package(ZLIB REQUIRED)
+set(ZLIB_LIBRARIES ZLIB::ZLIB)
 
 set(libsrc
     hrtf/reader.c
