diff --git a/CMakeLists.txt b/CMakeLists.txt
index 477064a..aa1a172 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -95,8 +95,12 @@ message("## ${OATPP_THIS_MODULE_NAME} module. Resolving dependencies...\n")
 ##############################
 ## Find libcurl dependency
 
-include(FindPkgConfig)
-pkg_check_modules(PKG_CURL REQUIRED libcurl)
+if (VCPKG_TOOLCHAIN)
+    find_package(CURL REQUIRED)
+else()
+    include(FindPkgConfig)
+    pkg_check_modules(PKG_CURL REQUIRED libcurl)
+endif()
 
 message("[libcurl] libcurl found:")
 message("[libcurl] LIBRARIES=${PKG_CURL_LIBRARIES}")
