diff --git a/CMakeLists.txt b/CMakeLists.txt
index c2141bd..dc1c81c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -363,7 +363,7 @@ if (TARGET crypto)
     message(STATUS "S2N found target: crypto")
     set(LINK_LIB "crypto")
 else()
-    find_package(crypto REQUIRED)
+    find_package(OpenSSL COMPONENTS Crypto REQUIRED)
     message(STATUS "Using libcrypto from the cmake path")
     set(LINK_LIB "AWS::crypto")
 endif()
@@ -497,7 +497,7 @@ if (S2N_INTERN_LIBCRYPTO)
     endif()
 else()
     # LINK_LIB is set above after checking targets. It handles the find_package craziness.
-    target_link_libraries(${PROJECT_NAME} PUBLIC ${LINK_LIB})
+    target_link_libraries(${PROJECT_NAME} PUBLIC OpenSSL::${LINK_LIB})
 endif()
 
 target_link_libraries(${PROJECT_NAME} PUBLIC ${OS_LIBS} m)
