diff --git a/vendor/core/CMakeLists.txt b/vendor/core/CMakeLists.txt
index e30b6b3..9dfa331 100644
--- a/vendor/core/CMakeLists.txt
+++ b/vendor/core/CMakeLists.txt
@@ -41,12 +41,12 @@ if(SOURCEMETA_CORE_INSTALL)
 endif()
 
 if(SOURCEMETA_CORE_REGEX)
-  find_package(BoostRegex REQUIRED)
+  find_package(Boost REQUIRED COMPONENTS regex) 
   add_subdirectory(src/core/regex)
 endif()
 
 if(SOURCEMETA_CORE_URI)
-  find_package(UriParser REQUIRED)
+  find_package(uriparser CONFIG REQUIRED char wchar_t)
   add_subdirectory(src/core/uri)
 endif()
 
