diff --git a/tools/gwenbuild/Makefile.am b/tools/gwenbuild/Makefile.am
index 14aa0e2..eb60988 100644
--- a/tools/gwenbuild/Makefile.am
+++ b/tools/gwenbuild/Makefile.am
@@ -28,11 +28,11 @@ noinst_HEADERS=\
 
 
 gwbuild_LDADD = \
-  $(top_builddir)/src/$(gwenhywfar_internal_libname) \
   types/libgwb_types.la \
   parser/libgwb_parser.la \
   tools/libgwb_tools.la \
   builders/libgwb_builders.la \
   buildctx/libgwb_build.la \
+  $(top_builddir)/src/$(gwenhywfar_internal_libname) \
   $(PROG_LIBRARIES)
 
