diff --git a/src/cmake/Config.cmake.in b/src/cmake/Config.cmake.in
index 4aee211..fa71d86 100644
--- a/src/cmake/Config.cmake.in
+++ b/src/cmake/Config.cmake.in
@@ -25,6 +25,9 @@ endif()
 if(@USE_WEBP@)
     find_dependency(WebP CONFIG)
 endif()
+if(@USE_LIBRAW@)
+    find_dependency(LibRaw)
+endif()
 
 # add here all the find_dependency() whenever switching to config based dependencies
 if (NOT @OPENIMAGEIO_CONFIG_DO_NOT_FIND_IMATH@ AND NOT OPENIMAGEIO_CONFIG_DO_NOT_FIND_IMATH)
