diff --git a/src/cmake/Config.cmake.in b/src/cmake/Config.cmake.in
index 0a6afeb..9017e02 100644
--- a/src/cmake/Config.cmake.in
+++ b/src/cmake/Config.cmake.in
@@ -1,6 +1,21 @@
 @PACKAGE_INIT@
 
 include(CMakeFindDependencyMacro)
+find_dependency(libheif CONFIG)
+find_dependency(PNG)
+find_dependency(unofficial-libsquish CONFIG)
+if(@USE_OPENCV@)
+    find_dependency(OpenCV CONFIG)
+endif()
+if(@USE_OPENCOLORIO@)
+    find_dependency(OpenColorIO CONFIG)
+endif()
+if(@USE_OPENJPEG@)
+    find_dependency(OpenJPEG CONFIG)
+endif()
+if(@USE_WEBP@)
+    find_dependency(WebP CONFIG)
+endif()
 
 # add here all the find_dependency() whenever switching to config based dependencies
 if (NOT @OPENIMAGEIO_CONFIG_DO_NOT_FIND_IMATH@ AND NOT OPENIMAGEIO_CONFIG_DO_NOT_FIND_IMATH)
