diff --git a/src/cmake/Config.cmake.in b/src/cmake/Config.cmake.in
index 9ee9c9d..6b9e7c2 100644
--- a/src/cmake/Config.cmake.in
+++ b/src/cmake/Config.cmake.in
@@ -31,6 +31,8 @@ endif()
 if (NOT @OPENIMAGEIO_CONFIG_DO_NOT_FIND_IMATH@ AND NOT OPENIMAGEIO_CONFIG_DO_NOT_FIND_IMATH)
     find_dependency(Imath @Imath_VERSION@
                     HINTS @Imath_DIR@)
+    find_dependency(OpenEXR @OpenEXR_VERSION@
+                        HINTS @OpenEXR_DIR@)
 endif ()

 if (NOT @fmt_LOCAL_BUILD@ AND NOT @OIIO_INTERNALIZE_FMT@)
