diff --git a/src/cmake/externalpackages.cmake b/src/cmake/externalpackages.cmake
index 1a17d1e..2ba4264 100644
--- a/src/cmake/externalpackages.cmake
+++ b/src/cmake/externalpackages.cmake
@@ -259,7 +259,7 @@ checked_find_package (Field3D
                    DEPS         HDF5
                    DEFINITIONS  -DUSE_FIELD3D=1)
 checked_find_package (GIF 4)
-checked_find_package (Libheif 1.3)  # For HEIF/HEIC format
+checked_find_package (libheif CONFIG)  # For HEIF/HEIC format
 checked_find_package (LibRaw)
 checked_find_package (OpenJPEG CONFIG)
 checked_find_package (OpenVDB 5.0
diff --git a/src/heif.imageio/CMakeLists.txt b/src/heif.imageio/CMakeLists.txt
index cb5e5f9..f773fd5 100644
--- a/src/heif.imageio/CMakeLists.txt
+++ b/src/heif.imageio/CMakeLists.txt
@@ -5,7 +5,7 @@
 if (LIBHEIF_FOUND)
     add_oiio_plugin (heifinput.cpp heifoutput.cpp
                      INCLUDE_DIRS ${LIBHEIF_INCLUDES}
-                     LINK_LIBRARIES ${LIBHEIF_LIBRARIES}
+                     LINK_LIBRARIES heif
                      DEFINITIONS "-DUSE_HEIF=1")
 else ()
     message (WARNING "heif plugin will not be built")
