diff --git a/src/cmake/externalpackages.cmake b/src/cmake/externalpackages.cmake
index 1240f22..b7772e1 100644
--- a/src/cmake/externalpackages.cmake
+++ b/src/cmake/externalpackages.cmake
@@ -387,5 +387,7 @@ macro (find_or_download_fmt)
     checked_find_package (fmt REQUIRED)
 endmacro()
 
-find_or_download_fmt()
+find_package(fmt CONFIG REQUIRED)
+get_property(FMT_INCLUDES TARGET fmt::fmt PROPERTY INTERFACE_INCLUDE_DIRECTORIES)
+checked_find_package (fmt CONFIG REQUIRED)
 include_directories (${FMT_INCLUDES})
