diff --git a/src/cmake/oiio_macros.cmake b/src/cmake/oiio_macros.cmake
index 9e54f67..d9c9437 100644
--- a/src/cmake/oiio_macros.cmake
+++ b/src/cmake/oiio_macros.cmake
@@ -142,7 +142,7 @@ macro (fancy_add_executable)
             target_link_libraries (${_target_NAME} PRIVATE ${_target_LINK_LIBRARIES})
         endif ()
         set_target_properties (${_target_NAME} PROPERTIES FOLDER "Tools")
-        install_targets (${_target_NAME})
+        install(TARGETS ${_target_NAME} RUNTIME DESTINATION tools/openimageio)
     else ()
         message (STATUS "${ColorRed}Disabling ${_target_NAME} ${ColorReset}")
     endif ()
