diff --git a/src/cmake/externalpackages.cmake b/src/cmake/externalpackages.cmake
index 1fc2059..62c4efb 100644
--- a/src/cmake/externalpackages.cmake
+++ b/src/cmake/externalpackages.cmake
@@ -222,7 +222,7 @@ if (LibRaw_FOUND AND LibRaw_VERSION VERSION_LESS 0.20 AND CMAKE_CXX_STANDARD VER
     # set (LIBRAW_FOUND 0)
 endif ()
 
-checked_find_package (OpenJPEG VERSION_MIN 2.0)
+checked_find_package (OpenJPEG PREFER_CONFIG VERSION_MIN 2.0)
 
 checked_find_package (OpenVDB
                       VERSION_MIN 5.0
