diff --git a/src/cmake/externalpackages.cmake b/src/cmake/externalpackages.cmake
index 67a3629..86a480f 100644
--- a/src/cmake/externalpackages.cmake
+++ b/src/cmake/externalpackages.cmake
@@ -230,9 +230,12 @@ if (APPLE AND LIBHEIF_VERSION VERSION_GREATER_EQUAL 1.10 AND LIBHEIF_VERSION VER
     set (Libheif_FOUND 0)
 endif ()
 
-checked_find_package (LibRaw
+checked_find_package (libraw CONFIG
                       VERSION_MIN 0.18
                       PRINT LibRaw_r_LIBRARIES)
+set(LibRaw_FOUND ${libraw_FOUND})
+set(LibRaw_VERSION ${libraw_VERSION})
+set(LibRaw_r_LIBRARIES libraw::raw_r)
 if (LibRaw_FOUND AND LibRaw_VERSION VERSION_LESS 0.20 AND CMAKE_CXX_STANDARD VERSION_GREATER_EQUAL 17)
     message (STATUS "${ColorYellow}WARNING When building for C++17, LibRaw should be 0.20 or higher (found ${LibRaw_VERSION}). You may get errors, depending on the compiler.${ColorReset}")
     # Currently, we issue the above warning and let them take their chances.
