diff --git a/src/cmake/externalpackages.cmake b/src/cmake/externalpackages.cmake
index b50a923..cec6e43 100644
--- a/src/cmake/externalpackages.cmake
+++ b/src/cmake/externalpackages.cmake
@@ -168,6 +168,11 @@ set(OPENCOLORIO_FOUND "${OpenColorIO_FOUND}")
 if (NOT OPENCOLORIO_FOUND)
     set (OPENCOLORIO_FOUND 0)
 endif ()
+if(USE_OPENCOLORIO AND TARGET Imath::Imath AND OIIO_USING_IMATH STREQUAL "2")
+    message(FATAL_ERROR
+        "OpenColorIO and OpenEXR use incompatible versions of Imath. "
+        "You cannot use openimageio[opencolorio] for this configuration.")
+endif()
 set(OPENCV_INCLUDES "")
 set(OPENCV_LIBRARIES opencv_core)
 checked_find_package (OpenCV CONFIG
