diff --git a/extensions/CMakeLists.txt b/extensions/CMakeLists.txt
index e545b6d..281e71b 100644
--- a/extensions/CMakeLists.txt
+++ b/extensions/CMakeLists.txt
@@ -1,3 +1,8 @@
+if(BUILD_SHARED_LIBS)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Dlibcmark_gfm_EXPORTS")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Dlibcmark_gfm_EXPORTS")
+endif()
+
 set(LIBRARY "libcmark-gfm-extensions")
 set(STATICLIBRARY "libcmark-gfm-extensions_static")
 set(LIBRARY_SOURCES
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 84dd2a0..e5e237e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -154,6 +154,10 @@ endif()
 
 set(libdir lib${LIB_SUFFIX})
 
+if(NOT BUILD_TOOLS)
+  set_target_properties(${PROGRAM} PROPERTIES EXCLUDE_FROM_ALL 1)
+  set(PROGRAM "")
+endif()
 install(TARGETS ${PROGRAM} ${CMARK_INSTALL}
   EXPORT cmark-gfm
   RUNTIME DESTINATION bin
