diff --git a/CMakeLists.txt b/CMakeLists.txt
index e90fcdd..17c79e8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,12 +26,19 @@ add_library(LercLib ${SOURCES})
 endif()
 
 install(
     TARGETS Lerc
+    EXPORT LercTargets
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
     PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
 )
+target_include_directories(Lerc PUBLIC "$<INSTALL_INTERFACE:include>")
+install(EXPORT LercTargets
+  FILE unofficial-lerc-config.cmake
+  NAMESPACE unofficial::Lerc::
+  DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/unofficial-lerc"
+)
 
 # Configure and install pkgconfig file
 configure_file(Lerc.pc.in ${CMAKE_CURRENT_BINARY_DIR}/Lerc.pc @ONLY)
