diff --git a/msvc/hunspell.vcxproj b/msvc/hunspell.vcxproj
index 660a082..e1757db 100644
--- a/msvc/hunspell.vcxproj
+++ b/msvc/hunspell.vcxproj
@@ -122,7 +122,6 @@
       <Optimization>Disabled</Optimization>
       <AdditionalIncludeDirectories>..\src\hunspell;..\src\parsers;.;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
       <PreprocessorDefinitions>W32;WIN32;_DEBUG;_CONSOLE;HUNSPELL_STATIC;_CRT_SECURE_NO_WARNINGS;%(PreprocessorDefinitions)</PreprocessorDefinitions>
-      <MinimalRebuild>true</MinimalRebuild>
       <BasicRuntimeChecks>EnableFastChecks</BasicRuntimeChecks>
       <RuntimeLibrary>MultiThreadedDebug</RuntimeLibrary>
       <PrecompiledHeader />
