diff --git a/src/libqhull/global.c b/src/libqhull/global.c
index e7de8e6..81be9df 100644
--- a/src/libqhull/global.c
+++ b/src/libqhull/global.c
@@ -636,7 +636,7 @@ void qh_init_qhull_command(int argc, char *argv[]) {
 */
 void qh_initflags(char *command) {
   int k, i, lastproject;
-  char *s= command, *t, *prev_s, *start, key, *lastwarning= NULL;
+  char *s= command, *t = NULL, *prev_s, *start, key, *lastwarning= NULL;
   boolT isgeom= False, wasproject;
   realT r;
 
diff --git a/src/libqhull_r/global_r.c b/src/libqhull_r/global_r.c
index 3e6919f..8d13e47 100644
--- a/src/libqhull_r/global_r.c
+++ b/src/libqhull_r/global_r.c
@@ -610,7 +610,7 @@ void qh_init_qhull_command(qhT *qh, int argc, char *argv[]) {
 */
 void qh_initflags(qhT *qh, char *command) {
   int k, i, lastproject;
-  char *s= command, *t, *prev_s, *start, key, *lastwarning= NULL;
+  char *s= command, *t = NULL, *prev_s, *start, key, *lastwarning= NULL;
   boolT isgeom= False, wasproject;
   realT r;
 
