diff --git a/CMakeLists.txt b/CMakeLists.txt
index 122fe93..b571c38 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -360,11 +360,6 @@ ENDIF()
 # embedded builds this is not an issue as they will only build the configuration
 # used in the host application.
 MARK_AS_ADVANCED(PA_LIBNAME_ADD_SUFFIX)
-IF(MSVC AND PA_BUILD_STATIC AND PA_BUILD_SHARED AND NOT PA_LIBNAME_ADD_SUFFIX)
-  MESSAGE(WARNING "Building both shared and static libraries, and avoiding the suffix _static will lead to a name conflict")
-  SET(PA_LIBNAME_ADD_SUFFIX ON CACHE BOOL "Forcing use of suffix _static to avoid name conflict between static and import library" FORCE)
-  MESSAGE(WARNING "PA_LIBNAME_ADD_SUFFIX was set to ON")
-ENDIF()
 
 SET(PA_TARGETS "")
 
