From 086a76ad7c17060d504371ea724cf5d651d43eb3 Mon Sep 17 00:00:00 2001
From: Mikhail Paulyshka <me@mixaill.tk>
Date: Thu, 28 Dec 2017 16:07:51 +0300
Subject: [PATCH] fix trietool for MSVC runtime

---
 tools/trietool.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tools/trietool.c b/tools/trietool.c
index 4cfde2a..7c90905 100644
--- a/tools/trietool.c
+++ b/tools/trietool.c
@@ -589,12 +589,12 @@ string_trim (char *s)
     char   *p;
 
     /* skip leading white spaces */
-    while (*s && isspace (*s))
+    while (*s && isspace ((unsigned char)*s))
         ++s;
 
     /* trim trailing white spaces */
     p = s + strlen (s) - 1;
-    while (isspace (*p))
+    while (isspace ((unsigned char)*p))
         --p;
     *++p = '\0';
 
-- 
2.14.1.windows.1

