diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3f5c0d3..87f98a7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -467,6 +467,7 @@ endif()
 # XXX for a normal full distribution we'll need to figure out
 # XXX how to build both shared and static libraries.
 # Probably leverage that to build a local VALGRIND=1 library for testing too.
+if (BUILD_SHARED_LIBS)
 add_library(${PROJECT_NAME}
     ${JSON_C_SOURCES}
     ${JSON_C_HEADERS}
@@ -484,9 +485,10 @@ target_include_directories(${PROJECT_NAME}
 )
 
 target_link_libraries(${PROJECT_NAME} PUBLIC ${CMAKE_REQUIRED_LIBRARIES})
+endif ()
 
 # Allow to build static and shared libraries at the same time
-if (BUILD_STATIC_LIBS AND BUILD_SHARED_LIBS)
+if (BUILD_STATIC_LIBS)
     set(STATIC_LIB ${PROJECT_NAME}-static)
     add_library(${STATIC_LIB} STATIC
         ${JSON_C_SOURCES}
