diff --git a/meson.build b/meson.build
index 25d42ce..60a8354 100644
--- a/meson.build
+++ b/meson.build
@@ -1,6 +1,6 @@
 project('wayland-protocols',
 	version: '1.49',
-	meson_version: '>= 0.58.0',
+	meson_version: '>= 0.62.0',
 	license: 'MIT/Expat',
 )
 
@@ -8,15 +8,7 @@ wayland_protocols_version = meson.project_version()
 
 fs = import('fs')
 
-dep_scanner = dependency('wayland-scanner',
-    version: get_option('tests') ? '>=1.25.0' : '>=1.22.90',
-    required: get_option('tests'),
-    native: true,
-    fallback: 'wayland'
-)
-if dep_scanner.found()
-	prog_scanner = find_program(dep_scanner.get_variable(pkgconfig: 'wayland_scanner', internal: 'wayland_scanner'))
-endif
+prog_scanner = find_program('wayland_scanner')
 
 stable_protocols = {
 	'linux-dmabuf': {'v1': []},
@@ -178,7 +170,7 @@ endforeach
 
 include_dirs = []
 headers = []
-if dep_scanner.found()
+if true
 	subdir('include/wayland-protocols')
 	include_dirs = ['include']
 endif
