diff --git a/symengine/CMakeLists.txt b/symengine/CMakeLists.txt
index 7eeb4960..3fb8e7b5 100644
--- a/symengine/CMakeLists.txt
+++ b/symengine/CMakeLists.txt
@@ -263,6 +263,7 @@ install(TARGETS symengine
             RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
             ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
             LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+            INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
         )
 install(FILES
     "${symengine_BINARY_DIR}/symengine/symengine_config.h"
