diff --git a/src/bin/CMakeLists.txt b/src/bin/CMakeLists.txt
index d0592a320..07df9daf7 100644
--- a/src/bin/CMakeLists.txt
+++ b/src/bin/CMakeLists.txt
@@ -269,6 +269,8 @@ set(PUBLIC_HEADERS
     ../inc/msquic_posix.h
     ../inc/quic_sal_stub.h)
 
+file(GLOB PUBLIC_HEADERS "../inc/*.h" "../inc/*.hpp")
+
 install(TARGETS msquic EXPORT msquic
     RUNTIME DESTINATION bin
     ARCHIVE DESTINATION lib
