diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8406ab9..24c6328 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -144,6 +144,7 @@ install_basic_package_files(${PROJECT_NAME}
   NO_CHECK_REQUIRED_COMPONENTS_MACRO
   DEPENDENCIES ${OSQP_EIGEN_EXPORTED_DEPENDENCIES})
 
+if(0)
 # Install package.xml to share
 install(FILES package.xml DESTINATION share/cmake/osqp-eigen)
 
@@ -152,3 +153,4 @@ include(AddOsqpEigenUnitTest)
 add_subdirectory(tests)
 
 include(AddUninstallTarget)
+endif()
