diff --git a/gloo/transport/tcp/device.cc b/gloo/transport/tcp/device.cc
index cda44c8..2df6167 100644
--- a/gloo/transport/tcp/device.cc
+++ b/gloo/transport/tcp/device.cc
@@ -12,6 +12,7 @@
 #include <netdb.h>
 #include <netinet/in.h>
 #include <string.h>
+#include <array>
 
 #include "gloo/common/linux.h"
 #include "gloo/common/logging.h"
diff --git a/gloo/transport/tcp/tls/pair.cc b/gloo/transport/tcp/tls/pair.cc
index e8f95f7..b0277aa 100644
--- a/gloo/transport/tcp/tls/pair.cc
+++ b/gloo/transport/tcp/tls/pair.cc
@@ -17,6 +17,7 @@
 
 #include <cstring>
 #include <poll.h>
+#include <array>
 
 namespace gloo {
 namespace transport {
