diff --git a/urdf_parser/CMakeLists.txt b/urdf_parser/CMakeLists.txt
index 8623edf..5eeaa81 100644
--- a/urdf_parser/CMakeLists.txt
+++ b/urdf_parser/CMakeLists.txt
@@ -39,24 +39,44 @@ if(BUILD_TESTING)
   add_subdirectory(test)
 endif()
 
-INSTALL(TARGETS urdfdom_model DESTINATION ${CMAKE_INSTALL_LIBDIR} EXPORT urdfdom_modelTargets)
+INSTALL(TARGETS urdfdom_model
+  EXPORT urdfdom_modelTargets
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
 install(EXPORT urdfdom_modelTargets
     NAMESPACE urdfdom::
     DESTINATION ${CMAKE_CONFIG_INSTALL_DIR}
 )
-INSTALL(TARGETS urdfdom_world DESTINATION ${CMAKE_INSTALL_LIBDIR} EXPORT urdfdom_worldTargets)
+INSTALL(TARGETS urdfdom_world
+  EXPORT urdfdom_worldTargets
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
 install(EXPORT urdfdom_worldTargets
     NAMESPACE urdfdom::
     DESTINATION ${CMAKE_CONFIG_INSTALL_DIR}
 )
 INSTALL(TARGETS check_urdf urdf_to_graphiz urdf_mem_test
   DESTINATION ${CMAKE_INSTALL_BINDIR})
-INSTALL(TARGETS urdfdom_sensor DESTINATION ${CMAKE_INSTALL_LIBDIR} EXPORT urdfdom_sensorTargets)
+INSTALL(TARGETS urdfdom_sensor
+  EXPORT urdfdom_sensorTargets
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
 install(EXPORT urdfdom_sensorTargets
     NAMESPACE urdfdom::
     DESTINATION ${CMAKE_CONFIG_INSTALL_DIR}
 )
-INSTALL(TARGETS urdfdom_model_state DESTINATION ${CMAKE_INSTALL_LIBDIR} EXPORT urdfdom_model_stateTargets)
+INSTALL(TARGETS urdfdom_model_state
+  EXPORT urdfdom_model_stateTargets
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
 install(EXPORT urdfdom_model_stateTargets
     NAMESPACE urdfdom::
     DESTINATION ${CMAKE_CONFIG_INSTALL_DIR}
