diff --git a/src/vanillapdf/utils/util.h b/src/vanillapdf/utils/util.h
index aae1052..d695e49 100644
--- a/src/vanillapdf/utils/util.h
+++ b/src/vanillapdf/utils/util.h
@@ -50,6 +50,7 @@ private:
     ScopeGuardFactory();
 };
 
+#if 0
 // The AutoSubscribe concept is something I was thinking about for a long time.
 // Unfortunately I am still not able to create a working POC, however I want to keep the code.
 // In case I will be thinking about such functionality in the future, this could be the starting point.
@@ -102,6 +103,7 @@ private:
     T _observable;
     U* _observer = nullptr;
 };
+#endif
 
 #if (__cplusplus < 201402L) && !defined(COMPILER_MICROSOFT_VISUAL_STUDIO)
     // Use custom implementation if not
