diff -r 7c4281fe1229 -r c71622e89674 cmake/SearchForStuff.cmake
--- a/cmake/SearchForStuff.cmake
+++ b/cmake/SearchForStuff.cmake
@@ -357,6 +357,8 @@
   pkg_check_modules(OGRE OGRE>=${MIN_OGRE_VERSION})
 
   if (NOT OGRE_FOUND)
+    # Workaround for CMake bug https://gitlab.kitware.com/cmake/cmake/issues/17135
+    unset(OGRE_FOUND CACHE)
     # If OGRE was not found, try with the standard find_package(OGRE)
     find_package(OGRE COMPONENTS RTShaderSystem Terrain Overlay Paging)
     # Add each component include directories to OGRE_INCLUDE_DIRS because
@@ -370,6 +372,7 @@
     list(APPEND OGRE_LIBRARIES ${OGRE_Terrain_LIBRARIES})
     list(APPEND OGRE_LIBRARIES ${OGRE_Overlay_LIBRARIES})
     list(APPEND OGRE_LIBRARIES ${OGRE_Paging_LIBRARIES})
+    set(OGRE_PLUGINDIR ${OGRE_PLUGIN_DIR})
   endif ()
 
   if (NOT OGRE_FOUND)
