diff -r 7c4281fe1229 -r b7e6f837cc8b cmake/SearchForStuff.cmake
--- a/cmake/SearchForStuff.cmake
+++ b/cmake/SearchForStuff.cmake
@@ -91,12 +91,6 @@
 ########################################
 # Find packages
 
-# In Visual Studio we use configure.bat to trick all path cmake
-# variables so let's consider that as a replacement for pkgconfig
-if (MSVC)
-  set (PKG_CONFIG_FOUND TRUE)
-endif()
-
 find_package(CURL)
 if (CURL_FOUND)
   # FindCURL.cmake distributed with CMake exports
@@ -109,6 +103,12 @@
   set(CURL_INCLUDEDIR ${CURL_INCLUDE_DIRS})
 endif ()
 
+# In Visual Studio we use configure.bat to trick all path cmake
+# variables so let's consider that as a replacement for pkgconfig
+if (MSVC)
+  set (PKG_CONFIG_FOUND TRUE)
+endif()
+
 if (PKG_CONFIG_FOUND)
   if (NOT CURL_FOUND)
     pkg_check_modules(CURL libcurl)
