Subject: [PATCH] patch 0007
---
Index: configure
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/configure b/configure
--- a/configure	(revision 7e37a389341db853f0116849f1c12331a12dd215)
+++ b/configure	(revision eb9a92addf12136c78bd0e558b68fa45b0c963ae)
@@ -4587,6 +4587,10 @@
             -march=*)             ;;
             -lz)                  echo zlib.lib ;;
             -lx264)               echo libx264.lib ;;
+            -lx265)               echo libx265.lib ;;
+            -lmp3lame)            echo libmp3lame.lib ;;
+            -liconv)              echo iconv.lib ;;
+            -lm)                  ;;
             -lstdc++)             ;;
             -l*)                  echo ${flag#-l}.lib ;;
             -LARGEADDRESSAWARE)   echo $flag ;;
