From 7ca15b7f337b00a71384f748d05fce89fcf15a8d Mon Sep 17 00:00:00 2001
From: "Matthias C. M. Troffaes" <matthias.troffaes@gmail.com>
Date: Tue, 4 May 2021 13:03:17 +0100
Subject: [PATCH] configure: dnn needs avformat

---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index 9e45c0822c..8725a94f8a 100755
--- a/configure
+++ b/configure
@@ -2660,7 +2660,7 @@ cbs_vp9_select="cbs"
 dct_select="rdft"
 dirac_parse_select="golomb"
 dnn_suggest="libtensorflow libopenvino"
-dnn_deps="swscale"
+dnn_deps="avformat swscale"
 error_resilience_select="me_cmp"
 faandct_deps="faan"
 faandct_select="fdctdsp"
-- 
2.25.1

