
diff --git a/libavcodec/qsv.c b/libavcodec/qsv.c
--- a/libavcodec/qsv.c
+++ b/libavcodec/qsv.c
@@ -383,7 +383,7 @@
 int ff_qsv_init_internal_session(AVCodecContext *avctx, QSVSession *qs,
                                  const char *load_plugins, int gpu_copy)
 {
-    mfxIMPL          impl = MFX_IMPL_AUTO_ANY;
+    mfxIMPL           impl = MFX_IMPL_AUTO_ANY;
     mfxVersion        ver = { { QSV_VERSION_MINOR, QSV_VERSION_MAJOR } };
     mfxInitParam init_par = { MFX_IMPL_AUTO_ANY };
 
@@ -393,12 +393,38 @@
 #if QSV_VERSION_ATLEAST(1, 16)
     init_par.GPUCopy        = gpu_copy;
 #endif
-    init_par.Implementation = impl;
+
+    // https://trac.ffmpeg.org/ticket/7933
+    // try d3d11
+    init_par.Implementation = MFX_IMPL_HARDWARE_ANY | MFX_IMPL_VIA_D3D11;
+    impl = init_par.Implementation;
     init_par.Version        = ver;
     ret = MFXInitEx(init_par, &qs->session);
-    if (ret < 0)
+    if (ret >= 0)
+        av_log(avctx, AV_LOG_DEBUG, "MFX Initialized with HARDWARE_ANY | VIA_D3D11\n");
+
+    if (ret < 0) {
+        // try d3d9
+        init_par.Implementation = MFX_IMPL_HARDWARE_ANY | MFX_IMPL_VIA_D3D9;
+        impl = init_par.Implementation;
+        ret = MFXInitEx(init_par, &qs->session);
+        if(ret >= 0)
+            av_log(avctx, AV_LOG_DEBUG, "MFX Initialized with HARDWARE_ANY | VIA_D3D9\n");
+    }
+
+    if (ret < 0) {
+        // try any
+        init_par.Implementation = MFX_IMPL_AUTO_ANY;
+        impl = init_par.Implementation;
+        ret = MFXInitEx(init_par, &qs->session);
+        if(ret >= 0)
+            av_log(avctx, AV_LOG_DEBUG, "MFX Initialized with AUTO_ANY\n");
+    }
+
+    if (ret < 0) {
         return ff_qsv_print_error(avctx, ret,
                                   "Error initializing an internal MFX session");
+    }
 
 #ifdef AVCODEC_QSV_LINUX_SESSION_HANDLE
     ret = ff_qsv_set_display_handle(avctx, qs);
