--- a/libavdevice/opengl_enc.c
+++ b/libavdevice/opengl_enc.c
@@ -25,7 +25,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <unistd.h>
+//#include <unistd.h>
 #include <stddef.h>

 #include "config.h"
@@ -33,6 +33,8 @@
 #if HAVE_WINDOWS_H
 #define WIN32_LEAN_AND_MEAN
 #include <windows.h>
+#else
+#include <unistd.h>
 #endif
 #if HAVE_OPENGL_GL3_H
 #include <OpenGL/gl3.h>
