From 5f7c72d6c93fb1f4a730e2315fff6e5eb9bb95c8 Mon Sep 17 00:00:00 2001
From: Anonymous Maarten <anonymous.maarten@gmail.com>
Date: Wed, 27 Sep 2017 15:50:40 +0200
Subject: [PATCH] configure: create .lib static libraries using MSVC compiler
 (instead of .a)

---
 configure | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure b/configure
index ba38a73906..0a3f308323 100755
--- a/configure
+++ b/configure
@@ -4890,15 +4890,17 @@
         ;;
     win32|win64)
         disable symver
-        if enabled shared; then
+#        if enabled shared; then
             # Link to the import library instead of the normal static library
             # for shared libs.
             LD_LIB='%.lib'
             # Cannot build both shared and static libs with MSVC or icl.
-            disable static
-        fi
+#            disable static
+#        fi
         enabled x86_32 && check_ldflags -LARGEADDRESSAWARE
         shlibdir_default="$bindir_default"
+        LIBPREF=""
+        LIBSUF=".lib"
         SLIBPREF=""
         SLIBSUF=".dll"
         SLIBNAME_WITH_VERSION='$(SLIBPREF)$(FULLNAME)-$(LIBVERSION)$(SLIBSUF)'
-- 
2.14.1.windows.1

