--- a/configure
+++ b/configure
@@ -3856,6 +3856,7 @@
             -lx265)               echo libx265.lib ;;
             -lfontconfig)         echo fontconfig.lib ;;
             -lopenjpeg)           echo openjp2.lib ;;
+            -lmp3lame)            echo libmp3lame.lib ;;
             -l*)                  echo ${flag#-l}.lib ;;
             -LARGEADDRESSAWARE)   echo $flag ;;
             -L*)                  echo -libpath:${flag#-L} ;;
