diff --git a/configure b/configure
index 19a2556..6f99e4e 100644
--- a/configure
+++ b/configure
@@ -1882,6 +1882,7 @@ HEADERS_LIST="
     openjpeg_2_1_openjpeg_h
     openjpeg_2_0_openjpeg_h
     openjpeg_1_5_openjpeg_h
+    opj_config_h
     OpenGL_gl3_h
     poll_h
     sndio_h
@@ -5820,7 +5821,7 @@ enabled libopenjpeg       && { { check_lib openjpeg-2.1/openjpeg.h opj_version -
                                { check_lib openjpeg-2.0/openjpeg.h opj_version -lopenjp2 -DOPJ_STATIC && add_cppflags -DOPJ_STATIC; } ||
                                { check_lib openjpeg-1.5/openjpeg.h opj_version -lopenjpeg -DOPJ_STATIC && add_cppflags -DOPJ_STATIC; } ||
                                { check_lib openjpeg.h opj_version -lopenjpeg -DOPJ_STATIC && add_cppflags -DOPJ_STATIC; } ||
-                               die "ERROR: libopenjpeg not found"; }
+                               die "ERROR: libopenjpeg not found"; } && check_header opj_config.h
 enabled libopenmpt        && require_pkg_config "libopenmpt >= 0.2.6557" libopenmpt/libopenmpt.h openmpt_module_create
 enabled libopus           && { use_pkg_config opus opus_multistream.h opus_multistream_decoder_create ||
                                { require libopus opus/opus_multistream.h opus_multistream_decoder_create -lopus &&
diff --git a/libavcodec/libopenjpegdec.c b/libavcodec/libopenjpegdec.c
index ce4e2b0..45c4cb7 100644
--- a/libavcodec/libopenjpegdec.c
+++ b/libavcodec/libopenjpegdec.c
@@ -48,8 +48,15 @@
 #  define OPENJPEG_MAJOR_VERSION 2
 #  define OPJ(x) OPJ_##x
 #else
-#  define OPENJPEG_MAJOR_VERSION 1
-#  define OPJ(x) x
+#  if HAVE_OPJ_CONFIG_H
+#    include <opj_config.h>
+#    define OPENJPEG_MAJOR_VERSION OPJ_VERSION_MAJOR
+#    define OPJ(x) OPJ_##x
+#    define HAVE_OPENJPEG_2_1_OPENJPEG_H 1
+#  else
+#    define OPENJPEG_MAJOR_VERSION 1
+#    define OPJ(x) x
+#  endif
 #endif
 
 #define JP2_SIG_TYPE    0x6A502020
diff --git a/libavcodec/libopenjpegenc.c b/libavcodec/libopenjpegenc.c
index 4a12729..c44ee16 100644
--- a/libavcodec/libopenjpegenc.c
+++ b/libavcodec/libopenjpegenc.c
@@ -46,8 +46,15 @@
 #  define OPENJPEG_MAJOR_VERSION 2
 #  define OPJ(x) OPJ_##x
 #else
-#  define OPENJPEG_MAJOR_VERSION 1
-#  define OPJ(x) x
+#  if HAVE_OPJ_CONFIG_H
+#    include <opj_config.h>
+#    define OPENJPEG_MAJOR_VERSION OPJ_VERSION_MAJOR
+#    define OPJ(x) OPJ_##x
+#    define HAVE_OPENJPEG_2_1_OPENJPEG_H 1
+#  else
+#    define OPENJPEG_MAJOR_VERSION 1
+#    define OPJ(x) x
+#  endif
 #endif
 
 typedef struct LibOpenJPEGContext {
