--- a/ffmpeg_dxva2.c	2017-07-29 19:49:29.000000000 +0200
+++ b/ffmpeg_dxva2.c	2018-09-02 20:33:48.248716100 +0200
@@ -16,12 +16,14 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
-#include <windows.h>
-
 #ifdef _WIN32_WINNT
 #undef _WIN32_WINNT
 #endif
 #define _WIN32_WINNT 0x0600
+#define WIN32_LEAN_AND_MEAN
+
+#include <windows.h>
+
 #define DXVA2API_USE_BITFIELDS
 #define COBJMACROS
 
