diff --git a/Makefile.in b/Makefile.in
index d00b565..c465d0b 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -294,7 +294,7 @@ libhogweed.a: $(hogweed_OBJS)
 
 %.$(OBJEXT): %.asm $(srcdir)/m4-utils.m4 $(srcdir)/asm.m4 config.m4 machine.m4
 	$(M4) $(srcdir)/m4-utils.m4 $(srcdir)/asm.m4 config.m4 machine.m4 $< >$*.s
-	$(COMPILE) -c $*.s
+	$(COMPILE_AS) -c $*.s -o $@
 
 %.$(OBJEXT): %.c
 	$(COMPILE) -c $< \
diff --git a/aclocal.m4 b/aclocal.m4
index 1d218a0..1ff02c8 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -301,7 +301,7 @@ AC_DEFUN([GMP_TRY_ASSEMBLE],
 [cat >conftest.s <<EOF
 [$1]
 EOF
-gmp_assemble="$CC $CFLAGS $CPPFLAGS -c conftest.s >conftest.out 2>&1"
+gmp_assemble="$CCAS $CPPFLAGS $ASMFLAGS -c conftest.s >conftest.out 2>&1"
 if AC_TRY_EVAL(gmp_assemble); then
   cat conftest.out >&AC_FD_CC
   ifelse([$2],,:,[$2])
diff --git a/config.make.in b/config.make.in
index f8e1f74..4668884 100644
--- a/config.make.in
+++ b/config.make.in
@@ -73,6 +73,9 @@ TEST_SHLIB_DIR = ${abs_top_builddir}/.lib
 # flags before CPPFLAGS and LDFLAGS. While EXTRA_CFLAGS are added at the end.
 
 COMPILE = $(CC) $(PRE_CPPFLAGS) $(CPPFLAGS) $(DEFS) $(CFLAGS) $(EXTRA_CFLAGS) $(DEP_FLAGS)
+CCAS = @CCAS@
+ASMFLAGS = @ASMFLAGS@
+COMPILE_AS = $(CCAS) $(PRE_CPPFLAGS) $(CPPFLAGS) $(DEFS) $(ASMFLAGS) $(DEP_FLAGS)
 COMPILE_CXX = $(CXX) $(PRE_CPPFLAGS) $(CPPFLAGS) $(DEFS) $(CXXFLAGS) $(DEP_FLAGS)
 LINK = $(CC) $(CFLAGS) $(PRE_LDFLAGS) $(LDFLAGS)
 LINK_CXX = $(CXX) $(CXXFLAGS) $(PRE_LDFLAGS) $(LDFLAGS)
diff --git a/configure.ac b/configure.ac
index 1012718..8db139d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -134,6 +134,10 @@ LSH_RPATH_INIT([`echo $with_lib_path | sed 's/:/ /g'` \
 
 # Checks for programs.
 AC_PROG_CC
+# Copied from gmp
+test -n "$CCAS" || CCAS='$(CC)'
+AC_SUBST(CCAS)
+AC_SUBST(ASMFLAGS)
 
 NETTLE_CHECK_IFUNC
 
@@ -321,7 +325,7 @@ W64_ABI=no   # For x86_64 windows
 case "$host_cpu" in
   [x86_64 | amd64])
     AC_TRY_COMPILE([
-#if defined(__x86_64__) || defined(__arch64__)
+#if defined(__x86_64__) || defined(__arch64__) || defined(_M_AMD64)
 #error 64-bit x86
 #endif
     ], [], [
@@ -374,7 +378,7 @@ case "$host_cpu" in
     ;;
   aarch64*)
     AC_TRY_COMPILE([
-#if defined(__aarch64__)
+#if defined(__aarch64__) || defined(_M_ARM64)
 #error 64-bit arm
 #endif
     ], [], [
