diff --git a/configure.ac b/configure.ac
index ea92c1f97..9621a70e5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -693,8 +693,9 @@ See 'config.log' for details (search for GMP_NUMB_BITS).])],
 
 dnl Check if we can link with GMP
 AC_CHECK_LIB(gmp, __gmpz_init, [LIBS="-lgmp $LIBS"],
+ [AC_CHECK_LIB(gmpd, __gmpz_init, [LIBS="-lgmpd $LIBS"],
  [AC_MSG_ERROR([libgmp not found or uses a different ABI (including static vs shared).
-Please read the INSTALL file -- see "In case of problem".])])
+Please read the INSTALL file -- see "In case of problem".])])])
 
 dnl Check for corresponding 'gmp.h' and libgmp
 AC_MSG_CHECKING(if gmp.h version and libgmp version are the same)
