diff --git a/cmake/CompilerWarnings.cmake b/cmake/CompilerWarnings.cmake
index 9a9e507..eb1a1c5 100644
--- a/cmake/CompilerWarnings.cmake
+++ b/cmake/CompilerWarnings.cmake
@@ -86,6 +86,8 @@ endif()
 
 # Treat warnings as errors
 function( set_warning_as_error )
+    return()
+
     message( STATUS "[${PROJECT_NAME}] Treating warnings as errors")
 
     if ( CMAKE_VERSION VERSION_GREATER_EQUAL "3.24" )
