diff --git a/CMakeLists.txt b/CMakeLists.txt
index fb6d040..e8118ec 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,5 +16,7 @@ install(TARGETS indica EXPORT indicaConfig)
 install(EXPORT indicaConfig
         NAMESPACE indica::
         DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/indica)
-install(FILES ${CMAKE_CURRENT_LIST_DIR}/include/indica.hpp
-        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/indica)
+install(DIRECTORY ${CMAKE_CURRENT_LIST_DIR}/include/indicators
+        DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+        USE_SOURCE_PERMISSIONS
+        PATTERN "*.hpp")
