From b399342016573d6f45bd5bb1aa8d4df358e0761c Mon Sep 17 00:00:00 2001
From: Wei Dai <wei.dai@microsoft.com>
Date: Wed, 16 Jun 2021 18:21:51 -0700
Subject: [PATCH] Fixed config.

---
 cmake/HEXLConfig.cmake.in | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/cmake/HEXLConfig.cmake.in b/cmake/HEXLConfig.cmake.in
index 57089bd..f112380 100644
--- a/cmake/HEXLConfig.cmake.in
+++ b/cmake/HEXLConfig.cmake.in
@@ -11,6 +11,14 @@
 
 @PACKAGE_INIT@
 
+include(CMakeFindDependencyMacro)
+find_dependency(CpuFeatures)
+if(NOT CpuFeatures_FOUND)
+    message(WARNING "Could not find dependency `CpuFeatures` required by this configuration")
+    set(HEXL_FOUND FALSE)
+    return()
+endif()
+
 include(${CMAKE_CURRENT_LIST_DIR}/HEXLTargets.cmake)
 
 # Defines HEXL_FOUND: If Intel HEXL library was found
-- 
2.25.1

