diff --git a/src/HConfig.h.in b/src/HConfig.h.in
index 652f6651f..62b81f83c 100644
--- a/src/HConfig.h.in
+++ b/src/HConfig.h.in
@@ -4,7 +4,7 @@
 #cmakedefine FAST_BUILD
 #cmakedefine ZLIB_FOUND
 #cmakedefine CMAKE_BUILD_TYPE "@CMAKE_BUILD_TYPE@"
-#cmakedefine CMAKE_INSTALL_PREFIX "@CMAKE_INSTALL_PREFIX@"
+#cmakedefine CMAKE_INSTALL_PREFIX ""
 #cmakedefine HIGHSINT64
 #cmakedefine HIGHS_HAVE_MM_PAUSE
 #cmakedefine HIGHS_HAVE_BUILTIN_CLZ
@@ -15,6 +15,6 @@
 #define HIGHS_VERSION_MAJOR @HIGHS_VERSION_MAJOR@
 #define HIGHS_VERSION_MINOR @HIGHS_VERSION_MINOR@
 #define HIGHS_VERSION_PATCH @HIGHS_VERSION_PATCH@
-#define HIGHS_DIR "@CMAKE_SOURCE_DIR@"
+#define HIGHS_DIR ""
 
 #endif /* HCONFIG_H_ */
