diff --git a/highs.pc.in b/highs.pc.in
index 4c00116..792ec17 100644
--- a/highs.pc.in
+++ b/highs.pc.in
@@ -7,6 +7,6 @@ Description: Linear Optimization Software
 URL: https://github.com/ERGO-Code/HiGHS
 Version: @HIGHS_VERSION_MAJOR@.@HIGHS_VERSION_MINOR@.@HIGHS_VERSION_PATCH@
 Libs: -L${libdir} -lhighs
-#TODO Libs.private (when HiGHS lib is build as static lib)
 Cflags: -I${includedir}
 Requires:
+Requires.private: zlib
