diff --git a/cmake/TPLs/FindTPLMETIS.cmake b/cmake/TPLs/FindTPLMETIS.cmake
index 90911a3..bde1e25 100644
--- a/cmake/TPLs/FindTPLMETIS.cmake
+++ b/cmake/TPLs/FindTPLMETIS.cmake
@@ -53,8 +53,8 @@
 # ************************************************************************
 # @HEADER
 
-
-TRIBITS_TPL_FIND_INCLUDE_DIRS_AND_LIBRARIES( METIS
-  REQUIRED_HEADERS metis.h
-  REQUIRED_LIBS_NAMES "metis"
-  )
+find_package(metis REQUIRED)
+tribits_extpkg_create_imported_all_libs_target_and_config_file(
+  METIS
+  INNER_FIND_PACKAGE_NAME metis
+  IMPORTED_TARGETS_FOR_ALL_LIBS metis )
diff --git a/cmake/TPLs/FindTPLMatio.cmake b/cmake/TPLs/FindTPLMatio.cmake
index c76713a..e57b5be 100644
--- a/cmake/TPLs/FindTPLMatio.cmake
+++ b/cmake/TPLs/FindTPLMatio.cmake
@@ -56,7 +56,9 @@ if (${CMAKE_VERSION} GREATER "3.13")
      cmake_policy(SET CMP0074 NEW)
 endif()
 
-find_package(Matio REQUIRED)
-TRIBITS_TPL_FIND_INCLUDE_DIRS_AND_LIBRARIES( Matio
-  REQUIRED_HEADERS matio.h
-  REQUIRED_LIBS_NAMES "matio")
+find_package(matio REQUIRED)
+tribits_extpkg_create_imported_all_libs_target_and_config_file(
+  Matio
+  INNER_FIND_PACKAGE_NAME matio
+  IMPORTED_TARGETS_FOR_ALL_LIBS matio::matio )
+
diff --git a/cmake/TPLs/FindTPLParMETIS.cmake b/cmake/TPLs/FindTPLParMETIS.cmake
index 2dc2c14..9bc04ae 100644
--- a/cmake/TPLs/FindTPLParMETIS.cmake
+++ b/cmake/TPLs/FindTPLParMETIS.cmake
@@ -54,7 +54,8 @@
 # @HEADER
 
 
-TRIBITS_TPL_FIND_INCLUDE_DIRS_AND_LIBRARIES( ParMETIS
-  REQUIRED_HEADERS parmetis.h
-  REQUIRED_LIBS_NAMES "parmetis;metis"
-  )
+find_package(parmetis REQUIRED)
+tribits_extpkg_create_imported_all_libs_target_and_config_file(
+  ParMETIS
+  INNER_FIND_PACKAGE_NAME parmetis
+  IMPORTED_TARGETS_FOR_ALL_LIBS parmetis )
diff --git a/cmake/TPLs/FindTPLZlib.cmake b/cmake/TPLs/FindTPLZlib.cmake
index cc607e9..069a37c 100644
--- a/cmake/TPLs/FindTPLZlib.cmake
+++ b/cmake/TPLs/FindTPLZlib.cmake
@@ -53,8 +53,8 @@
 # ************************************************************************
 # @HEADER
 
-
-TRIBITS_TPL_FIND_INCLUDE_DIRS_AND_LIBRARIES( Zlib
-  REQUIRED_HEADERS zlib.h
-  REQUIRED_LIBS_NAMES  z
-  )
+find_package(ZLIB REQUIRED)
+tribits_extpkg_create_imported_all_libs_target_and_config_file(
+  Zlib
+  INNER_FIND_PACKAGE_NAME ZLIB
+  IMPORTED_TARGETS_FOR_ALL_LIBS ZLIB::ZLIB )
diff --git a/cmake/tribits/common_tpls/FindTPLCGNS.cmake b/cmake/tribits/common_tpls/FindTPLCGNS.cmake
index 649ab49..f9a4a5d 100644
--- a/cmake/tribits/common_tpls/FindTPLCGNS.cmake
+++ b/cmake/tribits/common_tpls/FindTPLCGNS.cmake
@@ -55,7 +55,7 @@ if ((CGNS_ALLOW_MODERN AND HDF5_FOUND_MODERN_CONFIG_FILE) OR CGNS_FORCE_MODERN)
     endif()
     tribits_extpkg_create_imported_all_libs_target_and_config_file(
       CGNS
-      INNER_FIND_PACKAGE_NAME  CGNS
+      INNER_FIND_PACKAGE_NAME  cgns
       IMPORTED_TARGETS_FOR_ALL_LIBS  ${CGNS_TARGET})
     set(TPL_CGNS_NOT_FOUND FALSE)
   endif()
diff --git a/cmake/tribits/common_tpls/FindTPLHDF5.cmake b/cmake/tribits/common_tpls/FindTPLHDF5.cmake
index 71e0dd6..9eb7b65 100644
--- a/cmake/tribits/common_tpls/FindTPLHDF5.cmake
+++ b/cmake/tribits/common_tpls/FindTPLHDF5.cmake
@@ -16,7 +16,7 @@ if (Netcdf_ALLOW_MODERN)
     tribits_extpkg_create_imported_all_libs_target_and_config_file(
       HDF5
       INNER_FIND_PACKAGE_NAME  HDF5
-      IMPORTED_TARGETS_FOR_ALL_LIBS   ${HDF5_EXPORT_LIBRARIES})
+      IMPORTED_TARGETS_FOR_ALL_LIBS   $<IF:$<TARGET_EXISTS:hdf5::hdf5-shared>,hdf5::hdf5-shared,hdf5::hdf5-static>)
     set(HDF5_INTERNAL_IS_MODERN TRUE)
     set(TPL_HDF5_NOT_FOUND FALSE)
   else()
diff --git a/packages/seacas/libraries/aprepro_lib/CMakeLists.txt b/packages/seacas/libraries/aprepro_lib/CMakeLists.txt
index 1efc331..bfdd0f4 100644
--- a/packages/seacas/libraries/aprepro_lib/CMakeLists.txt
+++ b/packages/seacas/libraries/aprepro_lib/CMakeLists.txt
@@ -110,6 +110,10 @@ if (${CMAKE_PROJECT_NAME} STREQUAL "Seacas")
    endif()
 endif()
 
+if(BUILD_SHARED_LIBS)
+    set_target_properties(aprepro_lib PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS ON)
+endif()
+
 if (${PACKAGE_NAME}_ENABLE_TESTS)
   TRIBITS_ADD_EXECUTABLE(aprepro_lib_test NOEXEPREFIX NOEXESUFFIX INSTALLABLE SOURCES apr_test.cc LINKER_LANGUAGE CXX)
   TRIBITS_ADD_EXECUTABLE(aprepro_strings_test NOEXEPREFIX NOEXESUFFIX INSTALLABLE SOURCES strings.cc LINKER_LANGUAGE CXX)
diff --git a/packages/seacas/libraries/chaco/CMakeLists.txt b/packages/seacas/libraries/chaco/CMakeLists.txt
index 576abc8..94a59af 100644
--- a/packages/seacas/libraries/chaco/CMakeLists.txt
+++ b/packages/seacas/libraries/chaco/CMakeLists.txt
@@ -56,5 +56,8 @@ TRIBITS_ADD_LIBRARY(
   HEADERS ${HEADERS}
   SOURCES ${SOURCES}
   )
+if(BUILD_SHARED_LIBS)
+    set_target_properties(chaco PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS ON)
+endif()
 
 TRIBITS_SUBPACKAGE_POSTPROCESS()
diff --git a/packages/seacas/libraries/exodus/include/exodusII.h b/packages/seacas/libraries/exodus/include/exodusII.h
index 5c5149e..cc97e28 100644
--- a/packages/seacas/libraries/exodus/include/exodusII.h
+++ b/packages/seacas/libraries/exodus/include/exodusII.h
@@ -571,6 +571,8 @@ typedef struct ex_var_params
 #ifndef EXODUS_EXPORT
 #if defined(_WIN32) && defined(exodus_shared_EXPORTS)
 #define EXODUS_EXPORT __declspec(dllexport)
+#elif defined(_WIN32) && defined(exodus_EXPORTS)
+#define EXODUS_EXPORT __declspec(dllexport)
 #endif
 #endif /* EXODUS_EXPORT */
 
diff --git a/packages/seacas/libraries/nemesis/CMakeLists.txt b/packages/seacas/libraries/nemesis/CMakeLists.txt
index b03a9e9..4ab0952 100644
--- a/packages/seacas/libraries/nemesis/CMakeLists.txt
+++ b/packages/seacas/libraries/nemesis/CMakeLists.txt
@@ -1,5 +1,6 @@
 TRIBITS_SUBPACKAGE(Nemesis)
 
+set(NEMESIS_BUILD_SHARED_LIBS "${BUILD_SHARED_LIBS}" CACHE BOOL "")
 SET(NEMESIS_VERSION_MAJOR "5")
 SET(NEMESIS_VERSION_MINOR "15")
 SET(NEMESIS_VERSION_PATCH "0")
@@ -27,6 +28,7 @@ TRIBITS_ADD_LIBRARY(
   HEADERS ${HEADERS}
   SOURCES ${SOURCES}
   )
+target_compile_definitions(nemesis PRIVATE nemc_EXPORTS)
 
 IF (${PACKAGE_NAME}_ENABLE_TESTS)
 TRIBITS_ADD_EXECUTABLE_AND_TEST(
diff --git a/packages/seacas/libraries/nemesis/ne_nemesisI.h b/packages/seacas/libraries/nemesis/ne_nemesisI.h
index f07b316..770235c 100644
--- a/packages/seacas/libraries/nemesis/ne_nemesisI.h
+++ b/packages/seacas/libraries/nemesis/ne_nemesisI.h
@@ -12,6 +12,8 @@
  ****************************************************************************/
 #pragma once
 
+#include "nemesis_cfg.h"
+
 #ifdef __cplusplus
 extern "C" {
 #endif
@@ -33,24 +35,24 @@ typedef void void_int;
 /*=============================================================================
  *     Initial Information Routines
  *===========================================================================*/
-extern int ne_get_init_info(int   neid,          /* NemesisI file ID */
+NEMESIS_EXPORT int ne_get_init_info(int   neid,          /* NemesisI file ID */
                             int  *num_proc,      /* Number of processors */
                             int  *num_proc_in_f, /* Number of procs in this file */
                             char *ftype);
 
-extern int ne_put_init_info(int   neid,          /* NemesisI file ID */
+NEMESIS_EXPORT int ne_put_init_info(int   neid,          /* NemesisI file ID */
                             int   num_proc,      /* Number of processors */
                             int   num_proc_in_f, /* Number of procs in this file */
                             char *ftype);
 
-extern int ne_get_init_global(int       neid,            /* NemesisI file ID */
+NEMESIS_EXPORT int ne_get_init_global(int       neid,            /* NemesisI file ID */
                               void_int *num_nodes_g,     /* Number of global FEM nodes */
                               void_int *num_elems_g,     /* Number of global FEM elements */
                               void_int *num_elem_blks_g, /* Number of global elem blocks */
                               void_int *num_node_sets_g, /* Number of global node sets */
                               void_int *num_side_sets_g  /* Number of global side sets */
 );
-extern int ne_put_init_global(int     neid,            /* NemesisI file ID */
+NEMESIS_EXPORT int ne_put_init_global(int     neid,            /* NemesisI file ID */
                               int64_t num_nodes_g,     /* Number of global FEM nodes */
                               int64_t num_elems_g,     /* Number of global FEM elements */
                               int64_t num_elem_blks_g, /* Number of global elem blocks */
@@ -58,12 +60,12 @@ extern int ne_put_init_global(int     neid,            /* NemesisI file ID */
                               int64_t num_side_sets_g  /* Number of global side sets */
 );
 
-extern int ne_put_version(int neid);
+NEMESIS_EXPORT int ne_put_version(int neid);
 
 /*=============================================================================
  *     Loadbalance Parameter Routines
  *===========================================================================*/
-extern int ne_get_loadbal_param(int       neid,           /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_get_loadbal_param(int       neid,           /* NetCDF/Exodus file ID */
                                 void_int *num_int_nodes,  /* Number of internal FEM nodes */
                                 void_int *num_bor_nodes,  /* Number of border FEM nodes */
                                 void_int *num_ext_nodes,  /* Number of external FEM nodes */
@@ -74,7 +76,7 @@ extern int ne_get_loadbal_param(int       neid,           /* NetCDF/Exodus file
                                 int       processor       /* Processor ID */
 );
 
-extern int ne_put_loadbal_param(int     neid,           /* NemesisI file ID  */
+NEMESIS_EXPORT int ne_put_loadbal_param(int     neid,           /* NemesisI file ID  */
                                 int64_t num_int_nodes,  /* Number of internal FEM nodes */
                                 int64_t num_bor_nodes,  /* Number of border FEM nodes */
                                 int64_t num_ext_nodes,  /* Number of external FEM nodes */
@@ -85,7 +87,7 @@ extern int ne_put_loadbal_param(int     neid,           /* NemesisI file ID  */
                                 int     processor       /* Processor ID */
 );
 
-extern int ne_put_loadbal_param_cc(int       neid,           /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_put_loadbal_param_cc(int       neid,           /* NetCDF/Exodus file ID */
                                    void_int *num_int_nodes,  /* Number of internal node IDs */
                                    void_int *num_bor_nodes,  /* Number of border node IDs */
                                    void_int *num_ext_nodes,  /* Number of external node IDs */
@@ -98,26 +100,26 @@ extern int ne_put_loadbal_param_cc(int       neid,           /* NetCDF/Exodus fi
 /*=============================================================================
  *     NS, SS & EB Global Parameter Routines
  *===========================================================================*/
-extern int ne_get_ns_param_global(int       neid,          /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_get_ns_param_global(int       neid,          /* NetCDF/Exodus file ID */
                                   void_int *ns_ids_glob,   /* Global IDs of node sets */
                                   void_int *ns_n_cnt_glob, /* Count of nodes in node sets */
                                   void_int *ns_df_cnt_glob /* Count of dist. factors in ns */
 );
 
-extern int
+NEMESIS_EXPORT int
 ne_put_ns_param_global(int       neid,          /* NemesisI file ID */
                        void_int *global_ids,    /* Vector of global node-set IDs */
                        void_int *global_n_cnts, /* Vector of node counts in node-sets */
                        void_int *global_df_cnts /* Vector of dist factor counts in node-sets */
 );
 
-extern int ne_get_ss_param_global(int       neid,          /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_get_ss_param_global(int       neid,          /* NetCDF/Exodus file ID */
                                   void_int *ss_ids_glob,   /* Global side-set IDs */
                                   void_int *ss_s_cnt_glob, /* Global side count */
                                   void_int *ss_df_cnt_glob /* Global dist. factor count */
 );
 
-extern int ne_put_ss_param_global(int       neid,           /* NemesisI file ID */
+NEMESIS_EXPORT int ne_put_ss_param_global(int       neid,           /* NemesisI file ID */
                                   void_int *global_ids,     /* Vector of global side-set IDs */
                                   void_int *global_el_cnts, /* Vector of element/side */
                                                             /* counts in each side set */
@@ -125,12 +127,12 @@ extern int ne_put_ss_param_global(int       neid,           /* NemesisI file ID
                                                             /* counts in each side set */
 );
 
-extern int ne_get_eb_info_global(int       neid,       /* NemesisI file ID                 */
+NEMESIS_EXPORT int ne_get_eb_info_global(int       neid,       /* NemesisI file ID                 */
                                  void_int *el_blk_ids, /* Vector of global element IDs     */
                                  void_int *el_blk_cnts /* Vector of global element counts  */
 );
 
-extern int ne_put_eb_info_global(int       neid,       /* NemesisI file ID */
+NEMESIS_EXPORT int ne_put_eb_info_global(int       neid,       /* NemesisI file ID */
                                  void_int *el_blk_ids, /* Vector of global element IDs     */
                                  void_int *el_blk_cnts /* Vector of global element counts  */
 );
@@ -138,7 +140,7 @@ extern int ne_put_eb_info_global(int       neid,       /* NemesisI file ID */
 /*=============================================================================
  *     NS, SS & EB Subset Routines
  *===========================================================================*/
-extern int ne_get_n_side_set(int          neid,               /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_get_n_side_set(int          neid,               /* NetCDF/Exodus file ID */
                              ex_entity_id side_set_id,        /* Side-set ID to read */
                              int64_t      start_side_num,     /* Starting element number */
                              int64_t      num_sides,          /* Number of sides to read */
@@ -146,7 +148,7 @@ extern int ne_get_n_side_set(int          neid,               /* NetCDF/Exodus f
                              void_int    *side_set_side_list  /* List of side IDs */
 );
 
-extern int ne_put_n_side_set(int             neid,               /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_put_n_side_set(int             neid,               /* NetCDF/Exodus file ID */
                              ex_entity_id    side_set_id,        /* Side-set ID to write */
                              int64_t         start_side_num,     /* Starting element number */
                              int64_t         num_sides,          /* Number of sides to write */
@@ -154,49 +156,49 @@ extern int ne_put_n_side_set(int             neid,               /* NetCDF/Exodu
                              const void_int *side_set_side_list  /* List of side IDs */
 );
 
-extern int ne_get_n_side_set_df(int          neid,          /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_get_n_side_set_df(int          neid,          /* NetCDF/Exodus file ID */
                                 ex_entity_id side_set_id,   /* Side-set ID */
                                 int64_t      start_num,     /* Starting df number */
                                 int64_t      num_df_to_get, /* Number of df's to read */
                                 void        *side_set_df    /* Distribution factors */
 );
 
-extern int ne_put_n_side_set_df(int          neid,          /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_put_n_side_set_df(int          neid,          /* NetCDF/Exodus file ID */
                                 ex_entity_id side_set_id,   /* Side-set ID */
                                 int64_t      start_num,     /* Starting df number */
                                 int64_t      num_df_to_get, /* Number of df's to write */
                                 void        *side_set_df    /* Distribution factors */
 );
 
-extern int ne_get_n_node_set(int          neid,              /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_get_n_node_set(int          neid,              /* NetCDF/Exodus file ID */
                              ex_entity_id node_set_id,       /* Node set ID */
                              int64_t      start_node_num,    /* Node index to start reading at */
                              int64_t      num_node,          /* Number of nodes to read */
                              void_int    *node_set_node_list /* List of nodes in node set */
 );
 
-extern int ne_put_n_node_set(int             neid,              /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_put_n_node_set(int             neid,              /* NetCDF/Exodus file ID */
                              ex_entity_id    node_set_id,       /* Node set ID */
                              int64_t         start_node_num,    /* Node index to start writing at */
                              int64_t         num_node,          /* Number of nodes to write */
                              const void_int *node_set_node_list /* List of nodes in node set */
 );
 
-extern int ne_get_n_node_set_df(int          neid,          /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_get_n_node_set_df(int          neid,          /* NetCDF/Exodus file ID */
                                 ex_entity_id node_set_id,   /* Node-set ID */
                                 int64_t      start_num,     /* Starting df number */
                                 int64_t      num_df_to_get, /* Number of df's to read */
                                 void        *node_set_df    /* Distribution factors */
 );
 
-extern int ne_put_n_node_set_df(int          neid,          /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_put_n_node_set_df(int          neid,          /* NetCDF/Exodus file ID */
                                 ex_entity_id node_set_id,   /* Node-set ID */
                                 int64_t      start_num,     /* Starting df number */
                                 int64_t      num_df_to_get, /* Number of df's to write */
                                 void        *node_set_df    /* Distribution factors */
 );
 
-extern int ne_get_n_coord(int     neid,           /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_get_n_coord(int     neid,           /* NetCDF/Exodus file ID */
                           int64_t start_node_num, /* Starting position to read from */
                           int64_t num_nodes,      /* Number of coords to read */
                           void   *x_coor,         /* Vector of X coordinates */
@@ -204,7 +206,7 @@ extern int ne_get_n_coord(int     neid,           /* NetCDF/Exodus file ID */
                           void   *z_coor          /* Vector of Z coordinates */
 );
 
-extern int ne_put_n_coord(int     neid,           /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_put_n_coord(int     neid,           /* NetCDF/Exodus file ID */
                           int64_t start_node_num, /* Starting position to write to */
                           int64_t num_nodes,      /* Number of coords to write */
                           void   *x_coor,         /* Vector of X coordinates */
@@ -212,35 +214,35 @@ extern int ne_put_n_coord(int     neid,           /* NetCDF/Exodus file ID */
                           void   *z_coor          /* Vector of Z coordinates */
 );
 
-extern int ne_get_n_elem_conn(int          neid,           /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_get_n_elem_conn(int          neid,           /* NetCDF/Exodus file ID */
                               ex_entity_id elem_blk_id,    /* Element block ID */
                               int64_t      start_elem_num, /* Starting position to read from */
                               int64_t      num_elems,      /* Number of elements to read */
                               void_int    *connect         /* Connectivity vector */
 );
 
-extern int ne_put_n_elem_conn(int             neid,           /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_put_n_elem_conn(int             neid,           /* NetCDF/Exodus file ID */
                               ex_entity_id    elem_blk_id,    /* Element block ID */
                               int64_t         start_elem_num, /* Starting position to write to */
                               int64_t         num_elems,      /* Number of elements to write */
                               const void_int *connect         /* Connectivity vector */
 );
 
-extern int ne_get_n_elem_attr(int          neid,           /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_get_n_elem_attr(int          neid,           /* NetCDF/Exodus file ID */
                               ex_entity_id elem_blk_id,    /* Element block ID */
                               int64_t      start_elem_num, /* Starting position to read from */
                               int64_t      num_elems,      /* Number of elements to read */
                               void        *attrib          /* Attribute */
 );
 
-extern int ne_put_n_elem_attr(int          neid,           /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_put_n_elem_attr(int          neid,           /* NetCDF/Exodus file ID */
                               ex_entity_id elem_blk_id,    /* Element block ID */
                               int64_t      start_elem_num, /* Starting position to write to */
                               int64_t      num_elems,      /* Number of elements to write */
                               void        *attrib          /* Attribute */
 );
 
-extern int ne_get_elem_type(int          neid,        /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_get_elem_type(int          neid,        /* NetCDF/Exodus file ID */
                             ex_entity_id elem_blk_id, /* Element block ID */
                             char        *elem_type    /* The name of the element type */
 );
@@ -248,7 +250,7 @@ extern int ne_get_elem_type(int          neid,        /* NetCDF/Exodus file ID *
 /*=============================================================================
  *     Variable Routines
  *===========================================================================*/
-extern int ne_get_n_elem_var(int          neid,              /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_get_n_elem_var(int          neid,              /* NetCDF/Exodus file ID */
                              int          time_step,         /* time index */
                              int          elem_var_index,    /* elemental variable index */
                              ex_entity_id elem_blk_id,       /* elemental block id */
@@ -258,7 +260,7 @@ extern int ne_get_n_elem_var(int          neid,              /* NetCDF/Exodus fi
                              void        *elem_var_vals      /* variable values */
 );
 
-extern int ne_put_elem_var_slab(int          neid,           /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_put_elem_var_slab(int          neid,           /* NetCDF/Exodus file ID */
                                 int          time_step,      /* time index */
                                 int          elem_var_index, /* elemental variable index */
                                 ex_entity_id elem_blk_id,    /* elemental block id */
@@ -267,7 +269,7 @@ extern int ne_put_elem_var_slab(int          neid,           /* NetCDF/Exodus fi
                                 void        *elem_var_vals   /* variable values */
 );
 
-extern int ne_get_n_nodal_var(int     neid,            /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_get_n_nodal_var(int     neid,            /* NetCDF/Exodus file ID */
                               int     time_step,       /* whole time step number */
                               int     nodal_var_index, /* index of desired nodal var */
                               int64_t start_node_num,  /* starting node number */
@@ -275,7 +277,7 @@ extern int ne_get_n_nodal_var(int     neid,            /* NetCDF/Exodus file ID
                               void   *nodal_vars       /* array of nodal var values */
 );
 
-extern int ne_put_nodal_var_slab(int     neid,            /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_put_nodal_var_slab(int     neid,            /* NetCDF/Exodus file ID */
                                  int     time_step,       /* The time step index */
                                  int     nodal_var_index, /* Nodal variable index */
                                  int64_t start_pos,       /* Start position for write */
@@ -286,51 +288,51 @@ extern int ne_put_nodal_var_slab(int     neid,            /* NetCDF/Exodus file
 /*=============================================================================
  *     Number Map Routines
  *===========================================================================*/
-extern int ne_get_n_elem_num_map(int       neid,      /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_get_n_elem_num_map(int       neid,      /* NetCDF/Exodus file ID */
                                  int64_t   start_ent, /* Starting position to read from */
                                  int64_t   num_ents,  /* Number of elements to read */
                                  void_int *elem_map   /* element map numbers */
 );
 
-extern int ne_put_n_elem_num_map(int             neid,      /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_put_n_elem_num_map(int             neid,      /* NetCDF/Exodus file ID */
                                  int64_t         start_ent, /* Starting position to read from */
                                  int64_t         num_ents,  /* Number of elements to read */
                                  const void_int *elem_map   /* element map numbers */
 );
 
-extern int ne_get_n_node_num_map(int       neid,      /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_get_n_node_num_map(int       neid,      /* NetCDF/Exodus file ID */
                                  int64_t   start_ent, /* starting node number */
                                  int64_t   num_ents,  /* number of nodes to read */
                                  void_int *node_map   /* vector for node map */
 );
 
-extern int ne_put_n_node_num_map(int             neid,      /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_put_n_node_num_map(int             neid,      /* NetCDF/Exodus file ID */
                                  int64_t         start_ent, /* starting node number */
                                  int64_t         num_ents,  /* number of nodes to read */
                                  const void_int *node_map   /* vector for node map */
 );
 
-extern int ne_get_node_map(int       neid,      /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_get_node_map(int       neid,      /* NetCDF/Exodus file ID */
                            void_int *node_mapi, /* Internal FEM node IDs */
                            void_int *node_mapb, /* Border FEM node IDs */
                            void_int *node_mape, /* External FEM node IDs */
                            int       processor  /* Processor IDs */
 );
 
-extern int ne_put_node_map(int       neid,      /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_put_node_map(int       neid,      /* NetCDF/Exodus file ID */
                            void_int *node_mapi, /* Internal FEM node IDs */
                            void_int *node_mapb, /* Border FEM node IDs */
                            void_int *node_mape, /* External FEM node IDs */
                            int       processor  /* This processor ID */
 );
 
-extern int ne_get_elem_map(int       neid,      /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_get_elem_map(int       neid,      /* NetCDF/Exodus file ID */
                            void_int *elem_mapi, /* Internal element IDs */
                            void_int *elem_mapb, /* Border element IDs */
                            int       processor  /* Processor ID */
 );
 
-extern int ne_put_elem_map(int       neid,      /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_put_elem_map(int       neid,      /* NetCDF/Exodus file ID */
                            void_int *elem_mapi, /* Internal FEM element IDs */
                            void_int *elem_mapb, /* Border FEM element IDs */
                            int       processor  /* This processor ID */
@@ -340,7 +342,7 @@ extern int ne_put_elem_map(int       neid,      /* NetCDF/Exodus file ID */
  *     Communications Maps Routines
  *===========================================================================*/
 
-extern int ne_get_cmap_params(int       neid,                /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_get_cmap_params(int       neid,                /* NetCDF/Exodus file ID */
                               void_int *node_cmap_ids,       /* Nodal comm. map IDs */
                               void_int *node_cmap_node_cnts, /* Number of nodes in each map */
                               void_int *elem_cmap_ids,       /* Elemental comm. map IDs */
@@ -348,7 +350,7 @@ extern int ne_get_cmap_params(int       neid,                /* NetCDF/Exodus fi
                               int       processor            /* This processor ID */
 );
 
-extern int ne_put_cmap_params(int       neid,               /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_put_cmap_params(int       neid,               /* NetCDF/Exodus file ID */
                               void_int *node_map_ids,       /* Node map IDs */
                               void_int *node_map_node_cnts, /* Nodes in nodal comm */
                               void_int *elem_map_ids,       /* Elem map IDs */
@@ -356,7 +358,7 @@ extern int ne_put_cmap_params(int       neid,               /* NetCDF/Exodus fil
                               int64_t   processor           /* This processor ID */
 );
 
-extern int ne_put_cmap_params_cc(int       neid,               /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_put_cmap_params_cc(int       neid,               /* NetCDF/Exodus file ID */
                                  void_int *node_map_ids,       /* Node map IDs */
                                  void_int *node_map_node_cnts, /* Nodes in nodal comm */
                                  void_int *node_proc_ptrs,     /* Pointer into array for */
@@ -367,21 +369,21 @@ extern int ne_put_cmap_params_cc(int       neid,               /* NetCDF/Exodus
                                                                /* elem maps               */
 );
 
-extern int ne_get_node_cmap(int          neid,     /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_get_node_cmap(int          neid,     /* NetCDF/Exodus file ID */
                             ex_entity_id map_id,   /* Map ID */
                             void_int    *node_ids, /* FEM node IDs */
                             void_int    *proc_ids, /* Processor IDs */
                             int          processor /* This processor ID */
 );
 
-extern int ne_put_node_cmap(int          neid,     /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_put_node_cmap(int          neid,     /* NetCDF/Exodus file ID */
                             ex_entity_id map_id,   /* Nodal comm map ID */
                             void_int    *node_ids, /* FEM node IDs */
                             void_int    *proc_ids, /* Processor IDs */
                             int          processor /* This processor ID */
 );
 
-extern int ne_get_elem_cmap(int          neid,     /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_get_elem_cmap(int          neid,     /* NetCDF/Exodus file ID */
                             ex_entity_id map_id,   /* Elemental comm map ID */
                             void_int    *elem_ids, /* Element IDs */
                             void_int    *side_ids, /* Element side IDs */
@@ -389,7 +391,7 @@ extern int ne_get_elem_cmap(int          neid,     /* NetCDF/Exodus file ID */
                             int          processor /* This processor ID */
 );
 
-extern int ne_put_elem_cmap(int          neid,     /* NetCDF/Exodus file ID */
+NEMESIS_EXPORT int ne_put_elem_cmap(int          neid,     /* NetCDF/Exodus file ID */
                             ex_entity_id map_id,   /* Elemental comm map ID */
                             void_int    *elem_ids, /* Vector of element IDs */
                             void_int    *side_ids, /* Vector of side IDs */
diff --git a/packages/seacas/libraries/nemesis/nemesis_cfg.h.in b/packages/seacas/libraries/nemesis/nemesis_cfg.h.in
index b3a53a7..e854744 100755
--- a/packages/seacas/libraries/nemesis/nemesis_cfg.h.in
+++ b/packages/seacas/libraries/nemesis/nemesis_cfg.h.in
@@ -1,8 +1,10 @@
+/*
 # Copyright(C) 1999-2020, 2022 National Technology & Engineering Solutions
 # of Sandia, LLC (NTESS).  Under the terms of Contract DE-NA0003525 with
 # NTESS, the U.S. Government retains certain rights in this software.
 #
 # See packages/seacas/LICENSE for details
+*/
 #pragma once
 
 #define NEMESIS_VERSION_MAJOR @NEMESIS_VERSION_MAJOR@
diff --git a/packages/seacas/libraries/suplib_c/CMakeLists.txt b/packages/seacas/libraries/suplib_c/CMakeLists.txt
index 52f7c57..0dce939 100644
--- a/packages/seacas/libraries/suplib_c/CMakeLists.txt
+++ b/packages/seacas/libraries/suplib_c/CMakeLists.txt
@@ -29,5 +29,8 @@ TRIBITS_ADD_LIBRARY(
   NOINSTALLHEADERS
   SOURCES ${SOURCES}
   )
+if(BUILD_SHARED_LIBS)
+    set_target_properties(suplib_c PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS ON)
+endif()
 
 TRIBITS_SUBPACKAGE_POSTPROCESS()
diff --git a/packages/seacas/libraries/suplib_cpp/CMakeLists.txt b/packages/seacas/libraries/suplib_cpp/CMakeLists.txt
index 47c7b0d..6a57f63 100644
--- a/packages/seacas/libraries/suplib_cpp/CMakeLists.txt
+++ b/packages/seacas/libraries/suplib_cpp/CMakeLists.txt
@@ -31,5 +31,8 @@ TRIBITS_ADD_LIBRARY(
   NOINSTALLHEADERS
   SOURCES ${SOURCES}
   )
+if(BUILD_SHARED_LIBS)
+    set_target_properties(suplib_cpp PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS ON)
+endif()
 
 TRIBITS_SUBPACKAGE_POSTPROCESS()
