diff --git a/packages/seacas/libraries/ioss/src/Ioss_Decomposition.C b/packages/seacas/libraries/ioss/src/Ioss_Decomposition.C
index fa2719a..0a7cfff 100644
--- a/packages/seacas/libraries/ioss/src/Ioss_Decomposition.C
+++ b/packages/seacas/libraries/ioss/src/Ioss_Decomposition.C
@@ -14,6 +14,7 @@
 #include <algorithm>
 #include <cassert>
 #include <fmt/ostream.h>
+#include <fmt/ranges.h>
 #include <numeric>
 
 #if !defined(NO_ZOLTAN_SUPPORT)
diff --git a/packages/seacas/libraries/ioss/src/Ioss_IOFactory.C b/packages/seacas/libraries/ioss/src/Ioss_IOFactory.C
index ef20a45..faa7943 100644
--- a/packages/seacas/libraries/ioss/src/Ioss_IOFactory.C
+++ b/packages/seacas/libraries/ioss/src/Ioss_IOFactory.C
@@ -11,6 +11,7 @@
 #include <Ioss_Version.h>
 #include <cstddef> // for nullptr
 #include <fmt/ostream.h>
+#include <fmt/ranges.h>
 #include <map>     // for _Rb_tree_iterator, etc
 #include <ostream> // for basic_ostream, etc
 #include <set>
diff --git a/packages/seacas/libraries/ioss/src/Ioss_StructuredBlock.C b/packages/seacas/libraries/ioss/src/Ioss_StructuredBlock.C
index 488d28a..0a1e5e2 100644
--- a/packages/seacas/libraries/ioss/src/Ioss_StructuredBlock.C
+++ b/packages/seacas/libraries/ioss/src/Ioss_StructuredBlock.C
@@ -14,6 +14,7 @@
 #include <Ioss_SmartAssert.h>
 #include <Ioss_StructuredBlock.h>
 #include <fmt/ostream.h>
+#include <fmt/ranges.h>
 
 #include <cstddef> // for size_t
 #include <numeric>
diff --git a/packages/seacas/libraries/ioss/src/Ioss_Utils.C b/packages/seacas/libraries/ioss/src/Ioss_Utils.C
index 241ff80..d234b8b 100644
--- a/packages/seacas/libraries/ioss/src/Ioss_Utils.C
+++ b/packages/seacas/libraries/ioss/src/Ioss_Utils.C
@@ -20,6 +20,7 @@
 #include <fmt/chrono.h>
 #include <fmt/format.h>
 #include <fmt/ostream.h>
+#include <fmt/ranges.h>
 #include <fstream>
 #include <sstream>
 #include <string>
diff --git a/packages/seacas/libraries/ioss/src/Ioss_ZoneConnectivity.C b/packages/seacas/libraries/ioss/src/Ioss_ZoneConnectivity.C
index 6888e17..f53c5e5 100644
--- a/packages/seacas/libraries/ioss/src/Ioss_ZoneConnectivity.C
+++ b/packages/seacas/libraries/ioss/src/Ioss_ZoneConnectivity.C
@@ -8,6 +8,7 @@
 #include <Ioss_ZoneConnectivity.h>
 #include <cstddef> // for size_t
 #include <fmt/ostream.h>
+#include <fmt/ranges.h>
 #include <string> // for string
 #include <vector> // for vector
 
diff --git a/packages/seacas/libraries/ioss/src/heartbeat/Iohb_Layout.h b/packages/seacas/libraries/ioss/src/heartbeat/Iohb_Layout.h
index c486d69..b67a5c6 100644
--- a/packages/seacas/libraries/ioss/src/heartbeat/Iohb_Layout.h
+++ b/packages/seacas/libraries/ioss/src/heartbeat/Iohb_Layout.h
@@ -9,6 +9,7 @@
 #include "iohb_export.h"
 
 #include <fmt/ostream.h>
+#include <fmt/ranges.h>
 #include <sstream>
 #include <string>
 #include <vector>
diff --git a/packages/seacas/libraries/ioss/src/main/io_info.C b/packages/seacas/libraries/ioss/src/main/io_info.C
index 53a162b..bcadb55 100644
--- a/packages/seacas/libraries/ioss/src/main/io_info.C
+++ b/packages/seacas/libraries/ioss/src/main/io_info.C
@@ -11,6 +11,7 @@
 #define FMT_DEPRECATED_OSTREAM
 #include <fmt/format.h>
 #include <fmt/ostream.h>
+#include <fmt/ranges.h>
 #if defined(SEACAS_HAVE_CGNS)
 #include <cgnslib.h>
 #endif
diff --git a/packages/seacas/libraries/ioss/src/main/io_modify.C b/packages/seacas/libraries/ioss/src/main/io_modify.C
index 50a217d..c01191b 100644
--- a/packages/seacas/libraries/ioss/src/main/io_modify.C
+++ b/packages/seacas/libraries/ioss/src/main/io_modify.C
@@ -58,6 +58,7 @@
 #include <fmt/color.h>
 #include <fmt/format.h>
 #include <fmt/ostream.h>
+#include <fmt/ranges.h>
 
 #if defined(SEACAS_HAVE_EXODUS)
 #include <exodus/Ioex_Internals.h>
diff --git a/packages/seacas/libraries/ioss/src/main/sphgen.C b/packages/seacas/libraries/ioss/src/main/sphgen.C
index f633d55..24c2e2f 100644
--- a/packages/seacas/libraries/ioss/src/main/sphgen.C
+++ b/packages/seacas/libraries/ioss/src/main/sphgen.C
@@ -14,6 +14,7 @@
 #include <cstdlib>
 #include <cstring>
 #include <fmt/ostream.h>
+#include <fmt/ranges.h>
 #include <iostream>
 #include <string>
 #include <vector>
