diff --git a/packages/seacas/libraries/ioss/src/Ioss_Decomposition.h b/packages/seacas/libraries/ioss/src/Ioss_Decomposition.h
index 4244ad5..9ce5eab 100644
--- a/packages/seacas/libraries/ioss/src/Ioss_Decomposition.h
+++ b/packages/seacas/libraries/ioss/src/Ioss_Decomposition.h
@@ -94,6 +94,8 @@ namespace Ioss {
     SetDecompositionData()                             = default;
     SetDecompositionData(const SetDecompositionData &) = delete;
     SetDecompositionData(SetDecompositionData &&)      = default;
+    SetDecompositionData &operator =(const SetDecompositionData &) = delete;
+    SetDecompositionData &operator =(SetDecompositionData &&) = default;
 
     ~SetDecompositionData()
     {
