diff --git a/src/record/sdk/CMakeLists.txt b/src/record/sdk/CMakeLists.txt
index 4be0697..d30c6de 100644
--- a/src/record/sdk/CMakeLists.txt
+++ b/src/record/sdk/CMakeLists.txt
@@ -81,7 +81,7 @@ install(
             development
     RUNTIME
         DESTINATION
-            ${CMAKE_INSTALL_BINDIR}
+            tools
         COMPONENT
             runtime
 )
diff --git a/src/sdk/CMakeLists.txt b/src/sdk/CMakeLists.txt
index 3616f40..a5019cb 100644
--- a/src/sdk/CMakeLists.txt
+++ b/src/sdk/CMakeLists.txt
@@ -90,7 +90,7 @@ install(
             development
     RUNTIME
         DESTINATION
-            ${CMAKE_INSTALL_BINDIR}
+            tools
         COMPONENT
             runtime
 )
