diff --git a/cmake/FindMETIS.cmake b/cmake/FindMETIS.cmake
index 5f41792..9c62022 100644
--- a/cmake/FindMETIS.cmake
+++ b/cmake/FindMETIS.cmake
@@ -106,5 +106,5 @@ if (METIS_LIBRARY_DEBUG)
     DEBUG)
 endif (METIS_LIBRARY_DEBUG)
 
-find_package_handle_standard_args (METIS REQUIRED_VARS
+find_package_handle_standard_args (metis REQUIRED_VARS
   METIS_INCLUDE_DIR METIS_LIBRARY VERSION_VAR METIS_VERSION)
diff --git a/internal/ceres/CMakeLists.txt b/internal/ceres/CMakeLists.txt
index 299b373..81f1be9 100644
--- a/internal/ceres/CMakeLists.txt
+++ b/internal/ceres/CMakeLists.txt
@@ -108,14 +108,14 @@ endif (NOT MINIGLOG)
 if (SUITESPARSE AND SuiteSparse_FOUND)
   # Define version information for use in Solver::FullReport.
   add_definitions(-DCERES_SUITESPARSE_VERSION="${SuiteSparse_VERSION}")
-  list(APPEND CERES_LIBRARY_PRIVATE_DEPENDENCIES SuiteSparse::CHOLMOD
-    SuiteSparse::SPQR)
+  list(APPEND CERES_LIBRARY_PRIVATE_DEPENDENCIES SuiteSparse::cholmod
+    SuiteSparse::spqr)
 endif (SUITESPARSE AND SuiteSparse_FOUND)
 
 if (CXSPARSE AND CXSparse_FOUND)
   # Define version information for use in Solver::FullReport.
   add_definitions(-DCERES_CXSPARSE_VERSION="${CXSparse_VERSION}")
-  list(APPEND CERES_LIBRARY_PRIVATE_DEPENDENCIES CXSparse::CXSparse)
+  list(APPEND CERES_LIBRARY_PRIVATE_DEPENDENCIES SuiteSparse::cxsparse)
 endif (CXSPARSE AND CXSparse_FOUND)
 
 if (ACCELERATESPARSE AND AccelerateSparse_FOUND)
