diff --git a/src/Makefile.am b/src/Makefile.am
index fc3acc3..9a86251 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -279,14 +279,14 @@ gpg-error.def: Makefile gpg-error.def.in
 # It is correct to use $(CC_FOR_BUILD) here.  We want to run the
 # program at build time.
 mkerrcodes$(EXEEXT_FOR_BUILD): mkerrcodes.c mkerrcodes.h Makefile
-	$(CC_FOR_BUILD) $(CFLAGS_FOR_BUILD) $(LDFLAGS_FOR_BUILD) \
+	$(CC_FOR_BUILD) $(CFLAGS_FOR_BUILD) $(CFLAGS) $(LDFLAGS_FOR_BUILD) \
 	$(CPPFLAGS_FOR_BUILD) -I. -I$(srcdir) -o $@ $(srcdir)/mkerrcodes.c
 
 if HAVE_W32CE_SYSTEM
 # It is correct to use $(CC_FOR_BUILD) here.  We want to run the
 # program at build time.
 mkw32errmap$(EXEEXT_FOR_BUILD): mkw32errmap.c mkw32errmap.tab.h Makefile
-	$(CC_FOR_BUILD) $(CFLAGS_FOR_BUILD) $(LDFLAGS_FOR_BUILD) \
+	$(CC_FOR_BUILD) $(CFLAGS_FOR_BUILD) $(CFLAGS) $(LDFLAGS_FOR_BUILD) \
 	$(CPPFLAGS_FOR_BUILD) -I. -I$(srcdir) -o $@ $(srcdir)/mkw32errmap.c
 endif
 
@@ -300,7 +300,7 @@ errnos-sym.h: Makefile mkstrtable.awk errnos.in
 
 
 mkheader$(EXEEXT_FOR_BUILD): mkheader.c Makefile
-	$(CC_FOR_BUILD) $(CFLAGS_FOR_BUILD) $(LDFLAGS_FOR_BUILD) \
+	$(CC_FOR_BUILD) $(CFLAGS_FOR_BUILD) $(CFLAGS) $(LDFLAGS_FOR_BUILD) \
 	$(CPPFLAGS_FOR_BUILD) -g -I. -I$(srcdir) -o $@ $(srcdir)/mkheader.c
 
 parts_of_gpg_error_h = 	 	\
