diff --git a/cmake/tribits/common_tpls/FindTPLHDF5.cmake b/cmake/tribits/common_tpls/FindTPLHDF5.cmake
index b1997e92b..4d008be71 100644
--- a/cmake/tribits/common_tpls/FindTPLHDF5.cmake
+++ b/cmake/tribits/common_tpls/FindTPLHDF5.cmake
@@ -9,7 +9,7 @@ if (Netcdf_ALLOW_MODERN)
-  set(minimum_modern_HDF5_version 1.13.2)
+  set(minimum_modern_HDF5_version 1.12.0)
   print_var(Netcdf_ALLOW_MODERN)
   message("-- Using find_package(HDF5 ${minimum_modern_HDF5_version} CONFIG) ...")
-  find_package(HDF5  ${minimum_modern_HDF5_version}  CONFIG)
+  find_package(HDF5 ${minimum_modern_HDF5_version} MODULE)
   if (HDF5_FOUND)
     message("-- Found HDF5_CONFIG=${HDF5_CONFIG}")
     message("-- Generating Netcdf::all_libs and NetcdfConfig.cmake")
@@ -17,7 +17,7 @@ if (Netcdf_ALLOW_MODERN)
     tribits_extpkg_create_imported_all_libs_target_and_config_file(
       HDF5
       INNER_FIND_PACKAGE_NAME  HDF5
-      IMPORTED_TARGETS_FOR_ALL_LIBS   ${HDF5_EXPORT_LIBRARIES})
+      IMPORTED_TARGETS_FOR_ALL_LIBS  hdf5::hdf5)
     set(HDF5_INTERNAL_IS_MODERN TRUE)
   endif()
 
diff --git a/cmake/TPLs/FindTPLBLAS.cmake b/cmake/TPLs/FindTPLBLAS.cmake
index dc04355fd..7af0deac4 100644
--- a/cmake/TPLs/FindTPLBLAS.cmake
+++ b/cmake/TPLs/FindTPLBLAS.cmake
@@ -54,27 +54,8 @@
 # @HEADER
 
 
-if (MSVC AND NOT
-    (BLAS_LIBRARY_DIRS  OR
-     (NOT "${BLAS_LIBRARY_NAMES}" STREQUAL "blas blas_win32" AND
-      NOT "${BLAS_LIBRARY_NAMES}" STREQUAL "") OR
-     BLAS_INCLUDE_DIRS  OR
-     BLAS_INCLUDE_NAMES OR
-     (NOT "${TPL_BLAS_LIBRARIES}" STREQUAL "blas" AND
-      NOT "${TPL_BLAS_LIBRARIES}" STREQUAL "") OR
-     TPL_BLAS_INCLUDE_DIRS)
-   )
-  # Find the CLAPACK built by CMake on the machine for MSVC if found it will
-  # set the BLAS and LAPACK libraries.  NOTE: This the FindCLAPACK module must
-  # be called every configure or this does not work!
-  # If the user has specified alternate name or location of their blas that
-  # will be used instead.
-  find_package(CLAPACK 3.2.1 NO_MODULE)
-  if (CLAPACK_FOUND)
-    advanced_set(TPL_BLAS_LIBRARIES blas
-      CACHE FILEPATH "Set from MSVC CLAPACK specialization")
-  endif()
-endif()
-
-tribits_tpl_find_include_dirs_and_libraries( BLAS
-  REQUIRED_LIBS_NAMES "blas blas_win32")
+find_package(BLAS REQUIRED)
+tribits_extpkg_create_imported_all_libs_target_and_config_file(
+  BLAS
+  INNER_FIND_PACKAGE_NAME BLAS
+  IMPORTED_TARGETS_FOR_ALL_LIBS BLAS::BLAS)
diff --git a/cmake/TPLs/FindTPLBoostLib.cmake b/cmake/TPLs/FindTPLBoostLib.cmake
index 6ebc9c847..1098ece9c 100644
--- a/cmake/TPLs/FindTPLBoostLib.cmake
+++ b/cmake/TPLs/FindTPLBoostLib.cmake
@@ -54,8 +54,9 @@
 # @HEADER
 
 
-TRIBITS_TPL_FIND_INCLUDE_DIRS_AND_LIBRARIES( BoostLib
-  REQUIRED_HEADERS boost/version.hpp boost/mpl/at.hpp
-  REQUIRED_LIBS_NAMES boost_program_options boost_system
-  )
+find_package(Boost COMPONENTS system program_options REQUIRED)
+tribits_extpkg_create_imported_all_libs_target_and_config_file(
+  Boost
+  INNER_FIND_PACKAGE_NAME Boost
+  IMPORTED_TARGETS_FOR_ALL_LIBS Boost::system Boost::program_options)
 
diff --git a/cmake/TPLs/FindTPLHWLOC.cmake b/cmake/TPLs/FindTPLHWLOC.cmake
index 715b3e9bd..d4840870c 100644
--- a/cmake/TPLs/FindTPLHWLOC.cmake
+++ b/cmake/TPLs/FindTPLHWLOC.cmake
@@ -63,9 +63,9 @@
 #    Source:        http://www.open-mpi.org/projects/hwloc/
 #    Version:       1.3
 #
-
-TRIBITS_TPL_FIND_INCLUDE_DIRS_AND_LIBRARIES( HWLOC
-  REQUIRED_HEADERS hwloc.h
-  REQUIRED_LIBS_NAMES "hwloc"
-  )
-
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(HWLOC REQUIRED IMPORTED_TARGET hwloc)
+tribits_extpkg_create_imported_all_libs_target_and_config_file(
+  HWLOC
+  INNER_FIND_PACKAGE_NAME HWLOC
+  IMPORTED_TARGETS_FOR_ALL_LIBS PkgConfig::HWLOC )
diff --git a/cmake/TPLs/FindTPLLAPACK.cmake b/cmake/TPLs/FindTPLLAPACK.cmake
index 443924df0..d5ad9d744 100644
--- a/cmake/TPLs/FindTPLLAPACK.cmake
+++ b/cmake/TPLs/FindTPLLAPACK.cmake
@@ -54,21 +54,9 @@
 # @HEADER
 
 
-if (MSVC AND NOT
-    (LAPACK_LIBRARY_DIRS  OR
-     (NOT "${LAPACK_LIBRARY_NAMES}" STREQUAL "lapack lapack_win32" AND
-      NOT "${LAPACK_LIBRARY_NAMES}" STREQUAL "") OR
-     LAPACK_INCLUDE_DIRS  OR
-     LAPACK_INCLUDE_NAMES OR
-     (NOT "${TPL_LAPACK_LIBRARIES}" STREQUAL "lapack" AND
-      NOT "${TPL_LAPACK_LIBRARIES}" STREQUAL "") OR
-     TPL_LAPACK_INCLUDE_DIRS)
-   )
-  if(CLAPACK_FOUND)
-    advanced_set(TPL_LAPACK_LIBRARIES lapack
-        CACHE FILEPATH "Set from MSVC CLAPACK specialization")
-  endif()
-endif()
 
-tribits_tpl_find_include_dirs_and_libraries( LAPACK
-  REQUIRED_LIBS_NAMES "lapack lapack_win32")
+find_package(LAPACK REQUIRED)
+tribits_extpkg_create_imported_all_libs_target_and_config_file(
+  LAPACK
+  INNER_FIND_PACKAGE_NAME LAPACK
+  IMPORTED_TARGETS_FOR_ALL_LIBS LAPACK::LAPACK)
diff --git a/cmake/TPLs/FindTPLMETIS.cmake b/cmake/TPLs/FindTPLMETIS.cmake
index 90911a3cc..72529e063 100644
--- a/cmake/TPLs/FindTPLMETIS.cmake
+++ b/cmake/TPLs/FindTPLMETIS.cmake
@@ -54,7 +54,8 @@
 # @HEADER
 
 
-TRIBITS_TPL_FIND_INCLUDE_DIRS_AND_LIBRARIES( METIS
-  REQUIRED_HEADERS metis.h
-  REQUIRED_LIBS_NAMES "metis"
-  )
+find_package(metis REQUIRED)
+tribits_extpkg_create_imported_all_libs_target_and_config_file(
+  METIS
+  INNER_FIND_PACKAGE_NAME metis
+  IMPORTED_TARGETS_FOR_ALL_LIBS metis)
diff --git a/cmake/TPLs/FindTPLMatio.cmake b/cmake/TPLs/FindTPLMatio.cmake
index ede07c2b3..32160130a 100644
--- a/cmake/TPLs/FindTPLMatio.cmake
+++ b/cmake/TPLs/FindTPLMatio.cmake
@@ -53,7 +53,7 @@
 # ************************************************************************
 # @HEADER
 
-
+list(APPEND CMAKE_FIND_LIBRARY_PREFIXES lib)
 TRIBITS_TPL_FIND_INCLUDE_DIRS_AND_LIBRARIES( Matio
   REQUIRED_HEADERS matio.h
   REQUIRED_LIBS_NAMES "matio")
diff --git a/cmake/TPLs/FindTPLX11.cmake b/cmake/TPLs/FindTPLX11.cmake
index 3ad956ae7..65e538d4e 100644
--- a/cmake/TPLs/FindTPLX11.cmake
+++ b/cmake/TPLs/FindTPLX11.cmake
@@ -53,8 +53,8 @@
 # ************************************************************************
 # @HEADER
 
-
-TRIBITS_TPL_FIND_INCLUDE_DIRS_AND_LIBRARIES( X11
-  REQUIRED_HEADERS X11/Xlib.h X11/Xutil.h X11/Xos.h X11/keysym.h X11/Xresource.h X11/Xatom.h
-  REQUIRED_LIBS_NAMES X11
-  )
+find_package(X11 REQUIRED)
+tribits_extpkg_create_imported_all_libs_target_and_config_file(
+  X11
+  INNER_FIND_PACKAGE_NAME X11
+  IMPORTED_TARGETS_FOR_ALL_LIBS X11::X11 )
diff --git a/cmake/TPLs/FindTPLZlib.cmake b/cmake/TPLs/FindTPLZlib.cmake
index cc607e9d8..3bbc823b3 100644
--- a/cmake/TPLs/FindTPLZlib.cmake
+++ b/cmake/TPLs/FindTPLZlib.cmake
@@ -54,7 +54,8 @@
 # @HEADER
 
 
-TRIBITS_TPL_FIND_INCLUDE_DIRS_AND_LIBRARIES( Zlib
-  REQUIRED_HEADERS zlib.h
-  REQUIRED_LIBS_NAMES  z
-  )
+find_package(ZLIB REQUIRED)
+tribits_extpkg_create_imported_all_libs_target_and_config_file(
+  Zlib
+  INNER_FIND_PACKAGE_NAME ZLIB
+  IMPORTED_TARGETS_FOR_ALL_LIBS ZLIB::ZLIB )
diff --git a/cmake/TPLs/FindTPLgtest.cmake b/cmake/TPLs/FindTPLgtest.cmake
index e68dc50db..5cf86afa2 100644
--- a/cmake/TPLs/FindTPLgtest.cmake
+++ b/cmake/TPLs/FindTPLgtest.cmake
@@ -54,7 +54,8 @@
 # @HEADER
 
 
-TRIBITS_TPL_FIND_INCLUDE_DIRS_AND_LIBRARIES( gtest
-  REQUIRED_HEADERS gtest/gtest.h 
-  REQUIRED_LIBS_NAMES gtest
-  )
+find_package(GTest REQUIRED)
+tribits_extpkg_create_imported_all_libs_target_and_config_file(
+  GTest
+  INNER_FIND_PACKAGE_NAME GTest
+  IMPORTED_TARGETS_FOR_ALL_LIBS GTest::GTest )
\ No newline at end of file
diff --git a/packages/seacas/applications/slice/CMakeLists.txt b/packages/seacas/applications/slice/CMakeLists.txt
index c6e4e1b1a..8f963c681 100644
--- a/packages/seacas/applications/slice/CMakeLists.txt
+++ b/packages/seacas/applications/slice/CMakeLists.txt
@@ -1,9 +1,6 @@
 TRIBITS_SUBPACKAGE(Slice)
 
-ASSERT_DEFINED(TPL_ENABLE_METIS)
-IF (TPL_ENABLE_METIS)
-   ADD_DEFINITIONS(-DUSE_METIS)
-ENDIF()
+
 
 INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR})
 
@@ -19,4 +16,10 @@ TRIBITS_ADD_EXECUTABLE(
   INSTALLABLE
   )
 
+ASSERT_DEFINED(TPL_ENABLE_METIS)
+IF (TPL_ENABLE_METIS)
+   ADD_DEFINITIONS(-DUSE_METIS)
+   target_link_libraries(slice PUBLIC metis)
+ENDIF()
+
 TRIBITS_SUBPACKAGE_POSTPROCESS()
diff --git a/packages/seacas/applications/slice/cmake/Dependencies.cmake b/packages/seacas/applications/slice/cmake/Dependencies.cmake
index 253b73411..bff87b656 100644
--- a/packages/seacas/applications/slice/cmake/Dependencies.cmake
+++ b/packages/seacas/applications/slice/cmake/Dependencies.cmake
@@ -1,4 +1,4 @@
 TRIBITS_PACKAGE_DEFINE_DEPENDENCIES(
   LIB_REQUIRED_PACKAGES SEACASExodus SEACASIoss SEACASSuplibC SEACASSuplibCpp
-  LIB_OPTIONAL_TPLS METIS
+  LIB_OPTIONAL_TPLS metis
 )
