diff --git a/src/rerun/time_column.cpp b/src/rerun/time_column.cpp
index 91fd6f9..8777d8e 100644
--- a/src/rerun/time_column.cpp
+++ b/src/rerun/time_column.cpp
@@ -4,6 +4,7 @@
 #include "c/rerun.h"
 
 #include <arrow/array/array_base.h>
+#include <arrow/array/util.h>
 #include <arrow/buffer.h>
 #include <arrow/c/bridge.h>
 
@@ -20,7 +21,9 @@ namespace rerun {
         // which may or may not be another copy (if the collection already owns the data this is just a move).
         auto length = static_cast<int64_t>(times.size());
         auto buffer = arrow_buffer_from_vector(std::move(times).to_vector());
-        array = std::make_shared<arrow::PrimitiveArray>(datatype, length, buffer);
+        auto buffers = std::vector<std::shared_ptr<arrow::Buffer>>{nullptr, buffer};
+        auto array_data = std::make_shared<arrow::ArrayData>(datatype, length, std::move(buffers));
+        array = arrow::MakeArray(array_data);
     }
 
     TimeColumn TimeColumn::from_nanoseconds(
