diff --git a/src/FieldMapping.cpp b/src/FieldMapping.cpp
index b1f1a1f..90612b1 100644
--- a/src/FieldMapping.cpp
+++ b/src/FieldMapping.cpp
@@ -49,9 +49,6 @@
 #include "FieldMapping.h"
 #include "Types.h"
 
-#ifdef WIN32
-#define isnan(__x__) _isnan(__x__) 
-#endif
 
 //----------------------------------------------------------------------------//
 
diff --git a/src/Log.cpp b/src/Log.cpp
index f78229c..d095a50 100644
--- a/src/Log.cpp
+++ b/src/Log.cpp
@@ -40,8 +40,9 @@
 */
 
 //----------------------------------------------------------------------------//
-
+#ifndef _WIN32
 #include <unistd.h>
+#endif
 #include <ios>
 #include <fstream>
 
