diff --git a/librabbitmq/CMakeLists.txt b/librabbitmq/CMakeLists.txt
index bd5369a..f6d703f 100644
--- a/librabbitmq/CMakeLists.txt
+++ b/librabbitmq/CMakeLists.txt
@@ -113,7 +113,7 @@ set(RABBITMQ_SOURCES
     ${AMQP_SSL_SRCS}
 )
 
-add_definitions(-DAMQP_BUILD)
+add_definitions(-DAMQP_BUILD -D_CRT_SECURE_NO_WARNINGS)
 
 set(RMQ_LIBRARIES ${AMQP_SSL_LIBS} ${SOCKET_LIBRARIES} ${LIBRT} ${CMAKE_THREAD_LIBS_INIT})
 
