diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e736ae5..9aa61eb 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -93,6 +93,10 @@ target_compile_definitions(foonathan_memory PUBLIC
 if(NOT MSVC)
 target_compile_features(foonathan_memory PUBLIC cxx_constexpr)
 endif()
+target_compile_definitions(foonathan_memory INTERFACE
+    $<$<CONFIG:DEBUG>:Z_VCPKG_FOONATHAN_MEMORY_DEBUG=1>
+    $<$<NOT:$<CONFIG:DEBUG>>:Z_VCPKG_FOONATHAN_MEMORY_DEBUG=0>
+)
 
 if(${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang")
     if("${CMAKE_CXX_SIMULATE_ID}" STREQUAL "MSVC")
