diff --git a/cdk/cmake/DepFindProtobuf.cmake b/cdk/cmake/DepFindProtobuf.cmake
index 1fc785e..d9eaecc 100644
--- a/cdk/cmake/DepFindProtobuf.cmake
+++ b/cdk/cmake/DepFindProtobuf.cmake
@@ -45,8 +45,6 @@
 #
 
 if(TARGET ext::protobuf)
-  return()
-endif()
 
 message(STATUS "Setting up Protobuf.")
 
@@ -65,6 +63,10 @@ add_ext_targets(protobuf
   EXECUTABLE protoc pb_protoc
 )
 
+endif()
+if(COMMAND mysqlx_protobuf_generate_cpp)
+  return()
+endif()
 
 # Standard PROTOBUF_GENERATE_CPP modified to our usage
 function(mysqlx_protobuf_generate_cpp SRCS HDRS)
