diff --git a/ThirdParty/netcdf/vtknetcdf/ncconfig.h.in b/ThirdParty/netcdf/vtknetcdf/ncconfig.h.in
index 0a19c41..0d92371 100644
--- a/ThirdParty/netcdf/vtknetcdf/ncconfig.h.in
+++ b/ThirdParty/netcdf/vtknetcdf/ncconfig.h.in
@@ -110,7 +110,7 @@
 #cmakedefine HAVE_LIBHDF5_HL
 
 /* Define to `int' if system doesn't define.  */
-#cmakedefine ssize_t @ssize_t@
+/* #cmakedefine ssize_t @ssize_t@ */
 
 /* Define to `int' if system doesn't define.  */
 #cmakedefine ptrdiff_t @ptrdiff_t@
