From 0655297a0500e8789d0d455e2ae01f08f1402385 Mon Sep 17 00:00:00 2001
From: "DESKTOP-6M1EIQI\\Raahil" <raahil.sha@zimaging.io>
Date: Wed, 2 Feb 2022 13:08:01 -0500
Subject: [PATCH] FindLZ4

---
 CMake/FindLZ4.cmake | 43 +++++--------------------------------------
 1 file changed, 5 insertions(+), 38 deletions(-)

diff --git a/CMake/FindLZ4.cmake b/CMake/FindLZ4.cmake
index 8c94e3bcd1..4fadd98a14 100644
--- a/CMake/FindLZ4.cmake
+++ b/CMake/FindLZ4.cmake
@@ -1,38 +1,5 @@
-find_path(LZ4_INCLUDE_DIR
-  NAMES lz4.h
-  DOC "lz4 include directory")
-mark_as_advanced(LZ4_INCLUDE_DIR)
-find_library(LZ4_LIBRARY
-  NAMES lz4 liblz4
-  DOC "lz4 library")
-mark_as_advanced(LZ4_LIBRARY)
-
-if (LZ4_INCLUDE_DIR)
-  file(STRINGS "${LZ4_INCLUDE_DIR}/lz4.h" _lz4_version_lines
-    REGEX "#define[ \t]+LZ4_VERSION_(MAJOR|MINOR|RELEASE)")
-  string(REGEX REPLACE ".*LZ4_VERSION_MAJOR *\([0-9]*\).*" "\\1" _lz4_version_major "${_lz4_version_lines}")
-  string(REGEX REPLACE ".*LZ4_VERSION_MINOR *\([0-9]*\).*" "\\1" _lz4_version_minor "${_lz4_version_lines}")
-  string(REGEX REPLACE ".*LZ4_VERSION_RELEASE *\([0-9]*\).*" "\\1" _lz4_version_release "${_lz4_version_lines}")
-  set(LZ4_VERSION "${_lz4_version_major}.${_lz4_version_minor}.${_lz4_version_release}")
-  unset(_lz4_version_major)
-  unset(_lz4_version_minor)
-  unset(_lz4_version_release)
-  unset(_lz4_version_lines)
-endif ()
-
-include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(LZ4
-  REQUIRED_VARS LZ4_LIBRARY LZ4_INCLUDE_DIR
-  VERSION_VAR LZ4_VERSION)
-
-if (LZ4_FOUND)
-  set(LZ4_INCLUDE_DIRS "${LZ4_INCLUDE_DIR}")
-  set(LZ4_LIBRARIES "${LZ4_LIBRARY}")
-
-  if (NOT TARGET LZ4::LZ4)
-    add_library(LZ4::LZ4 UNKNOWN IMPORTED)
-    set_target_properties(LZ4::LZ4 PROPERTIES
-      IMPORTED_LOCATION "${LZ4_LIBRARY}"
-      INTERFACE_INCLUDE_DIRECTORIES "${LZ4_INCLUDE_DIR}")
-  endif ()
-endif ()
+find_package(LZ4 CONFIG REQUIRED)
+set_target_properties(lz4::lz4 PROPERTIES IMPORTED TRUE GLOBAL TRUE)
+if(NOT TARGET LZ4::LZ4)
+add_library(LZ4::LZ4 ALIAS lz4::lz4)
+endif()
-- 
2.32.0.windows.1

