diff --git a/CMake/FindLibHaru.cmake b/CMake/FindLibHaru.cmake
index 1f5b881..8a4c426 100644
--- a/CMake/FindLibHaru.cmake
+++ b/CMake/FindLibHaru.cmake
@@ -1,3 +1,11 @@
+find_package(unofficial-libharu CONFIG REQUIRED)
+set(LibHaru_FOUND 1)
+if(NOT TARGET LibHaru::LibHaru)
+  add_library(LibHaru::LibHaru INTERFACE IMPORTED)
+  target_link_libraries(LibHaru::LibHaru INTERFACE unofficial::libharu::haru)
+endif()
+
+if(0)
 find_path(LibHaru_INCLUDE_DIR
   NAMES hpdf.h
   DOC "libharu include directory")
@@ -85,3 +93,4 @@ if (LibHaru_FOUND)
     endif ()
   endif ()
 endif ()
+endif (0)
