From 7ea05601c9439d58575089e5d8f40fa27c5d4ad1 Mon Sep 17 00:00:00 2001
From: "DESKTOP-6M1EIQI\\Raahil" <raahil.sha@zimaging.io>
Date: Wed, 2 Feb 2022 13:09:16 -0500
Subject: [PATCH] FindPROJ

---
 CMake/FindLibPROJ.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMake/FindLibPROJ.cmake b/CMake/FindLibPROJ.cmake
index e2344bb171..bac4c747e3 100644
--- a/CMake/FindLibPROJ.cmake
+++ b/CMake/FindLibPROJ.cmake
@@ -9,7 +9,7 @@ find_library(LibPROJ_LIBRARY_RELEASE
 mark_as_advanced(LibPROJ_LIBRARY_RELEASE)
 
 find_library(LibPROJ_LIBRARY_DEBUG
-  NAMES projd
+  NAMES projd proj_d proj NAMES_PER_DIR
   DOC "libproj debug library")
 mark_as_advanced(LibPROJ_LIBRARY_DEBUG)
 
-- 
2.32.0.windows.1

