diff --git a/CMake/FindLibHaru.cmake b/CMake/FindLibHaru.cmake
index 7ef877a49..2a18e0694 100644
--- a/CMake/FindLibHaru.cmake
+++ b/CMake/FindLibHaru.cmake
@@ -45,8 +45,13 @@ if (LibHaru_FOUND)
 
   if (NOT TARGET LibHaru::LibHaru)
     include(vtkDetectLibraryType)
-    vtk_detect_library_type(libharu_library_type
-      PATH "${LibHaru_LIBRARY}")
+    if(LibHaru_LIBRARY_RELEASE)
+        vtk_detect_library_type(libharu_library_type
+            PATH "${LibHaru_LIBRARY_RELEASE}")
+    else()
+        vtk_detect_library_type(libharu_library_type
+            PATH "${LibHaru_LIBRARY_DEBUG}")
+    endif()
     add_library(LibHaru::LibHaru "${libharu_library_type}" IMPORTED)
     unset(libharu_library_type)
     set_target_properties(LibHaru::LibHaru PROPERTIES
