diff --git a/ThirdParty/hdf5/CMakeLists.txt b/ThirdParty/hdf5/CMakeLists.txt
index 74a965a..6a8b216 100644
--- a/ThirdParty/hdf5/CMakeLists.txt
+++ b/ThirdParty/hdf5/CMakeLists.txt
@@ -16,7 +16,7 @@ vtk_module_third_party(
     STANDARD_INCLUDE_DIRS)
 
 set(vtkhdf5_is_parallel 0)
-if (VTK_MODULE_USE_EXTERNAL_vtkhdf5 AND HDF5_IS_PARALLEL)
+if (VTK_MODULE_USE_EXTERNAL_vtkhdf5 AND HDF5_IS_PARALLEL AND ENABLE_HDF5_PARALLEL)
   if (NOT TARGET VTK::mpi)
     message(FATAL_ERROR
       "An external MPI-aware HDF5 requires that VTK be built with MPI support "
