diff --git a/CMake/patches/99/FindHDF5.cmake b/CMake/patches/99/FindHDF5.cmake
index ecc7f568..ae4ef466 100644
--- a/CMake/patches/99/FindHDF5.cmake
+++ b/CMake/patches/99/FindHDF5.cmake
@@ -509,7 +509,7 @@ if(NOT HDF5_FOUND AND NOT HDF5_NO_FIND_PACKAGE_CONFIG_FILE)
         set(HDF5_IS_PARALLEL ${HDF5_ENABLE_PARALLEL})
         set(HDF5_INCLUDE_DIRS ${HDF5_INCLUDE_DIR})
         set(HDF5_LIBRARIES)
-        if (NOT TARGET hdf5 AND NOT TARGET hdf5-static AND NOT TARGET hdf5-shared)
+        if (TARGET hdf5::hdf5-static OR TARGET hdf5::hdf5-shared)
             # Some HDF5 versions (e.g. 1.8.18) used hdf5::hdf5 etc
             set(_target_prefix "hdf5::")
         endif()
