diff --git a/IO/ExportPDF/vtkPDFContextDevice2D.cxx b/IO/ExportPDF/vtkPDFContextDevice2D.cxx
index a51945dc9b..1cbc3945af 100644
--- a/IO/ExportPDF/vtkPDFContextDevice2D.cxx
+++ b/IO/ExportPDF/vtkPDFContextDevice2D.cxx
@@ -1631,21 +1631,21 @@ void vtkPDFContextDevice2D::ApplyLineWidth(float width)
 void vtkPDFContextDevice2D::ApplyLineType(int type)
 {
   // These match the OpenGL2 implementation:
-  static const HPDF_UINT16 noPen[] = { 0, 10 };
+  static const HPDF_REAL noPen[] = { 0, 10 };
   static const HPDF_UINT noPenLen = 2;
 
-  static const HPDF_UINT16 dash[] = { 8 };
+  static const HPDF_REAL dash[] = { 8 };
   static const HPDF_UINT dashLen = 1;
 
-  static const HPDF_UINT16 dot[] = { 1, 7 };
-  static const HPDF_UINT16 denseDot[] = { 1, 3 };
+  static const HPDF_REAL dot[] = { 1, 7 };
+  static const HPDF_REAL denseDot[] = { 1, 3 };
   static const HPDF_UINT dotLen = 2;
 
-  static const HPDF_UINT16 dashDot[] = { 4, 6, 2, 4 };
+  static const HPDF_REAL dashDot[] = { 4, 6, 2, 4 };
   static const HPDF_UINT dashDotLen = 4;
 
   // This is dash-dot-dash, but eh. It matches the OpenGL2 0x1C47 pattern.
-  static const HPDF_UINT16 dashDotDot[] = { 3, 3, 1, 3, 3, 3 };
+  static const HPDF_REAL dashDotDot[] = { 3, 3, 1, 3, 3, 3 };
   static const HPDF_UINT dashDotDotLen = 6;
 
   switch (type)
