diff --git a/CMake/vtkInstallCMakePackage.cmake b/CMake/vtkInstallCMakePackage.cmake
index bcb20446c2..24c06ed8ad 100644
--- a/CMake/vtkInstallCMakePackage.cmake
+++ b/CMake/vtkInstallCMakePackage.cmake
@@ -62,7 +62,6 @@ set(vtk_cmake_module_files
   FindGLEW.cmake
   FindJOGL.cmake
   FindJsonCpp.cmake
-  FindLibHaru.cmake
   FindLibPROJ.cmake
   FindLibXml2.cmake
   FindLZ4.cmake
diff --git a/ThirdParty/libharu/CMakeLists.txt b/ThirdParty/libharu/CMakeLists.txt
index bac8dce188..44004fd9bb 100644
--- a/ThirdParty/libharu/CMakeLists.txt
+++ b/ThirdParty/libharu/CMakeLists.txt
@@ -4,16 +4,15 @@ vtk_module_third_party(
     VERSION       "2.4.0"
     STANDARD_INCLUDE_DIRS
   EXTERNAL
-    PACKAGE LibHaru
+    PACKAGE unofficial-libharu CONFIG_MODE
     # Unreleased. Requires these PRs:
     # https://github.com/libharu/libharu/pull/157
     # https://github.com/libharu/libharu/pull/187
-    VERSION 2.4.0
-    TARGETS LibHaru::LibHaru
+    TARGETS unofficial::libharu::hpdf
     STANDARD_INCLUDE_DIRS)
 
 include(vtkDetectLibraryType)
-vtk_detect_library_shared(vtklibharu LibHaru::LibHaru)
+vtk_detect_library_shared(vtklibharu unofficial::libharu::hpdf)
 
 set(HPDF_DLL 0)
 if (WIN32 AND vtklibharu_is_shared)
