diff --git a/CMake/FindHDF5.cmake b/CMake/FindHDF5.cmake
index 6d558e39b1..9511e9a1ad 100644
--- a/CMake/FindHDF5.cmake
+++ b/CMake/FindHDF5.cmake
@@ -3,11 +3,11 @@
 # (BUG #0014363).
 
 # include the default FindHDF5.cmake.
-if(CMAKE_VERSION VERSION_LESS 3.6.1)
+#if(CMAKE_VERSION VERSION_LESS 3.6.1)
   include(${CMAKE_CURRENT_LIST_DIR}/NewCMake/FindHDF5.cmake)
-else()
-  include(${CMAKE_ROOT}/Modules/FindHDF5.cmake)
-endif()
+#else()
+#  include(${CMAKE_ROOT}/Modules/FindHDF5.cmake)
+#endif()
 
 if(HDF5_FOUND AND (HDF5_IS_PARALLEL OR HDF5_ENABLE_PARALLEL))
   include(vtkMPI)
