diff --git a/GUISupport/Qt/QVTKOpenGLWindow.cxx b/GUISupport/Qt/QVTKOpenGLWindow.cxx
index 9d14b1d..0ac2f95 100644
--- a/GUISupport/Qt/QVTKOpenGLWindow.cxx
+++ b/GUISupport/Qt/QVTKOpenGLWindow.cxx
@@ -14,6 +14,7 @@
 =========================================================================*/
 #include "QVTKOpenGLWindow.h"
 
+#include <GL/gl.h>
 #include <QApplication>
 #include <QOpenGLContext>
 #include <QOpenGLFramebufferObject>
diff --git a/GUISupport/Qt/vtkQWidgetTexture.cxx b/GUISupport/Qt/vtkQWidgetTexture.cxx
index 8cde5fc..9413d46 100644
--- a/GUISupport/Qt/vtkQWidgetTexture.cxx
+++ b/GUISupport/Qt/vtkQWidgetTexture.cxx
@@ -18,14 +18,14 @@
 #include "vtkOpenGLResourceFreeCallback.h"
 #include "vtkOpenGLState.h"
 
-#include "vtkOpenGLError.h"
-
 #include <QGraphicsProxyWidget>
 #include <QGraphicsScene>
 #include <QOpenGLPaintDevice>
 #include <QPainter>
 #include <QWidget>
 
+#include "vtkOpenGLError.h"
+
 //------------------------------------------------------------------------------
 vtkStandardNewMacro(vtkQWidgetTexture);
 
