diff --git a/Rendering/ZSpace/vtk.module b/Rendering/ZSpace/vtk.module
index cf6b6a2d6365b763f80e2e166928ebc2176febdf..1e6869ee05da1de1eea38207d8cfab7cce10d646 100644
--- a/Rendering/ZSpace/vtk.module
+++ b/Rendering/ZSpace/vtk.module
@@ -8,6 +8,8 @@ SPDX_LICENSE_IDENTIFIER
   BSD-3-Clause
 SPDX_COPYRIGHT_TEXT
   Copyright (c) Ken Martin, Will Schroeder, Bill Lorensen
+CONDITION
+  WIN32
 DEPENDS
   VTK::CommonCore
   VTK::CommonDataModel
