diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e911c28..bbeb99b 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -206,6 +206,11 @@ if(WITH_ZLIB)
   target_link_libraries(rdkafka PUBLIC ZLIB::ZLIB)
 endif()
 
+if(WITH_CURL)
+  find_package(CURL CONFIG REQUIRED)
+  target_link_libraries(rdkafka PUBLIC CURL::libcurl)
+endif()
+
 if(WITH_ZSTD)
   target_link_libraries(rdkafka PRIVATE ${ZSTD_LIBRARY})
   target_include_directories(rdkafka PRIVATE ${ZSTD_INCLUDE_DIR})
