diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6d300da..e8627d1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -61,20 +61,18 @@ set(leptonica_INSTALL_CMAKE_DIR "lib/cmake/leptonica"
     CACHE STRING "Install destination for CMake package files")
 
 if(NOT SW_BUILD)
-    find_package(GIF)
-    find_package(JPEG)
-    find_package(PNG)
-    find_package(TIFF)
-    find_package(ZLIB)
+    find_package(GIF REQUIRED)
+    find_package(JPEG REQUIRED)
+    find_package(PNG REQUIRED)
+    find_package(TIFF REQUIRED)
+    find_package(ZLIB REQUIRED)
     find_package(PkgConfig)
     if (PKG_CONFIG_FOUND)
         pkg_check_modules(WEBP libwebp QUIET)
         pkg_check_modules(WEBPMUX libwebpmux>=${MINIMUM_WEBPMUX_VERSION} QUIET)
-        pkg_check_modules(JP2K libopenjp2>=2.0 QUIET)
     endif()
     if(NOT WEBP)
-        find_path(WEBP_INCLUDE_DIR /webp/decode.h)
-        find_library(WEBP_LIBRARY NAMES webp)
+        find_package(WebP CONFIG REQUIRED)
         if (WEBP_INCLUDE_DIR AND WEBP_LIBRARY)
             set(WEBP 1)
             set(WEBP_FOUND TRUE)
@@ -100,8 +98,7 @@ if(NOT SW_BUILD)
         endif()
     endif()
     if(NOT JP2K)
-        find_path(JP2K_INCLUDE_DIR /openjpeg-2.3/openjpeg.h)
-        find_library(JP2K_LIBRARY NAMES openjp2)
+        find_package(OpenJPEG CONFIG REQUIRED)
         if (JP2K_INCLUDE_DIR AND JP2K_LIBRARY)
              set(JP2K 1)
              set(JP2K_FOUND TRUE)
@@ -110,7 +107,7 @@ if(NOT SW_BUILD)
              set(HAVE_LIBJP2K 1)
         endif()
     endif()
-    if(NOT JP2K)
+    if(0)
         find_path(JP2K_INCLUDE_DIR /openjpeg-2.4/openjpeg.h)
         find_library(JP2K_LIBRARY NAMES openjp2)
         if (JP2K_INCLUDE_DIR AND JP2K_LIBRARY)
@@ -216,7 +213,7 @@ include(Configure)
 
 configure_file(${AUTOCONFIG_SRC} ${AUTOCONFIG} @ONLY)
 
-set(INCLUDE_DIR "${CMAKE_INSTALL_PREFIX}/include" "${CMAKE_INSTALL_PREFIX}/include/leptonica")
+set(INCLUDE_DIR "\${CMAKE_CURRENT_LIST_DIR}/../../include" "\${CMAKE_CURRENT_LIST_DIR}/../../include/leptonica")
 
 ###############################################################################
 #
