diff --git a/CMakeLists.txt b/CMakeLists.txt
index debc232..d9e0b4f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -61,7 +61,10 @@ if(NOT EXISTS ${PROJECT_SOURCE_DIR}/.cppan)
     if(NOT WEBP)
         message(STATUS "Looking for WEBP")
         find_path(WEBP_INCLUDE_DIR /webp/decode.h)
-        find_library(WEBP_LIBRARY NAMES webp)
+        include(SelectLibraryConfigurations)
+        find_library(WEBP_LIBRARY_RELEASE NAMES webp)
+        find_library(WEBP_LIBRARY_DEBUG NAMES webpd)
+        select_library_configurations(WEBP)
         if (WEBP_INCLUDE_DIR AND WEBP_LIBRARY)
             set(WEBP 1)
             set(WEBP_FOUND TRUE)
