diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8b9ca9d..8a5519f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,7 +8,7 @@
 #
 ###############################################################################
 
-cmake_minimum_required(VERSION 3.1.3)
+cmake_minimum_required(VERSION 3.13)
 cmake_policy(SET CMP0054 NEW)
 
 # In-source builds are disabled.
@@ -64,7 +64,7 @@ if(NOT SW_BUILD)
     find_package(TIFF REQUIRED)
     find_package(ZLIB REQUIRED)
     find_package(PkgConfig)
-    find_package(OpenJPEG REQUIRED)
+    find_package(OpenJPEG CONFIG REQUIRED)
     if (PKG_CONFIG_FOUND)
         pkg_check_modules(WEBP libwebp QUIET)
         pkg_check_modules(WEBPMUX libwebpmux>=${MINIMUM_WEBPMUX_VERSION} QUIET)
