--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -58,20 +58,19 @@ if(UNIX)
 endif()
 
 if(NOT SW_BUILD)
-    find_package(GIF)
-    find_package(JPEG)
-    find_package(PNG)
-    find_package(TIFF)
-    find_package(ZLIB)
+    find_package(GIF REQUIRED)
+    find_package(JPEG REQUIRED)
+    find_package(PNG REQUIRED)
+    find_package(TIFF REQUIRED)
+    find_package(ZLIB REQUIRED)
     find_package(PkgConfig)
+    find_package(OpenJPEG REQUIRED)
     if (PKG_CONFIG_FOUND)
         pkg_check_modules(WEBP libwebp QUIET)
         pkg_check_modules(WEBPMUX libwebpmux>=${MINIMUM_WEBPMUX_VERSION} QUIET)
-        pkg_check_modules(JP2K libopenjp2>=2.0 QUIET)
     endif()
     if(NOT WEBP)
-        find_path(WEBP_INCLUDE_DIR /webp/decode.h)
-        find_library(WEBP_LIBRARY NAMES webp)
+        find_package(WebP CONFIG REQUIRED)
         if (WEBP_INCLUDE_DIR AND WEBP_LIBRARY)
             set(WEBP 1)
             set(WEBP_FOUND TRUE)
@@ -213,7 +212,7 @@ include(Configure)
 
 configure_file(${AUTOCONFIG_SRC} ${AUTOCONFIG} @ONLY)
 
-set(INCLUDE_DIR "${CMAKE_INSTALL_PREFIX}/include" "${CMAKE_INSTALL_PREFIX}/include/leptonica")
+set(INCLUDE_DIR "\${CMAKE_CURRENT_LIST_DIR}/../../include" "\${CMAKE_CURRENT_LIST_DIR}/../../include/leptonica")
 
 ###############################################################################
 #
