diff --git a/configure.ac b/configure.ac
index 920612f..06d5dfa 100644
--- a/configure.ac
+++ b/configure.ac
@@ -154,9 +154,10 @@ AC_CHECK_TYPES([struct termios2], [], [], [[#include <asm/termbits.h>]])
 AC_CHECK_TYPES([struct termios2], [], [], [[#include <asm/termbits.h>]])
 
+test -n "$WARNING_CFLAGS" || \
 WARNING_CFLAGS="-Wall \
 -Wmissing-declarations -Wmissing-prototypes \
 -Wnested-externs -Wpointer-arith \
 -Wsign-compare -Wchar-subscripts \
 -Wstrict-prototypes -Wshadow \
 -Wformat-security"
 AC_SUBST([WARNING_CFLAGS])
