diff --git a/configure.ac b/configure.ac
index 06d5dfa..f3122ac 100644
--- a/configure.ac
+++ b/configure.ac
@@ -133,6 +133,8 @@ if test "$os_cygwin" = "false"; then
     AC_CHECK_HEADERS([winsock2.h], HAVE_WINSOCK2_H=yes)
     if test "x$HAVE_WINSOCK2_H" = "xyes"; then
         LIBS="$LIBS -lws2_32"
+        PC_LIBS_PRIVATE="$PC_LIBS_PRIVATE -lws2_32"
+        AC_SUBST(PC_LIBS_PRIVATE)
     AC_SUBST(LIBS)
     fi
 fi
diff --git a/libmodbus.pc.in b/libmodbus.pc.in
index f1a9cbf..f7b4d08 100644
--- a/libmodbus.pc.in
+++ b/libmodbus.pc.in
@@ -7,4 +7,5 @@ Name: modbus
 Description: Modbus library
 Version: @VERSION@
 Libs: -L${libdir} -lmodbus
+Libs.private: @PC_LIBS_PRIVATE@
 Cflags: -I${includedir}/modbus
