diff -Naur a/cmake/project-config-version.cmake.in b/cmake/project-config-version.cmake.in
--- a/cmake/project-config-version.cmake.in	2023-07-25 07:37:40.000000000 -0500
+++ b/cmake/project-config-version.cmake.in	2023-11-10 16:58:52.835325073 -0600
@@ -48,10 +48,6 @@
   # Reject if the user asks for an incompatible precsision.
   set (REASON "GEOGRAPHICLIB_PRECISION = @GEOGRAPHICLIB_PRECISION@")
   set (PACKAGE_VERSION_UNSUITABLE TRUE)
-elseif (NOT CMAKE_CROSSCOMPILING_STR STREQUAL "@CMAKE_CROSSCOMPILING_STR@")
-  # Reject if there's a mismatch in ${CMAKE_CROSSCOMPILING}.
-  set (REASON "cross-compiling = @CMAKE_CROSSCOMPILING@")
-  set (PACKAGE_VERSION_UNSUITABLE TRUE)
 elseif (CMAKE_CROSSCOMPILING AND
     NOT (CMAKE_SYSTEM_NAME STREQUAL "@CMAKE_SYSTEM_NAME@" AND
       CMAKE_SYSTEM_PROCESSOR STREQUAL "@CMAKE_SYSTEM_PROCESSOR@"))
