diff --git a/src/msvcbuild.bat b/src/msvcbuild.bat
index 69c0c61a..2abb7749 100644
--- a/src/msvcbuild.bat
+++ b/src/msvcbuild.bat
@@ -102,10 +102,10 @@ buildvm -m folddef -o lj_folddef.h lj_opt_fold.c
 @set LJLINKTYPE=%LJLINKTYPE_DEBUG%
 :NODEBUG
 @set LJCOMPILE=%LJCOMPILE% %LJCOMPILETARGET%
-@set LJLINK=%LJLINK% %LJLINKTYPE% %LJLINKTARGET%
+@set LJLINK=%LJLINK% %LJLINKTARGET%
 @if "%1"=="amalg" goto :AMALGDLL
 @if "%1"=="static" goto :STATIC
-%LJCOMPILE% %LJDYNBUILD% lj_*.c lib_*.c
+%LJCOMPILE% %LJDYNBUILD% lj_*.c lib_*.c /Fdlua51.pdb
 @if errorlevel 1 goto :BAD
 @if "%1"=="mixed" goto :STATICLIB
 %LJLINK% /DLL /OUT:%LJDLLNAME% lj_*.obj lib_*.obj
@@ -136,7 +136,7 @@ buildvm -m folddef -o lj_folddef.h lj_opt_fold.c
 if exist %LJDLLNAME%.manifest^
   %LJMT% -manifest %LJDLLNAME%.manifest -outputresource:%LJDLLNAME%;2
 
-%LJCOMPILE% luajit.c
+%LJCOMPILE% luajit.c /Fdluajit.pdb
 @if errorlevel 1 goto :BAD
 %LJLINK% /OUT:luajit.exe luajit.obj %LJLIBNAME%
 @if errorlevel 1 goto :BAD
