diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 709a1b5..bd3085f 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -150,11 +150,11 @@ foreach(target_name libconfig libconfig++)
   install(EXPORT ${target_name}Targets
     FILE "${target_name}Config.cmake"
     NAMESPACE libconfig::
-    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/libconfig
+    DESTINATION share/${target_name}
     )
 
   install(FILES
     "${CMAKE_CURRENT_BINARY_DIR}/${target_name}ConfigVersion.cmake"
-    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/libconfig"
+    DESTINATION share/${target_name}
     )
 endforeach()
