diff --git a/third_party/aws/aws-checksums.bazel b/third_party/aws/aws-checksums.bazel
--- a/third_party/aws/aws-checksums.bazel
+++ b/third_party/aws/aws-checksums.bazel
@@ -7,6 +7,13 @@
 
 exports_files(["LICENSE"])
 
+load("@bazel_skylib//lib:selects.bzl", "selects")
+
+selects.config_setting_group(
+    name = "is_linux_debug",
+    match_all = ["@org_tensorflow//tensorflow:linux_x86_64", "@org_tensorflow//tensorflow:debug"],
+)
+
 cc_library(
     name = "aws-checksums",
     srcs = select({
@@ -27,4 +34,10 @@
     deps = [
         "@aws-c-common",
     ],
+    defines = select({
+        ":is_linux_debug": [
+            "DEBUG_BUILD"
+        ],
+        "//conditions:default": [],
+    }),
 )

diff --git a/tensorflow/core/kernels/data/experimental/io_ops.cc b/tensorflow/core/kernels/data/experimental/io_ops.cc
--- a/tensorflow/core/kernels/data/experimental/io_ops.cc
+++ b/tensorflow/core/kernels/data/experimental/io_ops.cc
@@ -28,6 +28,12 @@
 namespace data {
 namespace experimental {

+/*static*/ constexpr const char* const LoadDatasetOp::kCompression;
+/*static*/ constexpr const char* const LoadDatasetOp::kReaderFunc;
+/*static*/ constexpr const char* const LoadDatasetOp::kReaderFuncTarguments;
+
+/*static*/ constexpr const int SaveDatasetOp::kFileFormatVersion;
+
 SaveDatasetOp::SaveDatasetOp(OpKernelConstruction* ctx)
     : HybridAsyncOpKernel(ctx, "tf_data_save_dataset") {
   OP_REQUIRES_OK(ctx, ctx->GetAttr(kCompression, &compression_));

diff --git a/tensorflow/core/kernels/data/experimental/snapshot_dataset_op.cc b/tensorflow/core/kernels/data/experimental/snapshot_dataset_op.cc
--- a/tensorflow/core/kernels/data/experimental/snapshot_dataset_op.cc
+++ b/tensorflow/core/kernels/data/experimental/snapshot_dataset_op.cc
@@ -63,6 +63,14 @@
 namespace data {
 namespace experimental {

+/*static*/ constexpr const char* const SnapshotDatasetV2Op::kCompression;
+/*static*/ constexpr const char* const SnapshotDatasetV2Op::kReaderFunc;
+/*static*/ constexpr const char* const SnapshotDatasetV2Op::kShardFunc;
+/*static*/ constexpr const char* const SnapshotDatasetV2Op::kReaderFuncTarguments;
+/*static*/ constexpr const char* const SnapshotDatasetV2Op::kShardFuncTarguments;
+
+/*static*/ constexpr const int SnapshotDatasetV2Op::kFileFormatVersion;
+
 // ==== Snapshot Implementation ====

 /* The current snapshot on-disk layout is as follows:

diff --git a/tensorflow/core/kernels/data/experimental/data_service_dataset_op.cc b/tensorflow/core/kernels/data/experimental/data_service_dataset_op.cc
--- a/tensorflow/core/kernels/data/experimental/data_service_dataset_op.cc
+++ b/tensorflow/core/kernels/data/experimental/data_service_dataset_op.cc
@@ -52,6 +52,8 @@
 /* static */ constexpr const char* const
     DataServiceDatasetOp::kMaxOutstandingRequests;
 /* static */ constexpr const char* const
+    DataServiceDatasetOp::kTaskRefreshIntervalHintMs;
+/* static */ constexpr const char* const
     DataServiceDatasetOp::kIterationCounter;
 /* static */ constexpr const char* const DataServiceDatasetOp::kOutputTypes;
 /* static */ constexpr const char* const DataServiceDatasetOp::kOutputShapes;
