diff --git a/src/OpenColorIO/res/OpenColorIO.pc.in b/src/OpenColorIO/res/OpenColorIO.pc.in
index f25c280..d9a450b 100644
--- a/src/OpenColorIO/res/OpenColorIO.pc.in
+++ b/src/OpenColorIO/res/OpenColorIO.pc.in
@@ -9,3 +9,5 @@ Description: @PROJECT_DESCRIPTION@
 URL: @PROJECT_HOMEPAGE_URL@
 Libs: -L${libdir} -l@PROJECT_NAME@
 Cflags: -I${includedir}
+Libs.private: -lpystring
+Requires.private: expat Imath yaml-cpp
