diff --git a/qwtconfig.pri b/qwtconfig.pri
index 0b054e0..f93fef6 100644
--- a/qwtconfig.pri
+++ b/qwtconfig.pri
@@ -72,7 +72,10 @@ QWT_INSTALL_FEATURES  = $${QWT_INSTALL_PREFIX}/features
 # it will be a static library.
 ######################################################################
 
+CONFIG(dynamic, dynamic|static) {
+
 QWT_CONFIG           += QwtDll
+}
 
 ######################################################################
 # QwtPlot enables all classes, that are needed to use the QwtPlot
