diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1cf379c..650fc72 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -169,7 +169,7 @@ check_function_exists(symlink HAVE_SYMLINK)
 # Check for readlink support
 check_function_exists(readlink HAVE_READLINK)
 
-configure_file(mz_config.h.cmakein ${CMAKE_CURRENT_BINARY_DIR}/mz_config.h @ONLY NEWLINE_STYLE UNIX)
+configure_file(mz_config.h.cmakein mz_config.h @ONLY NEWLINE_STYLE UNIX)
 list(APPEND MINIZIP_HDR ${CMAKE_CURRENT_BINARY_DIR}/mz_config.h)
 
 if(MZ_LIBCOMP)
@@ -206,7 +206,7 @@ if(MZ_ZLIB)
         if(BUILD_SHARED_LIBS)
             list(APPEND MINIZIP_LIB_PUBLIC zlib-ng::zlib)
         else()
-            list(APPEND MINIZIP_LIB_PUBLIC zlib-ng::zlibstatic)
+            list(APPEND MINIZIP_LIB_PUBLIC zlib-ng::zlib)
         endif()
 
         set(PC_PRIVATE_DEPS "zlib-ng")
@@ -363,6 +363,7 @@ endif()
 if(MZ_PPMD)
     message(STATUS "Using PPMD")
 
+if(0)
     clone_repo(ppmd https://github.com/ip7z/7zip "26.00")
 
     # get the 7zip/PPMd version number
@@ -370,6 +371,7 @@ if(MZ_PPMD)
     string(REGEX MATCH "MY_VERSION_NUMBERS \"([0-9]*\.[0-9]*)\"" _ ${version})
     set(7ZIP_VERSION_NUMBER ${CMAKE_MATCH_1})
     message(STATUS "  Found 7Zip/PPMd ${7ZIP_VERSION_NUMBER}")
+endif()
 
     # The PPMd repository does not support cmake so we have to create
     # a PPMd library ourselves
@@ -399,7 +401,7 @@ endif()
 if(MZ_ZSTD)
     # Check if zstd is present
     if(NOT MZ_FORCE_FETCH_LIBS)
-        find_package(zstd QUIET CONFIG)
+        find_package(ZSTD NAMES zstd CONFIG REQUIRED)
     endif()
 
     if(zstd_FOUND AND NOT MZ_FORCE_FETCH_LIBS)
