diff --git "a/include/mgl2/define.h" "b/include/mgl2/define.h"
index e23ec04..8e79086 100644
--- "a/include/mgl2/define.h"
+++ "b/include/mgl2/define.h"
@@ -141,7 +140,9 @@ typedef long msize;
 #define getcwd	_getcwd
 #define chdir	_chdir // BORLAND has chdir
 #endif
+#if (_MSC_VER<1500)
 #define snprintf _snprintf
+#endif
 #if (_MSC_VER<1600) // based on https://hg.python.org/cpython/rev/9aedb876c2d7
 #define hypot	_hypot
 #endif
