diff --git "a/src/canvas_cf.cpp" "b/src/canvas_cf.cpp"
index 5e1443a..26a0a8e 100644
--- "a/src/canvas_cf.cpp"
+++ "b/src/canvas_cf.cpp"
@@ -50,7 +50,6 @@ void MGL_EXPORT MGL_FUNC_INIT mgl_init()	// TODO try to add ld option: "-init mg
 #endif
 		for(long i=0;i<360;i++)	mgl_cos[i] = cos(i*M_PI/180.);
 		if(!mglDefFont)		mglDefFont = new mglFont(MGL_DEF_FONT_NAME);
-		if(!mglDefaultGr)	mglDefaultGr = new mglCanvas(600,400);
 	}
 }
 void MGL_EXPORT MGL_FUNC_FINI mgl_fini()	// TODO try to add ld option: "-fini mgl_fini"
@@ -434,7 +433,7 @@ void MGL_EXPORT mgl_set_legend_marks(HMGL gr, int num)
 uintptr_t MGL_EXPORT mgl_create_graph_(int *width, int *height)
 {	return uintptr_t(new mglCanvas(*width,*height));	}
 void MGL_EXPORT mgl_delete_graph_(uintptr_t *gr)	{	delete _GR_;	}
-uintptr_t MGL_EXPORT mgl_default_graph_()	{	return uintptr_t(&mglDefaultGr);	}
+uintptr_t MGL_EXPORT mgl_default_graph_()	{	return uintptr_t(mgl_default_graph());	}
 void MGL_EXPORT mgl_set_size_scl_(double *scl)	{	mgl_set_size_scl(*scl);	}
 void MGL_EXPORT mgl_set_size_(uintptr_t *gr, int *width, int *height)
 {	mgl_set_size(_GR_,*width,*height);	}
