diff --git a/Lib/include/DBow3/BowVector.h b/Lib/include/DBow3/BowVector.h
index cdbf517..90862b9 100644
--- a/Lib/include/DBow3/BowVector.h
+++ b/Lib/include/DBow3/BowVector.h
@@ -10,13 +10,11 @@
 #ifndef __D_T_BOW_VECTOR__
 #define __D_T_BOW_VECTOR__
 
+#include <cstdint>
 #include <map>
 #include <vector>
 #include <ostream>
 #include "exports.h"
-#if _WIN32
-#include <cstdint>
-#endif
 namespace DBoW3 {
 
 /// Id of words
