diff --git a/plugins/qca-gnupg/CMakeLists.txt b/plugins/qca-gnupg/CMakeLists.txt
index cbe33b6..09d1123 100644
--- a/plugins/qca-gnupg/CMakeLists.txt
+++ b/plugins/qca-gnupg/CMakeLists.txt
@@ -53,6 +53,4 @@ if(NOT DEVELOPER_MODE)
           LIBRARY DESTINATION "${QCA_CRYPTO_INSTALL_DIR}"
           ARCHIVE DESTINATION "${QCA_CRYPTO_INSTALL_DIR}"
           RUNTIME DESTINATION "${QCA_CRYPTO_INSTALL_DIR}")
-
-  install_pdb(qca-gnupg ${QCA_CRYPTO_INSTALL_DIR})
 endif()
diff --git a/plugins/qca-logger/CMakeLists.txt b/plugins/qca-logger/CMakeLists.txt
index 9f8b434..9a62b55 100644
--- a/plugins/qca-logger/CMakeLists.txt
+++ b/plugins/qca-logger/CMakeLists.txt
@@ -19,6 +19,4 @@ if(NOT DEVELOPER_MODE)
           LIBRARY DESTINATION "${QCA_CRYPTO_INSTALL_DIR}"
           ARCHIVE DESTINATION "${QCA_CRYPTO_INSTALL_DIR}"
           RUNTIME DESTINATION "${QCA_CRYPTO_INSTALL_DIR}")
-
-  install_pdb(qca-logger ${QCA_CRYPTO_INSTALL_DIR})
 endif()
diff --git a/plugins/qca-ossl/CMakeLists.txt b/plugins/qca-ossl/CMakeLists.txt
index efffbb5..bd6522e 100644
--- a/plugins/qca-ossl/CMakeLists.txt
+++ b/plugins/qca-ossl/CMakeLists.txt
@@ -64,8 +64,6 @@ if(OPENSSL_FOUND)
             LIBRARY DESTINATION "${QCA_CRYPTO_INSTALL_DIR}"
             ARCHIVE DESTINATION "${QCA_CRYPTO_INSTALL_DIR}"
             RUNTIME DESTINATION "${QCA_CRYPTO_INSTALL_DIR}")
-
-    install_pdb(qca-ossl ${QCA_CRYPTO_INSTALL_DIR})
   endif()
 
 else()
diff --git a/plugins/qca-softstore/CMakeLists.txt b/plugins/qca-softstore/CMakeLists.txt
index 0896801..069bfb0 100644
--- a/plugins/qca-softstore/CMakeLists.txt
+++ b/plugins/qca-softstore/CMakeLists.txt
@@ -19,6 +19,4 @@ if(NOT DEVELOPER_MODE)
           LIBRARY DESTINATION "${QCA_CRYPTO_INSTALL_DIR}"
           ARCHIVE DESTINATION "${QCA_CRYPTO_INSTALL_DIR}"
           RUNTIME DESTINATION "${QCA_CRYPTO_INSTALL_DIR}")
-
-  install_pdb(qca-softstore ${QCA_CRYPTO_INSTALL_DIR})
 endif()
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 7066839..41877d0 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -167,6 +167,5 @@ if(NOT DEVELOPER_MODE)
           FRAMEWORK DESTINATION "${QCA_LIBRARY_INSTALL_DIR}"
           PUBLIC_HEADER DESTINATION "${QCA_FULL_INCLUDE_INSTALL_DIR}" INCLUDES DESTINATION "${QCA_FULL_INCLUDE_INSTALL_DIR}"
   )
-  install_pdb(${QCA_LIB_NAME} ${QCA_BINARY_INSTALL_DIR})
 endif()
 
