diff --git a/gdk-pixbuf/meson.build b/gdk-pixbuf/meson.build
index d7aa127b48..c313dc272d 100644
--- a/gdk-pixbuf/meson.build
+++ b/gdk-pixbuf/meson.build
@@ -214,6 +214,7 @@ gdkpixbuf = library('gdk_pixbuf-2.0',
                     dependencies: [
                       gdk_pixbuf_deps,
                       included_loaders_deps,
+                      loaders_deps
                     ],
                     install: true)
 
