diff --git a/build-aux/post-install.bat b/build-aux/post-install.bat
index 6994d49e1..b4789dd62 100644
--- a/build-aux/post-install.bat
+++ b/build-aux/post-install.bat
@@ -7,8 +7,8 @@ set binary_version=%3
 set libdir_windows=%libdir:/=\%
 
 if not "%DESTDIR%" == "" goto warn_msg
-if not exist %libdir_windows%\gdk-pixbuf-2.0\%binary_version%\ mkdir %libdir_windows%\gdk-pixbuf-2.0\%binary_version%
-%bindir%\gdk-pixbuf-query-loaders > %libdir_windows%\gdk-pixbuf-2.0\%binary_version%\loaders.cache
+if not exist %libdir_windows%\..\gdk-pixbuf-2.0\%binary_version%\ mkdir %libdir_windows%\..\gdk-pixbuf-2.0\%binary_version%
+%bindir%\gdk-pixbuf-query-loaders > %libdir_windows%\..\gdk-pixbuf-2.0\%binary_version%\loaders.cache
 
 goto end
 
@@ -18,7 +18,7 @@ echo *** Warning: loaders.cache not built
 echo ***
 echo *** You should generate this file manually on the host system
 echo *** using:
-echo ***   gdk-pixbuf-query-loaders ^> %libdir_windows%\gdk-pixbuf-2.0\%binary_version%\loaders.cache
+echo ***   gdk-pixbuf-query-loaders ^> %libdir_windows%\..\gdk-pixbuf-2.0\%binary_version%\loaders.cache
 echo ***
 
 :end
diff --git a/build-aux/post-install.sh b/build-aux/post-install.sh
index 661bceeef..0b1a46527 100644
--- a/build-aux/post-install.sh
+++ b/build-aux/post-install.sh
@@ -5,14 +5,14 @@ libdir="$2"
 binary_version="$3"
 
 if [ -z "$DESTDIR" ]; then
-        mkdir -p "$libdir/gdk-pixbuf-2.0/$binary_version"
-        $bindir/gdk-pixbuf-query-loaders > "$libdir/gdk-pixbuf-2.0/$binary_version/loaders.cache"
+        mkdir -p "$libdir/../gdk-pixbuf-2.0/$binary_version"
+        $bindir/gdk-pixbuf-query-loaders > "$libdir/../gdk-pixbuf-2.0/$binary_version/loaders.cache"
 else
         echo "***"
         echo "*** Warning: loaders.cache not built"
         echo "***"
         echo "*** You should generate this file manually on the host system"
         echo "*** using:"
-        echo "***   gdk-pixbuf-query-loaders > $libdir/gdk-pixbuf-2.0/$binary_version/loaders.cache"
+        echo "***   gdk-pixbuf-query-loaders > $libdir/../gdk-pixbuf-2.0/$binary_version/loaders.cache"
         echo "***"
 fi
diff --git a/meson.build b/meson.build
index c38607029..4b5df91ae 100644
--- a/meson.build
+++ b/meson.build
@@ -308,12 +310,12 @@ endif
 # Don't check and build the jpeg loader if native_windows_loaders is true
 if get_option('jpeg') and not native_windows_loaders
   if cc.has_header('jpeglib.h')
-    jpeg_dep = cc.find_library('jpeg', required: false)
-    if cc.get_id() == 'msvc' and not jpeg_dep.found()
+    jpeg_dep = dependency('libjpeg', required: false)
+    #if cc.get_id() == 'msvc' and not jpeg_dep.found()
       # The IJG JPEG library builds the .lib file as libjpeg.lib in its MSVC build system,
       # so look for it as well when jpeg.lib cannot be found
-      jpeg_dep = cc.find_library('libjpeg', required: false)
-    endif
+    #  jpeg_dep = cc.find_library('libjpeg', required: false)
+    #endif
     if jpeg_dep.found() and cc.has_function('jpeg_destroy_decompress', dependencies: jpeg_dep)
       enabled_loaders += 'jpeg'
       loaders_deps += jpeg_dep
