diff --git a/meson_options.txt b/meson_options.txt
index dff9be8..828fd25 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -1,3 +1,7 @@
+option('VCPKG_HOST_TRIPLET', type: 'string',
+  description: 'vcpkg host triplet for pylibdir search'
+)
+
 option('cairo', type: 'feature', value : 'auto',
   description: 'Use cairo for tests'
 )
diff --git a/tools/g-ir-tool-template.in b/tools/g-ir-tool-template.in
index 469af0c..f165a98 100755
--- a/tools/g-ir-tool-template.in
+++ b/tools/g-ir-tool-template.in
@@ -1,4 +1,4 @@
-#!@PYTHON_CMD@
+#!/usr/bin/env python3
 # -*- Mode: Python -*-
 # GObject-Introspection - a framework for introspecting GObject libraries
 # Copyright (C) 2008  Johan Dahlin
@@ -46,22 +46,21 @@ if debug:
 filedir = os.path.dirname(__file__)
 
 # Try using relative paths first so that the installation prefix is relocatable
-datadir = os.path.abspath(os.path.join(filedir, '..', 'share'))
+datadir = os.path.abspath(os.path.join(filedir, '..', '..', 'share'))
 # Fallback to hard-coded paths if the relocatable paths are wrong
+datadir = os.getenv('VCPKG_GI_DATADIR', datadir)
 if not os.path.isdir(os.path.join(datadir, 'gir-1.0')):
-    datadir = "@datarootdir@"
+    raise Exception('Failed to determine datadir')
 
 builtins.__dict__['DATADIR'] = datadir
 
-gir_dir = os.path.abspath(os.path.join(filedir, '..', '@gir_dir_prefix@', 'gir-1.0'))
-# Fallback to hard-coded paths if the relocatable paths are wrong
-if not os.path.isdir(gir_dir):
-    gir_dir = "@GIR_DIR@"
+gir_dir = os.path.abspath(os.path.join(datadir, 'gir-1.0'))
 
 builtins.__dict__['GIR_DIR'] = gir_dir
 
 # Again, relative paths first so that the installation prefix is relocatable
-pylibdir = os.path.abspath(os.path.join(filedir, '..', 'lib', 'gobject-introspection'))
+pylibdir = os.path.abspath(os.path.join(filedir, '..', '..', 'lib', 'gobject-introspection'))
+host_pylibdir = os.path.abspath(os.path.join(filedir, '..', '..', '..', '@VCPKG_HOST_TRIPLET@', 'lib', 'gobject-introspection'))
 
 # EXT_SUFFIX for py3 SO for py2
 py_mod_suffix = sysconfig.get_config_var('EXT_SUFFIX') or sysconfig.get_config_var('SO')
@@ -89,13 +88,21 @@ if not os.path.isfile(os.path.join(pylibdir, 'giscanner', '_giscanner' + py_mod_
         gdump_path = os.path.join(builddir, 'giscanner', 'gdump.c')
         if os.path.isfile(gdump_path):
             builtins.__dict__['GDUMP_PATH'] = gdump_path
+    elif os.path.isfile(os.path.join(host_pylibdir, 'giscanner', '_giscanner' + py_mod_suffix)):
+        pylibdir = host_pylibdir
     else:
-        # Okay, we're not running uninstalled and the prefix is not
-        # relocatable. Use hard-coded libdir.
-        pylibdir = os.path.join('@libdir@', 'gobject-introspection')
+        raise Exception('Could not determine pylibdir')
 
 sys.path.insert(0, pylibdir)
 
+vcpkg_gi_lib_path     = os.getenv('VCPKG_GI_LIBDIR', None)
+vcpkg_gi_lib_path_var = os.getenv('VCPKG_GI_LIBDIR_VAR', None)
+if vcpkg_gi_lib_path_var is not None and vcpkg_gi_lib_path is not None:
+    ld_lib_path = os.getenv(vcpkg_gi_lib_path_var, '')
+    if ld_lib_path != '':
+        ld_lib_path = os.pathsep + ld_lib_path
+    os.environ[vcpkg_gi_lib_path_var] = vcpkg_gi_lib_path + ld_lib_path
+
 from giscanner.utils import dll_dirs
 dll_dirs = dll_dirs()
 dll_dirs.add_dll_dirs(['gio-2.0'])
diff --git a/tools/meson.build b/tools/meson.build
index 7ac43cd..33eee12 100644
--- a/tools/meson.build
+++ b/tools/meson.build
@@ -23,6 +23,7 @@ foreach tool : tools
   tools_conf.set('gir_dir_prefix', gir_dir_prefix)
   tools_conf.set('GIR_DIR', girdir)
   tools_conf.set('PYTHON_CMD', python_cmd)
+  tools_conf.set('VCPKG_HOST_TRIPLET', get_option('VCPKG_HOST_TRIPLET'))
 
   tools_conf.set('TOOL_MODULE', tool[1])
   tools_conf.set('TOOL_FUNCTION', tool[2])
