diff --git a/CMakeLists.txt b/CMakeLists.txt
index 67b61712ec..73711753ae 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -253,10 +253,12 @@ if(SFML_OS_MACOSX)
         endif()
     endif()
 
-    # only the default architecture (i.e. 64-bit) is supported
-    if(NOT CMAKE_OSX_ARCHITECTURES STREQUAL "x86_64")
-        message(FATAL_ERROR "Only 64-bit architecture is supported")
-    endif()
+    # only the default x86_64 or arm64 are supported
+    foreach (arch IN LISTS CMAKE_OSX_ARCHITECTURES)
+        if (NOT (arch STREQUAL "x86_64" OR arch STREQUAL "arm64"))
+            message(FATAL_ERROR "Invalid arch ${arch}")
+        endif()
+    endforeach()
 
     # configure Xcode templates
     set(XCODE_TEMPLATES_ARCH "\$(NATIVE_ARCH_ACTUAL)")
