diff --git a/CMakeLists.txt b/CMakeLists.txt
index fc9a268..e38a3bf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -123,10 +123,9 @@ if(SFML_OS_ANDROID)
     endif()

     # install everything in $NDK/sources/ because this path is appended by the NDK (convenient)
-    set(CMAKE_INSTALL_PREFIX ${CMAKE_ANDROID_NDK}/sources/third_party/sfml)
-
+
     # we install libs in a subdirectory named after the ABI
-    set(CMAKE_INSTALL_LIBDIR "${CMAKE_INSTALL_LIBDIR}/${CMAKE_ANDROID_ARCH_ABI}")
+    set(CMAKE_INSTALL_LIBDIR "${CMAKE_INSTALL_LIBDIR}")
     # pass shared STL configuration (if any)
     if (CMAKE_ANDROID_STL_TYPE MATCHES "_shared")
         add_definitions("-DSTL_LIBRARY=${CMAKE_ANDROID_STL_TYPE}")
@@ -315,14 +314,14 @@ endif()

 # setup the install rules
 if(NOT SFML_BUILD_FRAMEWORKS)
-    install(DIRECTORY include
-            DESTINATION .
+    install(DIRECTORY include/
+            DESTINATION "include"
             COMPONENT devel
             FILES_MATCHING PATTERN "*.hpp" PATTERN "*.inl")

     if(SFML_GENERATE_PDB)
         install(DIRECTORY ${PROJECT_BINARY_DIR}/lib
-                DESTINATION .
+                DESTINATION lib
                 COMPONENT devel
                 FILES_MATCHING PATTERN "*.pdb")
     endif()
@@ -522,7 +521,6 @@ elseif(SFML_OS_ANDROID)
     endif()

     # install Android.mk so the NDK knows how to set up SFML
-    install(FILES src/SFML/Android.mk DESTINATION .)

 endif()
 
diff --git a/src/SFML/Audio/CMakeLists.txt b/src/SFML/Audio/CMakeLists.txt
index d27dc6d..f8883a1 100644
--- a/src/SFML/Audio/CMakeLists.txt
+++ b/src/SFML/Audio/CMakeLists.txt
@@ -82,6 +82,9 @@ sfml_add_library(sfml-audio

 # setup dependencies
 target_link_libraries(sfml-audio PRIVATE OpenAL)
+if(SFML_OS_ANDROID)
+    target_link_libraries(sfml-audio PRIVATE android log)
+endif()

 # minimp3 sources
 target_include_directories(sfml-audio SYSTEM PRIVATE "${PROJECT_SOURCE_DIR}/extlibs/headers/minimp3")
