diff --git a/viskores/exec/cuda/internal/WrappedOperators.h b/viskores/exec/cuda/internal/WrappedOperators.h
index c00f1414b..de09ad9b0 100644
--- a/viskores/exec/cuda/internal/WrappedOperators.h
+++ b/viskores/exec/cuda/internal/WrappedOperators.h
@@ -33,6 +33,9 @@ VISKORES_THIRDPARTY_POST_INCLUDE
 #if THRUST_VERSION >= 200500
 #include <cuda/std/type_traits>
 #endif
+#if THRUST_VERSION >= 300300
+#include <cuda/functional>
+#endif
 
 namespace viskores
 {
@@ -199,30 +202,40 @@ struct WrappedBinaryPredicate
 }
 } //namespace viskores::exec::cuda::internal
 
-namespace thrust
-{
-namespace detail
-{
 //
 // We tell Thrust that our WrappedBinaryOperator is commutative so that we
 // activate numerous fast paths inside thrust which are only available when
 // the binary functor is commutative and the T type is is_arithmetic
 //
 //
+#if THRUST_VERSION >= 300300
+namespace cuda
+{
+template <typename T, typename F>
+inline constexpr bool
+  is_commutative_v<::viskores::exec::cuda::internal::WrappedBinaryOperator<T, F>, T> =
+    ::cuda::std::is_arithmetic<T>::value;
+}
+#else // THRUST_VERSION < 300300
+namespace thrust
+{
+namespace detail
+{
 #if THRUST_VERSION >= 200500
 template <typename T, typename F>
 struct is_commutative<viskores::exec::cuda::internal::WrappedBinaryOperator<T, F>>
   : public ::cuda::std::is_arithmetic<T>
 {
 };
-#else
+#else  // THRUST_VERSION < 200500
 template <typename T, typename F>
 struct is_commutative<viskores::exec::cuda::internal::WrappedBinaryOperator<T, F>>
-  : public thrust::detail::is_arithmetic<T>
+  : public ::thrust::detail::is_arithmetic<T>
 {
 };
-#endif
+#endif // THRUST_VERSION < 200500
 }
 } //namespace thrust::detail
+#endif // THRUST_VERSION < 300300
 
 #endif //viskores_exec_cuda_internal_WrappedOperators_h
