diff --git a/CMakeLists.txt b/CMakeLists.txt
index c78bd16..bf31091 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -193,7 +193,9 @@ OPTION(OSGEARTH_INSTALL_SHADERS "Whether to deploy GLSL shaders when doing a Mak
 # TinyXML is an XML parsing library
 SET (WITH_EXTERNAL_TINYXML FALSE CACHE BOOL "Use bundled or system wide version of TinyXML")
 IF (WITH_EXTERNAL_TINYXML)
-    find_package(TinyXML REQUIRED)
+    find_package(tinyxml CONFIG REQUIRED)
+    set(TINYXML_FOUND 1)
+    set(TINYXML_LIBRARY unofficial-tinyxml::unofficial-tinyxml)
 ENDIF (WITH_EXTERNAL_TINYXML)
 
 # postfix settings for various configs
