diff --git a/CMakeLists.txt b/CMakeLists.txt
index 049e37e..c78bd16 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -121,9 +121,9 @@ option(OSGEARTH_ENABLE_GEOCODER "Enable the geocoder (GDAL/OGR must be built wit
 
 # Mobile/GLES:
 IF (OSGEARTH_USE_GLES)
-    find_package(OpenGLES) 
+    find_package(OpenGLES REQUIRED)
 ELSE ()
-    find_package(OpenGL)
+    find_package(OpenGL REQUIRED)
 ENDIF (OSGEARTH_USE_GLES)
 
 # required
@@ -132,28 +132,28 @@ find_package(CURL REQUIRED)
 find_package(GDAL REQUIRED)
 
 # optional
-find_package(GEOS)
-find_package(Sqlite3)
-find_package(Draco)
-find_package(BASISU)
-find_package(GLEW)
-find_package(Protobuf)
-find_package(WEBP)
+find_package(GEOS REQUIRED)
+find_package(Sqlite3 REQUIRED)
+find_package(Draco REQUIRED)
+find_package(BASISU REQUIRED)
+find_package(GLEW REQUIRED)
+find_package(Protobuf REQUIRED)
+find_package(WEBP REQUIRED)
 
 if(OSGEARTH_ENABLE_PROFILING)
-    find_package(Tracy)
+    find_package(Tracy REQUIRED)
 endif()
 
 if(OSGEARTH_BUILD_ZIP_PLUGIN)
-    find_package(LIBZIP)
+    find_package(LIBZIP REQUIRED)
 endif()
 
 if(OSGEARTH_BUILD_TRITON_NODEKIT)
-    find_package(Triton QUIET)
+    find_package(Triton QUIET REQUIRED)
 endif()
 
 if(OSGEARTH_BUILD_SILVERLINING_NODEKIT)
-    find_package(SilverLining QUIET)
+    find_package(SilverLining QUIET REQUIRED)
 endif()
 
 # Sqlite enables the MBTiles format:
@@ -182,18 +182,18 @@ SET (PROTOBUF_USE_DLLS FALSE CACHE BOOL "Set this to true if Protobuf is compile
 # Duktape is the JavaScript interpreter
 SET (WITH_EXTERNAL_DUKTAPE FALSE CACHE BOOL "Use bundled or system wide version of Duktape")
 IF (WITH_EXTERNAL_DUKTAPE)
-    find_package(Duktape)
+    find_package(Duktape REQUIRED)
 ENDIF (WITH_EXTERNAL_DUKTAPE)
 
 # Whether to install shaders (glsl files).
-# If true, shaders install into a resources folder. If false, they are inlined in the 
+# If true, shaders install into a resources folder. If false, they are inlined in the
 # code and you cannot tweak them after install.
 OPTION(OSGEARTH_INSTALL_SHADERS "Whether to deploy GLSL shaders when doing a Make INSTALL" OFF)
 
 # TinyXML is an XML parsing library
 SET (WITH_EXTERNAL_TINYXML FALSE CACHE BOOL "Use bundled or system wide version of TinyXML")
 IF (WITH_EXTERNAL_TINYXML)
-    find_package(TinyXML)
+    find_package(TinyXML REQUIRED)
 ENDIF (WITH_EXTERNAL_TINYXML)
 
 # postfix settings for various configs
@@ -272,4 +272,4 @@ ADD_SUBDIRECTORY(src)
 
 if (OSGEARTH_BUILD_DOCS)
     ADD_SUBDIRECTORY(docs)
-endif()    
+endif()
