--- a/libparmetis/CMakeLists.txt	Sat Mar 30 17:24:50 2013
+++ b/libparmetis/CMakeLists.txt	Wed Dec 21 19:40:28 2016
@@ -5,10 +5,10 @@
 # Create libparmetis
 add_library(parmetis ${ParMETIS_LIBRARY_TYPE} ${parmetis_sources})
 # Link with metis and MPI libraries.
-target_link_libraries(parmetis metis ${MPI_LIBRARIES})
+target_link_libraries(parmetis ${METIS_LIBRARY} ${MPI_LIBRARIES})
 set_target_properties(parmetis PROPERTIES LINK_FLAGS "${MPI_LINK_FLAGS}")
 
 install(TARGETS parmetis
   LIBRARY DESTINATION lib
-  RUNTIME DESTINATION lib
+  RUNTIME DESTINATION bin
   ARCHIVE DESTINATION lib)
