diff --git a/src/Makefile.shlib b/src/Makefile.shlib
index fa81f6ffdd..a9675ab88c 100644
--- a/src/Makefile.shlib
+++ b/src/Makefile.shlib
@@ -108,6 +108,11 @@ endif
 
 ifeq ($(PORTNAME), darwin)
   ifdef soname
+    ifneq (,$(findstring linux-android,$(host_os)))
+      # Android uses unversioned shared libraries
+      shlib            = $(shlib_bare)
+     soname           = $(shlib_bare)
+    endif
     # linkable library
     ifneq ($(SO_MAJOR_VERSION), 0)
       version_link	= -compatibility_version $(SO_MAJOR_VERSION) -current_version $(SO_MAJOR_VERSION).$(SO_MINOR_VERSION)
