diff --git a/src/libmongoc/src/mongoc/mongoc-client.c b/src/libmongoc/src/mongoc/mongoc-client.c
index 15517e6..bd762d0 100644
--- a/src/libmongoc/src/mongoc/mongoc-client.c
+++ b/src/libmongoc/src/mongoc/mongoc-client.c
@@ -18,8 +18,8 @@
 #include "mongoc-config.h"
 #ifdef MONGOC_HAVE_DNSAPI
 /* for DnsQuery_UTF8 */
-#include <Windows.h>
-#include <WinDNS.h>
+#include <windows.h>
+#include <windns.h>
 #include <ws2tcpip.h>
 #else
 #if defined(MONGOC_HAVE_RES_NSEARCH) || defined(MONGOC_HAVE_RES_SEARCH)
diff --git a/src/libmongoc/src/mongoc/mongoc-socket.c b/src/libmongoc/src/mongoc/mongoc-socket.c
index a77d805..15d39b4 100644
--- a/src/libmongoc/src/mongoc/mongoc-socket.c
+++ b/src/libmongoc/src/mongoc/mongoc-socket.c
@@ -25,7 +25,7 @@
 #include "mongoc-socket-private.h"
 #include "mongoc-trace-private.h"
 #ifdef _WIN32
-#include <Mstcpip.h>
+#include <mstcpip.h>
 #include <process.h>
 #endif
 
diff --git a/src/libmongoc/src/mongoc/mongoc-sspi-private.h b/src/libmongoc/src/mongoc/mongoc-sspi-private.h
index 60996fc..3745eff 100644
--- a/src/libmongoc/src/mongoc/mongoc-sspi-private.h
+++ b/src/libmongoc/src/mongoc/mongoc-sspi-private.h
@@ -28,7 +28,7 @@ BSON_BEGIN_DECLS
 
 #define SECURITY_WIN32 1 /* Required for SSPI */
 
-#include <Windows.h>
+#include <windows.h>
 #include <limits.h>
 #include <sspi.h>
 #include <string.h>
