diff --git a/CMakeLists.txt b/CMakeLists.txt
index d027a19..aac7f3d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -346,10 +346,7 @@ if (USE_SYSTEM_LIBBSON)
    # The input variable BSON_ROOT_DIR is respected for backwards compatibility,
    # but you should use the standard CMAKE_PREFIX_PATH instead.
    message (STATUS "Searching for libbson CMake packages")
-   find_package (bson-1.0
-      "${PROJECT_VERSION}"
-      HINTS
-      ${BSON_ROOT_DIR})
+   find_package(bson-1.0 CONFIG REQUIRED)
 
    if (NOT bson-1.0_FOUND)
       message (FATAL_ERROR "System libbson not found")
diff --git a/src/libmongoc/src/mongoc-config.cmake b/src/libmongoc/src/mongoc-config.cmake
index cdbb756..a873af0 100644
--- a/src/libmongoc/src/mongoc-config.cmake
+++ b/src/libmongoc/src/mongoc-config.cmake
@@ -1,5 +1,5 @@
 include(CMakeFindDependencyMacro)
-find_dependency(bson-1.0 @libmongoc_VERSION@)
+find_dependency(bson-1.0)
 if("@ENABLE_SNAPPY@")
     find_dependency(Snappy CONFIG)
 endif()
