diff --git a/src/libmongoc/CMakeLists.txt b/src/libmongoc/CMakeLists.txt
index 1b32e6c..2e1f854 100644
--- a/src/libmongoc/CMakeLists.txt
+++ b/src/libmongoc/CMakeLists.txt
@@ -1207,6 +1207,22 @@ set (VERSION "${MONGOC_VERSION}")
 set (prefix "${CMAKE_INSTALL_PREFIX}")
 set (libdir "\${prefix}/${CMAKE_INSTALL_LIBDIR}")
 
+set(MONGOC_REQUIRES_PRIVATE zlib)
+set(ZLIB_LIBRARIES "")
+if(ENABLE_SSL STREQUAL "OPENSSL")
+   string(APPEND MONGOC_REQUIRES_PRIVATE " openssl")
+   set(SSL_LIBRARIES "")
+endif()
+if(ENABLE_ICU)
+   string(APPEND MONGOC_REQUIRES_PRIVATE " icu-uc")
+   set(ICU_LIBRARIES "")
+endif()
+if(ENABLE_SNAPPY)
+   set(SNAPPY_LIBRARIES "-lsnappy")
+   if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+      string(APPEND SNAPPY_LIBRARIES "d")
+   endif()
+endif()
 foreach (
       FLAG
       ${SASL_LIBRARIES} ${SSL_LIBRARIES} ${SHM_LIBRARIES} ${RESOLV_LIBRARIES}
diff --git a/src/libmongoc/src/libmongoc-static-1.0.pc.in b/src/libmongoc/src/libmongoc-static-1.0.pc.in
index c23bb62..7b38f0a 100644
--- a/src/libmongoc/src/libmongoc-static-1.0.pc.in
+++ b/src/libmongoc/src/libmongoc-static-1.0.pc.in
@@ -9,3 +9,4 @@ Version: @VERSION@
 Requires: libbson-static-1.0
 Libs: -L${libdir} -lmongoc-static-1.0 @MONGOC_LIBRARIES@
 Cflags: -I${includedir}/libmongoc -DMONGOC_STATIC
+Requires.private: @MONGOC_REQUIRES_PRIVATE@
