diff --git a/CMakeLists.txt b/CMakeLists.txt
index 45550a8..1f98148 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,7 +42,7 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH}
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
 find_package(Boost REQUIRED)
-find_package(Eigen3 3.3.7 REQUIRED) # v3.3.7 adds support for cmake targets
+find_package(Eigen3 CONFIG REQUIRED)
 
 add_library(Interpolate INTERFACE)
 add_library(libInterpolate::Interpolate ALIAS Interpolate)
@@ -117,7 +117,7 @@ file(
   WRITE ${CMAKE_CURRENT_BINARY_DIR}/libInterpolateConfig.cmake
   "include(CMakeFindDependencyMacro)
 find_dependency(Boost)
-find_dependency(Eigen3 3.3.7 REQUIRED)
+find_dependency(Eigen3 CONFIG REQUIRED)
 include(\${CMAKE_CURRENT_LIST_DIR}/libInterpolateTargets.cmake)
 ")
 write_basic_package_version_file(
 