index 3aea49a..b21d330 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,9 +17,18 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH}
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)

 if("${libInterpolate_VERSION}" STREQUAL "UNKNOWN")
-  message(STATUS "Detecting version number using git...")
-  include(macro-git_version)
-  git_version(libInterpolate)
+  find_package(Git QUIET)
+  if(Git_FOUND AND EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/.git")
+    message(STATUS "Detecting version number using git...")
+    include(macro-git_version)
+    git_version(libInterpolate)
+  else()
+    message(STATUS "Git not found or not a git repository. Setting default version.")
+    set(libInterpolate_VERSION "0.0.0" CACHE STRING "Library version number" FORCE)
+    set(libInterpolate_VERSION_MAJOR "0" CACHE INTERNAL "Library major version number")
+    set(libInterpolate_VERSION_MINOR "0" CACHE INTERNAL "Library minor version number")
+    set(libInterpolate_VERSION_PATCH "0" CACHE INTERNAL "Library patch version number")
+  endif()
   message(STATUS "Version: ${libInterpolate_VERSION}")
   message(STATUS "Version (Full): ${libInterpolate_VERSION_FULL}")
 endif()
 