diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4a3d31e..82611ef 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -58,12 +58,15 @@ if(nanopb_BUILD_GENERATOR)
         install(
             FILES ${PROJECT_BINARY_DIR}/${generator_proto_py_file}
                   ${generator_proto_file}
-            DESTINATION ${PYTHON_INSTDIR}/proto/
+            DESTINATION ${CMAKE_INSTALL_BINDIR}/proto/
         )
     endforeach()
 
     install( FILES generator/proto/_utils.py
-             DESTINATION ${PYTHON_INSTDIR}/proto/ )
+             DESTINATION ${CMAKE_INSTALL_BINDIR}/proto/ )
+
+    install( FILES generator/proto/__init__.py
+             DESTINATION ${CMAKE_INSTALL_BINDIR}/proto/ )
 endif()
 
 if(WIN32)
