diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2d4641902..742722d8f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -151,7 +151,7 @@ if(BUILD_LAYERMGR)
         add_subdirectory(vkconfig_cmd)
         add_subdirectory(vkconfig_gui)
 
-        if(WIN32 AND (QT_TARGET_TYPE STREQUAL STATIC_LIBRARY))
+        if(WIN32 AND (QT_TARGET_TYPE STREQUAL STATIC_LIBRARY) AND (NOT DEFINED VCPKG_CRT_LINKAGE OR VCPKG_CRT_LINKAGE STREQUAL "static"))
             set_property(TARGET vkconfig-gui vkconfig-cmd vkconfig-core PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
             message(STATUS "INFO: vkconfig will link against static runtime")
         endif()
