diff --git a/CMakeLists.txt b/CMakeLists.txt
index ed97ca0..e9751cb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -278,7 +278,6 @@ IF(MESHLIB_PYTHON_SUPPORT)
 ENDIF()
 
 IF(MSVC)
-  add_subdirectory(${PROJECT_SOURCE_DIR}/OpenCTM)
 ENDIF()
 
 IF(NOT MR_EMSCRIPTEN)
diff --git a/thirdparty/CMakeLists.txt b/thirdparty/CMakeLists.txt
index e4734cf..4aca508 100644
--- a/thirdparty/CMakeLists.txt
+++ b/thirdparty/CMakeLists.txt
@@ -83,7 +83,6 @@ IF(NOT APPLE AND NOT MR_EMSCRIPTEN)
   ENDIF()
 ENDIF()
 
-add_subdirectory(./OpenCTM-git ./OpenCTM)
 
 option(PHMAP_INSTALL "" ON)
 add_subdirectory(./parallel-hashmap)
