diff --git a/src/stc/scintilla/include/Scintilla.h b/src/stc/scintilla/include/Scintilla.h
index 8c57520e1..54ffb27d2 100644
--- a/src/stc/scintilla/include/Scintilla.h	
+++ b/src/stc/scintilla/include/Scintilla.h
@@ -27,7 +27,7 @@ int Scintilla_LinkLexers(void);
 #endif
 
 // Include header that defines basic numeric types.
-#if defined(_MSC_VER)
+#if defined(_MSC_VER) && !defined(__clang__)
 // Older releases of MSVC did not have stdint.h.
 #include <stddef.h>
 #elif defined( __VMS )
