diff --git a/build/cmake/init.cmake b/build/cmake/init.cmake
--- a/build/cmake/init.cmake
+++ b/build/cmake/init.cmake
@@ -201,11 +201,11 @@ endif()
 wx_get_install_dir(include)
 if(WIN32_MSVC_NAMING)
     if(wxBUILD_SHARED)
-        set(lib_suffix "_dll")
+        # set(lib_suffix "_dll")
     else()
-        set(lib_suffix "_lib")
+        # set(lib_suffix "_lib")
     endif()
-    set(wxPLATFORM_LIB_DIR "${wxCOMPILER_PREFIX}${wxARCH_SUFFIX}${lib_suffix}")
+    # set(wxPLATFORM_LIB_DIR "${wxCOMPILER_PREFIX}${wxARCH_SUFFIX}${lib_suffix}")
     set(wxINSTALL_INCLUDE_DIR "${include_dir}")
 else()
     wx_get_flavour(lib_flavour "-")
