diff --git a/build/cmake/lib/nanosvg.cmake b/build/cmake/lib/nanosvg.cmake
index 401bf48..b9e4b57 100644
--- a/build/cmake/lib/nanosvg.cmake
+++ b/build/cmake/lib/nanosvg.cmake
@@ -16,9 +16,9 @@ elseif(wxUSE_NANOSVG)
     set(NANOSVG_INCLUDE_DIRS )
     set(wxUSE_NANOSVG_EXTERNAL_ENABLE_IMPL TRUE)
 
-    find_package(NanoSVG REQUIRED)
+    find_package(unofficial-nanosvg CONFIG REQUIRED)
 
-    foreach(TARGETNAME NanoSVG::nanosvg NanoSVG::nanosvgrast unofficial::nanosvg)
+    foreach(TARGETNAME unofficial::nanosvg::nanosvg)
         if(NOT TARGET ${TARGETNAME})
             continue()
         endif()
diff --git a/build/cmake/options.cmake b/build/cmake/options.cmake
index 49c536a..5630526 100644
--- a/build/cmake/options.cmake
+++ b/build/cmake/options.cmake
@@ -114,7 +114,7 @@ wx_add_thirdparty_library(wxUSE_EXPAT EXPAT "use expat for XML parsing" DEFAULT_
 wx_add_thirdparty_library(wxUSE_LIBJPEG JPEG "use libjpeg (JPEG file format)")
 wx_add_thirdparty_library(wxUSE_LIBPNG PNG "use libpng (PNG image format)")
 wx_add_thirdparty_library(wxUSE_LIBTIFF TIFF "use libtiff (TIFF file format)")
-wx_add_thirdparty_library(wxUSE_NANOSVG NanoSVG "use NanoSVG for rasterizing SVG")
+wx_add_thirdparty_library(wxUSE_NANOSVG unofficial-nanosvg "use NanoSVG for rasterizing SVG")
 
 wx_option(wxUSE_LIBLZMA "use LZMA compression" OFF)
 set(wxTHIRD_PARTY_LIBRARIES ${wxTHIRD_PARTY_LIBRARIES} wxUSE_LIBLZMA "use liblzma for LZMA compression")
diff --git a/build/cmake/wxWidgetsConfig.cmake.in b/build/cmake/wxWidgetsConfig.cmake.in
index 6ef5a6e..248a701 100644
--- a/build/cmake/wxWidgetsConfig.cmake.in
+++ b/build/cmake/wxWidgetsConfig.cmake.in
@@ -1,5 +1,8 @@
 @PACKAGE_INIT@
 
+include(CMakeFindDependencyMacro)
+find_dependency(unofficial-nanosvg CONFIG)
+
 # determine target from compiler, platform and library type
 if(WIN32 AND NOT CYGWIN AND NOT MSYS)
     if(${CMAKE_CXX_COMPILER_ID} STREQUAL MSVC)
