diff --git a/build/cmake/init.cmake b/build/cmake/init.cmake
index 0726ea2..2f8652d 100644
--- a/build/cmake/init.cmake
+++ b/build/cmake/init.cmake
@@ -684,7 +684,9 @@ if(wxUSE_GUI)
     endif()
 
     if(wxUSE_SOUND AND wxUSE_LIBSDL AND UNIX AND NOT APPLE)
-        find_package(SDL2)
+        find_package(SDL2 CONFIG REQUIRED)
+        set(SDL2_INCLUDE_DIR "" CACHE INTERNAL "")
+        set(SDL2_LIBRARY SDL2::SDL2 CACHE INTERNAL "")
         if(NOT SDL2_FOUND)
             find_package(SDL)
             mark_as_advanced(SDL_INCLUDE_DIR SDLMAIN_LIBRARY)
diff --git a/build/cmake/wxWidgetsConfig.cmake.in b/build/cmake/wxWidgetsConfig.cmake.in
index 8e4e2be..9a3cad2 100644
--- a/build/cmake/wxWidgetsConfig.cmake.in
+++ b/build/cmake/wxWidgetsConfig.cmake.in
@@ -2,6 +2,9 @@
 
 include(CMakeFindDependencyMacro)
 find_dependency(NanoSVG CONFIG)
+if("@wxUSE_LIBSDL@")
+    find_dependency(SDL2 CONFIG)
+endif()
 
 cmake_policy(PUSH)
 # Set policies to prevent warnings
