From 65c7fc253b40de34df46edfb42d0f32749a5a0aa Mon Sep 17 00:00:00 2001
From: Russell Greene <russell@shotover.com>
Date: Tue, 18 Oct 2022 09:07:07 -0500
Subject: [PATCH 5/7] fix-error-c3861

---
 XML/include/Poco/XML/ParserEngine.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/XML/include/Poco/XML/ParserEngine.h b/XML/include/Poco/XML/ParserEngine.h
index e0c8455..363654c 100644
--- a/XML/include/Poco/XML/ParserEngine.h
+++ b/XML/include/Poco/XML/ParserEngine.h
@@ -19,6 +19,7 @@
 
 #include "Poco/XML/XML.h"
 #if defined(POCO_UNBUNDLED)
+#include <expat_config.h>
 #include <expat.h>
 #else
 #include "Poco/XML/expat.h"
