diff --git a/Data/CMakeLists.txt b/Data/CMakeLists.txt
index 72da6df..12a2870 100644
--- a/Data/CMakeLists.txt
+++ b/Data/CMakeLists.txt
@@ -49,6 +49,20 @@ target_include_directories(Data
 POCO_INSTALL(Data)
 POCO_GENERATE_PACKAGE(Data)
 
+if (NOT POCO_DATA_NO_SQL_PARSER)
+    file(GLOB HEADERS "${CMAKE_CURRENT_LIST_DIR}/src/sql-parser/src/*.h")
+    install(FILES ${HEADERS} DESTINATION "include/Poco/Data/sql-parser/src")
+    
+    file(GLOB HEADERS "${CMAKE_CURRENT_LIST_DIR}/src/sql-parser/src/parser/*.h")
+    install(FILES ${HEADERS} DESTINATION "include/Poco/Data/sql-parser/src/parser")
+    
+    file(GLOB HEADERS "${CMAKE_CURRENT_LIST_DIR}/src/sql-parser/src/sql/*.h")
+    install(FILES ${HEADERS} DESTINATION "include/Poco/Data/sql-parser/src/sql")
+    
+    file(GLOB HEADERS "${CMAKE_CURRENT_LIST_DIR}/src/sql-parser/src/util/*.h")
+    install(FILES ${HEADERS} DESTINATION "include/Poco/Data/sql-parser/src/util")
+endif()
+
 if(ENABLE_TESTS)
 	add_subdirectory(testsuite)
 endif()
