From e2576ff5ab9f4a69a7d560cd378a377022501a9a Mon Sep 17 00:00:00 2001
From: Aleksandar Fabijanic <aleks-f@users.noreply.github.com>
Date: Sat, 7 Dec 2024 17:35:39 -0500
Subject: [PATCH] fix(cmake): remove libatomic dependency (#4811)

---
 Foundation/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Foundation/CMakeLists.txt b/Foundation/CMakeLists.txt
index 5898d22f5c..e985cc39a7 100644
--- a/Foundation/CMakeLists.txt
+++ b/Foundation/CMakeLists.txt
@@ -159,7 +159,7 @@ else()
 				target_link_libraries(Foundation PUBLIC ${CMAKE_DL_LIBS} rt Threads::Threads)
 			else()
 				target_compile_definitions(Foundation PUBLIC _XOPEN_SOURCE=500 POCO_HAVE_FD_EPOLL)
-				target_link_libraries(Foundation PUBLIC pthread atomic ${CMAKE_DL_LIBS} rt)
+				target_link_libraries(Foundation PUBLIC pthread ${CMAKE_DL_LIBS} rt)
 			endif()
 		endif(APPLE)
 	endif(UNIX AND NOT ANDROID)
