diff --git a/cmake/FindPCRE.cmake b/cmake/FindPCRE.cmake
index c5bb73c..bfeaa24 100644
--- a/cmake/FindPCRE.cmake
+++ b/cmake/FindPCRE.cmake
@@ -65,9 +65,9 @@ find_path(PCRE_INCLUDE_DIR
         ${PC_PCRE_INCLUDE_DIRS}
   DOC "Specify the include directory containing pcre.h"
 )
-
+SET(PCRE_NAMES pcred pcre)
 find_library(PCRE_LIBRARY
-  NAMES pcre
+  NAMES ${PCRE_NAMES}
   HINTS
         ${PCRE_ROOT_DIR}/lib
         ${PCRE_ROOT_LIBRARY_DIRS}
