diff --git a/cmake/FindPCRE.cmake b/cmake/FindPCRE.cmake
index c5bb73c..ab5ada7 100644
--- a/cmake/FindPCRE.cmake
+++ b/cmake/FindPCRE.cmake
@@ -66,15 +66,15 @@ find_path(PCRE_INCLUDE_DIR
   DOC "Specify the include directory containing pcre.h"
 )
 
-find_library(PCRE_LIBRARY
-  NAMES pcre
-  HINTS
-        ${PCRE_ROOT_DIR}/lib
-        ${PCRE_ROOT_LIBRARY_DIRS}
-  PATHS
-        ${PC_PCRE_LIBRARY_DIRS}
-  DOC "Specify the lib directory containing pcre"
-)
+if(WIN32)
+  include(SelectLibraryConfigurations)
+  find_library(PCRE_LIBRARY_RELEASE NAMES pcre)
+  find_library(PCRE_LIBRARY_DEBUG NAMES pcred)
+  select_library_configurations(PCRE)
+else()
+  SET(PCRE_NAMES pcred pcre)
+  find_library(PCRE_LIBRARY NAMES ${PCRE_NAMES})
+endif()
 
 set(PCRE_VERSION ${PC_PCRE_VERSION})
 
