diff --git a/Foundation/CMakeLists.txt b/Foundation/CMakeLists.txt
index 0b8f7bc..c362890 100644
--- a/Foundation/CMakeLists.txt
+++ b/Foundation/CMakeLists.txt
@@ -53,6 +53,21 @@ if (POCO_UNBUNDLED)
 	include_directories(${ZLIB_INCLUDE_DIRS})
 
 	add_definitions(-DPOCO_UNBUNDLED)
+    add_definitions(
+        -D_pcre_utf8_table1=_poco_pcre_utf8_table1
+        -D_pcre_utf8_table1_size=_poco_pcre_utf8_table1_size
+        -D_pcre_utf8_table2=_poco_pcre_utf8_table2
+        -D_pcre_utf8_table3=_poco_pcre_utf8_table3
+        -D_pcre_utf8_table4=_poco_pcre_utf8_table4
+        -D_pcre_utt_names=_poco_pcre_utt_names
+        -D_pcre_utt=_poco_pcre_utt
+        -D_pcre_utt_size=_poco_pcre_utt_size
+        -D_pcre_OP_lengths=_poco_pcre_OP_lengths
+        -D_pcre_hspace_list=_poco_pcre_hspace_list
+        -D_pcre_vspace_list=_poco_pcre_vspace_list
+        -D_pcre_ucp_gentype=_poco_pcre_ucp_gentype
+        -D_pcre_ucp_gbtable=_poco_pcre_ucp_gbtable
+    )
 
 else()
 	# pcre
