diff --git a/src/Makefile.shlib b/src/Makefile.shlib
index fa81f6ffdd..7d2edb3ef4 100644
--- a/src/Makefile.shlib
+++ b/src/Makefile.shlib
@@ -231,10 +231,14 @@ endif
 
 .PHONY: all-lib all-static-lib all-shared-lib
 
+ifndef LIBPQ_LIBRARY_TYPE
 all-lib: all-shared-lib
 ifdef soname
 # no static library when building a dynamically loadable module
 all-lib: all-static-lib
+endif
+else
+all-lib: all-$(LIBPQ_LIBRARY_TYPE)-lib
 all-lib: lib$(NAME).pc
 endif
 
@@ -362,9 +366,13 @@ endif # PORTNAME == cygwin || PORTNAME == win32
 ##
 
 .PHONY: install-lib install-lib-static install-lib-shared installdirs-lib
+ifndef LIBPQ_LIBRARY_TYPE
 install-lib: install-lib-shared
 ifdef soname
 install-lib: install-lib-static
+endif
+else
+install-lib: install-lib-$(LIBPQ_LIBRARY_TYPE)
 install-lib: install-lib-pc
 endif
 
