diff --git a/CMakeLists.txt b/CMakeLists.txt
index ea113e1..149e437 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,7 +2,7 @@
 # Set minimum version of CMake. Since command 'project' use
 # VERSION sub-option we need at least 3.0.
 # Note: If you use 2.6 or 2.4, God kills a kitten. Seriously.
-cmake_minimum_required(VERSION 3.2 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
 
 ####
 # Set variables:
