diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index 11fa675..34cc9ed 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -20,6 +20,11 @@ set(${PROJECT_NAME}_INSTALL_SAMPLES OFF CACHE BOOL
 set(${PROJECT_NAME}_ONLY_LIBRARY OFF CACHE BOOL
     "Set to ON to only build markdown library (default is OFF)")

+# MSVC deprecated warnings (C4996,strdup, ...)
+if(MSVC)
+    add_definitions(-D_CRT_SECURE_NO_DEPRECATE -D_CRT_NONSTDC_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS)
+endif()
+
 # Check headers
 include(CheckIncludeFile)
 check_include_file(libgen.h HAVE_LIBGEN_H)
