diff --git a/common/include/guiddef.h b/common/include/jxrguiddef.h
similarity index 100%
rename from common/include/guiddef.h
rename to common/include/jxrguiddef.h
diff --git a/jxrgluelib/JXRGlue.h b/jxrgluelib/JXRGlue.h
index d0b219c..c3e5d2b 100644
--- a/jxrgluelib/JXRGlue.h
+++ b/jxrgluelib/JXRGlue.h
@@ -32,7 +32,11 @@ extern "C" {
 #endif
 
 #include <JXRMeta.h>
+#ifdef _WIN32
 #include <guiddef.h>
+#else
+#include <jxrguiddef.h>
+#endif
 
 //================================================================
 #define WMP_SDK_VERSION 0x0101
