diff --git a/Findlibzmq.cmake b/Findlibzmq.cmake
index 0a563dd8..6c543598 100644
--- a/Findlibzmq.cmake
+++ b/Findlibzmq.cmake
@@ -23,11 +23,61 @@ find_path (
     HINTS ${PC_LIBZMQ_INCLUDE_HINTS}
 )
 
-find_library (
-    LIBZMQ_LIBRARIES
-    NAMES zmq
-    HINTS ${PC_LIBZMQ_LIBRARY_HINTS}
-)
+if (NOT MSVC)
+    find_library (
+        LIBZMQ_LIBRARIES
+        NAMES zmq
+        HINTS ${PC_LIBZMQ_LIBRARY_HINTS}
+    )
+else ()
+    # libzmq dll/lib built with MSVC is named using the Boost convention.
+    # https://github.com/zeromq/czmq/issues/577
+    # https://bitbucket.org/ignitionrobotics/ign-transport/pull-requests/19/detec-zmq-under-windows-using-a-findzmq/diff
+    # https://github.com/zeromq/libzmq/blob/master/CMakeLists.txt
+    if (MSVC_IDE)
+        set(MSVC_TOOLSET "-${CMAKE_VS_PLATFORM_TOOLSET}")
+    else ()
+        set(MSVC_TOOLSET "")
+    endif ()
+
+    set(_zmq_versions
+        "4_3_1"
+        "4_3_0")
+
+    set(_zmq_debug_names)
+    set(_zmq_release_names)
+
+    foreach(ver ${_zmq_versions})
+        list(APPEND _zmq_debug_names
+            "libzmq${MSVC_TOOLSET}-mt-gd-${ver}" # Debug, BUILD_SHARED
+            "libzmq${MSVC_TOOLSET}-mt-sgd-${ver}" # Debug, BUILD_STATIC
+        )
+
+        list(APPEND _zmq_release_names
+            "libzmq${MSVC_TOOLSET}-mt-${ver}" # Release|RelWithDebInfo|MinSizeRel, BUILD_SHARED
+            "libzmq${MSVC_TOOLSET}-mt-s-${ver}" # Release|RelWithDebInfo|MinSizeRel, BUILD_STATIC
+        ) 
+    endforeach()
+
+    find_library (ZeroMQ_LIBRARY_DEBUG
+        NAMES zmq libzmq ${_zmq_debug_names}
+    )
+
+    find_library (ZeroMQ_LIBRARY_RELEASE
+        NAMES zmq libzmq ${_zmq_release_names}
+    )
+
+    if (ZeroMQ_LIBRARY_RELEASE AND ZeroMQ_LIBRARY_DEBUG)
+        set(LIBZMQ_LIBRARIES
+            debug ${ZeroMQ_LIBRARY_DEBUG}
+            optimized ${ZeroMQ_LIBRARY_RELEASE}
+        )
+    elseif (ZeroMQ_LIBRARY_RELEASE)
+        set(LIBZMQ_LIBRARIES ${ZeroMQ_LIBRARY_RELEASE})
+    elseif (ZeroMQ_LIBRARY_DEBUG)
+        set(LIBZMQ_LIBRARIES ${ZeroMQ_LIBRARY_DEBUG})
+    endif ()
+endif ()
 
 include(FindPackageHandleStandardArgs)
 
