diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2d2b305a..8100d0cd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -138,7 +138,8 @@ ENDIF (LIBZMQ_FOUND)
 # UUID dependency
 ########################################################################
 find_package(uuid)
-IF (UUID_FOUND)
+option(CZMQ_WITH_UUID "Build czmq with uuid" ${UUID_FOUND})
+IF (CZMQ_WITH_UUID AND UUID_FOUND)
     include_directories(${UUID_INCLUDE_DIRS})
     list(APPEND MORE_LIBRARIES ${UUID_LIBRARIES})
   IF (PC_UUID_FOUND)
@@ -149,13 +150,14 @@ IF (UUID_FOUND)
   ENDIF (PC_UUID_FOUND)
     add_definitions(-DHAVE_UUID)
     list(APPEND OPTIONAL_LIBRARIES ${UUID_LIBRARIES})
-ENDIF (UUID_FOUND)
+ENDIF (CZMQ_WITH_UUID AND UUID_FOUND)
 
 ########################################################################
 # SYSTEMD dependency
 ########################################################################
 find_package(systemd)
-IF (SYSTEMD_FOUND)
+option(CZMQ_WITH_SYSTEMD "Build czmq with systemd" ${SYSTEMD_FOUND})
+IF (CZMQ_WITH_SYSTEMD AND SYSTEMD_FOUND)
     include_directories(${SYSTEMD_INCLUDE_DIRS})
     list(APPEND MORE_LIBRARIES ${SYSTEMD_LIBRARIES})
   IF (PC_SYSTEMD_FOUND)
@@ -166,13 +168,14 @@ IF (SYSTEMD_FOUND)
   ENDIF (PC_SYSTEMD_FOUND)
     add_definitions(-DHAVE_LIBSYSTEMD)
     list(APPEND OPTIONAL_LIBRARIES ${SYSTEMD_LIBRARIES})
-ENDIF (SYSTEMD_FOUND)
+ENDIF (CZMQ_WITH_SYSTEMD AND SYSTEMD_FOUND)
 
 ########################################################################
 # LZ4 dependency
 ########################################################################
 find_package(lz4)
-IF (LZ4_FOUND)
+option(CZMQ_WITH_LZ4 "Build czmq with lz4" ${LZ4_FOUND})
+IF (CZMQ_WITH_LZ4 AND LZ4_FOUND)
     include_directories(${LZ4_INCLUDE_DIRS})
     list(APPEND MORE_LIBRARIES ${LZ4_LIBRARIES})
   IF (PC_LZ4_FOUND)
@@ -183,13 +186,14 @@ IF (LZ4_FOUND)
   ENDIF (PC_LZ4_FOUND)
     add_definitions(-DHAVE_LIBLZ4)
     list(APPEND OPTIONAL_LIBRARIES ${LZ4_LIBRARIES})
-ENDIF (LZ4_FOUND)
+ENDIF (CZMQ_WITH_LZ4 AND LZ4_FOUND)
 
 ########################################################################
 # LIBCURL dependency
 ########################################################################
 find_package(libcurl)
-IF (LIBCURL_FOUND)
+option(CZMQ_WITH_LIBCURL "Build czmq with libcurl" ${LIBCURL_FOUND})
+IF (CZMQ_WITH_LIBCURL AND LIBCURL_FOUND)
     include_directories(${LIBCURL_INCLUDE_DIRS})
     list(APPEND MORE_LIBRARIES ${LIBCURL_LIBRARIES})
   IF (PC_LIBCURL_FOUND)
@@ -200,13 +204,14 @@ IF (LIBCURL_FOUND)
   ENDIF (PC_LIBCURL_FOUND)
     add_definitions(-DHAVE_LIBCURL)
     list(APPEND OPTIONAL_LIBRARIES ${LIBCURL_LIBRARIES})
-ENDIF (LIBCURL_FOUND)
+ENDIF (CZMQ_WITH_LIBCURL AND LIBCURL_FOUND)
 
 ########################################################################
 # LIBMICROHTTPD dependency
 ########################################################################
 find_package(libmicrohttpd)
-IF (LIBMICROHTTPD_FOUND)
+option(CZMQ_WITH_LIBMICROHTTPD "Build czmq with libmicrohttpd" ${LIBMICROHTTPD_FOUND})
+IF (CZMQ_WITH_LIBMICROHTTPD AND LIBMICROHTTPD_FOUND)
     include_directories(${LIBMICROHTTPD_INCLUDE_DIRS})
     list(APPEND MORE_LIBRARIES ${LIBMICROHTTPD_LIBRARIES})
   IF (PC_LIBMICROHTTPD_FOUND)
@@ -217,7 +222,7 @@ IF (LIBMICROHTTPD_FOUND)
   ENDIF (PC_LIBMICROHTTPD_FOUND)
     add_definitions(-DHAVE_LIBMICROHTTPD)
     list(APPEND OPTIONAL_LIBRARIES ${LIBMICROHTTPD_LIBRARIES})
-ENDIF (LIBMICROHTTPD_FOUND)
+ENDIF (CZMQ_WITH_LIBMICROHTTPD AND LIBMICROHTTPD_FOUND)
 
 ########################################################################
 # version
