diff --git a/src/unix/core.c b/src/unix/core.c
index def5b8f..e44298e 100644
--- a/src/unix/core.c
+++ b/src/unix/core.c
@@ -1636,6 +1636,7 @@ unsigned int uv_available_parallelism(void) {
    * glibc it's... complicated... so for consistency try sched_getaffinity()
    * before falling back to sysconf(_SC_NPROCESSORS_ONLN).
    */
+#ifdef CPU_COUNT   
   if (0 == sched_getaffinity(0, sizeof(set), &set))
     rc = CPU_COUNT(&set);
   else
@@ -1645,6 +1646,8 @@ unsigned int uv_available_parallelism(void) {
     rc = 1;
 
   return (unsigned) rc;
+#endif
+  
 #elif defined(__MVS__)
   return 1;  /* TODO(bnoordhuis) Read from CSD_NUMBER_ONLINE_CPUS? */
 #else  /* __linux__ */
