diff --git a/cairomm/meson.build b/cairomm/meson.build
index 8580562..8ec2c51 100644
--- a/cairomm/meson.build
+++ b/cairomm/meson.build
@@ -73,7 +73,7 @@ cairomm_private_h = [
 # docs/reference/meson.build needs this.
 source_h_files = cairomm_public_h
 
-install_headers(cairomm_public_h, subdir: cairomm_pcname / 'cairomm')
+install_headers(cairomm_public_h, subdir: 'cairomm')
 
 # Make sure we are exporting the symbols from the DLL
 cairomm_cpp_args = ['-DCAIROMM_BUILD=1']
diff --git a/data/cairomm.pc.in b/data/cairomm.pc.in
index 1ee5b42..c7e33b5 100644
--- a/data/cairomm.pc.in
+++ b/data/cairomm.pc.in
@@ -16,4 +16,4 @@ Version: @PACKAGE_VERSION@
 URL: https://www.cairographics.org/cairomm/
 Requires: @CAIROMM_MODULES@
 Libs: -L${libdir} -lcairomm@MSVC_TOOLSET_VER@-@CAIROMM_API_VERSION@ @CAIROMM_EXTRA_LIBS@
-Cflags: -I${includedir}/@CAIROMM_MODULE_NAME@ -I${libdir}/@CAIROMM_MODULE_NAME@/include
+Cflags: -I${includedir}
diff --git a/meson.build b/meson.build
index 9f10967..6949360 100644
--- a/meson.build
+++ b/meson.build
@@ -299,7 +299,7 @@ mm_conf_data.set('CAIROMM_MICRO_VERSION', cairomm_micro_version)
 mm_conf_data.set('VERSION', meson.project_version()) # for MSVC_NMake/cairomm/cairomm.rc
 
 cairommconfig_h_meson = files('cairommconfig.h.meson')
-install_includeconfigdir = install_libdir / cairomm_pcname / 'include'
+install_includeconfigdir = install_includedir
 configure_file(
   input: cairommconfig_h_meson,
   output: 'cairommconfig.h',
@@ -387,7 +387,7 @@ summary = [
   'Directories:',
   '                  prefix: @0@'.format(install_prefix),
   '              includedir: @0@'.format(install_prefix / install_includedir),
-  '       includecairommdir: @0@'.format(install_prefix / install_includedir / cairomm_pcname),
+  '       includecairommdir: @0@'.format(install_prefix / install_includedir),
   '                  libdir: @0@'.format(install_prefix / install_libdir),
   '        includeconfigdir: @0@'.format(install_prefix / install_includeconfigdir),
   '            pkgconfigdir: @0@'.format(install_prefix / install_pkgconfigdir),
