diff --git a/SMP/libgnutls.vcxproj b/SMP/libgnutls.vcxproj
index 1de15721..56ce1ca4 100644
--- a/SMP/libgnutls.vcxproj
+++ b/SMP/libgnutls.vcxproj
@@ -1239,6 +1239,7 @@
       <MinimalRebuild>false</MinimalRebuild>
       <TreatSpecificWarningsAsErrors>4113;%(TreatSpecificWarningsAsErrors)</TreatSpecificWarningsAsErrors>
       <DisableSpecificWarnings>4267;4244;4018;4090;4047;4024;4311;4116;4146;4312;4028;4334;4101;4013;4020;%(DisableSpecificWarnings)</DisableSpecificWarnings>
+      <RuntimeLibrary>$(RuntimeLibrary)</RuntimeLibrary>
     </ClCompile>
     <Lib>
       <OutputFile>$(OutDir)\lib\x86\$(TargetName)$(TargetExt)</OutputFile>
@@ -1305,6 +1306,7 @@ del /f /q $(OutDir)\licenses\gnutls.txt
       <CompileAsWinRT>false</CompileAsWinRT>
       <TreatSpecificWarningsAsErrors>4113;%(TreatSpecificWarningsAsErrors)</TreatSpecificWarningsAsErrors>
       <DisableSpecificWarnings>4267;4244;4018;4090;4047;4024;4311;4116;4146;4312;4028;4334;4101;4013;4020;%(DisableSpecificWarnings)</DisableSpecificWarnings>
+      <RuntimeLibrary>$(RuntimeLibrary)</RuntimeLibrary>
       <SDLCheck>false</SDLCheck>
     </ClCompile>
     <Lib>
@@ -1370,6 +1372,7 @@ del /f /q $(OutDir)\licenses\gnutls.txt
       <MinimalRebuild>false</MinimalRebuild>
       <TreatSpecificWarningsAsErrors>4113;%(TreatSpecificWarningsAsErrors)</TreatSpecificWarningsAsErrors>
       <DisableSpecificWarnings>4267;4244;4018;4090;4047;4024;4311;4116;4146;4312;4028;4334;4101;4013;4020;%(DisableSpecificWarnings)</DisableSpecificWarnings>
+      <RuntimeLibrary>$(RuntimeLibrary)</RuntimeLibrary>
     </ClCompile>
     <Lib>
       <OutputFile>$(OutDir)\lib\x64\$(TargetName)$(TargetExt)</OutputFile>
@@ -1436,6 +1439,7 @@ del /f /q $(OutDir)\licenses\gnutls.txt
       <CompileAsWinRT>false</CompileAsWinRT>
       <TreatSpecificWarningsAsErrors>4113;%(TreatSpecificWarningsAsErrors)</TreatSpecificWarningsAsErrors>
       <DisableSpecificWarnings>4267;4244;4018;4090;4047;4024;4311;4116;4146;4312;4028;4334;4101;4013;4020;%(DisableSpecificWarnings)</DisableSpecificWarnings>
+      <RuntimeLibrary>$(RuntimeLibrary)</RuntimeLibrary>
       <SDLCheck>false</SDLCheck>
     </ClCompile>
     <Lib>
@@ -1502,6 +1506,7 @@ del /f /q $(OutDir)\licenses\gnutls.txt
       <MinimalRebuild>false</MinimalRebuild>
       <TreatSpecificWarningsAsErrors>4113;%(TreatSpecificWarningsAsErrors)</TreatSpecificWarningsAsErrors>
       <DisableSpecificWarnings>4267;4244;4018;4090;4047;4024;4311;4116;4146;4312;4028;4334;4101;4013;4020;%(DisableSpecificWarnings)</DisableSpecificWarnings>
+      <RuntimeLibrary>$(RuntimeLibrary)</RuntimeLibrary>
     </ClCompile>
     <Link>
       <OutputFile>$(OutDir)\bin\x86\$(TargetName)$(TargetExt)</OutputFile>
@@ -1575,6 +1580,7 @@ del /f /q $(OutDir)\licenses\gnutls.txt
       <CompileAsWinRT>false</CompileAsWinRT>
       <TreatSpecificWarningsAsErrors>4113;%(TreatSpecificWarningsAsErrors)</TreatSpecificWarningsAsErrors>
       <DisableSpecificWarnings>4267;4244;4018;4090;4047;4024;4311;4116;4146;4312;4028;4334;4101;4013;4020;%(DisableSpecificWarnings)</DisableSpecificWarnings>
+      <RuntimeLibrary>$(RuntimeLibrary)</RuntimeLibrary>
       <SDLCheck>false</SDLCheck>
     </ClCompile>
     <Link>
@@ -1648,6 +1654,7 @@ del /f /q $(OutDir)\licenses\gnutls.txt
       <MinimalRebuild>false</MinimalRebuild>
       <TreatSpecificWarningsAsErrors>4113;%(TreatSpecificWarningsAsErrors)</TreatSpecificWarningsAsErrors>
       <DisableSpecificWarnings>4267;4244;4018;4090;4047;4024;4311;4116;4146;4312;4028;4334;4101;4013;4020;%(DisableSpecificWarnings)</DisableSpecificWarnings>
+      <RuntimeLibrary>$(RuntimeLibrary)</RuntimeLibrary>
     </ClCompile>
     <Link>
       <OutputFile>$(OutDir)\bin\x64\$(TargetName)$(TargetExt)</OutputFile>
@@ -1719,6 +1726,7 @@ del /f /q $(OutDir)\licenses\gnutls.txt
       <CompileAsWinRT>false</CompileAsWinRT>
       <TreatSpecificWarningsAsErrors>4113;%(TreatSpecificWarningsAsErrors)</TreatSpecificWarningsAsErrors>
       <DisableSpecificWarnings>4267;4244;4018;4090;4047;4024;4311;4116;4146;4312;4028;4334;4101;4013;4020;%(DisableSpecificWarnings)</DisableSpecificWarnings>
+      <RuntimeLibrary>$(RuntimeLibrary)</RuntimeLibrary>
       <SDLCheck>false</SDLCheck>
     </ClCompile>
     <Link>
@@ -1798,6 +1806,7 @@ del /f /q $(OutDir)\licenses\gnutls.txt
       <ProgramDataBaseFileName>$(OutDir)\lib\x86\$(TargetName).pdb</ProgramDataBaseFileName>
       <TreatSpecificWarningsAsErrors>4113;%(TreatSpecificWarningsAsErrors)</TreatSpecificWarningsAsErrors>
       <DisableSpecificWarnings>4267;4244;4018;4090;4047;4024;4311;4116;4146;4312;4028;4334;4101;4013;4020;%(DisableSpecificWarnings)</DisableSpecificWarnings>
+      <RuntimeLibrary>$(RuntimeLibrary)</RuntimeLibrary>
     </ClCompile>
     <Lib>
       <OutputFile>$(OutDir)\lib\x86\$(TargetName)$(TargetExt)</OutputFile>
@@ -1870,6 +1879,7 @@ del /f /q $(OutDir)\licenses\gnutls.txt
       <CompileAsWinRT>false</CompileAsWinRT>
       <TreatSpecificWarningsAsErrors>4113;%(TreatSpecificWarningsAsErrors)</TreatSpecificWarningsAsErrors>
       <DisableSpecificWarnings>4267;4244;4018;4090;4047;4024;4311;4116;4146;4312;4028;4334;4101;4013;4020;%(DisableSpecificWarnings)</DisableSpecificWarnings>
+      <RuntimeLibrary>$(RuntimeLibrary)</RuntimeLibrary>
       <SDLCheck>false</SDLCheck>
     </ClCompile>
     <Lib>
@@ -1941,6 +1951,7 @@ del /f /q $(OutDir)\licenses\gnutls.txt
       <ProgramDataBaseFileName>$(OutDir)\lib\x64\$(TargetName).pdb</ProgramDataBaseFileName>
       <TreatSpecificWarningsAsErrors>4113;%(TreatSpecificWarningsAsErrors)</TreatSpecificWarningsAsErrors>
       <DisableSpecificWarnings>4267;4244;4018;4090;4047;4024;4311;4116;4146;4312;4028;4334;4101;4013;4020;%(DisableSpecificWarnings)</DisableSpecificWarnings>
+      <RuntimeLibrary>$(RuntimeLibrary)</RuntimeLibrary>
     </ClCompile>
     <Lib>
       <OutputFile>$(OutDir)\lib\x64\$(TargetName)$(TargetExt)</OutputFile>
@@ -2013,6 +2024,7 @@ del /f /q $(OutDir)\licenses\gnutls.txt
       <CompileAsWinRT>false</CompileAsWinRT>
       <TreatSpecificWarningsAsErrors>4113;%(TreatSpecificWarningsAsErrors)</TreatSpecificWarningsAsErrors>
       <DisableSpecificWarnings>4267;4244;4018;4090;4047;4024;4311;4116;4146;4312;4028;4334;4101;4013;4020;%(DisableSpecificWarnings)</DisableSpecificWarnings>
+      <RuntimeLibrary>$(RuntimeLibrary)</RuntimeLibrary>
       <SDLCheck>false</SDLCheck>
     </ClCompile>
     <Lib>
@@ -2083,6 +2095,7 @@ del /f /q $(OutDir)\licenses\gnutls.txt
       <ProgramDataBaseFileName>$(IntDir)$(TargetName).pdb</ProgramDataBaseFileName>
       <TreatSpecificWarningsAsErrors>4113;%(TreatSpecificWarningsAsErrors)</TreatSpecificWarningsAsErrors>
       <DisableSpecificWarnings>4267;4244;4018;4090;4047;4024;4311;4116;4146;4312;4028;4334;4101;4013;4020;%(DisableSpecificWarnings)</DisableSpecificWarnings>
+      <RuntimeLibrary>$(RuntimeLibrary)</RuntimeLibrary>
     </ClCompile>
     <Link>
       <EnableCOMDATFolding>true</EnableCOMDATFolding>
@@ -2160,6 +2173,7 @@ del /f /q $(OutDir)\licenses\gnutls.txt
       <ProgramDataBaseFileName>$(IntDir)$(TargetName).pdb</ProgramDataBaseFileName>
       <TreatSpecificWarningsAsErrors>4113;%(TreatSpecificWarningsAsErrors)</TreatSpecificWarningsAsErrors>
       <DisableSpecificWarnings>4267;4244;4018;4090;4047;4024;4311;4116;4146;4312;4028;4334;4101;4013;4020;%(DisableSpecificWarnings)</DisableSpecificWarnings>
+      <RuntimeLibrary>$(RuntimeLibrary)</RuntimeLibrary>
     </ClCompile>
     <Link>
       <EnableCOMDATFolding>true</EnableCOMDATFolding>
@@ -2239,6 +2253,7 @@ del /f /q $(OutDir)\licenses\gnutls.txt
       <CompileAsWinRT>false</CompileAsWinRT>
       <TreatSpecificWarningsAsErrors>4113;%(TreatSpecificWarningsAsErrors)</TreatSpecificWarningsAsErrors>
       <DisableSpecificWarnings>4267;4244;4018;4090;4047;4024;4311;4116;4146;4312;4028;4334;4101;4013;4020;%(DisableSpecificWarnings)</DisableSpecificWarnings>
+      <RuntimeLibrary>$(RuntimeLibrary)</RuntimeLibrary>
       <SDLCheck>false</SDLCheck>
     </ClCompile>
     <Link>
@@ -2321,6 +2336,7 @@ del /f /q $(OutDir)\licenses\gnutls.txt
       <CompileAsWinRT>false</CompileAsWinRT>
       <TreatSpecificWarningsAsErrors>4113;%(TreatSpecificWarningsAsErrors)</TreatSpecificWarningsAsErrors>
       <DisableSpecificWarnings>4267;4244;4018;4090;4047;4024;4311;4116;4146;4312;4028;4334;4101;4013;4020;%(DisableSpecificWarnings)</DisableSpecificWarnings>
+      <RuntimeLibrary>$(RuntimeLibrary)</RuntimeLibrary>
       <SDLCheck>false</SDLCheck>
     </ClCompile>
     <Link>
@@ -2401,6 +2417,7 @@ del /f /q $(OutDir)\licenses\gnutls.txt
       <ProgramDataBaseFileName>$(IntDir)$(TargetName).pdb</ProgramDataBaseFileName>
       <TreatSpecificWarningsAsErrors>4113;%(TreatSpecificWarningsAsErrors)</TreatSpecificWarningsAsErrors>
       <DisableSpecificWarnings>4267;4244;4018;4090;4047;4024;4311;4116;4146;4312;4028;4334;4101;4013;4020;%(DisableSpecificWarnings)</DisableSpecificWarnings>
+      <RuntimeLibrary>$(RuntimeLibrary)</RuntimeLibrary>
     </ClCompile>
     <Link>
       <EnableCOMDATFolding>true</EnableCOMDATFolding>
@@ -2477,6 +2494,7 @@ del /f /q $(OutDir)\licenses\gnutls.txt
       <ProgramDataBaseFileName>$(IntDir)$(TargetName).pdb</ProgramDataBaseFileName>
       <TreatSpecificWarningsAsErrors>4113;%(TreatSpecificWarningsAsErrors)</TreatSpecificWarningsAsErrors>
       <DisableSpecificWarnings>4267;4244;4018;4090;4047;4024;4311;4116;4146;4312;4028;4334;4101;4013;4020;%(DisableSpecificWarnings)</DisableSpecificWarnings>
+      <RuntimeLibrary>$(RuntimeLibrary)</RuntimeLibrary>
     </ClCompile>
     <Link>
       <EnableCOMDATFolding>true</EnableCOMDATFolding>
@@ -2555,6 +2573,7 @@ del /f /q $(OutDir)\licenses\gnutls.txt
       <CompileAsWinRT>false</CompileAsWinRT>
       <TreatSpecificWarningsAsErrors>4113;%(TreatSpecificWarningsAsErrors)</TreatSpecificWarningsAsErrors>
       <DisableSpecificWarnings>4267;4244;4018;4090;4047;4024;4311;4116;4146;4312;4028;4334;4101;4013;4020;%(DisableSpecificWarnings)</DisableSpecificWarnings>
+      <RuntimeLibrary>$(RuntimeLibrary)</RuntimeLibrary>
       <SDLCheck>false</SDLCheck>
     </ClCompile>
     <Link>
@@ -2636,6 +2655,7 @@ del /f /q $(OutDir)\licenses\gnutls.txt
       <CompileAsWinRT>false</CompileAsWinRT>
       <TreatSpecificWarningsAsErrors>4113;%(TreatSpecificWarningsAsErrors)</TreatSpecificWarningsAsErrors>
       <DisableSpecificWarnings>4267;4244;4018;4090;4047;4024;4311;4116;4146;4312;4028;4334;4101;4013;4020;%(DisableSpecificWarnings)</DisableSpecificWarnings>
+      <RuntimeLibrary>$(RuntimeLibrary)</RuntimeLibrary>
       <SDLCheck>false</SDLCheck>
     </ClCompile>
     <Link>
