diff --git a/m4/hooks.m4 b/m4/hooks.m4
index e1248c7..4656c3a 100644
--- a/m4/hooks.m4
+++ b/m4/hooks.m4
@@ -91,13 +91,14 @@ AC_MSG_ERROR([[
       mini_nettle=$withval,
       mini_nettle=no)
 
+  PKG_CHECK_MODULES(GMP, [gmp], [], [AC_MSG_ERROR([[gmp is required]])])
   AC_ARG_VAR(GMP_CFLAGS, [C compiler flags for gmp])
   AC_ARG_VAR(GMP_LIBS, [linker flags for gmp])
   if test "$mini_nettle" != no;then
     GMP_CFLAGS=""
     GMP_LIBS=""
   else
-    if test x$GMP_LIBS = x; then
+    if test "x$GMP_LIBS" = "x"; then
 	AC_CHECK_LIB(gmp, __gmpz_cmp, [GMP_LIBS="-lgmp"], [AC_MSG_ERROR([[
 ***
 *** gmp was not found.
