From aa0fa5f737b8ea3d2dfb396243be79af49274b6e Mon Sep 17 00:00:00 2001
From: abc <a@b.com>
Date: Fri, 18 Dec 2020 14:32:31 +0800
Subject: [PATCH 2/2] remove test subdirectory

---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ec66b515..cce875a0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -749,7 +749,6 @@ endif()
 if(ENABLE_GLIB)
   add_subdirectory(glib)
 endif()
-add_subdirectory(test)
 if(ENABLE_QT5)
   add_subdirectory(qt5)
 endif()
-- 
2.29.2.windows.2

