From 8d2aec6756830a1d42e0ea36ea6dbf5dce06e875 Mon Sep 17 00:00:00 2001
From: Matthias Kuhn <matthias@opengis.ch>
Date: Thu, 20 Jan 2022 08:25:44 +0100
Subject: [PATCH] Export cmake targets

---
 CMakeLists.txt | 21 ++++++++++++++++++++-
 1 file changed, 20 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1573249a..fdebefc8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -579,7 +579,26 @@ target_link_libraries(poppler LINK_PRIVATE ${poppler_LIBS})
 if(CMAKE_USE_PTHREADS_INIT)
    target_link_libraries(poppler LINK_PRIVATE Threads::Threads)
 endif()
-install(TARGETS poppler RUNTIME DESTINATION bin LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS poppler EXPORT unofficial-poppler-targets RUNTIME DESTINATION bin LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}" ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+
+install(EXPORT unofficial-poppler-targets
+    NAMESPACE unofficial::
+    DESTINATION share/unofficial-poppler
+)
+file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/unofficial-poppler-config.cmake.in"
+[[include(CMakeFindDependencyMacro)
+find_dependency(Freetype)
+find_dependency(LCMS2)
+if(@ENABLE_CURL@)
+    find_dependency(CURL)
+endif()
+if(@ENABLE_ZLIB@)
+    find_dependency(ZLIB)
+endif()
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-poppler-targets.cmake")
+]])
+configure_file("${CMAKE_CURRENT_BINARY_DIR}/unofficial-poppler-config.cmake.in" "${CMAKE_CURRENT_BINARY_DIR}/unofficial-poppler-config.cmake" @ONLY)
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/unofficial-poppler-config.cmake" DESTINATION share/unofficial-poppler)
 
 if(ENABLE_UNSTABLE_API_ABI_HEADERS)
   install(FILES
-- 
2.31.1

