diff --color -ur /home/mkuhn/dev/vcpkg/buildtrees/poppler/src/er-20.12.1-0f846aceea.clean/CMakeLists.txt /home/mkuhn/dev/vcpkg/buildtrees/poppler/src/er-20.12.1-0f846aceea/CMakeLists.txt
--- /home/mkuhn/dev/vcpkg/buildtrees/poppler/src/er-20.12.1-0f846aceea.clean/CMakeLists.txt	2022-01-19 18:49:42.309363752 +0100
+++ /home/mkuhn/dev/vcpkg/buildtrees/poppler/src/er-20.12.1-0f846aceea/CMakeLists.txt	2022-01-19 20:34:22.171542334 +0100
@@ -578,7 +578,26 @@
 if(CMAKE_USE_PTHREADS_INIT)
    target_link_libraries(poppler LINK_PRIVATE Threads::Threads)
 endif()
-install(TARGETS poppler RUNTIME DESTINATION bin LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS poppler EXPORT unofficial-poppler-targets RUNTIME DESTINATION bin LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}" ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+
+install(EXPORT unofficial-poppler-targets
+    NAMESPACE unofficial::
+    DESTINATION share/unofficial-poppler
+)
+file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/unofficial-poppler-config.cmake.in"
+[[include(CMakeFindDependencyMacro)
+find_dependency(Freetype)
+find_dependency(LCMS2)
+if(@ENABLE_CURL@)
+    find_dependency(CURL)
+endif()
+if(@ENABLE_ZLIB@)
+    find_dependency(ZLIB)
+endif()
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-poppler-targets.cmake")
+]])
+configure_file("${CMAKE_CURRENT_BINARY_DIR}/unofficial-poppler-config.cmake.in" "${CMAKE_CURRENT_BINARY_DIR}/unofficial-poppler-config.cmake" @ONLY)
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/unofficial-poppler-config.cmake" DESTINATION share/unofficial-poppler)
 
 if(ENABLE_UNSTABLE_API_ABI_HEADERS)
   install(FILES
