diff --git a/cmd/gvedit/main.cpp b/cmd/gvedit/main.cpp
index cc951df..6097de3 100644
--- a/cmd/gvedit/main.cpp
+++ b/cmd/gvedit/main.cpp
@@ -29,6 +29,7 @@
 #include <common/globals.h>
 
 #ifdef _MSC_VER
+#elif 0
 #pragma comment(lib, "cgraph.lib")
 #pragma comment(lib, "gvc.lib")
 #endif
diff --git a/plugin/gdk/gvloadimage_gdk.c b/plugin/gdk/gvloadimage_gdk.c
index d76f024..e8d6dff 100644
--- a/plugin/gdk/gvloadimage_gdk.c
+++ b/plugin/gdk/gvloadimage_gdk.c
@@ -23,6 +23,7 @@
 #include <gdk/gdk.h>
 
 #ifdef _MSC_VER //*dependencies
+#elif 0
     #pragma comment( lib, "gvc.lib" )
     #pragma comment( lib, "glib-2.0.lib" )
     #pragma comment( lib, "cairo.lib" )
