diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1ea6465..a3e7857 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -327,8 +327,8 @@ if(PkgConfig_FOUND)
       message(FATAL_ERROR "-DWITH_GDK=ON and GDK not found")
     endif()
   endif()
-  pkg_check_modules(GDK_PIXBUF gdk-pixbuf-2.0)
-  pkg_check_modules(LASI lasi)
+  set(GDK_PIXBUF_FOUND 0)
+  set(LASI_FOUND 0)
   if(AUTO_POPPLER)
     pkg_check_modules(POPPLER poppler-glib)
   elseif(WITH_POPPLER)
@@ -682,7 +682,12 @@ endif()
 
 if(UNIX)
   find_library(MATH_LIB m)
-  link_libraries(${MATH_LIB})
+  if(MATH_LIB)
+    set(MATH_LIB m CACHE INTERNAL "")
+    link_libraries(${MATH_LIB})
+  else()
+    set(MATH_LIB "")
+  endif()
 endif()
 
 if(WIN32)
diff --git a/cmake/FindCAIRO.cmake b/cmake/FindCAIRO.cmake
index a6fdf8e..a469065 100644
--- a/cmake/FindCAIRO.cmake
+++ b/cmake/FindCAIRO.cmake
@@ -1,8 +1,11 @@
 include(FindPackageHandleStandardArgs)
 
 find_package(PkgConfig)
-pkg_check_modules(CAIRO cairo)
-if(MINGW)
+pkg_check_modules(CAIRO cairo IMPORTED_TARGET)
+if(CAIRO_FOUND)
+  set(CAIRO_LIBRARIES PkgConfig::CAIRO)
+  set(CAIRO_LINK_LIBRARIES PkgConfig::CAIRO)
+elseif(MINGW)
   find_program(CAIRO_RUNTIME_LIBRARY NAMES libcairo-2.dll)
   find_program(EXPAT_RUNTIME_LIBRARY NAMES libexpat-1.dll)
   find_program(FONTCONFIG_RUNTIME_LIBRARY NAMES libfontconfig-1.dll)
diff --git a/cmake/FindGD.cmake b/cmake/FindGD.cmake
index 49d372c..2ed7e42 100644
--- a/cmake/FindGD.cmake
+++ b/cmake/FindGD.cmake
@@ -1,3 +1,11 @@
+find_package(PkgConfig)
+pkg_check_modules(GD gdlib IMPORTED_TARGET)
+set(GD_LIBRARIES PkgConfig::GD)
+foreach(item IN ITEMS FONTCONFIG FREETYPE GIF JPEG PNG)
+  set(HAVE_GD_${item} 1)
+endforeach()
+return()
+
 find_path(GD_INCLUDE_DIR gd.h)
 find_library(GD_LIBRARY NAMES gd libgd)
 
diff --git a/cmake/FindGTS.cmake b/cmake/FindGTS.cmake
index bc1d5ea..8114884 100644
--- a/cmake/FindGTS.cmake
+++ b/cmake/FindGTS.cmake
@@ -1,9 +1,11 @@
 include(FindPackageHandleStandardArgs)
 
 find_package(PkgConfig)
-pkg_check_modules(GTS gts)
+pkg_check_modules(GTS gts IMPORTED_TARGET)
 
-if(MINGW)
+if(GTS_FOUND)
+  set(GTS_LINK_LIBRARIES PkgConfig::GTS)
+elseif(MINGW)
   find_package(GLIB)
 
   find_program(GTS_RUNTIME_LIBRARY NAMES libgts-0-7-5.dll)
diff --git a/cmake/FindLTDL.cmake b/cmake/FindLTDL.cmake
index e74d23f..42d9ae2 100644
--- a/cmake/FindLTDL.cmake
+++ b/cmake/FindLTDL.cmake
@@ -20,4 +20,7 @@ mark_as_advanced(LTDL_INCLUDE_DIR LTDL_LIBRARY)
 set(LTDL_INCLUDE_DIRS ${LTDL_INCLUDE_DIR})
 if(NOT WIN32 OR MINGW)
   set(LTDL_LIBRARIES ${LTDL_LIBRARY})
+  if(CMAKE_DL_LIBS AND NOT BUILD_SHARED_LIBS)
+    set(LTDL_LIBRARIES "${LTDL_LIBRARIES};${CMAKE_DL_LIBS}")
+  endif()
 endif()
diff --git a/cmake/FindPANGOCAIRO.cmake b/cmake/FindPANGOCAIRO.cmake
index d185f0b..222789a 100644
--- a/cmake/FindPANGOCAIRO.cmake
+++ b/cmake/FindPANGOCAIRO.cmake
@@ -1,9 +1,12 @@
 include(FindPackageHandleStandardArgs)
 find_package(PkgConfig)
 
-pkg_check_modules(PANGOCAIRO pangocairo)
+pkg_check_modules(PANGOCAIRO pangocairo IMPORTED_TARGET)
 
-if(MINGW)
+if(1)
+  set(PANGOCAIRO_LIBRARIES PkgConfig::PANGOCAIRO)
+  set(PANGOCAIRO_LINK_LIBRARIES PkgConfig::PANGOCAIRO) # https://gitlab.kitware.com/cmake/cmake/-/issues/16154
+elseif(MINGW)
   find_package(GLIB)
 
   find_program(GOBJECT_RUNTIME_LIBRARY NAMES libgobject-2.0-0.dll)
