diff --git a/plugin/pango/CMakeLists.txt b/plugin/pango/CMakeLists.txt
index cadff33..c05608f 100644
--- a/plugin/pango/CMakeLists.txt
+++ b/plugin/pango/CMakeLists.txt
@@ -60,6 +60,7 @@ if(CAIRO_FOUND AND PANGOCAIRO_FOUND)
   set_target_properties(gvplugin_pango PROPERTIES
     VERSION ${GRAPHVIZ_PLUGIN_VERSION}.0.0
     SOVERSION ${GRAPHVIZ_PLUGIN_VERSION}
+    LINKER_LANGUAGE CXX # for cairo
   )
 
   if(MINGW)
