diff --git a/cmake/FindGObject.cmake b/cmake/FindGObject.cmake
index dc207a2..19a8a67 100644
--- a/cmake/FindGObject.cmake
+++ b/cmake/FindGObject.cmake
@@ -27,13 +27,13 @@
                             GLIB2::GLIB2
         )
     else()
-        find_path(GGObject_INCLUDE_DIR
+        find_path(GObject_INCLUDE_DIR
             NAMES gobject.h
             PATH_SUFFIXES glib-2.0/gobject/
         )
         find_library(GObject_LIBRARY NAMES gobject-2.0)
         if (GObject_LIBRARY AND GObject_INCLUDE_DIR)
-            add_library(GObject::GObject IMPORTED)
+            add_library(GObject::GObject INTERFACE IMPORTED)
             target_include_directories(GObject::GObject INTERFACE
                                     ${GObject_INCLUDE_DIR}
             )
