diff --git a/src/plugins/multimedia/darwin/CMakeLists.txt b/src/plugins/multimedia/darwin/CMakeLists.txt
index a1a0cb1..242ceac 100644
--- a/src/plugins/multimedia/darwin/CMakeLists.txt
+++ b/src/plugins/multimedia/darwin/CMakeLists.txt
@@ -34,6 +34,6 @@ qt_internal_add_plugin(QDarwinMediaPlugin
         ${FWMetal}
         ${FWQuartzCore}
-        AVFoundation::AVFoundation
+        ${FWAVFoundation}
 )
 
 qt_internal_extend_target(QDarwinMediaPlugin CONDITION NOT TVOS
