diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 97fe87669..0440e845e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -43,6 +43,14 @@ if (DBUS)
         PURPOSE "Required by the DBUS feature"
     )
 endif()
+# TODO: workaround for building with vcpkg. Bug?
+if(MSVC)
+    if (MSVC_RUNTIME_DYNAMIC)
+        add_library(ZSTD::ZSTD ALIAS zstd::libzstd_shared)
+    else()
+        add_library(ZSTD::ZSTD ALIAS zstd::libzstd_static)
+    endif()
+endif()
 
 # automatically call Qt moc, rcc and uic as needed for all targets by default
 set(CMAKE_AUTORCC_OPTIONS --compress 9 --threshold 5)
