diff --git a/CMakeLists.txt b/CMakeLists.txt
index a7b75f7..794af1d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -545,6 +545,9 @@ if(NOT CAF_NO_OPENSSL)
       message(STATUS
               "Disable OpenSSL. Required >= 1.0.1 due to TLSv1.2 support.")
       set(CAF_NO_OPENSSL yes)
+    elseif(CAF_OPENSSL_VERSION_OVERRIDE)
+      include_directories(BEFORE ${OPENSSL_INCLUDE_DIR})
+      add_optional_caf_lib(openssl)
     else()
       # Check if openssl headers and library versions match
       set(CMAKE_REQUIRED_LIBRARIES ${OPENSSL_LIBRARIES})
