diff --git a/fizz/CMakeLists.txt b/fizz/CMakeLists.txt
index 80c5051..9f2e3e6 100644
--- a/fizz/CMakeLists.txt
+++ b/fizz/CMakeLists.txt
@@ -92,10 +92,10 @@ endif()

 find_package(ZLIB REQUIRED)

-find_package(Libevent CONFIG QUIET)
-if(TARGET event)
+find_package(Libevent CONFIG REQUIRED)
+if(TARGET libevent::core)
   message(STATUS "Found libevent from package config")
-  list(APPEND FIZZ_SHINY_DEPENDENCIES event)
+  list(APPEND FIZZ_SHINY_DEPENDENCIES libevent::core)
 else()
   find_package(Libevent MODULE REQUIRED)
   list(APPEND FIZZ_LINK_LIBRARIES ${LIBEVENT_LIB})
diff --git a/fizz/cmake/fizz-config.cmake.in b/fizz/cmake/fizz-config.cmake.in
index b28750f..282c42d 100644
--- a/fizz/cmake/fizz-config.cmake.in
+++ b/fizz/cmake/fizz-config.cmake.in
@@ -29,6 +29,7 @@ include(CMakeFindDependencyMacro)
 find_dependency(unofficial-sodium CONFIG REQUIRED)
 find_dependency(folly CONFIG)
 find_dependency(ZLIB)
+find_dependency(Libevent CONFIG)

 if (NOT fizz_FIND_QUIETLY)
   message(STATUS "Found fizz: ${PACKAGE_PREFIX_DIR}")
