diff --git a/fizz/CMakeLists.txt b/fizz/CMakeLists.txt
index ab3b48f..c84890e 100644
--- a/fizz/CMakeLists.txt
+++ b/fizz/CMakeLists.txt
@@ -40,6 +40,7 @@ endif()
 
 find_package(Boost REQUIRED COMPONENTS system thread filesystem regex context)
 find_package(OpenSSL REQUIRED)
+find_package(ZLIB REQUIRED)
 find_package(Glog REQUIRED)
 find_package(DoubleConversion REQUIRED)
 find_package(Threads REQUIRED)
@@ -209,6 +209,7 @@ target_link_libraries(fizz
  PUBLIC
      ${FOLLY_LIBRARIES}
      ${Boost_LIBRARIES}
+     ${ZLIB_LIBRARIES}
      ${OPENSSL_LIBRARIES}
      # Don't use the sodium target here because it will break clients that
      # consume fizz's exported targets (fizz-targets.cmake) since the sodium
