diff --git a/fizz/CMakeLists.txt b/fizz/CMakeLists.txt
index ab3b48f..c84890e 100644
--- a/fizz/CMakeLists.txt
+++ b/fizz/CMakeLists.txt
@@ -35,6 +35,7 @@ endif()
 
 find_package(Boost REQUIRED COMPONENTS system thread filesystem regex context)
 find_package(OpenSSL REQUIRED)
+find_package(ZLIB REQUIRED)
 find_package(Glog REQUIRED)
 find_package(Gflags REQUIRED)
 find_package(Libevent REQUIRED)
@@ -169,6 +170,7 @@ target_link_libraries(fizz
     ${FOLLY_LIBRARIES}
     ${Boost_LIBRARIES}
     ${OPENSSL_LIBRARIES}
+    ${ZLIB_LIBRARIES}
     # Don't use the sodium target here because it will break clients that
     # consume fizz's exported targets (fizz-targets.cmake) since the sodium
     # target is not exported.
