diff --git a/cmake/version.cmake b/cmake/version.cmake
index b349b75..667b5c3 100644
--- a/cmake/version.cmake
+++ b/cmake/version.cmake
@@ -180,7 +180,7 @@ function( create_version_header dest_path target )
         add_custom_command(
             OUTPUT ${version_h_output}
             # On Windows this command has to be invoked by a shell in order to work
-            COMMAND ${BASH_EXECUTABLE} -c "\"scripts/mkversion\" \"-v\" \"${KTX_GIT_VERSION_FULL}\" \"-o\" \"version.h\" \"${dest_path}\""
+            COMMAND "${BASH_EXECUTABLE}" -- ./scripts/mkversion -v "${KTX_GIT_VERSION_FULL}" -o version.h "${dest_path}"
             WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
             COMMENT "Generate ${version_h_output}"
             VERBATIM
