diff --git a/CMakeLists.txt b/CMakeLists.txt
index ccf4af61..4b232a50 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -485,9 +485,9 @@ macro(common_libktx_settings target enable_write library_type)
 
     )
     if(IOS)
-        set_target_properties(${target} PROPERTIES
-            FRAMEWORK TRUE
-        )
+        #set_target_properties(${target} PROPERTIES
+        #    FRAMEWORK TRUE
+        #)
     endif()
 
     if( NOT ${library_type} STREQUAL STATIC )
