diff --git a/mk/vs/Common++.vcxproj.template b/mk/vs/Common++.vcxproj.template
index 907be1e..c42c62b 100644
--- a/mk/vs/Common++.vcxproj.template
+++ b/mk/vs/Common++.vcxproj.template
@@ -77,20 +77,20 @@
   </ImportGroup>
   <PropertyGroup Label="UserMacros" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
-    <OutDir>$(PcapPlusPlusHome)\Common++\Lib</OutDir>
-    <IntDir>$(PcapPlusPlusHome)\Common++\Obj</IntDir>
+    <OutDir>Common++\Lib</OutDir>
+    <IntDir>Common++\Obj</IntDir>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|Win32'">
-    <OutDir>$(PcapPlusPlusHome)\Common++\Lib</OutDir>
-    <IntDir>$(PcapPlusPlusHome)\Common++\Obj</IntDir>
+    <OutDir>Common++\Lib</OutDir>
+    <IntDir>Common++\Obj</IntDir>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
-    <OutDir>$(PcapPlusPlusHome)\Common++\Lib</OutDir>
-    <IntDir>$(PcapPlusPlusHome)\Common++\Obj</IntDir>
+    <OutDir>Common++\Lib</OutDir>
+    <IntDir>Common++\Obj</IntDir>
   </PropertyGroup>
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
-    <OutDir>$(PcapPlusPlusHome)\Common++\Lib</OutDir>
-    <IntDir>$(PcapPlusPlusHome)\Common++\Obj</IntDir>
+    <OutDir>Common++\Lib</OutDir>
+    <IntDir>Common++\Obj</IntDir>
   </PropertyGroup>
   <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
     <ClCompile>
