diff --git a/CMakeLists.txt b/CMakeLists.txt
index 49de04d..1e67a43 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -115,7 +115,7 @@ endif ()
 if (SIMSIMD_BUILD_SHARED)
     set(SIMSIMD_SOURCES ${SIMSIMD_SOURCES} c/lib.c)
     add_library(simsimd_shared SHARED ${SIMSIMD_SOURCES})
-    target_include_directories(simsimd_shared PUBLIC "${PROJECT_SOURCE_DIR}/include")
+    target_include_directories(simsimd_shared PUBLIC $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include> $<INSTALL_INTERFACE:include>)
     set_target_properties(simsimd_shared PROPERTIES OUTPUT_NAME simsimd)
 
     install(
@@ -129,7 +129,21 @@ if (SIMSIMD_BUILD_SHARED)
         PUBLIC_HEADER
         RESOURCE
         RUNTIME)
+
+    install(TARGETS simsimd_shared EXPORT unofficial-simsimd-config
+        LIBRARY DESTINATION lib
+        ARCHIVE DESTINATION lib
+        RUNTIME DESTINATION bin)
+
+    install(EXPORT unofficial-simsimd-config
+        FILE unofficial-simsimd-config.cmake
+        NAMESPACE unofficial::simsimd::
+        DESTINATION share/unofficial-simsimd
+    )
 endif ()
 
-install(DIRECTORY ./include/ DESTINATION /usr/include/)
-install(DIRECTORY ./c/ DESTINATION /usr/src/${PROJECT_NAME}/)
+install(DIRECTORY ${CMAKE_SOURCE_DIR}/include/
+   DESTINATION include
+   FILES_MATCHING
+   PATTERN *.h
+)
