--- a/cmake/modules/FindLibRaw.cmake	2016-11-02 07:09:50.000000000 +0100
+++ b/cmake/modules/FindLibRaw.cmake	2019-06-03 22:05:28.759307500 +0200
@@ -31,18 +31,36 @@
           PATH_SUFFIXES libraw
          )
 
-FIND_LIBRARY(LibRaw_LIBRARIES NAMES raw
+FIND_LIBRARY(LibRaw_LIBRARY_RELEASE NAMES raw
              HINTS
              ${PC_LIBRAW_LIBDIR}
              ${PC_LIBRAW_LIBRARY_DIRS}
             )
 
-FIND_LIBRARY(LibRaw_r_LIBRARIES NAMES raw_r
+FIND_LIBRARY(LibRaw_LIBRARY_DEBUG NAMES rawd
+             HINTS
+             ${PC_LIBRAW_LIBDIR}
+             ${PC_LIBRAW_LIBRARY_DIRS}
+            )
+
+INCLUDE(SelectLibraryConfigurations)
+
+select_library_configurations(LibRaw)
+
+FIND_LIBRARY(LibRaw_r_LIBRARY_RELEASE NAMES raw_r
              HINTS
              ${PC_LIBRAW_R_LIBDIR}
              ${PC_LIBRAW_R_LIBRARY_DIRS}
             )
 
+FIND_LIBRARY(LibRaw_r_LIBRARY_DEBUG NAMES raw_rd
+             HINTS
+             ${PC_LIBRAW_R_LIBDIR}
+             ${PC_LIBRAW_R_LIBRARY_DIRS}
+            )
+
+select_library_configurations(LibRaw_r)
+
 IF(LibRaw_INCLUDE_DIR)
    FILE(READ ${LibRaw_INCLUDE_DIR}/libraw_version.h _libraw_version_content)
    
