diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1afe304..88c0294 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -584,7 +584,7 @@ SET_TARGET_PROPERTIES(raw_r PROPERTIES COMPILE_PDB_NAME "raw_r")
 if (LIBRAW_INSTALL)
 
 # Configure and install data file for packaging.
-IF(NOT WIN32)
+IF(1)
     CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/cmake/data/libraw.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/libraw.pc @ONLY)
     INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/libraw.pc DESTINATION lib${LIB_SUFFIX}/pkgconfig)
 
@@ -610,8 +610,8 @@ INSTALL(FILES       ${LIBRAW_PATH}/libraw/libraw.h
 # Install Shared binary files.
 INSTALL(TARGETS raw
         RUNTIME DESTINATION bin
-        LIBRARY DESTINATION lib${LIB_SUFFIX}
-        ARCHIVE DESTINATION lib${LIB_SUFFIX}
+        LIBRARY DESTINATION lib${LIB_SUFFIX}/manual-link
+        ARCHIVE DESTINATION lib${LIB_SUFFIX}/manual-link
        )
 
 INSTALL(TARGETS raw_r
diff --git a/cmake/data/libraw.pc.cmake b/cmake/data/libraw.pc.cmake
index aede5f2..ffeefc2 100644
--- a/cmake/data/libraw.pc.cmake
+++ b/cmake/data/libraw.pc.cmake
@@ -1,6 +1,6 @@
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-libdir=${prefix}/lib@LIB_SUFFIX@
+libdir=${prefix}/lib@LIB_SUFFIX@/manual-link
 includedir=${prefix}/include/libraw
 
 Name: @PROJECT_NAME@
