diff --git a/src/libraw_cxx.cpp b/src/libraw_cxx.cpp
index 0abd5329a..5eea00ea9 100644
--- a/src/libraw_cxx.cpp
+++ b/src/libraw_cxx.cpp
@@ -3500,7 +3500,7 @@ int LibRaw::phase_one_subtract_black(ushort *src, ushort *dest)
     {
       if (!imgdata.rawdata.ph1_cblack || !imgdata.rawdata.ph1_rblack)
       {
-        register int bl = imgdata.color.phase_one_data.t_black;
+        int bl = imgdata.color.phase_one_data.t_black;
         for (int row = 0; row < S.raw_height; row++)
         {
           checkCancel();
@@ -3514,7 +3514,7 @@ int LibRaw::phase_one_subtract_black(ushort *src, ushort *dest)
       }
       else
       {
-        register int bl = imgdata.color.phase_one_data.t_black;
+        int bl = imgdata.color.phase_one_data.t_black;
         for (int row = 0; row < S.raw_height; row++)
         {
           checkCancel();
@@ -6467,7 +6467,7 @@ static void utf2char(utf16_t *str, char *buffer, unsigned bufsz)
 
 static void *lr_memmem(const void *l, size_t l_len, const void *s, size_t s_len)
 {
-  register char *cur, *last;
+  char *cur, *last;
   const char *cl = (const char *)l;
   const char *cs = (const char *)s;
 
