diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 14d5e85..7e46870 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -65,6 +65,14 @@ if(NOT DEFINED CMAKE_INSTALL_LIBDIR)
 endif(NOT DEFINED CMAKE_INSTALL_LIBDIR)
 set(libdir ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR})
 
+foreach(t ${install_libs})
+    target_include_directories(
+        ${t}
+        INTERFACE $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+                  $<INSTALL_INTERFACE:include>
+    )
+endforeach()
+
 install(
     DIRECTORY "include/gainput"
     DESTINATION "include"
@@ -73,7 +81,15 @@ install(
 
 install(
     TARGETS ${install_libs}
+    EXPORT  unofficial-${PROJECT_NAME}-Targets
     LIBRARY DESTINATION "${libdir}"
     ARCHIVE DESTINATION "${libdir}"
     RUNTIME DESTINATION "bin"
 )
+
+install(
+    EXPORT unofficial-${PROJECT_NAME}-Targets
+    FILE unofficial-${PROJECT_NAME}Config.cmake
+    NAMESPACE unofficial::${PROJECT_NAME}::
+    DESTINATION share/unofficial-${PROJECT_NAME}
+)
\ No newline at end of file
