diff --git a/cmake/FindFeature.cmake b/cmake/FindFeature.cmake
index 11bd9f1..231dcf4 100644
--- a/cmake/FindFeature.cmake
+++ b/cmake/FindFeature.cmake
@@ -15,7 +15,7 @@ macro(find_feature _feature _type _purpose _description)
       set(_feature_default "ON")
       message(STATUS "Finding ${_type_lower} feature ${_feature} for ${_purpose} (${_description})")
       find_package(${_feature} REQUIRED)
-    elseif(${_type} STREQUAL "RECOMMENDED")
+    elseif(0)
       if(NOT ${WITH_${_feature_upper}})
         set(_feature_default "OFF")
         message(STATUS "Skipping ${_type_lower} feature ${_feature} for ${_purpose} (${_description})")
@@ -31,7 +31,7 @@ macro(find_feature _feature _type _purpose _description)
           )
         endif()
       endif()
-    elseif(${_type} STREQUAL "OPTIONAL")
+    elseif(1)
       if(${WITH_${_feature_upper}})
         set(_feature_default "ON")
         message(STATUS "Finding ${_type_lower} feature ${_feature} for ${_purpose} (${_description})")
