diff --git a/client/FreeRDP-ClientConfig.cmake.in b/client/FreeRDP-ClientConfig.cmake.in
index 35b74c1..a4e48c9 100644
--- a/client/FreeRDP-ClientConfig.cmake.in
+++ b/client/FreeRDP-ClientConfig.cmake.in
@@ -1,6 +1,7 @@
 include(CMakeFindDependencyMacro)
 find_dependency(WinPR @FREERDP_VERSION@)
 find_dependency(FreeRDP @FREERDP_VERSION@)
+find_dependency(ZLIB)
 
 @PACKAGE_INIT@
 
diff --git a/libfreerdp/FreeRDPConfig.cmake.in b/libfreerdp/FreeRDPConfig.cmake.in
index 21ecd59..2613a59 100644
--- a/libfreerdp/FreeRDPConfig.cmake.in
+++ b/libfreerdp/FreeRDPConfig.cmake.in
@@ -1,5 +1,7 @@
 include(CMakeFindDependencyMacro)
 find_dependency(WinPR @FREERDP_VERSION@)
+find_dependency(cJSON)
+find_dependency(ZLIB)
 
 @PACKAGE_INIT@
 
diff --git a/libfreerdp/freerdp.pc.in b/libfreerdp/freerdp.pc.in
index 21ea0c7..71b8e18 100644
--- a/libfreerdp/freerdp.pc.in
+++ b/libfreerdp/freerdp.pc.in
@@ -14,7 +14,7 @@ Description: FreeRDP: A Remote Desktop Protocol Implementation
 URL: http://www.freerdp.com/
 Version: @FREERDP_VERSION@
 Requires: 
-Requires.private: winpr@FREERDP_API_VERSION@
+Requires.private: winpr@FREERDP_API_VERSION@ libcjson zlib
 Libs: -L${libdir} ${libs}
 Libs.private: -ldl -lpthread
 Cflags: -I${includedir}
