diff --git a/CMakeCPack.cmake b/CMakeCPack.cmake
index 6ccfc7d..75479bf 100644
--- a/CMakeCPack.cmake
+++ b/CMakeCPack.cmake
@@ -67,9 +67,9 @@ if(MSVC)
 		set(CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP TRUE)
 		include(InstallRequiredSystemLibraries)
 
-		install(PROGRAMS ${CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS}
-			DESTINATION ${CMAKE_INSTALL_BINDIR}
-			COMPONENT libraries)
+		# install(PROGRAMS ${CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS}
+		# 	DESTINATION ${CMAKE_INSTALL_BINDIR}
+		# 	COMPONENT libraries)
 	endif()
 endif()
 
