diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2bff2e8..db346d3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -711,6 +711,9 @@ else()
 	set(FREERDP_LIBRARY_PATH "${CMAKE_INSTALL_LIBDIR}")
 	set(FREERDP_PLUGIN_PATH "${CMAKE_INSTALL_LIBDIR}/${FREERDP_MAJOR_DIR}")
 endif()
+set(FREERDP_DATA_PATH "share/freerdp")
+set(FREERDP_LIBRARY_PATH "${CMAKE_INSTALL_LIBDIR}")
+set(FREERDP_PLUGIN_PATH "${CMAKE_INSTALL_LIBDIR}/freerdp${FREERDP_VERSION_MAJOR}")
 set(FREERDP_ADDIN_PATH "${FREERDP_PLUGIN_PATH}")
 
 # Path to put extensions
diff --git a/client/Windows/CMakeLists.txt b/client/Windows/CMakeLists.txt
index cc94b99..0423242 100644
--- a/client/Windows/CMakeLists.txt
+++ b/client/Windows/CMakeLists.txt
@@ -107,7 +107,7 @@ endif()
 target_link_libraries(${MODULE_NAME} PRIVATE ${LIBS})
 
 if(WITH_CLIENT_INTERFACE)
-	install(TARGETS ${MODULE_NAME} DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT libraries)
+	install(TARGETS ${MODULE_NAME} COMPONENT libraries)
 	if (WITH_DEBUG_SYMBOLS AND MSVC AND BUILD_SHARED_LIBS)
 		install(FILES ${CMAKE_PDB_BINARY_DIR}/${MODULE_NAME}.pdb DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT symbols)
 	endif()
