diff --git a/libfreerdp/CMakeLists.txt b/libfreerdp/CMakeLists.txt
index 10a0921..7e9a63d 100644
--- a/libfreerdp/CMakeLists.txt
+++ b/libfreerdp/CMakeLists.txt
@@ -363,7 +363,6 @@ endif()
 # /primitives
 
 list(REMOVE_DUPLICATES LIBFREERDP_DEFINITIONS)
-list(REMOVE_DUPLICATES LIBFREERDP_LIBS)
 list(REMOVE_DUPLICATES LIBFREERDP_INCLUDES)
 include_directories(${LIBFREERDP_INCLUDES})
 
diff --git a/winpr/libwinpr/CMakeLists.txt b/winpr/libwinpr/CMakeLists.txt
index da23a1b..2616439 100644
--- a/winpr/libwinpr/CMakeLists.txt
+++ b/winpr/libwinpr/CMakeLists.txt
@@ -113,8 +113,6 @@ endforeach()
 
 set(MODULE_NAME winpr)
 list(REMOVE_DUPLICATES WINPR_DEFINITIONS)
-list(REMOVE_DUPLICATES WINPR_LIBS_PRIVATE)
-list(REMOVE_DUPLICATES WINPR_LIBS_PUBLIC)
 list(REMOVE_DUPLICATES WINPR_INCLUDES)
 include_directories(${WINPR_INCLUDES})
 
diff --git a/winpr/tools/CMakeLists.txt b/winpr/tools/CMakeLists.txt
index b89a954..13f4c13 100644
--- a/winpr/tools/CMakeLists.txt
+++ b/winpr/tools/CMakeLists.txt
@@ -82,7 +82,6 @@ add_subdirectory(makecert)
 
 set(MODULE_NAME winpr-tools)
 list(REMOVE_DUPLICATES WINPR_TOOLS_DEFINITIONS)
-list(REMOVE_DUPLICATES WINPR_TOOLS_LIBS)
 list(REMOVE_DUPLICATES WINPR_TOOLS_INCLUDES)
 include_directories(${WINPR_TOOLS_INCLUDES})
 
