diff --git a/CMakeLists.txt b/CMakeLists.txt
index c1428a9..ecfe068 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -85,7 +85,7 @@ set(CMAKE_CXX_STANDARD 17)
 # don't generally produce global targets. To guarantee all imported targets are 
 # global, this module is included at the project root level.
 
-include(FindExtPackages)
+find_package(nlohmann_json CONFIG REQUIRED)
 
 add_subdirectory(casbin)
 
