diff --git "a/Main/StackWalker/StackWalker.h" "b/Main/StackWalker/StackWalker.h"
index 67eb90575..e906dbc26 100644
--- "a/Main/StackWalker/StackWalker.h"
+++ "b/Main/StackWalker/StackWalker.h"
@@ -229,13 +229,11 @@ protected:
   {                                                               \
     memset(&c, 0, sizeof(CONTEXT));                               \
     c.ContextFlags = contextFlags;                                \
-    __asm {                                                       \
-        call x                                                    \
-     x: pop eax                                                   \
-        mov c.Eip, eax                                            \
-        mov c.Ebp, ebp                                            \
-        mov c.Esp, esp                                            \
-    };                                                            \
+    __asm    call x                                               \
+    __asm x: pop eax                                              \
+    __asm    mov c.Eip, eax                                       \
+    __asm    mov c.Ebp, ebp                                       \
+    __asm    mov c.Esp, esp                                       \
   } while (0);
 // clang-format on
 #endif
