diff --git a/QGLViewer/QGLViewer.pro b/QGLViewer/QGLViewer.pro
index e0205e5..6db4c81 100644
--- a/QGLViewer/QGLViewer.pro
+++ b/QGLViewer/QGLViewer.pro
@@ -108,9 +108,6 @@ contains( DEFINES, NO_VECTORIAL_RENDER ) {
 # --  U n i x  --
 # ---------------
 unix {
-	CONFIG -= debug debug_and_release
-	CONFIG *= release
-
 	# INCLUDE_DIR and LIB_DIR specify where to install the include files and the library.
 	# Use qmake INCLUDE_DIR=... LIB_DIR=... , or qmake PREFIX=... to customize your installation.
 	isEmpty( PREFIX ) {
@@ -250,9 +247,6 @@ macx|darwin-g++ {
 # --  W i n d o w s  --
 # ---------------------
 win32 {
-	# Windows requires a debug lib version to link against debug applications
-	CONFIG *= debug_and_release build_all
-
 	# Needed by Intel C++, (icl.exe) so that WINGDIAPI is a defined symbol in gl.h.
 	DEFINES *= WIN32
 
@@ -279,7 +273,7 @@ win32 {
 		QMAKE_CXXFLAGS *= -TP -GR
 		DEFINES += NOMINMAX
 		win32-msvc {
-			QMAKE_CXXFLAGS *= -EH -FS
+			QMAKE_CXXFLAGS *= -EHs -FS
 		} else {
 			QMAKE_CXXFLAGS *= -EHs
 		}
