diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6b1f1dc..7b034dd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -396,7 +396,9 @@ if (NSYNC_ENABLE_TESTS)
 	endforeach (t)
 endif ()
 
-set (CMAKE_SKIP_INSTALL_ALL_DEPENDENCY ON)
+if(WIN32 AND NOT VCPKG_TARGET_TRIPLET STREQUAL "x64-windows-static")
+    set (CMAKE_SKIP_INSTALL_ALL_DEPENDENCY ON)
+endif()
 
 install (TARGETS nsync
 	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT RuntimeLibraries
