diff --git a/folly/memory/UninitializedMemoryHacks.h b/folly/memory/UninitializedMemoryHacks.h
index bd31c88..9f640a8 100644
--- a/folly/memory/UninitializedMemoryHacks.h
+++ b/folly/memory/UninitializedMemoryHacks.h
@@ -101,6 +101,9 @@ template <
         typename std::enable_if<std::is_trivially_destructible<T>::value>::type>
 inline void resizeWithoutInitialization(
     std::basic_string<T>& s, std::size_t n) {
+#if defined(_MSVC_STL_UPDATE) && _MSVC_STL_UPDATE >= 202206L
+    s.resize(n);
+#else
   if (n <= s.size()) {
     s.resize(n);
   } else {
@@ -111,6 +114,7 @@ inline void resizeWithoutInitialization(
     }
     detail::unsafeStringSetLargerSize(s, n);
   }
+#endif // STL workaround
 }
 
 /**
