diff --git a/CMake/folly-deps.cmake b/CMake/folly-deps.cmake
index 396fa6c..9d80d99 100644
--- a/CMake/folly-deps.cmake
+++ b/CMake/folly-deps.cmake
@@ -22,7 +22,7 @@ list(APPEND FOLLY_LINK_LIBRARIES ${DOUBLE_CONVERSION_LIBRARY})
 list(APPEND FOLLY_INCLUDE_DIRECTORIES ${DOUBLE_CONVERSION_INCLUDE_DIR})
 
 set(FOLLY_HAVE_LIBGFLAGS OFF)
-find_package(GFlags CONFIG QUIET)
+find_package(gflags CONFIG REQUIRED)
 if (gflags_FOUND)
   message(STATUS "Found gflags from package config")
   set(FOLLY_HAVE_LIBGFLAGS ON)
