diff --git a/CMake/GenPkgConfig.cmake b/CMake/GenPkgConfig.cmake
index 0e93175..1637d9f 100644
--- a/CMake/GenPkgConfig.cmake
+++ b/CMake/GenPkgConfig.cmake
@@ -93,7 +93,7 @@ function(gen_pkgconfig_vars)
   # Set the output variables
   string(REPLACE ";" " " cflags "${cflags}")
   string(REPLACE ";" " " private_libs "${private_libs}")
-
+  string(REPLACE "<$<COMPILE_LANGUAGE:CXX>:/utf-8>" "/utf-8" cflags "${cflags}")
   # Since CMake 3.18 FindThreads may include a generator expression requiring
   # a target, which gets propagated to us through INTERFACE_COMPILE_OPTIONS.
   # Before CMake 3.19 there's no way to solve this in a general way, so we
