--- a/folly/stats/Histogram.h
+++ b/folly/stats/Histogram.h
@@ -485,7 +485,7 @@
 // with /permissive- (the default for the CMake build), so we directly include
 // the -defs as if it were -inl, and don't provide the explicit instantiations.
 // https://developercommunity.visualstudio.com/content/problem/81223/incorrect-error-c5037-with-permissive.html
-#if defined(_MSC_VER) && _MSC_FULL_VER >= 191125506 && _MSC_FULL_VER < 191125542
+#if defined(_MSC_VER) && _MSC_FULL_VER >= 191125506 && _MSC_FULL_VER <= 191125547
 #define FOLLY_MSVC_USE_WORKAROUND_FOR_C5037 1
 #else
 #define FOLLY_MSVC_USE_WORKAROUND_FOR_C5037 0
