diff --git a/CMake/FollyCompilerUnix.cmake b/CMake/FollyCompilerUnix.cmake
index b7d994b..a518cbe 100644
--- a/CMake/FollyCompilerUnix.cmake
+++ b/CMake/FollyCompilerUnix.cmake
@@ -16,7 +16,6 @@ function(apply_folly_compile_options_to_target THETARGET)
       -std=gnu++14
       -finput-charset=UTF-8
       -fsigned-char
-      -Werror
       -Wall
       -Wno-deprecated
       -Wdeprecated-declarations
