diff --git a/CMakeLists.txt b/CMakeLists.txt
index abc9be0..eb999f5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -190,7 +190,6 @@ endif()
 
 include_directories( "${INCLUDE_PATH}" )
 include_directories( "${SOURCE_PATH}" )
-include_directories( SYSTEM "${SFML_INCLUDE_DIR}" )
 include_directories( SYSTEM "${PROJECT_SOURCE_DIR}/extlibs/libELL/include" )
 
 # Set the library output directory
@@ -244,7 +243,7 @@ if( WIN32 )
 	endif()
 
 	add_definitions( -DWIN32 )
-	target_link_libraries( sfgui ${SFML_LIBRARIES} ${SFML_DEPENDENCIES} ${OPENGL_gl_LIBRARY} )
+	target_link_libraries( sfgui sfml-graphics sfml-window sfml-system ${OPENGL_gl_LIBRARY} )
 
 	if( CMAKE_COMPILER_IS_GNUCXX )
 		if( SFGUI_BUILD_SHARED_LIBS )
@@ -261,11 +260,11 @@ elseif( APPLE )
 	mark_as_advanced( COREFOUNDATION_LIBRARY )
 
 	include_directories( SYSTEM /System/Library/Frameworks/CoreFoundation.framework/Headers )
-	target_link_libraries( sfgui ${SFML_LIBRARIES} ${SFML_DEPENDENCIES} ${OPENGL_gl_LIBRARY} ${COREFOUNDATION_LIBRARY} )
+	target_link_libraries( sfgui sfml-graphics sfml-window sfml-system ${OPENGL_gl_LIBRARY} ${COREFOUNDATION_LIBRARY} )
 	set( SHARE_PATH "${CMAKE_INSTALL_PREFIX}/share/SFGUI" )
 	set( LIB_PATH "lib" )
 elseif( "${CMAKE_SYSTEM_NAME}" MATCHES "Linux" )
-	target_link_libraries( sfgui ${SFML_LIBRARIES} ${SFML_DEPENDENCIES} ${OPENGL_gl_LIBRARY} ${X11_LIBRARIES} )
+	target_link_libraries( sfgui sfml-graphics sfml-window sfml-system ${OPENGL_gl_LIBRARY} ${X11_LIBRARIES} )
 	set( SHARE_PATH "${CMAKE_INSTALL_PREFIX}/share/SFGUI" )
 	
 	if( LIB_SUFFIX )
@@ -274,7 +273,7 @@ elseif( "${CMAKE_SYSTEM_NAME}" MATCHES "Linux" )
 		set( LIB_PATH "lib" )
 	endif()
 else()
-	target_link_libraries( sfgui ${SFML_LIBRARIES} ${SFML_DEPENDENCIES} ${OPENGL_gl_LIBRARY} )
+	target_link_libraries( sfgui sfml-graphics sfml-window sfml-system ${OPENGL_gl_LIBRARY} )
 	set( SHARE_PATH "${CMAKE_INSTALL_PREFIX}/share/SFGUI" )
 	set( LIB_PATH "lib" )
 endif()
