diff --git a/CMakeLists.txt b/CMakeLists.txt
index 085cc82..7da39f2 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -33,6 +33,11 @@ project(
   HOMEPAGE_URL "https://libcopp.atframe.work"
   LANGUAGES C CXX ASM)
 
+# libcopp requires x86 to be called i386
+if(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86")
+  set(CMAKE_SYSTEM_PROCESSOR i386)
+endif()
+
 # ######################################################################################################################
 include("${PROJECT_SOURCE_DIR}/project/cmake/ProjectBuildOption.cmake")
 
