diff --git a/tools/pybind11NewTools.cmake b/tools/pybind11NewTools.cmake
index 5ab3142..de0e63b 100644
--- a/tools/pybind11NewTools.cmake
+++ b/tools/pybind11NewTools.cmake
@@ -245,10 +245,7 @@ if(TARGET ${_Python}::Module)
     APPEND
     PROPERTY INTERFACE_LINK_LIBRARIES ${_Python}::Module)
 else()
-  set_property(
-    TARGET pybind11::module
-    APPEND
-    PROPERTY INTERFACE_LINK_LIBRARIES pybind11::python_link_helper)
+  target_link_libraries(${_Python}::Module INTERFACE ${${_Python}_LIBRARIES})
 endif()
 
 # WITHOUT_SOABI and WITH_SOABI will disable the custom extension handling used by pybind11.
