diff --git a/imgui-SFML.cpp b/imgui-SFML.cpp
index 6ca64ae..94af0c8 100644
--- a/imgui-SFML.cpp
+++ b/imgui-SFML.cpp
@@ -128,7 +128,7 @@ void initDefaultJoystickMapping();
 // Returns first id of connected joystick
 unsigned int getConnectedJoystickId();
 
-void updateJoystickActionState(ImGuiIO& io, ImGuiNavInput_ action);
+void updateJoystickActionState(ImGuiIO& io, ImGuiNavInput action);
 void updateJoystickDPadState(ImGuiIO& io);
 void updateJoystickLStickState(ImGuiIO& io);
 
@@ -1030,7 +1030,7 @@ void initDefaultJoystickMapping() {
     ImGui::SFML::SetJoystickLStickThreshold(5.f);
 }
 
-void updateJoystickActionState(ImGuiIO& io, ImGuiNavInput_ action) {
+void updateJoystickActionState(ImGuiIO& io, ImGuiNavInput action) {
     bool isPressed = sf::Joystick::isButtonPressed(s_currWindowCtx->joystickId,
                                                    s_currWindowCtx->joystickMapping[action]);
     io.NavInputs[action] = isPressed ? 1.0f : 0.0f;
