diff --git a/CMakeLists.txt b/CMakeLists.txt
index f5c0bf34..579b95f8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -454,6 +454,9 @@ if(WARNINGS_AS_ERRORS)
        endif()
 endif()
 
+target_compile_options(datachannel PRIVATE /wd4996)
+target_compile_options(datachannel-static PRIVATE /wd4996)
+
 if(DATACHANNEL_STATIC_LINKAGE)
 install(TARGETS datachannel-static EXPORT LibDataChannelTargets
        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
