diff --git a/CMakeLists.txt b/CMakeLists.txt
index c8e2f1e..1dfcd62 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -471,6 +471,12 @@ if(WARNINGS_AS_ERRORS)
 	endif()
 endif()
 
+if(BUILD_SHARED_LIBS)
+    target_compile_options(datachannel PRIVATE /wd4996)
+elseif(BUILD_STATIC_LIBS)
+    target_compile_options(datachannel-static PRIVATE /wd4996)
+endif()
+
 if(BUILD_SHARED_LIBS)
 install(TARGETS datachannel EXPORT LibDataChannelTargets
 	RUNTIME DESTINATION bin
