diff --git a/msvc-scripts/sodium.props b/msvc-scripts/sodium.props
index 41e1e47..3126455 100644
--- a/msvc-scripts/sodium.props
+++ b/msvc-scripts/sodium.props
@@ -17,9 +17,9 @@
     <PreBuildEvent>
       <Message>Process .in files</Message>
     </PreBuildEvent>
-    <PostBuildEvent>
+    <!-- <PostBuildEvent>
       <Command>"$(SolutionDir)/test/default/wintest.bat" $(Configuration) $(Platform)</Command>
-    </PostBuildEvent>
+    </PostBuildEvent> -->
     <PostBuildEvent>
       <Message>Run the test suite</Message>
     </PostBuildEvent>
