diff --git a/rapidxml.hpp b/rapidxml.hpp
index ae91e08..853bc36 100644
--- a/rapidxml.hpp
+++ b/rapidxml.hpp
@@ -382,7 +382,7 @@ namespace rapidxml
     public:
 
         //! \cond internal
-        typedef void *(alloc_func)(std::size_t);       // Type of user-defined function used to allocate memory
-        typedef void (free_func)(void *);              // Type of user-defined function used to free memory
+        typedef void *  alloc_func(std::size_t);       // Type of user-defined function used to allocate memory
+        typedef void   free_func(void *);              // Type of user-defined function used to free memory
         //! \endcond
         
