diff --git a/share/cmake/Qt5Gui/Qt5GuiConfigExtras.cmake b/share/cmake/Qt5Gui/Qt5GuiConfigExtras.cmake
index c9c4dccb..461cb2ae 100644
--- a/share/cmake/Qt5Gui/Qt5GuiConfigExtras.cmake	
+++ b/share/cmake/Qt5Gui/Qt5GuiConfigExtras.cmake
@@ -17,12 +17,13 @@ macro(_qt5gui_find_extra_libs Name Libs LibDir IncDirs)
         string(REGEX REPLACE [^_A-Za-z0-9] _ _cmake_lib_name ${_libFile})
         if (NOT TARGET Qt5::Gui_${_cmake_lib_name} AND NOT _Qt5Gui_${_cmake_lib_name}_LIBRARY_DONE)
             if (IS_ABSOLUTE ${_lib})
-                set(Qt5Gui_${_cmake_lib_name}_LIBRARY ${_lib})
+                set(Qt5Gui_${_cmake_lib_name}_LIBRARY_RELEASE ${_lib})
+                _qt5_Gui_check_file_exists("${Qt5Gui_${_cmake_lib_name}_LIBRARY_RELEASE}")
             else()
-                find_library(Qt5Gui_${_cmake_lib_name}_LIBRARY ${_lib}
+                find_library(Qt5Gui_${_cmake_lib_name}_LIBRARY_RELEASE ${_lib}
                 )
             endif()
-            if (NOT Qt5Gui_${_cmake_lib_name}_LIBRARY)
+            if (NOT Qt5Gui_${_cmake_lib_name}_LIBRARY_RELEASE)
                 # The above find_library call doesn't work for finding
                 # libraries in Windows SDK paths outside of the proper
                 # environment, even if the libraries are present.  In other
@@ -36,17 +37,16 @@ macro(_qt5gui_find_extra_libs Name Libs LibDir IncDirs)
                 # in the interface of QtGui), so an effort is made to do so
                 # above with paths known to qmake.
                 set(_Qt5Gui_${_cmake_lib_name}_LIBRARY_DONE TRUE)
-                unset(Qt5Gui_${_cmake_lib_name}_LIBRARY CACHE)
+                unset(Qt5Gui_${_cmake_lib_name}_LIBRARY_RELEASE CACHE)
             else()
                 add_library(Qt5::Gui_${_cmake_lib_name} SHARED IMPORTED)
                 set_property(TARGET Qt5::Gui_${_cmake_lib_name} APPEND PROPERTY INTERFACE_INCLUDE_DIRECTORIES ${Qt5Gui_${Name}_INCLUDE_DIRS})
 
                 set_property(TARGET Qt5::Gui_${_cmake_lib_name} APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
-                _qt5_Gui_check_file_exists("${Qt5Gui_${_cmake_lib_name}_LIBRARY}")
-                set_property(TARGET Qt5::Gui_${_cmake_lib_name} PROPERTY IMPORTED_LOCATION_RELEASE "${Qt5Gui_${_cmake_lib_name}_LIBRARY}")
+                set_property(TARGET Qt5::Gui_${_cmake_lib_name} PROPERTY IMPORTED_LOCATION_RELEASE "${Qt5Gui_${_cmake_lib_name}_LIBRARY_RELEASE}")
 
-                set_property(TARGET Qt5::Gui_${_cmake_lib_name} PROPERTY IMPORTED_IMPLIB_RELEASE "${Qt5Gui_${_cmake_lib_name}_LIBRARY}")
-                unset(Qt5Gui_${_cmake_lib_name}_LIBRARY CACHE)
+                set_property(TARGET Qt5::Gui_${_cmake_lib_name} PROPERTY IMPORTED_IMPLIB_RELEASE "${Qt5Gui_${_cmake_lib_name}_LIBRARY_RELEASE}")
+                unset(Qt5Gui_${_cmake_lib_name}_LIBRARY_RELEASE CACHE)
 
                 find_library(Qt5Gui_${_cmake_lib_name}_LIBRARY_DEBUG ${_lib}d
                     PATHS "${LibDir}"
