diff --git a/cmake/ShogunFindLAPACK.cmake b/cmake/ShogunFindLAPACK.cmake
index fe3d6db9c2..882a1f4218 100644
--- a/cmake/ShogunFindLAPACK.cmake	
+++ b/cmake/ShogunFindLAPACK.cmake
@@ -6,7 +6,8 @@ IF (LAPACK_FOUND)
   SET(HAVE_LAPACK 1)
 
   # find out the type of Lapack/BLAS implementation we are dealing with
-  IF("${LAPACK_LIBRARIES}" MATCHES ".*/Accelerate.framework$")
+  message(STATUS "LAPACK_LIBRARIES:${LAPACK_LIBRARIES}")
+  IF("${LAPACK_LIBRARIES}" MATCHES ".*/Accelerate.framework(;|$)")
     # Accelerate.framework we found for LaPack/BLAS
     SET(HAVE_MVEC 1)
     SET(HAVE_CATLAS 1)
