diff --git a/CMakeLists.txt b/CMakeLists.txt
index 509e634..8b5a30b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -392,7 +392,7 @@ if(MSVC)
     set(CMAKE_CXX_FLAGS "\
         /EHsc /GF /nologo /std:c++latest /utf-8 /W4 /WX \
         /Zc:inline /Zc:rvalueCast /Zc:strictStrings \
-        /Fa$(IntDir)")
+        /Fa${CMAKE_SOURCE_DIR}/libs/dimcli")
     if(NOT MSVC_VERSION LESS 1910)
         # /permissive-  // reject non-conforming backward compatibility-isms
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /permissive-")
@@ -431,8 +431,8 @@ elseif(CMAKE_COMPILER_IS_GNUCXX)
     if(BUILD_COVERAGE)
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fprofile-arcs -ftest-coverage")
         set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} --coverage")
     endif()
-elseif(CMAKE_CXX_COMPILER_ID STREQUAL Clang)
+elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
     if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS "5")
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++1z")
     else()
@@ -533,7 +533,6 @@ endforeach()
 # test targets
 if(BUILD_TESTING)
     enable_testing()
-endif()
 file(GLOB allnames tests/*)
 foreach(var ${allnames})
     if(IS_DIRECTORY "${var}")
@@ -548,6 +547,7 @@ foreach(var ${allnames})
         endif()
     endif()
 endforeach()
+endif()
 
 # update deps file
 update_deps_file(${deps})
