diff --git a/CMakeLists.txt b/CMakeLists.txt
index 335b1ad..e2e8a4d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,8 +1,9 @@
 cmake_minimum_required(VERSION 2.8.0 FATAL_ERROR)
 
 set(PROJECT_NAME wavelib)
-project(${PROJECT_NAME} CXX C)
+project(${PROJECT_NAME} C)
 
+include(GNUInstallDirs)
 
 # src root path
 set(WAVELIB_SRC_ROOT ${PROJECT_SOURCE_DIR} CACHE PATH "Wavelib source root")
@@ -41,6 +42,7 @@ include_directories(${COMMON_BIN_PATH})
 if(BUILD_UT)
     include(CTest)
     enable_testing()
+    enable_language(CXX)
     add_subdirectory(unitTests)
 endif()
 
@@ -48,4 +50,6 @@ add_subdirectory(src)
 add_subdirectory(auxiliary)
 add_subdirectory(test) 
 
-install(DIRECTORY ${WAVELIB_SRC_ROOT}/include/ DESTINATION include FILES_MATCHING PATTERN "*.h")
+install(DIRECTORY ${WAVELIB_SRC_ROOT}/header/
+    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+    FILES_MATCHING PATTERN "*.h")
diff --git a/auxiliary/CMakeLists.txt b/auxiliary/CMakeLists.txt
index 51c836d..25823e6 100644
--- a/auxiliary/CMakeLists.txt
+++ b/auxiliary/CMakeLists.txt
@@ -12,5 +12,9 @@ target_link_libraries(wauxlib wavelib)
 
 set_property(TARGET wauxlib PROPERTY FOLDER "lib")
 
-target_include_directories(wauxlib PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+target_include_directories(wauxlib PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
 
+install(TARGETS wauxlib
+    EXPORT wavelib-targets
+    ARCHIVE DESTINATION lib
+    INCLUDES DESTINATION include)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index dc9592a..0bb6940 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -27,5 +27,14 @@ add_library(wavelib STATIC ${SOURCE_FILES} ${HEADER_FILES})
         
 set_property(TARGET wavelib PROPERTY FOLDER "lib")
 
-target_include_directories(wavelib PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}../header)
+target_include_directories(wavelib PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}../header)
 
+install(TARGETS wavelib
+    EXPORT wavelib-targets
+    ARCHIVE DESTINATION lib
+    INCLUDES DESTINATION include)
+
+install(EXPORT wavelib-targets
+    FILE wavelib-targets.cmake
+    NAMESPACE wavelib::
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/wavelib)
