diff --git a/CMakeLists.txt b/CMakeLists.txt
index e440b4b..5ad91ba 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -725,7 +725,7 @@ IF(USE_HDF5 OR ENABLE_NETCDF_4)
   #Check to see if H5Z_SZIP exists in HDF5_Libraries. If so, we must use szip.
   CHECK_LIBRARY_EXISTS(${HDF5_C_LIBRARY_hdf5} H5Z_SZIP "" USE_SZIP)
   IF(USE_SZIP)
-    FIND_LIBRARY(SZIP NAMES szip sz)
+    GET_PROPERTY(SZIP TARGET ${HDF5_C_LIBRARY_hdf5} PROPERTY INTERFACE_LINK_LIBRARIES)
     IF(SZIP)
       SET(HAVE_H5Z_SZIP 1)
       SET(SZIP_LIBRARY ${SZIP})
