diff --git a/CMakeLists.txt b/CMakeLists.txt
index 35c694f..e2c03e5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -839,7 +839,7 @@ ENDIF()
 
 # Check for the math library so it can be explicitly linked.
 IF(NOT WIN32)
-  FIND_LIBRARY(HAVE_LIBM NAMES math m libm)
+  FIND_LIBRARY(HAVE_LIBM NAMES m libm math)
   MESSAGE(STATUS "Found Math library: ${HAVE_LIBM}")
   IF(NOT HAVE_LIBM)
     MESSAGE(FATAL_ERROR "Unable to find the math library.")
diff --git a/netCDFConfig.cmake.in b/netCDFConfig.cmake.in
index 102828f..c5052cc 100644
--- a/netCDFConfig.cmake.in
+++ b/netCDFConfig.cmake.in
@@ -16,6 +16,7 @@ include(CMakeFindDependencyMacro)
 find_dependency(ZLIB)
 find_dependency(HDF5 CONFIG)
 find_dependency(CURL CONFIG)
+find_dependency(embree CONFIG)
 include("${CMAKE_CURRENT_LIST_DIR}/netCDFTargets.cmake")
 
 # Compiling Options
