diff --git a/CMakeLists.txt b/CMakeLists.txt
index 96fbaf979..e5089ad4f 100644
--- a/CMakeLists.txt	
+++ b/CMakeLists.txt
@@ -261,7 +261,7 @@ ENDIF()
 INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/include)
 INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/oc2)
 INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/libsrc)
-SET(CMAKE_REQUIRED_INCLUDES ${CMAKE_SOURCE_DIR}/libsrc)
+list(APPEND CMAKE_REQUIRED_INCLUDES ${CMAKE_SOURCE_DIR}/libsrc)
 
 ################################
 # End Compiler Configuration
@@ -726,7 +726,7 @@ IF(USE_HDF5 OR ENABLE_NETCDF_4)
     FIND_LIBRARY(SZIP NAMES szip sz)
     IF(SZIP)
       SET(HAVE_H5Z_SZIP 1)
-      SET(SZIP_LIBRARY ${SZIP})
+      #SET(SZIP_LIBRARY ${SZIP})
       SET(CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES} ${SZIP})
     ELSE()
       MESSAGE(FATAL_ERROR "HDF5 Requires SZIP, but cannot find libszip or libsz.")
diff --git a/liblib/CMakeLists.txt b/liblib/CMakeLists.txt
index 6a81a4801..74173c1d3 100644
--- a/liblib/CMakeLists.txt	
+++ b/liblib/CMakeLists.txt
@@ -61,7 +61,7 @@ SET(TLL_LIBS "")
 SET(TLL_LIBS ${TLL_LIBS} ${HAVE_LIBM} ${ZLIB_LIBRARY})
 
 # Add extra dependencies specified via NC_EXTRA_DEPS
-SET(TLL_LIBS ${TLL_LIBS} ${EXTRA_DEPS})
+#SET(TLL_LIBS ${TLL_LIBS} ${EXTRA_DEPS})
 
 IF(HAVE_LIBDL)
   SET(TLL_LIBS ${LIBDL} ${TLL_LIBS})
@@ -100,7 +102,7 @@ IF(ENABLE_PNETCDF AND PNETCDF)
 ENDIF()
 
 IF(TLL_LIBS)
-  LIST(REMOVE_DUPLICATES TLL_LIBS)
+  #LIST(REMOVE_DUPLICATES TLL_LIBS) //removes deuplicate debug/optimize
 ENDIF()
 
 TARGET_LINK_LIBRARIES(netcdf ${TLL_LIBS})
