diff --git a/wpimath/src/main/native/include/frc/system/LinearSystem.h b/wpimath/src/main/native/include/frc/system/LinearSystem.h
index d2b0213..382a0a1 100644
--- a/wpimath/src/main/native/include/frc/system/LinearSystem.h
+++ b/wpimath/src/main/native/include/frc/system/LinearSystem.h
@@ -229,8 +229,8 @@ class LinearSystem {
     }

     return LinearSystem<States, Inputs, sizeof...(OutputIndices)>{
-        m_A, m_B, m_C(outputIndicesArray, Eigen::placeholders::all),
-        m_D(outputIndicesArray, Eigen::placeholders::all)};
+        m_A, m_B, m_C(outputIndicesArray, Eigen::all),
+        m_D(outputIndicesArray, Eigen::all)};
   }

  private:
