diff --git a/src/MagnumPlugins/OpenGexImporter/OpenDdl/OpenDdl.cpp b/src/MagnumPlugins/OpenGexImporter/OpenDdl/OpenDdl.cpp
index c94ee98..843d1ec 100644
--- a/src/MagnumPlugins/OpenGexImporter/OpenDdl/OpenDdl.cpp
+++ b/src/MagnumPlugins/OpenGexImporter/OpenDdl/OpenDdl.cpp
@@ -25,6 +25,7 @@
 
 #include <algorithm>
 #include <tuple>
+#include <memory>
 #include <Corrade/Utility/Debug.h>
 
 #include "MagnumPlugins/OpenGexImporter/OpenDdl/Document.h"
