From 4d59a1eafc7a5828a0dedc92a713b3394d5f64fc Mon Sep 17 00:00:00 2001
From: Squareys <squareys@googlemail.com>
Date: Wed, 6 Nov 2019 18:45:17 +0100
Subject: [PATCH] fix vcpkg build

---
 src/MagnumPlugins/BasisImageConverter/BasisImageConverter.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/MagnumPlugins/BasisImageConverter/BasisImageConverter.cpp b/src/MagnumPlugins/BasisImageConverter/BasisImageConverter.cpp
index db3a39cb..d5c80b11 100644
--- a/src/MagnumPlugins/BasisImageConverter/BasisImageConverter.cpp
+++ b/src/MagnumPlugins/BasisImageConverter/BasisImageConverter.cpp
@@ -106,7 +106,9 @@ Containers::Array<char> BasisImageConverter::doExportToData(const ImageView2D& i
 
     /* Hierarchical virtual selector codebook options */
     PARAM_CONFIG_FIX_NAME(global_sel_pal, bool, "global_selector_palette");
-    PARAM_CONFIG_FIX_NAME(auto_global_sel_pal, bool, "auto_global_selector_palette");
+    /* vcpkg version is begind 54304fa01443280bbaa9854f16c9f1756b17b40b, which changes name
+     * to m_auto_global_sel_pal */
+    params.m_no_auto_global_sel_pal = !configuration().value<bool>("auto_global_selector_palette");
     PARAM_CONFIG_FIX_NAME(no_hybrid_sel_cb, bool, "no_hybrid_selector_codebook");
     PARAM_CONFIG_FIX_NAME(global_pal_bits, int, "global_palette_bits");
     PARAM_CONFIG_FIX_NAME(global_mod_bits, int, "global_modifier_bits");
-- 
2.17.0.windows.1

