diff --git a/src/external/dr/dr_mp3.h b/src/external/dr/dr_mp3.h
index 8191da768..6c77cf1d0 100644
--- a/src/external/dr/dr_mp3.h
+++ b/src/external/dr/dr_mp3.h
@@ -534,7 +534,7 @@ end:
     return g_have_simd - 1;
 #endif
 }
-#elif defined(__ARM_NEON) || defined(__aarch64__)
+#elif defined(__ARM_NEON) || defined(__aarch64__) || defined(_M_ARM64)
 #include <arm_neon.h>
 #define DRMP3_HAVE_SIMD 1
 #define DRMP3_VSTORE vst1q_f32
