diff --git a/CMakeLists.txt b/CMakeLists.txt
index bcfe91d8..cc2b66d7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -306,7 +306,7 @@ endif()
 if(CMAKE_C_COMPILER_ID MATCHES "AppleClang|Clang|GNU")
   list(APPEND mi_cflags -Wno-unknown-pragmas -fvisibility=hidden)
   if(NOT MI_USE_CXX)
-    list(APPEND mi_cflags -Wstrict-prototypes)
+    list(APPEND mi_cflags -Wstrict-prototypes -std=c99)
   endif()
   if(CMAKE_C_COMPILER_ID MATCHES "AppleClang|Clang")
     list(APPEND mi_cflags -Wno-static-in-inline)
