From 7d7fae33029d359423b26b60ea12174028ebcf3a Mon Sep 17 00:00:00 2001
From: Dmitry Baryshev <dmitrymq@gmail.com>
Date: Wed, 2 Jun 2021 21:01:59 +0300
Subject: [PATCH] Enable CMake configs in VCPKG mode

---
 CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d773025..dd80d22 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -564,7 +564,8 @@ if(NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL)
         LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
     )
 
-    if (BUILD_SHARED_LIBS)
+    # Enable CMake configs in VCPKG mode
+    if (BUILD_SHARED_LIBS OR VCPKG_TARGET_TRIPLET)
         install(EXPORT ${PROJECT_NAME}-config
                 DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
 
-- 
2.22.0.windows.1

