diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index f410480..9a95d1f 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -134,9 +134,12 @@ endif()
 
 # Thread support in Scotch
 if(THREADS)
-  find_package(Threads)
-  if(Threads_FOUND)
+set(USE_PTHREAD OFF)
+if(NOT WIN32)
+  find_package(Threads REQUIRED)
+  if(CMAKE_USE_PTHREADS_INIT)
     add_definitions(-DCOMMON_PTHREAD -DSCOTCH_PTHREAD)
+    set(USE_PTHREAD ON)
   endif()
   if(CMAKE_USE_WIN32_THREADS_INIT)
     add_definitions(-DCOMMON_THREAD_WIN32)
@@ -145,6 +148,12 @@ if(THREADS)
   if(PTHREAD_AFFINITY_LINUX_OK)
     add_definitions(-DCOMMON_PTHREAD_AFFINITY_LINUX)
   endif()
+else()
+  find_package(PThreads4W REQUIRED)
+  add_definitions(-DCOMMON_PTHREAD -DSCOTCH_PTHREAD)
+  link_libraries(PThreads4W::PThreads4W)
+  set(USE_PTHREAD ON)
+endif()
 endif()
 
 # Detect and remove GCC _FORTIFY_SOURCE
diff --git a/src/libscotch/CMakeLists.txt b/src/libscotch/CMakeLists.txt
index c142c22..15ed0d8 100644
--- a/src/libscotch/CMakeLists.txt
+++ b/src/libscotch/CMakeLists.txt
@@ -512,6 +512,22 @@ set(SCOTCH_C_SOURCES
   wgraph_part_zr.h
   wgraph_store.c)
 
+add_library(scotcherr library_error.c)
+target_include_directories(scotcherr PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+  $<BUILD_INTERFACE:${GENERATED_INCLUDE_DIR}>
+  $<INSTALL_INTERFACE:include>)
+
+add_dependencies(scotcherr scotch_h)
+
+add_library(scotcherrexit library_error_exit.c)
+target_include_directories(scotcherrexit PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+  $<BUILD_INTERFACE:${GENERATED_INCLUDE_DIR}>
+  $<INSTALL_INTERFACE:include>)
+
+add_dependencies(scotcherrexit scotch_h)
+
 add_library(scotch
   ${SCOTCH_C_SOURCES})
 add_library(SCOTCH::scotch ALIAS scotch)
@@ -573,10 +589,14 @@ if(LIBLZMA_FOUND)
 endif()
 
 # Add thread library
-if(Threads_FOUND)
-  target_compile_definitions(scotch PRIVATE COMMON_PTHREAD SCOTCH_PTHREAD)
-  target_link_libraries(scotch PRIVATE Threads::Threads ${THREADS_PTHREADS_WIN32_LIBRARY})
-endif()
+if(USE_PTHREAD)
+  target_compile_definitions(scotch PUBLIC COMMON_PTHREAD SCOTCH_PTHREAD)
+  if(NOT WIN32)
+    target_link_libraries(scotch PUBLIC Threads::Threads)
+  else()
+    target_link_libraries(scotch PUBLIC PThreads4W::PThreads4W)
+  endif()
+endif(USE_PTHREAD)
 
 # Include files
 add_dependencies(scotch scotch_h scotchf_h)
@@ -589,27 +609,7 @@ if(CMAKE_BUILD_TYPE STREQUAL Debug)
   target_compile_definitions(scotch PRIVATE SCOTCH_DEBUG_ALL)
 endif()
 
-add_library(scotcherr library_error.c)
-add_library(SCOTCH::scotcherr ALIAS scotcherr)
-set_target_properties(scotcherr PROPERTIES VERSION ${SCOTCH_VERSION_LONG}
-  SOVERSION ${SCOTCH_VERSION}.${SCOTCH_RELEASE})
-target_include_directories(scotcherr PUBLIC
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
-  $<BUILD_INTERFACE:${GENERATED_INCLUDE_DIR}>
-  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
-
-add_dependencies(scotcherr scotch_h)
-
-add_library(scotcherrexit library_error_exit.c)
-add_library(SCOTCH::scotcherrexit ALIAS scotcherrexit)
-set_target_properties(scotcherrexit PROPERTIES VERSION ${SCOTCH_VERSION_LONG}
-  SOVERSION ${SCOTCH_VERSION}.${SCOTCH_RELEASE})
-target_include_directories(scotcherrexit PUBLIC
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
-  $<BUILD_INTERFACE:${GENERATED_INCLUDE_DIR}>
-  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
-
-add_dependencies(scotcherrexit scotch_h)
+target_link_libraries(scotch PRIVATE scotcherr)
 
 #################
 #  libPTScotch  #
@@ -854,10 +854,14 @@ if(BUILD_PTSCOTCH)
     set_target_properties(ptscotch PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
   endif()
 
-  if(Threads_FOUND)
-    target_compile_definitions(ptscotch PRIVATE SCOTCH_PTHREAD COMMON_PTHREAD)
-    target_link_libraries(ptscotch PRIVATE Threads::Threads ${THREADS_PTHREADS_WIN32_LIBRARY})
-  endif()
+  if(USE_PTHREAD)
+    target_compile_definitions(ptscotch PUBLIC COMMON_PTHREAD SCOTCH_PTHREAD)
+    if(NOT WIN32)
+      target_link_libraries(ptscotch PUBLIC Threads::Threads)
+    else()
+      target_link_libraries(ptscotch PUBLIC PThreads4W::PThreads4W)
+    endif()
+  endif(USE_PTHREAD)
 
   add_library(ptscotcherr library_error.c)
   add_library(SCOTCH::ptscotcherr ALIAS ptscotcherr)
diff --git a/src/libscotch/common_file_decompress.c b/src/libscotch/common_file_decompress.c
index 6e80222..3048d90 100644
--- a/src/libscotch/common_file_decompress.c
+++ b/src/libscotch/common_file_decompress.c
@@ -373,7 +373,11 @@ FileCompress * const        compptr)
   decodat.avail_out = FILECOMPRESSDATASIZE;
   do {
     if ((decodat.avail_in == 0) && (deacval == LZMA_RUN)) {
+#ifdef _MSC_VER
+      size_t              bytenbr; //According to MSDN fread returns size_t
+#else
       ssize_t             bytenbr;
+#endif
 
       bytenbr = fread (compptr->bufftab, 1, FILECOMPRESSDATASIZE, compptr->oustptr); /* Read from pipe */
       if (ferror (compptr->oustptr)) {
diff --git a/src/libscotch/common_thread.h b/src/libscotch/common_thread.h
index a8d3284..2e9ae28 100644
--- a/src/libscotch/common_thread.h
+++ b/src/libscotch/common_thread.h
@@ -82,6 +82,8 @@ struct ThreadContext_ {
     int                         dummval;          /*+ Dummy value if no affinity enabled  +*/
 #ifdef COMMON_PTHREAD_AFFINITY_LINUX
     cpu_set_t                   cpusdat;          /*+ Original thread mask of main thread +*/
+#else
+    void*                       dummy;
 #endif /* COMMON_PTHREAD_AFFINITY_LINUX */
   }                             savedat;          /*+ Save area for affinity mask         +*/
 #endif /* COMMON_PTHREAD */
diff --git a/src/libscotch/context.c b/src/libscotch/context.c
index acda708..ac0aa58 100644
--- a/src/libscotch/context.c
+++ b/src/libscotch/context.c
@@ -70,8 +70,11 @@ static struct ContextValuesData_ {
 #else /* ((defined SCOTCH_DETERMINISTIC) || (defined COMMON_RANDOM_FIXED_SEED)) */
                               0
 #endif /* ((defined SCOTCH_DETERMINISTIC) || (defined COMMON_RANDOM_FIXED_SEED)) */
-  }, { -1.0 } };                                  /* Temporary hack: dummy value since ISO C does not accept zero-sized arrays */
-
+  }
+#ifndef _MSC_VER
+  , { }
+#endif
+  };
 /***********************************/
 /*                                 */
 /* These routines handle contexts. */
@@ -91,8 +94,8 @@ Context * const             contptr)
   int                 o;
 
   o = contextValuesInit (contptr, &contextvaluesdat, sizeof (contextvaluesdat),
-                         CONTEXTOPTIONNUMNBR, (byte *) &contextvaluesdat.vinttab - (byte *) &contextvaluesdat,
-                         CONTEXTOPTIONDBLNBR, (byte *) &contextvaluesdat.vdbltab - (byte *) &contextvaluesdat);
+                             CONTEXTOPTIONNUMNBR, (char *) &contextvaluesdat.vinttab - (char *) &contextvaluesdat,
+                             CONTEXTOPTIONDBLNBR, (char *) &contextvaluesdat.vdbltab - (char *) &contextvaluesdat);
 
   if (o == 0) {                                   /* If values array allocated, update it with existing environment variables */
     INT                 deteval;
diff --git a/src/libscotch/parser_ll.l b/src/libscotch/parser_ll.l
index 7b690e0..823623e 100644
--- a/src/libscotch/parser_ll.l
+++ b/src/libscotch/parser_ll.l
@@ -83,6 +83,9 @@
 #undef INTEGER                                    /* In case someone defined them */
 #undef DOUBLE
 
+#ifdef _MSC_VER
+#define YY_NO_UNISTD_H
+#endif
 #include "parser.h"
 #include "parser_yy.h"
 #include "parser_ly.h"                            /*+ Definitions produced by yacc +*/
diff --git a/src/scotch/CMakeLists.txt b/src/scotch/CMakeLists.txt
index 243e21d..7d67c01 100644
--- a/src/scotch/CMakeLists.txt
+++ b/src/scotch/CMakeLists.txt
@@ -67,10 +67,15 @@ function(add_scotch_exe)
   add_dependencies(${file_we} scotch_h)
   target_include_directories(${file_we} PRIVATE ${GENERATED_INCLUDE_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
   target_compile_definitions(${file_we} PUBLIC "SCOTCH_CODENAME=\"${SCOTCH_CODENAME}\"")
-  if(Threads_FOUND)
-    target_compile_definitions(${file_we} PUBLIC SCOTCH_PTHREAD COMMON_PTHREAD)
-    target_link_libraries(${file_we} PUBLIC Threads::Threads ${THREADS_PTHREADS_WIN32_LIBRARY})
-  endif(Threads_FOUND)
+
+  if(USE_PTHREAD)
+    target_compile_definitions(${file_we} PUBLIC COMMON_PTHREAD SCOTCH_PTHREAD)
+    if(NOT WIN32)
+      target_link_libraries(${file_we} PUBLIC Threads::Threads)
+    else()
+      target_link_libraries(${file_we} PUBLIC PThreads4W::PThreads4W)
+    endif()
+  endif(USE_PTHREAD)
   target_link_libraries(${file_we} PUBLIC scotch scotcherrexit)
 endfunction(add_scotch_exe)
 
@@ -145,10 +150,14 @@ if(BUILD_PTSCOTCH)
       ${GENERATED_INCLUDE_DIR}/ptscotch.h)
     add_dependencies(${file_we} ptscotch_h)
     target_compile_definitions(${file_we} PUBLIC "SCOTCH_CODENAME=\"${SCOTCH_CODENAME}\"")
-    if(Threads_FOUND)
-      target_compile_definitions(${file_we} PUBLIC SCOTCH_PTHREAD COMMON_PTHREAD)
-      target_link_libraries(${file_we} PUBLIC Threads::Threads ${THREADS_PTHREADS_WIN32_LIBRARY})
-    endif(Threads_FOUND)
+    if(USE_PTHREAD)
+      target_compile_definitions(${file_we} PUBLIC COMMON_PTHREAD SCOTCH_PTHREAD)
+      if(NOT WIN32)
+        target_link_libraries(${file_we} PUBLIC Threads::Threads)
+      else()
+        target_link_libraries(${file_we} PUBLIC PThreads4W::PThreads4W)
+      endif()
+    endif(USE_PTHREAD)
     target_include_directories(${file_we} PRIVATE ${GENERATED_INCLUDE_DIR})
     target_link_libraries(${file_we} PUBLIC scotch ptscotch ptscotcherrexit)
   endfunction(add_ptscotch_exe)
