diff --git a/include/boost/process/detail/traits/wchar_t.hpp b/include/boost/process/detail/traits/wchar_t.hpp
index 812a92c..98026d3 100644
--- a/include/boost/process/detail/traits/wchar_t.hpp
+++ b/include/boost/process/detail/traits/wchar_t.hpp
@@ -12,6 +12,8 @@
 #include <boost/process/detail/traits/env.hpp>
 #include <boost/process/locale.hpp>
 
+#include <algorithm>
+
 namespace boost { namespace process { namespace detail {
 
 //template
