diff --git a/src/3rdparty/chromium/build/config/linux/pkg_config.gni b/src/3rdparty/chromium/build/config/linux/pkg_config.gni
index 0d873903c..09bf2e874 100644
--- a/src/3rdparty/chromium/build/config/linux/pkg_config.gni
+++ b/src/3rdparty/chromium/build/config/linux/pkg_config.gni
@@ -54,7 +54,7 @@ pkg_config_script = "//build/config/linux/pkg-config.py"
 pkg_config_args = []
 
 common_pkg_config_args = []
-if (sysroot != "") {
+if (sysroot != "" && !is_android) {
   # Pass the sysroot if we're using one (it requires the CPU arch also).
   common_pkg_config_args += [
     "-s",
