diff --git a/cmake/SalomeMacros.cmake b/cmake/SalomeMacros.cmake
index bb38859ee9..3b7662dcae 100644
--- a/cmake/SalomeMacros.cmake
+++ b/cmake/SalomeMacros.cmake
@@ -23,9 +23,6 @@
 # Set-up global policies
 #----------------------------------------------------------------------------
 CMAKE_POLICY(SET CMP0003 NEW)   # Ensure proper linker behavior
-IF(WIN32)
-  CMAKE_POLICY(SET CMP0020 OLD) # Disable automatic linking to qtmain.lib
-ENDIF(WIN32)
 CMAKE_POLICY(SET CMP0053 NEW)   # For correct Qt 5 detection procedure
 IF(CMAKE_VERSION VERSION_GREATER_EQUAL 3.12)
   CMAKE_POLICY(SET CMP0074 NEW) # Use ROOT variables when detecting packages
