diff --git a/cfp/src/CMakeLists.txt b/cfp/src/CMakeLists.txt
index 386698c..979be58 100644
--- a/cfp/src/CMakeLists.txt
+++ b/cfp/src/CMakeLists.txt
@@ -33,6 +33,8 @@ set_property(TARGET cfp PROPERTY SOVERSION ${ZFP_VERSION_MAJOR})
 set_property(TARGET cfp PROPERTY OUTPUT_NAME ${ZFP_LIBRARY_PREFIX}cfp)
 
 install(TARGETS cfp EXPORT cfp-targets
-  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+  RUNTIME DESTINATION bin
+  LIBRARY DESTINATION lib
+  ARCHIVE DESTINATION lib)
+  
+install(DIRECTORY ${ZFP_SOURCE_DIR}/include DESTINATION include)
