diff --git a/CMakeLists.txt b/CMakeLists.txt
index 69cdb8b..4c985fb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -149,10 +149,10 @@ ENABLE_TESTING()
 
 # Targets
 ADD_SUBDIRECTORY(lib)
-ADD_SUBDIRECTORY(man)
-ADD_SUBDIRECTORY(src)
-ADD_SUBDIRECTORY(regress)
-ADD_SUBDIRECTORY(examples)
+#ADD_SUBDIRECTORY(man)
+#ADD_SUBDIRECTORY(src)
+#ADD_SUBDIRECTORY(regress)
+#ADD_SUBDIRECTORY(examples)
 
 # pkgconfig file
 SET(prefix ${CMAKE_INSTALL_PREFIX})
