diff --git a/CMakeLists.txt b/CMakeLists.txt
index a2cb47d..57a1cf5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -81,14 +81,8 @@ install(FILES ${ads_INSTALL_INCLUDE}
     DESTINATION include 
     COMPONENT headers
 )
-install(FILES 
-    "${CMAKE_CURRENT_SOURCE_DIR}/LICENSE"
-    "${CMAKE_CURRENT_SOURCE_DIR}/gnu-lgpl-v2.1.md"
-    DESTINATION license 
-    COMPONENT license
-)
 install(TARGETS qtadvanceddocking
     EXPORT qt-advanced-docking-systemConfig
     RUNTIME DESTINATION bin
     LIBRARY DESTINATION lib
     ARCHIVE DESTINATION lib
