diff --git a/src/cpp/CMakeLists.txt b/src/cpp/CMakeLists.txt
index eb84c8a..e6c8ae0 100644
--- a/src/cpp/CMakeLists.txt
+++ b/src/cpp/CMakeLists.txt
@@ -2,7 +2,13 @@
 
 add_definitions(-D_FLANN_VERSION=${FLANN_VERSION})
 
-configure_file(${CMAKE_CURRENT_SOURCE_DIR}/flann/config.h.in ${CMAKE_CURRENT_SOURCE_DIR}/flann/config.h)
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/flann/config.h.in ${CMAKE_CURRENT_BINARY_DIR}/flann/config.h)
+include_directories(${CMAKE_CURRENT_BINARY_DIR}/flann)
+
+install (
+    FILES ${CMAKE_CURRENT_BINARY_DIR}/flann/config.h
+    DESTINATION include/flann
+)
 
 file(GLOB_RECURSE C_SOURCES flann.cpp lz4.c lz4hc.c)
 file(GLOB_RECURSE CPP_SOURCES flann_cpp.cpp lz4.c lz4hc.c)
