diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9d06cb4..593bbfd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,6 +43,8 @@ endif (NOT USE_MINIZ)
 
 set(EXAMPLE_CFLAGS "-std=c++11 -pedantic -Werror -Wall -g")
 #include_directories("${PROJECT_SOURCE_DIR}/include")
+include_directories("${PROJECT_BINARY_DIR}")
+include_directories("${PROJECT_SOURCE_DIR}/src")
 
 add_library(tmxparser_static STATIC ${SOURCES} ${HEADERS})
 set_target_properties(tmxparser_static PROPERTIES
@@ -67,8 +69,6 @@ add_executable(run_tests test/test.cpp)
 set_target_properties(run_tests PROPERTIES
   COMPILE_FLAGS ${EXAMPLE_CFLAGS})
 target_link_libraries(run_tests tmxparser ${TINYXML2_LIBRARIES})
-include_directories("${PROJECT_BINARY_DIR}")
-include_directories("${PROJECT_SOURCE_DIR}/src")
 
 install(FILES ${HEADERS} DESTINATION include/tmxparser)
 install(TARGETS tmxparser tmxparser_static DESTINATION lib)
