diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2be8ae8..cda1957 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -194,6 +194,13 @@ if (BUILD_UVW_LIBS)
     )
 endif()
 
+find_package(libuv CONFIG REQUIRED)
+if (TARGET libuv::uv)
+    target_link_libraries(uvw PRIVATE libuv::uv)
+else()
+    target_link_libraries(uvw PRIVATE libuv::uv_a)
+endif()
+
 install(
     EXPORT uvwConfig 
     NAMESPACE uvw:: 
