diff --git a/src/windows.hpp b/src/windows.hpp
index 24cce83d11..c6d625188d 100644
--- a/src/windows.hpp
+++ b/src/windows.hpp
@@ -90,7 +90,6 @@ static inline int poll (struct pollfd *pfd, unsigned long nfds, int timeout)
 #ifndef AI_NUMERICSERV
 #define AI_NUMERICSERV 0x0400
 #endif
-#endif
 
 //  In MSVC prior to v14, snprintf is not available
 //  The closest implementation is the _snprintf_s function
@@ -108,3 +107,5 @@ struct sockaddr_un
     char sun_path[108];        /* pathname */
 };
 #endif
+
+#endif
