diff --git a/source/config/mh-msys-msvc b/source/config/mh-msys-msvc
index 6f50798eb..172543fd3 100644
--- a/source/config/mh-msys-msvc
+++ b/source/config/mh-msys-msvc
@@ -122,7 +122,7 @@ LIBCTESTFW=	$(top_builddir)/tools/ctestfw/$(LIBICU)$(CTESTFW_STUBNAME)$(ICULIBSU
 LIBICUTOOLUTIL=	$(LIBDIR)/$(LIBICU)$(TOOLUTIL_STUBNAME)$(ICULIBSUFFIX).lib
 
 ## These are the library specific LDFLAGS
-LDFLAGSICUDT+=	-NOENTRY# The NOENTRY option is required for creating a resource-only DLL.
+## LDFLAGSICUDT+=	-NOENTRY# The NOENTRY option is required for creating a resource-only DLL.
 LDFLAGSICUUC=# Unused for now.
 LDFLAGSICUI18N=# Unused for now.
 LDFLAGSICUIO=# Unused for now.
diff --git a/source/tools/pkgdata/pkgdata.cpp b/source/tools/pkgdata/pkgdata.cpp
index c2ac112f6..30a0ea65f 100644
--- a/source/tools/pkgdata/pkgdata.cpp
+++ b/source/tools/pkgdata/pkgdata.cpp
@@ -1805,7 +1805,7 @@ static int32_t pkg_createWithoutAssemblyCode(UPKGOptions *o, const char *targetD
 
 #ifdef WINDOWS_WITH_MSVC
 #define LINK_CMD "link.exe /nologo /release /out:"
-#define LINK_FLAGS "/NXCOMPAT /DYNAMICBASE /DLL /NOENTRY /MANIFEST:NO /implib:"
+#define LINK_FLAGS "/NXCOMPAT /DYNAMICBASE /DLL /MANIFEST:NO /implib:"
 
 #define LINK_EXTRA_UWP_FLAGS "/APPCONTAINER "
 #define LINK_EXTRA_UWP_FLAGS_X86_ONLY "/SAFESEH "
