diff --git a/ace/config-win32-common.h b/ace/config-win32-common.h
index 56567eead08..d219bd5f1ae 100644
--- a/ace/config-win32-common.h
+++ b/ace/config-win32-common.h
@@ -39,12 +39,14 @@
 #if defined (_WIN64) || defined (WIN64)
 #  define ACE_WIN64
 
+
 // MPC template adds _AMD64_ but user projects not generated using MPC
-// may want to use _AMD64_ as well. Ensure it's there in all cases.
+// may want to use _AMD64_ as well. Ensure it's there in all non ARM cases.
+#if !defined(_ARM64_) && !defined(_ARM_)
 #  ifndef _AMD64_
 #    define _AMD64_
 #  endif
-
+#endif
 // Use 64-bit file offsets by default in the WIN64 case, similar to
 // what 64-bit UNIX systems do.
 //
