diff --git a/MPC/templates/vs2019platforms.mpt b/MPC/templates/vs2019platforms.mpt
index 4f5dbfa8..1f4671b1 100644
--- a/MPC/templates/vs2019platforms.mpt
+++ b/MPC/templates/vs2019platforms.mpt
@@ -3,9 +3,9 @@
 // Any of these platforms can set using value_template platforms=
 // when generating solutions/projects.
 //
-//platforms = Win32 x64
+//platforms = Win32 x64 ARM ARM64
 
-platforms = Win32 x64
+platforms = Win32 x64 ARM ARM64
 default_platform = Win32
 PlatformToolset = v142
 WindowsTargetPlatformVersion = 10.0.17763.0
@@ -22,3 +22,18 @@ x64 {
   link_options     = /machine:AMD64
 }
 
+ARM {
+  output_subdir    = ARM
+  cpu_defines      = _ARM_
+  platform_defines = _ARM
+  TargetMachine    = "0"             // We let VC++ figure it out for us.
+  link_options     = /machine:ARM
+}
+
+ARM64 {
+  output_subdir    = ARM64
+  cpu_defines      = _ARM64_
+  platform_defines = _ARM64
+  TargetMachine    = "0"             // We let VC++ figure it out for us.
+  link_options     = /machine:ARM64
+}
\ No newline at end of file
