diff --git a/ace/Process_Manager.cpp b/ace/Process_Manager.cpp
index ee0418b8fca..1eeae31d2a8 100644
--- a/ace/Process_Manager.cpp
+++ b/ace/Process_Manager.cpp
@@ -129,12 +129,12 @@ ACE_Process_Manager::instance (void)
           ACE_Object_Manager::at_exit (ACE_Process_Manager::instance_,
                                        ACE_Process_Manager_cleanup,
                                        0,
-                                       typeid (*ACE_Process_Manager::instance_).name ());
+                                       typeid (ACE_Process_Manager).name ());
 #else
           ACE_Object_Manager::at_exit (ACE_Process_Manager::instance_,
                                        ACE_Process_Manager::cleanup,
                                        0,
-                                       typeid (*ACE_Process_Manager::instance_).name ());
+                                       typeid (ACE_Process_Manager).name ());
 #endif /* ACE_HAS_SIG_C_FUNC */

         }
@@ -162,12 +162,12 @@ ACE_Process_Manager::instance (ACE_Process_Manager *tm)
   ACE_Object_Manager::at_exit (ACE_Process_Manager::instance_,
                                 ACE_Process_Manager_cleanup,
                                 0,
-                                typeid (*ACE_Process_Manager::instance_).name ());
+                                typeid (ACE_Process_Manager).name ());
 #else
   ACE_Object_Manager::at_exit (ACE_Process_Manager::instance_,
                                 ACE_Process_Manager::cleanup,
                                 0,
-                                typeid (*ACE_Process_Manager::instance_).name ());
+                                typeid (ACE_Process_Manager).name ());
 #endif /* ACE_HAS_SIG_C_FUNC */

   ACE_Process_Manager::instance_ = tm;
