diff --git a/opus_functions.cmake b/opus_functions.cmake
index a3ac1c0..ef2d4c0 100644
--- a/opus_functions.cmake
+++ b/opus_functions.cmake
@@ -43,6 +43,8 @@ function(get_library_version OPUS_LIBRARY_VERSION OPUS_LIBRARY_VERSION_MAJOR)
 endfunction()
 
 function(get_package_version PACKAGE_VERSION)
+  set(PACKAGE_VERSION "0" CACHE STRING "opus package version")
+  return()
   find_package(Git)
   if(Git_FOUND AND EXISTS "${CMAKE_CURRENT_LIST_DIR}/.git")
     execute_process(COMMAND ${GIT_EXECUTABLE} describe --tags --match "v*"
