--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -13,10 +13,12 @@
 # directory, but can also be used for custom projects.
 #############################################################
 
-if(WIN32 OR APPLE)
-	set(OGRE_CMAKE_DIR "CMake")
-else()
-	set(OGRE_CMAKE_DIR "${OGRE_LIB_DIRECTORY}/OGRE/cmake")
+if (NOT DEFINED OGRE_CMAKE_DIR)
+  if(WIN32 OR APPLE)
+    set(OGRE_CMAKE_DIR "CMake")
+  else()
+    set(OGRE_CMAKE_DIR "${OGRE_LIB_DIRECTORY}/OGRE/cmake")
+  endif()
 endif()
 
 set(INST_FILES
--- a/cmake/InstallResources.cmake
+++ b/cmake/InstallResources.cmake
@@ -241,10 +241,12 @@ endif ()
 # Create the CMake package files
 include(CMakePackageConfigHelpers)
 
-if(WIN32 OR APPLE)
-  set(OGRE_CMAKE_DIR "CMake")
-else()
-  set(OGRE_CMAKE_DIR "${OGRE_LIB_DIRECTORY}/OGRE/cmake")
+if (NOT DEFINED OGRE_CMAKE_DIR)
+  if(WIN32 OR APPLE)
+    set(OGRE_CMAKE_DIR "CMake")
+  else()
+    set(OGRE_CMAKE_DIR "${OGRE_LIB_DIRECTORY}/OGRE/cmake")
+  endif()
 endif()
 configure_package_config_file(${OGRE_TEMPLATES_DIR}/OGREConfig.cmake.in ${OGRE_BINARY_DIR}/cmake/OGREConfig.cmake
     INSTALL_DESTINATION ${OGRE_CMAKE_DIR}
