diff --git a/CMake/Dependencies.cmake b/CMake/Dependencies.cmake
index 068dd27..721121b 100644
--- a/CMake/Dependencies.cmake
+++ b/CMake/Dependencies.cmake
@@ -346,7 +346,9 @@ if(NOT ANDROID AND NOT EMSCRIPTEN)
     )
   endif()
 
+  if (0)
   find_package(Qt5 COMPONENTS Core Gui QUIET)
+  endif()
   macro_log_feature(Qt5_FOUND "Qt" "optional integration with the Qt Library for window creation and input" "http://www.qt.io/" FALSE "" "")
 endif()
 
