diff --git a/CMake/Dependencies.cmake b/CMake/Dependencies.cmake
index 959fdf5..dcd28bb 100644
--- a/CMake/Dependencies.cmake
+++ b/CMake/Dependencies.cmake
@@ -217,7 +217,7 @@ find_package(FreeImage)
 macro_log_feature(FreeImage_FOUND "freeimage" "Support for commonly used graphics image formats" "http://freeimage.sourceforge.net" FALSE "" "")
 
 # Find FreeType
-find_package(Freetype)
+find_package(freetype CONFIG REQUIRED)
 macro_log_feature(FREETYPE_FOUND "freetype" "Portable font engine" "http://www.freetype.org" FALSE "" "")
 
 # Find X11
@@ -291,7 +291,7 @@ macro_log_feature(SWIG_FOUND "SWIG" "Language bindings (Python, Java, C#) for OG
 # Find sdl2
 if(NOT ANDROID AND NOT EMSCRIPTEN)
   # find script does not work in cross compilation environment
-  find_package(SDL2 QUIET)
+  find_package(SDL2 CONFIG REQUIRED)
   macro_log_feature(SDL2_FOUND "SDL2" "Simple DirectMedia Library needed for input handling in samples" "https://www.libsdl.org/" FALSE "" "")
   if(SDL2_FOUND AND NOT TARGET SDL2::SDL2)
     add_library(SDL2::SDL2 INTERFACE IMPORTED)
diff --git a/Components/Bites/src/OgreImGuiInputListener.cpp b/Components/Bites/src/OgreImGuiInputListener.cpp
index 8c0264c..12a7c53 100644
--- a/Components/Bites/src/OgreImGuiInputListener.cpp
+++ b/Components/Bites/src/OgreImGuiInputListener.cpp
@@ -7,6 +7,12 @@
 #include <OgreMath.h>
 
 #include <imgui.h>
+#include <imgui/imgui.cpp>
+#include <imgui/imgui_draw.cpp>
+#include <imgui/imgui_widgets.cpp>
+#include <imgui/imgui_demo.cpp>
+#include <imgui/imgui_stdlib.cpp>
+#include <imgui/imgui_freetype.cpp>
 
 // map sdl2 mouse buttons to imgui
 static int sdl2imgui(int b)
diff --git a/Components/Overlay/CMakeLists.txt b/Components/Overlay/CMakeLists.txt
index 1cea873..edb3c74 100644
--- a/Components/Overlay/CMakeLists.txt
+++ b/Components/Overlay/CMakeLists.txt
@@ -19,12 +19,15 @@ list(APPEND HEADER_FILES
 file(GLOB SOURCE_FILES "${CMAKE_CURRENT_SOURCE_DIR}/src/*.cpp")
 
 if(OGRE_BUILD_COMPONENT_OVERLAY_IMGUI)
+  find_package(imgui CONFIG REQUIRED)
+  find_path(IMGUI_INCLUDE_DIR imgui.h)
   list(APPEND SOURCE_FILES
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/imgui/imgui.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/imgui/imgui_draw.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/imgui/imgui_widgets.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/imgui/imgui_demo.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/imgui/misc/freetype/imgui_freetype.cpp)
+    ${IMGUI_INCLUDE_DIR}/imgui/imgui.cpp
+    ${IMGUI_INCLUDE_DIR}/imgui/imgui_draw.cpp
+    ${IMGUI_INCLUDE_DIR}/imgui/imgui_widgets.cpp
+    ${IMGUI_INCLUDE_DIR}/imgui/imgui_demo.cpp
+    ${IMGUI_INCLUDE_DIR}/imgui/imgui_stdlib.cpp
+    ${IMGUI_INCLUDE_DIR}/imgui/imgui_freetype.cpp)
 else()
     list(REMOVE_ITEM SOURCE_FILES "${CMAKE_CURRENT_SOURCE_DIR}/src/OgreImGuiOverlay.cpp")
 endif()
@@ -32,7 +35,7 @@ endif()
 # setup target
 add_library(OgreOverlay ${OGRE_COMP_LIB_TYPE} ${HEADER_FILES} ${SOURCE_FILES})
 set_target_properties(OgreOverlay PROPERTIES VERSION ${OGRE_SOVERSION} SOVERSION ${OGRE_SOVERSION})
-target_link_libraries(OgreOverlay PUBLIC OgreMain PRIVATE ${FREETYPE_LIBRARIES} ZLIB::ZLIB)
+target_link_libraries(OgreOverlay PUBLIC OgreMain PRIVATE ${FREETYPE_LIBRARIES} ZLIB::ZLIB imgui::imgui)
 target_include_directories(OgreOverlay PUBLIC 
   "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
   $<INSTALL_INTERFACE:include/OGRE/Overlay>
@@ -44,10 +47,10 @@ if(OGRE_BUILD_COMPONENT_OVERLAY_IMGUI)
     PRIVATE "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/imgui/misc/freetype>")
 
   if(UNIX)
-    set_source_files_properties(src/imgui/misc/freetype/imgui_freetype.cpp PROPERTIES
+    set_source_files_properties(${IMGUI_INCLUDE_DIR}/imgui/imgui_freetype.cpp PROPERTIES
       COMPILE_FLAGS "-Wno-missing-declarations")
 
-    set_source_files_properties(src/imgui/imgui_widgets.cpp PROPERTIES
+    set_source_files_properties(${IMGUI_INCLUDE_DIR}/imgui/imgui_widgets.cpp PROPERTIES
       COMPILE_FLAGS "-Wno-cast-qual")
   endif()
 
@@ -58,8 +61,8 @@ if(OGRE_BUILD_COMPONENT_OVERLAY_IMGUI)
     #define IMGUI_API _OgreOverlayExport")
 
   list(APPEND HEADER_FILES
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/imgui/imgui.h
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/imgui/imconfig.h)
+    ${IMGUI_INCLUDE_DIR}/imgui.h
+    ${IMGUI_INCLUDE_DIR}/imconfig.h)
 endif()
 
 generate_export_header(OgreOverlay 
diff --git a/PlugIns/DotScene/CMakeLists.txt b/PlugIns/DotScene/CMakeLists.txt
index fd285df..ab8e9c1 100644
--- a/PlugIns/DotScene/CMakeLists.txt
+++ b/PlugIns/DotScene/CMakeLists.txt
@@ -1,4 +1,4 @@
-find_package(pugixml QUIET)
+find_package(pugixml CONFIG REQUIRED)
 
 file(GLOB HEADER_FILES "${CMAKE_CURRENT_SOURCE_DIR}/include/*.h")
 list(APPEND HEADER_FILES ${PROJECT_BINARY_DIR}/include/OgreDotScenePluginExports.h)
diff --git a/Tools/XMLConverter/CMakeLists.txt b/Tools/XMLConverter/CMakeLists.txt
index 8fe9683..2a516b2 100644
--- a/Tools/XMLConverter/CMakeLists.txt
+++ b/Tools/XMLConverter/CMakeLists.txt
@@ -22,7 +22,7 @@ set(SOURCE_FILES
 )
 
 # allow override by cmake
-find_package(pugixml QUIET)
+find_package(pugixml CONFIG REQUIRED)
 
 add_executable(OgreXMLConverter ${HEADER_FILES} ${SOURCE_FILES})
 target_include_directories(OgreXMLConverter PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/include)
