diff --git a/external/crashpad/crashpad-config.cmake.in b/external/crashpad/crashpad-config.cmake.in
index 846797d..d7c4ae6 100644
--- a/external/crashpad/crashpad-config.cmake.in
+++ b/external/crashpad/crashpad-config.cmake.in
@@ -1,6 +1,9 @@
 include("${CMAKE_CURRENT_LIST_DIR}/crashpad-targets.cmake")
 
 if(@CRASHPAD_ZLIB_SYSTEM@)
+    include(CMakeFindDependencyMacro)
+    find_dependency(ZLIB)
+elseif(0)
     find_package(ZLIB REQUIRED)
     target_include_directories(crashpad::zlib INTERFACE ${ZLIB_INCLUDE_DIRS})
     target_compile_definitions(crashpad::zlib INTERFACE ${ZLIB_COMPILE_DEFINITIONS})
diff --git a/external/crashpad/third_party/zlib/CMakeLists.txt b/external/crashpad/third_party/zlib/CMakeLists.txt
index eeb449f..59a6ff1 100644
--- a/external/crashpad/third_party/zlib/CMakeLists.txt
+++ b/external/crashpad/third_party/zlib/CMakeLists.txt
@@ -1,15 +1,10 @@
 if(CRASHPAD_ZLIB_SYSTEM)
     add_library(crashpad_zlib INTERFACE)
-    string(REPLACE ";" "$<SEMICOLON>" GENEX_ZLIB_LIBRARIES "${ZLIB_LIBRARIES}")
-    target_include_directories(crashpad_zlib INTERFACE
-        $<BUILD_INTERFACE:${ZLIB_INCLUDE_DIRS}>
-    )
     target_compile_definitions(crashpad_zlib INTERFACE
         ZLIB_CONST
         CRASHPAD_ZLIB_SOURCE_SYSTEM
-        $<BUILD_INTERFACE:${ZLIB_COMPILE_DEFINITIONS}>
     )
-    target_link_libraries(crashpad_zlib INTERFACE $<BUILD_INTERFACE:${GENEX_ZLIB_LIBRARIES}>)
+    target_link_libraries(crashpad_zlib INTERFACE ZLIB::ZLIB)
 else()
     add_library(crashpad_zlib STATIC
         zlib/adler32.c
diff --git a/sentry-config.cmake.in b/sentry-config.cmake.in
index 89ea345..acbd5e2 100644
--- a/sentry-config.cmake.in
+++ b/sentry-config.cmake.in
@@ -3,9 +3,14 @@
 set(SENTRY_BACKEND @SENTRY_BACKEND@)
 set(SENTRY_TRANSPORT @SENTRY_TRANSPORT@)
 
+include(CMakeFindDependencyMacro)
+if("@SENTRY_LINK_PTHREAD@")
+	find_dependency(Threads)
+endif()
+
 if(SENTRY_BACKEND STREQUAL "crashpad")
 	if(@SENTRY_CRASHPAD_SYSTEM@)
-		find_package(crashpad REQUIRED)
+		find_dependency(crashpad)
 	else()
 		include("${CMAKE_CURRENT_LIST_DIR}/sentry_crashpad-targets.cmake")
 	endif()
@@ -14,7 +19,5 @@ endif()
 include("${CMAKE_CURRENT_LIST_DIR}/sentry-targets.cmake")
 
 if(SENTRY_TRANSPORT STREQUAL "curl" AND NOT @BUILD_SHARED_LIBS@)
-	find_package(CURL REQUIRED)
-	set_property(TARGET sentry::sentry APPEND
-		PROPERTY INTERFACE_LINK_LIBRARIES ${CURL_LIBRARIES})
+	find_dependency(CURL)
 endif()
