diff --git a/sentry-config.cmake.in b/sentry-config.cmake.in
index 70ce7d3..796d428 100644
--- a/sentry-config.cmake.in
+++ b/sentry-config.cmake.in
@@ -9,7 +9,7 @@ set(SENTRY_LINK_PTHREAD @SENTRY_LINK_PTHREAD@)

 if(SENTRY_BACKEND STREQUAL "crashpad")
 	include("${CMAKE_CURRENT_LIST_DIR}/sentry_crashpad-targets.cmake")
-	if(NOT MSVC AND NOT SENTRY_BUILD_SHARED_LIBS)
+	if(NOT SENTRY_BUILD_SHARED_LIBS)
 		find_dependency(ZLIB REQUIRED)
 	endif()
 endif()
@@ -23,7 +23,5 @@ endif()
 include("${CMAKE_CURRENT_LIST_DIR}/sentry-targets.cmake")

 if(SENTRY_TRANSPORT STREQUAL "curl" AND (NOT @BUILD_SHARED_LIBS@ OR NOT SENTRY_BUILD_SHARED_LIBS))
 	find_dependency(CURL REQUIRED)
-	set_property(TARGET sentry::sentry APPEND
-		PROPERTY INTERFACE_LINK_LIBRARIES ${CURL_LIBRARIES})
 endif()
