diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index 39cdf8b..a28b198 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -182,6 +182,8 @@ if(NOT MINIMAL_BUILD AND USD_SUPPORT)
     UNIFY_ENVVARS_AND_CACHE(USD_INCLUDE_DIR)
     UNIFY_ENVVARS_AND_CACHE(USD_LIBRARY_DIR)
 
+    find_package(pxr CONFIG REQUIRED)
+    if(0)
     if("${USD_INCLUDE_DIR}" STREQUAL "")
         find_path(USD_INCLUDE_DIR pxr.h
             [PATHS
@@ -199,6 +201,7 @@ if(NOT MINIMAL_BUILD AND USD_SUPPORT)
         set(USD_INCLUDE_DIR ${USD_INCLUDE_DIR} CACHE FILEPATH "USD header files")
         message(STATUS "Looking for USD include files in: ${USD_INCLUDE_DIR}")
     endif()
+    endif()
 
     set(USD_LIBRARIES
             usd_usd 
@@ -336,6 +339,9 @@ IF(MSVC)
 ENDIF()
 
 # Open CASCADE
+find_package(OpenCASCADE CONFIG REQUIRED)
+set(OCC_VERSION_STRING ${OpenCASCADE_VERSION})
+if (0)
 IF("${OCC_INCLUDE_DIR}" STREQUAL "")
     FIND_PATH(OCC_INCLUDE_DIR Standard_Version.hxx
         [PATHS
@@ -354,6 +360,7 @@ ELSE()
     SET(OCC_INCLUDE_DIR ${OCC_INCLUDE_DIR} CACHE FILEPATH "Open CASCADE header files")
     MESSAGE(STATUS "Looking for Open CASCADE include files in: ${OCC_INCLUDE_DIR}")
 ENDIF()
+endif()
 
 SET(OPENCASCADE_LIBRARY_NAMES
     TKernel TKMath TKBRep TKGeomBase TKGeomAlgo TKG3d TKG2d TKShHealing TKTopAlgo TKMesh TKPrim TKBool TKBO
@@ -519,6 +526,14 @@ IF(NOT MINIMAL_BUILD AND COLLADA_SUPPORT)
 ENDIF()
 
 if(NOT MINIMAL_BUILD AND HDF5_SUPPORT)
+    find_package(hdf5 CONFIG REQUIRED)
+    if(BUILD_SHARED_LIBS)
+        set(HDF5_LIBRARIES hdf5::hdf5-shared hdf5::hdf5_cpp-shared)
+    else()
+        set(HDF5_LIBRARIES hdf5::hdf5-static hdf5::hdf5_cpp-static)
+    endif(BUILD_SHARED_LIBS)
+    
+    if(0)
     IF("${HDF5_INCLUDE_DIR}" STREQUAL "")
 		MESSAGE(STATUS "No HDF5 include directory specified")
 	ElSE()
@@ -595,6 +610,7 @@ if(NOT MINIMAL_BUILD AND HDF5_SUPPORT)
             z dl
         )
     endif()
+    endif()
     
     add_definitions(-DWITH_HDF5)
 	set(SWIG_DEFINES ${SWIG_DEFINES} -DWITH_HDF5)
@@ -995,7 +1011,7 @@ ENDIF()
 IF(NOT MINIMAL_BUILD AND BUILD_IFCMAX)
 	ADD_SUBDIRECTORY(../src/ifcmax ifcmax)
 ENDIF()
-if (NOT MINIMAL_BUILD)
+if (0)
 ADD_SUBDIRECTORY(../src/svgfill svgfill)
 endif()
 
