diff --git a/Makefile.nmake b/Makefile.nmake
index 56ae888..7260785 100644
--- a/Makefile.nmake
+++ b/Makefile.nmake
@@ -143,7 +143,7 @@ objs =  \
 
 INCLUDES  = -I./ -Ierasure_code/ -Iraid/ -Icrc/ -Iigzip/ -Iinclude/ -Imem/
 LINKFLAGS = /nologo
-CFLAGS   = -O2 -D NDEBUG /nologo -D_USE_MATH_DEFINES -Qstd=c99 $(INCLUDES) $(D)
+CFLAGS   = /nologo -D_USE_MATH_DEFINES -Qstd=c99 $(INCLUDES) $(D)
 AFLAGS   = -f win64 $(INCLUDES) $(D)
 CC       = icl
 AS       = yasm
