--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,10 +1,12 @@
 cmake_minimum_required(VERSION 3.10)
 project(MiniFB VERSION 0.9.1 LANGUAGES C CXX)
 
-include(CheckLanguage)
-check_language(OBJC)
-if(CMAKE_OBJC_COMPILER)
-  enable_language(OBJC)
+if(APPLE)
+  include(CheckLanguage)
+  check_language(OBJC)
+  if(CMAKE_OBJC_COMPILER)
+    enable_language(OBJC)
+  endif()
 endif()
 
 message(STATUS "Processing " ${PROJECT_NAME})
