diff --git a/include/aws/common/byte_buf.h b/include/aws/common/byte_buf.h
index 545b06d..c579c82 100644
--- a/include/aws/common/byte_buf.h
+++ b/include/aws/common/byte_buf.h
@@ -21,6 +21,8 @@
 
 #include <string.h>
 
+#pragma warning(disable: 4068)
+
 /**
  * Represents a length-delimited binary string or buffer. If byte buffer points
  * to constant memory or memory that should otherwise not be freed by this
