diff --git a/create_lib.bat b/create_lib.bat
index c70a179..4690d9e 100644
--- a/create_lib.bat
+++ b/create_lib.bat
@@ -7,10 +7,16 @@ IF ""=="%1" (set WINPCAPSOURCEDIR=.\) ELSE (set WINPCAPSOURCEDIR=%1)
 echo Creating \Lib folder
 mkdir %WPDPACKDESTDIR% 		>nul 2>nul
 mkdir %WPDPACKDESTDIR%\Lib 	>nul 2>nul
+mkdir %WPDPACKDESTDIR%\Lib\debug\ 	>nul 2>nul
 mkdir %WPDPACKDESTDIR%\Lib\x64	>nul 2>nul
+mkdir %WPDPACKDESTDIR%\Lib\x64\debug	>nul 2>nul
 
-xcopy /v /Y "%WINPCAPSOURCEDIR%\wpcap\PRJ\Release No AirPcap\x86\wpcap.lib"		%WPDPACKDESTDIR%\Lib\ >nul
-xcopy /v /Y "%WINPCAPSOURCEDIR%\wpcap\PRJ\Release No AirPcap\x64\wpcap.lib"		%WPDPACKDESTDIR%\Lib\x64 >nul
+xcopy /v /Y "%WINPCAPSOURCEDIR%\wpcap\PRJ\Debug No AirPcap\x86\wpcap.lib"		    %WPDPACKDESTDIR%\Lib\debug\ >nul
+xcopy /v /Y "%WINPCAPSOURCEDIR%\wpcap\PRJ\Debug No AirPcap\x64\wpcap.lib"		    %WPDPACKDESTDIR%\Lib\x64\debug\ >nul
+xcopy /v /Y "%WINPCAPSOURCEDIR%\wpcap\PRJ\Release No AirPcap\x86\wpcap.lib"		%WPDPACKDESTDIR%\Lib\ >nul
+xcopy /v /Y "%WINPCAPSOURCEDIR%\wpcap\PRJ\Release No AirPcap\x64\wpcap.lib"		%WPDPACKDESTDIR%\Lib\x64 >nul
+xcopy /v /Y %WINPCAPSOURCEDIR%\packetNtx\Dll\Project\Debug\x86\packet.lib	 	    %WPDPACKDESTDIR%\Lib\debug\ >nul
+xcopy /v /Y %WINPCAPSOURCEDIR%\packetNtx\Dll\Project\Debug\x64\packet.lib	 	    %WPDPACKDESTDIR%\Lib\x64\debug\ >nul
 xcopy /v /Y %WINPCAPSOURCEDIR%\packetNtx\Dll\Project\Release\x86\packet.lib	 	%WPDPACKDESTDIR%\Lib\ >nul
 xcopy /v /Y %WINPCAPSOURCEDIR%\packetNtx\Dll\Project\Release\x64\packet.lib	 	%WPDPACKDESTDIR%\Lib\x64 >nul
 
