diff --git a/clickhouse/base/compressed.cpp b/clickhouse/base/compressed.cpp
index 6b7af08..c3a6449 100644
--- a/clickhouse/base/compressed.cpp
+++ b/clickhouse/base/compressed.cpp
@@ -1,8 +1,8 @@
 #include "compressed.h"
 #include "wire_format.h"
 
-#include <cityhash/city.h>
-#include <lz4/lz4.h>
+#include <city.h>
+#include <lz4.h>
 
 #include <system_error>
 
diff --git a/clickhouse/client.cpp b/clickhouse/client.cpp
index 1053006..59950ab 100644
--- a/clickhouse/client.cpp
+++ b/clickhouse/client.cpp
@@ -8,8 +8,8 @@
 
 #include "columns/factory.h"
 
-#include <cityhash/city.h>
-#include <lz4/lz4.h>
+#include <city.h>
+#include <lz4.h>
 
 #include <assert.h>
 #include <atomic>
diff --git a/clickhouse/types/types.h b/clickhouse/types/types.h
index 8d453cd..69aa6a4 100644
--- a/clickhouse/types/types.h
+++ b/clickhouse/types/types.h
@@ -4,6 +4,7 @@
 #include <memory>
 #include <string>
 #include <vector>
+#include <stdexcept>
 
 namespace clickhouse {
 
