From 0f783f1a8eac7a78bc92198e140d4a5c745382ca Mon Sep 17 00:00:00 2001
From: Nicole Mazzuca <mazzucan@outlook.com>
Date: Wed, 23 Sep 2020 11:50:21 -0700
Subject: [PATCH] include unistd for gete?[ug]id

---
 loader/linux/icd_linux_envvars.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/loader/linux/icd_linux_envvars.c b/loader/linux/icd_linux_envvars.c
index 4af5a08..92b0698 100644
--- a/loader/linux/icd_linux_envvars.c
+++ b/loader/linux/icd_linux_envvars.c
@@ -25,6 +25,10 @@
 
 #include <stdlib.h>
 
+#if defined(__APPLE__)
+#include <unistd.h>
+#endif
+
 char *khrIcd_getenv(const char *name) {
     // No allocation of memory necessary for Linux.
     return getenv(name);
-- 
2.24.3 (Apple Git-128)

