diff --git a/src/cmake/DawnLibrary.cmake b/src/cmake/DawnLibrary.cmake
index f80426804a..e9fbeaf34d 100644
--- a/src/cmake/DawnLibrary.cmake
+++ b/src/cmake/DawnLibrary.cmake
@@ -184,9 +184,9 @@ function(dawn_install_target name)
     get_target_property(target_type "${name}" TYPE)
     # Only shared libraries and executables generate PDB files through the linker
     # Static libraries are created by the archiver and don't generate PDB files
-    if ((target_type STREQUAL "SHARED_LIBRARY") OR (target_type STREQUAL "EXECUTABLE"))
-      install(FILES $<TARGET_PDB_FILE:${name}> DESTINATION ${CMAKE_INSTALL_BINDIR} OPTIONAL)
-    endif()
+    #if ((target_type STREQUAL "SHARED_LIBRARY") OR (target_type STREQUAL "EXECUTABLE"))
+    #  install(FILES $<TARGET_PDB_FILE:${name}> DESTINATION ${CMAKE_INSTALL_BINDIR} OPTIONAL)
+    #endif()
   endif (MSVC)
 
   # Automatically determine where each header should go based on its subdirectory in the include dir.
