diff --git a/src/dawn/native/d3d/PlatformFunctions.cpp b/src/dawn/native/d3d/PlatformFunctions.cpp
index 12d65e0db5..d3abb05fb5 100644
--- a/src/dawn/native/d3d/PlatformFunctions.cpp
+++ b/src/dawn/native/d3d/PlatformFunctions.cpp
@@ -121,10 +121,14 @@ MaybeError PlatformFunctions::EnsureFXC(std::span<const std::string> searchPaths
 }
 
 void PlatformFunctions::InitWindowsVersion() {
+#if DAWN_PLATFORM_IS(WINUWP)
+    return;
+#else
     // Currently we only care about the build number of Windows 10 and Windows 11.
     if (!IsWindows10OrGreater()) {
         return;
     }
+#endif
 
     // Referenced from base/win/windows_version.cc in Chromium
     constexpr wchar_t kRegKeyWindowsNTCurrentVersion[] =
