diff --git a/src/tint/lang/null/writer/common/options.h b/src/tint/lang/null/writer/common/options.h
index 3401c5abb0..fdd7c85471 100644
--- a/src/tint/lang/null/writer/common/options.h
+++ b/src/tint/lang/null/writer/common/options.h
@@ -30,6 +30,8 @@
 
 #include <string>
 
+#include "src/tint/api/common/binding_point.h"
+#include "src/tint/api/common/bindings.h"
 #include "src/tint/api/common/substitute_overrides_config.h"
 #include "src/tint/utils/reflection/reflection.h"
 
