diff --git a/vs_toolchain.py b/vs_toolchain.py
index 61b322950..072065624 100755
--- a/vs_toolchain.py
+++ b/vs_toolchain.py
@@ -419,7 +419,7 @@ def CopyDlls(target_dir, configuration, target_cpu):
   _CopyRuntime(target_dir, runtime_dir, target_cpu, debug=False)
   if configuration == 'Debug':
     _CopyRuntime(target_dir, runtime_dir, target_cpu, debug=True)
-  _CopyDebugger(target_dir, target_cpu)
+  #_CopyDebugger(target_dir, target_cpu)
   if target_cpu == 'arm64':
     target_dir = os.path.join(target_dir, 'win_clang_x64')
     target_cpu = 'x64'
@@ -428,7 +428,7 @@ def CopyDlls(target_dir, configuration, target_cpu):
     _CopyRuntime(target_dir, runtime_dir, target_cpu, debug=False)
     if configuration == 'Debug':
       _CopyRuntime(target_dir, runtime_dir, target_cpu, debug=True)
-    _CopyDebugger(target_dir, target_cpu)
+   #_CopyDebugger(target_dir, target_cpu)
 
 
 def _CopyDebugger(target_dir, target_cpu):
