diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4456aa5..4ed061f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -385,7 +385,9 @@ else (build_errors)
     add_subdirectory(gazebo)
     add_subdirectory(media)
     add_subdirectory(tools)
+    if (BUILD_PLUGINS)
     add_subdirectory(plugins)
+    endif()
     add_subdirectory(interfaces)
     add_subdirectory(worlds)
     add_subdirectory(models)
