diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3e0ae3d..eeb1ba7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -384,16 +384,18 @@ else (build_errors)
     set(TEST_TYPE "UNIT")
     add_subdirectory(gazebo)
     add_subdirectory(media)
+    if (BUILD_TOOLS)
     add_subdirectory(tools)
+    endif()
+    if (BUILD_PLUGINS)
     add_subdirectory(plugins)
+    endif()
     add_subdirectory(interfaces)
     add_subdirectory(worlds)
     add_subdirectory(models)
 
     if (BUILD_TESTING)
       add_subdirectory(test)
-    else()
-      add_subdirectory(test EXCLUDE_FROM_ALL)
     endif()
 
   endif (BUILD_GAZEBO)
diff --git a/gazebo/CMakeLists.txt b/gazebo/CMakeLists.txt
index 9ad9d02..5c45753 100644
--- a/gazebo/CMakeLists.txt
+++ b/gazebo/CMakeLists.txt
@@ -41,7 +41,9 @@ add_subdirectory(gui)
 add_subdirectory(physics)
 add_subdirectory(sensors)
 add_subdirectory(util)
+if (BUILD_TESTING)
 add_subdirectory(test)
+endif()
 
 add_dependencies(gazebo_physics gazebo_msgs)
 add_dependencies(gazebo_gui gazebo_msgs)
