diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9d9ed21..799bc20 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,6 +14,7 @@ ENDIF( )
 project(kenlm)
 
 option(FORCE_STATIC "Build static executables" OFF)
+option(ENABLE_INTERPOLATE "Build interpolation program" OFF)
 if (FORCE_STATIC)
   #presumably overkill, is there a better way?
   #http://cmake.3232098.n2.nabble.com/Howto-compile-static-executable-td5580269.html
diff --git a/lm/CMakeLists.txt b/lm/CMakeLists.txt
index 25d22c3..395815b 100644
--- a/lm/CMakeLists.txt
+++ b/lm/CMakeLists.txt
@@ -43,6 +43,9 @@ install(TARGETS kenlm
   ARCHIVE DESTINATION lib
 )
 
+file(GLOB SRC_HEADERS ${CMAKE_CURRENT_LIST_DIR}/*.h ${CMAKE_CURRENT_LIST_DIR}/*.hh)
+install(FILES ${SRC_HEADERS} DESTINATION include/kenlm)
+
 # This directory has children that need to be processed
 add_subdirectory(builder)
 add_subdirectory(filter)
@@ -50,12 +53,14 @@ add_subdirectory(interpolate)
 
 # Explicitly list the executable files to be compiled
 set(EXE_LIST
-  query
   fragment
-  build_binary
   kenlm_benchmark
 )
 
+if (UNIX)
+    list(APPEND EXE_LIST query build_binary)
+endif()
+
 set(LM_LIBS kenlm kenlm_util ${Boost_LIBRARIES} Threads::Threads)
 
 AddExes(EXES ${EXE_LIST}
diff --git a/lm/interpolate/CMakeLists.txt b/lm/interpolate/CMakeLists.txt
index 04776be..7375ffd 100644
--- a/lm/interpolate/CMakeLists.txt
+++ b/lm/interpolate/CMakeLists.txt
@@ -1,5 +1,6 @@
+if (ENABLE_INTERPOLATE)
 # Eigen3 less than 3.1.0 has a race condition: http://eigen.tuxfamily.org/bz/show_bug.cgi?id=466
-find_package(Eigen3 3.1.0 CONFIG)
+find_package(Eigen3 3.1.0 CONFIG REQUIRED)
 
 if(EIGEN3_FOUND)
   include_directories(${EIGEN3_INCLUDE_DIR})
@@ -54,3 +55,4 @@ else()
       "(cd $HOME; wget -O - https://gitlab.com/libeigen/eigen/-/archive/3.3.7/eigen-3.3.7.tar.bz2 |tar xj)\n"
       "rm CMakeCache.txt\n")
 endif()
+endif()
\ No newline at end of file
diff --git a/util/CMakeLists.txt b/util/CMakeLists.txt
index b0a06b1..711aa32 100644
--- a/util/CMakeLists.txt
+++ b/util/CMakeLists.txt
@@ -84,6 +84,15 @@ AddExes(EXES probing_hash_table_benchmark
         LIBRARIES kenlm_util ${Boost_LIBRARIES} Threads::Threads)
 endif()
 
+install(TARGETS kenlm_util
+  RUNTIME DESTINATION bin
+  LIBRARY DESTINATION lib
+  ARCHIVE DESTINATION lib
+)
+
+file(GLOB UTILS_HEADER ${CMAKE_CURRENT_LIST_DIR}/*.h ${CMAKE_CURRENT_LIST_DIR}/*.hh)
+install(FILES ${UTILS_HEADER} DESTINATION include/kenlm/utils)
+
 # Only compile and run unit tests if tests should be run
 if(BUILD_TESTING)
   set(KENLM_BOOST_TESTS_LIST
