diff --git a/CMakeLists.txt b/CMakeLists.txt
index 827d7d8..0604127 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,8 +45,8 @@ if (slick-shm_FOUND)
     message(STATUS "Found slick-shm ${slick-shm_VERSION}: ${slick-shm_DIR}")
 endif()
 
-find_package(slick-stream-buffer-multiplexer 1.0.1 CONFIG QUIET)
-if (NOT slick-stream-buffer-multiplexer_FOUND)
+find_package(slick-stream-buffer-multiplexer CONFIG REQUIRED)
+if (FALSE)
     message(STATUS "Fetching slick-stream-buffer-multiplexer...")
     include(FetchContent)
     set(BUILD_SLICK_STREAM_BUFFER_MULTIPLEXER_TESTS OFF CACHE BOOL "" FORCE)
@@ -72,8 +72,8 @@ if (slick-stream-buffer_FOUND)
     message(STATUS "Found slick-stream-buffer ${slick-stream-buffer_VERSION}: ${slick-stream-buffer_DIR}")
 endif()
 
-find_package(slick-dynamic-buffer 1.0.1 CONFIG QUIET)
-if (NOT slick-dynamic-buffer_FOUND)
+find_package(slick-dynamic-buffer CONFIG REQUIRED)
+if (FALSE)
     message(STATUS "Fetching slick-dynamic-buffer...")
     include(FetchContent)
     set(BUILD_SLICK_DYNAMIC_BUFFER_TESTS OFF CACHE BOOL "" FORCE)
diff --git a/cmake/slick-net-config.cmake.in b/cmake/slick-net-config.cmake.in
index 1bde373..f8cd2c1 100644
--- a/cmake/slick-net-config.cmake.in
+++ b/cmake/slick-net-config.cmake.in
@@ -5,8 +5,8 @@ include(CMakeFindDependencyMacro)
 # Find required dependencies
 find_dependency(Boost CONFIG COMPONENTS beast context)
 find_dependency(OpenSSL CONFIG)
-find_dependency(slick-stream-buffer-multiplexer 1.0.1 CONFIG)
-find_dependency(slick-dynamic-buffer 1.0.1 CONFIG)
+find_dependency(slick-stream-buffer-multiplexer CONFIG)
+find_dependency(slick-dynamic-buffer CONFIG)
 
 # Include the targets file
 include("${CMAKE_CURRENT_LIST_DIR}/slick-net-targets.cmake")
