diff --git a/CMakeLists.txt b/CMakeLists.txt
index 97a6737..4ae28fb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,4 +48,4 @@ install(EXPORT ${PROJECT_NAME}_targets
 )
 
 # Install the headers at a standard cmake location.
-install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/include/${PROJECT_NAME}" DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
\ No newline at end of file
+install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/include/wil" DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}")
\ No newline at end of file
