diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5411be0..e92c495 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -97,7 +97,7 @@ install(TARGETS maxminddb
 install(EXPORT maxminddb
         FILE maxminddb-config.cmake
         NAMESPACE maxminddb::
-        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/maxminddb)
+        DESTINATION ${CMAKE_INSTALL_DATADIR}/cmake/maxminddb)
 
 # We always want to build mmdblookup
 add_subdirectory(bin)
