diff --git a/cmake/FindAVCODEC.cmake b/cmake/FindAVCODEC.cmake
--- a/cmake/FindAVCODEC.cmake
+++ b/cmake/FindAVCODEC.cmake
@@ -15,9 +15,9 @@
 #
 ########################################
 # Find avcodec
 include(GzPkgConfig)
-gz_pkg_check_modules_quiet(AVCODEC libavcodec)
+gz_pkg_check_modules_quiet(AVCODEC libavcodec INTERFACE)
 
 if(NOT AVCODEC_FOUND)
   include(GzManualSearch)
   gz_manual_search(AVCODEC
diff --git a/cmake/FindAVDEVICE.cmake b/cmake/FindAVDEVICE.cmake
--- a/cmake/FindAVDEVICE.cmake
+++ b/cmake/FindAVDEVICE.cmake
@@ -19,9 +19,9 @@
 set(av_minor ${AVDEVICE_FIND_VERSION_MINOR})
 set(av_patch ${AVDEVICE_FIND_VERSION_PATCH})
 
 include(GzPkgConfig)
-gz_pkg_check_modules_quiet(AVDEVICE "libavdevice >= ${av_major}.${av_minor}.${av_patch}")
+gz_pkg_check_modules_quiet(AVDEVICE "libavdevice >= ${av_major}.${av_minor}.${av_patch}" INTERFACE)
 
 if(NOT AVDEVICE_FOUND)
   include(GzManualSearch)
   gz_manual_search(AVDEVICE
diff --git a/cmake/FindAVFORMAT.cmake b/cmake/FindAVFORMAT.cmake
--- a/cmake/FindAVFORMAT.cmake
+++ b/cmake/FindAVFORMAT.cmake
@@ -15,9 +15,9 @@
 #
 ########################################
 # Find AV format
 include(GzPkgConfig)
-gz_pkg_check_modules_quiet(AVFORMAT libavformat)
+gz_pkg_check_modules_quiet(AVFORMAT libavformat INTERFACE)
 
 if(NOT AVFORMAT_FOUND)
   include(GzManualSearch)
   gz_manual_search(AVFORMAT
diff --git a/cmake/FindAVUTIL.cmake b/cmake/FindAVUTIL.cmake
--- a/cmake/FindAVUTIL.cmake
+++ b/cmake/FindAVUTIL.cmake
@@ -15,9 +15,9 @@
 #
 ########################################
 # Find avutil
 include(GzPkgConfig)
-gz_pkg_check_modules_quiet(AVUTIL libavutil)
+gz_pkg_check_modules_quiet(AVUTIL libavutil INTERFACE)
 
 if(NOT AVUTIL_FOUND)
   include(GzManualSearch)
   gz_manual_search(AVUTIL
diff --git a/cmake/FindSWSCALE.cmake b/cmake/FindSWSCALE.cmake
--- a/cmake/FindSWSCALE.cmake
+++ b/cmake/FindSWSCALE.cmake
@@ -15,9 +15,9 @@
 #
 ########################################
 # Find libswscale format
 include(GzPkgConfig)
-gz_pkg_check_modules_quiet(SWSCALE libswscale)
+gz_pkg_check_modules_quiet(SWSCALE libswscale INTERFACE)
 
 if(NOT SWSCALE_FOUND)
   include(GzManualSearch)
   gz_manual_search(SWSCALE
