diff --git a/CMakeLists.txt b/CMakeLists.txt
index d811dda..97ee8b8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,13 +39,7 @@ endif()
 # Install Targets
 # --------------------------------------------------------------------------------------------------------
 
-include("cmake/cpm.cmake")
-
-CPMFindPackage(
-  NAME           PackageProject
-  VERSION        1.13.0
-  GIT_REPOSITORY "https://github.com/TheLartians/PackageProject.cmake"
-)
+include("cmake/PackageProject.cmake/CMakeLists.txt")
 
 packageProject(
   NAMESPACE cr
