diff --git a/3fd/web_wws_impl_host.cpp b/3fd/web_wws_impl_host.cpp
index b87dc75..1cff45f 100644
--- a/3fd/web_wws_impl_host.cpp
+++ b/3fd/web_wws_impl_host.cpp
@@ -48,7 +48,7 @@ namespace wws
                 throw AppException<std::runtime_error>(oss.str());
             }
 
-            const auto fileSizeBytes = inputStream.seekg(0, std::ios::end).tellg().seekpos(); // move cursor to the end to get the zize
+            const auto fileSizeBytes = inputStream.seekg(0, std::ios::end).tellg(); // move cursor to the end to get the zize
 
             // File is not trunked:
             if (fileSizeBytes > 0)
