From 91fab7c4b8f7da3182f07f5392ebe9388f979157 Mon Sep 17 00:00:00 2001
From: Maki <mxmcube@gmail.com>
Date: Wed, 3 Jun 2020 01:40:58 +0100
Subject: [PATCH] Include cstdint

---
 cpp/types.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cpp/types.h b/cpp/types.h
index 8c7cc2b..122dc71 100644
--- a/cpp/types.h
+++ b/cpp/types.h
@@ -3,6 +3,8 @@
 #include "ffi.h"
 #include "event.h"
 
+#include <cstdint>
+
 namespace discord {
 
 enum class Result {
-- 
2.26.2

