diff --git a/config/Make.rules b/config/Make.rules
index 5cf7048..f817b79 100644
--- a/config/Make.rules
+++ b/config/Make.rules
@@ -87,6 +87,7 @@ CONFIGS                 ?= $(firstword $(supported-configs))
 os                      ?= $(shell uname)
 
 include $(top_srcdir)/config/Make.rules.$(os)
+cppflags    := $(filter-out -Werror,$(cppflags))
 include $(top_srcdir)/config/Make.project.rules
 include $(top_srcdir)/config/Make.tests.rules
 
