diff --git a/cpp/src/slice2swift/SwiftUtil.cpp b/cpp/src/slice2swift/SwiftUtil.cpp
index 404dee7670..01e144de90 100644
--- a/cpp/src/slice2swift/SwiftUtil.cpp
+++ b/cpp/src/slice2swift/SwiftUtil.cpp
@@ -1523,7 +1523,7 @@ SwiftGenerator::writeMemberwiseInitializer(IceUtilInternal::Output& out,
                                            const DataMemberList& baseMembers,
                                            const DataMemberList& allMembers,
                                            const ContainedPtr& p,
-                                           bool local,
+                                           bool /*local*/,
                                            bool rootClass,
                                            const StringPairList& extraParams)
 {
@@ -1531,11 +1531,6 @@ SwiftGenerator::writeMemberwiseInitializer(IceUtilInternal::Output& out,
     {
         out << sp;
         out << nl;
-        int typeCtx = TypeContextInParam;
-        if(local)
-        {
-            typeCtx |= TypeContextLocal;
-        }
         out << "public init" << spar;
         for(DataMemberList::const_iterator i = allMembers.begin(); i != allMembers.end(); ++i)
         {
