diff --git a/CMakeLists.txt b/CMakeLists.txt
index fb160b5..6235d89 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -34,7 +34,7 @@ set_target_properties(${SHARED_LIB} PROPERTIES OUTPUT_NAME ${PROJECT_NAME})
 set_target_properties(${STATIC_LIB} PROPERTIES CLEAN_DIRECT_OUTPUT 1)
 set_target_properties(${SHARED_LIB} PROPERTIES CLEAN_DIRECT_OUTPUT 1)
 
-add_subdirectory(test)
+#add_subdirectory(test)
 
 # Install static lib.
 install(TARGETS ${STATIC_LIB}
