diff --git a/CMakeLists.txt b/CMakeLists.txt
index b2c39d1..29bb533 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -35,10 +35,8 @@ if (APPLE)
     endif()
 endif()
 
-if(MSVC)
-    list (APPEND CMAKE_CXX_FLAGS " -std=c++11 -O3 ")
-else()
-    list (APPEND CMAKE_CXX_FLAGS " -std=c++11 -O3 -Wall -Werror ")
+if(NOT MSVC)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -O3 -Wall -Werror")
 endif()
 
 # Check for cryptopp (static)
@@ -64,21 +62,26 @@ set(LICENSEPP_SOURCE_FILES
     src/license.cc
 )
 
-if (BUILD_SHARED_LIBS)
-    add_library (licensepp-lib SHARED ${LICENSEPP_SOURCE_FILES})
-else()
-    add_library (licensepp-lib STATIC ${LICENSEPP_SOURCE_FILES})
-endif()
+add_library (licensepp-lib ${LICENSEPP_SOURCE_FILES})
 
 set_target_properties (licensepp-lib PROPERTIES
     VERSION ${LICENSEPP_SOVERSION}
 )
+target_include_directories (licensepp-lib PUBLIC $<INSTALL_INTERFACE:include>)
 target_link_libraries (licensepp-lib
     ${CRYPTOPP_LIBRARIES}
 )
 
 set_target_properties (licensepp-lib PROPERTIES OUTPUT_NAME "licensepp")
-install (TARGETS licensepp-lib EXPORT licensepp-config DESTINATION lib)
+
+include(GNUInstallDirs)
+
+install (TARGETS licensepp-lib
+    EXPORT licensepp-config
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+)
 install (FILES license++/base-license-manager.h DESTINATION "include/license++")
 install (FILES license++/issuing-authority.h DESTINATION "include/license++")
 install (FILES license++/license.h DESTINATION "include/license++")
