diff --git a/vcg/complex/algorithms/curve_on_manifold.h b/vcg/complex/algorithms/curve_on_manifold.h
index 5b9d858..f091b4b 100644
--- a/vcg/complex/algorithms/curve_on_manifold.h
+++ b/vcg/complex/algorithms/curve_on_manifold.h
@@ -36,7 +36,7 @@
 #include <vcg/space/index/kdtree/kdtree.h>
 #include <vcg/math/histogram.h>
 #include<vcg/space/distance3.h>
-#include<eigenlib/Eigen/Core>
+#include<eigen3/Eigen/Core>
 #include <vcg/complex/algorithms/attribute_seam.h>
 #include <wrap/io_trimesh/export_ply.h>
 
diff --git a/vcg/complex/algorithms/harmonic.h b/vcg/complex/algorithms/harmonic.h
index d004d7b..13d5167 100644
--- a/vcg/complex/algorithms/harmonic.h
+++ b/vcg/complex/algorithms/harmonic.h
@@ -24,7 +24,7 @@
 #define __VCGLIB_HARMONIC_FIELD
 
 #include <vcg/complex/complex.h>
-#include <eigenlib/Eigen/Sparse>
+#include <eigen3/Eigen/Sparse>
 
 namespace vcg {
 namespace tri {
diff --git a/vcg/complex/algorithms/implicit_smooth.h b/vcg/complex/algorithms/implicit_smooth.h
index 2e1e24f..76311e2 100644
--- a/vcg/complex/algorithms/implicit_smooth.h
+++ b/vcg/complex/algorithms/implicit_smooth.h
@@ -23,7 +23,7 @@
 #ifndef __VCG_IMPLICIT_SMOOTHER
 #define __VCG_IMPLICIT_SMOOTHER
 
-#include <eigenlib/Eigen/Sparse>
+#include <eigen3/Eigen/Sparse>
 #include <vcg/complex/algorithms/mesh_to_matrix.h>
 #include <vcg/complex/algorithms/update/quality.h>
 #include <vcg/complex/algorithms/smooth.h>
diff --git a/vcg/complex/algorithms/inertia.h b/vcg/complex/algorithms/inertia.h
index 41b5a29..f25d6f9 100644
--- a/vcg/complex/algorithms/inertia.h
+++ b/vcg/complex/algorithms/inertia.h
@@ -24,8 +24,8 @@
 #define _VCG_INERTIA_
 
 
-#include <eigenlib/Eigen/Core>
-#include <eigenlib/Eigen/Eigenvalues>
+#include <eigen3/Eigen/Core>
+#include <eigen3/Eigen/Eigenvalues>
 #include <vcg/complex/algorithms/update/normal.h>
 
 namespace vcg
diff --git a/vcg/complex/algorithms/parametrization/poisson_solver.h b/vcg/complex/algorithms/parametrization/poisson_solver.h
index cf5c36d..3b8f1a5 100644
--- a/vcg/complex/algorithms/parametrization/poisson_solver.h
+++ b/vcg/complex/algorithms/parametrization/poisson_solver.h
@@ -24,7 +24,7 @@
 #ifndef VCG_POISSON_SOLVER
 #define VCG_POISSON_SOLVER
 
-#include <eigenlib/Eigen/Sparse>
+#include <eigen3/Eigen/Sparse>
 
 #include <vcg/complex/algorithms/clean.h>
 #include <vcg/complex/algorithms/update/bounding.h>
diff --git a/vcg/complex/algorithms/update/curvature.h b/vcg/complex/algorithms/update/curvature.h
index 441121b..0a96146 100644
--- a/vcg/complex/algorithms/update/curvature.h
+++ b/vcg/complex/algorithms/update/curvature.h
@@ -32,7 +32,7 @@
 #include <vcg/complex/algorithms/point_sampling.h>
 #include <vcg/complex/algorithms/intersection.h>
 #include <vcg/complex/algorithms/inertia.h>
-#include <eigenlib/Eigen/Core>
+#include <eigen3/Eigen/Core>
 
 namespace vcg {
 namespace tri {
diff --git a/vcg/complex/algorithms/update/curvature_fitting.h b/vcg/complex/algorithms/update/curvature_fitting.h
index b2b8c63..ab04b6b 100644
--- a/vcg/complex/algorithms/update/curvature_fitting.h
+++ b/vcg/complex/algorithms/update/curvature_fitting.h
@@ -38,11 +38,11 @@
 #include <vcg/complex/algorithms/inertia.h>
 #include <vcg/complex/algorithms/nring.h>
 
-#include <eigenlib/Eigen/Core>
-#include <eigenlib/Eigen/QR>
-#include <eigenlib/Eigen/LU>
-#include <eigenlib/Eigen/SVD>
-#include <eigenlib/Eigen/Eigenvalues>
+#include <eigen3/Eigen/Core>
+#include <eigen3/Eigen/QR>
+#include <eigen3/Eigen/LU>
+#include <eigen3/Eigen/SVD>
+#include <eigen3/Eigen/Eigenvalues>
 
 
 namespace vcg {
diff --git a/vcg/complex/algorithms/update/fitmaps.h b/vcg/complex/algorithms/update/fitmaps.h
index bd2c94f..fd837d7 100644
--- a/vcg/complex/algorithms/update/fitmaps.h
+++ b/vcg/complex/algorithms/update/fitmaps.h
@@ -34,10 +34,10 @@
 #include <vcg/complex/algorithms/update/bounding.h>
 #include "vcg/complex/algorithms/update/curvature_fitting.h"
 
-#include <eigenlib/Eigen/Core>
-#include <eigenlib/Eigen/QR>
-#include <eigenlib/Eigen/LU>
-#include <eigenlib/Eigen/SVD>
+#include <eigen3/Eigen/Core>
+#include <eigen3/Eigen/QR>
+#include <eigen3/Eigen/LU>
+#include <eigen3/Eigen/SVD>
 
 #include <vcg/complex/algorithms/nring.h>
 
diff --git a/vcg/math/eigen.h b/vcg/math/eigen.h
index 5cae402..81f32ff 100644
--- a/vcg/math/eigen.h
+++ b/vcg/math/eigen.h
@@ -53,10 +53,10 @@ struct ei_to_vcgtype;
 }
 
 #include "base.h"
-#include "../../eigenlib/Eigen/LU"
-#include "../../eigenlib/Eigen/Geometry"
-#include "../../eigenlib/Eigen/Array"
-#include "../../eigenlib/Eigen/Core"
+#include "../../eigen3/Eigen/LU"
+#include "../../eigen3/Eigen/Geometry"
+#include "../../eigen3/Eigen/Array"
+#include "../../eigen3/Eigen/Core"
 
 // add support for unsigned char and short int
 namespace Eigen {
diff --git a/vcg/math/matrix44.h b/vcg/math/matrix44.h
index a138e57..5291e0e 100644
--- a/vcg/math/matrix44.h
+++ b/vcg/math/matrix44.h
@@ -30,8 +30,8 @@
 #include <vcg/space/point4.h>
 #include <vector>
 #include <iostream>
-#include <eigenlib/Eigen/Core>
-#include <eigenlib/Eigen/LU>
+#include <eigen3/Eigen/Core>
+#include <eigen3/Eigen/LU>
 
 namespace vcg {
 
diff --git a/vcg/math/polar_decomposition.h b/vcg/math/polar_decomposition.h
index deabf86..bd725d1 100644
--- a/vcg/math/polar_decomposition.h
+++ b/vcg/math/polar_decomposition.h
@@ -26,8 +26,8 @@
 
 #include <vcg/math/matrix33.h>
 #include <vcg/math/matrix44.h>
-#include <eigenlib/Eigen/Dense>
-#include <eigenlib/Eigen/SVD>
+#include <eigen3/Eigen/Dense>
+#include <eigen3/Eigen/SVD>
 
 namespace vcg{
 
diff --git a/vcg/math/quadric.h b/vcg/math/quadric.h
index 8522056..99a9de1 100644
--- a/vcg/math/quadric.h
+++ b/vcg/math/quadric.h
@@ -26,7 +26,7 @@
 #include <vcg/space/point3.h>
 #include <vcg/space/plane3.h>
 #include <vcg/math/matrix33.h>
-#include <eigenlib/Eigen/Core>
+#include <eigen3/Eigen/Core>
 
 namespace vcg {
 namespace math {
diff --git a/vcg/space/fitting3.h b/vcg/space/fitting3.h
index e9e238a..6861969 100644
--- a/vcg/space/fitting3.h
+++ b/vcg/space/fitting3.h
@@ -29,8 +29,8 @@
 #include <vcg/math/matrix44.h>
 #include <vcg/math/matrix33.h>
 
-#include <eigenlib/Eigen/Core>
-#include <eigenlib/Eigen/Eigenvalues>
+#include <eigen3/Eigen/Core>
+#include <eigen3/Eigen/Eigenvalues>
 
 namespace vcg {
 
diff --git a/vcg/space/point_matching.h b/vcg/space/point_matching.h
index e9ac0ff..c6ff376 100644
--- a/vcg/space/point_matching.h
+++ b/vcg/space/point_matching.h
@@ -27,8 +27,8 @@
 #include <vcg/math/quaternion.h>
 #include <vcg/math/matrix44.h>
 
-#include <eigenlib/Eigen/Dense>
-#include <eigenlib/Eigen/Eigenvalues>
+#include <eigen3/Eigen/Dense>
+#include <eigen3/Eigen/Eigenvalues>
 #include  <iostream>
 
 namespace vcg
diff --git a/wrap/igl/smooth_field.h b/wrap/igl/smooth_field.h
index e1208cf..a403d54 100644
--- a/wrap/igl/smooth_field.h
+++ b/wrap/igl/smooth_field.h
@@ -25,7 +25,7 @@
 #define SMOOTHER_FIELD_H
 
 //eigen stuff
-#include <eigenlib/Eigen/Sparse>
+#include <eigen3/Eigen/Sparse>
 
 //vcg stuff
 #include <vcg/complex/algorithms/update/color.h>
