--- openxr.hpp	2020-08-16 14:42:30.845279600 -0700
+++ openxr.hpp	2020-08-16 14:45:19.707229200 -0700
@@ -18249,9 +18249,9 @@
    public:
         // ctor
               SpatialGraphNodeSpaceCreateInfoMSFT (
-                  const SpatialGraphNodeTypeMSFT& nodeType_ = {},
-                  uint8_t nodeId[16]_ = 0,
-                  const Posef& pose_ = {}
+                  const SpatialGraphNodeTypeMSFT& nodeType_,
+                  uint8_t nodeId_[16],
+                  const Posef& pose_
                   )
             :
 
@@ -18259,9 +18259,9 @@
               ) ,
 
               nodeType {nodeType_} ,
-              nodeId {nodeId_} ,
               pose {pose_} 
         {
+            memcpy(nodeId, nodeId_, 16);
         }
         operator const XrSpatialGraphNodeSpaceCreateInfoMSFT&() const { return *reinterpret_cast<const XrSpatialGraphNodeSpaceCreateInfoMSFT*>(this); }
         operator XrSpatialGraphNodeSpaceCreateInfoMSFT &() {
