diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c75b145..386494c 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -89,7 +89,7 @@ if(NOT VULKAN_INCOMPATIBLE)
 endif()
 
 find_package(Threads REQUIRED)
-find_package(JsonCpp)
+find_package(jsoncpp CONFIG REQUIRED)
 
 ### All options defined here
 option(BUILD_LOADER "Build loader" ON)
diff --git a/src/loader/CMakeLists.txt b/src/loader/CMakeLists.txt
index 6a88cf4..0821a3d 100644
--- a/src/loader/CMakeLists.txt
+++ b/src/loader/CMakeLists.txt
@@ -68,7 +68,11 @@ add_library(openxr_loader ${LIBRARY_TYPE}
     ${openxr_loader_RESOURCE_FILE}
 )
 if(BUILD_WITH_SYSTEM_JSONCPP)
-    target_link_libraries(openxr_loader PRIVATE JsonCpp::JsonCpp)
+    if(BUILD_SHARED_LIBS)
+        target_link_libraries(openxr_loader PRIVATE jsoncpp_lib)
+    else()
+        target_link_libraries(openxr_loader PRIVATE jsoncpp_static)
+    endif()
 else()
     target_sources(openxr_loader
         PRIVATE
