diff --git a/Make.Rules b/Make.Rules
index b92de36..f1b506c 100644
--- a/Make.Rules
+++ b/Make.Rules
@@ -66,11 +66,11 @@ DEFINES := -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
 SYSTEM_HEADERS = /usr/include
 
 SUDO := sudo
-CC := $(CROSS_COMPILE)gcc
-LD := $(CC) -Wl,-x -shared
-AR := $(CROSS_COMPILE)ar
-RANLIB := $(CROSS_COMPILE)ranlib
-OBJCOPY := $(CROSS_COMPILE)objcopy
+#CC := $(CROSS_COMPILE)gcc
+LD := $(CC) -Wl,-x -shared
+#AR := $(CROSS_COMPILE)ar
+#RANLIB := $(CROSS_COMPILE)ranlib
+#OBJCOPY := $(CROSS_COMPILE)objcopy
 
 # Reference:
 #   CPPFLAGS used for building .o files from .c & .h files
