diff --git a/src/training/CMakeLists.txt b/src/training/CMakeLists.txt
index af291a5..f2e2ceb 100644
--- a/src/training/CMakeLists.txt
+++ b/src/training/CMakeLists.txt
@@ -63,7 +63,7 @@ endif()
 # LIBRARY tessopt
 ########################################
 
-add_library                 (tessopt tessopt.cpp tessopt.h)
+add_library                 (tessopt STATIC tessopt.cpp tessopt.h)
 project_group               (tessopt "Training Tools")
 
 
@@ -81,7 +81,7 @@ set(common_training_src
     mastertrainer.cpp
     mastertrainer.h
 )
-add_library                 (common_training ${common_training_src})
+add_library                 (common_training STATIC ${common_training_src})
 target_link_libraries       (common_training libtesseract tessopt)
 project_group               (common_training "Training Tools")
 
@@ -196,7 +196,7 @@ set(unicharset_training_src
     validate_javanese.cpp validate_myanmar.cpp validator.cpp
 
 )
-add_library                 (unicharset_training ${unicharset_training_src})
+add_library                 (unicharset_training STATIC ${unicharset_training_src})
 if(UNIX)
     list(APPEND ICU_LIBRARIES ${CMAKE_DL_LIBS})
 endif()
@@ -270,7 +270,7 @@ endif()
 
 find_package(unofficial-cairo CONFIG REQUIRED)
 find_package(unofficial-glib CONFIG REQUIRED)
-find_package(Intl CONFIG REQUIRED)
+find_package(Intl REQUIRED)
 find_package(Fontconfig REQUIRED)
 find_package(Freetype REQUIRED)
 if(UNIX OR BUILD_SHARED_LIBS)
