diff --git a/CMakeLists.txt b/CMakeLists.txt
index c07b827..85d30e6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -171,9 +171,17 @@ else()
 endif()
 
 find_package(OpenCL QUIET)
-find_package(LibArchive)
-if(LibArchive_FOUND)
-    set(HAVE_LIBARCHIVE ON)
+
+if(BUILD_LIBARCHIVE)
+    find_package(LibArchive)
+    if(LibArchive_FOUND)
+        set(HAVE_LIBARCHIVE ON)
+        find_package(BZip2 REQUIRED)
+        find_package(lz4 CONFIG REQUIRED)
+        find_package(LibXml2 REQUIRED)
+        find_package(LZO REQUIRED)
+        find_package(OpenSSL REQUIRED)
+    endif()
 endif()
 
 
@@ -285,6 +293,15 @@ add_definitions(-D_SILENCE_STDEXT_HASH_DEPRECATION_WARNINGS=1)
 
 include_directories(${Leptonica_INCLUDE_DIRS})
 
+if(HAVE_LIBARCHIVE)
+    include_directories(${BZIP2_INCLUDE_DIR})
+    include_directories(${LZ4_INCLUDE_DIR})
+    include_directories(${LIBXML2_INCLUDE_DIRS})
+    include_directories(${LZO_INCLUDE_DIR})
+    include_directories(${OPENSSL_INCLUDE_DIR})
+    include_directories(${LibArchive_INCLUDE_DIRS})
+endif()
+
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 include_directories(src/api)
@@ -405,10 +422,21 @@ if (CPPAN_BUILD)
     file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/TesseractTargets.cmake "include(${CMAKE_CURRENT_BINARY_DIR}/cppan.cmake)\n")
     export(TARGETS libtesseract APPEND FILE ${CMAKE_CURRENT_BINARY_DIR}/TesseractTargets.cmake)
 else()
-    target_link_libraries       (libtesseract PUBLIC
+    if(HAVE_LIBARCHIVE)
+        target_link_libraries       (libtesseract PUBLIC
         ${Leptonica_LIBRARIES}
         ${LibArchive_LIBRARIES}
-    )
+        ${BZIP2_LIBRARY}
+        lz4::lz4
+        ${LIBXML2_LIBRARIES}
+        ${LZO_LIBRARY}
+        OpenSSL::SSL
+        )
+    else()
+        target_link_libraries       (libtesseract PUBLIC
+        ${Leptonica_LIBRARIES}
+        )
+    endif()
     export(TARGETS libtesseract FILE ${CMAKE_CURRENT_BINARY_DIR}/TesseractTargets.cmake)
 endif()
 
