diff --git a/CMakeLists.txt b/CMakeLists.txt
index 16ced1b..ef1b158 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -69,7 +69,7 @@ include(CMake/install_config.cmake)
 add_subdirectory(wrappers)
 
 if (BUILD_EXAMPLES AND BUILD_GLSL_EXTENSIONS)
-    find_package(glfw3 3.3 QUIET)
+    find_package(glfw3 CONFIG REQUIRED)
     if(NOT TARGET glfw)
         message(STATUS "GLFW 3.3 not found; using internal version")
         set(GLFW_INSTALL ON CACHE BOOL "" FORCE)
