diff --git a/include/outcome/experimental/coroutine_support.hpp b/include/outcome/experimental/coroutine_support.hpp
index 21759ede58..6c4b152294 100644
--- a/include/outcome/experimental/coroutine_support.hpp
+++ b/include/outcome/experimental/coroutine_support.hpp
@@ -45,7 +45,11 @@ Distributed under the Boost Software License, Version 1.0.
   OUTCOME_V2_NAMESPACE_END
 
 #ifdef __cpp_exceptions
+#if 0
 #include "status-code/include/system_code_from_exception.hpp"
+#else
+#include <status-code/system_code_from_exception.hpp>
+#endif
 OUTCOME_V2_NAMESPACE_BEGIN
 namespace awaitables
 {
