diff --git a/CMakeLists.txt b/CMakeLists.txt
index c111dac..35b8a43 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required( VERSION 3.3 )
+cmake_minimum_required( VERSION 3.27 )
 
 #declare project
 project( rlottie VERSION 0.2 LANGUAGES C CXX ASM)
@@ -83,9 +83,9 @@ target_link_libraries(rlottie
                      )
 
 if (NOT APPLE AND NOT WIN32)
-    target_link_libraries(rlottie
+    target_link_options(rlottie
                         PRIVATE
-                            "-Wl,--version-script=${CMAKE_SOURCE_DIR}/rlottie.expmap"
+                            $<COMPILE_ONLY:-Wl,--version-script=${CMAKE_SOURCE_DIR}/rlottie.expmap>
                           )
 endif()
 
@@ -124,10 +124,6 @@ if (LOTTIE_ASAN)
     target_link_options(rlottie PUBLIC  -fsanitize=address)
 endif()
 
-if (NOT LIB_INSTALL_DIR)
-    set (LIB_INSTALL_DIR "/usr/lib")
-endif (NOT LIB_INSTALL_DIR)
-
 #declare source and include files
 add_subdirectory(inc)
 add_subdirectory(src)
@@ -139,9 +135,10 @@ if (LOTTIE_TEST)
 endif()
 
 SET(PREFIX ${CMAKE_INSTALL_PREFIX})
-SET(EXEC_DIR ${PREFIX})
-SET(LIBDIR ${LIB_INSTALL_DIR})
-SET(INCDIR ${PREFIX}/include)
+SET(EXEC_DIR bin)
+SET(LIBDIR lib)
+SET(INCDIR include)
+SET(LIB_INSTALL_DIR lib)
 
 CONFIGURE_FILE(${PROJECT_NAME}.pc.in ${PROJECT_NAME}.pc)
 INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.pc DESTINATION ${LIB_INSTALL_DIR}/pkgconfig)
@@ -156,8 +153,8 @@ install(FILES
 
 #install lib
 install( TARGETS rlottie EXPORT rlottie-targets
-         LIBRARY     DESTINATION    ${LIB_INSTALL_DIR}
-         ARCHIVE     DESTINATION    ${LIB_INSTALL_DIR}
+         LIBRARY     DESTINATION    lib
+         ARCHIVE     DESTINATION    lib
          INCLUDES    DESTINATION    include
        )
 
@@ -166,7 +163,7 @@ install( TARGETS rlottie EXPORT rlottie-targets
 install( EXPORT rlottie-targets
          FILE          rlottieTargets.cmake
          NAMESPACE     rlottie::
-         DESTINATION   ${LIB_INSTALL_DIR}/cmake/rlottie
+         DESTINATION   share/rlottie
        )
 
 
@@ -180,14 +177,14 @@ write_basic_package_version_file(
 
 configure_package_config_file(${CMAKE_CURRENT_LIST_DIR}/cmake/rlottieConfig.cmake.in
     ${CMAKE_CURRENT_BINARY_DIR}/rlottieConfig.cmake
-    INSTALL_DESTINATION ${LIB_INSTALL_DIR}/cmake/rlottie
+    INSTALL_DESTINATION share/rlottie
 )
 
 #Install the config, configversion and custom find modules
 install(FILES
     ${CMAKE_CURRENT_BINARY_DIR}/rlottieConfig.cmake
     ${CMAKE_CURRENT_BINARY_DIR}/rlottieConfigVersion.cmake
-    DESTINATION ${LIB_INSTALL_DIR}/cmake/rlottie
+    DESTINATION share/rlottie
 )
 
 
