diff --git a/CMakeLists.txt b/CMakeLists.txt
index 65484a7..a6c02c2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -75,6 +75,10 @@ function(config_library target)
         FILE llhttp-config.cmake
         NAMESPACE llhttp::
         DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/llhttp)
+  target_include_directories(${target}
+    PRIVATE include ${CMAKE_CURRENT_BINARY_DIR}
+    INTERFACE $<INSTALL_INTERFACE:include>
+)
 endfunction(config_library target)
 
 if(BUILD_SHARED_LIBS)
