diff --git a/SMP/libgpg-error_winrt.vcxproj b/SMP/llibgpg-error_winrt.vcxproj
index e1acf679..f9c302be 100644
--- a/SMP/libgpg-error_winrt.vcxproj
+++ b/SMP/libgpg-error_winrt.vcxproj
@@ -3,6 +3,7 @@
   <PropertyGroup Label="Globals">
     <ProjectGuid>{AA4A762D-A8D7-4F68-9B79-FBE63C6BE105}</ProjectGuid>
     <RootNamespace>gpg-error</RootNamespace>
+    <TargetPlatformMinVersion>10.0.17200.0</TargetPlatformMinVersion>
   </PropertyGroup>
   <ImportGroup Label="PropertySheets">
     <Import Project="smp_winrt.props" />
