diff --git a/Makefile.am b/Makefile.am
index 9f92a4f..e35629d 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -116,7 +116,7 @@ LIBGMPXX_LT_REVISION = 1
 LIBGMPXX_LT_AGE      = 6
 
 
-SUBDIRS = tests mpn mpz mpq mpf printf scanf rand cxx demos tune doc
+SUBDIRS = mpn mpz mpq mpf printf rand cxx
 
 EXTRA_DIST = configfsf.guess configfsf.sub .gdbinit INSTALL.autoconf \
 	     COPYING.LESSERv3 COPYINGv2 COPYINGv3
