From 653dae652bac66e4e96a94ba8ebfabb2227738b7 Mon Sep 17 00:00:00 2001
From: Daniel Olivier <dan_olivier@hotmail.com>
Date: Thu, 19 Oct 2017 21:26:09 -0400
Subject: [PATCH] Use external Glew and Freetype


diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4eb55e9..2546de6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -74,13 +74,6 @@ if(NOT HAVE_M_PI)
     endif()
 endif()
 
-if(NOT MINGW AND (WIN32 OR WIN64))
-    set(GLEW_ROOT_DIR
-        ${CMAKE_CURRENT_SOURCE_DIR}/windows/glew)
-    set(ENV{FREETYPE_DIR}
-        ${CMAKE_CURRENT_SOURCE_DIR}/windows/freetype)
-endif()
-
 find_package(OpenGL REQUIRED)
 find_package(Freetype REQUIRED)
 