diff -ur MUMPS_5.4.0/libseq/mpic.c MUMPS/libseq/mpic.c
--- MUMPS_5.4.0/libseq/mpic.c	2021-04-13 17:26:34.000000000 +0200
+++ MUMPS/libseq/mpic.c	2021-05-12 16:25:49.708906988 +0200
@@ -13,7 +13,7 @@
  *  https://cecill.info/licences/Licence_CeCILL-C_V1-en.html)
  *
  */
-#include "mpi.h"
+#include "mumps_mpi.h"
 #include "elapse.h"
 MPI_Comm LIBSEQ_CALL MPI_Comm_f2c(LIBSEQ_INT comm)
 {
   return 0;
diff -ur MUMPS_5.4.0/src/mumps_metis64.h MUMPS/src/mumps_metis64.h
--- MUMPS_5.4.0/src/mumps_metis64.h	2021-04-13 17:26:35.000000000 +0200
+++ MUMPS/src/mumps_metis64.h	2021-05-12 16:25:49.708906988 +0200
@@ -18,7 +18,11 @@
 /* Interfacing with 64-bit (par)metis, for METIS 4 or METIS 5 */
 #include "mumps_common.h" /* includes mumps_compat.h and mumps_c_types.h */
 #if defined(parmetis) || defined(parmetis3)
+#ifdef MPI
 #include "mpi.h"
+#else
+#include "mumps_mpi.h"
+#endif
 #define MUMPS_PARMETIS_64 \
   F_SYMBOL(parmetis_64,PARMETIS_64)
 void MUMPS_CALL
diff -ur MUMPS_5.4.0/src/mumps_metis.h MUMPS/src/mumps_metis.h
--- MUMPS_5.4.0/src/mumps_metis.h	2021-04-13 17:26:35.000000000 +0200
+++ MUMPS/src/mumps_metis.h	2021-05-12 16:25:49.708906988 +0200
@@ -18,7 +18,11 @@
 /* Interfacing with 32-bit (par)metis, for METIS 4 or METIS 5 */
 #include "mumps_common.h" /* includes mumps_compat.h and mumps_c_types.h */
 #if defined(parmetis) || defined(parmetis3)
+#ifdef MPI
 #include "mpi.h"
+#else
+#include "mumps_mpi.h"
+#endif
 #define MUMPS_PARMETIS \
   F_SYMBOL(parmetis,PARMETIS)
 void MUMPS_CALL
diff -ur MUMPS_5.4.0/src/mumps_scotch.h MUMPS/src/mumps_scotch.h
--- MUMPS_5.4.0/src/mumps_scotch.h	2021-04-13 17:26:35.000000000 +0200
+++ MUMPS/src/mumps_scotch.h	2021-05-12 16:25:49.708906988 +0200
@@ -82,7 +82,11 @@
 MUMPS_SCOTCH_SET_PTHREAD_NUMBER (MUMPS_INT *PTHREAD_NUMBER);
 #endif /*scotch or ptscotch*/
 #if defined(ptscotch)
+#ifdef MPI
 #include "mpi.h"
+#else
+#include "mumps_mpi.h"
+#endif
 #include "ptscotch.h"
 #define MUMPS_DGRAPHINIT \
   F_SYMBOL(dgraphinit,DGRAPHINIT)
