diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1ee66f1..7cec7ca 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -392,8 +392,9 @@ endif()
 set(LAPACK_INSTALL_EXPORT_NAME ${LAPACK_INSTALL_EXPORT_NAME_CACHE})
 unset(LAPACK_INSTALL_EXPORT_NAME_CACHE)
 
-add_subdirectory(LAPACKE)
-
+if(LAPACKE)
+    add_subdirectory(LAPACKE)
+endif()
 
 #-------------------------------------
 # BLAS++ / LAPACK++
