diff --git a/include/pqxx/tablereader.hxx b/include/pqxx/tablereader.hxx
index f5300cb..0cc0e43 100644
--- a/include/pqxx/tablereader.hxx
+++ b/include/pqxx/tablereader.hxx
@@ -34,13 +34,13 @@ public:
 	const std::string &Name,
 	const std::string &Null=std::string{});
   template<typename ITER>
-  PQXX_DEPRECATED tablereader(
+  tablereader(
 	transaction_base &,
 	const std::string &Name,
 	ITER begincolumns,
 	ITER endcolumns);
   template<typename ITER>
-  PQXX_DEPRECATED tablereader(
+  tablereader(
 	transaction_base &,
 	const std::string &Name,
 	ITER begincolumns,
diff --git a/include/pqxx/tablewriter.hxx b/include/pqxx/tablewriter.hxx
index 32e7a98..a25b356 100644
--- a/include/pqxx/tablewriter.hxx
+++ b/include/pqxx/tablewriter.hxx
@@ -36,13 +36,13 @@ public:
 	const std::string &WName,
 	const std::string &Null=std::string{});
   template<typename ITER>
-        PQXX_DEPRECATED tablewriter(
+        tablewriter(
 	transaction_base &,
 	const std::string &WName,
 	ITER begincolumns,
 	ITER endcolumns);
   template<typename ITER>
-        PQXX_DEPRECATED tablewriter(
+        tablewriter(
 	transaction_base &T,
 	const std::string &WName,
 	ITER begincolumns,
