diff --git a/Modules/ThirdParty/MINC/src/libminc/CMakeLists.txt b/Modules/ThirdParty/MINC/src/libminc/CMakeLists.txt
index 305b2c63..63f81a53 100644
--- a/Modules/ThirdParty/MINC/src/libminc/CMakeLists.txt
+++ b/Modules/ThirdParty/MINC/src/libminc/CMakeLists.txt
@@ -550,7 +550,7 @@ IF(LIBMINC_INSTALL_LIB_DIR AND NOT LIBMINC_INSTALL_NO_DEVELOPMENT)
      ${CMAKE_CURRENT_BINARY_DIR}/CMakeFiles/Use${LIBMINC_EXTERNAL_LIB_PREFIX}LIBMINC.cmake 
      ${CMAKE_CURRENT_BINARY_DIR}/CMakeFiles/${LIBMINC_EXTERNAL_LIB_PREFIX}LIBMINCConfig.cmake
     DESTINATION 
-     ${LIBMINC_INSTALL_LIB_DIR}/cmake
+     share/itk
     COMPONENT Development)
 ENDIF(LIBMINC_INSTALL_LIB_DIR AND NOT LIBMINC_INSTALL_NO_DEVELOPMENT)
 
