diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1267b1a..a17eeb9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -60,16 +60,6 @@ endif()
 include(itkCompilerChecks)
 include(itkSupportMacros)
 
-# Configure CMake variables that will be used in the module macros
-# ITK_USE_FILE is used when a remote module includes an example directory
-# that can be configured as an independent project.
-set(ITK_CONFIG_CMAKE_DIR "${ITK_SOURCE_DIR}/CMake")
-set(ITK_USE_FILE "${ITK_CONFIG_CMAKE_DIR}/UseITK.cmake")
-# ITKInternalConfig.cmake is used to handle find_package(ITK) calls. This is useful
-# to remote module examples which are set up as independent projects that can be copied
-# outside of their original project and used without any modification.
-configure_file(CMake/ITKInternalConfig.cmake ${ITK_BINARY_DIR}/CMakeTmp/ITKConfig.cmake COPYONLY)
-
 include(CMakeDependentOption)
 #
 # use ExternalProject
@@ -140,6 +130,16 @@ OFF (default) means link to regular, dynamic CRT (/MD and /MDd)." OFF)
   endif()
 endif()
 
+# Configure CMake variables that will be used in the module macros
+# ITK_USE_FILE is used when a remote module includes an example directory
+# that can be configured as an independent project.
+set(ITK_CONFIG_CMAKE_DIR "${ITK_SOURCE_DIR}/CMake")
+set(ITK_USE_FILE "${ITK_CONFIG_CMAKE_DIR}/UseITK.cmake")
+# ITKInternalConfig.cmake is used to handle find_package(ITK) calls. This is useful
+# to remote module examples which are set up as independent projects that can be copied
+# outside of their original project and used without any modification.
+configure_file(CMake/ITKInternalConfig.cmake ${ITK_BINARY_DIR}/CMakeTmp/ITKConfig.cmake COPYONLY)
+
 include(ITKDownloadSetup)
 include(PreventInSourceBuilds)
 include(PreventInBuildInstalls)
