diff --git a/Modules/Core/GPUCommon/CMakeLists.txt b/Modules/Core/GPUCommon/CMakeLists.txt
index da2d66b63..6fb476680 100644
--- a/Modules/Core/GPUCommon/CMakeLists.txt
+++ b/Modules/Core/GPUCommon/CMakeLists.txt
@@ -24,9 +24,13 @@ if(ITK_USE_GPU AND APPLE AND NOT ITK_COMPILER_HAS_BLOCKS)
 endif()
 
 if(ITK_USE_GPU)
+  message(STATUS "OPENCL_LIBRARIES:${OPENCL_LIBRARIES}")
+  if(TARGET OpenCL::OpenCL)
+	message(STATUS "OPENCL_LIBRARIES:${OpenCL_LIBRARIES}")
+  endif()
   set(ITKGPUCommon_LIBRARIES ITKGPUCommon)
-  set(ITKGPUCommon_SYSTEM_INCLUDE_DIRS ${OPENCL_INCLUDE_DIRS})
-  set(ITKGPUCommon_SYSTEM_LIBRARY_DIRS ${OPENCL_LIBRARIES})
+  set(ITKGPUCommon_SYSTEM_INCLUDE_DIRS ${OpenCL_INCLUDE_DIRS})
+  set(ITKGPUCommon_SYSTEM_LIBRARY_DIRS ${OpenCL_LIBRARIES})
 endif()
 
 if(NOT ITK_SOURCE_DIR)
diff --git a/Modules/Core/GPUCommon/src/CMakeLists.txt b/Modules/Core/GPUCommon/src/CMakeLists.txt
index 0b39f2a9a..160f7a752 100644
--- a/Modules/Core/GPUCommon/src/CMakeLists.txt
+++ b/Modules/Core/GPUCommon/src/CMakeLists.txt
@@ -14,5 +14,5 @@ if (ITK_USE_GPU)
   write_gpu_kernels("${ITKGPUCommon_Kernels}" ITKGPUCommon_SRCS)
 
   itk_module_add_library(ITKGPUCommon ${ITKGPUCommon_SRCS})
-  target_link_libraries(ITKGPUCommon LINK_PUBLIC ${OPENCL_LIBRARIES})
+  target_link_libraries(ITKGPUCommon LINK_PUBLIC ${OpenCL_LIBRARIES})
 endif()
diff --git a/Modules/Core/GPUFiniteDifference/CMakeLists.txt b/Modules/Core/GPUFiniteDifference/CMakeLists.txt
index c1bfde450..a642d9230 100644
--- a/Modules/Core/GPUFiniteDifference/CMakeLists.txt
+++ b/Modules/Core/GPUFiniteDifference/CMakeLists.txt
@@ -9,8 +9,8 @@ if(NOT ITK_SOURCE_DIR)
 endif()
 
 if (ITK_USE_GPU)
-  set(ITKGPUFiniteDifference_SYSTEM_INCLUDE_DIRS ${OPENCL_INCLUDE_DIRS})
-  set(ITKGPUFiniteDifference_SYSTEM_LIBRARY_DIRS ${OPENCL_LIBRARIES})
+  set(ITKGPUFiniteDifference_SYSTEM_INCLUDE_DIRS ${OpenCL_INCLUDE_DIRS})
+  set(ITKGPUFiniteDifference_SYSTEM_LIBRARY_DIRS ${OpenCL_LIBRARIES})
   set(ITKGPUFiniteDifference_LIBRARIES ITKGPUFiniteDifference)
 endif()
 
diff --git a/Modules/Core/GPUFiniteDifference/src/CMakeLists.txt b/Modules/Core/GPUFiniteDifference/src/CMakeLists.txt
index 8425cc025..8335c69a0 100644
--- a/Modules/Core/GPUFiniteDifference/src/CMakeLists.txt
+++ b/Modules/Core/GPUFiniteDifference/src/CMakeLists.txt
@@ -9,5 +9,5 @@ if (ITK_USE_GPU)
   write_gpu_kernels("${ITKGPUFiniteDifference_Kernels}" ITKGPUFiniteDifference_SRCS)
 
   itk_module_add_library(ITKGPUFiniteDifference ${ITKGPUFiniteDifference_SRCS})
-  target_link_libraries(ITKGPUFiniteDifference LINK_PUBLIC ${OPENCL_LIBRARIES})
+  target_link_libraries(ITKGPUFiniteDifference LINK_PUBLIC ${OpenCL_LIBRARIES})
 endif()
diff --git a/Modules/Filtering/GPUAnisotropicSmoothing/CMakeLists.txt b/Modules/Filtering/GPUAnisotropicSmoothing/CMakeLists.txt
index f36fd887d..310d983fa 100644
--- a/Modules/Filtering/GPUAnisotropicSmoothing/CMakeLists.txt
+++ b/Modules/Filtering/GPUAnisotropicSmoothing/CMakeLists.txt
@@ -10,8 +10,8 @@ endif()
 
 if (ITK_USE_GPU)
   set(ITKGPUAnisotropicSmoothing_LIBRARIES ITKGPUAnisotropicSmoothing)
-  set(ITKGPUAnisotropicSmoothing_SYSTEM_INCLUDE_DIRS ${OPENCL_INCLUDE_DIRS})
-  set(ITKGPUAnisotropicSmoothing_SYSTEM_LIBRARY_DIRS ${OPENCL_LIBRARIES})
+  set(ITKGPUAnisotropicSmoothing_SYSTEM_INCLUDE_DIRS ${OpenCL_INCLUDE_DIRS})
+  set(ITKGPUAnisotropicSmoothing_SYSTEM_LIBRARY_DIRS ${OpenCL_LIBRARIES})
 endif()
 
 if(NOT ITK_SOURCE_DIR)
diff --git a/Modules/Filtering/GPUImageFilterBase/CMakeLists.txt b/Modules/Filtering/GPUImageFilterBase/CMakeLists.txt
index 078905521..67fd7b832 100644
--- a/Modules/Filtering/GPUImageFilterBase/CMakeLists.txt
+++ b/Modules/Filtering/GPUImageFilterBase/CMakeLists.txt
@@ -10,8 +10,8 @@ endif()
 
 if (ITK_USE_GPU)
   set(ITKGPUImageFilterBase_LIBRARIES ITKGPUImageFilterBase)
-  set(ITKGPUImageFilterBase_SYSTEM_INCLUDE_DIRS ${OPENCL_INCLUDE_DIRS})
-  set(ITKGPUImageFilterBase_SYSTEM_LIBRARY_DIRS ${OPENCL_LIBRARIES})
+  set(ITKGPUImageFilterBase_SYSTEM_INCLUDE_DIRS ${OpenCL_INCLUDE_DIRS})
+  set(ITKGPUImageFilterBase_SYSTEM_LIBRARY_DIRS ${OpenCL_LIBRARIES})
 endif()
 
 if(NOT ITK_SOURCE_DIR)
diff --git a/Modules/Filtering/GPUImageFilterBase/src/CMakeLists.txt b/Modules/Filtering/GPUImageFilterBase/src/CMakeLists.txt
index 251493f5e..6d33fa964 100644
--- a/Modules/Filtering/GPUImageFilterBase/src/CMakeLists.txt
+++ b/Modules/Filtering/GPUImageFilterBase/src/CMakeLists.txt
@@ -10,5 +10,5 @@ if (ITK_USE_GPU)
   write_gpu_kernels("${ITKGPUImageFilterBase_Kernels}" ITKGPUImageFilterBase_SRCS)
 
   itk_module_add_library(ITKGPUImageFilterBase ${ITKGPUImageFilterBase_SRCS})
-  target_link_libraries(ITKGPUImageFilterBase LINK_PUBLIC ${OPENCL_LIBRARIES})
+  target_link_libraries(ITKGPUImageFilterBase LINK_PUBLIC ${OpenCL_LIBRARIES})
 endif()
diff --git a/Modules/Filtering/GPUSmoothing/CMakeLists.txt b/Modules/Filtering/GPUSmoothing/CMakeLists.txt
index 9bac40921..2ed493d40 100644
--- a/Modules/Filtering/GPUSmoothing/CMakeLists.txt
+++ b/Modules/Filtering/GPUSmoothing/CMakeLists.txt
@@ -10,8 +10,8 @@ endif()
 
 if (ITK_USE_GPU)
   set(ITKGPUSmoothing_LIBRARIES ITKGPUSmoothing)
-  set(ITKGPUSmoothing_SYSTEM_INCLUDE_DIRS ${OPENCL_INCLUDE_DIRS})
-  set(ITKGPUSmoothing_SYSTEM_LIBRARY_DIRS ${OPENCL_LIBRARIES})
+  set(ITKGPUSmoothing_SYSTEM_INCLUDE_DIRS ${OpenCL_INCLUDE_DIRS})
+  set(ITKGPUSmoothing_SYSTEM_LIBRARY_DIRS ${OpenCL_LIBRARIES})
 endif()
 
 if(NOT ITK_SOURCE_DIR)
diff --git a/Modules/Filtering/GPUSmoothing/src/CMakeLists.txt b/Modules/Filtering/GPUSmoothing/src/CMakeLists.txt
index d2120e842..2c6ca4a4a 100644
--- a/Modules/Filtering/GPUSmoothing/src/CMakeLists.txt
+++ b/Modules/Filtering/GPUSmoothing/src/CMakeLists.txt
@@ -9,5 +9,5 @@ if (ITK_USE_GPU)
   write_gpu_kernels("${ITKGPUSmoothing_Kernels}" ITKGPUSmoothing_SRCS)
 
   itk_module_add_library(ITKGPUSmoothing ${ITKGPUSmoothing_SRCS})
-  target_link_libraries(ITKGPUSmoothing LINK_PUBLIC ${OPENCL_LIBRARIES})
+  target_link_libraries(ITKGPUSmoothing LINK_PUBLIC ${OpenCL_LIBRARIES})
 endif()
diff --git a/Modules/Filtering/GPUThresholding/CMakeLists.txt b/Modules/Filtering/GPUThresholding/CMakeLists.txt
index 4390785f5..61786b7a8 100644
--- a/Modules/Filtering/GPUThresholding/CMakeLists.txt
+++ b/Modules/Filtering/GPUThresholding/CMakeLists.txt
@@ -2,8 +2,8 @@ project(ITKGPUThresholding)
 
 if (ITK_USE_GPU)
   set(ITKGPUThresholding_LIBRARIES ITKGPUThresholding)
-  set(ITKGPUThresholding_SYSTEM_INCLUDE_DIRS ${OPENCL_INCLUDE_DIRS})
-  set(ITKGPUThresholding_SYSTEM_LIBRARY_DIRS ${OPENCL_LIBRARIES})
+  set(ITKGPUThresholding_SYSTEM_INCLUDE_DIRS ${OpenCL_INCLUDE_DIRS})
+  set(ITKGPUThresholding_SYSTEM_LIBRARY_DIRS ${OpenCL_LIBRARIES})
 endif()
 
 itk_module_impl()
diff --git a/Modules/Filtering/GPUThresholding/src/CMakeLists.txt b/Modules/Filtering/GPUThresholding/src/CMakeLists.txt
index 326c86859..c07c80414 100644
--- a/Modules/Filtering/GPUThresholding/src/CMakeLists.txt
+++ b/Modules/Filtering/GPUThresholding/src/CMakeLists.txt
@@ -9,5 +9,5 @@ if (ITK_USE_GPU)
   write_gpu_kernels("${ITKGPUThresholding_Kernels}" ITKGPUThresholding_SRCS)
 
   itk_module_add_library(ITKGPUThresholding ${ITKGPUThresholding_SRCS})
-  target_link_libraries(ITKGPUThresholding LINK_PUBLIC ${OPENCL_LIBRARIES})
+  target_link_libraries(ITKGPUThresholding LINK_PUBLIC ${OpenCL_LIBRARIES})
 endif()
diff --git a/Modules/Registration/GPUPDEDeformable/CMakeLists.txt b/Modules/Registration/GPUPDEDeformable/CMakeLists.txt
index c8d70a423..097c974c5 100644
--- a/Modules/Registration/GPUPDEDeformable/CMakeLists.txt
+++ b/Modules/Registration/GPUPDEDeformable/CMakeLists.txt
@@ -2,8 +2,8 @@ project(ITKGPUPDEDeformableRegistration)
 
 if (ITK_USE_GPU)
   set(ITKGPUPDEDeformableRegistration_LIBRARIES ITKGPUPDEDeformableRegistration)
-  set(ITKGPUPDEDeformableRegistration_SYSTEM_INCLUDE_DIRS ${OPENCL_INCLUDE_DIRS})
-  set(ITKGPUPDEDeformableRegistration_SYSTEM_LIBRARY_DIRS ${OPENCL_LIBRARIES})
+  set(ITKGPUPDEDeformableRegistration_SYSTEM_INCLUDE_DIRS ${OpenCL_INCLUDE_DIRS})
+  set(ITKGPUPDEDeformableRegistration_SYSTEM_LIBRARY_DIRS ${OpenCL_LIBRARIES})
 endif()
 
 itk_module_impl()
diff --git a/Modules/Registration/GPUPDEDeformable/src/CMakeLists.txt b/Modules/Registration/GPUPDEDeformable/src/CMakeLists.txt
index 28b1d194c..1a049bdde 100644
--- a/Modules/Registration/GPUPDEDeformable/src/CMakeLists.txt
+++ b/Modules/Registration/GPUPDEDeformable/src/CMakeLists.txt
@@ -10,5 +10,5 @@ if (ITK_USE_GPU)
   write_gpu_kernels("${ITKGPUPDEDeformableRegistration_Kernels}" ITKGPUPDEDeformableRegistration_SRCS)
 
   itk_module_add_library(ITKGPUPDEDeformableRegistration ${ITKGPUPDEDeformableRegistration_SRCS})
-  target_link_libraries(ITKGPUPDEDeformableRegistration LINK_PUBLIC ${OPENCL_LIBRARIES})
+  target_link_libraries(ITKGPUPDEDeformableRegistration LINK_PUBLIC ${OpenCL_LIBRARIES})
 endif()
