--- a/cmake/macros/SundialsAddLibrary.cmake
+++ b/cmake/macros/SundialsAddLibrary.cmake
@@ -323,7 +323,7 @@
         set_target_properties(${_actual_target_name} PROPERTIES ${_properties})
       endif()
 
       # install phase
-      install(TARGETS ${_actual_target_name} DESTINATION ${CMAKE_INSTALL_LIBDIR} EXPORT sundials-targets)
+      install(TARGETS ${_actual_target_name} EXPORT sundials-targets)
 
     endif()
