diff --git a/CMakeLists.txt b/CMakeLists.txt
index 585eb0e..7515d98 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,8 +28,7 @@ enable_language(CXX)
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake/Modules/" ${CMAKE_MODULE_PATH})
 
-set(CMAKE_INSTALL_PREFIX "${CMAKE_CURRENT_LIST_DIR}" CACHE PATH "Install prefix" FORCE)
-set(INSTALL_BIN_DIR      "${CMAKE_CURRENT_LIST_DIR}" CACHE PATH "Path where exe and dll will be installed")
+set(INSTALL_BIN_DIR      "bin"                       CACHE PATH "Path where exe and dll will be installed")
 set(INSTALL_LIB_DIR      "lib"                       CACHE PATH "Path where lib will be installed")
 set(INSTALL_INCLUDE_DIR  "include"                   CACHE PATH "Path where headers will be installed")
 set(INSTALL_CMAKE_DIR    "share/darknet"             CACHE PATH "Path where cmake configs will be installed")
@@ -88,7 +87,6 @@ if(USE_INTEGRATED_LIBS)
   set(PThreads_windows_DIR ${CMAKE_CURRENT_LIST_DIR}/3rdparty/pthreads CACHE PATH "Path where pthreads for windows can be located")
   add_definitions(-D_TIMESPEC_DEFINED)
 endif()
-set(Stb_DIR ${CMAKE_CURRENT_LIST_DIR}/3rdparty/stb CACHE PATH "Path where Stb image library can be located")
 
 set(CMAKE_DEBUG_POSTFIX d)
 add_definitions(-DUSE_CMAKE_LIBS)
