diff --git a/CMakeLists.txt b/CMakeLists.txt
index a77d982..d65e65e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -184,6 +184,10 @@ include(QuickCppLibApplyDefaultDefinitions)
 
 # Set the C++ features this library requires
 all_compile_features(PUBLIC cxx_std_17)
+# for checking __cplusplus
+if(MSVC)
+  all_compile_options(PRIVATE /Zc:__cplusplus)
+endif()
 set(check_cxx_source_linkage_flags)
 if(NOT CMAKE_CXX_STANDARD)
   if(MSVC)
