diff --git a/Import/CMakeLists.txt b/Import/CMakeLists.txt
index 3547dae..23d6c39 100644
--- a/Import/CMakeLists.txt
+++ b/Import/CMakeLists.txt
@@ -184,7 +184,6 @@ if (GACUI_CORE)
 
     if (GACUI_REFLECTION)
         install(FILES GacUIReflection.h DESTINATION include)
-        install(FILES GacUIR.UnitTest.UIReflection.h DESTINATION include)
         install(FILES Skins/DarkSkin/DarkSkinReflection.h DESTINATION include/Skins/DarkSkin)
     endif()
     
