diff --git a/Import/Vlpp.h b/Import/Vlpp.h
index cf59b80..b00c9b8 100644
--- a/Import/Vlpp.h
+++ b/Import/Vlpp.h
@@ -49,10 +49,10 @@ static_assert(sizeof(wchar_t) == sizeof(char32_t), "wchar_t is not UTF-32.");
 static_assert(false, "wchar_t configuration is not right.");
 #endif
 
+#include <stdlib.h>
 #if defined VCZH_MSVC
 #include <intrin.h>
 #elif defined VCZH_GCC
-#include <x86intrin.h>
 #include <stdint.h>
 #include <stddef.h>
 #include <wchar.h>
@@ -149,8 +149,8 @@ x86 and x64 Compatbility
 #define INCRC(x)	(_InterlockedIncrement64(x))
 #define DECRC(x)	(_InterlockedDecrement64(x))
 #elif defined VCZH_GCC
-#define INCRC(x)	(__sync_add_and_fetch(x, 1))
-#define DECRC(x)	(__sync_sub_and_fetch(x, 1))
+#define INCRC(x)	(__atomic_add_fetch(x, 1, __ATOMIC_SEQ_CST))
+#define DECRC(x)	(__atomic_sub_fetch(x, 1, __ATOMIC_SEQ_CST))
 #endif
 #else
 #define ITOA_S		_itoa_s
