diff --git a/tests/fuzz/CMakeLists.txt b/tests/fuzz/CMakeLists.txt
index 81000dc..cf9f909 100644
--- a/tests/fuzz/CMakeLists.txt
+++ b/tests/fuzz/CMakeLists.txt
@@ -52,8 +52,8 @@ foreach(source ${SOURCES})
         target_link_libraries(${target} ${FUZZING_ENGINE})
     endif()
 
-    target_link_libraries(${target} blosc_static)
-    add_dependencies(${target} blosc_static)
+    target_link_libraries(${target} blosc)
+    add_dependencies(${target} blosc)
 
     # run standalone fuzzer against each file
     file(GLOB COMPAT_FILES ${PROJECT_SOURCE_DIR}/compat/*.cdata)
