diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index f6eb01b..25b23c8 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -192,6 +192,18 @@ else()
       VERSION "${OPENJPH_VERSION}")
 endif()
 
+if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+  set(OJPH_PKG_CONFIG_DEBUG_POSTFIX "${CMAKE_DEBUG_POSTFIX}")
+else()
+  set(OJPH_PKG_CONFIG_DEBUG_POSTFIX "")
+endif()
+
+if(MSVC)
+  set(OJPH_PKG_CONFIG_LIB_NAME "${OJPH_LIB_NAME_STRING}${OJPH_PKG_CONFIG_DEBUG_POSTFIX}" PARENT_SCOPE)
+else()
+  set(OJPH_PKG_CONFIG_LIB_NAME "openjph${OJPH_PKG_CONFIG_DEBUG_POSTFIX}" PARENT_SCOPE)
+endif()
+
 install(TARGETS openjph
   EXPORT openjph-targets
 )
diff --git a/src/openjph.pc.in b/src/openjph.pc.in
index 94ea3ab..06d8c2f 100644
--- a/src/openjph.pc.in
+++ b/src/openjph.pc.in
@@ -6,5 +6,5 @@ Name: @PROJECT_NAME@
 Description: @PROJECT_DESCRIPTION@
 Version: @PROJECT_VERSION@
 Requires: @PKG_CONFIG_REQUIRES@
-Libs: -L${libdir} -lopenjph
+Libs: -L${libdir} -l@OJPH_PKG_CONFIG_LIB_NAME@
 Cflags: -I${includedir} -D_FILE_OFFSET_BITS=64
