diff --git a/cmake/ChooseCudaCC.cmake b/cmake/ChooseCudaCC.cmake
index d9bc6c2..a20ed38 100755
--- a/cmake/ChooseCudaCC.cmake
+++ b/cmake/ChooseCudaCC.cmake
@@ -78,10 +78,23 @@ function(chooseCudaCC SUPPORTED_CC SUPPORTED_GENCODE_FLAGS)
   # Default setting of the CUDA CC versions to compile.
   # Shortening the lists saves a lot of compile time.
   #
-  set(CUDA_MIN_CC 20)
-  set(CUDA_MAX_CC 86)
-  if(CUDA_VERSION_MAJOR GREATER_EQUAL 11)
+
+  # The current version last time this list was updated was CUDA 12.1.
+  if(CUDA_VERSION VERSION_GREATER_EQUAL 12)
+    set(CUDA_MIN_CC 50)
+    set(CUDA_MAX_CC 90)
+  elseif(CUDA_VERSION VERSION_GREATER_EQUAL 11.8)
+    set(CUDA_MIN_CC 35)
+    set(CUDA_MAX_CC 90)
+  elseif(CUDA_VERSION VERSION_GREATER_EQUAL 11.6)
+    set(CUDA_MIN_CC 35)
+    set(CUDA_MAX_CC 87)
+  elseif(CUDA_VERSION VERSION_GREATER_EQUAL 11.1)
+    set(CUDA_MIN_CC 35)
+    set(CUDA_MAX_CC 86)
+  elseif(CUDA_VERSION_MAJOR GREATER_EQUAL 11)
     set(CUDA_MIN_CC 35)
+    set(CUDA_MAX_CC 80)
   elseif(CUDA_VERSION_MAJOR GREATER_EQUAL 10)
     set(CUDA_MIN_CC 30)
     set(CUDA_MAX_CC 75)
@@ -89,8 +102,10 @@ function(chooseCudaCC SUPPORTED_CC SUPPORTED_GENCODE_FLAGS)
     set(CUDA_MIN_CC 30)
     set(CUDA_MAX_CC 72)
   elseif(CUDA_VERSION_MAJOR GREATER_EQUAL 8)
+    set(CUDA_MIN_CC 20)
     set(CUDA_MAX_CC 62)
   elseif(CUDA_VERSION_MAJOR GREATER_EQUAL 7)
+    set(CUDA_MIN_CC 20)
     set(CUDA_MAX_CC 53)
   else()
     message(FATAL_ERROR "We do not support a CUDA SDK below version 7.0")
