diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5d27c57c4..7f37e5b5d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -148,13 +148,13 @@ find_package(fmt REQUIRED)
 
 # === Target: vcpkglib ===
 
-add_library(vcpkglib OBJECT
+add_library(vcpkglib STATIC
     ${VCPKGLIB_BASE_SOURCES}
     ${VCPKGLIB_SOURCES}
     ${VCPKGLIB_BASE_INCLUDES}
     ${VCPKGLIB_INCLUDES}
 )
-target_include_directories(vcpkglib PUBLIC include)
+target_include_directories(vcpkglib PUBLIC $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>)

 vcpkg_target_add_warning_options(vcpkglib)
 target_compile_definitions(vcpkglib PUBLIC
@@ -241,6 +241,7 @@ endif()
 
 # === Target: vcpkg ===
 
+if(0)
 add_executable(vcpkg ${VCPKG_SOURCES})
 target_link_libraries(vcpkg PRIVATE vcpkglib)
 vcpkg_target_add_warning_options(vcpkg)
@@ -256,6 +257,7 @@ endif()
 if(VCPKG_PDB_SUFFIX)
     set_property(TARGET vcpkg PROPERTY PDB_NAME "vcpkg${VCPKG_PDB_SUFFIX}")
 endif()
+endif()
 
 # === Target: generate-message-map ===
 
@@ -338,7 +349,23 @@ endif()
 # === Installation ===
 # ====================
 
-install(TARGETS vcpkg
-    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-    BUNDLE DESTINATION ${CMAKE_INSTALL_BINDIR}
+# install directory instead of using PUBLIC_HEADER to maintain directory heirarchy
+install(DIRECTORY "${CMAKE_SOURCE_DIR}/include/vcpkg"
+    DESTINATION "include"
+    FILES_MATCHING
+    PATTERN "*.h" 
+    PATTERN "fwd/*.h" 
+    PATTERN "base/*.h" 
+    PATTERN "base/fwd/*.h"
 )
+
+install(TARGETS vcpkglib EXPORT vcpkglib_targets RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
+export(EXPORT vcpkglib_targets NAMESPACE unofficial::vcpkglib:: FILE "${CMAKE_CURRENT_BINARY_DIR}/unofficial-vcpkglib-targets.cmake")
+set(INSTALL_CONFIG_DIR "${CMAKE_INSTALL_LIBDIR}/cmake/unofficial-vcpkglib")
+install(EXPORT vcpkglib_targets NAMESPACE unofficial::vcpkglib:: FILE unofficial-vcpkglib-targets.cmake DESTINATION "${INSTALL_CONFIG_DIR}")
+file(WRITE "${CMAKE_INSTALL_PREFIX}/share/vcpkglib/vcpkglib-config.cmake"
+[[
+find_package(Threads REQUIRED)
+find_package(fmt CONFIG REQUIRED)
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-vcpkglib-targets.cmake")
+]])
