diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5d27c57c4..7f37e5b5d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -142,7 +142,7 @@ find_package(fmt REQUIRED)
 
 # === Target: vcpkglib ===
 
-add_library(vcpkglib OBJECT
+add_library(vcpkglib STATIC
     ${VCPKGLIB_BASE_SOURCES}
     ${VCPKGLIB_SOURCES}
     ${VCPKGLIB_BASE_INCLUDES}
@@ -241,6 +241,7 @@ endif()
 
 # === Target: vcpkg ===
 
+if(0)
 add_executable(vcpkg ${VCPKG_SOURCES})
 target_link_libraries(vcpkg PRIVATE vcpkglib)
 vcpkg_target_add_warning_options(vcpkg)
@@ -256,6 +257,7 @@ endif()
 if(VCPKG_PDB_SUFFIX)
     set_property(TARGET vcpkg PROPERTY PDB_NAME "vcpkg${VCPKG_PDB_SUFFIX}")
 endif()
+endif()
 
 # === Target: generate-message-map ===
 
@@ -338,7 +340,14 @@ endif()
 # === Installation ===
 # ====================
 
-install(TARGETS vcpkg
-    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-    BUNDLE DESTINATION ${CMAKE_INSTALL_BINDIR}
+# install directory instead of using PUBLIC_HEADER to maintain directory heirarchy
+install(DIRECTORY "${CMAKE_SOURCE_DIR}/include/vcpkg"
+    DESTINATION "include"
+    FILES_MATCHING
+    PATTERN "*.h" 
+    PATTERN "fwd/*.h" 
+    PATTERN "base/*.h" 
+    PATTERN "base/fwd/*.h"
 )
+
+install(TARGETS vcpkglib RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
