diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e913cb2..107f777 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -424,14 +424,24 @@ set (GENTAB_BDIR ${CMAKE_CURRENT_BINARY_DIR}/gentables)
 # To fix cross-compiling fluidsynth from Win32 to ARM (using vcpkg), we need to pass the current generator
 # on to the external project, otherwise (for some unknown reason) the target compiler will be used rather
 # than the host compiler.
-ExternalProject_Add(gentables
-    DOWNLOAD_COMMAND ""
-    SOURCE_DIR ${GENTAB_SDIR}
-    BINARY_DIR ${GENTAB_BDIR}
-    CONFIGURE_COMMAND
-        "${CMAKE_COMMAND}" -DCMAKE_VERBOSE_MAKEFILE=${CMAKE_VERBOSE_MAKEFILE} -G "${CMAKE_GENERATOR}" -B "${GENTAB_BDIR}" "${GENTAB_SDIR}"
-    BUILD_COMMAND
-        "${CMAKE_COMMAND}" --build "${GENTAB_BDIR}"
-    INSTALL_COMMAND ${GENTAB_BDIR}/make_tables.exe "${CMAKE_BINARY_DIR}/"
-)
+set(CMAKE_DISABLE_SOURCE_CHANGES OFF)
+if(VCPKG_TARGET_ARCHITECTURE STREQUAL arm OR VCPKG_TARGET_ARCHITECTURE STREQUAL arm64)
+  ExternalProject_Add(gentables
+      DOWNLOAD_COMMAND ""
+      SOURCE_DIR ${GENTAB_SDIR}
+      BINARY_DIR ${GENTAB_BDIR}
+      CONFIGURE_COMMAND
+          "${CMAKE_COMMAND}" "${GENTAB_SDIR}" -A Win32 -B "${GENTAB_BDIR}"
+      BUILD_COMMAND
+          "${CMAKE_COMMAND}" --build "${GENTAB_BDIR}"
+      INSTALL_COMMAND ${GENTAB_BDIR}/make_tables.exe "${CMAKE_BINARY_DIR}/"
+  )
+else()
+  ExternalProject_Add(gentables
+      DOWNLOAD_COMMAND ""
+      SOURCE_DIR ${GENTAB_SDIR}
+      BINARY_DIR ${GENTAB_BDIR}
+      INSTALL_COMMAND ${GENTAB_BDIR}/make_tables.exe "${CMAKE_BINARY_DIR}/"
+  )
+endif()
 add_dependencies(libfluidsynth-OBJ gentables)
