diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e8fcb05..36f5371 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -7,3 +7,32 @@ else ()
   add_library (${BEA_TARGET} STATIC BeaEngine.c)
   set_target_properties (${BEA_TARGET} PROPERTIES COMPILE_FLAGS "-DBEA_ENGINE_STATIC") 
 endif ()
+
+install(DIRECTORY
+  "${CMAKE_CURRENT_SOURCE_DIR}/includes"
+  DESTINATION "include"
+  FILES_MATCHING PATTERN "*.h")
+install(DIRECTORY
+  "${CMAKE_CURRENT_SOURCE_DIR}/includes/instr_set"
+  DESTINATION "include/instr_set"
+  FILES_MATCHING PATTERN "*.h")
+install(TARGETS ${BEA_TARGET}
+ EXPORT beaengine-targets
+ RUNTIME DESTINATION "bin"
+ ARCHIVE DESTINATION "lib"
+ LIBRARY DESTINATION "lib"
+ )
+install(EXPORT beaengine-targets NAMESPACE libhat:: DESTINATION "share/libhat")
+include(CMakePackageConfigHelpers)
+configure_package_config_file(
+"${CMAKE_CURRENT_LIST_DIR}/beaengine-config.cmake.in"
+"${CMAKE_CURRENT_BINARY_DIR}/beaengine-config.cmake"
+INSTALL_DESTINATION "share/beaengine"
+)
+write_basic_package_version_file(
+  "${CMAKE_CURRENT_BINARY_DIR}/beaengine-config-version.cmake" VERSION "${VERSION}" COMPATIBILITY SameMajorVersion)
+install(FILES
+  "${CMAKE_CURRENT_BINARY_DIR}/beaengine-config.cmake"
+  "${CMAKE_CURRENT_BINARY_DIR}/beaengine-config-version.cmake"
+  DESTINATION "share/beaengine"
+)
