diff --git a/meson.build b/meson.build
index d946c49..eded01d 100644
--- a/meson.build
+++ b/meson.build
@@ -90,6 +90,7 @@ subdir('include')
 subdir('scripts')
 subdir('static')
 subdir('src')
+if false
 if get_option('examples')
   subdir('examples')
 endif
@@ -97,6 +98,7 @@ subdir('test')
 if get_option('doc')
   subdir('docs')
 endif
+endif
 
 pkg_mod = import('pkgconfig')
 pkg_mod.generate(libraries : libzim,
