diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 7e81ed9..39b95a9 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -31,9 +31,9 @@ target_include_directories(${OATPP_THIS_MODULE_NAME}
 )
 
 target_link_libraries(${OATPP_THIS_MODULE_NAME}
-        PRIVATE mbedtls::TLS
-        PRIVATE mbedtls::X509
-        PRIVATE mbedtls::Crypto
+        PRIVATE ${MBEDTLS_TLS_LIBRARY}
+        PRIVATE ${MBEDTLS_X509_LIBRARY}
+        PRIVATE ${MBEDTLS_CRYPTO_LIBRARY}
 )
 
 if(${OATPP_MBEDTLS_DEBUG} GREATER 0)
