diff --git a/ports/cmake/src/CMakeLists.txt b/ports/cmake/src/CMakeLists.txt
index 7147174..76e1743 100644
--- a/ports/cmake/src/CMakeLists.txt
+++ b/ports/cmake/src/CMakeLists.txt
@@ -238,7 +238,7 @@ if(BUILD_LIBOUT123)
             endif()
         endif()
 
-        if(NOT CHECK_MODULES OR NOT JACK_REQUIRED EQUAL -1)
+        if(0)
             pkg_search_module(JACK jack)
             if(JACK_FOUND)
                 list(APPEND OUTPUT_MODULES jack)
