diff --git a/meson.build b/meson.build
index 7419e2b..687450e 100644
--- a/meson.build
+++ b/meson.build
@@ -1909,12 +1909,11 @@ libsystemd_includes = [basic_includes, include_directories(
 includes = [libsystemd_includes, include_directories('src/shared')]
 
 subdir('po')
-subdir('catalog')
+support_url=''
 subdir('src/fundamental')
 subdir('src/basic')
 subdir('src/libsystemd')
-subdir('src/shared')
-subdir('src/libudev')
+subdir('src/systemd') # headers
 
 libsystemd = shared_library(
         'systemd',
@@ -1930,7 +1929,8 @@ libsystemd = shared_library(
                         threads,
                         userspace],
         link_depends : libsystemd_sym,
-        install : true,
+        build_by_default : static_libsystemd == 'false',
+        install : static_libsystemd == 'false',
         install_tag: 'libsystemd',
         install_dir : libdir)
 
@@ -1963,6 +1963,8 @@ install_libsystemd_static = static_library(
                         userspace],
         c_args : libsystemd_c_args + (static_libsystemd_pic ? [] : ['-fno-PIC']))
 
+if false
+
 libudev = shared_library(
         'udev',
         version : libudev_version,
@@ -1974,7 +1976,8 @@ libudev = shared_library(
         dependencies : [threads,
                         userspace],
         link_depends : libudev_sym,
-        install : true,
+        build_by_default : static_libudev == 'false',
+        install : static_libudev == 'false',
         install_tag: 'libudev',
         install_dir : libdir)
 
@@ -2661,6 +2664,13 @@ if not meson.is_cross_build()
                 command : [export_dbus_interfaces_py, '@OUTPUT@', dbus_programs])
 endif
 
+endif # false
+static_libudev='unused'
+bashcompletiondir='no'
+zshcompletiondir='no'
+want_html=false
+want_man=false
+
 ############################################################
 
 alt_time_epoch = run_command('date', '-Is', '-u', '-d', '@@0@'.format(time_epoch),
