diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index a9f8908..5c98f3f 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -17,6 +17,5 @@ install(EXPORT ${package_name} DESTINATION lib/cmake/${package_name}
 
 install(FILES
     "${CMAKE_CURRENT_BINARY_DIR}/${package_name}Config.cmake"
-    FindPahoMqttC.cmake
     "${CMAKE_CURRENT_BINARY_DIR}/${package_name}ConfigVersion.cmake"
     DESTINATION lib/cmake/${package_name})
diff --git a/cmake/PahoMqttCppConfig.cmake.in b/cmake/PahoMqttCppConfig.cmake.in
index 164e123..9beac1c 100644
--- a/cmake/PahoMqttCppConfig.cmake.in
+++ b/cmake/PahoMqttCppConfig.cmake.in
@@ -5,7 +5,7 @@ set(PAHO_WITH_SSL @PAHO_WITH_SSL@)
 
 include(CMakeFindDependencyMacro)
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR})
-find_dependency(PahoMqttC REQUIRED)
+find_dependency(eclipse-paho-mqtt-c REQUIRED)
 list(REMOVE_AT CMAKE_MODULE_PATH -1)
 find_dependency(Threads REQUIRED)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 3d43595..a25f74d 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -23,7 +23,7 @@
 #     Frank Pagliughi - made the shared library optional
 #*******************************************************************************/
 
-find_package(PahoMqttC REQUIRED)
+find_package(eclipse-paho-mqtt-c CONFIG REQUIRED)
 
 # --- The headers ---
 
@@ -81,13 +81,21 @@ target_include_directories(paho-cpp-objs
 ## --- Build the shared library, if requested ---
 
 if(PAHO_BUILD_SHARED)
+    ## set PAHO_C_LIB
+    if(PAHO_WITH_SSL)
+        find_package(OpenSSL REQUIRED)
+        set(_PAHO_MQTT_C_LIB_NAME eclipse-paho-mqtt-c::paho-mqtt3as eclipse-paho-mqtt-c::paho-mqtt3cs)
+    else()
+        set(_PAHO_MQTT_C_LIB_NAME eclipse-paho-mqtt-c::paho-mqtt3a eclipse-paho-mqtt-c::paho-mqtt3c)
+    endif()
+
     ## create the shared library
     add_library(paho-mqttpp3 SHARED $<TARGET_OBJECTS:paho-cpp-objs>)
 
     ## add dependencies to the shared library
     target_link_libraries(paho-mqttpp3
         PRIVATE ${LIBS_SYSTEM}
-        PUBLIC PahoMqttC::PahoMqttC Threads::Threads)
+        PUBLIC ${_PAHO_MQTT_C_LIB_NAME} Threads::Threads)
 
     # It would be nice to exort the include paths from the obj lib, but we
     # get an export error. Perhaps in a future version?
@@ -119,13 +127,22 @@ endif()
 ## --- Build static version of the library, if requested ---
 
 if(PAHO_BUILD_STATIC)
+
+    ## set PAHO_C_LIB
+    if(PAHO_WITH_SSL)
+        find_package(OpenSSL REQUIRED)
+        set(_PAHO_MQTT_C_LIB_NAME eclipse-paho-mqtt-c::paho-mqtt3as-static eclipse-paho-mqtt-c::paho-mqtt3cs-static)
+    else()
+        set(_PAHO_MQTT_C_LIB_NAME  eclipse-paho-mqtt-c::paho-mqtt3a-static eclipse-paho-mqtt-c::paho-mqtt3c-static)
+    endif()
+
     ## create the static library
     add_library(paho-mqttpp3-static STATIC $<TARGET_OBJECTS:paho-cpp-objs>)
 
     ## add dependencies to the shared library
     target_link_libraries(paho-mqttpp3-static
         PRIVATE ${LIBS_SYSTEM}
-        PUBLIC PahoMqttC::PahoMqttC Threads::Threads)
+        PUBLIC ${_PAHO_MQTT_C_LIB_NAME} Threads::Threads)
 
     target_include_directories(paho-mqttpp3-static PUBLIC
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
