diff --git a/include/sleepy_discord/zlib_compression.h b/include/sleepy_discord/zlib_compression.h
index 9d95c58..b2d418e 100644
--- a/include/sleepy_discord/zlib_compression.h
+++ b/include/sleepy_discord/zlib_compression.h
@@ -9,6 +9,7 @@
 #include <forward_list>
 #include <string>
 #include <mutex>
+#include <cstring>
 
 namespace SleepyDiscord {
 	//This Queue is basicly a single linked list with the back and size stored
