diff --git a/include/private/gpu/ganesh/GrD3DTypesMinimal.h b/include/private/gpu/ganesh/GrD3DTypesMinimal.h
index 84db9c1567..2bf9c2755d 100644
--- a/include/private/gpu/ganesh/GrD3DTypesMinimal.h
+++ b/include/private/gpu/ganesh/GrD3DTypesMinimal.h
@@ -48,7 +48,7 @@ struct GrD3DBackendSurfaceInfo {
     GrD3DTextureResourceInfo snapTextureResourceInfo() const;
 
     bool isProtected() const;
-#if defined(GPU_TEST_UTILS)
+#if 1
     bool operator==(const GrD3DBackendSurfaceInfo& that) const;
 #endif
 
diff --git a/src/gpu/ganesh/d3d/GrD3DTypesMinimal.cpp b/src/gpu/ganesh/d3d/GrD3DTypesMinimal.cpp
index a8388a80fd..f2ed2fa886 100644
--- a/src/gpu/ganesh/d3d/GrD3DTypesMinimal.cpp
+++ b/src/gpu/ganesh/d3d/GrD3DTypesMinimal.cpp
@@ -54,7 +54,7 @@ bool GrD3DBackendSurfaceInfo::isProtected() const {
     return fTextureResourceInfo->fProtected == GrProtected::kYes;
 }
 
-#if defined(GPU_TEST_UTILS)
+#if 1
 bool GrD3DBackendSurfaceInfo::operator==(const GrD3DBackendSurfaceInfo& that) const {
     GrD3DTextureResourceInfo cpyInfoThis = *fTextureResourceInfo;
     GrD3DTextureResourceInfo cpyInfoThat = *that.fTextureResourceInfo;
