diff --git a/BUILD.gn b/BUILD.gn
index def43b13f1..5863050248 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -1933,7 +1933,9 @@ skia_component("skia") {
     if (is_ios) {
       sources += [ "src/ports/SkOSFile_ios.h" ]
     }
-    libs += [ "dl" ]
+    if (skia_vcpkg_has_lib_dl) {
+      libs += [ "dl" ]
+    }
   }
 
   if (is_android) {
diff --git a/gn/skia.gni b/gn/skia.gni
index 13fbcce6ca..9ed1ea066e 100644
--- a/gn/skia.gni
+++ b/gn/skia.gni
@@ -105,6 +105,8 @@ declare_args() {
   }
 
   skia_build_rust_targets = false
+
+  skia_vcpkg_has_lib_dl = true
 }
 
 declare_args() {
