diff --git a/BUILD.gn b/BUILD.gn
index 577dd00..07f9882 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -1478,6 +1478,10 @@ skia_component("skia") {
   defines = []
   libs = []
 
+  if (is_win && target_cpu != "arm64" && !skia_enable_winuwp) {
+    libs += [ "OpenGL32.lib" ] # for src/utils/win/SkWGL_win.cpp
+  }
+
   if (skia_enable_sksl) {
     deps += [ ":minify_sksl" ]
     sources += skia_sksl_sources
