diff --git a/src/dawn/native/BUILD.gn b/src/dawn/native/BUILD.gn
index 24853ced..064b839b 100644
--- a/src/dawn/native/BUILD.gn
+++ b/src/dawn/native/BUILD.gn
@@ -882,6 +882,10 @@ dawn_json_generator("webgpu_dawn_native_proc_gen") {
   target = "webgpu_dawn_native_proc"
   outputs = [ "src/dawn/native/webgpu_dawn_native_proc.cpp" ]
 }
+source_set("webgpu_dawn_native_proc_gen_sources") {
+    sources = get_target_outputs(":webgpu_dawn_native_proc_gen")
+    deps = [ ":webgpu_dawn_native_proc_gen" ]
+}
 
 dawn_component("webgpu_dawn") {
   # For a single library - build `webgpu_dawn_shared` with GN args:
@@ -901,9 +905,8 @@ dawn_component("webgpu_dawn") {
   DEFINE_PREFIX = "WGPU"
   output_name = "webgpu_dawn"
 
-  sources = get_target_outputs(":webgpu_dawn_native_proc_gen")
   deps = [
     ":static",
-    ":webgpu_dawn_native_proc_gen",
+    ":webgpu_dawn_native_proc_gen_sources",
   ]
 }
