diff --git a/src/ports/SkDebug_stdio.cpp b/src/ports/SkDebug_stdio.cpp
index 78c7072b..3d5c134d 100644
--- a/src/ports/SkDebug_stdio.cpp
+++ b/src/ports/SkDebug_stdio.cpp
@@ -7,6 +7,7 @@
 
 #include "include/private/base/SkFeatures.h"
 #include "include/private/base/SkLoadUserConfig.h"
+#include "include/private/base/SkDebug.h"
 
 #if !defined(SK_BUILD_FOR_WIN) && !defined(SK_BUILD_FOR_ANDROID)
 
