diff --git a/BUILD.gn b/BUILD.gn
index 37cfe09694..def43b13f1 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -993,7 +993,9 @@ optional("gpu") {
         "src/gpu/ganesh/gl/glx/GrGLMakeGLXInterface.cpp",
         "src/gpu/ganesh/gl/glx/GrGLMakeNativeInterface_glx.cpp",
       ]
-      libs += [ "GL" ]
+      deps += [
+        "//third_party/gl"
+      ]
     } else if (is_mac) {
       sources += [ "src/gpu/ganesh/gl/mac/GrGLMakeNativeInterface_mac.cpp" ]
     } else if (is_ios) {
