diff --git a/CMakeLists.txt b/CMakeLists.txt
index b13ff3b..137c049 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -61,6 +61,7 @@ set(GZ_UTILS_VER ${gz-utils2_VERSION_MAJOR})
 # Find Tinyxml2
 if(USE_EXTERNAL_TINYXML2)
   gz_find_package(TINYXML2 PRETTY tinyxml2
+    PKGCONFIG tinyxml2
     REQUIRED_BY graphics
     PRIVATE_FOR graphics)
 else()
@@ -92,6 +93,7 @@ gz_find_package(FreeImage VERSION 3.9
 # Find GNU Triangulation Surface Library
 gz_find_package(
   GTS PRETTY gts PURPOSE "GNU Triangulation Surface library"
+  PKGCONFIG gts
   REQUIRED_BY graphics
   PRIVATE_FOR graphics)
 
@@ -99,6 +101,7 @@ gz_find_package(
 # Find GDAL
 gz_find_package(GDAL VERSION 3.0
   PKGCONFIG gdal
+  PKGCONFIG_VER_COMPARISON >=
   PRIVATE_FOR geospatial
   REQUIRED_BY geospatial)
 
