diff --git a/src/windows/sparsehash/internal/sparseconfig.h b/src/windows/sparsehash/internal/sparseconfig.h
index 3091559..f96e08c 100644
--- a/src/windows/sparsehash/internal/sparseconfig.h
+++ b/src/windows/sparsehash/internal/sparseconfig.h
@@ -7,10 +7,10 @@
 #define GOOGLE_NAMESPACE  ::google
 
 /* the location of the header defining hash functions */
-#define HASH_FUN_H  <hash_map>
+#define HASH_FUN_H  <unordered_map>
 
 /* the namespace of the hash<> function */
-#define HASH_NAMESPACE  stdext
+#define HASH_NAMESPACE  std
 
 /* Define to 1 if you have the <inttypes.h> header file. */
 #undef HAVE_INTTYPES_H
