diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2fa3908..1ff7438 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -58,6 +58,13 @@ add_library(${PROJECT_NAME}
 	src/script_span/fixunicodevalue.cc
 	)
 
+if (MSVC)
+	add_compile_definitions(COMPILER_MSVC)
+endif()
+
+install(TARGETS ${PROJECT_NAME} LIBRARY DESTINATION "${INSTALL_LIB_DIR}" )
+return()
+
 # unit tests exec:
 add_executable(language_identifier_main src/language_identifier_main.cc)
 target_link_libraries(language_identifier_main cld3 ${Protobuf_LITE_LIBRARIES})
