diff --git a/CMakeLists.txt b/CMakeLists.txt
index dc57bb1..1c4f1e6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -131,6 +131,7 @@ endif()
 
 # Installation target.
 if(LIBENVPP_INSTALL)
+	include(CMakePackageConfigHelpers)
 	# Libenvpp installation.
 	set(LIBENVPP_PROJECT_CONFIG_OUT "${CMAKE_CURRENT_BINARY_DIR}/libenvpp-config.cmake")
 	set(LIBENVPP_VERSION_CONFIG_FILE "${CMAKE_CURRENT_BINARY_DIR}/libenvpp-config-version.cmake")
