diff --git a/cmake/findDependencies.cmake b/cmake/findDependencies.cmake
index 67a6cfd..d5006ff 100644
--- a/cmake/findDependencies.cmake
+++ b/cmake/findDependencies.cmake
@@ -30,10 +30,12 @@ endif()
 
 list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake/")
 
+if(EXIV2_BUILD_UNIT_TESTS)
 find_package (Python3 COMPONENTS Interpreter)
 if (NOT Python3_Interpreter_FOUND)
     message(WARNING "Python3 was not found. Python tests under the 'tests' folder will not be executed")
 endif()
+endif()
 
 find_package(Filesystem COMPONENTS Experimental Final REQUIRED)
 
