diff --git a/cmake/findDependencies.cmake b/cmake/findDependencies.cmake
index 1075c3031..bec2128a5 100644
--- a/cmake/findDependencies.cmake
+++ b/cmake/findDependencies.cmake
@@ -77,7 +77,7 @@ if( ICONV_FOUND )
 endif()
 
 if( EXIV2_ENABLE_INIH )
-  find_package(inih)
+  find_package(unofficial-inih CONFIG REQUIRED)
   message ( "-- inih_INCLUDE_DIRS : " ${inih_INCLUDE_DIRS} )
   message ( "-- inih_LIBRARIES : " ${inih_LIBRARIES} )
   message ( "-- inih_inireader_INCLUDE_DIRS : " ${inih_inireader_INCLUDE_DIRS} )
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index dba7af99c..15bab8c8a 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -263,10 +263,10 @@ if( Iconv_FOUND AND Iconv_LIBRARIES )
 endif()
 
 if( EXIV2_ENABLE_INIH )
-  target_link_libraries( exiv2lib_int PRIVATE inih::libinih )
-  target_link_libraries( exiv2lib_int PRIVATE inih::inireader )
-  target_link_libraries( exiv2lib PRIVATE inih::libinih )
-  target_link_libraries( exiv2lib PRIVATE inih::inireader )
+  target_link_libraries( exiv2lib_int PRIVATE unofficial::inih::libinih )
+  target_link_libraries( exiv2lib_int PRIVATE unofficial::inih::inireader )
+  target_link_libraries( exiv2lib PRIVATE unofficial::inih::libinih )
+  target_link_libraries( exiv2lib PRIVATE unofficial::inih::inireader )
   list(APPEND requires_private_list "INIReader")
 endif()
 
