diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c52b352..b19c438 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -149,7 +149,7 @@ set_target_properties( exiv2lib_int PROPERTIES
 
 target_include_directories(exiv2lib_int PRIVATE ${ZLIB_INCLUDE_DIR})
 target_include_directories(exiv2lib PRIVATE 
-    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/xmpsdk/include>
+    ${PROJECT_SOURCE_DIR}/xmpsdk/include
     ${ZLIB_INCLUDE_DIR}
 )
 
@@ -215,7 +215,7 @@ else()
 endif()
 
 if( EXIV2_ENABLE_PNG )
-    target_link_libraries( exiv2lib PRIVATE $<BUILD_INTERFACE:${ZLIB_LIBRARIES}>)
+    target_link_libraries( exiv2lib PRIVATE ${ZLIB_LIBRARIES})
 endif()
 
 if( EXIV2_ENABLE_NLS )
diff --git a/xmpsdk/CMakeLists.txt b/xmpsdk/CMakeLists.txt
index a22698f..ca6ffe0 100644
--- a/xmpsdk/CMakeLists.txt
+++ b/xmpsdk/CMakeLists.txt
@@ -27,13 +27,12 @@ add_library(exiv2-xmp STATIC
 )
 
 target_link_libraries(exiv2-xmp
-    PRIVATE 
-        $<BUILD_INTERFACE:${EXPAT_LIBRARY}>
+    PRIVATE ${EXPAT_LIBRARY}
 )
 
 target_include_directories(exiv2-xmp
     PRIVATE 
-        $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/xmpsdk/include>
+        ${PROJECT_SOURCE_DIR}/xmpsdk/include
         ${EXPAT_INCLUDE_DIR}
 )
 
