diff --git a/cmake/findDependencies.cmake b/cmake/findDependencies.cmake
index ec3a43f..d8637cc 100644
--- a/cmake/findDependencies.cmake
+++ b/cmake/findDependencies.cmake
@@ -42,7 +42,10 @@ if (EXIV2_ENABLE_XMP AND EXIV2_ENABLE_EXTERNAL_XMP)
     message(FATAL_ERROR "EXIV2_ENABLE_XMP AND EXIV2_ENABLE_EXTERNAL_XMP are mutually exclusive.  You can only choose one of them")
 else()
     if (EXIV2_ENABLE_XMP)
-        find_package(EXPAT REQUIRED)
+        find_package(EXPAT NAMES expat CONFIG REQUIRED)
+        set(EXPAT_INCLUDE_DIR "")
+        set(EXPAT_LIBRARY expat::expat)
+        set(EXPAT_LIBRARIES expat::expat)
     elseif (EXIV2_ENABLE_EXTERNAL_XMP)
         find_package(XmpSdk REQUIRED)
     endif ()
