diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5417a62..4da0d6c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -123,10 +123,12 @@ IF(CMAKE_BUILD_TYPE STREQUAL "Debug")
     SET(MY_CMAKE_FLAGS "${MY_CMAKE_FLAGS} -Od")
     #SET (MY_CMAKE_FLAGS "${MY_CMAKE_FLAGS} -DFIU_ENABLE=1")
     #SET(LIBS ${LIBS} fiu)
+    SET(LIB_NAME ssld cryptod)
 ELSE()
     SET(MY_CMAKE_FLAGS "${MY_CMAKE_FLAGS} -Ox")
     # Comment out the following line to compile out debug messages:
     #SET(MY_CMAKE_FLAGS "${MY_CMAKE_FLAGS} -DLSQUIC_LOWEST_LOG_LEVEL=LSQ_LOG_INFO")
+    SET(LIB_NAME ssl crypto)
 ENDIF()
 
 ENDIF() #MSVC
@@ -197,20 +199,21 @@ MESSAGE(STATUS "Compiler flags: ${CMAKE_C_FLAGS}")
 
 IF (NOT DEFINED LIBSSL_LIB AND DEFINED LIBSSL_DIR)
     SET(LIBSSL_LIB ${LIBSSL_DIR})
-    FOREACH(LIB_NAME ssl crypto)
+    FOREACH(LIB ${LIB_NAME})
         IF (CMAKE_SYSTEM_NAME STREQUAL Windows)
-            FIND_LIBRARY(LIBSSL_LIB_${LIB_NAME}
-                NAMES ${LIB_NAME}
+            FIND_LIBRARY(BORINGSSL_LIB_${LIB}
+                NAMES ${LIB}
                 PATHS ${LIBSSL_LIB}/ ${LIBSSL_LIB}/lib/ ${LIBSSL_LIB}/ssl/ ${LIBSSL_LIB}/crypto/
                 PATH_SUFFIXES Debug Release MinSizeRel RelWithDebInfo
                 NO_DEFAULT_PATH)
         ELSE()
-            FIND_LIBRARY(LIBSSL_LIB_${LIB_NAME}
-                NAMES lib${LIB_NAME}${LIB_SUFFIX}
+            FIND_LIBRARY(BORINGSSL_LIB_${LIB}
+                NAMES lib${LIB}${LIB_SUFFIX}
                 PATHS ${LIBSSL_LIB} ${LIBSSL_LIB}/lib/ ${LIBSSL_LIB}/ssl/ ${LIBSSL_LIB}/crypto/
+                PATH_SUFFIXES ${LIB}
                 NO_DEFAULT_PATH)
         ENDIF()
-        IF(LIBSSL_LIB_${LIB_NAME})
+        IF(BORINGSSL_LIB_${LIB})
             MESSAGE(STATUS "Found ${LIB_NAME} library: ${LIBSSL_LIB_${LIB_NAME}}")
         ELSE()
             MESSAGE(STATUS "${LIB_NAME} library not found")
