diff --git a/lang/c/CMakeLists.txt b/lang/c/CMakeLists.txt
index e55e72d..c6981c0 100644
--- a/lang/c/CMakeLists.txt
+++ b/lang/c/CMakeLists.txt
@@ -127,11 +127,11 @@ include_directories(${AvroC_SOURCE_DIR}/src)
 
 # Enable codecs
 
-find_package(ZLIB)
+find_package(ZLIB REQUIRED)
 if (ZLIB_FOUND)
     set(ZLIB_PKG zlib)
     add_definitions(-DDEFLATE_CODEC)
-    include_directories(${ZLIB_INCLUDE_DIR})
+    link_libraries(ZLIB::ZLIB)
     message("Enabled deflate codec")
 else (ZLIB_FOUND)
     set(ZLIB_PKG "")
diff --git a/lang/c/tests/test_avro_data.c b/lang/c/tests/test_avro_data.c
index 8a0c013..1ebfb66 100644
--- a/lang/c/tests/test_avro_data.c
+++ b/lang/c/tests/test_avro_data.c
@@ -142,7 +142,7 @@ static void test_json(avro_datum_t datum, const char *expected)
 {
 	char  *json = NULL;
 	avro_datum_to_json(datum, 1, &json);
-	if (stricmp(json, expected) != 0) {
+	if (strcmp(json, expected) != 0) {
 		fprintf(stderr, "Unexpected JSON encoding: %s\n", json);
 		exit(EXIT_FAILURE);
 	}
