diff --git a/CMakeModules/ModuleInstall.cmake b/CMakeModules/ModuleInstall.cmake
index bb3be63..356edcd 100644
--- a/CMakeModules/ModuleInstall.cmake
+++ b/CMakeModules/ModuleInstall.cmake
@@ -21,7 +21,7 @@ SOURCE_GROUP(
     FILES ${TARGET_H}
 )
 
-IF(MSVC AND OSG_MSVC_VERSIONED_DLL)
+IF(0)
     HANDLE_MSVC_DLL()
 ENDIF()
 
diff --git a/src/OpenThreads/win32/CMakeLists.txt b/src/OpenThreads/win32/CMakeLists.txt
index 84550bd..0e816f7 100644
--- a/src/OpenThreads/win32/CMakeLists.txt
+++ b/src/OpenThreads/win32/CMakeLists.txt
@@ -39,7 +39,7 @@ LINK_DIRECTORIES(
     ${CMAKE_CURRENT_BINARY_DIR}
 )
 
-IF(MSVC AND OSG_MSVC_VERSIONED_DLL)
+IF(0)
     HANDLE_MSVC_DLL(ot ${OPENTHREADS_SOVERSION})
 ENDIF()
 
