diff --git a/CMakeLists.txt b/CMakeLists.txt
index 463d829b..8fc9b5e4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -100,8 +100,6 @@ mark_as_advanced(
   CUDA_HOST_COMPILER
   CUDA_USE_STATIC_CUDA_RUNTIME
   CUDA_rt_LIBRARY
-  SPDLOG_BUILD_EXAMPLES
-  SPDLOG_BUILD_TESTING
   ADDR2LINE_PROGRAM
   Backtrace_LIBRARY
   )
@@ -109,7 +107,7 @@ mark_as_advanced(
 #Configure forge submodule
 #forge is included in ALL target if AF_BUILD_FORGE is ON
 #otherwise, forge is not built at all
-include(AFconfigure_forge_submodule)
+# include(AFconfigure_forge_submodule)
 
 configure_file(
     ${ArrayFire_SOURCE_DIR}/CMakeModules/version.hpp.in
@@ -154,9 +152,12 @@ if(NOT LAPACK_FOUND)
     endif()
 endif()
 
-set(SPDLOG_BUILD_TESTING OFF CACHE INTERNAL "Disable testing in spdlog")
-add_subdirectory(extern/spdlog EXCLUDE_FROM_ALL)
-add_subdirectory(extern/glad)
+# set(SPDLOG_BUILD_TESTING OFF CACHE INTERNAL "Disable testing in spdlog")
+# add_subdirectory(extern/spdlog EXCLUDE_FROM_ALL)
+find_package(spdlog CONFIG REQUIRED)
+
+# add_subdirectory(extern/glad)
+
 add_subdirectory(src/backend/common)
 add_subdirectory(src/api/c)
 add_subdirectory(src/api/cpp)
@@ -260,9 +261,9 @@ install(DIRECTORY examples/ #NOTE The slash at the end is important
     DESTINATION ${AF_INSTALL_EXAMPLE_DIR}
     COMPONENT examples)
 
-install(DIRECTORY assets/examples/ #NOTE The slash at the end is important
-    DESTINATION ${AF_INSTALL_EXAMPLE_DIR}
-    COMPONENT examples)
+# install(DIRECTORY assets/examples/ #NOTE The slash at the end is important
+#     DESTINATION ${AF_INSTALL_EXAMPLE_DIR}
+#     COMPONENT examples)
 
 install(DIRECTORY "${ArrayFire_SOURCE_DIR}/LICENSES/"
     DESTINATION LICENSES
diff --git a/src/api/unified/CMakeLists.txt b/src/api/unified/CMakeLists.txt
index b0489be4..a82e9f83 100644
--- a/src/api/unified/CMakeLists.txt
+++ b/src/api/unified/CMakeLists.txt
@@ -74,7 +74,7 @@ target_include_directories(af
 target_link_libraries(af
   PRIVATE
     cpp_api_interface
-    spdlog
+    spdlog::spdlog spdlog::spdlog_header_only
     Threads::Threads
     ${CMAKE_DL_LIBS}
   )
diff --git a/src/backend/common/CMakeLists.txt b/src/backend/common/CMakeLists.txt
index 33aa64e6..2d07e4ea 100644
--- a/src/backend/common/CMakeLists.txt
+++ b/src/backend/common/CMakeLists.txt
@@ -69,16 +69,19 @@ else()
   target_sources(afcommon_interface INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/module_loading_unix.cpp)
 endif()
 
+find_package(glad CONFIG REQUIRED)
+
 target_link_libraries(afcommon_interface
   INTERFACE
-    spdlog
+    spdlog::spdlog spdlog::spdlog_header_only
     Boost::boost
-    af_glad_interface
+    glad::glad
     ${CMAKE_DL_LIBS}
 )
 
 if(AF_BUILD_FORGE)
-  add_dependencies(afcommon_interface forge)
+  find_package(Forge CONFIG REQUIRED)
+  target_link_libraries(afcommon_interface PRIVATE Forge::forge)
 endif()
 
 target_include_directories(afcommon_interface
@@ -87,8 +90,6 @@ target_include_directories(afcommon_interface
     ${ArrayFire_BINARY_DIR}
   SYSTEM INTERFACE
     $<$<PLATFORM_ID:Darwin>:${OPENGL_INCLUDE_DIR}>
-    ${ArrayFire_SOURCE_DIR}/extern/forge/include
-    ${ArrayFire_BINARY_DIR}/extern/forge/include
   )
 
 if(APPLE AND NOT USE_MKL)
diff --git a/src/backend/cpu/CMakeLists.txt b/src/backend/cpu/CMakeLists.txt
index bdd205bc..1d7fe462 100644
--- a/src/backend/cpu/CMakeLists.txt
+++ b/src/backend/cpu/CMakeLists.txt
@@ -266,9 +266,10 @@ endif(AF_WITH_CPUID)
 
 target_sources(afcpu
   PRIVATE
-    ${CMAKE_CURRENT_SOURCE_DIR}/threads/async_queue.hpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/threads/event.hpp
-  )
+    ${AF_CPU_THREAD_PATH}/include/threads/async_queue.hpp
+    ${AF_CPU_THREAD_PATH}/include/threads/event.hpp
+    )
+target_include_directories(afcpu PRIVATE ${AF_CPU_THREAD_PATH}/include)
 
 arrayfire_set_default_cxx_flags(afcpu)
 
diff --git a/src/backend/cuda/CMakeLists.txt b/src/backend/cuda/CMakeLists.txt
index ae29c43d..2c4d67ad 100644
--- a/src/backend/cuda/CMakeLists.txt
+++ b/src/backend/cuda/CMakeLists.txt
@@ -51,6 +51,12 @@ mark_as_advanced(
 
 get_target_property(COMMON_INTERFACE_DIRS afcommon_interface INTERFACE_INCLUDE_DIRECTORIES)
 
+# Cub
+find_path(CUB_INCLUDE_DIRS "cub/agent/agent_histogram.cuh")
+if (${CUB_INCLUDE_DIRS} EQUAL "CUB_INCLUDE_DIRS-NOTFOUND")
+  set(CUB_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/cub)
+endif()
+
 cuda_include_directories(
   ${CMAKE_CURRENT_BINARY_DIR}
   ${CMAKE_CURRENT_SOURCE_DIR}
@@ -58,7 +64,7 @@ cuda_include_directories(
   ${ArrayFire_BINARY_DIR}/include
   ${CMAKE_CURRENT_SOURCE_DIR}/kernel
   ${CMAKE_CURRENT_SOURCE_DIR}/jit
-  ${CMAKE_CURRENT_SOURCE_DIR}/cub
+  ${CUB_INCLUDE_DIRS}
   ${ArrayFire_SOURCE_DIR}/src/api/c
   ${ArrayFire_SOURCE_DIR}/src/backend
   ${COMMON_INTERFACE_DIRS}
diff --git a/src/backend/opencl/kernel/scan_by_key/CMakeLists.txt b/src/backend/opencl/kernel/scan_by_key/CMakeLists.txt
index 9a796c9e..f2fcdeda 100644
--- a/src/backend/opencl/kernel/scan_by_key/CMakeLists.txt
+++ b/src/backend/opencl/kernel/scan_by_key/CMakeLists.txt
@@ -27,6 +27,7 @@ foreach(SBK_BINARY_OP ${SBK_BINARY_OPS})
     add_dependencies(opencl_scan_by_key_${SBK_BINARY_OP}
                         ${cl_kernel_targets} OpenCL::cl2hpp Boost::boost)
 
+    find_package(glad CONFIG REQUIRED)
     target_include_directories(opencl_scan_by_key_${SBK_BINARY_OP}
       PRIVATE
         .
@@ -39,9 +40,9 @@ foreach(SBK_BINARY_OP ${SBK_BINARY_OPS})
         $<TARGET_PROPERTY:OpenCL::OpenCL,INTERFACE_INCLUDE_DIRECTORIES>
         $<TARGET_PROPERTY:OpenCL::cl2hpp,INTERFACE_INCLUDE_DIRECTORIES>
         $<TARGET_PROPERTY:Boost::boost,INTERFACE_INCLUDE_DIRECTORIES>
-        $<TARGET_PROPERTY:af_glad_interface,INTERFACE_INCLUDE_DIRECTORIES>
-        ${ArrayFire_SOURCE_DIR}/extern/forge/include
-        ${ArrayFire_BINARY_DIR}/extern/forge/include
+        $<TARGET_PROPERTY:glad::glad,INTERFACE_INCLUDE_DIRECTORIES>
+        # ${ArrayFire_SOURCE_DIR}/extern/forge/include
+        # ${ArrayFire_BINARY_DIR}/extern/forge/include
       )
 
     set_target_properties(opencl_scan_by_key_${SBK_BINARY_OP}
diff --git a/src/backend/opencl/kernel/sort_by_key/CMakeLists.txt b/src/backend/opencl/kernel/sort_by_key/CMakeLists.txt
index d618ff2f..8e5f4d8d 100644
--- a/src/backend/opencl/kernel/sort_by_key/CMakeLists.txt
+++ b/src/backend/opencl/kernel/sort_by_key/CMakeLists.txt
@@ -32,14 +32,15 @@ foreach(SBK_TYPE ${SBK_TYPES})
         ../../../include
         ${CMAKE_CURRENT_BINARY_DIR})
 
+    find_package(glad CONFIG REQUIRED)
     target_include_directories(opencl_sort_by_key_${SBK_TYPE}
       SYSTEM PRIVATE
         $<TARGET_PROPERTY:OpenCL::OpenCL,INTERFACE_INCLUDE_DIRECTORIES>
         $<TARGET_PROPERTY:OpenCL::cl2hpp,INTERFACE_INCLUDE_DIRECTORIES>
         $<TARGET_PROPERTY:Boost::boost,INTERFACE_INCLUDE_DIRECTORIES>
-        $<TARGET_PROPERTY:af_glad_interface,INTERFACE_INCLUDE_DIRECTORIES>
-        ${ArrayFire_SOURCE_DIR}/extern/forge/include
-        ${ArrayFire_BINARY_DIR}/extern/forge/include
+        $<TARGET_PROPERTY:glad::glad,INTERFACE_INCLUDE_DIRECTORIES>
+        # ${ArrayFire_SOURCE_DIR}/extern/forge/include
+        # ${ArrayFire_BINARY_DIR}/extern/forge/include
       )
 
     set_target_properties(opencl_sort_by_key_${SBK_TYPE}
