diff --git a/wangle/CMakeLists.txt b/wangle/CMakeLists.txt
index 8df7c76..f605f25 100644
--- a/wangle/CMakeLists.txt
+++ b/wangle/CMakeLists.txt
@@ -62,7 +62,7 @@ find_package(folly CONFIG REQUIRED)
 find_package(fizz CONFIG REQUIRED)
 find_package(fmt CONFIG REQUIRED)
 find_package(OpenSSL REQUIRED)
-find_package(Glog REQUIRED)
+find_package(glog CONFIG REQUIRED)
 find_package(gflags CONFIG QUIET)
 if (gflags_FOUND)
   message(STATUS "Found gflags from package config")
@@ -70,8 +70,8 @@ if (gflags_FOUND)
 else()
   find_package(Gflags REQUIRED)
 endif()
-find_package(LibEvent MODULE REQUIRED)
-find_package(DoubleConversion REQUIRED)
+find_package(Libevent CONFIG REQUIRED)
+find_package(double-conversion CONFIG REQUIRED)
 find_package(Threads REQUIRED)
 if (UNIX AND NOT APPLE)
   find_package(Librt)
@@ -153,24 +153,23 @@ target_include_directories(
   PUBLIC
     $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/..>
     $<INSTALL_INTERFACE:${INCLUDE_INSTALL_DIR}>
+  PRIVATE
     ${FIZZ_INCLUDE_DIR}
     ${FOLLY_INCLUDE_DIR}
     ${Boost_INCLUDE_DIR}
-    ${OPENSSL_INCLUDE_DIR}
-    ${GLOG_INCLUDE_DIRS}
     ${GFLAGS_INCLUDE_DIRS}
-    ${LIBEVENT_INCLUDE_DIR}
-    ${DOUBLE_CONVERSION_INCLUDE_DIRS}
 )
 target_link_libraries(wangle PUBLIC
   ${FOLLY_LIBRARIES}
   ${FIZZ_LIBRARIES}
   ${Boost_LIBRARIES}
-  ${OPENSSL_LIBRARIES}
-  ${GLOG_LIBRARIES}
+  OpenSSL::SSL
+  OpenSSL::Crypto
+  glog::glog
   ${GFLAGS_LIBRARIES}
-  ${LIBEVENT_LIB}
-  ${DOUBLE_CONVERSION_LIBRARIES}
+  libevent::core
+  libevent::extra
+  double-conversion::double-conversion
   ${CMAKE_DL_LIBS}
   ${LIBRT_LIBRARIES}
   Threads::Threads)
