diff --git a/opensubdiv/CMakeLists.txt b/opensubdiv/CMakeLists.txt
index ca24488..adb0191 100644
--- a/opensubdiv/CMakeLists.txt
+++ b/opensubdiv/CMakeLists.txt
@@ -241,7 +241,7 @@ if (NOT NO_LIB)
     endif()
 
     # Build frameworks  ----------------------------------
-    if(APPLE)
+    if(APPLE AND OSD_GPU)
         get_directory_property(OSD_HEADER_FILES DIRECTORY ${CMAKE_SOURCE_DIR}/opensubdiv/osd DEFINITION PUBLIC_HEADER_FILES)
         get_directory_property(FAR_HEADER_FILES DIRECTORY ${CMAKE_SOURCE_DIR}/opensubdiv/far DEFINITION PUBLIC_HEADER_FILES)
         get_directory_property(SDC_HEADER_FILES DIRECTORY ${CMAKE_SOURCE_DIR}/opensubdiv/sdc DEFINITION PUBLIC_HEADER_FILES)
