diff --git a/avahi-libevent.pc.in b/avahi-libevent.pc.in
index a1dca01b8..3356b0b39 100644
--- a/avahi-libevent.pc.in
+++ b/avahi-libevent.pc.in
@@ -6,6 +6,6 @@ includedir=${prefix}/include
 Name: avahi-libevent
 Description: Avahi Multicast DNS Responder (libevent Support)
 Version: @PACKAGE_VERSION@
-Requires: libevent-2.1.5
+Requires: libevent >= 2.1.5
 Libs: -L${libdir} -lavahi-libevent
 Cflags: -D_REENTRANT -I${includedir}

