diff --git a/cmake/coroutineOptions.cmake b/cmake/coroutineOptions.cmake
index 14a60c3..c13f3b3 100644
--- a/cmake/coroutineOptions.cmake
+++ b/cmake/coroutineOptions.cmake
@@ -14,6 +14,8 @@ function(target_coroutine_options TARGET)
     target_compile_options(${TARGET} PUBLIC -stdlib=libc++ -fcoroutines-ts)
     target_link_options(${TARGET} PUBLIC -stdlib=libc++)
     set_target_properties(${TARGET} PROPERTIES CXX_EXTENSIONS NO)
+  elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+    target_compile_options(${TARGET} PUBLIC -fcoroutines)
   else()
     message(FATAL_ERROR "Compiler not supported: ${CMAKE_CXX_COMPILER_ID}")
   endif()
