diff --git a/CMakeLists.txt b/CMakeLists.txt
index 351aa65..f25e440 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt	
@@ -73,7 +73,7 @@ set(concurrencpp_headers
         include/concurrencpp/timers/timer_queue.h
         include/concurrencpp/utils/bind.h)
 
-add_library(concurrencpp STATIC ${concurrencpp_headers} ${concurrencpp_sources})
+add_library(concurrencpp ${concurrencpp_headers} ${concurrencpp_sources})
 add_library(concurrencpp::concurrencpp ALIAS concurrencpp)
 
 target_include_directories(concurrencpp
