diff --git a/build-aux/ltmain.in b/build-aux/ltmain.in
index c750093..61533a0 100644
--- a/build-aux/ltmain.in
+++ b/build-aux/ltmain.in
@@ -3662,7 +3662,7 @@ EOF
 
 /* declarations of non-ANSI functions */
 #if defined __MINGW32__
-# ifdef __STRICT_ANSI__
+# if defined(__STRICT_ANSI__) && !defined(__MINGW64_VERSION_MAJOR) || defined(_POSIX_)
 int _putenv (const char *);
 # endif
 #elif defined __CYGWIN__
