diff --git a/CMakeLists.txt b/CMakeLists.txt
index fb0bc2f..e94a693 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,7 +45,7 @@ while (TRUE)
             message (FATAL_ERROR "The comms library found but the directory is not known")
         endif ()
 
-        set (CC_CMAKE_DIR ${LibComms_DIR}/lib/LibComms/cmake)
+        set (CC_CMAKE_DIR ${LibComms_DIR})
         break()
     endif ()
 
