From ddb89b07b09c5d9f1505d93b0d9b49f0cdd23592 Mon Sep 17 00:00:00 2001
From: quqiOnfree <quqiOnfree@outlook.com>
Date: Sun, 16 Jun 2024 01:09:52 +0800
Subject: [PATCH 7/9] add include2

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3994146..dacbb09 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -503,6 +503,7 @@ ENDIF()
 
 TARGET_LINK_LIBRARIES(${LIBRARY_NAME} ${MARIADB_CLIENT_TARGET_NAME} ${PLATFORM_DEPENDENCIES} unofficial::libmariadb)
 TARGET_LINK_LIBRARIES(${STATIC_LIBRARY_NAME} mariadbclient unofficial::libmariadb)
+TARGET_INCLUDE_DIRECTORIES(${LIBRARY_NAME} ${MARIADB_CLIENT_TARGET_NAME} PRIVATE ${CMAKE_SOURCE_DIR}/include)
 
 # install
 
-- 
2.45.2.windows.1

