diff --git a/backend/CMakeLists.txt b/backend/CMakeLists.txt
index d779033..133abef 100644
--- a/backend/CMakeLists.txt
+++ b/backend/CMakeLists.txt
@@ -80,7 +80,9 @@ if(NOT MSVC)
 endif()
 
 if(MSVC)
-    target_compile_definitions(zint PRIVATE DLL_EXPORT)
+    if(ZINT_SHARED)
+        target_compile_definitions(zint PRIVATE DLL_EXPORT)
+    endif()
 endif()
 
 zint_target_include_directories(PUBLIC
