diff --git a/include/libhat/compressed_pair.hpp b/include/libhat/compressed_pair.hpp
index 7bf839a..1b2be6c 100644
--- a/include/libhat/compressed_pair.hpp
+++ b/include/libhat/compressed_pair.hpp
@@ -4,6 +4,7 @@
     #include <tuple>
 #endif
 
+#include <cstddef>
 #include "defines.hpp"
 #include "type_traits.hpp"
 
diff --git a/include/libhat/strconv.hpp b/include/libhat/strconv.hpp
index f23d096..ad527fb 100644
--- a/include/libhat/strconv.hpp
+++ b/include/libhat/strconv.hpp
@@ -5,6 +5,7 @@
     #include <type_traits>
 #endif
 
+#include <cstdint>
 #include "concepts.hpp"
 #include "export.hpp"
 #include "result.hpp"
