diff --git a/include/libhat/compressed_pair.hpp b/include/libhat/compressed_pair.hpp
index 39cb8ea..341f9e3 100644
--- a/include/libhat/compressed_pair.hpp
+++ b/include/libhat/compressed_pair.hpp
@@ -5,6 +5,7 @@
     #include <utility>
 #endif
 
+#include <cstddef>
 #include "defines.hpp"
 #include "type_traits.hpp"
 
diff --git a/include/libhat/strconv.hpp b/include/libhat/strconv.hpp
index bbd786f..ad527fb 100644
--- a/include/libhat/strconv.hpp
+++ b/include/libhat/strconv.hpp
@@ -1,11 +1,11 @@
 #pragma once
 
 #ifndef LIBHAT_MODULE
-    #include <cstdint>
     #include <string_view>
     #include <type_traits>
 #endif
 
+#include <cstdint>
 #include "concepts.hpp"
 #include "export.hpp"
 #include "result.hpp"
