diff --git a/CMakeLists.txt b/CMakeLists.txt
index a6831d3..12915d1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -102,7 +102,9 @@ endif(MSVC)
 
 # Configure subdirectories.
 # We depend on these for later projects, so they should come first.
-add_subdirectory(third_party)
+find_package(SPIRV-Tools CONFIG REQUIRED)
+find_package(SPIRV-Tools-opt CONFIG REQUIRED)
+find_package(glslang CONFIG REQUIRED)
 
 if(SHADERC_ENABLE_SPVC)
 add_subdirectory(libshaderc_spvc)
diff --git a/glslc/CMakeLists.txt b/glslc/CMakeLists.txt
index d0df7db..1f14a5b 100644
--- a/glslc/CMakeLists.txt
+++ b/glslc/CMakeLists.txt
@@ -25,7 +25,7 @@ add_executable(glslc_exe src/main.cc)
 shaderc_default_compile_options(glslc_exe)
 target_include_directories(glslc_exe PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/.. ${spirv-tools_SOURCE_DIR}/include)
 set_target_properties(glslc_exe PROPERTIES OUTPUT_NAME glslc)
-target_link_libraries(glslc_exe PRIVATE glslc shaderc_util shaderc)
+target_link_libraries(glslc_exe PRIVATE glslc shaderc_util shaderc glslang)
 
 shaderc_add_tests(
   TEST_PREFIX glslc
