diff --git a/libshaderc/CMakeLists.txt b/libshaderc/CMakeLists.txt
index 0ffa06f..4b62aeb 100644
--- a/libshaderc/CMakeLists.txt
+++ b/libshaderc/CMakeLists.txt
@@ -34,10 +34,17 @@ if(SHADERC_ENABLE_INSTALL)
     DESTINATION
       ${CMAKE_INSTALL_INCLUDEDIR}/shaderc)
 
-  install(TARGETS shaderc shaderc_shared
+  if (NOT BUILD_SHARED_LIBS)
+  install(TARGETS shaderc
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+  else()
+  install(TARGETS shaderc_shared
+     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+  endif()
 endif(SHADERC_ENABLE_INSTALL)
 
 find_package(Threads)
@@ -71,6 +78,8 @@ shaderc_add_tests(
     shaderc_cpp
     shaderc_private)
 
+# Do not combine all dependencies to static library
+if (0)
 shaderc_combine_static_lib(shaderc_combined shaderc)
 
 if(SHADERC_ENABLE_INSTALL)
@@ -106,3 +115,4 @@ if(${SHADERC_ENABLE_TESTS})
   target_link_libraries(shaderc_c_smoke_test PRIVATE shaderc)
   add_test(NAME shaderc_c_smoke_test COMMAND shaderc_c_smoke_test)
 endif()
+endif()
