diff --git a/include/valijson/internal/optional.hpp b/include/valijson/internal/optional.hpp
index c9bc47b..c73c7fb 100644
--- a/include/valijson/internal/optional.hpp
+++ b/include/valijson/internal/optional.hpp
@@ -9,6 +9,6 @@ namespace opt = std;
 #    endif
 #  endif
 #else
-#  include <compat/optional.hpp>
+#  include <valijson/compat/optional.hpp>
 namespace opt = std::experimental;
 #endif
