diff --git a/CMakeLists.txt b/CMakeLists.txt
index 477064a..5da99e9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -95,8 +95,26 @@ message("## ${OATPP_THIS_MODULE_NAME} module. Resolving dependencies...\n")
 ##############################
 ## Find libcurl dependency
 
-include(FindPkgConfig)
-pkg_check_modules(PKG_CURL REQUIRED libcurl)
+if (VCPKG_TOOLCHAIN)
+    find_package(CURL REQUIRED)
+    if (CMAKE_BUILD_TYPE MATCHES "^[Dd][Ee][Bb][Uu][Gg]$")
+        get_filename_component(PKG_CURL_LIBRARIES ${CURL_LIBRARY_DEBUG} NAME)
+        get_filename_component(PKG_CURL_LIBRARY_DIR ${CURL_LIBRARY_DEBUG} DIRECTORY)
+    else()
+        get_filename_component(PKG_CURL_LIBRARIES ${CURL_LIBRARY_RELEASE} NAME)
+        get_filename_component(PKG_CURL_LIBRARY_DIR ${CURL_LIBRARY_RELEASE} DIRECTORY)
+    endif()
+    if (PKG_CURL_LIBRARIES MATCHES [[^.*\.a$]])
+        string(LENGTH ${PKG_CURL_LIBRARIES} _LEN)
+        math(EXPR _LEN "${_LEN} - 5")
+        string(SUBSTRING ${PKG_CURL_LIBRARIES} 3 ${_LEN} PKG_CURL_LIBRARIES)
+    endif()
+    set(PKG_CURL_VERSION ${CURL_VERSION})
+    set(PKG_CURL_INCLUDE_DIRS ${CURL_INCLUDE_DIRS})
+else()
+    include(FindPkgConfig)
+    pkg_check_modules(PKG_CURL REQUIRED libcurl)
+endif()
 
 message("[libcurl] libcurl found:")
 message("[libcurl] LIBRARIES=${PKG_CURL_LIBRARIES}")
