diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9dc2d0d..8488172 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -74,8 +74,9 @@ endif()
 ##################################################
 # Dependencies
 ##################################################
-include(dependencies)
+#include(dependencies)
 
+find_package(Boost REQUIRED COMPONENTS type_index hana assert charconv)
 
 ##################################################
 # Sanitizers
@@ -97,16 +98,42 @@ endif()
 # Parser
 ##################################################
 add_library(parser INTERFACE)
-target_include_directories(parser INTERFACE ${CMAKE_SOURCE_DIR}/include)
-if (Boost_FOUND)
-   target_include_directories(parser INTERFACE ${Boost_INCLUDE_DIRS})
-endif()
+target_include_directories(parser INTERFACE 
+  $<INSTALL_INTERFACE:include>
+  $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/include>)
 
+target_link_libraries(parser INTERFACE Boost::type_index Boost::hana Boost::assert Boost::charconv)
 
 ##################################################
 # Subdirectories
 ##################################################
+if(BUILD_TESTING)
 add_subdirectory(test)
 add_subdirectory(example)
+endif()
+
+include(CMakePackageConfigHelpers)
+
+configure_package_config_file(
+  cmake/unofficial-tzlaine-parser-config.cmake.in
+  "${CMAKE_CURRENT_BINARY_DIR}/cmake/unofficial-tzlaine-parser-config.cmake"
+  INSTALL_DESTINATION share/unofficial-tzlaine-parser/
+  NO_CHECK_REQUIRED_COMPONENTS_MACRO)
+
+# Install.
+install(TARGETS parser EXPORT unofficial-tzlaine-parser)
+
+install(
+  EXPORT unofficial-tzlaine-parser
+  NAMESPACE unofficial::tzlaine-parser::
+  FILE unofficial-tzlaine-parser-targets.cmake
+  DESTINATION share/unofficial-tzlaine-parser/)
+
+install(DIRECTORY include/boost DESTINATION include)
+
+install(
+  FILES
+  "${CMAKE_CURRENT_BINARY_DIR}/cmake/unofficial-tzlaine-parser-config.cmake"
+  DESTINATION share/unofficial-tzlaine-parser)
 
 endif()
diff --git a/cmake/unofficial-tzlaine-parser-config.cmake.in b/cmake/unofficial-tzlaine-parser-config.cmake.in
new file mode 100644
index 0000000..ab99c15
--- /dev/null
+++ b/cmake/unofficial-tzlaine-parser-config.cmake.in
@@ -0,0 +1,9 @@
+
+@PACKAGE_INIT@
+
+include(CMakeFindDependencyMacro)
+
+find_dependency(Boost COMPONENTS type_index hana assert charconv)
+
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-tzlaine-parser-targets.cmake")
+
