diff --git a/libgo/common/anys.h b/libgo/common/anys.h
index e5b2219..80c22e3 100644
--- a/libgo/common/anys.h
+++ b/libgo/common/anys.h
@@ -7,6 +7,7 @@
 #include <mutex>
 #include <assert.h>
 #include <memory>
+#include <stdexcept>
 
 namespace co
 {
diff --git a/libgo/common/spinlock.h b/libgo/common/spinlock.h
index b0adeb8..30789ad 100644
--- a/libgo/common/spinlock.h
+++ b/libgo/common/spinlock.h
@@ -1,6 +1,7 @@
 #pragma once
 #include "config.h"
 #include <exception>
+#include <stdexcept>
 
 namespace co
 {
