diff --git a/libgo/ctx_boost_context/context.h b/libgo/ctx_boost_context/context.h
index 2db71c0..2f321fb 100644
--- a/libgo/ctx_boost_context/context.h
+++ b/libgo/ctx_boost_context/context.h
@@ -1,4 +1,4 @@
-#include <boost/context/all.hpp>
+#include <boost/context/execution_context.hpp>
 
 #if BOOST_EXECUTION_CONTEXT == 2  // boost version >= 1.61
 # include "context_v2.h"
diff --git a/libgo/ctx_boost_context/context_v2.h b/libgo/ctx_boost_context/context_v2.h
index e415af4..59e2e29 100644
--- a/libgo/ctx_boost_context/context_v2.h
+++ b/libgo/ctx_boost_context/context_v2.h
@@ -1,4 +1,4 @@
-#include <boost/context/all.hpp>
+#include <boost/context/stack_context.hpp>
 
 namespace co
 {
