diff --git a/fitsio.h b/fitsio.h
index 719b187..2dacec8 100644
--- a/fitsio.h
+++ b/fitsio.h
@@ -39,6 +39,9 @@ SERVICES PROVIDED HEREUNDER."
 #define CFITSIO_MAJOR 3
 #define CFITSIO_SONAME 9
 
+/* Required for getcwd() */
+#define _POSIX_C_SOURCE 200112L
+
 /* the SONAME is incremented in a new release if the binary shared */
 /* library (on linux and Mac systems) is not backward compatible */
 /* with the previous release of CFITSIO */
diff --git a/group.c b/group.c
index af2f077..1cfd123 100644
--- a/group.c
+++ b/group.c
@@ -24,7 +24,7 @@
 #include <direct.h>   /* defines the getcwd function on Windows PCs */
 #endif
 
-#if defined(unix) || defined(__unix__)  || defined(__unix) || defined(HAVE_UNISTD_H)
+#if defined(unix) || defined(__unix__)  || defined(__unix) || defined(__APPLE__) || defined(HAVE_UNISTD_H)
 #include <unistd.h>  /* needed for getcwd prototype on unix machines */
 #endif
 
