diff --git a/cfitsio.pc.cmake b/cfitsio.pc.cmake
index 88d8ee3..bd6f3d9 100644
--- a/cfitsio.pc.cmake
+++ b/cfitsio.pc.cmake
@@ -6,7 +6,8 @@ includedir=@INCLUDE_INSTALL_DIR@
 Name: cfitsio
 Description: FITS File Subroutine Library
 URL: https://heasarc.gsfc.nasa.gov/fitsio/
-Version: @CFITSIO_MAJOR@.@CFITSIO_MINOR@.@CFITSIO_MICRO@
+Version: @CFITSIO_VERSION@
+Requires.private: @PKG_CONFIG_REQUIRES_PRIVATE@
 Libs: -L${libdir} -lcfitsio @PKG_CONFIG_LIBS@
 Libs.private: -lm
 Cflags: -I${includedir}
