diff -Naur postgresql-16.9/src/pl/tcl/pltcl.c postgresql-16.9-vcpkg/src/pl/tcl/pltcl.c
--- postgresql-16.9/src/pl/tcl/pltcl.c	2025-05-05 22:30:08.000000000 +0200
+++ postgresql-16.9-vcpkg/src/pl/tcl/pltcl.c	2025-05-23 22:53:43.485828671 +0200
@@ -56,7 +56,7 @@
 #define CONST86
 #endif
 
-#if !HAVE_TCL_VERSION(8,7)
+#if !HAVE_TCL_VERSION(8,7) || (TCL_MAJOR_VERSION == 9 && TCL_MINOR_VERSION == 0 && TCL_RELEASE_LEVEL == TCL_ALPHA_RELEASE)
 typedef int Tcl_Size;
 #endif
 
