diff --git a/cmake/IgnConfigureBuild.cmake b/cmake/IgnConfigureBuild.cmake
--- a/cmake/IgnConfigureBuild.cmake
+++ b/cmake/IgnConfigureBuild.cmake
@@ -109,7 +109,9 @@
     # Add all the source code directories
     add_subdirectory(src)
     add_subdirectory(include)
-    add_subdirectory(test)
+    if(BUILD_TESTING)
+      add_subdirectory(test)
+    endif()
 
 
     #--------------------------------------
