diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 7ca00ebc..fb0a42f1 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -382,7 +382,7 @@ install (TARGETS ${PROJECT_NAME}
 # Generate and install CMake package
 include(CMakePackageConfigHelpers)
 set(PACKAGE_INSTALL_DESTINATION
-    ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}-${NN_PACKAGE_VERSION}
+    ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}
 )
 install(EXPORT ${PROJECT_NAME}-target
     DESTINATION ${PACKAGE_INSTALL_DESTINATION}
