diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9efd451fb..7ff587631 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -719,9 +719,9 @@ else()
 endif()
 
 # Boost
-find_public_dependency(Boost REQUIRED COMPONENTS system)
+find_public_dependency(Boost REQUIRED COMPONENTS system date_time)
 target_include_directories(torrent-rasterbar PUBLIC ${Boost_INCLUDE_DIRS})
-target_link_libraries(torrent-rasterbar PUBLIC ${Boost_SYSTEM_LIBRARY})
+target_link_libraries(torrent-rasterbar PUBLIC ${Boost_LIBRARIES})
 
 if (exceptions)
 	if (MSVC)
