diff --git a/src/mpi.props b/src/mpi.props
index 8339a89..ba33015 100644
--- a/src/mpi.props
+++ b/src/mpi.props
@@ -96,7 +96,7 @@
       <Optimization>Full</Optimization>
       <WholeProgramOptimization>true</WholeProgramOptimization>
       <PreprocessorDefinitions>NDEBUG=1;%(PreprocessorDefinitions)</PreprocessorDefinitions>
-      <ControlFlowGuard>Guard</ControlFlowGuard>
+      <!-- <ControlFlowGuard>Guard</ControlFlowGuard> -->
     </ClCompile>
   </ItemDefinitionGroup>
 
diff --git a/src/mpi/msmpi/msmpi.props b/src/mpi/msmpi/msmpi.props
index 878a36a..7e3c4a6 100644
--- a/src/mpi/msmpi/msmpi.props
+++ b/src/mpi/msmpi/msmpi.props
@@ -13,7 +13,7 @@
       <Optimization>MaxSpeed</Optimization>
       <WholeProgramOptimization>true</WholeProgramOptimization>
       <PreprocessorDefinitions>NDEBUG=1;%(PreprocessorDefinitions)</PreprocessorDefinitions>
-      <ControlFlowGuard>Guard</ControlFlowGuard>
+      <!-- <ControlFlowGuard>Guard</ControlFlowGuard> -->
     </ClCompile>
   </ItemDefinitionGroup>
 
