diff --git a/src/mpi/msmpi/dll/msmpi.vcxproj b/src/mpi/msmpi/dll/msmpi.vcxproj
index 255b9f5..020a0cc 100644
--- a/src/mpi/msmpi/dll/msmpi.vcxproj
+++ b/src/mpi/msmpi/dll/msmpi.vcxproj
@@ -31,6 +31,7 @@
       <GenerateDebugInformation>true</GenerateDebugInformation>
       <EntryPointSymbol Condition="'$(BuildArchitecture)'=='amd64'">_DllMainCRTStartup</EntryPointSymbol>
       <EntryPointSymbol Condition="'$(BuildArchitecture)'=='i386'">_DllMainCRTStartup@12</EntryPointSymbol>
+      <ImageHasSafeExceptionHandlers Condition="'$(BuildArchitecture)'=='i386'">false</ImageHasSafeExceptionHandlers>
       <AdditionalDependencies>
         %(AdditionalDependencies);
         $(PUBLIC_SDK_LIB)\kernel32.lib;
