diff -urN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,8 +42,10 @@ if(CMAKE_VERSION VERSION_EQUAL "3.3.0" OR
 endif()
 
 if(CMAKE_SYSTEM_NAME MATCHES "Linux")
-  find_package(PkgConfig REQUIRED)
-  pkg_check_modules(UUID REQUIRED uuid)
+  find_path(UUID_INCLUDE_DIR uuid/uuid.h)
+  find_library(UUID_LIBRARY NAMES uuid)
+  include_directories(${UUID_INCLUDE_DIR})
+  link_libraries(${UUID_LIBRARY})
 endif()
 if(APPLE)
   find_library(COREFOUNDATION_LIBRARY CoreFoundation)
diff --git a/runtime/src/support/Guid.cpp b/runtime/src/support/Guid.cpp
--- a/runtime/src/support/Guid.cpp
+++ b/runtime/src/support/Guid.cpp
@@ -24,6 +24,7 @@
 
 #include <algorithm>
 
+#include "antlr4-common.h"
 #include "Guid.h"
 
 #ifdef GUID_LIBUUID
@@ -35,6 +36,7 @@
 #endif
 
 #ifdef GUID_WINDOWS
+#define NOMINMAX
 #include <objbase.h>
 #endif
 
