diff -urN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -40,8 +40,10 @@
 endif()

 if(CMAKE_SYSTEM_NAME MATCHES "Linux")
-  find_package(PkgConfig REQUIRED)
-  pkg_check_modules(UUID REQUIRED uuid)
+  find_path(UUID_INCLUDE_DIR uuid/uuid.h)
+  find_library(UUID_LIBRARY NAMES uuid)
+  include_directories(${UUID_INCLUDE_DIR})
+  link_libraries(${UUID_LIBRARY})
 endif()
 if(APPLE)
   find_library(COREFOUNDATION_LIBRARY CoreFoundation)
diff -urN a/runtime/src/support/guid.cpp b/runtime/src/support/guid.cpp
--- a/runtime/src/support/guid.cpp
+++ b/runtime/src/support/guid.cpp
@@ -22,6 +22,7 @@
  THE SOFTWARE.
  */

+#include "antlr4-common.h"
 #include "guid.h"

 #ifdef GUID_LIBUUID
 