diff --git a/sparrowConfig.cmake.in b/sparrowConfig.cmake.in
index 8b46226..a13454b 100644
--- a/sparrowConfig.cmake.in
+++ b/sparrowConfig.cmake.in
@@ -22,6 +22,10 @@
 
 include(CMakeFindDependencyMacro)
 
+if("@USE_DATE_POLYFILL@")
+    find_dependency(date)
+endif()
+
 if(NOT TARGET sparrow::sparrow)
     include("${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@Targets.cmake")
     get_target_property(@PROJECT_NAME@_INCLUDE_DIRS sparrow::sparrow INTERFACE_INCLUDE_DIRECTORIES)
