diff --git a/cmake/modules/FindAOM.cmake b/cmake/modules/FindAOM.cmake
index 61c5961..1ecf364 100644
--- a/cmake/modules/FindAOM.cmake
+++ b/cmake/modules/FindAOM.cmake
@@ -15,6 +15,7 @@ unset(CMAKE_REQUIRED_INCLUDES)
 
 find_library(AOM_LIBRARY
     NAMES libaom aom
+    NAMES_PER_DIR
     HINTS ${AOM_PKGCONF_LIBRARY_DIRS} ${AOM_PKGCONF_LIBDIR}
 )
 
diff --git a/cmake/modules/FindLIBDE265.cmake b/cmake/modules/FindLIBDE265.cmake
index c9a7fcb..95fc5bf 100644
--- a/cmake/modules/FindLIBDE265.cmake
+++ b/cmake/modules/FindLIBDE265.cmake
@@ -9,6 +9,7 @@ find_path(LIBDE265_INCLUDE_DIR
 
 find_library(LIBDE265_LIBRARY
     NAMES libde265 de265
+    NAMES_PER_DIR
     HINTS ${LIBDE265_PKGCONF_LIBRARY_DIRS} ${LIBDE265_PKGCONF_LIBDIR}
 )
 
diff --git a/cmake/modules/FindX265.cmake b/cmake/modules/FindX265.cmake
index ae8d8be..95898fe 100644
--- a/cmake/modules/FindX265.cmake
+++ b/cmake/modules/FindX265.cmake
@@ -8,7 +8,8 @@ find_path(X265_INCLUDE_DIR
 )
 
 find_library(X265_LIBRARY
-    NAMES libx265 x265
+    NAMES libx265 x265 x265-static
+    NAMES_PER_DIR
     HINTS ${X265_PKGCONF_LIBRARY_DIRS} ${X265_PKGCONF_LIBDIR}
 )
 
